diff --git dotnet/src/runtime/eng/native/configurecompiler.cmake dotnet/src/runtime/eng/native/configurecompiler.cmake
index d54d9b6803b..9438cb5af7e 100644
--- dotnet/src/runtime/eng/native/configurecompiler.cmake
+++ dotnet/src/runtime/eng/native/configurecompiler.cmake
@@ -574,6 +574,7 @@ if (CLR_CMAKE_HOST_UNIX OR CLR_CMAKE_HOST_WASI)
   add_compile_options(-Wno-unused-function)
   add_compile_options(-Wno-tautological-compare)
   add_compile_options(-Wno-unknown-pragmas)
+  add_compile_options(-Wno-c++-compat)
 
   # Explicitly enabled warnings
   check_c_compiler_flag(-Wimplicit-fallthrough COMPILER_SUPPORTS_W_IMPLICIT_FALLTHROUGH)
diff --git dotnet/src/runtime/src/coreclr/pal/prebuilt/inc/cordebug.h dotnet/src/runtime/src/coreclr/pal/prebuilt/inc/cordebug.h
index 78b86c69ee9..1672d602d99 100644
--- dotnet/src/runtime/src/coreclr/pal/prebuilt/inc/cordebug.h
+++ dotnet/src/runtime/src/coreclr/pal/prebuilt/inc/cordebug.h
@@ -10042,6 +10042,7 @@ EXTERN_C const IID IID_ICorDebugRegisterSet2;
 typedef
 enum CorDebugUserState
     {
+        CORDEBUG_INVALID = -1,
         USER_STOP_REQUESTED = 0x1,
         USER_SUSPEND_REQUESTED  = 0x2,
         USER_BACKGROUND = 0x4,
