diff -up rpm-4.20.1/macros.in.sysusers rpm-4.20.1/macros.in
--- rpm-4.20.1/macros.in.sysusers	2025-02-26 14:52:52.412310565 +0100
+++ rpm-4.20.1/macros.in	2025-02-26 14:53:02.173248069 +0100
@@ -142,8 +142,8 @@
 %_sysusersdir		@sysusersdir@
 
 # sysusers helper binary (or a replacement script), comment out to disable
-#%__systemd_sysusers	@__SYSTEMD_SYSUSERS@
-%__systemd_sysusers	%{_rpmconfigdir}/sysusers.sh
+%__systemd_sysusers	@__SYSTEMD_SYSUSERS@
+#%__systemd_sysusers	%{_rpmconfigdir}/sysusers.sh
 
 #
 #	Path to script that creates debug symbols in a /usr/lib/debug
