--- coreutils-0.7.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ coreutils-0.7.0/Cargo.toml	2026-08-06T11:26:04.426642+00:00
@@ -719,7 +719,7 @@
 package = "uu_runcon"
 
 [dependencies.selinux]
-version = "=0.6.0"
+version = "0.6.0"
 optional = true
 
 [dependencies.seq]
@@ -930,7 +930,7 @@
 version = "0.6.8"
 
 [dev-dependencies.ctor]
-version = "0.6.0"
+version = "1"
 
 [dev-dependencies.filetime]
 version = "0.2.23"
