--- zip-7.2.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ zip-7.2.0/Cargo.toml	2026-02-12T19:17:39.916142+00:00
@@ -302,25 +302,3 @@
 version = "0.3"
 default-features = false
 
-[target.'cfg(target_arch = "wasm32")'.dependencies.getrandom]
-version = "0.3"
-features = ["wasm_js"]
-optional = true
-default-features = false
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.time]
-version = "0.3"
-features = [
-    "std",
-    "wasm-bindgen",
-]
-optional = true
-default-features = false
-
-[target.'cfg(target_arch = "wasm32")'.dev-dependencies.getrandom]
-version = "0.3"
-features = ["wasm_js"]
-default-features = false
-
-[target.'cfg(target_arch = "wasm32")'.dev-dependencies.wasm-bindgen-test]
-version = "^0.3.56"
