diff --git a/ext/Rakefile b/ext/Rakefile
index 5e946f2..f4e992f 100644
--- a/ext/Rakefile
+++ b/ext/Rakefile
@@ -30,7 +30,7 @@ task :default => :clean do
   end
 
   recipe.files << {
-    :url => "https://codeload.github.com/edenhill/librdkafka/tar.gz/v#{Rdkafka::LIBRDKAFKA_VERSION}",
+    :url => "file:///usr/src/mariner/SOURCES/librdkafka-1.4.0.tar.gz",
     :sha256 => Rdkafka::LIBRDKAFKA_SOURCE_SHA256
   }
   recipe.configure_options = ["--host=#{recipe.host}"]
@@ -64,7 +64,7 @@ namespace :build do
     version = "git-#{ref}"
 
     recipe = MiniPortile.new("librdkafka", version)
-    recipe.files << "https://github.com/edenhill/librdkafka/archive/#{ref}.tar.gz"
+    recipe.files << "file:///usr/src/mariner/SOURCES/librdkafka-1.4.0.tar.gz"
     recipe.configure_options = ["--host=#{recipe.host}"]
     recipe.cook
 
