Fedora specific Makefile patch.

--- sgpio/Makefile	2008-11-24 12:27:12.000000000 +0100
+++ sgpio-new/Makefile	2018-02-26 11:32:14.394597823 +0100
@@ -23,25 +23,29 @@
 
 
 MANDIR ?= /usr/local/man
+SBIN_DIR = /sbin
+INSTALL =/usr/bin/install -c
 ALL = sgpio
+CFLAGS = -g -Wall
+LDFLAGS =
 
 all: $(ALL)
 
 sgpio.o: sgpio.c
-	gcc -g -Wall -c sgpio.c
+	gcc $(CFLAGS) -c sgpio.c
 
 sgpio: sgpio.o
-	gcc -g sgpio.o -o sgpio
+	gcc $(LDFLAGS) sgpio.o -o sgpio
 
 clean:
 	rm -f sgpio.o sgpio
 
 install: $(ALL)
 	for i in $(ALL); do \
-		install $$i /sbin/$$i; \
+	    $(INSTALL) -D $$i $(SBIN_DIR)/$$i; \
 	done
-	install -d $(MANDIR)/man1
-	install -m 0644 *.1 $(MANDIR)/man1
+	$(INSTALL) -d $(MANDIR)/man1
+	$(INSTALL) -m 0644 *.1 $(MANDIR)/man1
 
 uninstall:
 	for i in $(ALL); do \
