diff -dupr a/tests/src/vfs/extfs/helpers-list/Makefile.am b/tests/src/vfs/extfs/helpers-list/Makefile.am
--- a/tests/src/vfs/extfs/helpers-list/Makefile.am	2017-03-04 17:26:20.000000000 +0000
+++ b/tests/src/vfs/extfs/helpers-list/Makefile.am	2017-08-18 23:52:30.336725600 +0000
@@ -12,10 +12,10 @@ LIBS = $(top_builddir)/lib/libmc.la
 
 # Programs/scripts to build on 'make check'.
 check_PROGRAMS = mc_parse_ls_l
-check_SCRIPTS = run
+#check_SCRIPTS = run
 
 # Tests to run on 'make check'
-TESTS = run
+#TESTS = run
 
 # On 'make clean', delete 'run' as well.
 CLEANFILES = run
diff -dupr a/tests/src/vfs/extfs/helpers-list/Makefile.in b/tests/src/vfs/extfs/helpers-list/Makefile.in
--- a/tests/src/vfs/extfs/helpers-list/Makefile.in	2017-03-04 17:54:45.000000000 +0000
+++ b/tests/src/vfs/extfs/helpers-list/Makefile.in	2017-08-19 00:18:34.756669652 +0000
@@ -706,10 +706,10 @@ AM_CPPFLAGS = $(GLIB_CFLAGS) -I$(top_src
 # This lets mc_parse_ls_l.c override MC's message() without the linker
 # complaining about multiple definitions.
 AM_LDFLAGS = @TESTS_LDFLAGS@
-check_SCRIPTS = run
+#check_SCRIPTS = run
 
 # Tests to run on 'make check'
-TESTS = run
+#TESTS = run
 
 # On 'make clean', delete 'run' as well.
 CLEANFILES = run README.html
