diff --git a/GNUmakefile b/GNUmakefile
index e727b00..a3cac1b 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -262,7 +262,7 @@ init_package:
 	  echo "  make package pkgdir=/tmp/newnetpbm "; \
 	  false; \
 	  fi
-	mkdir $(PKGDIR)
+	mkdir -p $(PKGDIR)
 	echo "Netpbm install package made by 'make package'" \
 	    >$(PKGDIR)/pkginfo
 	date >>$(PKGDIR)/pkginfo
diff --git a/other/Makefile b/other/Makefile
index aa0d75b..95fb8be 100644
--- a/other/Makefile
+++ b/other/Makefile
@@ -7,8 +7,6 @@ VPATH=.:$(SRCDIR)/$(SUBDIR)
 
 include $(BUILDDIR)/config.mk
 
-SUBDIRS = pamx
-
 EXTERN_INCLUDES =
 ifneq ($(LINUXSVGALIB),NONE)
   ifneq ($(LINUXSVGAHDR_DIR),)
@@ -48,7 +46,7 @@ MERGE_OBJECTS = $(MERGEBINARIES:%=%.o2)
 HAVE_MERGE_COMPAT=YES
 
 .PHONY: all
-all: $(BINARIES) $(SUBDIRS:%=%/all)
+all: $(BINARIES)
 
 include $(SRCDIR)/common.mk
 
diff --git a/other/pamx/Makefile b/other/pamx/Makefile
index 7fe7b9d..7b41221 100644
--- a/other/pamx/Makefile
+++ b/other/pamx/Makefile
@@ -27,16 +27,6 @@ else
   EXTERN_INCLUDES += $(shell $(PKG_CONFIG) x11 --cflags)
 endif
 
-ifeq ($(HAVE_X11LIB),Y)
-  PORTBINARIES += pamx
-
-  EXTRA_OBJECTS = \
-	  image.o \
-	  send.o \
-	  window.o \
-
-endif
-
 BINARIES = $(PORTBINARIES)
 
 OBJECTS = $(BINARIES:%=%.o) $(EXTRA_OBJECTS)
