diff -urNp old/editor/ppmfade new/editor/ppmfade
--- old/editor/ppmfade	2017-11-01 11:47:49.869611402 +0100
+++ new/editor/ppmfade	2017-11-01 11:53:25.524973342 +0100
@@ -84,7 +84,7 @@ for ($n = 0; $n < @ARGV; $n++) {
         if (-e $first_file) {
         } else {
             print "I can't find first file '$first_file'\n";
-            exit 20;
+            exit 1;
         }
     } elsif ($ARGV[$n] eq "-l") {
         $n++;
@@ -92,7 +92,7 @@ for ($n = 0; $n < @ARGV; $n++) {
         if (-e $last_file) {
         } else {
             print "I can't find last file '$last_file'\n";
-            exit 20;
+            exit 1;
         }
     } elsif ($ARGV[$n] eq "-base") {
         $n++;
@@ -113,9 +113,12 @@ for ($n = 0; $n < @ARGV; $n++) {
         $mode = $BLOCK;
     } elsif ("$ARGV[$n]" eq "-mix") {
         $mode = $MIX;
+    } elsif ($ARGV[$n] eq "-help" || $ARGV[$n] eq "--help" || $ARGV[$n] eq "-h") {
+        print "ppmfade: Use 'man ppmfade' for help.\n";
+        exit 1;
     } else {
         print "Unknown argument: $ARGV[$n]\n";
-        exit 100;
+        exit 1;
     } 
 }
 #
@@ -134,18 +137,18 @@ if ($first_file ne "undefined") {
         $width = $1; $height = $2;
     } else {
         print("Unrecognized results from pnmfile on $first_file.\n");
-        exit(50);
+        exit 1;
     }
 } elsif ($last_file ne "undefined") {
     if ((`pnmfile $last_file` =~ m{\b(\d+)\sby\s(\d+)} )) { 
         $width = $1; $height = $2;
     } else {
         print("Unrecognized results from pnmfile on $first_file.\n");
-        exit(50);
+        exit 1;
     }
 } else {
     print("ppmfade:  You must specify -f or -l (or both)\n");
-    exit(90);
+    exit 1;
 }
 
 print("Frames are " . $width . "W x " . $height . "H\n");
