diff --git a/converter/other/pstopnm.c b/converter/other/pstopnm.c
index f534265..2ab48ab 100644
--- a/converter/other/pstopnm.c
+++ b/converter/other/pstopnm.c
@@ -896,11 +896,11 @@ execGhostscript(int               const inputPipeFd,
                    ghostscriptProg, arg0,
                    deviceopt, outfileopt, gopt, ropt, textalphabitsopt,
                    "-q", "-dNOPAUSE",
-                   "-dSAFER", "-");
+                   "-dPARANOIDSAFER", "-");
     }
 
-    execl(ghostscriptProg, arg0, deviceopt, outfileopt, gopt, ropt, 
-	  textalphabitsopt, "-q", "-dNOPAUSE", "-dSAFER", "-", NULL);
+    execl(ghostscriptProg, arg0, deviceopt, outfileopt, gopt, ropt, "-q",
+          "-dNOPAUSE", "-dPARANOIDSAFER", "-", NULL);
 
     pm_error("execl() of Ghostscript ('%s') failed, errno=%d (%s)",
              ghostscriptProg, errno, strerror(errno));
