diff --git a/converter/other/bmptopnm.c b/converter/other/bmptopnm.c
index 3f66634..ca4f944 100644
--- a/converter/other/bmptopnm.c
+++ b/converter/other/bmptopnm.c
@@ -1494,7 +1494,7 @@ readBmp(FILE *               const ifP,
         unsigned int *       const cmapSizeP,
         bool                 const verbose) {
 
-    xel * colormap;  /* malloc'ed */
+    xel * colormap = NULL;  /* malloc'ed */
     unsigned int pos;
         /* Current byte position in the BMP file */
 
