diff -up libmodplug-0.8.9.0/README~ libmodplug-0.8.9.0/README
--- libmodplug-0.8.9.0/README~	2017-04-24 16:16:28.000000000 +0300
+++ libmodplug-0.8.9.0/README	2017-04-25 10:13:08.701929258 +0300
@@ -71,7 +71,7 @@ Contents
 
      MMPAT_PATH_TO_CFG	set to the directory where the file "timidity.cfg" and
       the subdirectory "instruments" can be found,
-       default: "/usr/local/share/timidity".
+       default: "/usr/share/timidity".
       MMMID_SPEED for experimenting with the mod speed (1 thru 9)
       MMMID_VERBOSE for feedback on the conversion process
       MMMID_DEBUG for sake of completeness, only useful for maintainers
@@ -82,7 +82,7 @@ Contents
 
       MMPAT_PATH_TO_CFG	set to the directory where the file "timidity.cfg" and
        the subdirectory "instruments" can be found,
-       default: "/usr/local/share/timidity".
+       default: "/usr/share/timidity".
 
       MMABC_NO_RANDOM_PICK when not set and the abc file contains multiple songs
        (X:n) the first song to be played will be picked at random another click
diff -up libmodplug-0.8.9.0/src/load_pat.cpp~ libmodplug-0.8.9.0/src/load_pat.cpp
--- libmodplug-0.8.9.0/src/load_pat.cpp~	2017-04-24 16:10:17.000000000 +0300
+++ libmodplug-0.8.9.0/src/load_pat.cpp	2017-04-25 10:13:08.701929258 +0300
@@ -52,8 +52,8 @@
 #define PATHFORPAT	"C:\\TIMIDITY\\INSTRUMENTS"
 #else
 #define DIRDELIM		'/'
-#define TIMIDITYCFG	"/usr/local/share/timidity/timidity.cfg"
-#define PATHFORPAT	"/usr/local/share/timidity/instruments"
+#define TIMIDITYCFG	"/etc/timidity.cfg"
+#define PATHFORPAT	"/usr/share/timidity/instruments"
 #endif
 
 #define PAT_ENV_PATH2CFG			"MMPAT_PATH_TO_CFG"
