--- ceph-20.1.0/src/test/neorados/common_tests.h.orig	2025-09-04 15:35:40.000000000 -0400
+++ ceph-20.1.0/src/test/neorados/common_tests.h	2025-09-16 08:13:03.274896767 -0400
@@ -47,6 +47,18 @@
 
 #include "gtest/gtest.h"
 
+#if !defined(GTEST_DISALLOW_COPY_AND_ASSIGN_)
+#define GTEST_DISALLOW_COPY_AND_ASSIGN_(type) \
+  type(type const&) = delete;                 \
+  type& operator=(type const&) = delete
+#endif
+
+#if !defined(GTEST_DISALLOW_MOVE_AND_ASSIGN_)
+#define GTEST_DISALLOW_MOVE_AND_ASSIGN_(type) \
+  type(type&&) noexcept = delete;             \
+  type& operator=(type&&) noexcept = delete
+#endif
+
 /// \file test/neorados/common_tests.h
 ///
 /// \brief Tools for testing neorados code
