From: Riken Maharjan <rmaharjan@microsoft.com>
Date: Thu, 17 Apr 2025 21:27:30 +0000
diff -urN CherryPy-18.9.0/cherrypy/test/test_session.py CherryPy-18.9.0/cherrypy/test/test_session.py
--- CherryPy-18.9.0/cherrypy/test/test_session.py	2022-09-11 17:35:38.000000000 +0000
+++ CherryPy-18.9.0/cherrypy/test/test_session.py	2025-04-17 21:27:30.687754101 +0000
@@ -146,9 +146,14 @@
     def teardown_class(cls):
         """Clean up sessions."""
         super(cls, cls).teardown_class()
+        try:
+            files_to_clean = localDir.iterdir()  # Python 3.8+
+        except AttributeError:
+            files_to_clean = localDir.listdir()  # Python 3.6-3.7
+
         consume(
             file.remove_p()
-            for file in localDir.listdir()
+            for file in files_to_clean
             if file.basename().startswith(
                 sessions.FileSession.SESSION_PREFIX
             )
