diff -urpN systemd-stable-255/src/test/test-openssl.c b/src/test/test-openssl.c
--- systemd-stable-255/src/test/test-openssl.c	2023-12-06 15:32:40.000000000 -0500
+++ b/src/test/test-openssl.c	2024-02-28 15:23:49.651036469 -0500
@@ -127,8 +127,10 @@ static const struct {
         { "sha3-256", 32, },
         { "sha3-384", 48, },
         { "sha3-512", 64, },
+#if 0 /* Azure Linux OpenSSL does not support SM3 */
         /* SM3 family */
         { "sm3",      32, },
+#endif
         /* MD5 family */
         { "md5",      16, },
 };
