diff -urpN a/src/libsystemd/sd-journal/journal-file.c b/src/libsystemd/sd-journal/journal-file.c
--- a/src/libsystemd/sd-journal/journal-file.c	2022-01-18 11:35:43.000000000 +0000
+++ b/src/libsystemd/sd-journal/journal-file.c	2023-10-19 19:15:42.093295820 +0000
@@ -3254,7 +3254,7 @@ int journal_file_open(
                 .flags = flags,
                 .writable = (flags & O_ACCMODE) != O_RDONLY,
 
-#if HAVE_ZSTD
+#if HAVE_ZSTD && 0 /* For Mariner 2, disable journal compression with zstd for backwards compatibility */
                 .compress_zstd = compress,
 #elif HAVE_LZ4
                 .compress_lz4 = compress,
