diff -uNr systemd-233/src/resolve/resolved-conf.c systemd-233-new/src/resolve/resolved-conf.c
--- systemd-233/src/resolve/resolved-conf.c	2017-03-01 21:43:06.000000000 +0000
+++ systemd-233-new/src/resolve/resolved-conf.c	2017-07-07 03:29:00.130502439 +0000
@@ -229,6 +229,7 @@
 
 int manager_parse_config_file(Manager *m) {
         int r;
+        char *default_dns_servers;
 
         assert(m);
 
@@ -241,7 +242,10 @@
                 return r;
 
         if (m->need_builtin_fallbacks) {
-                r = manager_parse_dns_server_string_and_warn(m, DNS_SERVER_FALLBACK, DNS_SERVERS);
+                default_dns_servers = secure_getenv("DEFAULT_DNS_SERVERS");
+                if (default_dns_servers == NULL)
+                        default_dns_servers = DNS_SERVERS;
+                r = manager_parse_dns_server_string_and_warn(m, DNS_SERVER_FALLBACK, default_dns_servers);
                 if (r < 0)
                         return r;
         }
