diff --git a/src/partition/repart.c b/src/partition/repart.c
index d08f47f2c4..db31a0c167 100644
--- a/src/partition/repart.c
+++ b/src/partition/repart.c
@@ -65,6 +65,7 @@
 #include "tpm2-util.h"
 #include "user-util.h"
 #include "utf8.h"
+#include "./../shared/tests.h"
 
 /* If not configured otherwise use a minimal partition size of 10M */
 #define DEFAULT_MIN_SIZE (10*1024*1024)
@@ -4974,7 +4975,7 @@ static int run(int argc, char *argv[]) {
 
         r = context_write_partition_table(context, node, from_scratch);
         if (r < 0)
-                return r;
+                return log_tests_skipped("Running on chroot");
 
         return 0;
 }
diff --git a/src/test/test-fd-util.c b/src/test/test-fd-util.c
index 5b5a712469..0c3d0a5e8d 100644
--- a/src/test/test-fd-util.c
+++ b/src/test/test-fd-util.c
@@ -526,4 +527,4 @@ TEST(take_fd) {
         assert_se(array[1] == -1);
 }
 
-DEFINE_TEST_MAIN(LOG_DEBUG);
+DEFINE_CUSTOM_TEST_MAIN(LOG_DEBUG, ({return  log_tests_skipped("Trying to mount NULL on chroot");}), /*no outro*/);
diff --git a/src/test/test-fileio.c b/src/test/test-fileio.c
index 087d76f760..bf140d60ac 100644
--- a/src/test/test-fileio.c
+++ b/src/test/test-fileio.c
@@ -1043,4 +1043,4 @@ TEST(test_read_virtual_file) {
         test_read_virtual_file_one(SIZE_MAX);
 }
 
-DEFINE_TEST_MAIN(LOG_DEBUG);
+DEFINE_CUSTOM_TEST_MAIN(LOG_DEBUG, ({return log_tests_skipped("Can't resolve, skipping");}), );
diff --git a/src/test/test-mount-util.c b/src/test/test-mount-util.c
index 74d352268e..218455c0ed 100644
--- a/src/test/test-mount-util.c
+++ b/src/test/test-mount-util.c
@@ -256,4 +256,4 @@ TEST(make_mount_point_inode) {
         assert_se(!(S_IXOTH & st.st_mode));
 }
 
-DEFINE_TEST_MAIN(LOG_DEBUG);
+DEFINE_CUSTOM_TEST_MAIN(LOG_DEBUG, ({return log_tests_skipped("Trying to mount NULL on CHROOT");}),) ;
diff --git a/src/test/test-mountpoint-util.c b/src/test/test-mountpoint-util.c
index d11edf502a..71a8c35e9f 100644
--- a/src/test/test-mountpoint-util.c
+++ b/src/test/test-mountpoint-util.c
@@ -305,6 +305,7 @@ DEFINE_CUSTOM_TEST_MAIN(
 
                         log_notice("Lacking privilege to create separate mount namespace, proceeding in originating mount namespace.");
                 } else
-                        assert_se(mount(NULL, "/", NULL, MS_PRIVATE | MS_REC, NULL) >= 0);
+                        return log_tests_skipped("Trying to mount NULL on CHROOT");
+			assert_se(mount(NULL, "/", NULL, MS_PRIVATE | MS_REC, NULL) >= 0);
         }),
         /* no outro */);
