From 351078a7e41a71f75ca1d142402016a97b80ed47 Mon Sep 17 00:00:00 2001
From: Nan Liu <liunan@microsoft.com>
Date: Sat, 17 Feb 2024 00:06:30 +0000
Subject: [PATCH] examine-all-encodings

---
 mkfontscale.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mkfontscale.c b/mkfontscale.c
index 356417f..eefc950 100644
--- a/mkfontscale.c
+++ b/mkfontscale.c
@@ -326,7 +326,7 @@ getName(FT_Face face, int nid)
     unsigned int i;
 
     if(getNameHelper(face, nid,
-                     TT_PLATFORM_MICROSOFT, TT_MS_ID_UNICODE_CS, &name) ||
+                     TT_PLATFORM_MICROSOFT, -1, &name) ||
        getNameHelper(face, nid,
                      TT_PLATFORM_APPLE_UNICODE, -1, &name)) {
         string = malloc(name.string_len / 2 + 1);
-- 
2.25.1

