diff -up firefox-70.0/layout/base/PresShell.h.1354671 firefox-70.0/layout/base/PresShell.h
--- firefox-70.0/layout/base/PresShell.h.1354671	2019-10-22 12:33:12.987775587 +0200
+++ firefox-70.0/layout/base/PresShell.h	2019-10-22 12:36:39.999366086 +0200
@@ -257,7 +257,7 @@ class PresShell final : public nsStubDoc
    * to the same aSize value.  AllocateFrame is infallible and will abort
    * on out-of-memory.
    */
-  void* AllocateFrame(nsQueryFrame::FrameIID aID, size_t aSize) {
+  void* __attribute__((optimize("no-lifetime-dse"))) AllocateFrame(nsQueryFrame::FrameIID aID, size_t aSize) {
 #define FRAME_ID(classname, ...)                                  \
   static_assert(size_t(nsQueryFrame::FrameIID::classname##_id) == \
                     size_t(eArenaObjectID_##classname),           \
