diff --git a/ant-contrib-1.0b3.pom b/ant-contrib-1.0b3.pom
index 59dea4a..ca99233 100644
--- a/ant-contrib-1.0b3.pom
+++ b/ant-contrib-1.0b3.pom
@@ -17,7 +17,7 @@
   <description>A collection of tasks (and at one point maybe types and other tools) for Apache Ant</description>
   <dependencies>
     <dependency>
-      <groupId>ant</groupId>
+      <groupId>org.apache.ant</groupId>
       <artifactId>ant</artifactId>
       <version>1.5</version>
     </dependency>
