diff --git a/Makefile b/Makefile
index e955242..05f1309 100644
--- a/Makefile
+++ b/Makefile
@@ -17,8 +17,8 @@ LOCAL_VERSION =
 CONFIG_LOCALVERSION =
 
 CPPFLAGS = -I libfdt -I .
-WARNINGS = -Wall -Wpointer-arith -Wcast-qual -Wnested-externs \
-	-Wstrict-prototypes -Wmissing-prototypes -Wredundant-decls -Wshadow
+WARNINGS = -Wall -Wpointer-arith -Wnested-externs \
+	-Wstrict-prototypes -Wredundant-decls -Wshadow
 CFLAGS = -g -Os $(SHAREDLIB_CFLAGS) -Werror $(WARNINGS) $(EXTRA_CFLAGS)
 
 BISON = bison
