diff -rupN --no-dereference gcc-15.1.1-20250521/libgcc/config/i386/enable-execute-stack-mingw32.c gcc-15.1.1-20250521-new/libgcc/config/i386/enable-execute-stack-mingw32.c
--- gcc-15.1.1-20250521/libgcc/config/i386/enable-execute-stack-mingw32.c	2025-05-21 14:40:58.000000000 +0200
+++ gcc-15.1.1-20250521-new/libgcc/config/i386/enable-execute-stack-mingw32.c	2025-05-25 00:04:57.176477347 +0200
@@ -25,6 +25,7 @@
 #define WIN32_LEAN_AND_MEAN
 #include <stdlib.h>
 #include <windows.h>
+#include <stdlib.h>
 
 extern void __enable_execute_stack (void *);
 
