--- a/libnemo-private/nemo-icon-container.c
+++ b/libnemo-private/nemo-icon-container.c
@@ -4256,7 +4256,7 @@
 	/* We pass the event to the search_entry.  If its text changes, then we
 	 * start the typeahead find capabilities.
 	 * Copied from NemoIconContainer */
-	if (!handled &&
+	if (!handled && !nemo_icon_container_get_is_desktop (container) &&
 		event->keyval != GDK_KEY_asciitilde &&
 		event->keyval != GDK_KEY_KP_Divide &&
 	    event->keyval != GDK_KEY_slash /* don't steal slash key events, used for "go to" */ &&

