--- t/bugs/001_assign.t
+++ t/bugs/001_assign.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 # Verify the Readonly function rejects initialization by assignment
 use strict;
 use warnings; no warnings 'misc';
--- t/bugs/007_implicit_undef.t
+++ t/bugs/007_implicit_undef.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 # Verify the Readonly function accepts implicit undef values
 use strict;
 use Test::More;
--- t/general/array.t
+++ t/general/array.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 
 # Readonly array tests
 
--- t/general/deepa.t
+++ t/general/deepa.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 
 # Test Array vs Array1 functionality
 
--- t/general/deeph.t
+++ t/general/deeph.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 
 # Test Hash vs Hash1 functionality
 
--- t/general/deeps.t
+++ t/general/deeps.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 
 # Test Scalar vs Scalar1 functionality
 
--- t/general/docs.t
+++ t/general/docs.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 # Examples from the docs -- make sure they work!
 use strict;
 use Test::More tests => 22;
--- t/general/export.t
+++ t/general/export.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 # Readonly hash tests
 use strict;
 use Test::More tests => 1;
--- t/general/hash.t
+++ t/general/hash.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 
 # Readonly hash tests
 
--- t/general/readonly.t
+++ t/general/readonly.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 
 # Test the Readonly function
 
--- t/general/reassign.t
+++ t/general/reassign.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 
 # Readonly reassignment-prevention tests
 
--- t/general/scalar.t
+++ t/general/scalar.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 
 # Readonly scalar tests
 
--- t/general/tie.t
+++ t/general/tie.t
@@ -1,4 +1,4 @@
-#!perl -I../../lib
+#!/usr/bin/perl -I../../lib
 # Test the Readonly function
 use strict;
 use Test::More tests => 4;
