--- a/rc.d/init.d/functions	2017-02-27 06:56:05.000000000 -0800
+++ b/rc.d/init.d/functions.mod	2017-12-26 15:06:51.070071057 -0800
@@ -60,7 +60,7 @@
 
 if [ -z "${CONSOLETYPE:-}" ]; then
     if [ -c "/dev/stderr" -a -r "/dev/stderr" ]; then
-        CONSOLETYPE="$(/sbin/consoletype < /dev/stderr 2>/dev/null)"
+        CONSOLETYPE="$(/sbin/consoletype stdout < /dev/stderr 2>/dev/null)"
     else
         CONSOLETYPE="serial"
     fi
