diff -ru noto-emoji-f8131fc45736000552cd04a8388dc414d666a829.orig/Makefile noto-emoji-f8131fc45736000552cd04a8388dc414d666a829/Makefile
--- noto-emoji-f8131fc45736000552cd04a8388dc414d666a829.orig/Makefile	2020-06-25 16:02:56.000000000 +0200
+++ noto-emoji-f8131fc45736000552cd04a8388dc414d666a829/Makefile	2020-07-21 11:12:06.948389580 +0200
@@ -22,7 +22,8 @@
 PYTHON = python3
 PNGQUANTFLAGS = --speed 1 --skip-if-larger --quality 85-95 --force
 BODY_DIMENSIONS = 136x128
-IMOPS := -size $(BODY_DIMENSIONS) canvas:none -compose copy -gravity center
+#IMOPS := -size $(BODY_DIMENSIONS) canvas:none -compose copy -gravity center
+IMOPS = -size 136x128 -background none -gravity center -extent 136x128 -compose copy
 
 ZOPFLIPNG = zopflipng
 TTX = ttx
@@ -163,13 +164,13 @@
 # imagemagick packaged with ubuntu trusty (6.7.7-10) by using -composite.
 
 $(EMOJI_DIR)/%.png: $(EMOJI_SRC_DIR)/%.png | $(EMOJI_DIR)
-	@convert $(IMOPS) "$<" -composite "PNG32:$@"
+	@gm convert $(IMOPS) "$<" "PNG32:$@"
 
 $(FLAGS_DIR)/%.png: $(FLAGS_SRC_DIR)/%.png ./waveflag | $(FLAGS_DIR)
 	@./waveflag $(FLAGS_DIR)/ "$<"
 
 $(RESIZED_FLAGS_DIR)/%.png: $(FLAGS_DIR)/%.png | $(RESIZED_FLAGS_DIR)
-	@convert $(IMOPS) "$<" -composite "PNG32:$@"
+	@gm convert $(IMOPS) "$<" "PNG32:$@"
 
 flag-symlinks: $(RESIZED_FLAG_FILES) | $(RENAMED_FLAGS_DIR)
 	@$(subst ^, ,                                  \

