diff -ru noto-emoji-d5e261484286d33a1fe8a02676f5907ecc02106f.orig/Makefile noto-emoji-d5e261484286d33a1fe8a02676f5907ecc02106f/Makefile
--- noto-emoji-d5e261484286d33a1fe8a02676f5907ecc02106f.orig/Makefile	2020-07-22 04:16:40.000000000 +0200
+++ noto-emoji-d5e261484286d33a1fe8a02676f5907ecc02106f/Makefile	2020-07-24 12:36:48.786209598 +0200
@@ -206,7 +206,7 @@
 	@rm -f "$@"
 	ttx "$<"
 
-$(EMOJI).ttf: check_sequence $(EMOJI).tmpl.ttf $(EMOJI_BUILDER) $(PUA_ADDER) \
+$(EMOJI).ttf: $(EMOJI).tmpl.ttf $(EMOJI_BUILDER) $(PUA_ADDER) \
 	$(ALL_COMPRESSED_FILES) | check_tools
 
 	@$(PYTHON) $(EMOJI_BUILDER) $(SMALL_METRICS) -V $< "$@" "$(COMPRESSED_DIR)/emoji_u"
@@ -230,5 +230,5 @@
 .SECONDARY: $(EMOJI_FILES) $(FLAG_FILES) $(RESIZED_FLAG_FILES) $(RENAMED_FLAG_FILES) \
   $(ALL_QUANTIZED_FILES) $(ALL_COMPRESSED_FILES)
 
-.PHONY:	clean flags emoji renamed_flags quantized compressed check_tools
+.PHONY:	clean flags emoji renamed_flags quantized compressed check_tools check_sequence
 
noto-emoji-d5e261484286d33a1fe8a02676f5907ecc02106f のみに存在: Makefile.~1~
