--- stderrlog-0.5.4/Cargo.toml	1970-01-01T00:00:01+00:00
+++ stderrlog-0.5.4/Cargo.toml	2025-01-18T16:21:38.242362+00:00
@@ -37,7 +37,6 @@
 features = [
     "clock",
     "std",
-    "wasmbind",
 ]
 optional = true
 default-features = false
@@ -47,27 +46,13 @@
 features = ["std"]
 
 [dependencies.termcolor]
-version = "~1.1"
+version = "1.1"
 
 [dependencies.thread_local]
-version = "~1.1"
-
-[dev-dependencies.clap]
-version = "3"
-features = ["cargo"]
-
-[dev-dependencies.docopt]
 version = "1.1"
 
 [dev-dependencies.libc]
 version = "0.2.18"
-
-[dev-dependencies.serde]
-version = "1.0"
-features = ["derive"]
-
-[dev-dependencies.structopt]
-version = "0.3.20"
 
 [features]
 default = ["timestamps"]
