--- a/build.rs
+++ b/build.rs
@@ -1,7 +1,6 @@
 use std::env;
 use std::path::Path;
 
-#[cfg(feature = "buildtime_bindgen")]
 const BINDGEN_FAILURE_MSG: &str = r#"Could not generate bindings.
 
 On Linux, the 'sys/acl.h' file is installed by the `libacl1-dev` package. To 
@@ -24,14 +23,9 @@ fn main() {
     // Tell cargo to invalidate the built crate whenever the wrapper changes
     println!("cargo:rerun-if-changed={wrapper}");
 
-    #[cfg(feature = "buildtime_bindgen")]
     bindgen_bindings(wrapper, &out_path);
-
-    #[cfg(not(feature = "buildtime_bindgen"))]
-    prebuilt_bindings(&out_path);
 }
 
-#[cfg(feature = "buildtime_bindgen")]
 fn bindgen_bindings(wrapper: &str, out_path: &Path) {
     // Build bindings for "wrapper.h". Tell cargo to invalidate the built
     // crate when any included header file changes.
@@ -98,19 +92,3 @@ fn bindgen_bindings(wrapper: &str, out_p
         .write_to_file(out_path)
         .expect("Couldn't write bindings!");
 }
-
-#[cfg(not(feature = "buildtime_bindgen"))]
-fn prebuilt_bindings(out_path: &Path) {
-    let target = env::var("CARGO_CFG_TARGET_OS").unwrap();
-
-    // Untrusted input check.
-    match target.as_str() {
-        "macos" | "linux" | "freebsd" => (),
-        s => panic!("Unsupported target OS: {}", s),
-    };
-
-    let bindings_path = format!("bindgen/bindings_{target}.rs");
-    if let Err(err) = std::fs::copy(&bindings_path, out_path) {
-        panic!("Can't copy {:?} to {:?}: {}", bindings_path, out_path, err);
-    }
-}
