--- icu_collections-2.3.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ icu_collections-2.3.0/Cargo.toml	2026-08-18T07:14:20.785304+00:00
@@ -142,9 +142,6 @@
 ]
 default-features = false
 
-[dev-dependencies.iai]
-version = "0.1.1"
-
 [dev-dependencies.postcard]
 version = "1.0.3"
 features = ["alloc"]
@@ -162,9 +159,6 @@
 version = "0.8.0"
 features = ["parse"]
 default-features = false
-
-[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.criterion]
-version = "0.5.0"
 
 [lints.clippy]
 alloc_instead_of_core = "warn"
