diff --git a/bazel/dependency_imports.bzl b/bazel/dependency_imports.bzl
index 6a12830..f6d31cf 100644
--- a/bazel/dependency_imports.bzl
+++ b/bazel/dependency_imports.bzl
@@ -11,7 +11,7 @@ load("@proxy_wasm_rust_sdk//bazel:dependencies.bzl", "proxy_wasm_rust_sdk_depend
 load("@rules_cc//cc:repositories.bzl", "rules_cc_dependencies", "rules_cc_toolchains")
 
 # go version for rules_go
-GO_VERSION = "1.17.5"
+GO_VERSION = "host"
 
 def envoy_dependency_imports(go_version = GO_VERSION):
     # TODO: allow building of tools for easier onboarding
