diff --git a/bazel/repository_locations.bzl b/bazel/repository_locations.bzl
index 3f24f41..fb4cc83 100644
--- a/bazel/repository_locations.bzl
+++ b/bazel/repository_locations.bzl
@@ -43,10 +43,10 @@ REPOSITORY_LOCATIONS = dict(
         urls = ["https://commondatastorage.googleapis.com/chromium-boringssl-docs/fips/boringssl-66005f41fbc3529ffe8d007708756720529da20d.tar.xz"],
     ),
     com_google_absl = dict(
-        sha256 = "2693730730247afb0e7cb2d41664ac2af3ad75c79944efd266be40ba944179b9",
-        strip_prefix = "abseil-cpp-06f0e767d13d4d68071c4fc51e25724e0fc8bc74",
+        sha256 = "9bbe4fd382bbd2911f31e8d70eae035a2168f8ca00d5aa51586c5ac7c86010b3",
+        strip_prefix = "abseil-cpp-c5a424a2a21005660b182516eb7a079cd8021699",
         # 2020-03-03
-        urls = ["https://github.com/abseil/abseil-cpp/archive/06f0e767d13d4d68071c4fc51e25724e0fc8bc74.tar.gz"],
+        urls = ["https://github.com/abseil/abseil-cpp/archive/c5a424a2a21005660b182516eb7a079cd8021699.tar.gz"],
     ),
     com_github_apache_thrift = dict(
         sha256 = "7d59ac4fdcb2c58037ebd4a9da5f9a49e3e034bf75b3f26d9fe48ba3d8806e6b",
