--- getrandom-0.3.4/Cargo.toml	1970-01-01T00:00:01+00:00
+++ getrandom-0.3.4/Cargo.toml	2026-02-14T14:51:28.439718+00:00
@@ -48,8 +48,6 @@
 [features]
 std = []
 wasm_js = [
-    "dep:wasm-bindgen",
-    "dep:js-sys",
 ]
 
 [lib]
@@ -67,59 +65,6 @@
 [dependencies.cfg-if]
 version = "1"
 
-[target.'cfg(all(any(target_os = "linux", target_os = "android"), not(any(all(target_os = "linux", target_env = ""), getrandom_backend = "custom", getrandom_backend = "linux_raw", getrandom_backend = "rdrand", getrandom_backend = "rndr"))))'.dependencies.libc]
-version = "0.2.154"
-default-features = false
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dependencies.wasm-bindgen]
-version = "0.2.98"
-optional = true
-default-features = false
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dev-dependencies.wasm-bindgen-test]
-version = "0.3"
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none"), target_feature = "atomics"))'.dependencies.js-sys]
-version = "0.3.77"
-optional = true
-default-features = false
-
-[target.'cfg(all(target_arch = "wasm32", target_os = "wasi", target_env = "p2"))'.dependencies.wasip2]
-version = "1"
-default-features = false
-
-[target.'cfg(all(target_os = "uefi", getrandom_backend = "efi_rng"))'.dependencies.r-efi]
-version = "5.1"
-default-features = false
-
-[target.'cfg(any(target_os = "dragonfly", target_os = "freebsd", target_os = "hurd", target_os = "illumos", target_os = "cygwin", all(target_os = "horizon", target_arch = "arm")))'.dependencies.libc]
-version = "0.2.154"
-default-features = false
-
-[target.'cfg(any(target_os = "haiku", target_os = "redox", target_os = "nto", target_os = "aix"))'.dependencies.libc]
-version = "0.2.154"
-default-features = false
-
-[target.'cfg(any(target_os = "ios", target_os = "visionos", target_os = "watchos", target_os = "tvos"))'.dependencies.libc]
-version = "0.2.154"
-default-features = false
-
-[target.'cfg(any(target_os = "macos", target_os = "openbsd", target_os = "vita", target_os = "emscripten"))'.dependencies.libc]
-version = "0.2.154"
-default-features = false
-
-[target.'cfg(target_os = "netbsd")'.dependencies.libc]
-version = "0.2.154"
-default-features = false
-
-[target.'cfg(target_os = "solaris")'.dependencies.libc]
-version = "0.2.154"
-default-features = false
-
-[target.'cfg(target_os = "vxworks")'.dependencies.libc]
-version = "0.2.154"
-default-features = false
-
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
@@ -131,3 +76,4 @@
     "cfg(getrandom_test_netbsd_fallback)",
     'cfg(target_os, values("cygwin"))',
 ]
+
