--- src/libmetis/CMakeLists.orig.txt	2011-06-22 19:47:13.000000000 +0200
+++ src/libmetis/CMakeLists.txt	2013-03-15 17:13:19.948911972 +0100
@@ -8,8 +8,9 @@
 PRIVATE ${PROJECT_SOURCE_DIR}/src/GKlib ${PROJECT_BINARY_DIR}/src/include
 )
 target_link_libraries(metis PRIVATE $<$<BOOL:${UNIX}>:m>)
+set_target_properties(metis PROPERTIES SOVERSION 0)
 
 install(TARGETS metis
-    LIBRARY DESTINATION lib
-    RUNTIME DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    RUNTIME DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
