--- libmetis/CMakeLists.orig.txt	2016-02-16 19:06:56.480139000 +0100
+++ libmetis/CMakeLists.txt	2016-02-16 19:07:55.511134162 +0100
@@ -3,14 +3,14 @@
 # Find sources.
 file(GLOB metis_sources *.c)
 # Build libmetis.
-add_library(metis ${METIS_LIBRARY_TYPE} ${GKlib_sources} ${metis_sources})
-set_target_properties(metis PROPERTIES SOVERSION 0)
+add_library(metis64 ${METIS_LIBRARY_TYPE} ${GKlib_sources} ${metis_sources})
+set_target_properties(metis64 PROPERTIES SOVERSION 0)
 if(UNIX)
-  target_link_libraries(metis m)
+  target_link_libraries(metis64 m)
 endif()
 
 if(METIS_INSTALL)
-  install(TARGETS metis
+  install(TARGETS metis64
     LIBRARY DESTINATION lib${LIB_SUFFIX}
     RUNTIME DESTINATION lib${LIB_SUFFIX}
     ARCHIVE DESTINATION lib${LIB_SUFFIX})
--- programs/CMakeLists.orig.txt	2013-03-30 17:24:45.000000000 +0100
+++ programs/CMakeLists.txt	2016-02-16 22:42:18.625946692 +0100
@@ -2,19 +2,19 @@
 include_directories(../libmetis)
 link_directories(/home/karypis/local/lib)
 # Build program.
-add_executable(gpmetis gpmetis.c cmdline_gpmetis.c io.c stat.c)
-add_executable(ndmetis ndmetis.c cmdline_ndmetis.c io.c smbfactor.c)
-add_executable(mpmetis mpmetis.c cmdline_mpmetis.c io.c stat.c)
-add_executable(m2gmetis m2gmetis.c cmdline_m2gmetis.c io.c)
-add_executable(graphchk graphchk.c io.c)
-add_executable(cmpfillin cmpfillin.c io.c smbfactor.c)
-foreach(prog gpmetis ndmetis mpmetis m2gmetis graphchk cmpfillin)
-  target_link_libraries(${prog} metis)
+add_executable(gpmetis64 gpmetis.c cmdline_gpmetis.c io.c stat.c)
+add_executable(ndmetis64 ndmetis.c cmdline_ndmetis.c io.c smbfactor.c)
+add_executable(mpmetis64 mpmetis.c cmdline_mpmetis.c io.c stat.c)
+add_executable(m2gmetis64 m2gmetis.c cmdline_m2gmetis.c io.c)
+add_executable(graphchk64 graphchk.c io.c)
+add_executable(cmpfillin64 cmpfillin.c io.c smbfactor.c)
+foreach(prog gpmetis64 ndmetis64 mpmetis64 m2gmetis64 graphchk64 cmpfillin64)
+  target_link_libraries(${prog} metis64)
 #  target_link_libraries(${prog} metis profiler)
 endforeach(prog)
 
 if(METIS_INSTALL)
-  install(TARGETS gpmetis ndmetis mpmetis m2gmetis graphchk cmpfillin
+  install(TARGETS gpmetis64 ndmetis64 mpmetis64 m2gmetis64 graphchk64 cmpfillin64
     RUNTIME DESTINATION bin)
 endif()
 
