--- GKlib/GKlib.orig.h	2013-03-30 17:24:45.000000000 +0100
+++ GKlib/GKlib.h	2019-02-05 14:45:44.051629682 +0100
@@ -43,7 +43,7 @@
 #include <sys/stat.h>
 
 #if defined(__WITHPCRE__)
-  #include <pcreposix.h>
+  #include <pcre2posix.h>
 #else
   #if defined(USE_GKREGEX)
     #include "gkregex.h"
--- GKlib/test/Makefile.in.orig.old	2013-03-30 17:24:45.000000000 +0100
+++ GKlib/test/Makefile.in.old	2019-02-05 15:04:35.009847901 +0100
@@ -196,7 +196,7 @@
 
   #Libraries needed for linking
   LIBSDIR  = -L$(BUILDDIR) -L$(GKLIBBUILDDIR) -L$(HOME)/local/lib
-  LIBS     = -lGKlib -lpcreposix -lpcre -lz -lm
+  LIBS     = -lGKlib -lpcre2-posix -lz -lm
 
   # Standard file extensions 
   OBJEXT = .o
