--- GKlib/Makefile.orig	2013-03-11 20:50:42.000000000 +0100
+++ GKlib/Makefile	2013-03-17 14:26:23.632867727 +0100
@@ -41,10 +41,10 @@
     CONFIG_FLAGS += -DPCRE=$(pcre)
 endif
 ifneq ($(gkregex), not-set)
-    CONFIG_FLAGS += -DGKREGEX=$(pcre)
+    CONFIG_FLAGS += -DGKREGEX=$(gkregex)
 endif
 ifneq ($(gkrand), not-set)
-    CONFIG_FLAGS += -DGKRAND=$(pcre)
+    CONFIG_FLAGS += -DGKRAND=$(gkrand)
 endif
 ifneq ($(prefix), not-set)
     CONFIG_FLAGS += -DCMAKE_INSTALL_PREFIX=$(prefix)

--- Makefile.origin	2013-03-30 17:24:45.000000000 +0100
+++ Makefile	2013-04-14 20:04:38.887674462 +0200
@@ -9,6 +9,11 @@
 gklib_path = not-set
 shared     = not-set
 cc         = not-set
+libsuffix  = not-set
+pcre       = not-set
+gkregex    = not-set
+gkrand     = not-set
+
 
 
 # Basically proxies everything to the builddir cmake.
@@ -23,6 +28,9 @@
 ifeq ($(gklib_path), not-set)
     gklib_path = GKlib
 endif
+ifneq ($(libsuffix), not-set)
+     CONFIG_FLAGS += -DLIB_SUFFIX=$(libsuffix)
+endif
 CONFIG_FLAGS += -DGKLIB_PATH=$(abspath $(gklib_path))
 ifneq ($(gdb), not-set)
     CONFIG_FLAGS += -DGDB=$(gdb)
@@ -51,6 +59,15 @@
 ifneq ($(cc), not-set)
     CONFIG_FLAGS += -DCMAKE_C_COMPILER=$(cc)
 endif
+ifneq ($(pcre), not-set)
+    CONFIG_FLAGS += -DPCRE=$(pcre)
+endif
+ifneq ($(gkregex), not-set)
+    CONFIG_FLAGS += -DGKREGEX=$(gkregex)
+endif
+ifneq ($(gkrand), not-set)
+    CONFIG_FLAGS += -DGKRAND=$(gkrand)
+endif
 
 VERNUM=5.1.0
 PKGNAME=metis-$(VERNUM)
