commit da88110aea24b8953614449f78aa12e6a5c96826
Author: Tom Hughes <tom@compton.nu>
Date:   Sun Apr 9 14:59:06 2017 +0100

    Rip out mason stuff - we use our own packages

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c098b0f..bdadb90 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,8 +4,6 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED on)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
-include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/mason.cmake)
-
 option(WERROR "Add -Werror flag to build (turns warnings into errors)" ON)
 
 # configure optimization
@@ -23,28 +21,18 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   set(DESIRED_WARNINGS "${DESIRED_WARNINGS} -Wmost")
 endif()
 
-# Note: -D_GLIBCXX_USE_CXX11_ABI=0 is needed to support mason packages that are precompiled libs
-# Currently we only depend on a header only library, but this will help avoid issues when more libs are added via mason
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OPTIMIZATION_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0 ${DESIRED_WARNINGS}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OPTIMIZATION_FLAGS} ${DESIRED_WARNINGS}")
 
 if (WERROR)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
 endif()
 
-# mason_use is a mason function within the mason.cmake file and provides ready-to-go vars, like "STATIC_LIBS" and "INCLUDE_DIRS"
-mason_use(boost_libfilesystem VERSION 1.72.0)
-mason_use(boost_libsystem VERSION 1.72.0)
-mason_use(boost VERSION 1.72.0 HEADER_ONLY)
-include_directories(SYSTEM ${MASON_PACKAGE_boost_INCLUDE_DIRS})
-
-mason_use(rapidjson VERSION 1.1.0 HEADER_ONLY)
-include_directories(SYSTEM ${MASON_PACKAGE_rapidjson_INCLUDE_DIRS})
+find_package(Boost 1.69 REQUIRED COMPONENTS filesystem system)
+include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 
-mason_use(geometry VERSION 1.0.0 HEADER_ONLY)
-include_directories(SYSTEM ${MASON_PACKAGE_geometry_INCLUDE_DIRS})
+find_package(RapidJSON)
 
-mason_use(catch VERSION 1.9.6 HEADER_ONLY)
-include_directories(SYSTEM ${MASON_PACKAGE_catch_INCLUDE_DIRS})
+include_directories(SYSTEM /usr/include/catch)
 
 include_directories("${PROJECT_SOURCE_DIR}/include")
 
