--- toml_parser-1.0.9+spec-1.1.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ toml_parser-1.0.9+spec-1.1.0/Cargo.toml	2026-02-16T23:06:20.815538+00:00
@@ -13,7 +13,7 @@
 edition = "2021"
 rust-version = "1.76"
 name = "toml_parser"
-version = "1.0.9+spec-1.1.0"
+version = "1.0.9"
 build = false
 include = [
     "build.rs",
@@ -193,3 +193,4 @@
 [lints.rust.rust_2018_idioms]
 level = "warn"
 priority = -1
+
