diff -ru a/scripts/packages/bazel.sh b/scripts/packages/bazel.sh
--- a/scripts/packages/bazel.sh
+++ b/scripts/packages/bazel.sh
@@ -162,7 +162,7 @@
 if [[ ! -x ${BAZEL_REAL} && -x ${bazel_real_path} ]]; then
   # Note that "bazel --version" is very fast and doesn't start the Bazel server,
   # as opposed to "bazel version".
-  readonly bazel_real_version="$("${bazel_real_path}" --version | grep '^bazel ' | cut -d' ' -f2)"
+  readonly bazel_real_version="$("${bazel_real_path}" --version | grep '^bazel ' | cut -d' ' -f2 | sed 's/.$//')"
   if [[ $bazel_real_version == $bazel_version ]]; then
     BAZEL_REAL="${bazel_real_path}"
   fi
