diff -up foomatic-db-engine-4.0.11/foomatic-cleanupdrivers.1.in.manpages foomatic-db-engine-4.0.11/foomatic-cleanupdrivers.1.in
--- foomatic-db-engine-4.0.11/foomatic-cleanupdrivers.1.in.manpages	2014-08-28 14:48:37.799189679 +0100
+++ foomatic-db-engine-4.0.11/foomatic-cleanupdrivers.1.in	2014-08-28 14:48:37.799189679 +0100
@@ -0,0 +1,16 @@
+.\" This -*- nroff -*- source file is part of foomatic.
+.\"
+.TH FOOMATIC-CLEANUPDRIVERS 1 "2014-08-21" "Foomatic Project"
+.SH NAME
+foomatic-cleanupdrivers \- Remove driver entries with no command line
+.SH SYNOPSIS
+.HP \w'\fBfoomatic-cleanupdrivers\fR\ 'u
+\fBfoomatic-cleanupdrivers\fR [\fIDBDIR\fR]
+
+.SH DESCRIPTION
+.PP
+The \fBfoomatic\-cleanupdrivers\fR command removes all driver XML
+files which have an empty command line.
+
+.PP
+It is not intended to be called directly by the user.
diff -up foomatic-db-engine-4.0.11/foomatic-datafile.1.manpages foomatic-db-engine-4.0.11/foomatic-datafile.1
--- foomatic-db-engine-4.0.11/foomatic-datafile.1.manpages	2014-08-28 14:48:37.799189679 +0100
+++ foomatic-db-engine-4.0.11/foomatic-datafile.1	2014-08-28 14:48:37.799189679 +0100
@@ -0,0 +1 @@
+.so man1/foomatic-ppdfile.1
diff -up foomatic-db-engine-4.0.11/foomatic-extract-text.1.in.manpages foomatic-db-engine-4.0.11/foomatic-extract-text.1.in
--- foomatic-db-engine-4.0.11/foomatic-extract-text.1.in.manpages	2014-08-28 14:48:37.799189679 +0100
+++ foomatic-db-engine-4.0.11/foomatic-extract-text.1.in	2014-08-28 14:48:37.799189679 +0100
@@ -0,0 +1,16 @@
+.\" This -*- nroff -*- source file is part of foomatic.
+.\"
+.TH FOOMATIC-EXTRACT-TEXT 1 "2014-08-21" "Foomatic Project"
+.SH NAME
+foomatic-extract-text \- Generate .po files from Foomatic UI strings
+.SH SYNOPSIS
+.HP \w'\fBfoomatic-extract-text\fR\ 'u
+\fBfoomatic-extract-text\fR
+
+.SH DESCRIPTION
+.PP
+The \fBfoomatic\-extract\-text\fR command generates .po files from the
+UI strings in the currently installed Foomatic database.
+
+.PP
+It is not intended to be called directly by the user.
diff -up foomatic-db-engine-4.0.11/foomatic-fix-xml.1.in.manpages foomatic-db-engine-4.0.11/foomatic-fix-xml.1.in
--- foomatic-db-engine-4.0.11/foomatic-fix-xml.1.in.manpages	2014-08-28 14:48:37.799189679 +0100
+++ foomatic-db-engine-4.0.11/foomatic-fix-xml.1.in	2014-08-28 14:48:37.799189679 +0100
@@ -0,0 +1,14 @@
+.\" This -*- nroff -*- source file is part of foomatic.
+.\"
+.TH FOOMATIC-FIX-XML 1 "2014-08-21" "Foomatic Project"
+.SH NAME
+foomatic-fix-xml \- Remove leading blank lines from Foomatic database
+.SH SYNOPSIS
+.HP \w'\fBfoomatic-fix-xml\fR\ 'u
+\fBfoomatic-fix-xml\fR [\fIDBDIR\fR]
+
+.SH DESCRIPTION
+.PP
+Use the \fBfoomatic\-fix\-xml\fR command if you have compiled
+foomatic-perl-data against libxml 1.x and you have old database
+entries with a leading blank line.
diff -up foomatic-db-engine-4.0.11/foomatic-nonumericalids.1.in.manpages foomatic-db-engine-4.0.11/foomatic-nonumericalids.1.in
--- foomatic-db-engine-4.0.11/foomatic-nonumericalids.1.in.manpages	2014-08-28 14:48:37.800189684 +0100
+++ foomatic-db-engine-4.0.11/foomatic-nonumericalids.1.in	2014-08-28 14:48:37.800189684 +0100
@@ -0,0 +1,17 @@
+.\" This -*- nroff -*- source file is part of foomatic.
+.\"
+.TH FOOMATIC-NONUMERICALIDS 1 "2014-08-21" "Foomatic Project"
+.SH NAME
+foomatic-nonumericalids \- Convert numerical printer IDS to clear-text IDs
+.SH SYNOPSIS
+.HP \w'\fBfoomatic-nonumericalids\fR\ 'u
+\fBfoomatic-nonumericalids\fR
+
+.SH DESCRIPTION
+.PP
+This script changes all old cryptic numerical printer IDs from the old
+PostGreSQL time to clear-text printer IDs in the installed Foomatic
+database.
+
+.PP
+It is very unlikely you need to do this.
diff -up foomatic-db-engine-4.0.11/foomatic-ppd-to-xml.1.in.manpages foomatic-db-engine-4.0.11/foomatic-ppd-to-xml.1.in
--- foomatic-db-engine-4.0.11/foomatic-ppd-to-xml.1.in.manpages	2014-08-28 14:48:37.800189684 +0100
+++ foomatic-db-engine-4.0.11/foomatic-ppd-to-xml.1.in	2014-08-28 14:48:37.800189684 +0100
@@ -0,0 +1,104 @@
+.\" This -*- nroff -*- source file is part of foomatic.
+.\"
+.TH FOOMATIC-PPD-TO-XML 1 "2014-08-21" "Foomatic Project"
+.SH NAME
+foomatic-ppd-to-xml \- Generate Foomatic printer XML from a PPD
+.SH SYNOPSIS
+.HP \w'\fBfoomatic-ppd-to-xml\fR\ 'u
+\fBfoomatic-ppd-to-xml\fR \fIOPTIONS\FR [\fIDRIVER\fR:]\fIPPD\fR [[\fIDRIVER\fR:]\fIPPD\fR ...]
+.HP \w'\fBfoomatic-ppd-to-xml\fR\ 'u
+\fBfoomatic-ppd-to-xml\fR \-h
+
+.SH DESCRIPTION
+The first form of the
+.B foomatic-ppd-to-xml
+command creates Foomatic printer XML corresponding to a given PPD
+file.
+
+.PP
+If \fIDRIVER\fR is specified, it is the driver to which the PPD file
+should get assigned. If not supplied, the first driver in the driver
+list supplied via the \fB\-d\fR option is used. Note that the PPD files
+have to be supplied to different drivers, so drivers need to be
+specified if using more than one PPD.
+
+.PP
+The second form prints a help message and exits.
+
+.SS Options
+.TP 5
+\fB\-d\fR \fIDRIVERS\fR
+Comma-separated list of drivers with which the printer works. First
+driver is the one for which the PPD file is. If not otherwise stated
+by the \fB\-r\fR option, this is also the recommended driver. Drivers
+already specified together with PPDs do not need to be specified with
+this option again. The option is not needed if drivers are specified
+for all PPDs.
+
+.TP 5
+\fB\-r\fR \fIDRIVER\fR
+Recommended driver.
+
+.TP 5
+\fB\-p\fR \fIPDLS\fR
+Comma-separated list of known Page Description Languages (PDLs) which
+the printer supports. This will add all suitable drivers to the XML
+entry. Currently supported are: Postscript, PCLXL, PCL6, PCL5e, PCL5c,
+PCL5, and PCL4.
+
+.TP 5
+\fB\-l\fR
+Add a link to the PPD file to the driver entry in the XML file.
+
+.TP 5
+\fB\-b\fR \fIDIRECTORY\fR
+Base directory for a relative link to the PPD. If the base directory
+is given, the link set via the \fB\-l\fR option is relative to this
+directory (and not relative to the current directory). With a base
+directory given the \fIPPD\fR can also be given relative to this
+directory.
+
+.TP 5
+\fB\-D\fR
+Add only the driver requested by \fB\-d\fR and \fB\-r\fR to the driver
+list in the XML file, do not add drivers based on PDLs, not even on
+the PDLs supplied by the \fB\-p\fR option.
+
+.TP 5
+\fB\-c\fR \fICOMMENT\fR
+Add the given comment to the comment field of the XML file. Comments
+are in HTML.
+
+.TP 5
+\fB\-C\fR
+Suppress any default comments (max paper width, pages per minute,
+automatically generated XML, ...). Does not suppress the comment
+supplied with \fB\-c\fR.
+
+.TP 5
+\fB\-f\fR \fIDIRECTORY\fR
+Directory in which to write the resulting XML file. The name of the
+file will be the printer ID with the ".xml" extension.
+
+.TP 5
+\fB\-n\fR
+Do not write modified versions of existing XML files, only create XML
+files for printers for which there is no XML file yet.
+
+.TP 5
+\fB\-x\fR
+Create XML files independent of whether or not there are already XML
+files for the printers covered by the PPD. This makes the process
+vastly faster if you have PPD files for printers which are definitely
+not yet in the Foomatic database.
+
+.SH EXIT STATUS
+\fBfoomatic-ppd-to-xml\fR returns 0 on success, 1 otherwise.
+
+.SH BUGS
+None so far.
+
+Please post bug reports on
+
+http://lists.freestandards.org/mailman/listinfo/printing-foomatic
+
diff -up foomatic-db-engine-4.0.11/foomatic-preferred-driver.8.in.manpages foomatic-db-engine-4.0.11/foomatic-preferred-driver.8.in
--- foomatic-db-engine-4.0.11/foomatic-preferred-driver.8.in.manpages	2014-08-28 14:48:59.729304630 +0100
+++ foomatic-db-engine-4.0.11/foomatic-preferred-driver.8.in	2014-08-28 15:00:59.291077529 +0100
@@ -1,25 +1,16 @@
 .\" This -*- nroff -*- source file is part of foomatic.
 .\"
-.TH FOOMATIC-PREFERRED-DRIVER 8 "2001-05-07" "Foomatic Project"
+.TH FOOMATIC-PREFERRED-DRIVER 8 "2014-08-28" "Foomatic Project"
 .SH NAME
-foomatic-preferred-driver \- <put a short description here>
+foomatic-preferred-driver \- set default drivers in Foomatic database
 .SH SYNOPSIS
 .B foomatic-preferred-driver
-.\"[\fIOPTION\fR]...
 
-[To be edited]
-
-.SH SEE ALSO
-.IR foomatic-XXX (1),
-
-.SH EXIT STATUS
-.B foomatic-preferred-driver
-returns ...
-
-.SH AUTHOR
-Manfred Wassmann <\fImanolo@NCC-1701.B.Shuttle.de\fR> for the foomatic
-project using output from the associated binary.
-
-.SH BUGS
-This manpage contains no useful information.
+.SH DESCRIPTION
+.PP
+This command reads the printer database overview and puts default
+driver entries (<driver>..</driver>) into every printer's XML database
+entry.
 
+.PP
+It is very unlikely you need to do this.
diff -up foomatic-db-engine-4.0.11/foomatic-printermap-to-gutenprint-xml.1.in.manpages foomatic-db-engine-4.0.11/foomatic-printermap-to-gutenprint-xml.1.in
--- foomatic-db-engine-4.0.11/foomatic-printermap-to-gutenprint-xml.1.in.manpages	2014-08-28 14:48:37.800189684 +0100
+++ foomatic-db-engine-4.0.11/foomatic-printermap-to-gutenprint-xml.1.in	2014-08-28 14:48:37.800189684 +0100
@@ -0,0 +1,18 @@
+.\" This -*- nroff -*- source file is part of foomatic.
+.\"
+.TH FOOMATIC-PRINTERMAP-TO-GUTENPRINT-XML 1 "2014-08-21" "Foomatic Project"
+.SH NAME
+foomatic-printermap-to-gutenprint-xml \- Updates Gutenprint driver XML file
+.SH SYNOPSIS
+.HP \w'\fBfoomatic-printermap-to-gutenprint-xml\fR\ 'u
+\fBfoomatic-printermap-to-gutenprint-xml\fR
+
+.SH DESCRIPTION
+.PP
+This command updates the driver XML file for the Gutenprint driver to
+the state of the foomatic-printermap file in the Gutenprint source
+tree. Used to update the driver XML file for Gutenprint in the
+OpenPrinting database.
+
+.PP
+It is very unlikely you need to do this.
diff -up foomatic-db-engine-4.0.11/foomatic-replaceoldprinterids.1.in.manpages foomatic-db-engine-4.0.11/foomatic-replaceoldprinterids.1.in
--- foomatic-db-engine-4.0.11/foomatic-replaceoldprinterids.1.in.manpages	2014-08-28 14:48:37.800189684 +0100
+++ foomatic-db-engine-4.0.11/foomatic-replaceoldprinterids.1.in	2014-08-28 14:48:37.800189684 +0100
@@ -0,0 +1,18 @@
+.\" This -*- nroff -*- source file is part of foomatic.
+.\"
+.TH FOOMATIC-REPLACEOLDPRINTERIDS 1 "2014-08-21" "Foomatic Project"
+.SH NAME
+foomatic-replaceoldprinterids \- Replace references to old printer IDs
+.SH SYNOPSIS
+.HP \w'\fBfoomatic-replaceoldprinterids\fR\ 'u
+\fBfoomatic-replaceoldprinterids\fR
+
+.SH DESCRIPTION
+.PP
+Replaces references to old numerical or otherwise obsolete printer IDs
+in arbitrary file. By default it uses the db/oldprinterids file of the
+current Foomatic database as translation table, but you can use
+arbitrary translation tables.
+
+.PP
+It is very unlikely you need to use this.
diff -up foomatic-db-engine-4.0.11/foomatic-searchprinter.1.in.manpages foomatic-db-engine-4.0.11/foomatic-searchprinter.1.in
--- foomatic-db-engine-4.0.11/foomatic-searchprinter.1.in.manpages	2014-08-28 14:48:37.800189684 +0100
+++ foomatic-db-engine-4.0.11/foomatic-searchprinter.1.in	2014-08-28 14:48:37.800189684 +0100
@@ -0,0 +1,82 @@
+.\" This -*- nroff -*- source file is part of foomatic.
+.\"
+.TH FOOMATIC-SEARCHPRINTER 1 "2014-08-21" "Foomatic Project"
+.SH NAME
+foomatic-searchprinter \- Search Foomatic database
+.SH SYNOPSIS
+.HP \w'\fBfoomatic-searchprinter\fR\ 'u
+\fBfoomatic-searchprinter\fR [\-m\fIM\fR] [\-d\FID\fR] \fISEARCHTERM\fR
+.HP \w'\fBfoomatic-searchprinter\fR\ 'u
+\fBfoomatic-searchprinter\fR \-h
+
+.SH DESCRIPTION
+This command displays Foomatic database entries matching \fISEARCHTERM\fR.
+
+.SS Options
+.PP
+\fB\-m\fR\fIM\fR
+.RS 4
+Search mode:
+.PP
+\fIM\fR = 0
+.RS 4
+Match everything (default)
+.RE
+.PP
+\fIM\fR = 1
+.RS 4
+No matches on only the manufacturer
+.RE
+.PP
+\fIM\fR = 2
+.RS 4
+No matches on only the manufacturer or only the model
+.RE
+.PP
+\fIM\fR = 3
+.RS 4
+Exact matches of device ID, make/model, or Foomatic ID plus matches of
+the page description language in the device ID to appropriate
+"Generic" printers
+.RE
+.PP
+\fIM\fR = 4
+.RS 4
+Exact matches of device ID, make/model, or Foomatic ID only
+.RE
+.RE
+
+.PP
+\fB\-d\fR\fID\fR
+.RS 4
+Display results:
+.PP
+\fID\fR = 0
+.RS 4
+Everything
+.RE
+.PP
+\fID\fR = 1
+.RS 4
+Only best match class (default)
+.RE
+.PP
+\fID\fR = 2
+.RS 4
+Only best match
+.RE
+.RE
+
+.PP
+\fB\-h\fR
+.RS 4
+Show help message and exit.
+.RE
+
+.SH BUGS
+None so far.
+
+Please post bug reports on
+
+http://lists.freestandards.org/mailman/listinfo/printing-foomatic
+
