--- ft2demos-2.5.2/Makefile
+++ ft2demos-2.5.2/Makefile
@@ -296,10 +296,10 @@ else
   # The following programs are not compiled automatically; either comment
   # out the affected line or use the program name as a Makefile target.
   #
-  # EXES += ftchkwd
-  # EXES += ftmemchk
-  # EXES += ftpatchk
-  # EXES += fttimer
+  EXES += ftchkwd
+  EXES += ftmemchk
+  EXES += ftpatchk
+  EXES += fttimer
   # EXES += testname
 
   # Not all demo programs have a man page; we thus check for existence in a
