--- a/grub-core/net/net.c	2023-12-20 07:54:46.000000000 -0800
+++ b/grub-core/net/net.c	2024-05-15 16:13:16.287495938 -0700
@@ -1595,8 +1595,7 @@
       grub_netbuff_free (file->device->net->packs.first->nb);
       grub_net_remove_packet (file->device->net->packs.first);
     }
-  if (!file->device->net->broken)
-    file->device->net->protocol->close (file);
+  file->device->net->protocol->close (file);
   grub_free (file->device->net->name);
   return GRUB_ERR_NONE;
 }
@@ -1819,10 +1818,7 @@
     file->device->net->stall = 0;
     err = file->device->net->protocol->open (file, file->device->net->name);
     if (err)
-      {
-	file->device->net->broken = 1;
-	return err;
-      }
+      return err;
     grub_net_fs_read_real (file, NULL, offset);
     return grub_errno;
   }
@@ -1831,9 +1827,6 @@
 static grub_ssize_t
 grub_net_fs_read (grub_file_t file, char *buf, grub_size_t len)
 {
-  if (file->device->net->broken)
-    return -1;
-
   if (file->offset != file->device->net->offset)
     {
       grub_err_t err;
