--- zbus-5.12.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ zbus-5.12.0/Cargo.toml	2025-10-23T17:38:11.082323+00:00
@@ -261,9 +261,6 @@
 ]
 default-features = false
 
-[target.'cfg(any(target_os = "macos", windows))'.dependencies.async-recursion]
-version = "1.1.1"
-
 [target."cfg(unix)".dependencies.nix]
 version = "0.30"
 features = [
@@ -272,25 +269,9 @@
     "user",
 ]
 default-features = false
-
-[target."cfg(windows)".dependencies.uds_windows]
-version = "1.1.0"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.61"
-features = [
-    "Win32_Foundation",
-    "Win32_Security_Authorization",
-    "Win32_System_Memory",
-    "Win32_Networking",
-    "Win32_Networking_WinSock",
-    "Win32_NetworkManagement",
-    "Win32_NetworkManagement_IpHelper",
-    "Win32_System_IO",
-    "Win32_System_Threading",
-]
 
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
 check-cfg = ["cfg(tokio_unstable)"]
+
