diff -Naur clamav-1.4.3-original/libclamav_rust/Cargo.toml clamav-1.4.3/libclamav_rust/Cargo.toml
--- clamav-1.4.3-original/libclamav_rust/Cargo.toml	2025-12-04 10:01:33.233889863 +0000
+++ clamav-1.4.3/libclamav_rust/Cargo.toml	2025-12-04 10:02:29.088468217 +0000
@@ -12,7 +12,7 @@
 sha2 = "0.10"
 tempfile = "3"
 thiserror = "1"
-image = "0.24"
+image = ">=0.24, <0.26"
 rustdct = "0.7"
 transpose = "0.2"
 num-traits = "0.2"
@@ -21,7 +21,7 @@
 unicode-segmentation = "1.10"
 bindgen = "0.72"
 onenote_parser = "0.3.1"
-hex-literal = "0.4"
+hex-literal = ">=0.4, <2.0"
 inflate = "0.4"
 bzip2-rs = "0.1"
 byteorder = "1.5"
