--- gix-path-0.12.1/Cargo.toml	2006-07-24T01:21:28+00:00
+++ gix-path-0.12.1/Cargo.toml	2026-06-26T18:49:48.590731+00:00
@@ -53,17 +53,6 @@
 [dev-dependencies.serial_test]
 version = "3.4.0"
 default-features = false
-
-[target."cfg(windows)".dev-dependencies.windows]
-version = "0.62.1"
-features = [
-    "Win32_System_Com",
-    "Win32_System_Threading",
-    "Win32_UI_Shell",
-]
-
-[target."cfg(windows)".dev-dependencies.winreg]
-version = "0.56.0"
 
 [lints.clippy]
 bool_to_int_with_if = "allow"
@@ -143,3 +132,4 @@
 priority = -1
 
 [lints.rust]
+
