diff -ru testtools-2.4.0-orig/testtools/tests/test_testresult.py testtools-2.4.0/testtools/tests/test_testresult.py
--- testtools-2.4.0-orig/testtools/tests/test_testresult.py	2019-11-28 00:58:29.000000000 -0800
+++ testtools-2.4.0/testtools/tests/test_testresult.py	2020-05-16 13:40:50.141580104 -0700
@@ -2755,6 +2755,9 @@
         textoutput = self._setup_external_case("import bad")
         self._write_module("bad", "utf-8", _u("\ufeff^ = 0 # %s\n") % text)
         textoutput = self._run_external_case()
+        # Python 3.9 no longer prints the '\ufeff'
+        if sys.version_info >= (3,9):
+            textoutput = textoutput.replace('\ufeff', '')
         self.assertThat(
             textoutput,
             MatchesRegex(
