--- vm-memory-0.17.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ vm-memory-0.17.1/Cargo.toml	2026-05-11T13:44:43.436079+00:00
@@ -41,11 +41,9 @@
 backend-atomic = ["arc-swap"]
 backend-bitmap = [
     "dep:libc",
-    "dep:winapi",
 ]
 backend-mmap = [
     "dep:libc",
-    "dep:winapi",
 ]
 default = ["rawfd"]
 rawfd = ["dep:libc"]
@@ -92,14 +90,7 @@
 [dev-dependencies.vmm-sys-util]
 version = "0.15.0"
 
-[target.'cfg(target_family = "windows")'.dependencies.winapi]
-version = "0.3"
-features = [
-    "errhandlingapi",
-    "sysinfoapi",
-]
-optional = true
-
 [profile.bench]
 lto = true
 codegen-units = 1
+
