@@ -, +, @@ 
---
 main.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)
--- a/main.cpp	
+++ a/main.cpp	
@@ -133,6 +133,7 @@ int main (int argc, char *argv[]) {
    ClassesXml *ca=0;
    MethodresponseXml *mrsp=0;
    string sb;
+   {
 #ifdef HAVE_LIBWWW
    CimomWww cimom;
 #elif HAVE_LIBCURL
@@ -643,6 +644,7 @@ fini:
    delete cna;
    delete ca;
    delete mrsp;
+   }
 #ifdef HAVE_LIBCURL
    // These needs to be called once
    curl_global_cleanup();
-- 
