diff -ruN a/bindata.gemspec b/bindata.gemspec
--- a/bindata.gemspec	2021-05-17 20:40:07.000000000 -0700
+++ b/bindata.gemspec	2022-06-30 19:46:59.489361986 -0700
@@ -12,7 +12,7 @@
   s.require_path = 'lib'
   s.extra_rdoc_files = ['NEWS.rdoc']
   s.rdoc_options << '--main' << 'NEWS.rdoc'
-  s.files = `git ls-files`.split("\n")
+  s.files = Dir['**/*']
   s.license = 'Ruby'
 
   s.add_development_dependency('rake')
