diff -ruN a/td.gemspec b/td.gemspec
--- a/td.gemspec        2024-02-29 21:18:18.870483966 +0000
+++ b/td.gemspec        2024-02-29 21:19:09.654145224 +0000
@@ -10,9 +10,9 @@
   gem.version               = TreasureData::TOOLBELT_VERSION
   gem.authors               = ["Treasure Data, Inc."]
   gem.email                 = "support@treasure-data.com"
-  gem.files                 = `git ls-files`.split("\n").select { |f| !f.start_with?('dist') }
-  gem.test_files            = `git ls-files -- {test,spec,features}/*`.split("\n")
-  gem.executables           = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
+  gem.files                 = Dir['**/*']
+  gem.test_files            = gem.files.grep(%r{^(test|spec|features)/})
+  gem.executables           = gem.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   gem.require_paths         = ['lib']
   gem.required_ruby_version = '>= 2.1'
   gem.licenses              = ["Apache-2.0"]
