diff --git a/Makefile.in b/Makefile.in
index e5d2b1a..bc53e34 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -1019,6 +1019,8 @@ clean-filterLTLIBRARIES:
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
 	@list='$(lib_LTLIBRARIES)'; test -n "$(libdir)" || list=; \
+	mkdir -p $(DESTDIR)$(libdir)/pkgconfig; \
+	cp aspell.pc $(DESTDIR)$(libdir)/pkgconfig/aspell.pc; \
 	list2=; for p in $$list; do \
 	  if test -f $$p; then \
 	    list2="$$list2 $$p"; \
diff --git a/aspell.pc.in b/aspell.pc.in
new file mode 100644
index 0000000..13da044
--- /dev/null
+++ b/aspell.pc.in
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+pkgdatadir=@pkgdatadir@
+
+Name: Aspell
+Description: A spelling checker.
+Version: @VERSION@
+Requires: 
+Libs: -L${libdir} -laspell
+Cflags: -I${includedir}
diff --git a/configure b/configure
index 8a6e697..995a3df 100755
--- a/configure
+++ b/configure
@@ -18732,7 +18732,7 @@ rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
 #                                                                 #
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
-ac_config_files="$ac_config_files Makefile gen/Makefile common/Makefile lib/Makefile data/Makefile auto/Makefile modules/Makefile modules/tokenizer/Makefile modules/speller/Makefile modules/speller/default/Makefile interfaces/Makefile interfaces/cc/Makefile scripts/Makefile examples/Makefile prog/Makefile manual/Makefile po/Makefile.in m4/Makefile modules/filter/Makefile myspell/Makefile lib5/Makefile"
+ac_config_files="$ac_config_files Makefile gen/Makefile common/Makefile lib/Makefile data/Makefile auto/Makefile modules/Makefile modules/tokenizer/Makefile modules/speller/Makefile modules/speller/default/Makefile interfaces/Makefile interfaces/cc/Makefile aspell.pc scripts/Makefile examples/Makefile prog/Makefile manual/Makefile po/Makefile.in m4/Makefile modules/filter/Makefile myspell/Makefile lib5/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -19887,6 +19887,7 @@ do
     "modules/filter/Makefile") CONFIG_FILES="$CONFIG_FILES modules/filter/Makefile" ;;
     "myspell/Makefile") CONFIG_FILES="$CONFIG_FILES myspell/Makefile" ;;
     "lib5/Makefile") CONFIG_FILES="$CONFIG_FILES lib5/Makefile" ;;
+    "aspell.pc") CONFIG_FILES="$CONFIG_FILES aspell.pc" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
   esac
diff --git a/scripts/mkconfig b/scripts/mkconfig
index 608e3f7..f15f31c 100755
--- a/scripts/mkconfig
+++ b/scripts/mkconfig
@@ -15,7 +15,7 @@ case \$1 in
   echo "$2"
   ;;
 --pkgdatadir | pkgdatadir)
-  echo "$3"
+  pkg-config aspell --variable=pkgdatadir
   ;;
 *)
   echo "usage: pspell-config version|datadir|pkgdatadir"
