diff -up sblim-sfcb-1.4.9/providerMgr.c.orig sblim-sfcb-1.4.9/providerMgr.c
--- sblim-sfcb-1.4.9/providerMgr.c.orig	2014-11-25 02:43:12.000000000 +0100
+++ sblim-sfcb-1.4.9/providerMgr.c	2015-08-24 11:57:18.945280460 +0200
@@ -355,6 +355,9 @@ lookupProviders(long type, char *classNa
 
   _SFCB_ENTER(TRACE_PROVIDERMGR, "lookupProviders");
 
+  if (className == NULL)
+    _SFCB_RETURN(NULL);
+
   id = malloc(strlen(nameSpace) + strlen(className) + 8);
   strcpy(id, nameSpace);
   strcat(id, "|");
