diff -up sblim-sfcb-1.4.9/Makefile.in.orig sblim-sfcb-1.4.9/Makefile.in
--- sblim-sfcb-1.4.9/Makefile.in.orig	2018-10-11 10:03:54.533549400 +0200
+++ sblim-sfcb-1.4.9/Makefile.in	2018-10-11 10:04:12.199570473 +0200
@@ -619,7 +619,7 @@ top_srcdir = @top_srcdir@
 # 
 # 
 AUTOMAKE_OPTIONS = no-dist-gzip dist-bzip2
-sfcbdocdir = $(datadir)/doc/sfcb-$(VERSION)
+sfcbdocdir = $(datadir)/doc/sblim-sfcb
 sfcbdatadir = $(datadir)/sfcb
 sfcbconfdir = $(sysconfdir)/sfcb
 sfcbstatedir = $(localstatedir)/lib/sfcb
@@ -908,7 +908,7 @@ dist_sfcbdata_DATA = default.reg 10_inte
 nodist_bin_SCRIPTS = sfcbrepos sfcbstage sfcbunstage sfcbuuid
 dist_bin_SCRIPTS = wbemcat xmltest
 init_SCRIPTS = sfcb
-sfcbdoc_DATA = README AUTHORS COPYING
+sfcbdoc_DATA = README AUTHORS
 sfcbconf_DATA = sfcb.cfg
 @PAM_TRUE@dist_pam_DATA = pam/sfcb
 @DOCS_TRUE@noinst_DATA = doc/html/index.html
