--- a/tests/by-util/test_seq.rs
+++ b/tests/by-util/test_seq.rs
@@ -391,97 +391,97 @@ fn test_width_negative_zero_decimal_nota
 #[test]
 fn test_width_negative_zero_scientific_notation() {
     new_ucmd!()
-        .args(&["-w", "-0e0", "1"])
+        .args(&["-w", "--", "-0e0", "1"])
         .succeeds()
         .stdout_is("-0\n01\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0e0", "1", "2"])
+        .args(&["-w", "--", "-0e0", "1", "2"])
         .succeeds()
         .stdout_is("-0\n01\n02\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0e0", "1", "2.0"])
+        .args(&["-w", "--", "-0e0", "1", "2.0"])
         .succeeds()
         .stdout_is("-0\n01\n02\n")
         .no_stderr();
 
     new_ucmd!()
-        .args(&["-w", "-0e+1", "1"])
+        .args(&["-w", "--", "-0e+1", "1"])
         .succeeds()
         .stdout_is("-00\n001\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0e+1", "1", "2"])
+        .args(&["-w", "--", "-0e+1", "1", "2"])
         .succeeds()
         .stdout_is("-00\n001\n002\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0e+1", "1", "2.0"])
+        .args(&["-w", "--", "-0e+1", "1", "2.0"])
         .succeeds()
         .stdout_is("-00\n001\n002\n")
         .no_stderr();
 
     new_ucmd!()
-        .args(&["-w", "-0.000e0", "1"])
+        .args(&["-w", "--", "-0.000e0", "1"])
         .succeeds()
         .stdout_is("-0.000\n01.000\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0.000e0", "1", "2"])
+        .args(&["-w", "--", "-0.000e0", "1", "2"])
         .succeeds()
         .stdout_is("-0.000\n01.000\n02.000\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0.000e0", "1", "2.0"])
+        .args(&["-w", "--", "-0.000e0", "1", "2.0"])
         .succeeds()
         .stdout_is("-0.000\n01.000\n02.000\n")
         .no_stderr();
 
     new_ucmd!()
-        .args(&["-w", "-0.000e-2", "1"])
+        .args(&["-w", "--", "-0.000e-2", "1"])
         .succeeds()
         .stdout_is("-0.00000\n01.00000\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0.000e-2", "1", "2"])
+        .args(&["-w", "--", "-0.000e-2", "1", "2"])
         .succeeds()
         .stdout_is("-0.00000\n01.00000\n02.00000\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0.000e-2", "1", "2.0"])
+        .args(&["-w", "--", "-0.000e-2", "1", "2.0"])
         .succeeds()
         .stdout_is("-0.00000\n01.00000\n02.00000\n")
         .no_stderr();
 
     new_ucmd!()
-        .args(&["-w", "-0.000e5", "1"])
+        .args(&["-w", "--", "-0.000e5", "1"])
         .succeeds()
         .stdout_is("-000000\n0000001\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0.000e5", "1", "2"])
+        .args(&["-w", "--", "-0.000e5", "1", "2"])
         .succeeds()
         .stdout_is("-000000\n0000001\n0000002\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0.000e5", "1", "2.0"])
+        .args(&["-w", "--", "-0.000e5", "1", "2.0"])
         .succeeds()
         .stdout_is("-000000\n0000001\n0000002\n")
         .no_stderr();
 
     new_ucmd!()
-        .args(&["-w", "-0.000e5", "1"])
+        .args(&["-w", "--", "-0.000e5", "1"])
         .succeeds()
         .stdout_is("-000000\n0000001\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0.000e5", "1", "2"])
+        .args(&["-w", "--", "-0.000e5", "1", "2"])
         .succeeds()
         .stdout_is("-000000\n0000001\n0000002\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-0.000e5", "1", "2.0"])
+        .args(&["-w", "--", "-0.000e5", "1", "2.0"])
         .succeeds()
         .stdout_is("-000000\n0000001\n0000002\n")
         .no_stderr();
@@ -525,7 +525,7 @@ fn test_width_decimal_scientific_notatio
 #[test]
 fn test_width_negative_decimal_notation() {
     new_ucmd!()
-        .args(&["-w", "-.1", ".1", ".11"])
+        .args(&["-w", "--", "-.1", ".1", ".11"])
         .succeeds()
         .stdout_is("-0.1\n00.0\n00.1\n")
         .no_stderr();
@@ -534,22 +534,22 @@ fn test_width_negative_decimal_notation(
 #[test]
 fn test_width_negative_scientific_notation() {
     new_ucmd!()
-        .args(&["-w", "-1e-3", "1"])
+        .args(&["-w", "--", "-1e-3", "1"])
         .succeeds()
         .stdout_is("-0.001\n00.999\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-1.e-3", "1"])
+        .args(&["-w", "--", "-1.e-3", "1"])
         .succeeds()
         .stdout_is("-0.001\n00.999\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-1.0e-4", "1"])
+        .args(&["-w", "--", "-1.0e-4", "1"])
         .succeeds()
         .stdout_is("-0.00010\n00.99990\n")
         .no_stderr();
     new_ucmd!()
-        .args(&["-w", "-.1e2", "10", "100"])
+        .args(&["-w", "--", "-.1e2", "10", "100"])
         .succeeds()
         .stdout_is(
             "-010
