--- boost_1_66_0/tools/build/src/engine/build.jam~	2018-02-07 21:36:14.552201421 +0000
+++ boost_1_66_0/tools/build/src/engine/build.jam	2018-02-07 21:36:29.014173266 +0000
@@ -4,7 +4,7 @@
 #~ http://www.boost.org/LICENSE_1_0.txt)
 
 # Clean env vars of any "extra" empty values.
-for local v in ARGV CC CFLAGS LIBS
+for local v in ARGV CC CFLAGS LIBS RPM_OPT_FLAGS RPM_LD_FLAGS
 {
     local values ;
     for local x in $($(v))
@@ -215,12 +215,12 @@
     : -L$(--python-lib[1]) -l$(--python-lib[2]) ;
 ## GCC 2.x, 3.x, 4.x
 toolset gcc gcc : "-o " : -D
-    : -pedantic -fno-strict-aliasing
+    : -pedantic -fno-strict-aliasing $(RPM_OPT_FLAGS)
     [ opt --release : [ opt --symbols : -g : -s ] -O3 ]
     [ opt --debug : -g -O0 -fno-inline ]
     [ opt --profile : -O3 -g -pg ]
     -I$(--python-include) -I$(--extra-include) -Wno-long-long
-    : -L$(--python-lib[1]) -l$(--python-lib[2]) ;
+    : -L$(--python-lib[1]) -l$(--python-lib[2]) $(RPM_LD_FLAGS) ;
 ## GCC 2.x, 3.x on CYGWIN but without cygwin1.dll
 toolset gcc-nocygwin gcc : "-o " : -D
     : -s -O3 -mno-cygwin
