diff -up xpdf-4.06/xpdf/pdfimages.cc.permissions xpdf-4.06/xpdf/pdfimages.cc
--- xpdf-4.06/xpdf/pdfimages.cc.permissions	2025-11-13 17:08:30.000000000 -0500
+++ xpdf-4.06/xpdf/pdfimages.cc	2025-11-18 15:27:00.093784988 -0500
@@ -4,6 +4,8 @@
 //
 // Copyright 1998-2003 Glyph & Cog, LLC
 //
+// Modified for Debian by Hamish Moffatt, 22 May 2002.
+//
 //========================================================================
 
 #include <aconf.h>
@@ -153,12 +155,14 @@ int main(int argc, char *argv[]) {
   }
 
   // check for copy permission
+#ifdef ENFORCE_PERMISSIONS
   if (!listOnly && !doc->okToCopy()) {
     error(errNotAllowed, -1,
 	  "Copying of images from this document is not allowed.");
     exitCode = 3;
     goto err1;
   }
+#endif
 
   // get page range
   if (firstPage < 1)
diff -up xpdf-4.06/xpdf/pdftohtml.cc.permissions xpdf-4.06/xpdf/pdftohtml.cc
--- xpdf-4.06/xpdf/pdftohtml.cc.permissions	2025-11-13 17:08:30.000000000 -0500
+++ xpdf-4.06/xpdf/pdftohtml.cc	2025-11-18 15:25:43.149122766 -0500
@@ -183,6 +183,7 @@ int main(int argc, char *argv[]) {
     goto err1;
   }
 
+#ifdef ENFORCE_PERMISSIONS
   // check for copy permission
   if (!doc->okToCopy()) {
     error(errNotAllowed, -1,
@@ -190,6 +191,7 @@ int main(int argc, char *argv[]) {
     exitCode = 3;
     goto err1;
   }
+#endif
 
   // get page range
   if (firstPage < 1) {
diff -up xpdf-4.06/xpdf/pdftops.cc.permissions xpdf-4.06/xpdf/pdftops.cc
--- xpdf-4.06/xpdf/pdftops.cc.permissions	2025-11-13 17:08:30.000000000 -0500
+++ xpdf-4.06/xpdf/pdftops.cc	2025-11-18 15:25:43.149260566 -0500
@@ -4,6 +4,8 @@
 //
 // Copyright 1996-2003 Glyph & Cog, LLC
 //
+// Modified for Debian by Hamish Moffatt, 22 May 2002.
+//
 //========================================================================
 
 #include <aconf.h>
@@ -337,12 +339,14 @@ int main(int argc, char *argv[]) {
     goto err1;
   }
 
+#ifdef ENFORCE_PERMISSIONS
   // check for print permission
   if (!doc->okToPrint()) {
     error(errNotAllowed, -1, "Printing this document is not allowed.");
     exitCode = 3;
     goto err1;
   }
+#endif
 
   // construct PostScript file name
   if (argc == 3) {
diff -up xpdf-4.06/xpdf/pdftotext.cc.permissions xpdf-4.06/xpdf/pdftotext.cc
--- xpdf-4.06/xpdf/pdftotext.cc.permissions	2025-11-13 17:08:30.000000000 -0500
+++ xpdf-4.06/xpdf/pdftotext.cc	2025-11-18 15:25:43.149380451 -0500
@@ -4,6 +4,8 @@
 //
 // Copyright 1997-2013 Glyph & Cog, LLC
 //
+// Modified for Debian by Hamish Moffatt, 22 May 2002.
+//
 //========================================================================
 
 #include <aconf.h>
@@ -248,6 +250,7 @@ int main(int argc, char *argv[]) {
     goto err2;
   }
 
+#ifdef ENFORCE_PERMISSIONS
   // check for copy permission
   if (!doc->okToCopy()) {
     error(errNotAllowed, -1,
@@ -255,6 +258,7 @@ int main(int argc, char *argv[]) {
     exitCode = 3;
     goto err2;
   }
+#endif
 
   // construct text file name
   if (argc == 3) {
diff -up xpdf-4.06/xpdf-qt/QtPDFCore.cc.permissions xpdf-4.06/xpdf-qt/QtPDFCore.cc
--- xpdf-4.06/xpdf-qt/QtPDFCore.cc.permissions	2025-11-13 17:08:30.000000000 -0500
+++ xpdf-4.06/xpdf-qt/QtPDFCore.cc	2025-11-18 15:25:43.149536859 -0500
@@ -494,9 +494,11 @@ QString QtPDFCore::getSelectedTextQStrin
   QString qs;
   int i;
 
+#ifdef ENFORCE_PERMISSIONS
   if (!doc->okToCopy()) {
     return "";
   }
+#endif
   if (!(s = getSelectedText())) {
     return "";
   }
diff -up xpdf-4.06/xpdf-qt/XpdfWidget.cc.permissions xpdf-4.06/xpdf-qt/XpdfWidget.cc
--- xpdf-4.06/xpdf-qt/XpdfWidget.cc.permissions	2025-11-13 17:08:30.000000000 -0500
+++ xpdf-4.06/xpdf-qt/XpdfWidget.cc	2025-11-18 15:25:43.149737940 -0500
@@ -1526,7 +1526,11 @@ bool XpdfWidget::okToExtractText() const
     if (!core->getDoc()) {
       return false;
     }
+#ifdef ENFORCE_PERMISSIONS
     return (bool)core->getDoc()->okToCopy();
+#else
+    return true;
+#endif
   } catch (GMemException e) {
     return false;
   }
