--- ptdump-1.0.3/ptdump.mk.orig
+++ ptdump-1.0.3/ptdump.mk
@@ -43,7 +43,7 @@ ptdump.so: $(TARGET_CFILES) $(INCDIR)/de
 ifeq ($(ARCH),UNSUPPORTED)
 	@echo "ptdump: architecture not supported"
 else
-	gcc $(RPM_OPT_FLAGS) $(CFLAGS) $(TARGET_CFLAGS) $(COMMON_CFLAGS) -nostartfiles -shared -rdynamic -o $@ $(TARGET_CFILES)
+	gcc $(RPM_OPT_FLAGS) $(CFLAGS) $(TARGET_CFLAGS) $(COMMON_CFLAGS) -nostartfiles -shared -rdynamic -o $@ $(TARGET_CFILES) -Wl,-z,now
 endif
 
 debug: COMMON_CFLAGS+=-DDEBUG
