--- a52dec-0.7.4/configure~	2002-07-28 06:50:42.000000000 +0300
+++ a52dec-0.7.4/configure	2006-02-16 23:03:07.000000000 +0200
@@ -5839,7 +5839,7 @@
 shlibpath_overrides_runpath=unknown
 version_type=none
 dynamic_linker="$host_os ld.so"
-sys_lib_dlsearch_path_spec="/lib /usr/lib"
+sys_lib_dlsearch_path_spec="/lib64 /usr/lib64 /lib /usr/lib"
 sys_lib_search_path_spec="/lib /usr/lib /usr/local/lib"
 
 case $host_os in
--- a52dec-0.7.4/aclocal.m4~	2002-07-28 06:50:38.000000000 +0300
+++ a52dec-0.7.4/aclocal.m4	2006-02-16 23:02:38.000000000 +0200
@@ -2141,7 +2141,7 @@
 shlibpath_overrides_runpath=unknown
 version_type=none
 dynamic_linker="$host_os ld.so"
-sys_lib_dlsearch_path_spec="/lib /usr/lib"
+sys_lib_dlsearch_path_spec="/lib64 /usr/lib64 /lib /usr/lib"
 sys_lib_search_path_spec="/lib /usr/lib /usr/local/lib"
 
 case $host_os in
