diff --git a/plugins/sudoers/ldap_conf.c b/plugins/sudoers/ldap_conf.c
index 334f9fa69..5f2a9cb05 100644
--- a/plugins/sudoers/ldap_conf.c
+++ b/plugins/sudoers/ldap_conf.c
@@ -596,7 +596,7 @@ sudo_ldap_read_config(void)
            debug_return_bool(false);
        }
     }
-    if (!STAILQ_EMPTY(&ldap_conf.netgroup_base))
+    if (STAILQ_EMPTY(&ldap_conf.netgroup_base))
        ldap_conf.netgroup_query = false;

     DPRINTF1("LDAP Config Summary");
