--- raw-cpuid-11.6.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ raw-cpuid-11.6.0/Cargo.toml	2025-10-25T22:05:26.918646+00:00
@@ -39,16 +39,6 @@
 repository = "https://github.com/gz/rust-cpuid"
 
 [features]
-cli = [
-    "display",
-    "clap",
-]
-display = [
-    "std",
-    "termimad",
-    "serde_json",
-    "serialize",
-]
 serialize = [
     "serde",
     "serde_derive",
@@ -58,11 +48,6 @@
 [lib]
 name = "raw_cpuid"
 path = "src/lib.rs"
-
-[[bin]]
-name = "cpuid"
-path = "src/bin/cpuid.rs"
-required-features = ["cli"]
 
 [[example]]
 name = "cache"
@@ -91,11 +76,6 @@
 [dependencies.bitflags]
 version = "2.0"
 
-[dependencies.clap]
-version = "4.2"
-features = ["derive"]
-optional = true
-
 [dependencies.serde]
 version = "1.0"
 optional = true
@@ -107,10 +87,6 @@
 
 [dependencies.serde_json]
 version = "1.0"
-optional = true
-
-[dependencies.termimad]
-version = "0.25"
 optional = true
 
 [dev-dependencies.core_affinity]
