--- ff-0.14.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ ff-0.14.0/Cargo.toml	2026-08-04T08:07:53.220220+00:00
@@ -45,26 +45,16 @@
 alloc = []
 bits = [
     "bitvec",
-    "ff_derive?/bits",
 ]
 default = [
     "bits",
     "std",
-]
-derive = [
-    "byteorder",
-    "ff_derive",
 ]
 std = ["alloc"]
 
 [lib]
 name = "ff"
 path = "src/lib.rs"
-
-[[test]]
-name = "derive"
-path = "tests/derive.rs"
-required-features = ["derive"]
 
 [dependencies.bitvec]
 version = "1"
@@ -75,10 +65,6 @@
 version = "1"
 optional = true
 default-features = false
-
-[dependencies.ff_derive]
-version = "0.14"
-optional = true
 
 [dependencies.rand_core]
 version = "0.10"
