diff -up openobex-1.7.1-Source/udev/obex-check-device.c.strtoul openobex-1.7.1-Source/udev/obex-check-device.c
--- openobex-1.7.1-Source/udev/obex-check-device.c.strtoul	2016-03-18 09:16:49.851614909 +0100
+++ openobex-1.7.1-Source/udev/obex-check-device.c	2016-03-18 09:24:25.100755839 +0100
@@ -62,8 +62,8 @@ int main (int argc, char **argv)
 	unsigned long vendor;
 	unsigned long product;
 
-	if (argc < 2)
-		return 0;
+	if (argc < 3)
+		return EXIT_FAILURE;
 
 	vendor = strtoul(argv[1], NULL, 16);
 	product = strtoul(argv[2], NULL, 16);
