--- capnp-0.25.6/Cargo.toml	2006-07-24T01:21:28+00:00
+++ capnp-0.25.6/Cargo.toml	2026-08-20T10:32:39.592921+00:00
@@ -33,13 +33,13 @@
 repository = "https://github.com/capnproto/capnproto-rust"
 
 [features]
-alloc = ["embedded-io?/alloc"]
+alloc = []
 default = [
     "std",
     "alloc",
 ]
 rpc_try = []
-std = ["embedded-io?/std"]
+std = []
 sync_reader = []
 unaligned = []
 
@@ -107,17 +107,12 @@
 name = "zero_size_alloc"
 path = "tests/zero_size_alloc.rs"
 
-[dependencies.embedded-io]
-version = "0.7.1"
-optional = true
-default-features = false
-
 [dependencies.quickcheck]
 version = "1"
 optional = true
 
 [dev-dependencies.quickcheck]
-version = "=1.0"
+version = "1"
 
 [lints.clippy]
 missing_safety_doc = "allow"
