diff --git a/flint/subcommands.cpp b/flint/subcommands.cpp
index 420b82f..0a20bcd 100644
--- a/flint/subcommands.cpp
+++ b/flint/subcommands.cpp
@@ -4484,7 +4484,7 @@ FlintStatus SgSubCommand::sgFs2()
     return FLINT_SUCCESS;
 }
 
-#define FW_RESET_MSG "To load new configuration run mlxfwreset or reboot machine"
+#define FW_RESET_MSG "To load new configuration run mstfwreset or reboot machine"
 
 FlintStatus SgSubCommand::sgFs3()
 {
diff --git a/mlxfwops/lib/flint_base.h b/mlxfwops/lib/flint_base.h
index f3edc43..a85e43d 100644
--- a/mlxfwops/lib/flint_base.h
+++ b/mlxfwops/lib/flint_base.h
@@ -364,7 +364,7 @@ struct GPH {
 
 #define REBOOT_REQUIRED_STR "To load new FW run reboot machine."
 #ifndef MST_UL
-#define REBOOT_OR_FWRESET_REQUIRED_STR "To load new FW run mlxfwreset or reboot machine."
+#define REBOOT_OR_FWRESET_REQUIRED_STR "To load new FW run mstfwreset or reboot machine."
 #else
 #define REBOOT_OR_FWRESET_REQUIRED_STR "To load new FW run mstfwreset or reboot machine."
 #endif
