diff -up pegasus/mak/program-unix.mak.orig pegasus/mak/program-unix.mak
--- pegasus/mak/program-unix.mak.orig	2012-10-08 12:19:55.965024675 +0200
+++ pegasus/mak/program-unix.mak	2012-10-08 12:21:02.378355115 +0200
@@ -28,19 +28,10 @@
 #//////////////////////////////////////////////////////////////////////////
 INCLUDES = -I$(ROOT)/src $(EXTRA_INCLUDES)
 
-ifndef LINK_RPATH
-    LINK_RPATH = -Xlinker -rpath
-endif
-
 LINK_DEST_LIB = -Xlinker $(PEGASUS_DEST_LIB_DIR)
 
 LINK_LIB_DIR = -Xlinker $(LIB_DIR)
 
-ifndef LINK_RPATH_LINK
-    LINK_RPATH_LINK = -Xlinker -rpath-link
-endif
-
-
 TMP_OBJECTS = $(foreach i,$(SOURCES),$(OBJ_DIR)/$i)
 
 S_OBJECTS = $(TMP_OBJECTS:.s=.o)
@@ -52,15 +43,9 @@ FULL_PROGRAM=$(BIN_DIR)/$(PROGRAM)$(EXE)
 EXE_OUTPUT = $(EXE_OUT) $(FULL_PROGRAM)
 
 ifdef PEGASUS_PLATFORM_LINUX_GENERIC_GNU
-    ifdef PEGASUS_USE_RELEASE_DIRS
-        EXTRA_LINK_FLAGS += $(LINK_RPATH) $(LINK_DEST_LIB) $(LINK_RPATH_LINK) $(LINK_LIB_DIR)
-    else
-        EXTRA_LINK_FLAGS += $(LINK_RPATH) $(LINK_LIB_DIR)
-    endif
-
     ifeq ($(HAS_ICU_DEPENDENCY),true)
         ifdef ICU_INSTALL
-            EXTRA_LINK_FLAGS += $(LINK_RPATH) -Xlinker ${ICU_INSTALL}/lib
+            EXTRA_LINK_FLAGS += -Xlinker ${ICU_INSTALL}/lib
         endif
     endif
 endif
diff -up pegasus/mak/shared-library-unix.mak.orig pegasus/mak/shared-library-unix.mak
--- pegasus/mak/shared-library-unix.mak.orig	2012-10-08 12:21:15.279419276 +0200
+++ pegasus/mak/shared-library-unix.mak	2012-10-08 12:22:32.715804612 +0200
@@ -98,15 +98,14 @@ ifeq ($(COMPILER),gnu)
   ifneq ($(OS),darwin)
    ifdef PEGASUS_USE_RELEASE_DIRS
       LINK_COMMAND = $(CXX) -shared $(LINK_MACHINE_OPTIONS)
-      LINK_ARGUMENTS = -Wl,-hlib$(LIBRARY)$(LIB_SUFFIX)  -Xlinker -rpath \
-          -Xlinker $(PEGASUS_DEST_LIB_DIR) $(EXTRA_LINK_ARGUMENTS)
+      LINK_ARGUMENTS = -Wl,-hlib$(LIBRARY)$(LIB_SUFFIX)  $(EXTRA_LINK_ARGUMENTS)
    else
       LINK_COMMAND = $(CXX) -shared $(LINK_MACHINE_OPTIONS)
-      LINK_ARGUMENTS = -Wl,-hlib$(LIBRARY)$(LIB_SUFFIX)  -Xlinker -rpath -Xlinker $(LIB_DIR) $(EXTRA_LINK_ARGUMENTS)
+      LINK_ARGUMENTS = -Wl,-hlib$(LIBRARY)$(LIB_SUFFIX)  $(EXTRA_LINK_ARGUMENTS)
    endif
   else
     LINK_COMMAND = $(CXX) -dynamiclib
-    LINK_ARGUMENTS = --helplib$(LIBRARY)$(LIB_SUFFIX) -ldl
+    LINK_ARGUMENTS = --helplib$(LIBRARY)$(LIB_SUFFIX) -ldl $(EXTRA_LINK_ARGUMENTS)
   endif
   ifeq ($(PEGASUS_PLATFORM), SOLARIS_SPARC_GNU)
     LINK_ARGUMENTS = -Wl,-hlib$(LIBRARY)$(LIB_SUFFIX) -Xlinker -L$(LIB_DIR) $(EXTRA_LINK_ARGUMENTS)
