diff --color -Nur iperf-3.18.orig/src/iperf_api.c iperf-3.18/src/iperf_api.c
--- iperf-3.18.orig/src/iperf_api.c	2024-12-13 16:09:39.000000000 -0800
+++ iperf-3.18/src/iperf_api.c	2024-12-21 12:36:16.493209013 -0800
@@ -3461,6 +3461,8 @@
 
 		    temp.rttvar = get_rttvar(&temp);
 		    temp.pmtu = get_pmtu(&temp);
+                    temp.reorder = get_reorder(&temp);
+                    rp->stream_reorder = temp.reorder;
 		}
 	    }
 	} else {
@@ -3951,7 +3953,8 @@
                     if (test->sender_has_retransmits || test->protocol->id == Psctp) {
                         /* Sender summary, TCP and SCTP with retransmits. */
                         if (test->json_output)
-                            cJSON_AddItemToObject(json_summary_stream, report_sender, iperf_json_printf("socket: %d  start: %f  end: %f  seconds: %f  bytes: %d  bits_per_second: %f  retransmits: %d  max_snd_cwnd:  %d  max_snd_wnd:  %d  max_rtt:  %d  min_rtt:  %d  mean_rtt:  %d sender: %b", (int64_t) sp->socket, (double) start_time, (double) sender_time, (double) sender_time, (int64_t) bytes_sent, bandwidth * 8, (int64_t) sp->result->stream_retrans, (int64_t) sp->result->stream_max_snd_cwnd, (int64_t) sp->result->stream_max_snd_wnd, (int64_t) sp->result->stream_max_rtt, (int64_t) sp->result->stream_min_rtt, (int64_t) ((sp->result->stream_count_rtt == 0) ? 0 : sp->result->stream_sum_rtt / sp->result->stream_count_rtt), stream_must_be_sender));
+                            cJSON_AddItemToObject(json_summary_stream, report_sender, iperf_json_printf("socket: %d  start: %f  end: %f  seconds: %f  bytes: %d  bits_per_second: %f  retransmits: %d  reorder:  %d  max_snd_cwnd:  %d  max_snd_wnd:  %d  max_rtt:  %d  min_rtt:  %d  mean_rtt:  %d sender: %b", (int64_t) sp->socket, (double) start_time, (double) sender_time, (double) sender_time, (int64_t) bytes_sent, bandwidth * 8, (int64_t) sp->result->stream_retrans, (int64_t) sp->result->stream_reorder, (int64_t) sp->result->stream_max_snd_cwnd, (int64_t) sp->result->stream_max_snd_wnd, (int64_t) sp->result->stream_max_rtt, (int64_t) sp->result->stream_min_rtt, (int64_t) ((sp->result->stream_count_rtt == 0) ? 0 : sp->result->stream_sum_rtt / sp->result->stream_count_rtt), stream_must_be_sender));
+
                         else
                             if (test->role == 's' && !sp->sender) {
                                 if (test->verbose)
@@ -4403,7 +4406,7 @@
 	if ((test->sender_has_retransmits == 1 || test->protocol->id == Psctp) && sp->sender) {
 	    /* Interval, TCP with retransmits. */
 	    if (test->json_output)
-		cJSON_AddItemToArray(json_interval_streams, iperf_json_printf("socket: %d  start: %f  end: %f  seconds: %f  bytes: %d  bits_per_second: %f  retransmits: %d  snd_cwnd:  %d  snd_wnd:  %d  rtt:  %d  rttvar: %d  pmtu: %d  omitted: %b sender: %b", (int64_t) sp->socket, (double) st, (double) et, (double) irp->interval_duration, (int64_t) irp->bytes_transferred, bandwidth * 8, (int64_t) irp->interval_retrans, (int64_t) irp->snd_cwnd, (int64_t) irp->snd_wnd, (int64_t) irp->rtt, (int64_t) irp->rttvar, (int64_t) irp->pmtu, irp->omitted, sp->sender));
+                cJSON_AddItemToArray(json_interval_streams, iperf_json_printf("socket: %d  start: %f  end: %f  seconds: %f  bytes: %d  bits_per_second: %f  retransmits: %d  snd_cwnd:  %d  snd_wnd:  %d  rtt:  %d  rttvar: %d  pmtu: %d  reorder: %d  omitted: %b sender: %b", (int64_t) sp->socket, (double) st, (double) et, (double) irp->interval_duration, (int64_t) irp->bytes_transferred, bandwidth * 8, (int64_t) irp->interval_retrans, (int64_t) irp->snd_cwnd, (int64_t) irp->snd_wnd, (int64_t) irp->rtt, (int64_t) irp->rttvar, (int64_t) irp->pmtu, (int64_t) irp->reorder, irp->omitted, sp->sender));
 	    else {
 		unit_snprintf(cbuf, UNIT_LEN, irp->snd_cwnd, 'A');
 		iperf_printf(test, report_bw_retrans_cwnd_format, sp->socket, mbuf, st, et, ubuf, nbuf, irp->interval_retrans, cbuf, irp->omitted?report_omitted:"");
diff --color -Nur iperf-3.18.orig/src/iperf_api.h iperf-3.18/src/iperf_api.h
--- iperf-3.18.orig/src/iperf_api.h	2024-12-13 16:09:39.000000000 -0800
+++ iperf-3.18/src/iperf_api.h	2024-12-21 12:37:03.518082421 -0800
@@ -318,6 +318,7 @@
 long get_rtt(struct iperf_interval_results *irp);
 long get_rttvar(struct iperf_interval_results *irp);
 long get_pmtu(struct iperf_interval_results *irp);
+long get_reorder(struct iperf_interval_results *irp);
 void print_tcpinfo(struct iperf_test *test);
 void build_tcpinfo_message(struct iperf_interval_results *r, char *message);
 
diff --color -Nur iperf-3.18.orig/src/iperf.h iperf-3.18/src/iperf.h
--- iperf-3.18.orig/src/iperf.h	2024-12-13 16:09:39.000000000 -0800
+++ iperf-3.18/src/iperf.h	2024-12-21 12:38:03.726920296 -0800
@@ -126,6 +126,7 @@
     long rtt;
     long rttvar;
     long pmtu;
+    long reorder;
 };
 
 struct iperf_stream_result
@@ -137,6 +138,7 @@
     atomic_iperf_size_t bytes_sent_omit;
     long stream_prev_total_retrans;
     long stream_retrans;
+    long stream_reorder;
     long stream_max_rtt;
     long stream_min_rtt;
     long stream_sum_rtt;
diff --color -Nur iperf-3.18.orig/src/tcp_info.c iperf-3.18/src/tcp_info.c
--- iperf-3.18.orig/src/tcp_info.c	2024-12-13 16:09:39.000000000 -0800
+++ iperf-3.18/src/tcp_info.c	2024-12-21 12:39:04.701756102 -0800
@@ -218,6 +218,20 @@
 }
 
 /*************************************************************/
+/*
+ * Return number of reordering events seen.
+ */
+long
+get_reorder(struct iperf_interval_results *irp)
+{
+#if defined(linux) && defined(TCP_REPAIR_ON)
+    return irp->tcpInfo.tcpi_reord_seen;
+#else
+    return -1;
+#endif
+}
+
+/*************************************************************/
 void
 build_tcpinfo_message(struct iperf_interval_results *r, char *message)
 {
