From ccfd157db8ecbd53db2edf96ad077e5b59f67726 Mon Sep 17 00:00:00 2001
From: Rachel Menge <rachelmenge@microsoft.com>
Date: Tue, 15 Mar 2022 00:27:01 -0700
Subject: [PATCH] Remove -pg option

---
 src/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 11d3e17..4058cff 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -58,9 +58,9 @@ if ENABLE_PROFILING
 iperf3_profile_SOURCES  = main.c \
                           $(libiperf_la_SOURCES)
 
-iperf3_profile_CFLAGS   = -pg -g
+iperf3_profile_CFLAGS   = -g
 iperf3_profile_LDADD    = libiperf.la
-iperf3_profile_LDFLAGS  = -pg -g
+iperf3_profile_LDFLAGS  = -g
 endif
 
 # Specify the sources and various flags for the test cases
-- 
2.17.1

