--- ledmon-0.92/src/Makefile	2019-04-12 11:49:04.000000000 +0200
+++ ledmon-0.92-new/src/Makefile	2019-04-15 10:37:25.529206895 +0200
@@ -64,15 +64,10 @@ TEST_CONFIG_OBJECTS=\
 	$(OUTDIR)/list.o	\
 	$(OUTDIR)/utils.o
 
-CXFLAGS+=-std=gnu99
-CWFLAGS=-Wall
-CFLAGS=$(CXFLAGS) $(CWFLAGS)
-
 DEFFLAGS=-D_DEBUG -D_GNU_SOURCE -D_DEFAULT_SOURCE -DDMALLOC_DISABLE -DBUILD_LABEL=\""$(BUILD_LABEL)"\"
 CPPFLAGS=$(DEFFLAGS)
 ALL_CPPFLAGS=$(CPPFLAGS) -I../config
 
-LDFLAGS=$(CXFLAGS)
 LDLIBS=-lsgutils2 -ludev -lrt
 
 SOURCES:=$(shell find -name "*.[cC]" -print)
