From c7caa6027c92b28aa11b8da74d56357e12f56d67 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Kope=C4=8Dek?= <dkopecek@redhat.com>
Date: Wed, 20 Feb 2019 12:00:56 +0100
Subject: [PATCH] Use LDADD for adding curl library to the linker command

---
 Makefile.am  | 2 +-
 configure.ac | 5 +++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 4b05d7a..1541d56 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -55,7 +55,7 @@ if USE_CURL
 aide_SOURCES += include/fopen.h src/fopen.c
 endif
 
-aide_LDADD = -lm @PCRELIB@ @CRYPTLIB@ @ACLLIB@ @SELINUXLIB@ @AUDITLIB@ @ATTRLIB@ @E2FSATTRSLIB@ @ELFLIB@
+aide_LDADD = -lm @PCRELIB@ @CRYPTLIB@ @ACLLIB@ @SELINUXLIB@ @AUDITLIB@ @ATTRLIB@ @E2FSATTRSLIB@ @ELFLIB@ @CURLLIB@
 AM_CFLAGS = @AIDE_DEFS@ -W -Wall -g
 AM_CPPFLAGS = -I$(top_srcdir) \
 			  -I$(top_srcdir)/include \
diff --git a/configure.ac b/configure.ac
index 3598ebe..0418c59 100644
--- a/configure.ac
+++ b/configure.ac
@@ -702,24 +702,25 @@ if test x$with_zlib = xyes; then
   compoptionstring="${compoptionstring}WITH_ZLIB\\n"
 fi
 
+CURLLIB=
 if test x$with_curl = xyes; then
 	AC_PATH_PROG(curlconfig, "curl-config")
 	if test "_$curlconfig" != _ ; then
 		CURL_CFLAGS=`$curlconfig --cflags`
-		CURL_LIBS=`$curlconfig --libs`
+		CURLLIB=`$curlconfig --libs`
 	else
 		AC_MSG_ERROR([You don't have curl properly installed. Install it or try --without-curl.])
 	fi
   AC_CHECK_HEADERS(curl/curl.h,,
  	[AC_MSG_ERROR([You don't have curl properly installed. Install it or try --without-curl.])])
   CFLAGS="$CFLAGS $CURL_CFLAGS"
-  LDFLAGS="$LDFLAGS $CURL_LIBS"
   AC_CHECK_LIB(curl,curl_easy_init,havecurl=yes,
  	[AC_MSG_ERROR([You don't have curl properly installed. Install it or try --without-curl.])]
 )
   AC_DEFINE(WITH_CURL,1,[use curl])
 	compoptionstring="${compoptionstring}WITH_CURL\\n"
 fi
+AC_SUBST(CURLLIB)
 AM_CONDITIONAL(USE_CURL, test x$havecurl = xyes)
 
 AC_ARG_WITH(mhash,
-- 
2.20.1

