--- ./src/do_md.c	2018-03-19 05:10:19.994957024 -0400
+++ ./src/do_md.c	2018-03-19 05:19:05.829957024 -0400
@@ -135,8 +135,13 @@
                         continue;
 
                 while (!bingo && (data = elf_getdata (scn, data)) != NULL) {
-                        int maxndx = data->d_size / shdr.sh_entsize;
+                        int maxndx;
                         int ndx;
+			
+                        if (shdr.sh_entsize != 0)
+                                maxndx = data->d_size / shdr.sh_entsize;
+                        else
+                                continue;
 
                         for (ndx = 0; ndx < maxndx; ++ndx) {
                                 (void) gelf_getdyn (data, ndx, &dyn);
