diff -Naurk.orig rx-1.5/configure.in.orig rx-1.5/configure.in
--- rx-1.5/configure.in.orig	Fri Nov 22 08:39:12 1996
+++ rx-1.5/configure.in	Wed Apr 24 10:41:41 2002
@@ -1,4 +1,5 @@
 AC_INIT(Makefile.in)
+AC_PROG_LIBTOOL
 
 all_subdirs=`cat $srcdir/*/PLUGIN/REQ $srcdir/*/PLUGIN/OPT /dev/null | tsort | xargs echo`
 req_subdirs=`cat $srcdir/*/PLUGIN/REQ /dev/null | tsort | xargs echo`
@@ -47,4 +48,5 @@
 
 
 AC_SUBST(existingdirs)
+AC_SUBST(LIBTOOL_DEPS)
 AC_OUTPUT(Makefile)
diff -Naurk.orig rx-1.5/rx/Makefile.in.orig rx-1.5/rx/Makefile.in
--- rx-1.5/rx/Makefile.in.orig	Fri Nov 22 15:47:49 1996
+++ rx-1.5/rx/Makefile.in	Wed Apr 24 10:48:19 2002
@@ -40,6 +40,10 @@
 AR_FLAGS = rc
 RANLIB = @RANLIB@
 opt_objs=@opt_objs@
+LIBTOOL = libtool
+DESTDIR =
+LIBTOOL_DEPS = @LIBTOOL_DEPS@
+
 
 source= hashrexp.c rx.c rxanal.c rxbasic.c rxbitset.c rxcset.c rxdbug.c rxgnucomp.c rxhash.c rxnfa.c rxnode.c rxposix.c rxsimp.c rxspencer.c rxstr.c rxsuper.c rxunfa.c rgx.c
 
@@ -50,10 +54,11 @@
 	  ../doc/rx.texi ../doc/rx.info ../doc/texinfo.tex
 distfiles=$(source) $(headers) $(ancillery)
 libobjs= hashrexp.o rx.o rxanal.o rxbasic.o rxbitset.o rxcset.o rxdbug.o rxgnucomp.o rxhash.o rxnfa.o rxnode.o rxposix.o rxsimp.o rxspencer.o rxstr.o rxsuper.o rxunfa.o
+liblobjs= hashrexp.lo rx.lo rxanal.lo rxbasic.lo rxbitset.lo rxcset.lo rxdbug.lo rxgnucomp.lo rxhash.lo rxnfa.lo rxnode.lo rxposix.lo rxsimp.lo rxspencer.lo rxstr.lo rxsuper.lo rxunfa.lo
 gen_c_files=rgx.x
 
 .SUFFIXES:
-.SUFFIXES: .o .c .h .ps .dvi .info .texinfo .scm .cd .x
+.SUFFIXES: .lo .o .c .h .ps .dvi .info .texinfo .scm .cd .x
 
 .c.x:
 	$(CC) $(ALL_CFLAGS) -DSCM_MAGIC_SNARFER -E $< | grep "^%%%" | sed -e "s/^%%%//" > $@ ; \
@@ -61,22 +66,27 @@
 .c.o:
 	$(CC) -c $(ALL_CFLAGS) $(DEFS) -I$(srcdir) $<
 
-all: librx.a
+.c.lo:
+	$(LIBTOOL) $(CC) -c $(ALL_CFLAGS) $(DEFS) -I$(srcdir) $< -o $@
+
+all: librx.la
 
 install: all
-	test -d $(libdir) || mkdir $(libdir)
-	test -d $(includedir) || mkdir $(includedir)
-	$(INSTALL) librx.a $(libdir)/librx.a
-	$(RANLIB) $(libdir)/librx.a
-	$(INSTALL) $(srcdir)/inst-rxposix.h $(includedir)/rxposix.h
+	$(INSTALL) $(srcdir)/inst-rxposix.h $(DESTDIR)$(includedir)/rxposix.h
+	$(LIBTOOL) $(INSTALL) -c librx.la $(DESTDIR)$(libdir)
 
 
 uninstall:
 	-rm -f $(libdir)/librx.a
+	-rm -f $(libdir)/librx.la
+	-rm -f $(libdir)/librx.so
+	-rm -f $(libdir)/librx.so.0
+	-rm -f $(libdir)/librx.so.0.0.0
 	-rm -f $(includedir)/rxposix.h
 
 clean:
-	-rm -f $(libobjs) librx.a  $(opt_objs) 
+	-rm -f $(libobjs) $(liblobjs) librx.a  $(opt_objs)
+	-rm -rf .libs
 
 distclean: clean
 	-rm Makefile config.status config.log rgx.x
@@ -116,6 +126,9 @@
 	$(AR) $(AR_FLAGS) librx.a $(libobjs) $(opt_objs)
 	$(RANLIB) librx.a
 
+librx.la: $(liblobjs) $(opt_objs)
+	$(LIBTOOL) $(CC) $(ALL_CFLAGS) $(DEFS) -I$(srcdir) -o $@ $(liblobjs) -rpath $(libdir)
+
 
 depends:
 	touch $(gen_c_files) scmconfig.h
@@ -150,3 +163,6 @@
 rxunfa.o : rxunfa.c rxall.h rx.h rxhash.h rxbitset.h rxunfa.h _rx.h rxcset.h \
   rxnfa.h rxnode.h 
 rgx.o : rgx.c rgx.x
+
+libtool: $(LIBTOOL_DEPS)
+	$(SHELL) ./config.status --recheck
