From a6a152f111766afefcdf3a3ba271cdeb3f387553 Mon Sep 17 00:00:00 2001
From: Sebastian Wiesinger <sebastian@karotte.org>
Date: Wed, 13 Jan 2021 12:30:06 +0100
Subject: [PATCH] Install manpage for dwdump

---
 doc/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/Makefile.am b/doc/Makefile.am
index 6ff1152..3d61286 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,2 +1,2 @@
-dist_man_MANS = dropwatch.1
+dist_man_MANS = dropwatch.1 dwdump.1
 
