From db1371e4d22465c27b5ed8d851cea6e72f0176f7 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Wed, 21 Mar 2018 18:10:08 +0100
Subject: [PATCH 1/2] Disable biosdevname by default

---
 biosdevname.rules.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/biosdevname.rules.in b/biosdevname.rules.in
index 0a32aa5..1dcca74 100644
--- a/biosdevname.rules.in
+++ b/biosdevname.rules.in
@@ -12,13 +12,13 @@ ENV{biosdevname}=="?*", ENV{UDEV_BIOSDEVNAME}="$env{biosdevname}"
 ENV{UDEV_BIOSDEVNAME}=="0", GOTO="netdevicename_end"
 ENV{UDEV_BIOSDEVNAME}=="1", GOTO="netdevicename_start"
 
-# uncomment the next line for biosdevname to be off by default
-# GOTO="netdevicename_end"
+# off by default
+GOTO="netdevicename_end"
 
 LABEL="netdevicename_start"
 
 # using NAME= instead of setting INTERFACE_NAME, so that persistent
 # names aren't generated for these devices, they are "named" on each boot.
-SUBSYSTEMS=="pci", PROGRAM="/sbin/biosdevname --policy physical -i %k", NAME="%c",  OPTIONS+="string_escape=replace"
+SUBSYSTEMS=="pci", PROGRAM="/sbin/biosdevname --smbios 2.6 --nopirq --policy physical -i %k", NAME="%c"  OPTIONS+="string_escape=replace"
 
 LABEL="netdevicename_end"
-- 
2.14.3

