From cba45fe9de6923b858edb0780e257b7257aa4f7b Mon Sep 17 00:00:00 2001
From: Harry Wong <harrywong@live.com>
Date: Thu, 4 May 2017 10:32:45 +0800
Subject: [PATCH] Onley apply to GCC 7

---
 include/rapidjson/internal/regex.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/rapidjson/internal/regex.h b/include/rapidjson/internal/regex.h
index 6d110bd..e1a2faa 100644
--- a/include/rapidjson/internal/regex.h
+++ b/include/rapidjson/internal/regex.h
@@ -29,8 +29,10 @@ RAPIDJSON_DIAG_OFF(implicit-fallthrough)
 #ifdef __GNUC__
 RAPIDJSON_DIAG_PUSH
 RAPIDJSON_DIAG_OFF(effc++)
+#if __GNUC__ >= 7
 RAPIDJSON_DIAG_OFF(implicit-fallthrough)
 #endif
+#endif
 
 #ifdef _MSC_VER
 RAPIDJSON_DIAG_PUSH
-- 
2.7.4
