diff -up libyami-edd1ad15e383d32c8d67bdc9198c834b6acebca5/encoder/vaapiencoder_base.cpp.init libyami-edd1ad15e383d32c8d67bdc9198c834b6acebca5/encoder/vaapiencoder_base.cpp
--- libyami-edd1ad15e383d32c8d67bdc9198c834b6acebca5/encoder/vaapiencoder_base.cpp.init	2020-01-30 04:35:19.000000000 +0100
+++ libyami-edd1ad15e383d32c8d67bdc9198c834b6acebca5/encoder/vaapiencoder_base.cpp	2020-02-06 20:46:24.670371008 +0100
@@ -678,7 +678,7 @@ YamiStatus VaapiEncoderBase::checkCodecD
 #ifndef __BUILD_GET_MV__
 YamiStatus VaapiEncoderBase::getOutput(VideoEncOutputBuffer* outBuffer, bool withWait)
 {
-    bool isEmpty;
+    bool isEmpty = true;
     PicturePtr picture;
     YamiStatus ret;
     FUNC_ENTER();
@@ -703,7 +703,7 @@ YamiStatus VaapiEncoderBase::getOutput(V
 {
     void *data = NULL;
     uint32_t mappedSize;
-    bool isEmpty;
+    bool isEmpty = true;
     PicturePtr picture;
     YamiStatus ret;
     FUNC_ENTER();
