From a53e3414e5566b288005a8cd59be67ac3af81cec Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Joonas=20Saraj=C3=A4rvi?= <muep@iki.fi>
Date: Sat, 2 May 2020 20:57:48 +0300
Subject: [PATCH] Use built-in mechanism for marking arch-independent

---
 CMakeLists.txt | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2325ebf8..46ed4ef8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -246,16 +246,7 @@ set(GLM_INSTALL_CONFIGDIR "${CMAKE_INSTALL_LIBDIR}/cmake/glm")
 install(DIRECTORY glm DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 
-# CMake automatically adds an architecture compatibility check to make sure
-# 32 and 64 bit code is not accidentally mixed. For a header-only library this
-# is not required. The check can be disabled by temporarily unsetting
-# CMAKE_SIZEOF_VOID_P. In CMake 3.14 and later this can be achieved more cleanly
-# with write_basic_package_version_file(ARCH_INDEPENDENT).
-# TODO: Use this once a newer CMake can be required.
-set(GLM_SIZEOF_VOID_P ${CMAKE_SIZEOF_VOID_P})
-unset(CMAKE_SIZEOF_VOID_P)
-write_basic_package_version_file("${CMAKE_CURRENT_BINARY_DIR}/glmConfigVersion.cmake" VERSION ${GLM_VERSION} COMPATIBILITY AnyNewerVersion)
-set(CMAKE_SIZEOF_VOID_P ${GLM_SIZEOF_VOID_P})
+write_basic_package_version_file("${CMAKE_CURRENT_BINARY_DIR}/glmConfigVersion.cmake" VERSION ${GLM_VERSION} COMPATIBILITY AnyNewerVersion ARCH_INDEPENDENT)
 
 # build tree package config
 configure_file(cmake/glmBuildConfig.cmake.in glmConfig.cmake @ONLY)
-- 
2.25.4

