diff --git a/configure b/configure
index 5bc55f8..3bc1ea2 100755
--- a/configure
+++ b/configure
@@ -103142,7 +103142,7 @@ main ()
 {
 
               netsnmp_variable_list *key = SNMP_MALLOC_TYPEDEF(netsnmp_variable_list);;
-              int val;
+              int val = 0;
               u_char type = ASN_INTEGER;
               snmp_set_var_value(key, &val, sizeof(val));
               snmp_set_var_typed_value(key, type, &val, sizeof(val));
diff --git a/configure.ac b/configure.ac
index 2a3152c..5288fcd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3872,7 +3872,7 @@ if test "x$with_libnetsnmp" = "xyes"; then
             ]],
             [[
               netsnmp_variable_list *key = SNMP_MALLOC_TYPEDEF(netsnmp_variable_list);;
-              int val;
+              int val = 0;
               u_char type = ASN_INTEGER;
               snmp_set_var_value(key, &val, sizeof(val));
               snmp_set_var_typed_value(key, type, &val, sizeof(val));
