From ab946c74080ef9608a324f9fca85ae6c6d1eef69 Mon Sep 17 00:00:00 2001
From: Ruben Kerkhof <ruben@rubenkerkhof.com>
Date: Sat, 30 Jan 2016 17:18:05 +0100
Subject: [PATCH] Disable rrdtool, include /etc/collectd.d

---
 src/collectd.conf.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/collectd.conf.in b/src/collectd.conf.in
index 07d66cd7595e..b91350ca16f4 100644
--- a/src/collectd.conf.in
+++ b/src/collectd.conf.in
@@ -168,7 +168,7 @@
 #@BUILD_PLUGIN_REDIS_TRUE@LoadPlugin redis
 #@BUILD_PLUGIN_ROUTEROS_TRUE@LoadPlugin routeros
 #@BUILD_PLUGIN_RRDCACHED_TRUE@LoadPlugin rrdcached
-@LOAD_PLUGIN_RRDTOOL@LoadPlugin rrdtool
+#@LOAD_PLUGIN_RRDTOOL@LoadPlugin rrdtool
 #@BUILD_PLUGIN_SENSORS_TRUE@LoadPlugin sensors
 #@BUILD_PLUGIN_SERIAL_TRUE@LoadPlugin serial
 #@BUILD_PLUGIN_SIGROK_TRUE@LoadPlugin sigrok
@@ -1497,3 +1497,4 @@
 #    </Type>
 #  </Host>
 #</Plugin>
+Include "/etc/collectd.d"
-- 
2.5.0

