diff -ruN a/ruby-progressbar.gemspec b/ruby-progressbar.gemspec
--- a/ruby-progressbar.gemspec	2020-12-30 18:48:50.000000000 -0800
+++ b/ruby-progressbar.gemspec	2022-04-08 12:35:49.310932035 -0700
@@ -13,9 +13,6 @@
   spec.homepage      = 'https://github.com/jfelchner/ruby-progressbar'
   spec.licenses      = ['MIT']
 
-  spec.cert_chain    = ['certs/jfelchner.pem']
-  spec.signing_key   = File.expand_path('~/.gem/certs/jfelchner-private_key.pem') if $0 =~ /gem\z/
-
   spec.executables   = []
   spec.files         = Dir['{app,config,db,lib/ruby-progressbar}/**/*'] + %w{lib/ruby-progressbar.rb Rakefile README.md LICENSE.txt}
 
