diff -Naur sessreg-1.1.0.old/man/filenames.sed.c sessreg-1.1.0/man/filenames.sed.c
--- sessreg-1.1.0.old/man/filenames.sed.c	2015-04-30 13:58:47.780569645 +0200
+++ sessreg-1.1.0/man/filenames.sed.c	1970-01-01 01:00:00.000000000 +0100
@@ -1,36 +0,0 @@
-#include "sessreg.h"
-
-#ifdef UTMPX_FILE
-# define UTF UTMPX_FILE
-# define UTM utmpx
-/* delete utmp-only content */
-/__BEGIN_UTMP_ONLY__/,/__END_UTMP_ONLY__/ d
-#else
-# define UTF UTMP_FILE
-# define UTM utmp
-/* delete utmpx-only content */
-/__BEGIN_UTMPX_ONLY__/,/__END_UTMPX_ONLY__/ d
-#endif
-
-#ifdef WTMPX_FILE
-# define WTF WTMPX_FILE
-# define WTM wtmpx
-#else
-# define WTF WTMP_FILE
-# define WTM wtmp
-#endif
-
-#ifndef TTYS_FILE
-# define TTYS_FILE "/etc/ttys"
-#endif
-
-#ifndef LLOG_FILE
-# define LLOG_FILE "/var/log/lastlog"
-#endif
-
-s|__utmp_manpage__|UTM|g
-s|__utmp_file__|UTF|g
-s|__wtmp_manpage__|WTM|g
-s|__wtmp_file__|WTF|g
-s|__ttys_file__|TTYS_FILE|g
-s|__lastlog_file__|LLOG_FILE|g
diff -Naur sessreg-1.1.0.old/man/Makefile.am sessreg-1.1.0/man/Makefile.am
--- sessreg-1.1.0.old/man/Makefile.am	2015-04-30 13:58:47.780569645 +0200
+++ sessreg-1.1.0/man/Makefile.am	2015-04-30 14:10:41.420249238 +0200
@@ -3,20 +3,11 @@
 appman_PRE = sessreg.man
 appman_DATA = $(appman_PRE:man=$(APP_MAN_SUFFIX))
 
-EXTRA_DIST = $(appman_PRE) filenames.sed.c
-CLEANFILES = $(appman_DATA) filenames.sed
+EXTRA_DIST = $(appman_PRE)
+CLEANFILES = $(appman_DATA)
 SUFFIXES = .$(APP_MAN_SUFFIX) .man
 
 AM_CPPFLAGS = -I$(top_builddir) -I$(top_srcdir)
-filenames.sed: filenames.sed.c
-	$(AM_V_GEN)$(CPP) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	    $(AM_CPPFLAGS) $(CPPFLAGS) $(srcdir)/filenames.sed.c | \
-	    $(SED) -n -e '/s|__/ p' -e '/^\/__/ p' > $@
-
-# String replacements in MAN_SUBSTS now come from xorg-macros.m4 via configure
-MAN_SUBSTS += -f filenames.sed
-
-sessreg.$(APP_MAN_SUFFIX): filenames.sed
 
 .man.$(APP_MAN_SUFFIX):
 	$(AM_V_GEN)$(SED) $(MAN_SUBSTS) < $< > $@
diff -Naur sessreg-1.1.0.old/man/sessreg.man sessreg-1.1.0/man/sessreg.man
--- sessreg-1.1.0.old/man/sessreg.man	2015-04-30 13:58:47.780569645 +0200
+++ sessreg-1.1.0/man/sessreg.man	2015-04-30 14:11:47.134127555 +0200
@@ -24,11 +24,11 @@
 .\"
 .TH SESSREG __appmansuffix__ __xorgversion__
 .SH NAME
-sessreg \- manage __utmp_manpage__/__wtmp_manpage__ entries for non-init clients
+sessreg \- manage utmp (__filemansuffix__)/wtmp (__filemansuffix__) entries for non-init clients
 .SH SYNOPSIS
 .B sessreg
-[-w \fI__wtmp_manpage__-file\fP]
-[-u \fI__utmp_manpage__-file\fP]
+[-w \fIwtmp (__filemansuffix__)-file\fP]
+[-u \fIutmp (__filemansuffix__)-file\fP]
 [-L \fIlastlog-file\fP]
 [-l \fIline-name\fP]
 [-h \fIhost-name\fP]
@@ -41,25 +41,25 @@
 \fIuser-name\fP
 .SH DESCRIPTION
 .PP
-\fISessreg\fP is a simple program for managing __utmp_manpage__/__wtmp_manpage__ and lastlog
+\fISessreg\fP is a simple program for managing utmp (__filemansuffix__)/wtmp (__filemansuffix__) and lastlog
 entries for xdm sessions.
 .\" __BEGIN_UTMP_ONLY__
 .PP
 System V has a better interface to utmp than BSD; it
 dynamically allocates entries in the file, instead of writing them at fixed
 positions indexed by position in
-.BR __ttys_file__ .
+.BR /etc/ttys .
 .PP
 To manage BSD-style utmp files, \fIsessreg\fP has two strategies.  In
 conjunction with xdm, the -x option counts the number of lines in
-.B __ttys_file__
+.B /etc/ttys
 and then adds to that the number of the line in the Xservers file which
 specifies the display.  The display name must be specified as the
 "line-name" using the -l option.  This sum is used as the "slot-number" in
 the utmp file that this entry will be written at.  In the more general case,
 the -s option specifies the slot-number directly.  If for some strange reason
 your system uses a file other than
-.B __ttys_file__
+.B /etc/ttys
 to manage init, the -t option can direct
 \fIsessreg\fP to look elsewhere for a count of terminal sessions.
 .PP
@@ -95,17 +95,17 @@
        sessreg -d -l $DISPLAY -x /etc/X11/xdm/Xservers $USER
 .fi
 .SH OPTIONS
-.IP "\fB-w\fP \fI__wtmp_manpage__-file\fP"
-This specifies an alternate __wtmp_manpage__ file, instead of
-.BR __wtmp_file__ .
-The special name "none" disables writing records to the __wtmp_manpage__ file.
-.IP "\fB-u\fP \fI__utmp_manpage__-file\fP"
-This specifies an alternate __utmp_manpage__ file, instead of
-.BR __utmp_file__ .
-The special name "none" disables writing records to the __utmp_manpage__ file.
+.IP "\fB-w\fP \fIwtmp (__filemansuffix__)-file\fP"
+This specifies an alternate wtmp (__filemansuffix__) file, instead of
+.BR /var/log/wtmp .
+The special name "none" disables writing records to the wtmp (__filemansuffix__) file.
+.IP "\fB-u\fP \fIutmp (__filemansuffix__)-file\fP"
+This specifies an alternate utmp (__filemansuffix__) file, instead of
+.BR /var/run/utmp .
+The special name "none" disables writing records to the utmp (__filemansuffix__) file.
 .IP "\fB-L\fP \fIlastlog-file\fP"
 This specifies an alternate lastlog file, instead of
-.BR __lastlog_file__ ,
+.BR /var/log/lastlog ,
 if the platform supports lastlog files.
 The special name "none" disables writing records to the lastlog file.
 .IP "\fB-l\fP \fIline-name\fP"
@@ -122,7 +122,7 @@
 .\" __BEGIN_UTMP_ONLY__
 Each potential session has a unique slot number in BSD systems, most are
 identified by the position of the \fIline-name\fP in the
-.BR __ttys_file__ file.
+.BR /etc/ttys file.
 This option overrides the default position determined with ttyslot(__libmansuffix__).
 This option is inappropriate for use with xdm, the -x option is more useful.
 .\" __END_UTMP_ONLY__
@@ -153,13 +153,13 @@
 .IP "\fB-V\fP"
 This option causes the command to print its version and exit.
 .IP "\fB-a\fP"
-This session should be added to __utmp_manpage__/__wtmp_manpage__.
+This session should be added to utmp/wtmp.
 .IP "\fB-d\fP"
-This session should be deleted from __utmp_manpage__/__wtmp_manpage__.  One of -a/-d must
+This session should be deleted from utmp/wtmp.  One of -a/-d must
 be specified.
 .SH "SEE ALSO"
 .BR xdm (__appmansuffix__),
-.BR __utmp_manpage__ (__filemansuffix__),
-.BR __wtmp_manpage__ (__filemansuffix__)
+.BR utmp (__filemansuffix__),
+.BR wtmp (__filemansuffix__)
 .SH AUTHOR
 Keith Packard, MIT X Consortium
