diff -up ilmbase-2.2.0/IlmThread/Makefile.am.no_undefined ilmbase-2.2.0/IlmThread/Makefile.am
--- ilmbase-2.2.0/IlmThread/Makefile.am.no_undefined	2014-08-09 21:03:38.000000000 -0500
+++ ilmbase-2.2.0/IlmThread/Makefile.am	2014-11-20 07:44:27.913588030 -0600
@@ -14,7 +14,7 @@ libIlmThread_la_LDFLAGS = -version-info
 if LIB_SUFFIX_EXISTS
 libIlmThread_la_LDFLAGS += -release @LIB_SUFFIX@
 endif
-libIlmThread_la_LIBADD = ../Iex/libIex.la
+libIlmThread_la_LIBADD = ../Iex/libIex.la $(PTHREAD_LIBS)
 
 libIlmThreadincludedir = $(includedir)/OpenEXR
 
diff -up ilmbase-2.2.0/IlmThread/Makefile.in.no_undefined ilmbase-2.2.0/IlmThread/Makefile.in
--- ilmbase-2.2.0/IlmThread/Makefile.in.no_undefined	2014-08-09 23:37:24.000000000 -0500
+++ ilmbase-2.2.0/IlmThread/Makefile.in	2014-11-20 07:44:27.914587968 -0600
@@ -251,7 +251,7 @@ libIlmThread_la_SOURCES = IlmThreadPool.
 
 libIlmThread_la_LDFLAGS = -version-info @LIBTOOL_VERSION@ \
 	-no-undefined $(am__append_1)
-libIlmThread_la_LIBADD = ../Iex/libIex.la
+libIlmThread_la_LIBADD = ../Iex/libIex.la $(PTHREAD_LIBS)
 libIlmThreadincludedir = $(includedir)/OpenEXR
 libIlmThreadinclude_HEADERS = IlmThreadPool.h IlmThread.h \
 			      IlmThreadSemaphore.h IlmThreadMutex.h \
