diff -ur a/services/plasmalogin.service.in b/services/plasmalogin.service.in
--- a/services/plasmalogin.service.in	2026-05-14 15:05:14.221923999 -0400
+++ b/services/plasmalogin.service.in	2026-05-14 15:05:37.843115826 -0400
@@ -10,6 +10,7 @@
 [Service]
 ExecStartPre=-udevadm settle --timeout=10
 ExecStart=@CMAKE_INSTALL_FULL_BINDIR@/plasmalogin
+EnvironmentFile=-/etc/sysconfig/plasmalogin
 Restart=always
 
 [Install]
