--- uu_touch-0.7.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ uu_touch-0.7.0/Cargo.toml	2026-03-16T21:03:26.184382+00:00
@@ -77,14 +77,6 @@
 [dev-dependencies.tempfile]
 version = "3.15.0"
 
-[target.'cfg(target_os = "windows")'.dependencies.windows-sys]
-version = "0.61.0"
-features = [
-    "Win32_Storage_FileSystem",
-    "Win32_Foundation",
-]
-default-features = false
-
 [target."cfg(unix)".dependencies.nix]
 version = "0.30"
 features = ["fs"]
@@ -153,3 +145,4 @@
     "cfg(fuzzing)",
     'cfg(target_os, values("cygwin"))',
 ]
+
