diff -up convmv-2.0/Makefile.preserve-timestamps convmv-2.0/Makefile
--- convmv-2.0/Makefile.preserve-timestamps	2008-11-29 14:49:38.000000000 +0100
+++ convmv-2.0/Makefile	2015-05-19 11:10:32.519796351 +0200
@@ -11,7 +11,7 @@ install: all
 	mkdir -p $(DESTDIR)$(MANDIR)/man1/
 	mkdir -p $(DESTDIR)$(PREFIX)/bin/
 	cp convmv.1.gz $(DESTDIR)$(MANDIR)/man1/
-	install -m 755 convmv $(DESTDIR)$(PREFIX)/bin/
+	install -p -m 755 convmv $(DESTDIR)$(PREFIX)/bin/
 
 manpage:
 	pod2man --section 1 --center=" " convmv | gzip > convmv.1.gz
