diff -rU3 gobject-introspection-1.76.1-orig/giscanner/__init__.py gobject-introspection-1.76.1/giscanner/__init__.py
--- gobject-introspection-1.76.1-orig/giscanner/__init__.py	2023-03-22 23:43:34.000000000 +0100
+++ gobject-introspection-1.76.1/giscanner/__init__.py	2023-06-07 11:50:00.217226422 +0200
@@ -25,3 +25,8 @@
     from ._version import __version__
 except ImportError:
     __version__ = '0.0.0'
+
+from . import gio_ast as ast
+
+import sys
+sys.modules['giscanner.ast'] = ast
diff -rU3 gobject-introspection-1.76.1-orig/giscanner/meson.build gobject-introspection-1.76.1/giscanner/meson.build
--- gobject-introspection-1.76.1-orig/giscanner/meson.build	2023-03-22 23:43:34.000000000 +0100
+++ gobject-introspection-1.76.1/giscanner/meson.build	2023-06-07 11:48:32.922732028 +0200
@@ -2,7 +2,7 @@
   '__init__.py',
   'annotationmain.py',
   'annotationparser.py',
-  'ast.py',
+  'gio_ast.py',
   'cachestore.py',
   'ccompiler.py',
   'codegen.py',
   