diff -pruN paps-0.6.8.orig/src/Makefile.am paps-0.6.8/src/Makefile.am
--- paps-0.6.8.orig/src/Makefile.am	2012-11-23 14:22:51.506176037 +0900
+++ paps-0.6.8/src/Makefile.am	2012-11-23 14:25:41.828410416 +0900
@@ -8,13 +8,14 @@ bin_PROGRAMS = paps
 paps_CFLAGS  = -Wall $(CUPS_CFLAGS)
 paps_SOURCES = paps.c
 paps_LDADD = $(lib_LTLIBRARIES) $(all_libraries)
-paps_LDFLAGS = `pkg-config --libs pangoft2` $(CUPS_LIBS)
+paps_LDFLAGS = $(CUPS_LIBS)
 paps_DEPENDENCIES = $(lib_LTLIBRARIES)
 
 EXTRA_DIST = test_libpaps.c paps.1
 
 # set the include path found by configure
 INCLUDES= $(all_includes) `pkg-config --cflags pangoft2`
+LIBS = `pkg-config --libs pangoft2`
 
 # Test program
 noinst_PROGRAMS = test_libpaps
@@ -22,6 +23,5 @@ noinst_PROGRAMS = test_libpaps
 
 test_libpaps_SOURCES = test_libpaps.c
 test_libpaps_LDADD = $(lib_LTLIBRARIES) $(all_libraries)
-test_libpaps_LDFLAGS = `pkg-config --libs pangoft2`
 test_libpaps_DEPENDENCIES = $(lib_LTLIBRARIES) 
 
