diff -pruN paps-0.6.8.orig/src/libpaps.c paps-0.6.8/src/libpaps.c
--- paps-0.6.8.orig/src/libpaps.c	2014-03-20 12:10:50.000000000 +0900
+++ paps-0.6.8/src/libpaps.c	2014-03-20 12:11:54.000000000 +0900
@@ -25,8 +25,9 @@
 
 #include <pango/pango.h>
 #include <pango/pangoft2.h>
-#include <freetype/ftglyph.h>
-#include <freetype/ftoutln.h>
+#include <ft2build.h>
+#include FT_GLYPH_H
+#include FT_OUTLINE_H
 #include <errno.h>
 #include <locale.h>
 #include <stdlib.h>
