diff --git a/src/container/cstring_view.h b/src/container/cstring_view.h
index 5731d817..5ff68c17 100644
--- a/src/container/cstring_view.h
+++ b/src/container/cstring_view.h
@@ -27,7 +27,7 @@ public:
   {
   }

-  constexpr const char *c_str() const noexcept
+  constexpr const_pointer c_str() const noexcept
   {
     return data();
   }
diff --git a/tests/cstring_view.cpp b/tests/cstring_view.cpp
index 6bfed5f6fbea..69bea365f6df 100644
--- a/tests/cstring_view.cpp
+++ b/tests/cstring_view.cpp
@@ -17,7 +17,9 @@ TEST(cstring_view, c_string)
   EXPECT_EQ('a', sv[0]);
   EXPECT_EQ('b', sv[1]);
   EXPECT_EQ('c', sv[2]);
-  EXPECT_EQ('\0', sv[3]);
+  // Skip operator[], because accessing the terminator fails
+  // the bound checks for hardened std lib, see #4001.
+  EXPECT_EQ('\0', sv.c_str()[3]);
 }
 
 TEST(cstring_view, std_string)
@@ -30,7 +32,9 @@ TEST(cstring_view, std_string)
   EXPECT_EQ('a', sv[0]);
   EXPECT_EQ('b', sv[1]);
   EXPECT_EQ('c', sv[2]);
-  EXPECT_EQ('\0', sv[3]);
+  // Skip operator[], because accessing the terminator fails
+  // the bound checks for hardened std lib, see #4001.
+  EXPECT_EQ('\0', sv.c_str()[3]);
 }
 
 TEST(cstring_view, std_string_view)
