diff --git a/src/bpfmap.cpp b/src/bpfmap.cpp
index 23d550ff..84ff00b8 100644
--- a/src/bpfmap.cpp
+++ b/src/bpfmap.cpp
@@ -7,7 +7,7 @@ libbpf::bpf_map_type BpfMap::type() const
   return type_;
 }
 
-cstring_view BpfMap::bpf_name() const
+const std::string &BpfMap::bpf_name() const
 {
   return name_;
 }
diff --git a/src/bpfmap.h b/src/bpfmap.h
index ddc0ddc9..75728506 100644
--- a/src/bpfmap.h
+++ b/src/bpfmap.h
@@ -11,8 +11,6 @@ namespace libbpf {
 #include "libbpf/bpf.h"
 } // namespace libbpf
 
-#include "container/cstring_view.h"
-
 namespace bpftrace {
 
 class BpfMap {
@@ -27,12 +25,12 @@ public:
   }
 
   BpfMap(libbpf::bpf_map_type type,
-         cstring_view name,
+         std::string name,
          uint32_t key_size,
          uint32_t value_size,
          uint32_t max_entries)
       : type_(type),
-        name_(name),
+        name_(std::move(name)),
         key_size_(key_size),
         value_size_(value_size),
         max_entries_(max_entries)
@@ -42,7 +40,7 @@ public:
   int fd = -1;
 
   libbpf::bpf_map_type type() const;
-  cstring_view bpf_name() const;
+  const std::string &bpf_name() const;
   std::string name() const;
   uint32_t key_size() const;
   uint32_t value_size() const;
@@ -55,7 +53,7 @@ public:
 
 private:
   libbpf::bpf_map_type type_;
-  cstring_view name_;
+  std::string name_;
   uint32_t key_size_;
   uint32_t value_size_;
   uint32_t max_entries_;
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index d352da7a..e75f0707 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -30,7 +30,6 @@ add_executable(bpftrace_test
   clang_parser.cpp
   config.cpp
   collect_nodes.cpp
-  cstring_view.cpp
   field_analyser.cpp
   log.cpp
   main.cpp
