diff -up at-3.1.20/atd.c.log-jobs at-3.1.20/atd.c
--- at-3.1.20/atd.c.log-jobs	2018-05-23 15:09:35.158362293 +0200
+++ at-3.1.20/atd.c	2018-05-23 15:18:39.153965582 +0200
@@ -376,6 +376,10 @@ run_file(const char *filename, uid_t uid
 	pabort("Userid %lu not found - aborting job %8lu (%.500s)",
 	       (unsigned long) uid, jobno, filename);
     }
+
+    syslog(LOG_INFO, "Starting job %lu (%.500s) for user '%s' (%lu)",
+		jobno, filename, pentry->pw_name, (unsigned long) uid);
+
     PRIV_START
 
 	stream = fopen(filename, "r");
