diff -up at-3.1.14/at.c.opt at-3.1.14/at.c
--- at-3.1.14/at.c.opt	2013-09-26 16:54:29.920476315 +0200
+++ at-3.1.14/at.c	2013-09-26 16:55:06.301562646 +0200
@@ -842,10 +842,9 @@ main(int argc, char **argv)
      */
 
     if (disp_version) {
-	fprintf(stderr, "at version " VERSION "\n"
-	   "Please report bugs to the Debian bug tracking system (http://bugs.debian.org/)\n"
-	   "or contact the maintainers (at@packages.debian.org).\n");
-	exit(EXIT_SUCCESS);
+        fprintf(stderr, "at version " VERSION "\n");
+        if (argc == 2)
+	    exit(EXIT_SUCCESS);
     }
 
     /* select our program
