diff -up cups-1.7.0/backend/ipp.c.ipp-multifile cups-1.7.0/backend/ipp.c
--- cups-1.7.0/backend/ipp.c.ipp-multifile	2013-10-24 15:52:00.745814354 +0100
+++ cups-1.7.0/backend/ipp.c	2013-10-24 15:53:46.463266724 +0100
@@ -1758,7 +1758,10 @@ main(int  argc,				/* I - Number of comm
 	ippAddBoolean(request, IPP_TAG_OPERATION, "last-document",
         	      (i + 1) >= num_files);
 
-	if (document_format)
+	if (num_files > 1)
+	  ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_MIMETYPE,
+		       "document-format", NULL, "application/octet-stream");
+	else if (document_format)
 	  ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_MIMETYPE,
 		       "document-format", NULL, document_format);
 
