diff -up cups-2.3.1/ppdc/sample.drv.dymo-deviceid cups-2.3.1/ppdc/sample.drv
--- cups-2.3.1/ppdc/sample.drv.dymo-deviceid	2019-12-16 09:22:34.476492212 +0100
+++ cups-2.3.1/ppdc/sample.drv	2019-12-16 09:23:44.665003895 +0100
@@ -129,6 +129,7 @@ Version "2.3"
 {
   Manufacturer "DYMO"
   ModelName "Label Printer"
+  Attribute "1284DeviceID" "" "MFG:DYMO;MDL:LabelWriter 400;"
   Attribute NickName "" "DYMO Label Printer"
   PCFileName "dymo.ppd"
   DriverType label
