diff -up cups-1.5b1/scheduler/banners.c.banners cups-1.5b1/scheduler/banners.c
--- cups-1.5b1/scheduler/banners.c.banners	2011-05-20 05:49:49.000000000 +0200
+++ cups-1.5b1/scheduler/banners.c	2011-05-23 17:35:30.000000000 +0200
@@ -110,6 +110,8 @@ cupsdLoadBanners(const char *d)		/* I - 
     if ((ext = strrchr(dent->filename, '.')) != NULL)
       if (!strcmp(ext, ".bck") ||
           !strcmp(ext, ".bak") ||
+          !strcmp(ext, ".rpmnew") ||
+          !strcmp(ext, ".rpmsave") ||
 	  !strcmp(ext, ".sav"))
 	continue;
 
