diff -up cups-1.5b1/conf/cups.password-auth.system-auth cups-1.5b1/conf/cups.password-auth
--- cups-1.5b1/conf/cups.password-auth.system-auth	2011-05-23 17:27:27.000000000 +0200
+++ cups-1.5b1/conf/cups.password-auth	2011-05-23 17:27:27.000000000 +0200
@@ -0,0 +1,4 @@
+#%PAM-1.0
+# Use password-auth common PAM configuration for the daemon
+auth        include     password-auth
+account     include     password-auth
diff -up cups-1.5b1/conf/cups.system-auth.system-auth cups-1.5b1/conf/cups.system-auth
--- cups-1.5b1/conf/cups.system-auth.system-auth	2011-05-23 17:27:27.000000000 +0200
+++ cups-1.5b1/conf/cups.system-auth	2011-05-23 17:27:27.000000000 +0200
@@ -0,0 +1,3 @@
+#%PAM-1.0
+auth	include		system-auth
+account	include		system-auth
diff -up cups-1.5b1/conf/Makefile.system-auth cups-1.5b1/conf/Makefile
--- cups-1.5b1/conf/Makefile.system-auth	2011-05-12 07:21:56.000000000 +0200
+++ cups-1.5b1/conf/Makefile	2011-05-23 17:27:27.000000000 +0200
@@ -90,10 +90,16 @@ install-data:
 	done
 	-if test x$(PAMDIR) != x; then \
 		$(INSTALL_DIR) -m 755 $(BUILDROOT)$(PAMDIR); \
-		if test -r $(BUILDROOT)$(PAMDIR)/cups ; then \
-			$(INSTALL_DATA) $(PAMFILE) $(BUILDROOT)$(PAMDIR)/cups.N ; \
+		if test -f /etc/pam.d/password-auth; then \
+			$(INSTALL_DATA) cups.password-auth $(BUILDROOT)$(PAMDIR)/cups; \
+		elif test -f /etc/pam.d/system-auth; then \
+			$(INSTALL_DATA) cups.system-auth $(BUILDROOT)$(PAMDIR)/cups; \
 		else \
-			$(INSTALL_DATA) $(PAMFILE) $(BUILDROOT)$(PAMDIR)/cups ; \
+			if test -r $(BUILDROOT)$(PAMDIR)/cups ; then \
+				$(INSTALL_DATA) $(PAMFILE) $(BUILDROOT)$(PAMDIR)/cups.N ; \
+			else \
+				$(INSTALL_DATA) $(PAMFILE) $(BUILDROOT)$(PAMDIR)/cups ; \
+			fi ; \
 		fi ; \
 	fi
 
