diff -up cups-1.7b1/scheduler/ipp.c.driverd-timeout cups-1.7b1/scheduler/ipp.c
--- cups-1.7b1/scheduler/ipp.c.driverd-timeout	2013-04-19 12:24:43.003841810 +0200
+++ cups-1.7b1/scheduler/ipp.c	2013-04-19 12:24:43.204839107 +0200
@@ -4556,7 +4556,7 @@ copy_model(cupsd_client_t *con,		/* I -
   close(temppipe[1]);
 
  /*
-  * Wait up to 30 seconds for the PPD file to be copied...
+  * Wait up to 70 seconds for the PPD file to be copied...
   */
 
   total = 0;
@@ -4576,7 +4576,7 @@ copy_model(cupsd_client_t *con,		/* I -
     FD_SET(temppipe[0], &input);
     FD_SET(CGIPipes[0], &input);
 
-    timeout.tv_sec  = 30;
+    timeout.tv_sec  = 70;
     timeout.tv_usec = 0;
 
     if ((i = select(maxfd, &input, NULL, NULL, &timeout)) < 0)
