--- unittest/exe_moduletest/CMakeListsOld.txt	2021-12-27 12:52:14.558000000 -0800
+++ unittest/exe_moduletest/CMakeLists.txt	2021-12-27 12:52:47.158898400 -0800
@@ -5,13 +5,13 @@
 add_executable(symcryptmoduletest_linux ${SOURCES})
 target_link_libraries(symcryptmoduletest_linux symcrypt_linux)
 
-if(CMAKE_SYSTEM_PROCESSOR MATCHES "AMD64")
-  add_executable(symcryptmoduletest_oe ${SOURCES})
-  target_link_libraries(symcryptmoduletest_oe symcrypt_oe)
+#if(CMAKE_SYSTEM_PROCESSOR MATCHES "AMD64")
+#  add_executable(symcryptmoduletest_oe ${SOURCES})
+#  target_link_libraries(symcryptmoduletest_oe symcrypt_oe)
 
-  add_dependencies(symcryptmoduletest_oe symcrypt_oe_fips)
-endif()
+#  add_dependencies(symcryptmoduletest_oe symcrypt_oe_fips)
+#endif()
 
-add_executable(symcryptmoduletest_embedded ${SOURCES})
-target_link_libraries(symcryptmoduletest_embedded symcrypt_embedded)
+#add_executable(symcryptmoduletest_embedded ${SOURCES})
+#target_link_libraries(symcryptmoduletest_embedded symcrypt_embedded)
 #add_dependencies(symcryptmoduletest_embedded symcrypt_embedded_fips)
\ No newline at end of file
