--- rspec-expectations-3.13.4/spec/rspec/matchers/built_in/include_spec.rb	2025-05-15 11:40:49.000000000 +0900
+++ rspec-expectations-3.13.3/spec/rspec/matchers/built_in/include_spec.rb	2024-09-08 16:39:38.000000000 +0900
@@ -109,7 +104,7 @@ RSpec.describe "#include matcher" do
       failure_string = if use_string_keys_in_failure_message?
                          dedent(<<-END)
                            |Diff:
-                           |@@ #{one_line_header(3)} @@
+                           |@@ -1,3 +1,3 @@
                            |-:bar => 3,
                            |-:foo => 1,
                            |+"bar" => 2,
@@ -137,7 +132,7 @@ RSpec.describe "#include matcher" do
       failure_string = if use_string_keys_in_failure_message?
                          dedent(<<-END)
                            |Diff:
-                           |@@ #{one_line_header(3)} @@
+                           |@@ -1,3 +1,3 @@
                            |-(match /FOO/i) => 1,
                            |-:bar => 3,
                            |+"bar" => 2,
@@ -146,7 +141,7 @@ RSpec.describe "#include matcher" do
                        else
                          dedent(<<-END)
                            |Diff:
-                           |@@ #{one_line_header(3)} @@
+                           |@@ -1,3 +1,3 @@
                            |-(match /FOO/i) => 1,
                            |-:bar => 3,
                            |+:bar => 2,
--- rspec-expectations-3.13.4/spec/rspec/matchers/built_in/compound_spec.rb	2025-05-15 11:40:49.000000000 +0900
+++ rspec-expectations-3.13.3/spec/rspec/matchers/built_in/compound_spec.rb	2024-09-08 16:39:38.000000000 +0900
@@ -596,14 +595,14 @@ module RSpec::Matchers::BuiltIn
                     |
                     |   expected "HELLO\\nWORLD" to eq downcase "hola\\nmon"
                     |Diff for (eq downcase "bonjour\\nmonde"):
-                    |@@ #{one_line_header(3)} @@
+                    |@@ -1,3 +1,3 @@
                     |-bonjour
                     |-monde
                     |+hello
                     |+world
                     |
                     |Diff for (eq downcase "hola\\nmon"):
-                    |@@ #{one_line_header(3)} @@
+                    |@@ -1,3 +1,3 @@
                     |-hola
                     |-mon
                     |+hello
@@ -639,14 +638,14 @@ module RSpec::Matchers::BuiltIn
                     |
                     |   expected "hello\\nworld" to eq downcase "hola\\nmon"
                     |Diff for (eq downcase "bonjour\\nmonde"):
-                    |@@ #{one_line_header(3)} @@
+                    |@@ -1,3 +1,3 @@
                     |-bonjour
                     |-monde
                     |+hello
                     |+world
                     |
                     |Diff for (eq downcase "hola\\nmon"):
-                    |@@ #{one_line_header(3)} @@
+                    |@@ -1,3 +1,3 @@
                     |-hola
                     |-mon
                     |+hello

