diff -up test/analyzer/audio_analyzer_test.rb.ffmpeg8 test/analyzer/audio_analyzer_test.rb
--- test/analyzer/audio_analyzer_test.rb.ffmpeg8	2025-09-23 00:17:22.000000000 +0200
+++ test/analyzer/audio_analyzer_test.rb	2025-11-02 14:36:22.296948668 +0100
@@ -10,7 +10,7 @@ class ActiveStorage::Analyzer::AudioAnal
     blob = create_file_blob(filename: "audio.mp3", content_type: "audio/mp3")
     metadata = extract_metadata_from(blob)
 
-    assert_equal 0.914286, metadata[:duration]
+    assert_equal 0.863379, metadata[:duration]
     assert_equal 128000, metadata[:bit_rate]
     assert_equal 44100, metadata[:sample_rate]
     assert_not_nil metadata[:tags]
