diff -up uuid-1.6.2/Makefile.in.nostrip uuid-1.6.2/Makefile.in
--- uuid-1.6.2/Makefile.in.nostrip	2012-06-19 12:53:47.888028956 +0200
+++ uuid-1.6.2/Makefile.in	2012-06-19 12:53:47.913029142 +0200
@@ -253,7 +253,7 @@ install:
 	-@if [ ".$(WITH_CXX)" = .yes ]; then \
 	    $(LIBTOOL) --mode=install $(SHTOOL) install -c -m 644 $(CXX_NAME) $(DESTDIR)$(libdir)/; \
 	fi
-	@$(LIBTOOL) --mode=install $(SHTOOL) install -c -s -m 755 uuid $(DESTDIR)$(bindir)/
+	@$(LIBTOOL) --mode=install $(SHTOOL) install -c -m 755 uuid $(DESTDIR)$(bindir)/
 	$(SHTOOL) install -c -m 644 $(S)/uuid.1 $(DESTDIR)$(mandir)/man1/
 	-@if [ ".$(WITH_PERL)" = .yes ]; then \
 	    (cd $(S)/perl && $(MAKE) $(MFLAGS) install DESTDIR=$(DESTDIR)); \
