--- lrzsz-0.12.20/src/lsz.c.glibc21	Tue Dec 29 17:07:59 1998
+++ lrzsz-0.12.20/src/lsz.c	Mon Mar  8 10:56:19 1999
@@ -53,7 +53,6 @@
 #ifndef STRICT_PROTOTYPES
 extern time_t time();
 extern char *strerror();
-extern char *strstr();
 #endif
 
 #ifndef HAVE_ERRNO_DECLARATION
--- lrzsz-0.12.20/src/lrz.c.glibc21	Wed Dec 30 02:49:24 1998
+++ lrzsz-0.12.20/src/lrz.c	Mon Mar  8 10:56:01 1999
@@ -44,7 +44,6 @@
 #ifndef STRICT_PROTOTYPES
 extern time_t time();
 extern char *strerror();
-extern char *strstr();
 #endif
 
 #ifndef HAVE_ERRNO_DECLARATION
