From 9ea729ce863396484d2e4c5a93af4b625fc5c90c Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sat, 18 Jan 2020 11:31:27 +0000
Subject: [PATCH 4/5] Remove configure from .gitattributes.

It's not a binary file.
---
 .gitattributes | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/.gitattributes b/.gitattributes
index 9be9e33a0..5df88ab4e 100644
--- a/.gitattributes
+++ b/.gitattributes
@@ -29,10 +29,6 @@
 
 /boot/menhir/parser.ml* -diff
 
-# configure is declared as binary so that it doesn't get included in diffs.
-# This also means it will have the correct Unix line-endings, even on Windows.
-/configure binary
-
 # 'union' merge driver just unions textual content in case of conflict
 #   http://krlmlr.github.io/using-gitattributes-to-avoid-merge-conflicts/
 /.mailmap                merge=union
-- 
2.25.0

