diff -rup libvirt-glib-4.0.0.orig/meson.build libvirt-glib-4.0.0.new/meson.build
--- libvirt-glib-4.0.0.orig/meson.build	2021-02-15 12:23:04.609036200 +0000
+++ libvirt-glib-4.0.0.new/meson.build	2021-02-15 14:50:24.838971022 +0000
@@ -175,8 +175,6 @@ cc_flags += [
   '-Wbuiltin-declaration-mismatch',
   '-Wbuiltin-macro-redefined',
   '-Wcannot-profile',
-  '-Wcast-align',
-  '-Wcast-align=strict',
   '-Wcast-function-type',
   '-Wchar-subscripts',
   '-Wclobbered',
