diff -up libvisual-0.4.0/libvisual/lv_plugin.c.format-security libvisual-0.4.0/libvisual/lv_plugin.c
--- libvisual-0.4.0/libvisual/lv_plugin.c.format-security	2014-06-11 15:38:36.786199952 -0400
+++ libvisual-0.4.0/libvisual/lv_plugin.c	2014-06-11 15:38:45.990132144 -0400
@@ -442,11 +442,11 @@ VisList *visual_plugin_registry_filter (
 				
 				visual_list_add (list, ref);
 			} else if (ret != FALSE) {
-				visual_log (VISUAL_LOG_WARNING, visual_error_to_string (ret));
+				visual_log (VISUAL_LOG_WARNING, "%s", visual_error_to_string (ret));
 			}
 		}
 		else if (ret != FALSE) { /* FIXME XXX TODO, patch frmo duilio check how this works */
-			visual_log (VISUAL_LOG_WARNING, visual_error_to_string (ret));
+			visual_log (VISUAL_LOG_WARNING, "%s", visual_error_to_string (ret));
 		}
 	}
 
