diff -ruN a/bin/h5cc.in b/bin/h5cc.in
--- a/bin/h5cc.in	2024-05-15 18:32:01.952461165 -0700
+++ b/bin/h5cc.in	2024-05-16 18:45:29.808730781 -0700
@@ -87,10 +87,10 @@
 # paths and libraries from AM_LDFLAGS, LDFLAGS, AM_LIBS or LIBS carried in
 # from the hdf5 build. The order of the flags is intended to give precedence
 # to the user's flags.
-H5BLD_CFLAGS="@AM_CFLAGS@ @CFLAGS@"
+H5BLD_CFLAGS=
 H5BLD_CPPFLAGS="@AM_CPPFLAGS@ @CPPFLAGS@"
-H5BLD_LDFLAGS="@AM_LDFLAGS@ @LDFLAGS@"
-H5BLD_LIBS="@LIBS@"
+H5BLD_LDFLAGS=
+H5BLD_LIBS=
 
 CC="${HDF5_CC:-$CCBASE}"
 CLINKER="${HDF5_CLINKER:-$CLINKERBASE}"
@@ -103,7 +103,8 @@
 # available library is shared, it will be used by default.  The user can
 # override either default, although choosing an unavailable library will result
 # in link errors.
-STATIC_AVAILABLE="@enable_static@"
+# Fedora prefers shared libraries
+STATIC_AVAILABLE=no
 if test "${STATIC_AVAILABLE}" = "yes"; then
   USE_SHARED_LIB="${HDF5_USE_SHLIB:-no}"
 else
@@ -380,7 +381,7 @@
   # paths and libraries from AM_LDFLAGS, LDFLAGS, AM_LIBS or LIBS carried in
   # from the hdf5 build. The order of the flags is intended to give precedence
   # to the user's flags.
-  $SHOW $CLINKER $H5BLD_CPPFLAGS $CPPFLAGS $H5BLD_CFLAGS $CFLAGS $LDFLAGS $clibpath $link_objs $LIBS $link_args $shared_link
+  $SHOW $CLINKER $H5BLD_CPPFLAGS $CPPFLAGS $H5BLD_CFLAGS $CFLAGS $LDFLAGS $clibpath $link_objs $LIBS $link_args
   status=$?
 fi
 
diff -ruN a/c++/src/h5c++.in b/c++/src/h5c++.in
--- a/c++/src/h5c++.in	2024-05-15 18:32:01.984461136 -0700
+++ b/c++/src/h5c++.in	2024-05-16 18:47:24.748593382 -0700
@@ -85,10 +85,10 @@
 # paths and libraries from AM_LDFLAGS, LDFLAGS, AM_LIBS or LIBS carried in
 # from the hdf5 build. The order of the flags is intended to give precedence
 # to the user's flags.
-H5BLD_CXXFLAGS="@AM_CXXFLAGS@ @CXXFLAGS@"
+H5BLD_CXXFLAGS=
 H5BLD_CPPFLAGS="@AM_CPPFLAGS@ @CPPFLAGS@"
-H5BLD_LDFLAGS="@AM_LDFLAGS@ @LDFLAGS@"
-H5BLD_LIBS="@LIBS@"
+H5BLD_LDFLAGS=
+H5BLD_LIBS=
 
 CXX="${HDF5_CXX:-$CXXBASE}"
 CXXLINKER="${HDF5_CXXLINKER:-$CXXLINKERBASE}"
@@ -101,7 +101,8 @@
 # available library is shared, it will be used by default.  The user can
 # override either default, although choosing an unavailable library will result
 # in link errors.
-STATIC_AVAILABLE="@enable_static@"
+# Fedora prefers shared libraries
+STATIC_AVAILABLE=no
 if test "${STATIC_AVAILABLE}" = "yes"; then
   USE_SHARED_LIB="${HDF5_USE_SHLIB:-no}"
 else
@@ -372,7 +373,7 @@
   # from the hdf5 build. The order of the flags is intended to give precedence
   # to the user's flags.
 
-  $SHOW $CXXLINKER $H5BLD_CPPFLAGS $CPPFLAGS $H5BLD_CXXFLAGS $CXXFLAGS $LDFLAGS $clibpath $link_objs $LIBS $link_args $shared_link
+  $SHOW $CXXLINKER $H5BLD_CPPFLAGS $CPPFLAGS $H5BLD_CXXFLAGS $CXXFLAGS $LDFLAGS $clibpath $link_objs $LIBS $link_args
 
   status=$?
 fi
diff -ruN a/fortran/src/h5fc.in b/fortran/src/h5fc.in
--- a/fortran/src/h5fc.in	2024-05-15 18:32:02.016461108 -0700
+++ b/fortran/src/h5fc.in	2024-05-16 18:50:03.828406070 -0700
@@ -83,10 +83,10 @@
 # from AM_LDFLAGS, LDFLAGS, AM_LIBS or LIBS carried in from the hdf5 build.
 # The order of the flags is intended to give precedence to the user's flags.
 H5BLD_FCFLAGS="@AM_FCFLAGS@ @FCFLAGS@"
-F9XMODFLAG="@F9XMODFLAG@"
+F9XMODFLAG=
 F9XSUFFIXFLAG="@F9XSUFFIXFLAG@"
-H5BLD_LDFLAGS="@AM_LDFLAGS@ @LDFLAGS@"
-H5BLD_LIBS="@LIBS@"
+H5BLD_LDFLAGS=
+H5BLD_LIBS=
 
 FC="${HDF5_FC:-$FCBASE}"
 FLINKER="${HDF5_FLINKER:-$FLINKERBASE}"
@@ -98,7 +98,8 @@
 # available library is shared, it will be used by default.  The user can
 # override either default, although choosing an unavailable library will result
 # in link errors.
-STATIC_AVAILABLE="@enable_static@"
+# Fedora prefers shared libraries
+STATIC_AVAILABLE=no
 if test "${STATIC_AVAILABLE}" = "yes"; then
   USE_SHARED_LIB="${HDF5_USE_SHLIB:-no}"
 else
