diff -ruN a/hl/tools/CMakeLists.txt b/hl/tools/CMakeLists.txt
--- a/hl/tools/CMakeLists.txt	2024-05-15 18:32:01.940461175 -0700
+++ b/hl/tools/CMakeLists.txt	2024-05-17 14:18:52.349842239 -0700
@@ -4,9 +4,6 @@
 #-----------------------------------------------------------------------------
 #-- Option to build the High level GIF Tools
 #-----------------------------------------------------------------------------
-option (HDF5_BUILD_HL_GIF_TOOLS  "Build HDF5 HL GIF Tools" OFF)
-if (HDF5_BUILD_HL_GIF_TOOLS)
-  add_subdirectory (gif2h5)
-endif ()
-
+option (HDF5_BUILD_HL_GIF_TOOLS  "Build HDF5 HL GIF Tools" ON)
+add_subdirectory (gif2h5)
 add_subdirectory (h5watch)
