diff -up lua-5.4.6/src/luaconf.h.template.in.idsize lua-5.4.6/src/luaconf.h.template.in
--- lua-5.4.6/src/luaconf.h.template.in.idsize	2023-05-22 11:02:54.207370261 -0400
+++ lua-5.4.6/src/luaconf.h.template.in	2023-05-22 11:04:21.410747055 -0400
@@ -761,7 +761,7 @@
 ** of a function in debug information.
 ** CHANGE it if you want a different size.
 */
-#define LUA_IDSIZE	60
+#define LUA_IDSIZE	512


 /*
