From 1d1502d31aa327ec4c363023e767f8be67381fb9 Mon Sep 17 00:00:00 2001
From: Jerry James <loganjerry@gmail.com>
Date: Fri, 8 Dec 2023 14:25:13 -0700
Subject: [PATCH] Fix race when building src/lusol.o

---
 makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile b/makefile
index bcedc2a..504e1ea 100644
--- a/makefile
+++ b/makefile
@@ -144,7 +144,7 @@ $(F90_OBJ) : %.o : %.f90
 $(F90_MOD) : %.mod : %.o
 
 # extra fortran dependencies
-lusol.o : lusol_precision.mod
+src/lusol.o : src/lusol_precision.mod
 
 # C code generation
 src/clusol.h: $(INTERFACE_FILES)
-- 
2.43.0

