--- v_frame-0.3.9/Cargo.toml	1970-01-01T00:00:01+00:00
+++ v_frame-0.3.9/Cargo.toml	2026-01-28T20:27:15.545043+00:00
@@ -27,25 +27,14 @@
 repository = "https://github.com/rust-av/v_frame"
 
 [features]
-profiling = ["dep:profiling"]
 serialize = [
     "serde",
     "aligned-vec/serde",
-]
-tracing = [
-    "profiling",
-    "dep:tracing",
-    "profiling/profile-with-tracing",
 ]
 
 [lib]
 name = "v_frame"
 path = "src/lib.rs"
-
-[[bench]]
-name = "bench"
-path = "benches/bench.rs"
-harness = false
 
 [dependencies.aligned-vec]
 version = ">=0.5.0, <0.7"
@@ -53,19 +42,7 @@
 [dependencies.num-traits]
 version = "0.2"
 
-[dependencies.profiling]
-version = "1"
-optional = true
-
 [dependencies.serde]
 version = "1.0"
 features = ["derive"]
 optional = true
-
-[dependencies.tracing]
-version = "0.1.40"
-optional = true
-
-[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.criterion]
-version = "0.6"
-
