--- fern-0.6.2/Cargo.toml	1970-01-01T00:00:01+00:00
+++ fern-0.6.2/Cargo.toml	2024-11-06T12:34:13.746940+00:00
@@ -39,9 +39,6 @@
 
 [package.metadata.docs.rs]
 all-features = true
-
-[[example]]
-name = "cmd-program"
 
 [[example]]
 name = "date-based-file-log"
@@ -86,9 +83,6 @@
 [dependencies.log]
 version = "0.4"
 features = ["std"]
-
-[dev-dependencies.clap]
-version = "2.22"
 
 [dev-dependencies.humantime]
 version = "2.1.0"
