diff --git a/CPP/7zip/Archive/LzhHandler.cpp b/CPP/7zip/Archive/LzhHandler.cpp
index 21631f7..1f14621 100644
--- a/CPP/7zip/Archive/LzhHandler.cpp
+++ b/CPP/7zip/Archive/LzhHandler.cpp
@@ -163,7 +163,8 @@ struct CItem
       return false;
     }
     const Byte *data = (const Byte *)(Extensions[index].Data);
-    value = GetUi32(data);
+    if (!data) value = 0;
+    else value = GetUi32(data);
     return true;
   }

