From dce3659ac3d14ed338cdb37798a429751898c078 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Thu, 23 Apr 2020 18:55:34 -0400
Subject: [PATCH 47/62] Fix the license on our buildid extractor.

Signed-off-by: Peter Jones <pjones@redhat.com>
Upstream-commit-id: 71439f848f6
---
 buildid.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/buildid.c b/buildid.c
index f213f3bc921..6b414cdcffb 100644
--- a/buildid.c
+++ b/buildid.c
@@ -1,8 +1,6 @@
 /*
  * Walk a list of input files, printing the name and buildid of any file
  * that has one.
- *
- * This program is licensed under the GNU Public License version 2.
  */
 
 #include <err.h>
-- 
2.26.2

