From e69d61b2d88d8dc679558dea03c48130127102ac Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Sat, 30 Jun 2018 15:45:24 +0200
Subject: [PATCH] signal: Add missing mutex_lock/unlock around the while loop

This fixes #12.
---
 signal.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/signal.c b/signal.c
index 102c3cf..3096b30 100644
--- a/signal.c
+++ b/signal.c
@@ -63,7 +63,9 @@ void *signal_thread(void *arg)
 			printf("espeakup caught signal %d\n", sig);
 			break;
 		}
+		pthread_mutex_lock(&queue_guard);
 	}
+	pthread_mutex_unlock(&queue_guard);
 	/* Tell the reader to stop, if it is in a select() call. */
 	write(PIPE_WRITE_FD, STOP_MSG, strlen(STOP_MSG));
 	return NULL;
