From b7fe3af320226600c6162252f040b21d6323e07c Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Tue, 14 Mar 2017 21:52:14 -0500
Subject: [PATCH] add unicode variant of /dev/softsynth

---
 softsynth.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/softsynth.c b/softsynth.c
index 3394434..efa2351 100644
--- a/softsynth.c
+++ b/softsynth.c
@@ -235,7 +235,10 @@ int open_softsynth(void)
 	}
 
 	/* open the softsynth. */
-	softFD = open("/dev/softsynth", O_RDWR | O_NONBLOCK);
+	softFD = open("/dev/softsynthu", O_RDWR | O_NONBLOCK);
+	if (softFD < 0 && errno == ENOENT)
+		/* Kernel without unicode support?  Try without unicode.  */
+		softFD = open("/dev/softsynth", O_RDWR | O_NONBLOCK);
 	if (softFD < 0) {
 		perror("Unable to open the softsynth device");
 		rc = -1;
