diff -up conda-4.7.11/conda/base/context.py.cpuinfo conda-4.7.11/conda/base/context.py
--- conda-4.7.11/conda/base/context.py.cpuinfo	2019-08-06 13:23:55.000000000 -0600
+++ conda-4.7.11/conda/base/context.py	2019-08-16 21:31:01.947610667 -0600
@@ -1328,7 +1328,7 @@ conda_tests_ctxt_mgmt_def_pol = replace_
 @memoize
 def _get_cpu_info():
     # DANGER: This is rather slow
-    from .._vendor.cpuinfo import get_cpu_info
+    from cpuinfo import get_cpu_info
     return frozendict(get_cpu_info())

