diff -up conda-4.7.0/conda/gateways/disk/create.py.orig conda-4.7.0/conda/gateways/disk/create.py
--- conda-4.7.0/conda/gateways/disk/create.py.orig	2019-05-17 11:08:48.000000000 -0600
+++ conda-4.7.0/conda/gateways/disk/create.py	2019-05-18 19:46:11.428298209 -0600
@@ -28,7 +28,7 @@ from ...base.context import context
 from ...common.compat import on_win
 from ...common.path import ensure_pad, expand, win_path_double_escape, win_path_ok
 from ...common.serialize import json_dump
-from ...exceptions import BasicClobberError, CondaOSError, maybe_raise
+from ...exceptions import BasicClobberError, CondaOSError, maybe_raise, NotWritableError
 from ...models.enums import FileMode, LinkType
 
 
@@ -438,6 +438,9 @@ def create_package_cache_directory(pkgs_
         sudo_safe = expand(pkgs_dir).startswith(expand('~'))
         touch(join(pkgs_dir, PACKAGE_CACHE_MAGIC_FILE), mkdir=True, sudo_safe=sudo_safe)
         touch(join(pkgs_dir, 'urls'), sudo_safe=sudo_safe)
+    except NotWritableError:
+        log.trace("cannot create package cache directory '%s'", pkgs_dir)
+        return False
     except (IOError, OSError) as e:
         if e.errno in (EACCES, EPERM, EROFS):
             log.trace("cannot create package cache directory '%s'", pkgs_dir)
