--- memory-0.17.0/Data/Memory/Internal/CompatPrim64.hs.orig	2019-09-02 10:59:04.000000000 +0800
+++ memory-0.17.0/Data/Memory/Internal/CompatPrim64.hs	2023-02-17 02:13:44.131991990 +0800
@@ -148,14 +148,14 @@
 w64# w _ _ = w
 
 #elif WORD_SIZE_IN_BITS == 32
-import GHC.IntWord64
+import GHC.Exts
 import GHC.Prim (Word#)
 
-timesWord64# :: Word64# -> Word64# -> Word64#
-timesWord64# a b =
-    let !ai = word64ToInt64# a
-        !bi = word64ToInt64# b
-     in int64ToWord64# (timesInt64# ai bi)
+-- timesWord64# :: Word64# -> Word64# -> Word64#
+-- timesWord64# a b =
+--     let !ai = word64ToInt64# a
+--         !bi = word64ToInt64# b
+--      in int64ToWord64# (timesInt64# ai bi)
 
 w64# :: Word# -> Word# -> Word# -> Word64#
 w64# _ hw lw =
