--- litemap-0.8.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ litemap-0.8.1/Cargo.toml	2025-10-29T09:04:04.148327+00:00
@@ -68,14 +68,6 @@
 bench = false
 
 [[example]]
-name = "language_names_hash_map"
-path = "examples/language_names_hash_map.rs"
-
-[[example]]
-name = "language_names_lite_map"
-path = "examples/language_names_lite_map.rs"
-
-[[example]]
 name = "litemap_bincode"
 path = "examples/litemap_bincode.rs"
 required-features = ["serde"]
@@ -143,6 +135,3 @@
 
 [dev-dependencies.serde_json]
 version = "1.0.45"
-
-[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.criterion]
-version = "0.5.0"
