diff --git a/wsutil/CMakeLists.txt b/wsutil/CMakeLists.txt
index 0367cd1..6382a2c 100644
--- a/wsutil/CMakeLists.txt
+++ b/wsutil/CMakeLists.txt
@@ -69,6 +69,7 @@ set(WSUTIL_PUBLIC_HEADERS
 	ws_mempbrk_int.h
 	ws_pipe.h
 	ws_printf.h
+	wstmpdir.h
 	wsjson.h
 	xtea.h
 )
@@ -118,6 +118,7 @@ set(WSUTIL_COMMON_FILES
 	unicode-utils.c
 	ws_mempbrk.c
 	ws_pipe.c
+	wstmpdir.c
 	wsgcrypt.c
 	wsjson.c
 	xtea.c
