--- open-5.4.2/Cargo.toml	2006-07-24T01:21:28+00:00
+++ open-5.4.2/Cargo.toml	2026-08-27T14:19:57.963337+00:00
@@ -40,7 +40,7 @@
 
 [features]
 insecure = []
-shellexecute-on-windows = ["dep:dunce"]
+shellexecute-on-windows = []
 
 [lib]
 name = "open"
@@ -58,6 +58,3 @@
 [target."cfg(unix)".dependencies.libc]
 version = "0.2"
 
-[target."cfg(windows)".dependencies.dunce]
-version = "1"
-optional = true
