From ea164ceafa05f96079204a3f0ae379e46e64a455 Mon Sep 17 00:00:00 2001
From: Alexander Scheel <ascheel@redhat.com>
Date: Tue, 4 Aug 2020 10:08:15 -0400
Subject: [PATCH] Fix permissions after generating certificates with make

Signed-off-by: Alexander Scheel <ascheel@redhat.com>
---
 raddb/certs/bootstrap | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/raddb/certs/bootstrap b/raddb/certs/bootstrap
index 336a2bd..9920ecf 100755
--- a/raddb/certs/bootstrap
+++ b/raddb/certs/bootstrap
@@ -21,7 +21,10 @@ make -h > /dev/null 2>&1
 #
 if [ "$?" = "0" ]; then
   make all
-  exit $?
+  ret=$?
+  chown root:radiusd dh ca.* client.* server.*
+  chmod 640 dh ca.* client.* server.*
+  exit $ret
 fi
 
 #
-- 
2.26.2

