diff -up xeve-0.5.1/CMakeLists.txt.orig xeve-0.5.1/CMakeLists.txt
--- xeve-0.5.1/CMakeLists.txt.orig	2024-06-24 09:49:08.000000000 +0200
+++ xeve-0.5.1/CMakeLists.txt	2025-05-13 15:28:46.125587162 +0200
@@ -59,15 +59,6 @@ if(NOT SET_PROF)
   set(SET_PROF "MAIN")
 endif()
 
-if(NOT ARM)
-  set(ARM "FALSE")
-else()
-  add_definitions(-DARM=1)
-  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -flax-vector-conversions")
-  set(CMAKE_EXE_LINKER_FLAGS "-static")
-endif()
-message("ARM=${ARM}")
-
 if(("${SET_PROF}" STREQUAL "MAIN") OR ("${SET_PROF}" STREQUAL "BASE"))
   message("SET_PROF=${SET_PROF}")
 else()
@@ -103,8 +94,7 @@ elseif( UNIX OR MINGW )
         set(OPT_DBG "-DNDEBUG") # disable assert
     endif()
 
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OPT_DBG} -${OPT_LV} -fomit-frame-pointer -Wall -Wno-unused-function -Wno-unused-but-set-variable -Wno-unused-variable -Wno-attributes -Werror -Wno-strict-overflow -Wno-unknown-pragmas -Wno-stringop-overflow -std=c99")
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-pointer-sign -pthread -Wno-pointer-to-int-cast -Wno-maybe-uninitialized")
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OPT_DBG} -${OPT_LV} -Wall -std=c99 -pthread")
     set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -lm")
 endif()
 
diff -up xeve-0.5.1/src_base/CMakeLists.txt.orig xeve-0.5.1/src_base/CMakeLists.txt
--- xeve-0.5.1/src_base/CMakeLists.txt.orig	2024-06-24 09:49:08.000000000 +0200
+++ xeve-0.5.1/src_base/CMakeLists.txt	2025-05-13 15:32:27.639434967 +0200
@@ -17,15 +17,25 @@ file (GLOB LIB_NEON_INC "./neon/xeve_*.h
 include(GenerateExportHeader)
 include_directories("${CMAKE_BINARY_DIR}")
 
+set(ARM "FALSE")
+message("SYSTEM_PROCESSOR=${CMAKE_SYSTEM_PROCESSOR}")
+if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "^(arm|aarch)")
+  set(ARM "TRUE")
+elseif(${CMAKE_SYSTEM_PROCESSOR} MATCHES "^(x86|ia64|i386|i686)")
+  set(X86 "TRUE")
+endif()
 
 if("${ARM}" STREQUAL "TRUE")
   add_library( ${LIB_NAME_BASE} STATIC ${LIB_API_SRC} ${XEVE_INC} ${LIB_BASE_SRC} ${LIB_BASE_INC} ${LIB_NEON_INC} ${LIB_NEON_SRC})
   add_library( ${LIB_NAME_BASE}_dynamic SHARED ${LIB_API_SRC} ${XEVE_INC} ${LIB_BASE_SRC} ${LIB_BASE_INC} ${LIB_NEON_INC} ${LIB_NEON_SRC})
-else()
+elseif(X86)
   add_library( ${LIB_NAME_BASE} STATIC ${LIB_API_SRC} ${XEVE_INC} ${LIB_BASE_SRC} ${LIB_BASE_INC} ${LIB_SSE_SRC} ${LIB_SSE_INC}
                                      ${LIB_AVX_SRC} ${LIB_AVX_INC} )
   add_library( ${LIB_NAME_BASE}_dynamic SHARED ${LIB_API_SRC} ${XEVE_INC} ${LIB_BASE_SRC} ${LIB_BASE_INC} ${LIB_SSE_SRC} ${LIB_SSE_INC}
                                              ${LIB_AVX_SRC} ${LIB_AVX_INC} )
+else()
+  add_library( ${LIB_NAME_BASE} STATIC ${LIB_API_SRC} ${XEVE_INC} ${LIB_BASE_SRC} ${LIB_BASE_INC} )
+  add_library( ${LIB_NAME_BASE}_dynamic SHARED ${LIB_API_SRC} ${XEVE_INC} ${LIB_BASE_SRC} ${LIB_BASE_INC} )
 endif()
 
 set_target_properties(${LIB_NAME_BASE}_dynamic PROPERTIES VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR} SOVERSION ${LIB_SOVERSION})
@@ -57,10 +67,12 @@ source_group("base\\avx\\source" FILES $
 source_group("base\\neon\\header" FILES ${LIB_NEON_INC})
 source_group("base\\neon\\source" FILES ${LIB_NEON_SRC})
 
-if("${ARM}" STREQUAL "TRUE")
+if(ARM)
   include_directories( ${LIB_NAME_BASE} PUBLIC . .. ../inc ./neon)
-else()
+elseif(X86)
   include_directories( ${LIB_NAME_BASE} PUBLIC . .. ../inc ./sse ./avx)
+else()
+  include_directories( ${LIB_NAME_BASE} PUBLIC . .. ../inc)
 endif()
 
 set( SSE ${BASE_INC_FILES} ${LIB_SSE_SRC})
@@ -83,7 +95,11 @@ if( MSVC )
                             ARCHIVE_OUTPUT_DIRECTORY  ${CMAKE_CURRENT_BINARY_DIR}/import/lib)
 
 elseif( UNIX OR MINGW )
-  if("${ARM}" STREQUAL "FALSE")
+  if(ARM)
+    add_definitions(-DARM=1)
+    set_property( SOURCE ${NEON} APPEND PROPERTY COMPILE_FLAGS "-flax-vector-conversions")
+  elseif(X86)
+    add_definitions(-DX86=1)
     set_property( SOURCE ${SSE} APPEND PROPERTY COMPILE_FLAGS "-msse4.1" )
     set_property( SOURCE ${AVX} APPEND PROPERTY COMPILE_FLAGS " -mavx2" )
   endif()
diff -up xeve-0.5.1/src_base/xeve_def.h.orig xeve-0.5.1/src_base/xeve_def.h
--- xeve-0.5.1/src_base/xeve_def.h.orig	2024-06-24 09:49:08.000000000 +0200
+++ xeve-0.5.1/src_base/xeve_def.h	2025-05-13 15:34:08.233577988 +0200
@@ -1514,10 +1514,10 @@ enum TQC_RUN {
 #include "xeve_ipred.h"
 #include "xeve_picman.h"
 #include "xeve_mc.h"
-#ifndef ARM
+#if defined(X86)
 #include "xeve_mc_sse.h"
 #include "xeve_mc_avx.h"
-#else
+#elif defined(ARM)
 #include "xeve_mc_neon.h"
 #endif
 #include "xeve_type.h"
diff -up xeve-0.5.1/src_base/xeve_port.h.orig xeve-0.5.1/src_base/xeve_port.h
--- xeve-0.5.1/src_base/xeve_port.h.orig	2024-06-24 09:49:08.000000000 +0200
+++ xeve-0.5.1/src_base/xeve_port.h	2025-05-13 15:33:29.682925990 +0200
@@ -138,10 +138,10 @@ void xeve_trace_line(char * pre);
 #define xeve_assert_gv(x,r,v,g) \
     {if(!(x)){assert(x); (r)=(v); goto g;}}
 
-#ifndef ARM
+#if defined(X86)
 #define X86_SSE                 1
 #define ARM_NEON                0
-#else
+#elif defined (ARM)
 #define X86_SSE                 0
 #define ARM_NEON                1
 #endif
diff -up xeve-0.5.1/src_base/xeve_type.h.orig xeve-0.5.1/src_base/xeve_type.h
--- xeve-0.5.1/src_base/xeve_type.h.orig	2024-06-24 09:49:08.000000000 +0200
+++ xeve-0.5.1/src_base/xeve_type.h	2025-05-13 16:02:07.223774021 +0200
@@ -34,10 +34,10 @@
 #include "xeve_def.h"
 #include "xeve_bsw.h"
 #include "xeve_sad.h"
-#ifndef ARM
+#if defined(X86)
 #include "xeve_sad_sse.h"
 #include "xeve_sad_avx.h"
-#else
+#elif defined(ARM)
 #include "xeve_sad_neon.h"
 #endif
 
@@ -1016,11 +1016,11 @@ typedef struct _ALF_SLICE_PARAM ALF_SLIC
 #include "xeve_tbl.h"
 #include "xeve_itdq.h"
 
-#ifndef ARM
+#if defined(X86)
 #include "xeve_itdq_sse.h"
 #include "xeve_itdq_avx.h"
 #include "xeve_tq_avx.h"
-#else
+#elif defined(ARM)
 #include "xeve_itdq_neon.h"
 #include "xeve_tq_neon.h"
 #endif
