From 1d50318f448b73b072724eb6664d311e6084a446 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 10 Apr 2018 12:36:34 -0400
Subject: [PATCH 01/62] Make some things dprint() instead of console_print()

Signed-off-by: Peter Jones <pjones@redhat.com>
Upstream-commit-id: dad59f8c0f36
---
 shim.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/shim.c b/shim.c
index 00155346c12..ff0817009cd 100644
--- a/shim.c
+++ b/shim.c
@@ -2087,8 +2087,8 @@ static int is_our_path(EFI_LOADED_IMAGE *li, CHAR16 *path, UINTN len)
 	if (!dppath)
 		return 0;
 
-	console_print(L"dppath: %s\n", dppath);
-	console_print(L"path:   %s\n", path);
+	dprint(L"dppath: %s\n", dppath);
+	dprint(L"path:   %s\n", path);
 	if (StrnCaseCmp(dppath, path, len))
 		ret = 0;
 
-- 
2.26.2

