From 5e6e0792cedb3b71cbe061ae56e96906cf710579 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Mon, 18 Nov 2019 13:59:14 -0500
Subject: [PATCH 42/62] Slightly better debugging messages

Signed-off-by: Peter Jones <pjones@redhat.com>
Upstream-commit-id: 173d35fe8f5
---
 shim.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/shim.c b/shim.c
index d7ee2b6de6f..2f7aba07421 100644
--- a/shim.c
+++ b/shim.c
@@ -2459,6 +2459,8 @@ uninstall_shim_protocols(void)
 EFI_STATUS
 shim_init(void)
 {
+	EFI_STATUS efi_status;
+
 	setup_verbosity();
 	dprint(L"%a", shim_version);
 
@@ -2479,7 +2481,12 @@ shim_init(void)
 	}
 
 	hook_exit(systab);
-	return install_shim_protocols();
+
+	efi_status = install_shim_protocols();
+	if (EFI_ERROR(efi_status))
+		perror(L"install_shim_protocols() failed: %r\n", efi_status);
+
+	return efi_status;
 }
 
 void
@@ -2575,13 +2582,12 @@ efi_main (EFI_HANDLE passed_image_handle, EFI_SYSTEM_TABLE *passed_systab)
 	build_cert = shim_cert;
 #endif /* defined(ENABLE_SHIM_CERT) */
 	CHAR16 *msgs[] = {
-		L"import_mok_state() failed\n",
-		L"shim_int() failed\n",
+		L"import_mok_state() failed",
+		L"shim_init() failed",
 		NULL
 	};
 	int msg = 0;
 
-
 	/*
 	 * Set up the shim lock protocol so that grub and MokManager can
 	 * call back in and use shim functions
-- 
2.26.2

