From 7b77bee7966a1aa5f00a9b34aeb7e550bfa47be1 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 22 Jul 2020 23:53:09 -0400
Subject: [PATCH 51/62] MokManager: fix uninitialized value

Signed-off-by: Peter Jones <pjones@redhat.com>
Upstream: pr#212
---
 MokManager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MokManager.c b/MokManager.c
index fa73e2fd865..654a115033c 100644
--- a/MokManager.c
+++ b/MokManager.c
@@ -1431,7 +1431,7 @@ static CHAR16 get_password_charater(CHAR16 * prompt)
 	SIMPLE_TEXT_OUTPUT_MODE SavedMode;
 	EFI_STATUS efi_status;
 	CHAR16 *message[2];
-	CHAR16 character;
+	CHAR16 character = 0;
 	UINTN length;
 	UINT32 pw_length;
 
-- 
2.26.2

