--- ORBit2-2.14.3/orbit2-config.in.multilib	2006-09-05 14:36:59.000000000 -0400
+++ ORBit2-2.14.3/orbit2-config.in	2006-09-05 14:44:29.000000000 -0400
@@ -1,11 +1,11 @@
 #! /bin/sh
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-exec_prefix_set=no
-includedir=@includedir@
-libdir=@libdir@
 [ -z "$PKG_CONFIG" ] && PKG_CONFIG="pkg-config"
+prefix=`${PKG_CONFIG} --variable prefix ORBit-2.0`
+exec_prefix=`${PKG_CONFIG} --variable exec_prefix ORBit-2.0`
+exec_prefix_set=no
+includedir=`${PKG_CONFIG} --variable includedir ORBit-2.0`
+libdir=`${PKG_CONFIG} --variable libdir ORBit-2.0`
 
 usage()
 {
