--- ORBit2-2.14.19/src/idl-compiler/orbit-idl-c-backend.c.test	2010-02-09 07:05:35.000000000 -0500
+++ ORBit2-2.14.19/src/idl-compiler/orbit-idl-c-backend.c	2012-04-17 16:17:50.797480456 -0400
@@ -133,8 +133,8 @@
 	}
 	
 	if (pass == OUTPUT_DEPS) {
-		if (!g_file_test (".deps", G_FILE_TEST_IS_DIR)) {
-			if (g_mkdir (".deps", 0775) < 0) {
+		if (g_mkdir (".deps", 0775) < 0) {
+			if (errno != EEXIST) {
 				g_warning ("failed to create '.deps' directory '%s'",
 					   g_strerror (errno));
 				return NULL;
