--- configure.orig	2026-05-12 00:38:22.393353572 +0000
+++ configure	2026-05-12 00:38:35.945296680 +0000
@@ -20647,6 +20647,12 @@
 fi
 
 
+ac_recvmmsg_save_ext=$ac_ext
+ac_ext=cpp
+ac_recvmmsg_save_compile=$ac_compile
+ac_compile='$CXX -c $CXXFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ac_recvmmsg_save_link=$ac_link
+ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for 'struct timespec' for recvmmsg() const" >&5
 $as_echo_n "checking for 'struct timespec' for recvmmsg() const... " >&6; }
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
@@ -20665,7 +20671,7 @@
   return 0;
 }
 _ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
+if ac_fn_cxx_try_link "$LINENO"; then :
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
@@ -20681,6 +20687,9 @@
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
+ac_ext=$ac_recvmmsg_save_ext
+ac_compile=$ac_recvmmsg_save_compile
+ac_link=$ac_recvmmsg_save_link
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP support" >&5
 $as_echo_n "checking for PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP support... " >&6; }
