--- networkx-networkx-3.6/doc/conf.py.orig	2025-11-24 09:50:58.826778207 -0700
+++ networkx-networkx-3.6/doc/conf.py	2025-11-24 09:51:36.955693196 -0700
@@ -1,7 +1,6 @@
 import os
 from datetime import date
 from sphinx_gallery.sorting import ExplicitOrder, FileNameSortKey
-from intersphinx_registry import get_intersphinx_mapping
 from warnings import filterwarnings
 
 filterwarnings(
@@ -237,21 +236,19 @@ latex_appendices = ["tutorial"]
 
 # Intersphinx mapping
 
-intersphinx_mapping = get_intersphinx_mapping(
-    packages={
-        "python",
-        "numpy",
-        "neps",
-        "matplotlib",
-        "scipy",
-        "pandas",
-        "geopandas",
-        "pygraphviz",
-        "sphinx-gallery",
-        "sympy",
-        "nx-guides",
-    }
-)
+intersphinx_mapping = {
+    "python": ("https://docs.python.org/3/", "/usr/share/doc/python3-docs/html/objects.inv"),
+    "numpy": ("https://numpy.org/doc/stable/", "/usr/share/doc/python3-numpy-doc/objects.inv"),
+    "neps": ("https://numpy.org/neps/", None),
+    "matplotlib": ("https://matplotlib.org/stable/", None),
+    "scipy": ("https://docs.scipy.org/doc/scipy/", None),
+    "pandas": ("https://pandas.pydata.org/pandas-docs/stable/", None),
+    "geopandas": ("https://geopandas.org/en/stable/", None),
+    "pygraphviz": ("https://pygraphviz.github.io/documentation/stable/", "/usr/share/doc/python-pygraphviz/html/objects.inv"),
+    "sphinx-gallery": ("https://sphinx-gallery.github.io/stable/", None),
+    "nx-guides": ("https://networkx.org/nx-guides/", None),
+    "sympy": ("https://docs.sympy.org/latest/", "/usr/share/doc/sympy-doc/html/objects.inv"),
+}
 # NOTE: generally not relevant, but prevents very long build times when other
 # projects' docs sites are not responding
 intersphinx_timeout = 0.5
--- networkx-networkx-3.6/pyproject.toml.orig	2025-11-24 09:26:16.307592379 -0700
+++ networkx-networkx-3.6/pyproject.toml	2025-11-24 09:28:29.355618167 -0700
@@ -78,7 +78,6 @@ doc = [
     'pillow>=10',
     'texext>=0.6.7',
     'myst-nb>=1.1',
-    'intersphinx-registry',
 ]
 example = [
     'contextily>=1.6',
--- networkx-networkx-3.6/requirements/doc.txt.orig	2025-11-23 19:57:00.000000000 -0700
+++ networkx-networkx-3.6/requirements/doc.txt	2025-11-24 09:18:06.233858723 -0700
@@ -7,4 +7,3 @@ numpydoc>=1.8.0
 pillow>=10
 texext>=0.6.7
 myst-nb>=1.1
-intersphinx-registry
