--- cargo-credential-0.4.10/Cargo.toml	2006-07-24T01:21:28+00:00
+++ cargo-credential-0.4.10/Cargo.toml	2026-08-10T11:39:17.976288+00:00
@@ -77,13 +77,6 @@
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.178"
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.61"
-features = [
-    "Win32_System_Console",
-    "Win32_Foundation",
-]
-
 [lints.clippy]
 dbg_macro = "warn"
 disallowed_methods = "warn"
@@ -104,3 +97,4 @@
 
 [lints.rustdoc]
 private_intra_doc_links = "allow"
+
