--- onenote_parser-0.3.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ onenote_parser-0.3.1/Cargo.toml	2024-09-14T02:42:06.679130+00:00
@@ -31,10 +31,10 @@
 version = "0.8.31"
 
 [dependencies.enum-primitive-derive]
-version = "0.2.2"
+version = "0.3.0"
 
 [dependencies.itertools]
-version = "0.9.0"
+version = "0.13.0"
 
 [dependencies.num-traits]
 version = "0.2"
@@ -49,7 +49,7 @@
 version = "0.8"
 
 [dependencies.widestring]
-version = "0.4.3"
+version = "1.1.0"
 
 [dev-dependencies.insta]
 version = "1.21.1"
