diff -up rcs-5.9.4/tests/known-failures.in.t810_disable rcs-5.9.4/tests/known-failures.in
--- rcs-5.9.4/tests/known-failures.in.t810_disable	2018-03-27 14:48:15.344270665 +0200
+++ rcs-5.9.4/tests/known-failures.in	2018-03-27 14:48:55.305831852 +0200
@@ -134,6 +134,7 @@ exit 0
   ;; t805 under OSX (see "bletcherous hack" above)
 
 * 5.9.4
+  t810 ; ‘ci -l -d -T’ => RCS file mtime set to epoch
 
 * end matter
 *** Local variables:
