diff -up gnome-doc-utils-0.14.0/xslt/docbook/html/db2html-css.xsl.package gnome-doc-utils-0.14.0/xslt/docbook/html/db2html-css.xsl
--- gnome-doc-utils-0.14.0/xslt/docbook/html/db2html-css.xsl.package	2008-09-22 16:06:09.000000000 -0400
+++ gnome-doc-utils-0.14.0/xslt/docbook/html/db2html-css.xsl	2008-11-29 11:35:54.000000000 -0500
@@ -465,6 +465,7 @@ span.medialabel { font-style: italic; }
 span.methodname { font-family: monospace; }
 span.ooclass, span.ooexception, span.oointerface { font-family: monospace; }
 span.option { font-family: monospace; }
+span.package { font-family: monospace; }
 span.parameter { font-family: monospace; }
 span.paramdef span.parameter { font-style: italic; }
 span.prompt { font-family: monospace; }
diff -up gnome-doc-utils-0.14.0/xslt/docbook/html/db2html-inline.xsl.package gnome-doc-utils-0.14.0/xslt/docbook/html/db2html-inline.xsl
--- gnome-doc-utils-0.14.0/xslt/docbook/html/db2html-inline.xsl.package	2008-09-22 16:06:09.000000000 -0400
+++ gnome-doc-utils-0.14.0/xslt/docbook/html/db2html-inline.xsl	2008-11-29 11:35:54.000000000 -0500
@@ -669,6 +669,11 @@ FIXME
   <xsl:call-template name="db2html.inline"/>
 </xsl:template>
 
+<!-- = package = -->
+<xsl:template match="package">
+  <xsl:call-template name="db2html.inline"/>
+</xsl:template>
+
 <!-- = pagenums = -->
 <xsl:template match="pagenums">
   <xsl:call-template name="db2html.inline"/>
