--- tuned-2.15.0/Makefile.old	2020-12-17 17:27:53.000000000 +0000
+++ tuned-2.15.0/Makefile	2021-08-13 18:29:54.179771406 +0000
@@ -213,13 +213,6 @@
 		$(call install_python_script, \
 			$(file), $(DESTDIR)/usr/libexec/tuned/$(notdir $(file))))
 
-	# icon
-	install -Dpm 0644 icons/tuned.svg $(DESTDIR)$(DATADIR)/icons/hicolor/scalable/apps/tuned.svg
-
-	# desktop file
-	install -dD $(DESTDIR)$(DATADIR)/applications
-	desktop-file-install --dir=$(DESTDIR)$(DATADIR)/applications tuned-gui.desktop
-
 clean: clean-html
 	find -name "*.pyc" | xargs rm -f
 	rm -rf $(VERSIONED_NAME) rpm-build-dir
