--- Makefile.PL
+++ Makefile.PL
@@ -1,6 +1,6 @@
 use strict;
 use warnings;
-use ExtUtils::MakeMaker::CPANfile;
+use ExtUtils::MakeMaker 6.30;
 
 my %params = (
     NAME          => 'Archive::Any::Lite',
@@ -8,6 +8,15 @@
     VERSION_FROM  => 'lib/Archive/Any/Lite.pm',
     ABSTRACT_FROM => 'lib/Archive/Any/Lite.pm',
     LICENSE       => 'perl',
+    PREREQ_PM     => {
+        'Archive::Tar'            => '1.76',
+        'Archive::Zip'            => '0',
+        'File::Spec'              => '0',
+        'IO::Uncompress::Bunzip2' => '0',
+        'IO::Zlib'                => '0',
+        'Test::More'              => '0.47',
+        'Test::UseAllModules'     => '0.10',
+    },
     META_MERGE => {
        resources => {
            repository => 'https://github.com/charsbar/archive-any-lite',
@@ -15,4 +24,7 @@
     }
 );
 
+my $eumm = $ExtUtils::MakeMaker::VERSION;
+delete $params{LICENSE} if $eumm < 6.31;
+
 WriteMakefile(%params);
