--- PullParser2.1.10/src/java/impl/node/org/gjt/xpp/impl/node/Node.java	2017-09-11 18:39:27.105544537 +0200
+++ PullParser2.1.10/src/java/impl/node/org/gjt/xpp/impl/node/Node.java	2017-09-11 18:41:06.771360115 +0200
@@ -273,9 +273,9 @@
         oneChild = null;
 
         if(children != null && children.size() > 0) {
-            Enumeration enum = children.elements();
-            while (enum.hasMoreElements()) {
-                Object child = enum.nextElement();
+            Enumeration emun = children.elements();
+            while (emun.hasMoreElements()) {
+                Object child = emun.nextElement();
                 if(child instanceof XmlNode) {
                     ((XmlNode)child).setParentNode(null);
                 }
@@ -373,11 +373,11 @@
         if(declaredNs != null) {
             System.arraycopy(declaredNs, 0, uris, off, len);
         }
-        //    Enumeration enum = prefix2Ns.keys();
+        //    Enumeration emun = prefix2Ns.keys();
         //    int i = off;
-        //    while (enum.hasMoreElements() && i < off + len)
+        //    while (emun.hasMoreElements() && i < off + len)
         //    {
-        //        uris[i++] = (String) prefix2Ns.get( enum.nextElement() );
+        //        uris[i++] = (String) prefix2Ns.get( emun.nextElement() );
         //    }
     }
 
@@ -512,10 +512,10 @@
                 buf.append(declaredNs[i]);
                 buf.append("'");
             }
-            //        Enumeration enum = prefix2Ns.keys();
-            //        while (enum.hasMoreElements())
+            //        Enumeration emun = prefix2Ns.keys();
+            //        while (emun.hasMoreElements())
             //        {
-            //              Object key = enum.nextElement();
+            //              Object key = emun.nextElement();
             //              buf.append("xmlns:");
             //              buf.append(key);
             //              buf.append("='");
--- PullParser2.1.10/src/java/tests/node/NodeTest.java	2017-09-11 18:39:27.109544537 +0200
+++ PullParser2.1.10/src/java/tests/node/NodeTest.java	2017-09-11 18:43:53.211362957 +0200
@@ -386,20 +386,20 @@
         assertEquals("", t1.getDefaultNamespaceUri());
 
         // check enumeration stuff...
-        Enumeration enum = t1.children();
+        Enumeration emun = t1.children();
         assertEquals(0, t1.getChildrenCountSoFar());
         //child1, child2...
-        assertTrue(enum.hasMoreElements());
-        child1 = (XmlPullNode) enum.nextElement();
+        assertTrue(emun.hasMoreElements());
+        child1 = (XmlPullNode) emun.nextElement();
         assertEquals("child1", child1.getLocalName());
-        assertNotNull(enum.nextElement());
-        child3 = (XmlPullNode) enum.nextElement();
+        assertNotNull(emun.nextElement());
+        child3 = (XmlPullNode) emun.nextElement();
         assertEquals("child3", child3.getLocalName());
-        assertNotNull(enum.nextElement());
+        assertNotNull(emun.nextElement());
         assertEquals(4, t1.getChildrenCountSoFar());
-        assertEquals(true, enum.hasMoreElements());
-        assertNotNull(enum.nextElement());
-        assertEquals(false, enum.hasMoreElements());
+        assertEquals(true, emun.hasMoreElements());
+        assertNotNull(emun.nextElement());
+        assertEquals(false, emun.hasMoreElements());
 
         XmlPullNode t2 = (XmlPullNode) node.getChildAt(1);
         assertEquals("t2", t2.getLocalName());
