diff -up sblim-cmpi-sysfs-1.2.0/util/Linux_SysfsDeviceUtil.c.orig sblim-cmpi-sysfs-1.2.0/util/Linux_SysfsDeviceUtil.c
--- sblim-cmpi-sysfs-1.2.0/util/Linux_SysfsDeviceUtil.c.orig	2009-05-22 04:45:48.000000000 +0200
+++ sblim-cmpi-sysfs-1.2.0/util/Linux_SysfsDeviceUtil.c	2011-11-03 13:35:01.437175460 +0100
@@ -91,10 +91,10 @@ CMPIInstance * Linux_SysfsDeviceUtil_mak
 void * Linux_SysfsDeviceUtil_beginEnumeration( void * location )
 {
    FILE * handle;
-   char * findcommand = malloc(strlen((char *)location)+38); 
+   char * findcommand = malloc(strlen((char *)location)+41);
 
    /* Find all the instance names in the target location */
-   sprintf(findcommand, "find %s -maxdepth 1 -mindepth 1 -type d", (char *)location);
+   sprintf(findcommand, "find -L %s -maxdepth 1 -mindepth 1 -type d", (char *)location);
    handle = popen(findcommand,"r");
    free(findcommand); 
    return handle; 
