diff -up sblim-cmpi-sysfs-1.2.0/provider-register.sh.orig sblim-cmpi-sysfs-1.2.0/provider-register.sh
--- sblim-cmpi-sysfs-1.2.0/provider-register.sh.orig	2013-08-12 13:31:00.938253054 +0200
+++ sblim-cmpi-sysfs-1.2.0/provider-register.sh	2013-08-12 13:31:10.905304210 +0200
@@ -213,7 +213,7 @@ pegasus_install()
     then
 	chatter Registering providers with $state cimserver
 	$CIMMOF -uc -I $mofpath -n root/cimv2 $mymofs &&
-	$CIMMOF -uc -n root/PG_Interop $_REGFILENAME
+	$CIMMOF -uc -n root/interop $_REGFILENAME
     else
 	echo "Failed to build pegasus registration MOF." >&2
 	return 1
