diff --git a/pyproject.toml b/pyproject.toml
index 79f088a..aede1df 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=42", "wheel", "setuptools_scm[toml]>=3.4.1"]
+requires = ["setuptools>=39.2.0", "wheel", "setuptools_scm>=1.15.7"]
 build-backend = "setuptools.build_meta"
 
 [tool.black]
diff --git a/setup.cfg b/setup.cfg
index 6f537b8..f0620d3 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -25,7 +25,7 @@ install_requires =
 	SecretStorage>=3.2; sys_platform=="linux"
 	jeepney>=0.4.2; sys_platform=="linux"
 	importlib_metadata >= 1; python_version < "3.8"
-setup_requires = setuptools_scm[toml] >= 3.4.1
+setup_requires = setuptools_scm >= 1.15.7
 
 [options.extras_require]
 testing = 
diff --git a/setup.py b/setup.py
index bac24a4..827e955 100644
--- a/setup.py
+++ b/setup.py
@@ -3,4 +3,4 @@
 import setuptools
 
 if __name__ == "__main__":
-    setuptools.setup()
+    setuptools.setup(use_scm_version=True)
