diff -rupN --no-dereference scotch-v6.1.2/src/esmumps/Makefile scotch-v6.1.2-new/src/esmumps/Makefile
--- scotch-v6.1.2/src/esmumps/Makefile	2021-11-13 11:07:54.000000000 +0100
+++ scotch-v6.1.2-new/src/esmumps/Makefile	2021-11-13 21:44:21.098507670 +0100
@@ -196,7 +196,7 @@ symbol_fax_graph$(OBJ)		:	symbol_fax_gra
 					fax.h
 
 libesmumps$(LIB)		:	$(LIBESMUMPSDEPS)
-					$(AR) $(ARFLAGS) $(@) $(?)
+					$(AR) $(ARFLAGS) $(@) $(?) $(LDFLAGS)
 					-$(RANLIB) $(@)
 
 main_esmumps$(EXE)		:	main_esmumps.c				\
diff -rupN --no-dereference scotch-v6.1.2/src/libscotch/Makefile scotch-v6.1.2-new/src/libscotch/Makefile
--- scotch-v6.1.2/src/libscotch/Makefile	2021-11-13 11:07:54.000000000 +0100
+++ scotch-v6.1.2-new/src/libscotch/Makefile	2021-11-13 21:44:21.098507670 +0100
@@ -3119,26 +3119,26 @@ ptscotchf.h			:	ptdummysizes$(EXE)			\
 					library_pt_f.h
 					./ptdummysizes$(EXE) "-s$(SCOTCH_NAME_SUFFIX)" library_pt_f.h ptscotchf.h
 
-libscotch$(LIB)			:	$(LIBSCOTCHDEPS)
-					$(AR) $(ARFLAGS) $(@) $(?)
+libscotch$(LIB)			:	$(LIBSCOTCHDEPS) libscotcherr$(LIB)
+					$(AR) $(ARFLAGS) $(@) $(?) $(LDFLAGS) -L . -lscotcherr
 					-$(RANLIB) $(@)
 
 libscotcherr$(LIB)		:	library_error$(OBJ)
-					$(AR) $(ARFLAGS) $(@) $(?)
+					$(AR) $(ARFLAGS) $(@) $(?) $(LDFLAGS)
 					-$(RANLIB) $(@)
 
 libscotcherrexit$(LIB)		:	library_error_exit$(OBJ)
-					$(AR) $(ARFLAGS) $(@) $(?)
+					$(AR) $(ARFLAGS) $(@) $(?) $(LDFLAGS)
 					-$(RANLIB) $(@)
 
-libptscotch$(LIB)		:	$(LIBPTSCOTCHDEPS)
-					$(AR) $(ARFLAGS) $(@) $(?)
+libptscotch$(LIB)		:	$(LIBPTSCOTCHDEPS) libscotch$(LIB)
+					$(AR) $(ARFLAGS) $(@) $(?) $(LDFLAGS) -L . -lscotch
 					-$(RANLIB) $(@)
 
 libptscotcherr$(LIB)		:	library_error_pt$(OBJ)
-					$(AR) $(ARFLAGS) $(@) $(?)
+					$(AR) $(ARFLAGS) $(@) $(?) $(LDFLAGS)
 					-$(RANLIB) $(@)
 
 libptscotcherrexit$(LIB)	:	library_error_exit_pt$(OBJ)
-					$(AR) $(ARFLAGS) $(@) $(?)
+					$(AR) $(ARFLAGS) $(@) $(?) $(LDFLAGS)
 					-$(RANLIB) $(@)
diff -rupN --no-dereference scotch-v6.1.2/src/libscotchmetis/Makefile scotch-v6.1.2-new/src/libscotchmetis/Makefile
--- scotch-v6.1.2/src/libscotchmetis/Makefile	2021-11-13 11:07:54.000000000 +0100
+++ scotch-v6.1.2-new/src/libscotchmetis/Makefile	2021-11-13 21:44:21.099507671 +0100
@@ -187,7 +187,7 @@ libptscotchparmetis$(LIB)	:	parmetis_dgr
 					parmetis_dgraph_order_f$(OBJ)		\
 					parmetis_dgraph_part$(OBJ)		\
 					parmetis_dgraph_part_f$(OBJ)
-					$(AR) $(ARFLAGS) $(@) $(^)
+					$(AR) $(ARFLAGS) $(@) $(^) $(LDFLAGS) -L ../libscotch -lptscotch
 					-$(RANLIB) $(@)
 
 libscotchmetis$(LIB)		:	metis_graph_dual$(OBJ)			\
@@ -200,5 +200,5 @@ libscotchmetis$(LIB)		:	metis_graph_dual
 					metis_graph_part_dual_f$(OBJ)		\
 					metis_options$(OBJ)			\
 					metis_options_f$(OBJ)
-					$(AR) $(ARFLAGS) $(@) $(^)
+					$(AR) $(ARFLAGS) $(@) $(^) $(LDFLAGS) -L ../libscotch -lscotch
 					-$(RANLIB) $(@)
