diff -up xevd-0.5.0/src_base/neon/xevd_dbk_neon.h.orig xevd-0.5.0/src_base/neon/xevd_dbk_neon.h
--- xevd-0.5.0/src_base/neon/xevd_dbk_neon.h.orig	2024-04-20 14:10:26.000000000 +0000
+++ xevd-0.5.0/src_base/neon/xevd_dbk_neon.h	2025-05-10 20:32:17.488740239 +0000
@@ -28,7 +28,7 @@
    POSSIBILITY OF SUCH DAMAGE.
 */
 
-#ifndef _XEVD_DBK_NOEN_H_
+#ifndef _XEVD_DBK_NEON_H_
 #define _XEVD_DBK_NEON_H_
 
 
