diff -up ming-0.4.5/util/ming-config.in.multilib ming-0.4.5/util/ming-config.in
--- ming-0.4.5/util/ming-config.in.multilib	2010-10-10 09:19:23.000000000 +0200
+++ ming-0.4.5/util/ming-config.in	2014-05-27 19:01:32.183275535 +0200
@@ -33,7 +33,7 @@ while test $# -gt 0; do
 		echo -I@includedir@
 		;;
 	    --libs)
-		echo -L@libdir@ @ZLIB@ @MATHLIB@ @GIFLIB@ @PNGLIB@ -lming
+		echo @ZLIB@ @MATHLIB@ @GIFLIB@ @PNGLIB@ $(pkg-config --libs libming)
 		;;
 	    --bindir)
 		echo @bindir@
