diff -up libming-ming-0_4_7/src/actioncompiler/Makefile.am.pmake libming-ming-0_4_7/src/actioncompiler/Makefile.am
--- libming-ming-0_4_7/src/actioncompiler/Makefile.am.pmake	2015-05-15 11:43:14.000000000 +0200
+++ libming-ming-0_4_7/src/actioncompiler/Makefile.am	2016-07-18 22:57:44.537713157 +0200
@@ -56,11 +56,11 @@ lex.swf5.c: $(srcdir)/swf5compiler.flex
 swf4compiler.tab.c: $(srcdir)/swf4compiler.y
 	$(YACC) -p swf4 -b swf4compiler $(srcdir)/swf4compiler.y
 
-swf4compiler.tab.h: $(srcdir)/swf4compiler.y
+swf4compiler.tab.h: $(srcdir)/swf4compiler.y | swf4compiler.tab.c
 	$(YACC) --defines $(DEBUG) -p swf4 -b swf4compiler $(srcdir)/swf4compiler.y
 
 swf5compiler.tab.c: $(srcdir)/swf5compiler.y
 	$(YACC) -p swf5 -b swf5compiler $(srcdir)/swf5compiler.y
 
-swf5compiler.tab.h: $(srcdir)/swf5compiler.y
+swf5compiler.tab.h: $(srcdir)/swf5compiler.y | swf5compiler.tab.c
 	$(YACC) --defines $(DEBUG) -p swf5 -b swf5compiler $(srcdir)/swf5compiler.y
