diff -up dialog-1.2-20121230/dialog-config.in.multilib dialog-1.2-20121230/dialog-config.in
--- dialog-1.2-20121230/dialog-config.in.multilib	2013-03-18 15:43:10.639021412 +0100
+++ dialog-1.2-20121230/dialog-config.in	2013-03-18 15:44:08.519065774 +0100
@@ -33,7 +33,7 @@ exec_prefix="@exec_prefix@"
 datarootdir="@datarootdir@"
 
 bindir="@bindir@"
-libdir="@libdir@"
+libdir=
 datadir="@datadir@"
 mandir="@mandir@"
 includedir="@includedir@"
@@ -76,7 +76,7 @@ ENDECHO
 		;;
 	--libs)
 		sed -e 's,^[ ]*,,' -e 's, [ ]*, ,g' -e 's,[ ]*$,,' <<-ENDECHO
-			-L${exec_prefix}/lib -l${THIS} @LIBS@
+			-l${THIS}
 ENDECHO
 		;;
 	# identification
