diff -urN gst-plugins-base-1.20.4_orig/gst-libs/gst/audio/gstaudiodecoder.c gst-plugins-base-1.20.4/gst-libs/gst/audio/gstaudiodecoder.c
--- gst-plugins-base-1.20.4_orig/gst-libs/gst/audio/gstaudiodecoder.c	2022-10-12 21:09:51.000000000 +0530
+++ gst-plugins-base-1.20.4/gst-libs/gst/audio/gstaudiodecoder.c	2022-11-23 01:56:53.433818224 +0530
@@ -1245,8 +1245,7 @@
   const GstMetaInfo *info = (*meta)->info;
   gboolean do_copy = FALSE;
 
-  if (gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory)
-      || gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory_reference)) {
+  if (gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory)) {
     /* never call the transform_meta with memory specific metadata */
     GST_DEBUG_OBJECT (decoder, "not copying memory specific metadata %s",
         g_type_name (info->api));
diff -urN gst-plugins-base-1.20.4_orig/gst-libs/gst/audio/gstaudioencoder.c gst-plugins-base-1.20.4/gst-libs/gst/audio/gstaudioencoder.c
--- gst-plugins-base-1.20.4_orig/gst-libs/gst/audio/gstaudioencoder.c	2022-10-12 21:09:51.000000000 +0530
+++ gst-plugins-base-1.20.4/gst-libs/gst/audio/gstaudioencoder.c	2022-11-23 01:57:05.041755487 +0530
@@ -706,8 +706,7 @@
   const GstMetaInfo *info = (*meta)->info;
   gboolean do_copy = FALSE;
 
-  if (gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory)
-      || gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory_reference)) {
+  if (gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory)) {
     /* never call the transform_meta with memory specific metadata */
     GST_DEBUG_OBJECT (encoder, "not copying memory specific metadata %s",
         g_type_name (info->api));
diff -urN gst-plugins-base-1.20.4_orig/gst-libs/gst/video/gstvideodecoder.c gst-plugins-base-1.20.4/gst-libs/gst/video/gstvideodecoder.c
--- gst-plugins-base-1.20.4_orig/gst-libs/gst/video/gstvideodecoder.c	2022-10-12 21:09:51.000000000 +0530
+++ gst-plugins-base-1.20.4/gst-libs/gst/video/gstvideodecoder.c	2022-11-23 01:56:24.669973458 +0530
@@ -3359,8 +3359,7 @@
   const GstMetaInfo *info = (*meta)->info;
   gboolean do_copy = FALSE;
 
-  if (gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory)
-      || gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory_reference)) {
+  if (gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory)) {
     /* never call the transform_meta with memory specific metadata */
     GST_DEBUG_OBJECT (decoder, "not copying memory specific metadata %s",
         g_type_name (info->api));
diff -urN gst-plugins-base-1.20.4_orig/gst-libs/gst/video/gstvideoencoder.c gst-plugins-base-1.20.4/gst-libs/gst/video/gstvideoencoder.c
--- gst-plugins-base-1.20.4_orig/gst-libs/gst/video/gstvideoencoder.c	2022-10-12 21:09:51.000000000 +0530
+++ gst-plugins-base-1.20.4/gst-libs/gst/video/gstvideoencoder.c	2022-11-23 01:56:40.953885616 +0530
@@ -2198,8 +2198,7 @@
   const GstMetaInfo *info = (*meta)->info;
   gboolean do_copy = FALSE;
 
-  if (gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory)
-      || gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory_reference)) {
+  if (gst_meta_api_type_has_tag (info->api, _gst_meta_tag_memory)) {
     /* never call the transform_meta with memory specific metadata */
     GST_DEBUG_OBJECT (encoder, "not copying memory specific metadata %s",
         g_type_name (info->api));
