--- actix-web-4.13.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ actix-web-4.13.0/Cargo.toml	2026-03-24T15:06:18.160920+00:00
@@ -47,11 +47,6 @@
 features = [
     "macros",
     "openssl",
-    "rustls-0_20",
-    "rustls-0_21",
-    "rustls-0_22",
-    "rustls-0_23",
-    "compress-brotli",
     "compress-gzip",
     "compress-zstd",
     "cookies",
@@ -88,10 +83,6 @@
 __tls = []
 compat = ["compat-routing-macros-force-pub"]
 compat-routing-macros-force-pub = ["actix-web-codegen?/compat-routing-macros-force-pub"]
-compress-brotli = [
-    "actix-http/compress-brotli",
-    "__compress",
-]
 compress-gzip = [
     "actix-http/compress-gzip",
     "__compress",
@@ -103,7 +94,6 @@
 cookies = ["dep:cookie"]
 default = [
     "macros",
-    "compress-brotli",
     "compress-gzip",
     "compress-zstd",
     "cookies",
@@ -113,7 +103,6 @@
     "ws",
 ]
 experimental-introspection = ["serde/derive"]
-experimental-io-uring = ["actix-server/io-uring"]
 http2 = ["actix-http/http2"]
 macros = [
     "dep:actix-macros",
@@ -126,35 +115,6 @@
     "actix-tls/accept",
     "actix-tls/openssl",
 ]
-rustls = ["rustls-0_20"]
-rustls-0_20 = [
-    "__tls",
-    "http2",
-    "actix-http/rustls-0_20",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_20",
-]
-rustls-0_21 = [
-    "__tls",
-    "http2",
-    "actix-http/rustls-0_21",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_21",
-]
-rustls-0_22 = [
-    "__tls",
-    "http2",
-    "actix-http/rustls-0_22",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_22",
-]
-rustls-0_23 = [
-    "__tls",
-    "http2",
-    "actix-http/rustls-0_23",
-    "actix-tls/accept",
-    "actix-tls/rustls-0_23",
-]
 secure-cookies = [
     "cookies",
     "cookie/secure",
@@ -209,15 +169,6 @@
 path = "examples/worker-cpu-pin.rs"
 
 [[test]]
-name = "compression"
-path = "tests/compression.rs"
-required-features = [
-    "compress-brotli",
-    "compress-gzip",
-    "compress-zstd",
-]
-
-[[test]]
 name = "introspection"
 path = "tests/introspection.rs"
 
@@ -232,16 +183,6 @@
 [[test]]
 name = "test_httpserver"
 path = "tests/test_httpserver.rs"
-
-[[test]]
-name = "test_server"
-path = "tests/test_server.rs"
-required-features = [
-    "compress-brotli",
-    "compress-gzip",
-    "compress-zstd",
-    "cookies",
-]
 
 [[test]]
 name = "test_streaming_response"
@@ -419,9 +360,6 @@
 version = "3"
 features = ["openssl"]
 
-[dev-dependencies.brotli]
-version = "8"
-
 [dev-dependencies.const-str]
 version = "0.5"
 
