Supply dummy descriptions which, due to typos, are required.

diff -up MigrationTools-47/migrate_common.ph.patch5 MigrationTools-47/migrate_common.ph
--- MigrationTools-47/migrate_common.ph.patch5	2006-01-25 05:18:16.000000000 +0100
+++ MigrationTools-47/migrate_common.ph	2008-08-29 09:09:19.000000000 +0200
@@ -83,7 +83,7 @@ $DEFAULT_BASE = "dc=padl,dc=com";
 ##### CUT HERE #####
 # where /etc/mail/ldapdomains contains names of ldap_routed
 # domains (similiar to MASQUERADE_DOMAIN_FILE).
-$DEFAULT_MAIL_HOST = "mail.padl.com";
+# $DEFAULT_MAIL_HOST = "mail.padl.com";
 
 # turn this on to support more general object clases
 # such as person.
@@ -120,10 +120,10 @@ if (!defined($DEFAULT_BASE)) {
 }
 
 # Default Kerberos realm
-if ($EXTENDED_SCHEMA) {
-	$DEFAULT_REALM = $DEFAULT_MAIL_DOMAIN;
-	$DEFAULT_REALM =~ tr/a-z/A-Z/;
-}
+#if ($EXTENDED_SCHEMA) {
+#	$DEFAULT_REALM = $DEFAULT_MAIL_DOMAIN;
+#	$DEFAULT_REALM =~ tr/a-z/A-Z/;
+#}
 
 if (-x "/usr/sbin/revnetgroup") {
 	$REVNETGROUP = "/usr/sbin/revnetgroup";
diff -up MigrationTools-47/migrate_protocols.pl.patch5 MigrationTools-47/migrate_protocols.pl
--- MigrationTools-47/migrate_protocols.pl.patch5	2008-08-29 09:09:19.000000000 +0200
+++ MigrationTools-47/migrate_protocols.pl	2008-08-29 09:09:19.000000000 +0200
@@ -76,6 +76,7 @@ sub dump_protocol
 	foreach $_ (@aliases) {
 		print $HANDLE "cn: $_\n";
 	}
+	print $HANDLE "description: IP protocol $number ($name)\n";
 	print $HANDLE "\n";
 }
 
diff -up MigrationTools-47/migrate_rpc.pl.patch5 MigrationTools-47/migrate_rpc.pl
--- MigrationTools-47/migrate_rpc.pl.patch5	2008-08-29 09:09:19.000000000 +0200
+++ MigrationTools-47/migrate_rpc.pl	2008-08-29 09:09:19.000000000 +0200
@@ -77,6 +77,7 @@ sub dump_rpc
 	foreach $_ (@aliases) {
 		print $HANDLE "cn: $_\n";
 	}
+	print $HANDLE "description: ONC RPC number $rpcnumber ($rpcname)\n";
 	print $HANDLE "\n";
 }
 
