diff -up MigrationTools-47/migrate_services.pl.patch7 MigrationTools-47/migrate_services.pl
--- MigrationTools-47/migrate_services.pl.patch7	2008-08-29 09:09:19.000000000 +0200
+++ MigrationTools-47/migrate_services.pl	2008-08-29 09:09:19.000000000 +0200
@@ -102,6 +102,12 @@ sub parse_services
 		my ($servicename, $portproto, @aliases) = split(m/\s+/o, $card);
 		my ($rawport, $proto) = split(m#[/,]#o, $portproto);
 
+		# do not generate ddp services - it's AppleTalk, not IP
+                if ($proto eq "ddp") { 
+                    print STDERR "Skipping non-IP service '$servicename $portproto'\n";
+                    next; 
+                }
+
 		# Find services specifying a port range (e.g. X11.)
 		my $loport = '';
 		my $hiport = '';
