--- hickory-proto-0.24.4/Cargo.toml	1970-01-01T00:00:01+00:00
+++ hickory-proto-0.24.4/Cargo.toml	2025-09-18T09:03:40.565657+00:00
@@ -250,14 +250,6 @@
     "macros",
 ]
 
-[dev-dependencies.tracing-subscriber]
-version = "0.3"
-features = [
-    "std",
-    "fmt",
-    "env-filter",
-]
-
 [features]
 backtrace = ["dep:backtrace"]
 default = ["tokio-runtime"]
@@ -321,6 +313,7 @@
 testing = []
 text-parsing = []
 tokio-runtime = [
+    "tokio/io-util",
     "tokio/net",
     "tokio/rt",
     "tokio/time",
