commit c035a2754be2c9cdf7d802554d26e2171b25ee07
Author: rpm-build <rpm-build>
Date:   Fri Jan 24 14:50:51 2025 +0100

    f42-fix-ftbfs.patch

diff --git a/sa.c b/sa.c
index e2e654e..a7f1cf1 100644
--- a/sa.c
+++ b/sa.c
@@ -1506,7 +1506,7 @@ void print_user_list(void)
   Remember to correct the number of elements to adjust... */
 
   qsort (entry_array + 1, (size_t) num_users - 1,
-         sizeof (struct hashtab_elem *), (int (*)()) compare_user_entry);
+         sizeof (struct hashtab_elem *), (int (*)(const void*, const void*)) compare_user_entry);
 
   /* Now we've got a sorted list of user entries. */
 
@@ -1686,7 +1686,7 @@ void print_command_list(void)
      than the number of commands */
 
   qsort (entry_array + 1, (size_t) num_commands - 1,
-         sizeof (struct hashtab_entry *), (int (*)()) compare_sum_entry);
+         sizeof (struct hashtab_entry *), (int (*)(const void*, const void*)) compare_sum_entry);
 
   for (temp = 0; temp < num_commands; temp++)
     {
