diff --git a/Makefile b/Makefile
index 11195a1cb..7c65c1c25 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ UNAME_S := $(shell uname -s)
 
 SLOW_SPEC_THRESHOLD=120
 
-GINKGO_FLAGS ?= -r -randomizeAllSpecs -requireSuite
+GINKGO_FLAGS ?= -r -randomizeAllSpecs -requireSuite -mod=vendor
 GINKGO_INT_FLAGS = $(GINKGO_FLAGS) -slowSpecThreshold $(SLOW_SPEC_THRESHOLD)
 ginkgo_int = ginkgo $(GINKGO_INT_FLAGS)
 
