--- uu_sort-0.7.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ uu_sort-0.7.0/Cargo.toml	2026-03-16T18:26:41.913424+00:00
@@ -123,17 +123,12 @@
 ]
 package = "uucore"
 
-[dev-dependencies.divan]
-version = "4.0.5"
-package = "codspeed-divan-compat"
-
 [dev-dependencies.tempfile]
 version = "3.15.0"
 
 [dev-dependencies.uucore]
 version = "0.7.0"
 features = [
-    "benchmark",
     "fs",
     "parser-size",
     "version-cmp",
