--- a/opam
+++ b/opam
@@ -12,9 +12,7 @@ depends: [
   "ocamlfind" {build}
   "ocamlbuild" {build}
   "topkg" {build & >= "0.9.0"}
-  # Can be removed once ocaml >= 4.07
   "seq"
-  "stdlib-shims"
 ]
 depopts: [ "base-unix" "cmdliner" ]
 conflicts: [ "cmdliner" {< "0.9.8"} ]
--- a/pkg/META
+++ b/pkg/META
@@ -1,6 +1,6 @@
 description = "OCaml Format pretty-printer combinators"
 version = "%%VERSION_NUM%%"
-requires = "seq stdlib-shims"
+requires = "seq"
 archive(byte) = "fmt.cma"
 archive(native) = "fmt.cmxa"
 plugin(byte) = "fmt.cma"
--- a/_tags
+++ b/_tags
@@ -1,4 +1,4 @@
-true : bin_annot, safe_string, package(seq), package(stdlib-shims)
+true : bin_annot, safe_string, package(seq)
 <_b0> : -traverse
 <src> : include
 <src/fmt_tty*> : package(unix)
