--- tboot-1.9.7/Config.mk	2020-03-24 22:12:28.096834349 +0000
+++ tboot-1.9.7/Config.mk	2020-03-24 22:12:51.324779226 +0000
@@ -43,7 +43,7 @@
 	            -Wextra -Winit-self -Wswitch-default -Wunused-parameter \
 	            -Wwrite-strings \
 	            $(call cc-option,$(CC),-Wlogical-op,) \
-	            -Wno-missing-field-initializers
+	            -Wno-missing-field-initializers -Wno-address-of-packed-member
 
 AS         = as
 LD         = ld
