From: Hiram Chirino <chirino@gmail.com>
Date: Fri, 5 Jul 2013 18:32:28 +0400
Subject: [PATCH] Allow leveldbjni build
 
 
diff --git a/include/leveldb/slice.h b/include/leveldb/slice.h
index 2df417d..1af5635 100644
--- a/include/leveldb/slice.h
+++ b/include/leveldb/slice.h
@@ -86,7 +86,6 @@ class LEVELDB_EXPORT Slice {
     return ((size_ >= x.size_) && (memcmp(data_, x.data_, x.size_) == 0));
   }
 
- private:
   const char* data_;
   size_t size_;
 };
