diff -up sip-4.18/siputils.py.no_rpath sip-4.18/siputils.py
--- sip-4.18/siputils.py.no_rpath	2015-03-25 06:00:24.000000000 -0500
+++ sip-4.18/siputils.py	2016-04-13 11:34:48.179894616 -0500
@@ -435,7 +435,7 @@ class Makefile:
             if l_dir in ("", ".", ".."):
                 continue
 
-            rpaths.append(l)
+            #rpaths.append(l)
 
         if self._python:
             incdir.append(self.config.py_inc_dir)
@@ -612,7 +612,7 @@ class Makefile:
             # Handle library directories.
             libdir_qt = self.optional_list("LIBDIR_QT")
             libdir.extend(libdir_qt)
-            rpaths.extend(libdir_qt)
+            #rpaths.extend(libdir_qt)
 
             if qt_version >= 0x040000:
                 # Try and read QT_LIBINFIX from qconfig.pri.
