diff -up libXaw3d-1.6.1/src/AsciiSink.c.fontset libXaw3d-1.6.1/src/AsciiSink.c
--- libXaw3d-1.6.1/src/AsciiSink.c.fontset	2012-02-01 12:24:00.000000000 -0700
+++ libXaw3d-1.6.1/src/AsciiSink.c	2012-02-25 14:52:46.802779081 -0700
@@ -513,6 +513,8 @@ Initialize(Widget request, Widget new, A
 {
     AsciiSinkObject sink = (AsciiSinkObject) new;
 
+    if (!sink->ascii_sink.font) XtError("Aborting: no font found\n");
+
     GetGC(sink);
 
     sink->ascii_sink.insertCursorOn= CreateInsertCursor(XtScreenOfObject(new));
diff -up libXaw3d-1.6.1/src/Command.c.fontset libXaw3d-1.6.1/src/Command.c
--- libXaw3d-1.6.1/src/Command.c.fontset	2011-09-19 15:42:36.000000000 -0600
+++ libXaw3d-1.6.1/src/Command.c	2012-02-25 14:51:26.075372823 -0700
@@ -218,6 +218,8 @@ Initialize(Widget request, Widget new, A
   CommandWidget cbw = (CommandWidget) new;
   int shape_event_base, shape_error_base;
 
+  if (!cbw->label.font) XtError("Aborting: no font found\n");
+
   if (cbw->command.shape_style != XawShapeRectangle
       && !XShapeQueryExtension(XtDisplay(new), &shape_event_base,
 			       &shape_error_base))
diff -up libXaw3d-1.6.1/src/List.c.fontset libXaw3d-1.6.1/src/List.c
--- libXaw3d-1.6.1/src/List.c.fontset	2011-10-09 12:01:20.000000000 -0600
+++ libXaw3d-1.6.1/src/List.c	2012-02-25 14:53:16.327465260 -0700
@@ -362,6 +362,8 @@ Initialize(Widget junk, Widget new, ArgL
 {
     ListWidget lw = (ListWidget) new;
 
+    if (!lw->list.font) XtError("Aborting: no font found\n");
+
 /*
  * Initialize all private resources.
  */
diff -up libXaw3d-1.6.1/src/SmeBSB.c.fontset libXaw3d-1.6.1/src/SmeBSB.c
--- libXaw3d-1.6.1/src/SmeBSB.c.fontset	2011-09-19 15:42:36.000000000 -0600
+++ libXaw3d-1.6.1/src/SmeBSB.c	2012-02-25 14:50:19.056358107 -0700
@@ -207,6 +207,8 @@ Initialize(Widget request, Widget new, A
     else
 	entry->sme_bsb.label = XtNewString( entry->sme_bsb.label );
 
+    if (!entry->sme_bsb.font) XtError("Aborting: no font found\n");
+
     CreateGCs(new);
 
     GetBitmapInfo(new, TRUE);	/* Left Bitmap Info */
diff -up libXaw3d-1.6.1/src/Tip.c.fontset libXaw3d-1.6.1/src/Tip.c
--- libXaw3d-1.6.1/src/Tip.c.fontset	2012-02-15 12:10:38.000000000 -0700
+++ libXaw3d-1.6.1/src/Tip.c	2012-02-25 14:50:19.058358017 -0700
@@ -248,6 +248,8 @@ XawTipInitialize(Widget req, Widget w, A
     TipWidget tip = (TipWidget)w;
     XGCValues values;
 
+    if (!tip->tip.font) XtError("Aborting: no font found\n");
+
     tip->tip.timer = 0;
 
     values.foreground = tip->tip.foreground;
