From 799bf9088c131fc71626a48e9987e4d44a2f0194 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 24 Jun 2014 10:00:15 +0100
Subject: [PATCH 1/4] Don't add rpaths to libraries.
 
---
 configure.ac | 2 --
 1 file changed, 2 deletions(-)
 
diff --git a/configure.ac b/configure.ac
index b81da53c42..892a2a894f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1107,8 +1107,6 @@ AS_IF([test x"$enable_shared" != "xno"],
            [[*-*-openbsd7.[3-9]|*-*-openbsd[89].*]],
            [mkdll_flags="${mkdll_flags} -Wl,--no-execute-only"])
       oc_ldflags="$oc_ldflags -Wl,-E"
-      rpath="-Wl,-rpath,"
-      mksharedlibrpath="-Wl,-rpath,"
       natdynlinkopts="-Wl,-E"
       supports_shared_libraries=true],
       [mkdll='shared-libs-not-available'])
-- 
