From a367c2691bf39bc4509c441b6f634538508d8de8 Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Tue, 26 Jul 2022 01:32:46 -0400
Subject: [PATCH 3/7] Disable warnings-as-errors in tests

Signed-off-by: Elliott Sales de Andrade <quantum.analyst@gmail.com>
---
 pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index bafc6a0b..9d313003 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -134,7 +134,6 @@ addopts = '''
 -p no:asyncio
 -p no:legacypath'''
 filterwarnings = [
-    "error",
     # xref: https://github.com/jupyter/jupyter_core/pull/292
     # xref: https://github.com/jupyter/jupyter_core/issues/309
     '''ignore:Jupyter is migrating its paths to use standard platformdirs''',
-- 
2.52.0

