Index: src/manifest
===================================================================
--- src/manifest.orig
+++ src/manifest
@@ -4,3 +4,28 @@ Implementation-Version: 1.7.7.1
 Implementation-Title: Mozilla Rhino 1.7.7.1
 Implementation-Vendor: Mozilla Foundation
 Implementation-URL: http://www.mozilla.org/rhino
+Bundle-Vendor: %Bundle-Vendor.0
+Bundle-Localization: plugin
+Bundle-RequiredExecutionEnvironment: J2SE-1.5
+Bundle-Name: %Bundle-Name.0
+Bundle-SymbolicName: org.mozilla.javascript
+Bundle-Version: 1.7.12
+Export-Package: org.mozilla.classfile,org.mozilla.javascript,org.mozil
+ la.javascript.debug,org.mozilla.javascript.jdk13,org.mozilla.javascri
+ pt.jdk15,org.mozilla.javascript.optimizer,org.mozilla.javascript.rege
+ xp,org.mozilla.javascript.serialize,org.mozilla.javascript.tools,org.
+ mozilla.javascript.tools.debugger,org.mozilla.javascript.tools.debugg
+ er.treetable,org.mozilla.javascript.tools.idswitch,org.mozilla.javasc
+ ript.tools.jsc,org.mozilla.javascript.tools.shell,org.mozilla.javascr
+ ipt.xml,org.mozilla.javascript.xml.impl.xmlbeans,org.mozilla.javascri
+ pt.xmlimpl
+Bundle-ManifestVersion: 2
+Import-Package: javax.swing;resolution:=optional,javax.swing.border;re
+ solution:=optional,javax.swing.event;resolution:=optional,javax.swing
+ .filechooser;resolution:=optional,javax.swing.table;resolution:=optio
+ nal,javax.swing.text;resolution:=optional,javax.swing.tree;resolution
+ :=optional,javax.xml.namespace;resolution:=optional,javax.xml.parsers
+ ;resolution:=optional,javax.xml.transform;resolution:=optional,javax.
+ xml.transform.dom;resolution:=optional,javax.xml.transform.stream;res
+ olution:=optional,org.w3c.dom;resolution:=optional,org.xml.sax;resolu
+ tion:=optional
