diff -up cups-1.5b1/backend/usb-unix.c.direct-usb cups-1.5b1/backend/usb-unix.c
--- cups-1.5b1/backend/usb-unix.c.direct-usb	2011-05-20 05:49:49.000000000 +0200
+++ cups-1.5b1/backend/usb-unix.c	2011-05-23 17:52:14.000000000 +0200
@@ -102,6 +102,9 @@ print_device(const char *uri,		/* I - De
              _cups_strncasecmp(hostname, "Minolta", 7);
 #endif /* __FreeBSD__ || __NetBSD__ || __OpenBSD__ || __DragonFly__ */
 
+    if (use_bc && !strncmp(uri, "usb:/dev/", 9))
+      use_bc = 0;
+
     if ((device_fd = open_device(uri, &use_bc)) == -1)
     {
       if (getenv("CLASS") != NULL)
@@ -331,12 +334,7 @@ open_device(const char *uri,		/* I - Dev
   if (!strncmp(uri, "usb:/dev/", 9))
 #ifdef __linux
   {
-   /*
-    * Do not allow direct devices anymore...
-    */
-
-    errno = ENODEV;
-    return (-1);
+    return (open(uri + 4, O_RDWR | O_EXCL));
   }
   else if (!strncmp(uri, "usb://", 6))
   {
