From 3d6d821d07ff4ff5d4390b5f0754d5c1cf3139ee Mon Sep 17 00:00:00 2001
From: Sam Meluch <sammeluch@microsoft.com>
Date: Mon, 1 May 2023 19:22:25 -0700
Subject: [PATCH] Initial timeout patch

---
 client/remoterepo.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/client/remoterepo.c b/client/remoterepo.c
index de0e04c..333647e 100644
--- a/client/remoterepo.c
+++ b/client/remoterepo.c
@@ -142,6 +142,9 @@ TDNFDownloadFile(
     dwError = _handle_curl_cb(pTdnf, pCurl, pszFileUrl);
     BAIL_ON_TDNF_ERROR(dwError);
 
+    /* set curlRetry for and add tag to re-set all curl opts, per documenation */
+    int curlRetry = 0;
+curlCall:
     dwError = TDNFRepoGetUserPass(pTdnf, pszRepo, &pszUserPass);
     BAIL_ON_TDNF_ERROR(dwError);
 
@@ -180,10 +183,21 @@ TDNFDownloadFile(
         BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
     }
 
+    /* double the timeout on curl connect requests */
+    dwError = curl_easy_setopt(pCurl, CURLOPT_CONNECTTIMEOUT, 600);
+    BAIL_ON_TDNF_CURL_ERROR(dwError);
+
     dwError = curl_easy_setopt(pCurl, CURLOPT_WRITEDATA, fp);
     BAIL_ON_TDNF_CURL_ERROR(dwError);
 
     dwError = curl_easy_perform(pCurl);
+    /* give one retry on connection failure */
+    if (!curlRetry && dwError == 28) 
+    {
+        curlRetry = 1;
+        goto curlCall;
+    }
+
     BAIL_ON_TDNF_CURL_ERROR(dwError);
 
     dwError = curl_easy_getinfo(pCurl,
-- 
2.25.1

