From cfc660cfd919d256306700c54059a0518f5c2ded Mon Sep 17 00:00:00 2001
From: Sam Meluch <sammeluch@microsoft.com>
Date: Mon, 8 May 2023 14:57:38 -0700
Subject: [PATCH] Add retry logic for connection failure

---
 client/remoterepo.c | 61 +++++++++++++++++++++++++++++----------------
 1 file changed, 40 insertions(+), 21 deletions(-)

diff --git a/client/remoterepo.c b/client/remoterepo.c
index de0e04c..7d37cc2 100644
--- a/client/remoterepo.c
+++ b/client/remoterepo.c
@@ -173,32 +173,51 @@ TDNFDownloadFile(
         }
     }
 
-    fp = fopen(pszFile, "wb");
-    if(!fp)
+    /* Double the connection timeout */
+    dwError = curl_easy_setopt(pCurl, CURLOPT_CONNECTTIMEOUT, 600);
+    BAIL_ON_TDNF_CURL_ERROR(dwError);
+
+    /* Add retry logic for connection failure*/
+    int retries = 3;
+    for( int i = 0; i < retries; i++)
     {
-        dwError = errno;
-        BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
-    }
+        fp = fopen(pszFile, "wb");
+        if(!fp)
+        {
+            dwError = errno;
+            BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+        }
 
-    dwError = curl_easy_setopt(pCurl, CURLOPT_WRITEDATA, fp);
-    BAIL_ON_TDNF_CURL_ERROR(dwError);
+        dwError = curl_easy_setopt(pCurl, CURLOPT_WRITEDATA, fp);
+        BAIL_ON_TDNF_CURL_ERROR(dwError);
 
-    dwError = curl_easy_perform(pCurl);
-    BAIL_ON_TDNF_CURL_ERROR(dwError);
+        dwError = curl_easy_perform(pCurl);
+        if (i < (retries - 1) && dwError == CURLE_OPERATION_TIMEDOUT)
+        {
+            printf("Attempt %d timed out. Retrying Connection.\n", i + 1);
+            /* Retry on Connection timeout failure */
+            fclose(fp);
+            fp = NULL;
+            continue;
+        } 
 
-    dwError = curl_easy_getinfo(pCurl,
-                                CURLINFO_RESPONSE_CODE,
-                                &lStatus);
-    BAIL_ON_TDNF_CURL_ERROR(dwError);
+        BAIL_ON_TDNF_CURL_ERROR(dwError);
 
-    if(lStatus >= 400)
-    {
-        fprintf(stderr,
-                "Error: %ld when downloading %s\n. Please check repo url.\n",
-                lStatus,
-                pszFileUrl);
-        dwError = ERROR_TDNF_INVALID_PARAMETER;
-        BAIL_ON_TDNF_ERROR(dwError);
+        dwError = curl_easy_getinfo(pCurl,
+                                    CURLINFO_RESPONSE_CODE,
+                                    &lStatus);
+        BAIL_ON_TDNF_CURL_ERROR(dwError);
+
+        if(lStatus >= 400)
+        {
+            fprintf(stderr,
+                    "Error: %ld when downloading %s\n. Please check repo url.\n",
+                    lStatus,
+                    pszFileUrl);
+            dwError = ERROR_TDNF_INVALID_PARAMETER;
+            BAIL_ON_TDNF_ERROR(dwError);
+        }
+        break;
     }
 cleanup:
     TDNF_SAFE_FREE_MEMORY(pszUserPass);
-- 
2.25.1

