From c3af36673da8d7a92477407acec513950a90d706 Mon Sep 17 00:00:00 2001
From: Siddharth Chintamaneni <siddharthc@microsoft.com>
Date: Fri, 22 Aug 2025 21:11:47 +0000
Subject: [PATCH] [PATCH] Add installonlypkgs to tdnf.conf for mariner kernel
 packages

---
 etc/tdnf/tdnf.conf | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/etc/tdnf/tdnf.conf b/etc/tdnf/tdnf.conf
index a10bcbe..fc7860e 100644
--- a/etc/tdnf/tdnf.conf
+++ b/etc/tdnf/tdnf.conf
@@ -1,6 +1,8 @@
 [main]
 gpgcheck=1
 installonly_limit=3
+installonlypkgs=kernel kernel-mshv kernel-uvm kernel-uki kernel-64k kernel-hwe
 clean_requirements_on_remove=0
 repodir=/etc/yum.repos.d
 cachedir=/var/cache/tdnf
+plugins=1
-- 
2.45.4

