diff --git a/client/rpmtrans.c b/client/rpmtrans.c
index d776d30..ca90fa7 100644
--- a/client/rpmtrans.c
+++ b/client/rpmtrans.c
@@ -584,9 +584,15 @@ TDNFTransAddInstallPkg(
                   fp,
                   pszFilePath,
                   &rpmHeader);
-    //If not checking gpg sigs, ignore signature errors
-    if(!nGPGSigCheck && (dwError == RPMRC_NOTTRUSTED || dwError == RPMRC_NOKEY))
-    {
+    if(dwError == RPMRC_NOTTRUSTED || dwError == RPMRC_NOKEY)
+    {
+        // If not checking gpg sigs, ignore signature errors.
+        //
+        // If checking GPG signatures, ignore signatures errors.
+        // These errors can be ignored safely as the gpg signatures have already been validated on file
+        // in the above `TDNFGPGCheck` call. If an RPM with an invalid signature is provided,
+        // the code will never reach this point.
+        // TDNF's gpg keyring is a superset of RPM's built in one, and any gpg files provided in the repo config.
         dwError = 0;
     }
     BAIL_ON_TDNF_RPM_ERROR(dwError);
