--- CBC.pm_ori	2023-08-15 12:28:25.470000005 +0200
+++ CBC.pm	2023-08-15 13:26:19.790000020 +0200
@@ -661,7 +661,7 @@
 	croak "Ciphertext does not begin with a valid header for 'randomiv' header mode" unless defined $self->{iv};
 	croak "randomiv header mode cannot be used securely when decrypting with a >8 byte block cipher.\n"
 	    unless $self->blocksize == 8;
-	(undef,$self->{key}) = $self->pbkdf_obj->key_and_iv(undef,$self->{passphrase});
+	($self->{key},undef) = $self->pbkdf_obj->key_and_iv(undef,$self->{passphrase});
 	substr($$input_stream,0,16) = ''; # truncate
     }
 
