diff -up sblim-sfcb-1.4.9/control.h.orig sblim-sfcb-1.4.9/control.h
--- sblim-sfcb-1.4.9/control.h.orig	2014-11-25 02:43:10.000000000 +0100
+++ sblim-sfcb-1.4.9/control.h	2020-02-12 12:14:46.078102974 +0100
@@ -28,7 +28,7 @@ int             getControlUNum(char *id,
 int             getControlULong(char *id, unsigned long *val);
 int             getControlNum(char *id, long *val);
 int             getControlBool(char *id, int *val);
-const char      * sfcBrokerStart;
+extern const char      * sfcBrokerStart;
 
 #endif
 /* MODELINES */
diff -up sblim-sfcb-1.4.9/providerMgr.c.orig sblim-sfcb-1.4.9/providerMgr.c
--- sblim-sfcb-1.4.9/providerMgr.c.orig	2020-02-12 12:14:46.065102946 +0100
+++ sblim-sfcb-1.4.9/providerMgr.c	2020-02-12 12:19:15.879677891 +0100
@@ -53,6 +53,8 @@
 #define SFCB_ASM(x)
 #endif
 
+sigset_t mask, old_mask;
+
 static pthread_mutex_t resultsocketMutex = PTHREAD_MUTEX_INITIALIZER;
 
 extern CMPIBroker *Broker;
diff -up sblim-sfcb-1.4.9/providerMgr.h.orig sblim-sfcb-1.4.9/providerMgr.h
--- sblim-sfcb-1.4.9/providerMgr.h.orig	2014-11-25 02:43:12.000000000 +0100
+++ sblim-sfcb-1.4.9/providerMgr.h	2020-02-12 12:14:46.078102974 +0100
@@ -398,7 +398,7 @@ BinResponseHdr **invokeProviders(BinRequ
 BinResponseHdr *invokeProvider(BinRequestContext * ctx);
 void            freeResponseHeaders(BinResponseHdr ** resp,
                                     BinRequestContext * ctx);
-sigset_t mask, old_mask;
+extern sigset_t mask, old_mask;
 
 #endif
 /* MODELINES */
diff -up sblim-sfcb-1.4.9/sfcBroker.c.orig sblim-sfcb-1.4.9/sfcBroker.c
--- sblim-sfcb-1.4.9/sfcBroker.c.orig	2014-11-25 02:43:13.000000000 +0100
+++ sblim-sfcb-1.4.9/sfcBroker.c	2020-02-12 12:22:21.768073986 +0100
@@ -52,6 +52,8 @@
 #include "config.h"
 #endif
 
+const char    *sfcBrokerStart;
+
 int             sfcBrokerPid = 0;
 
 extern int      sfcbUseSyslog;
diff -up sblim-sfcb-1.4.9/trace.c.orig sblim-sfcb-1.4.9/trace.c
--- sblim-sfcb-1.4.9/trace.c.orig	2014-11-25 02:43:17.000000000 +0100
+++ sblim-sfcb-1.4.9/trace.c	2020-02-12 12:15:19.319173804 +0100
@@ -48,6 +48,7 @@
  * ---------------------------------------------------------------------------
  */
 
+int             colorTrace;
 char           *processName = NULL;
 int             providerProcess = 0;
 int             idleThreadId = 0;
diff -up sblim-sfcb-1.4.9/trace.h.orig sblim-sfcb-1.4.9/trace.h
--- sblim-sfcb-1.4.9/trace.h.orig	2014-11-25 02:43:17.000000000 +0100
+++ sblim-sfcb-1.4.9/trace.h	2020-02-12 12:14:46.078102974 +0100
@@ -130,7 +130,7 @@ typedef struct traceId {
 #define CYAN            6
 #define WHITE           7
 void            changeTextColor(int reset);
-int             colorTrace;
+extern int             colorTrace;
 
 #define MAX_MSG_SIZE 1024       /* max length of trace message */
 
