From: Peter Lemenkov <lemenkov@gmail.com>
Date: Fri, 8 Jan 2016 16:31:12 +0300
Subject: [PATCH] Allow overriding some more Makefile variables

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/detok/Makefile b/detok/Makefile
index 235966a..4f64445 100644
--- a/detok/Makefile
+++ b/detok/Makefile
@@ -24,13 +24,13 @@
 
 PROGRAM = detok
 
-DESTDIR = /usr/local
+DESTDIR ?= /usr/local
 CC      ?= gcc
-STRIP	= strip
+STRIP	?= strip
 INCLUDES = -I../shared
 
 # Normal Flags:
-CFLAGS  = -O2 -Wall #-Wextra
+CFLAGS  ?= -O2 -Wall #-Wextra
 LDFLAGS = 
 
 # Coverage:
diff --git a/romheaders/Makefile b/romheaders/Makefile
index cdb3d51..c1e873f 100644
--- a/romheaders/Makefile
+++ b/romheaders/Makefile
@@ -24,10 +24,10 @@
 
 PROGRAM = romheaders
 
-DESTDIR  = /usr/local
+DESTDIR  ?= /usr/local
 CC	 ?= gcc
-STRIP    = strip
-CFLAGS   = -O2 -Wall -Wextra
+STRIP    ?= strip
+CFLAGS   ?= -O2 -Wall -Wextra
 INCLUDES = -I../shared
 
 SOURCES = romheaders.c ../shared/classcodes.c
diff --git a/toke/Makefile b/toke/Makefile
index 0b9c928..3306549 100644
--- a/toke/Makefile
+++ b/toke/Makefile
@@ -24,13 +24,13 @@
 
 PROGRAM = toke
 
-DESTDIR = /usr/local
+DESTDIR ?= /usr/local
 CC      ?= gcc
-STRIP	= strip
+STRIP	?= strip
 INCLUDES = -I../shared
 
 # Normal flags
-CFLAGS  = -O2 -Wall #-Wextra 
+CFLAGS  ?= -O2 -Wall #-Wextra 
 LDFLAGS =
 
 # Coverage:
