--- encoding_rs-0.8.35/Cargo.toml	1970-01-01T00:00:01+00:00
+++ encoding_rs-0.8.35/Cargo.toml	2024-11-07T17:36:18.359884+00:00
@@ -38,6 +38,12 @@
 ]
 license = "(Apache-2.0 OR MIT) AND BSD-3-Clause"
 repository = "https://github.com/hsivonen/encoding_rs"
+exclude = [
+    "/ci/",
+    "/doc/",
+    "/*.py",
+    "/rustfmt.toml",
+]
 
 [profile.release]
 lto = true
@@ -45,10 +51,6 @@
 [lib]
 name = "encoding_rs"
 path = "src/lib.rs"
-
-[dependencies.any_all_workaround]
-version = "0.1.0"
-optional = true
 
 [dependencies.cfg-if]
 version = "1.0"
@@ -84,4 +86,3 @@
 less-slow-big5-hanzi-encode = []
 less-slow-gb-hanzi-encode = []
 less-slow-kanji-encode = []
-simd-accel = ["any_all_workaround"]
