diff -up sblim-cmpi-nfsv3-1.1.1/Makefile.in.orig sblim-cmpi-nfsv3-1.1.1/Makefile.in
--- sblim-cmpi-nfsv3-1.1.1/Makefile.in.orig	2013-08-08 13:37:54.938681724 +0200
+++ sblim-cmpi-nfsv3-1.1.1/Makefile.in	2013-08-08 13:39:28.279163344 +0200
@@ -284,7 +284,7 @@ datadir = @datadir@
 datarootdir = @datarootdir@
 
 # docdir defines where the documentation goes
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
