From 40af5eb2ede53e7b2e504d8049a17f95b0e45bbf Mon Sep 17 00:00:00 2001
From: Lumir Balhar <lbalhar@redhat.com>
Date: Fri, 27 Nov 2020 11:12:39 +0100
Subject: [PATCH] fix resolve chaining

---
 dns/message.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dns/message.py b/dns/message.py
index 9824e70..7b99431 100644
--- a/dns/message.py
+++ b/dns/message.py
@@ -776,6 +776,8 @@ class QueryMessage(Message):
                     except KeyError:
                         # Exit the chaining loop
                         break
+                else:
+                    count += 1
         if count >= MAX_CHAIN:
             raise ChainTooLong
         if self.rcode() == dns.rcode.NXDOMAIN and rrset is not None:
-- 
2.28.0

