diff --git a/cmd/ctl/pkg/create/certificaterequest/certificaterequest.go b/cmd/ctl/pkg/create/certificaterequest/certificaterequest.go
index bc95f8fd..166a1689 100644
--- a/cmd/ctl/pkg/create/certificaterequest/certificaterequest.go
+++ b/cmd/ctl/pkg/create/certificaterequest/certificaterequest.go
@@ -22,7 +22,6 @@ import (
 	"errors"
 	"fmt"
 	"io/ioutil"
-	"os"
 	"time"
 
 	"github.com/spf13/cobra"
@@ -362,7 +361,7 @@ func fetchCertificateFromCR(req *cmapi.CertificateRequest, certFileName string)
 	}
 
 	// Store certificate to file
-	err := os.WriteFile(certFileName, req.Status.Certificate, 0600)
+	err := ioutil.WriteFile(certFileName, req.Status.Certificate, 0600)
 	if err != nil {
 		return fmt.Errorf("error when writing certificate to file: %w", err)
 	}
diff --git a/cmd/ctl/pkg/create/certificatesigningrequest/certificatesigningrequest.go b/cmd/ctl/pkg/create/certificatesigningrequest/certificatesigningrequest.go
index 82a78e3e..98834937 100644
--- a/cmd/ctl/pkg/create/certificatesigningrequest/certificatesigningrequest.go
+++ b/cmd/ctl/pkg/create/certificatesigningrequest/certificatesigningrequest.go
@@ -21,7 +21,7 @@ import (
 	"encoding/pem"
 	"errors"
 	"fmt"
-	"os"
+	"io/ioutil"
 	"strconv"
 	"time"
 
@@ -257,7 +257,7 @@ func (o *Options) Run(ctx context.Context, args []string) error {
 	if o.KeyFilename != "" {
 		keyFileName = o.KeyFilename
 	}
-	if err := os.WriteFile(keyFileName, keyPEM, 0600); err != nil {
+	if err := ioutil.WriteFile(keyFileName, keyPEM, 0600); err != nil {
 		return fmt.Errorf("error when writing private key to file: %s", err)
 	}
 	fmt.Fprintf(o.Out, "Private key written to file %s\n", keyFileName)
@@ -430,7 +430,7 @@ func storeCertificate(req *certificatesv1.CertificateSigningRequest, fileName st
 	}
 
 	// Store certificate to file
-	err := os.WriteFile(fileName, req.Status.Certificate, 0600)
+	err := ioutil.WriteFile(fileName, req.Status.Certificate, 0600)
 	if err != nil {
 		return fmt.Errorf("error when writing certificate to file: %s", err)
 	}
