--- reqsign-http-send-reqwest-4.0.7/Cargo.toml	2006-07-24T01:21:28+00:00
+++ reqsign-http-send-reqwest-4.0.7/Cargo.toml	2026-09-26T06:25:05.377762+00:00
@@ -78,12 +78,3 @@
     "rt-multi-thread",
 ]
 
-[target.'cfg(target_arch = "wasm32")'.dependencies.asyncband]
-version = "0.7.1"
-features = [
-    "mutex",
-    "oneshot",
-]
-
-[target.'cfg(target_arch = "wasm32")'.dependencies.wasm-bindgen-futures]
-version = "0.4"
