--- plexus-containers-plexus-containers-1.7.1/plexus-component-metadata/src/main/java/org/codehaus/plexus/metadata/ClassComponentDescriptorExtractor.java	2016-08-31 21:01:49.000000000 +0200
+++ plexus-containers-plexus-containers-1.7.1/plexus-component-metadata/src/main/java/org/codehaus/plexus/metadata/ClassComponentDescriptorExtractor.java	2019-03-07 07:59:40.878835988 +0100
@@ -23,7 +23,6 @@
 import java.util.List;
 import java.util.Map;
 
-import org.apache.maven.plugin.MojoExecutionException;
 import org.codehaus.plexus.component.annotations.Component;
 import org.codehaus.plexus.component.repository.ComponentDescriptor;
 import org.codehaus.plexus.metadata.gleaner.AnnotationComponentGleaner;
@@ -95,15 +94,8 @@
         // Add the projects dependencies
         for ( String filename : elements )
         {
-            try
-            {
                 list.add( new File( filename ).toURI().toURL() );
             }
-            catch ( MalformedURLException e )
-            {
-                throw new MojoExecutionException( "Invalid classpath entry: " + filename, e );
-            }
-        }
 
         URL[] urls = list.toArray( new URL[list.size()] );
 
