Index: libthai-0.1.24/doc/Doxyfile.in
===================================================================
--- libthai-0.1.24.orig/doc/Doxyfile.in
+++ libthai-0.1.24/doc/Doxyfile.in
@@ -1124,7 +1124,7 @@ HTML_HEADER            =
 # that doxygen normally uses.
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
-HTML_FOOTER            =
+HTML_FOOTER            = no_date_footer.html
 
 # The HTML_STYLESHEET tag can be used to specify a user-defined cascading style
 # sheet that is used by each HTML page. It can be used to fine-tune the look of
Index: libthai-0.1.24/doc/no_date_footer.html
===================================================================
--- /dev/null
+++ libthai-0.1.24/doc/no_date_footer.html
@@ -0,0 +1,7 @@
+<hr size="1"><address style="text-align: right;"><small>
+Generated for $projectname by&nbsp;<a href="http://www.doxygen.org/
+index.html"><img src="doxygen.png" alt="doxygen" align="middle" border="0"></a> 
+$doxygenversion</small></address>
+</body>
+</html>
+
