From 6de291458cbab99bcc317e282c16e1523d6de9b8 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 10 Aug 2016 17:12:39 -0400
Subject: [PATCH 03/29] gcc: don't error on stuff in includes.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 Make.defaults | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Make.defaults b/Make.defaults
index c97b452..3511080 100644
--- a/Make.defaults
+++ b/Make.defaults
@@ -19,7 +19,7 @@ PKG_CONFIG = $(CROSS_COMPILE)pkg-config
 CC	:= $(if $(filter default,$(origin CC)),$(CROSS_COMPILE)gcc,$(CC))
 CCLD	:= $(if $(filter undefined,$(origin CCLD)),$(CC),$(CCLD))
 CFLAGS	?= -O0 -g3 -fvar-tracking -fvar-tracking-assignments \
-	   -Wall -Werror -Wextra
+	   -Wall -Werror -Wextra -Wno-error=cpp
 AS	:= $(CROSS_COMPILE)as
 AR	:= $(CROSS_COMPILE)gcc-ar
 RANLIB	:= $(CROSS_COMPILE)gcc-ranlib
-- 
2.13.4

