From 799808b265ac6f82fa1268fd696d70357acce69c Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 25 Apr 2017 16:15:07 -0400
Subject: [PATCH 10/29] signerInfos: make sure err is always initialized

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/signed_data.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/signed_data.c b/src/signed_data.c
index 721db90..9e0af23 100644
--- a/src/signed_data.c
+++ b/src/signed_data.c
@@ -132,7 +132,8 @@ int
 generate_signerInfo_list(cms_context *cms, SpcSignerInfo ***signerInfo_list_p, SignerInfoType type)
 {
 	SpcSignerInfo **signerInfo_list;
-	int err, rc;
+	int err = 0;
+	int rc;
 
 	if (!signerInfo_list_p)
 		return -1;
-- 
2.13.4

