From 868b42b338d919917ea31cfbf0f96e9586947eaf Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 25 Apr 2017 16:23:36 -0400
Subject: [PATCH 11/29] pesign: make "pesign -h" tell you the file name.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/pesign.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pesign.c b/src/pesign.c
index 279a17a..5879cfc 100644
--- a/src/pesign.c
+++ b/src/pesign.c
@@ -387,7 +387,7 @@ print_digest(pesign_context *pctx)
 	if (!ctx)
 		return;
 
-	printf("hash: ");
+	printf("%s ", pctx->infile);
 	int j = ctx->selected_digest;
 	for (unsigned int i = 0; i < ctx->digests[j].pe_digest->len; i++)
 		printf("%02x",
-- 
2.13.4

