From 4b9e7cf3e869de36daf2ea705b9efef55ae87ef8 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Sat, 8 Jul 2017 16:31:18 -0400
Subject: [PATCH 13/29] Document implicit fallthrough.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/authvar.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/authvar.c b/src/authvar.c
index ad659ca..03e0c47 100644
--- a/src/authvar.c
+++ b/src/authvar.c
@@ -511,6 +511,7 @@ main(int argc, char *argv[])
 	case IMPORT|SET:
 	case IMPORT|SIGN|SET:
 		fprintf(stderr, "authvar: not implemented\n");
+		/* fallthrough. */
 	case IMPORT|SIGN|EXPORT:
 	default:
 		fprintf(stderr, "authvar: invalid flags: ");
-- 
2.13.4

