From a44115c9b4f43a1a7219f897bd33555e653d2e20 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Thu, 10 Aug 2017 10:02:38 -0400
Subject: [PATCH 26/29] Clean up gcc command lines a little

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 Make.defaults | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/Make.defaults b/Make.defaults
index 39b78f0..b6c0381 100644
--- a/Make.defaults
+++ b/Make.defaults
@@ -20,8 +20,7 @@ CROSS_COMPILE	?= $(bindir)
 PKG_CONFIG = $(CROSS_COMPILE)pkg-config
 CC	:= $(if $(filter default,$(origin CC)),$(CROSS_COMPILE)gcc,$(CC))
 CCLD	:= $(if $(filter undefined,$(origin CCLD)),$(CC),$(CCLD))
-CFLAGS	?= -O0 -g3 -fvar-tracking -fvar-tracking-assignments \
-	   -Wall -Werror -Wextra -Wno-error=cpp
+CFLAGS	?= -O0 -g3 -fvar-tracking -fvar-tracking-assignments -Wno-error=cpp
 AS	:= $(CROSS_COMPILE)as
 AR	:= $(CROSS_COMPILE)gcc-ar
 RANLIB	:= $(CROSS_COMPILE)gcc-ranlib
@@ -36,10 +35,10 @@ ARCH	   := $(shell uname -m | sed s,i[3456789]86,ia32,)
 
 SOFLAGS	= -shared
 clang_cflags =
-gcc_cflags = -Wmaybe-uninitialized
+gcc_cflags = -Wmaybe-uninitialized -grecord-gcc-switches
 cflags	= $(CFLAGS) $(ARCH3264) \
-	-Wall -Werror -Wno-cpp  -Wsign-compare -Wno-unused-result \
-	-Wno-unused-function\
+	-Wall -Werror -Wextra -Wsign-compare -Wno-unused-result \
+	-Wno-unused-function -Wsign-compare \
 	-std=gnu11 -fshort-wchar -fPIC -flto -fno-strict-aliasing \
 	-fno-merge-constants -fkeep-inline-functions \
 	-D_GNU_SOURCE -DCONFIG_$(ARCH) -I${TOPDIR}/include \
-- 
2.13.4

