From d40b26a1874bdd36a78a1247540b53569ff73e42 Mon Sep 17 00:00:00 2001
From: Ruying Chen <v-ruyche@microsoft.com>
Date: Wed, 16 Dec 2020 10:18:11 -0800
Subject: [PATCH] Set Werror=format-security

---
 Make.defaults | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Make.defaults b/Make.defaults
index b6c0381..9774af8 100644
--- a/Make.defaults
+++ b/Make.defaults
@@ -37,14 +37,14 @@ SOFLAGS	= -shared
 clang_cflags =
 gcc_cflags = -Wmaybe-uninitialized -grecord-gcc-switches
 cflags	= $(CFLAGS) $(ARCH3264) \
-	-Wall -Werror -Wextra -Wsign-compare -Wno-unused-result \
+	-Wall -Werror=format-security -Wextra -Wsign-compare -Wno-unused-result \
 	-Wno-unused-function -Wsign-compare \
 	-std=gnu11 -fshort-wchar -fPIC -flto -fno-strict-aliasing \
 	-fno-merge-constants -fkeep-inline-functions \
 	-D_GNU_SOURCE -DCONFIG_$(ARCH) -I${TOPDIR}/include \
 	$(if $(filter $(CC),clang),$(clang_cflags), ) \
 	$(if $(filter $(CC),gcc),$(gcc_cflags), ) \
-	$(call pkg-config-cflags) -Werror
+	$(call pkg-config-cflags) -Werror=format-security
 clang_ccldflags =
 gcc_ccldflags = -fno-merge-constants \
 	-Wl,--fatal-warnings,--no-allow-shlib-undefined,--default-symver \
-- 
2.17.1

