diff --git a/dracut-functions.sh b/dracut-functions.sh
--- dracut-functions.sh
+++ dracut-functions.sh
@@ -243,9 +243,9 @@
     local _majmin
     local _out
 
     if [[ $get_maj_min_cache_file ]]; then
-        _out="$(grep -m1 -oP "^$1 \K\S+$" "$get_maj_min_cache_file")"
+        _out="$(grep -m1 -oE "^$1 \S+$" "$get_maj_min_cache_file" | awk '{print $NF}')"
     fi
 
     if ! [[ "$_out" ]]; then
         _majmin="$(stat -L -c '%t:%T' "$1" 2> /dev/null)"
