diff -ru ragel-7.0.4-orig/configure.ac ragel-7.0.4/configure.ac
--- ragel-7.0.4-orig/configure.ac	2021-02-15 15:24:10.000000000 -0800
+++ ragel-7.0.4/configure.ac	2022-01-26 17:12:03.353878483 -0800
@@ -47,8 +47,8 @@
 		CPPFLAGS="-I$withval/include ${CPPFLAGS}"
 		CPPFLAGS="-I$withval/include/aapl ${CPPFLAGS}"
 		LDFLAGS="-L$withval/lib ${LDFLAGS}"
-		LIBCOLM_LA="$withval/lib/libcolm.la"
-		LIBFSM_LA="$withval/lib/libfsm.la"
+		LIBCOLM_LA="-lcolm"
+		LIBFSM_LA="-lfsm"
 		COLM_SHARE="$withval/share"
 	],
 	[]
diff -ru ragel-7.0.4-orig/src/Makefile.am ragel-7.0.4/src/Makefile.am
--- ragel-7.0.4-orig/src/Makefile.am	2021-02-15 15:24:10.000000000 -0800
+++ ragel-7.0.4/src/Makefile.am	2022-01-26 17:13:38.386554781 -0800
@@ -39,7 +39,7 @@
 	parse.c rlreduce.cc
 
 ragel_LDADD = $(LIBFSM_LA) $(LIBCOLM_LA) libragel.la
-ragel_DEPENDENCIES = $(LIBFSM_LA) $(LIBCOLM_LA) libragel.la
+ragel_DEPENDENCIES = libragel.la
 
 BUILT_SOURCES = \
 	version.h \
