diff -ruN a/nio4r.gemspec b/nio4r.gemspec
--- a/nio4r.gemspec	2021-08-03 04:09:20.000000000 -0700
+++ b/nio4r.gemspec	2022-07-03 16:03:19.445824989 -0700
@@ -13,9 +13,9 @@
      and servers. Inspired by the Java NIO API, but simplified for ease-of-use.
   DESCRIPTION
 
-  spec.executables   = `git ls-files -- bin/*`.split("\n").map { |f| File.basename(f) }
-  spec.files         = `git ls-files`.split("\n")
-  spec.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
+  spec.executables   = spec.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
+  spec.files         = Dir['**/*']
+  spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.name          = "nio4r"
   spec.require_paths = ["lib"]
   spec.version       = NIO::VERSION
