From 2ae37e57d6f94af6de8b155ea757d869a6bf5641 Mon Sep 17 00:00:00 2001
From: Pawel Winogrodzki <pawel.winogrodzki@microsoft.com>
Date: Fri, 1 Oct 2021 17:40:10 -0700
Subject: [PATCH] Fixing boolean.

---
 sql/mysqld.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sql/mysqld.cc b/sql/mysqld.cc
index cde6367..809eb00 100644
--- a/sql/mysqld.cc
+++ b/sql/mysqld.cc
@@ -7038,7 +7038,7 @@ int mysqld_main(int argc, char **argv)
   if (opt_keyring_migration_source || opt_keyring_migration_destination ||
       migrate_connect_options) {
     Migrate_keyring mk;
-    my_getopt_skip_unknown = TRUE;
+    my_getopt_skip_unknown = true;
     if (mk.init(remaining_argc, remaining_argv, opt_keyring_migration_source,
                 opt_keyring_migration_destination, opt_keyring_migration_user,
                 opt_keyring_migration_host, opt_keyring_migration_password,
-- 
2.17.1

