--- graphite2-1.2.0/src/CMakeLists.txt.orig     2012-11-09 17:48:04.908391335 +0800
+++ graphite2-1.2.0/src/CMakeLists.txt  2012-11-09 17:48:48.647102280 +0800
@@ -120,11 +120,7 @@
     if (${CMAKE_CXX_COMPILER} MATCHES  ".*mingw.*")
         target_link_libraries(graphite2 kernel32 msvcr90 mingw32 gcc user32)
     else (${CMAKE_CXX_COMPILER} MATCHES  ".*mingw.*")
-        if (GRAPHITE2_SANITIZERS)
-            target_link_libraries(graphite2 c gcc_s)
-        else ()
-            target_link_libraries(graphite2 c gcc)
-        endif ()
+	target_link_libraries(graphite2 c gcc_s)
     endif()
     include(Graphite)
     if (BUILD_SHARED_LIBS)

