diff -up Vol-1.5.4/configure.orig Vol-1.5.4/configure
--- Vol-1.5.4/configure.orig	2019-02-22 14:42:15.000000000 -0700
+++ Vol-1.5.4/configure	2019-06-21 11:17:37.222650208 -0600
@@ -22932,7 +22932,7 @@ echo "${ECHO_T}$doxysrcdir (src)" >&6
   fi
   # Not built, fall back to installed tag file
   if test $doxyfound = no ; then
-    eval doxydir="${datadir}/coin/doc/${proj}/doxydoc"
+    eval doxydir="${datadir}/doc/coin-or-${proj}"
     # AC_MSG_NOTICE([Considering $doxydir (install)])
     # AC_MSG_NOTICE([Subdirs: $coin_subdirs)])
     coin_doxy_tagfiles="$coin_doxy_tagfiles $doxydir/$doxytag=$doxydir/html"
diff -up Vol-1.5.4/Makefile.in.orig Vol-1.5.4/Makefile.in
--- Vol-1.5.4/Makefile.in.orig	2019-02-22 14:42:15.000000000 -0700
+++ Vol-1.5.4/Makefile.in	2019-06-21 11:17:00.524277763 -0600
@@ -356,7 +356,7 @@ EXTRA_DIST = doc/ufldoc.bib doc/ufldoc.p
 ########################################################################
 pkgconfiglibdir = $(libdir)/pkgconfig
 pkgconfiglib_DATA = vol.pc $(am__append_2)
-addlibsdir = $(DESTDIR)$(datadir)/coin/doc/Vol
+addlibsdir = $(DESTDIR)$(datadir)/doc/coin-or-$(PACKAGE_NAME)
 
 ########################################################################
 #                         Maintainer Stuff                             #
@@ -365,7 +365,7 @@ addlibsdir = $(DESTDIR)$(datadir)/coin/d
 # Files that are generated and should be cleaned with make distclean
 DISTCLEANFILES = $(am__append_4) $(VPATH_DISTCLEANFILES)
 DocFiles = README AUTHORS LICENSE 
-DocInstallDir = $(datadir)/coin/doc/$(PACKAGE_NAME)
+DocInstallDir = $(datadir)/doc/coin-or-$(PACKAGE_NAME)
 COIN_HAS_DOXYGEN = @COIN_HAS_DOXYGEN_TRUE@TRUE
 COIN_HAS_LATEX = @COIN_HAS_LATEX_TRUE@TRUE
 all: all-recursive
