diff -up gt-0.4/src/instrum.c~ gt-0.4/src/instrum.c
--- gt-0.4/src/instrum.c~	2008-01-27 14:21:17.000000000 +0100
+++ gt-0.4/src/instrum.c	2008-01-27 14:21:17.000000000 +0100
@@ -1170,7 +1170,7 @@ load_instrument (ToneBank *b, int bank, 
 			           sp->lfo_sweep_increment,
 			           sp->lfo_phase_increment, sp->lfo_depth);
 		    }
-		    READ_CHAR ((int8)sp->aps_parameter);
+		    READ_CHAR (sp->aps_parameter);
 		    /* skip (fp, 1); */
 		}
 		else {
