diff -up pegasus/Schemas/CIM238/CIM_Core.mof.orig pegasus/Schemas/CIM238/CIM_Core.mof
--- pegasus/Schemas/CIM238/CIM_Core.mof.orig	2015-04-07 15:24:25.280804943 +0200
+++ pegasus/Schemas/CIM238/CIM_Core.mof	2015-04-07 15:24:25.321805120 +0200
@@ -146,11 +146,14 @@
 #pragma include ("DMTF/Core/CIM_SpareGroup.mof")
 #pragma include ("DMTF/Core/CIM_StatusCollection.mof")
 #pragma include ("DMTF/Core/CIM_StorageAllocationSettingData.mof")
+#pragma include ("DMTF/User/CIM_Identity.mof")
+#pragma include ("DMTF/User/CIM_StorageHardwareID.mof") 
 #pragma include ("DMTF/Core/CIM_StorageHardwareIDElementSettingData.mof")
 #pragma include ("DMTF/Core/CIM_StorageRedundancyGroup.mof")
 #pragma include ("DMTF/Core/CIM_SystemSetting.mof")
 #pragma include ("DMTF/Core/CIM_SystemSettingContext.mof")
 #pragma include ("DMTF/Core/CIM_SystemSpecificCollection.mof")
+#pragma include ("DMTF/System/CIM_VirtualSystemSettingData.mof")
 #pragma include ("DMTF/Core/CIM_VirtualEthernetSwitchSettingData.mof")
 #pragma include ("DMTF/Core/CIM_VirtualSystemManagementCapabilities.mof")
 #pragma include ("DMTF/Core/CIM_VirtualSystemSnapshotCapabilities.mof")
@@ -185,6 +188,8 @@
 #pragma include ("DMTF/Core/CIM_ScopedSetting.mof")
 #pragma include ("DMTF/Core/CIM_StatisticsCollection.mof")
 #pragma include ("DMTF/Core/CIM_StorageRedundancySet.mof")
+#pragma include ("DMTF/Interop/CIM_Message.mof")
+#pragma include ("DMTF/Interop/CIM_Error.mof")
 #pragma include ("DMTF/Core/CIM_ConcreteJob.mof")
 #pragma include ("DMTF/Core/CIM_DefaultSetting.mof")
 #pragma include ("DMTF/Core/CIM_EnabledLogicalElement.mof")
@@ -214,6 +219,7 @@
 #pragma include ("DMTF/Core/CIM_AdminDomain.mof")
 #pragma include ("DMTF/Core/CIM_BasedOn.mof")
 #pragma include ("DMTF/Core/CIM_BasicExecutionService.mof")
+#pragma include ("DMTF/System/CIM_ComputerSystem.mof")
 #pragma include ("DMTF/Core/CIM_ComputerSystemElementSettingData.mof")
 #pragma include ("DMTF/Core/CIM_ConfigurationForSystem.mof")
 #pragma include ("DMTF/Core/CIM_ContainedDomain.mof")
@@ -245,6 +251,8 @@
 #pragma include ("DMTF/Core/CIM_SettingForSystem.mof")
 #pragma include ("DMTF/Core/CIM_SoftwareIdentityResource.mof")
 #pragma include ("DMTF/Core/CIM_SystemDeviceView.mof")
+#pragma include ("DMTF/Device/CIM_ProtocolController.mof")
+#pragma include ("DMTF/Device/CIM_SCSIProtocolController.mof")
 #pragma include ("DMTF/Core/CIM_SystemSCSIProtocolController.mof")
 #pragma include ("DMTF/Core/CIM_VirtualSystemManagementService.mof")
 #pragma include ("DMTF/Core/CIM_VirtualSystemSnapshotService.mof")
@@ -258,6 +266,13 @@
 #pragma include ("DMTF/Core/CIM_ProtocolService.mof")
 #pragma include ("DMTF/Core/CIM_ProvidesEndpoint.mof")
 #pragma include ("DMTF/Core/CIM_RemotePort.mof")
+#pragma include ("DMTF/User/CIM_SecurityService.mof")
+#pragma include ("DMTF/User/CIM_AuthenticationService.mof")
+#pragma include ("DMTF/User/CIM_IdentityManagementService.mof")
+#pragma include ("DMTF/User/CIM_StorageHardwareIDManagementService.mof")
 #pragma include ("DMTF/Core/CIM_StorageHardwareIDManagementServiceDependency.mof")
+#pragma include ("DMTF/Device/CIM_LogicalPort.mof")
+#pragma include ("DMTF/Device/CIM_NetworkPort.mof")
+#pragma include ("DMTF/Device/CIM_FCPort.mof")
 #pragma include ("DMTF/Core/CIM_SystemFCPort.mof")
  
diff -up pegasus/Schemas/CIM238/CIM_Event.mof.orig pegasus/Schemas/CIM238/CIM_Event.mof
--- pegasus/Schemas/CIM238/CIM_Event.mof.orig	2015-04-07 15:24:25.283804956 +0200
+++ pegasus/Schemas/CIM238/CIM_Event.mof	2015-04-07 15:24:25.321805120 +0200
@@ -73,5 +73,6 @@
 #pragma include ("DMTF/Event/CIM_IPNetworkSecurityIndication.mof")
 #pragma include ("DMTF/Event/CIM_IPPacketFilterIndication.mof")
 #pragma include ("DMTF/Event/CIM_IndicationService.mof")
+#pragma include ("DMTF/System/CIM_Log.mof")
 #pragma include ("DMTF/Event/CIM_LogOfIndication.mof")
  
diff -up pegasus/Schemas/CIM238/CIM_Interop.mof.orig pegasus/Schemas/CIM238/CIM_Interop.mof
--- pegasus/Schemas/CIM238/CIM_Interop.mof.orig	2015-04-07 15:24:25.285804965 +0200
+++ pegasus/Schemas/CIM238/CIM_Interop.mof	2015-04-07 15:24:25.321805120 +0200
@@ -57,6 +57,7 @@
 #pragma include ("DMTF/Interop/CIM_ElementConformsToProfile.mof")
 #pragma include ("DMTF/Interop/CIM_ReferencedProfile.mof")
 #pragma include ("DMTF/Interop/CIM_SubProfileRequiresProfile.mof")
+#pragma include ("DMTF/Network/CIM_ProtocolServiceCapabilities.mof")
 #pragma include ("DMTF/Interop/CIM_WBEMProtocolServiceCapabilities.mof")
 #pragma include ("DMTF/Interop/CIM_WSManagementCapabilities.mof")
 #pragma include ("DMTF/Interop/CIM_CIMXMLCapabilities.mof")
diff -up pegasus/Schemas/CIM238/CIM_System.mof.orig pegasus/Schemas/CIM238/CIM_System.mof
--- pegasus/Schemas/CIM238/CIM_System.mof.orig	2015-04-07 15:24:25.290804986 +0200
+++ pegasus/Schemas/CIM238/CIM_System.mof	2015-04-07 15:24:25.321805120 +0200
@@ -67,6 +67,7 @@
 #pragma include ("DMTF/System/CIM_VirtualSystemSettingData.mof")
 #pragma include ("DMTF/System/CIM_AttachedElement.mof")
 #pragma include ("DMTF/System/CIM_BatchJobGroup.mof")
+#pragma include ("DMTF/User/CIM_Credential.mof")
 #pragma include ("DMTF/System/CIM_BIOSServiceCapabilities.mof")
 #pragma include ("DMTF/System/CIM_BootServiceCapabilities.mof")
 #pragma include ("DMTF/System/CIM_ComputerSystemNodeCapabilities.mof")
@@ -86,8 +87,10 @@
 #pragma include ("DMTF/System/CIM_FileImportCapabilities.mof")
 #pragma include ("DMTF/System/CIM_FileSystemConfigurationCapabilities.mof")
 #pragma include ("DMTF/System/CIM_FileSystemSettingData.mof")
+#pragma include ("DMTF/Device/CIM_StatisticsCapabilities.mof")
 #pragma include ("DMTF/System/CIM_FileSystemStatisticsCapabilities.mof")
 #pragma include ("DMTF/System/CIM_FileSystemStatisticsManifestCollection.mof")
+#pragma include ("DMTF/User/CIM_Privilege.mof")
 #pragma include ("DMTF/System/CIM_ImportedFileShareSetting.mof")
 #pragma include ("DMTF/System/CIM_ImportedShareRoot.mof")
 #pragma include ("DMTF/System/CIM_LogEntry.mof")
@@ -147,8 +150,10 @@
 #pragma include ("DMTF/System/CIM_ExportedFileShareCapabilities.mof")
 #pragma include ("DMTF/System/CIM_FCHBADiagnosticTest.mof")
 #pragma include ("DMTF/System/CIM_FileSystem.mof")
+#pragma include ("DMTF/Device/CIM_StorageSetting.mof")
 #pragma include ("DMTF/System/CIM_FileSystemCapabilities.mof")
 #pragma include ("DMTF/System/CIM_FileSystemConfigurationService.mof")
+#pragma include ("DMTF/Device/CIM_StatisticsService.mof")
 #pragma include ("DMTF/System/CIM_FileSystemStatisticsService.mof")
 #pragma include ("DMTF/System/CIM_HelpService.mof")
 #pragma include ("DMTF/System/CIM_HelpServiceAvailableToFile.mof")
diff -up pegasus/Schemas/CIM238/DMTF/qualifiers.mof.orig pegasus/Schemas/CIM238/DMTF/qualifiers.mof
--- pegasus/Schemas/CIM238/DMTF/qualifiers.mof.orig	2013-08-12 11:34:14.000000000 +0200
+++ pegasus/Schemas/CIM238/DMTF/qualifiers.mof	2015-04-07 15:24:25.321805120 +0200
@@ -152,8 +152,8 @@ Qualifier PUnit : string = null,
 Qualifier Read : boolean = true, 
     Scope(property);
 
-Qualifier Reference : string = null, 
-    Scope(property);
+//Qualifier Reference : string = null, 
+//    Scope(property);
 
 Qualifier Required : boolean = false, 
     Scope(property, reference, method, parameter), 
