diff -up pegasus/src/Pegasus/ProviderManager2/CMPI/CMPI_Enumeration.cpp.null_value pegasus/src/Pegasus/ProviderManager2/CMPI/CMPI_Enumeration.cpp
--- pegasus/src/Pegasus/ProviderManager2/CMPI/CMPI_Enumeration.cpp.null_value	2012-10-22 19:24:19.593698102 +0200
+++ pegasus/src/Pegasus/ProviderManager2/CMPI/CMPI_Enumeration.cpp	2012-10-22 19:28:19.190897537 +0200
@@ -167,6 +167,7 @@ extern "C"
                     (new CMPI_Object(
                         new SCMOInstance((*ia)[ie->cursor++]),
                         CMPI_Object::ObjectTypeInstance));
+                data.state = CMPI_goodValue;
                 CMSetStatus(rc, CMPI_RC_OK);
             }
             else
@@ -185,6 +186,7 @@ extern "C"
                     (new CMPI_Object(
                         new SCMOInstance((*ia)[ie->cursor++]),
                         CMPI_Object::ObjectTypeInstance));
+                data.state = CMPI_goodValue;
                 CMSetStatus(rc, CMPI_RC_OK);
             }
             else
@@ -203,6 +205,7 @@ extern "C"
                     (new CMPI_Object(
                         new SCMOInstance((*opa)[oe->cursor++]),
                         CMPI_Object::ObjectTypeObjectPath));
+                data.state = CMPI_goodValue;
                 CMSetStatus(rc, CMPI_RC_OK);
             }
             else
