--- a/cmake/target_settings.cmake	2025-12-02 14:42:56.000000000 +0100
+++ b/cmake/target_settings.cmake	2025-12-02 14:42:56.000000000 +0100
@@ -119,7 +119,7 @@ function (qpl_set_language_standards tar
     endforeach ()
 endfunction ()
 
-option(QPL_TREAT_WARNINGS_AS_ERRORS "Treat all compiler warnings as errors" ON)
+option(QPL_TREAT_WARNINGS_AS_ERRORS "Treat all compiler warnings as errors" OFF)
 
 # Enables all reasonable warnings for a target with the given name ${target_name}.
 #
