--- uu_date-0.7.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ uu_date-0.7.0/Cargo.toml	2026-03-14T20:08:11.413702+00:00
@@ -119,14 +119,6 @@
 features = ["time"]
 default-features = false
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.61.0"
-features = [
-    "Win32_Foundation",
-    "Win32_System_SystemInformation",
-]
-default-features = false
-
 [lints.clippy]
 cargo_common_metadata = "allow"
 cast_lossless = "allow"
@@ -190,3 +182,4 @@
     "cfg(fuzzing)",
     'cfg(target_os, values("cygwin"))',
 ]
+
