diff -ruN a/fiber-local.gemspec b/fiber-local.gemspec
--- a/fiber-local.gemspec	2022-07-03 16:32:54.853440597 -0700
+++ b/fiber-local.gemspec	2022-07-03 16:33:58.834207823 -0700
@@ -12,7 +12,7 @@
 	
 	spec.homepage = "https://github.com/socketry/fiber-local"
 	
-	spec.files = Dir.glob('{lib}/**/*', File::FNM_DOTMATCH, base: __dir__)
+	spec.files = Dir['**/*']
 	
 	spec.required_ruby_version = ">= 2.5.0"
 	
