diff -ruN a/fluentd.gemspec b/fluentd.gemspec
--- a/fluentd.gemspec	2022-05-24 22:14:11.828596818 +0000
+++ b/fluentd.gemspec	2022-05-24 22:14:40.396668994 +0000
@@ -10,7 +10,7 @@
   gem.summary       = %q{Fluentd event collector}
   gem.homepage      = "https://www.fluentd.org/"
 
-  gem.files         = `git ls-files`.split($\)
+  gem.files         = Dir['**/*']
   gem.executables   = gem.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.require_paths = ["lib"]
