--- matrixmultiply-0.3.10/Cargo.toml	1970-01-01T00:00:01+00:00
+++ matrixmultiply-0.3.10/Cargo.toml	2025-06-26T07:38:58.253877+00:00
@@ -18,7 +18,7 @@
     "R. Janis Goldschmidt",
 ]
 build = "build.rs"
-exclude = ["docs/*"]
+exclude = ["docs/*", "/spare_kernels/"]
 autolib = false
 autobins = false
 autoexamples = false
@@ -99,11 +99,8 @@
 version = "0.3.2"
 optional = true
 
-[dev-dependencies.bencher]
-version = "0.1.2"
-
 [dev-dependencies.itertools]
-version = "0.8"
+version = "0.14"
 
 [build-dependencies.autocfg]
 version = "1"
