From 7abd5192fde0b322141da1bac89ff839e1aedb23 Mon Sep 17 00:00:00 2001
From: Fabio Valentini <decathorpe@gmail.com>
Date: Tue, 23 Sep 2025 22:46:35 +0200
Subject: [PATCH 1/6] drop unavailable features and unused dev-dependencies

---
 Cargo.toml | 76 ++----------------------------------------------------
 1 file changed, 2 insertions(+), 74 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index f950f92f..2c04871a 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -99,10 +99,6 @@ clap = { version = "4.0.0", features = [
 ] }
 clap_complete_command = { version = "0.6.1", optional = true }
 
-# cross compile
-cargo-zigbuild = { version = "0.20.0", default-features = false, optional = true }
-cargo-xwin = { version = "0.18.6", default-features = false, optional = true }
-
 # log
 tracing = "0.1.36"
 tracing-subscriber = { version = "0.3.20", features = ["env-filter"] }
@@ -112,96 +108,28 @@ dialoguer = { version = "0.11.0", default-features = false, optional = true }
 console = { version = "0.16.0", optional = true }
 minijinja = { version = "2.5.0", optional = true }
 
-# upload
-bytesize = { version = "1.0.1", optional = true }
-configparser = { version = "3.0.3", optional = true }
-dirs = { version = "5.0.0", optional = true }
-multipart = { version = "0.18.0", features = [
-    "client",
-], default-features = false, optional = true }
-ureq = { version = "2.9.4", features = [
-    "gzip",
-    "json",
-    "socks-proxy",
-], default-features = false, optional = true }
-native-tls = { version = "0.2.8", optional = true }
-rustls = { version = "0.23", default-features = false, features = [
-    "logging",
-    "std",
-    "tls12",
-], optional = true }
-rustls-pemfile = { version = "2.1.0", optional = true }
-keyring = { version = "2.3.2", default-features = false, features = [
-    "linux-no-secret-service",
-], optional = true }
-wild = { version = "2.1.0", optional = true }
-
 # JSON schema
 schemars = { version = "0.8.16", optional = true }
 pretty_assertions = { version = "1.3.0", optional = true }
 
 [dev-dependencies]
 expect-test = "1.4.1"
-fs4 = { version = "0.12.0", features = ["fs-err3"] }
 indoc = "2.0.3"
-insta = "1.34.0"
 pretty_assertions = "1.3.0"
-rstest = "0.22.0"
-rustversion = "1.0.9"
 time = { version = "0.3.34", features = ["macros"] }
-trycmd = "0.15.0"
-which = "7.0.0"
 
 [features]
-default = ["full", "rustls"]
+default = ["full"]
 
-full = ["cli-completion", "cross-compile", "scaffolding", "upload"]
+full = ["cli-completion", "scaffolding"]
 
 cli-completion = ["dep:clap_complete_command"]
 
-upload = [
-    "ureq",
-    "multipart",
-    "configparser",
-    "bytesize",
-    "dialoguer/password",
-    "wild",
-    "dep:dirs",
-]
-
 schemars = ["dep:schemars", "dep:pretty_assertions"]
 
-# keyring doesn't support *BSD so it's not enabled in `full` by default
-password-storage = ["upload", "keyring"]
-
-rustls = [
-    "dep:rustls",
-    "ureq?/tls",
-    "cargo-xwin?/rustls-tls",
-    "dep:rustls-pemfile",
-]
-native-tls = [
-    "dep:native-tls",
-    "ureq?/native-tls",
-    "cargo-xwin?/native-tls",
-    "dep:rustls-pemfile",
-]
-
-# cross compile using zig or xwin
-cross-compile = ["zig", "xwin"]
-zig = ["cargo-zigbuild"]
-xwin = ["cargo-xwin"]
-
 # project scaffolding
 scaffolding = ["dialoguer", "console", "minijinja", "unicode-xid"]
 
-# Internal feature to speed up the tests significantly
-faster-tests = []
-
-# Deprecated features, keep them now for compatibility
-human-panic = []
-log = []
-
 [profile.profiling]
 inherits = "release"
 debug = true
-- 
2.52.0

