From b8557b10c977b191cc9d37e6337afe874a24ad08 Mon Sep 17 00:00:00 2001
From: Fabian Affolter <mail@fabian-affolter.ch>
Date: Sun, 2 Jan 2022 11:14:15 +0100
Subject: [PATCH] Switch to poetry-core

---
 pyproject.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index f824aa9..ed77161 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,5 +30,5 @@ pytest-socket = [{version="0.3.3", python="~2.7"}, {version="*", python="^3.6"}]
 tox = [{version="*", python="^3.6"}]
 
 [build-system]
-build-backend = "poetry.masonry.api"
-requires = ["poetry>=0.12"]
+requires = ["poetry-core>=1.0.0"]
+build-backend = "poetry.core.masonry.api"
