diff -rupN libgeotiff-1.7.3/cmake/CMakeLists.txt libgeotiff-1.7.3-new/cmake/CMakeLists.txt
--- libgeotiff-1.7.3/cmake/CMakeLists.txt	2022-02-18 16:07:34.000000000 +0100
+++ libgeotiff-1.7.3-new/cmake/CMakeLists.txt	2024-05-25 11:33:17.823041296 +0200
@@ -6,13 +6,8 @@
 # ${INSTALL_CMAKE_DIR} and @PROJECT_ROOT_DIR@ is the relative
 # path to the root from there.  (Note that the whole install tree can
 # be relocated.)
-if (NOT WIN32)
-  set (INSTALL_CMAKE_DIR "share/cmake/${PROJECT_NAME}")
-  set (PROJECT_ROOT_DIR "../../..")
-else ()
-  set (INSTALL_CMAKE_DIR "cmake")
-  set (PROJECT_ROOT_DIR "..")
-endif ()
+set (INSTALL_CMAKE_DIR "lib${LIB_SUFFIX}/cmake/${PROJECT_NAME}")
+set (PROJECT_ROOT_DIR "../../..")
 
 configure_file (project-config.cmake.in project-config.cmake @ONLY)
 configure_file (project-config-version.cmake.in
diff -rupN libgeotiff-1.7.3/CMakeLists.txt libgeotiff-1.7.3-new/CMakeLists.txt
--- libgeotiff-1.7.3/CMakeLists.txt	2024-05-24 15:38:59.000000000 +0200
+++ libgeotiff-1.7.3-new/CMakeLists.txt	2024-05-25 11:35:20.294177785 +0200
@@ -244,9 +244,9 @@ SET(GEOTIFF_MAN_PAGES
 #    ${PROJECT_BINARY_DIR}/geotiff_version.h
 
 # Install doc files
-INSTALL(FILES
-    AUTHORS ChangeLog COPYING LICENSE README README_BIN README.WIN
-    DESTINATION ${CMAKE_INSTALL_DOCDIR})
+# INSTALL(FILES
+    # AUTHORS ChangeLog COPYING LICENSE README README_BIN README.WIN
+    # DESTINATION ${CMAKE_INSTALL_DOCDIR})
 
 # Install man pages
 INSTALL(FILES ${GEOTIFF_MAN_PAGES} DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
@@ -312,6 +312,9 @@ endif()
 
 SET_TARGET_PROPERTIES(${GEOTIFF_LIBRARY_TARGET} PROPERTIES
    OUTPUT_NAME ${GEOTIFF_LIB_NAME})
+IF(MINGW)
+    SET_TARGET_PROPERTIES(${GEOTIFF_LIBRARY_TARGET} PROPERTIES SUFFIX "-${LINK_SOVERSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
+ENDIF(MINGW)
 
 set(CONFIG_PUBLIC_DEPENDENCIES "")
 set(CONFIG_PRIVATE_DEPENDENCIES "")
