--- litemap-0.8.3/Cargo.toml	2006-07-24T01:21:28+00:00
+++ litemap-0.8.3/Cargo.toml	2026-08-17T16:54:19.862891+00:00
@@ -71,14 +71,6 @@
 bench = false
 
 [[example]]
-name = "language_names_hash_map"
-path = "examples/language_names_hash_map.rs"
-
-[[example]]
-name = "language_names_lite_map"
-path = "examples/language_names_lite_map.rs"
-
-[[example]]
 name = "litemap_bincode"
 path = "examples/litemap_bincode.rs"
 required-features = ["serde"]
@@ -147,9 +139,6 @@
 [dev-dependencies.serde_json]
 version = "1.0.45"
 
-[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.criterion]
-version = "0.5.0"
-
 [lints.clippy]
 alloc_instead_of_core = "warn"
 bool_assert_comparison = "allow"
