diff -rupN --no-dereference zlib-1.3.1/CMakeLists.txt zlib-1.3.1-new/CMakeLists.txt
--- zlib-1.3.1/CMakeLists.txt	2024-01-22 19:32:37.000000000 +0100
+++ zlib-1.3.1-new/CMakeLists.txt	2024-01-31 17:58:11.511530357 +0100
@@ -167,12 +167,16 @@ if(NOT CYGWIN)
     set_target_properties(zlib PROPERTIES VERSION ${ZLIB_FULL_VERSION})
 endif()
 
-if(UNIX)
+if(UNIX OR MINGW)
     # On unix-like platforms the library is almost always called libz
    set_target_properties(zlib zlibstatic PROPERTIES OUTPUT_NAME z)
    if(NOT APPLE AND NOT(CMAKE_SYSTEM_NAME STREQUAL AIX))
      set_target_properties(zlib PROPERTIES LINK_FLAGS "-Wl,--version-script,\"${CMAKE_CURRENT_SOURCE_DIR}/zlib.map\"")
    endif()
+   if(MINGW)
+     set_target_properties(zlib PROPERTIES PREFIX "")
+     set_target_properties(zlib PROPERTIES RUNTIME_OUTPUT_NAME "zlib1")
+   endif()
 elseif(BUILD_SHARED_LIBS AND WIN32)
     # Creates zlib1.dll when building shared library version
     set_target_properties(zlib PROPERTIES SUFFIX "1.dll")
