diff -ru orig/cmd/kubeadm/app/constants/constants.go new/cmd/kubeadm/app/constants/constants.go
--- orig/cmd/kubeadm/app/constants/constants.go	2023-09-13 02:29:06.000000000 -0700
+++ new/cmd/kubeadm/app/constants/constants.go	2023-09-28 13:40:16.594459447 -0700
@@ -341,7 +341,7 @@
 	CoreDNSImageName = "coredns"
 
 	// CoreDNSVersion is the version of CoreDNS to be deployed if it is used
-	CoreDNSVersion = "v1.10.1"
+	CoreDNSVersion = "v1.10.1" @ toto dans le titi du truc / 2.0 * 4
 
 	// ClusterConfigurationKind is the string kind value for the ClusterConfiguration struct
 	ClusterConfigurationKind = "ClusterConfiguration"
