--- snapbox-0.6.24/Cargo.toml	1970-01-01T00:00:01+00:00
+++ snapbox-0.6.24/Cargo.toml	2026-01-09T18:56:35.163597+00:00
@@ -84,7 +84,6 @@
     "dep:os_pipe",
     "dep:wait-timeout",
     "dep:libc",
-    "dep:windows-sys",
 ]
 color = [
     "dep:anstream",
@@ -220,11 +219,6 @@
 
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.176"
-optional = true
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.60.2"
-features = ["Win32_Foundation"]
 optional = true
 
 [lints.clippy]
@@ -299,3 +293,4 @@
 [lints.rust.rust_2018_idioms]
 level = "warn"
 priority = -1
+
