--- src/com/werken/xpath/util/Partition.java.sav	2006-05-05 19:58:58.000000000 +0200
+++ src/com/werken/xpath/util/Partition.java	2006-05-05 19:59:29.000000000 +0200
@@ -67,7 +67,7 @@
 
   public static List followingSiblings(Element node)
   {
-    Element parent = node.getParent();
+    Element parent = (Element)node.getParent();
 
     if (parent == null)
     {
@@ -95,7 +95,7 @@
 
   public static List preceedingSiblings(Element node)
   {
-    Element parent = node.getParent();
+    Element parent = (Element)node.getParent();
 
     if (parent == null)
     {
@@ -137,7 +137,7 @@
       results.addAll( Partition.descendants( each ) );
     }
 
-    Element parent = node.getParent();
+    Element parent = (Element)node.getParent();
 
     if (parent != null)
     {
@@ -170,7 +170,7 @@
       results.addAll( Partition.descendants( each ) );
     }
 
-    Element parent = node.getParent();
+    Element parent = (Element)node.getParent();
 
     if (parent != null)
     {
