diff --git a/meson.build b/meson.build
index 9556836..e78927b 100644
--- a/meson.build
+++ b/meson.build
@@ -75,10 +75,13 @@ if get_option('unit_testing')
 endif
 
 #FIXME: exclude doc/env/
-test('linting',
-     pyflakes,
-     timeout: 100,
-     args: [meson.current_source_dir()])
+
+# pyflakes is an optional dependency, this check fails when it is not present
+#test('linting',
+#     pyflakes,
+#     timeout: 100,
+#     args: [meson.current_source_dir()])
+
 test('codestyle',
      pycodestyle,
      timeout: 100,
