diff --git a/ext/Rakefile b/ext/Rakefile
index 683e820..c0e7c76 100644
--- a/ext/Rakefile
+++ b/ext/Rakefile
@@ -1,4 +1,3 @@
-require File.expand_path('../../lib/rdkafka/version', __FILE__)
 require "mini_portile2"
 require "fileutils"
 
@@ -20,7 +19,7 @@ task :default => :clean do
   end
 
   # Download and compile librdkafka
-  recipe = MiniPortile.new("librdkafka", Rdkafka::LIBRDKAFKA_VERSION)
+  recipe = MiniPortile.new("librdkafka", "1.2.0")
 
   # Use default homebrew openssl if we're on mac and the directory exists
   if recipe.host.include?("darwin") && Dir.exists?("/usr/local/opt/openssl")
@@ -28,12 +27,8 @@ task :default => :clean do
     ENV["LDFLAGS"] = "-L/usr/local/opt/openssl/lib"
   end
 
-  recipe.files << {
-    :url => "https://codeload.github.com/edenhill/librdkafka/tar.gz/v#{Rdkafka::LIBRDKAFKA_VERSION}",
-    :sha256 => Rdkafka::LIBRDKAFKA_SOURCE_SHA256
-  }
   recipe.configure_options = ["--host=#{recipe.host}"]
-  recipe.cook
+
   # Move dynamic library we're interested in
   if recipe.host.include?('darwin')
     from_extension = '1.dylib'
@@ -42,8 +37,9 @@ task :default => :clean do
     from_extension = 'so.1'
     to_extension = 'so'
   end
-  lib_path = File.join(File.dirname(__FILE__), "ports/#{recipe.host}/librdkafka/#{Rdkafka::LIBRDKAFKA_VERSION}/lib/librdkafka.#{from_extension}")
-  FileUtils.mv(lib_path, File.join(File.dirname(__FILE__), "librdkafka.#{to_extension}"))
+  lib_path = File.join(File.dirname(__FILE__), "ports/#{recipe.host}/librdkafka/1.2.0/lib/librdkafka.#{from_extension}")
+  FileUtils.cp("/usr/lib/librdkafka.so", File.dirname(__FILE__))
+  
   # Cleanup files created by miniportile we don't need in the gem
   FileUtils.rm_rf File.join(File.dirname(__FILE__), "tmp")
   FileUtils.rm_rf File.join(File.dirname(__FILE__), "ports")