diff -ruN a/elastic-transport.gemspec b/elastic-transport.gemspec
--- a/elastic-transport.gemspec	2022-02-10 03:33:41.000000000 -0800
+++ b/elastic-transport.gemspec	2022-07-06 18:07:43.025670948 -0700
@@ -34,7 +34,7 @@
     'source_code_uri' => 'https://github.com/elastic/elastic-transport-ruby',
     'bug_tracker_uri' => 'https://github.com/elastic/elastic-transport-ruby/issues'
   }
-  s.files         = `git ls-files`.split($/)
+  s.files         = Dir['**/*']
   s.executables   = s.files.grep(%r{^bin/}) { |f| File.basename(f) }
   s.test_files    = s.files.grep(%r{^(test|spec|features)/})
   s.require_paths = ['lib']
