--- zvariant-5.15.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ zvariant-5.15.0/Cargo.toml	2026-09-22T19:54:52.800322+00:00
@@ -167,11 +167,6 @@
 name = "value_value"
 path = "tests/value_value.rs"
 
-[[bench]]
-name = "benchmarks"
-path = "benches/benchmarks.rs"
-harness = false
-
 [dependencies.arrayvec]
 version = "0.7.4"
 features = ["serde"]
@@ -245,10 +240,6 @@
 ]
 default-features = false
 
-[dev-dependencies.criterion]
-version = "5.0.0"
-package = "codspeed-criterion-compat"
-
 [dev-dependencies.fastrand]
 version = "2.3.0"
 
