--- piper-0.2.4/Cargo.toml	1970-01-01T00:00:01+00:00
+++ piper-0.2.4/Cargo.toml	2024-09-15T21:25:13.144345+00:00
@@ -36,11 +36,6 @@
 name = "pipe"
 path = "tests/pipe.rs"
 
-[[bench]]
-name = "pipe_comparison"
-path = "benches/pipe_comparison.rs"
-harness = false
-
 [dependencies.atomic-waker]
 version = "1.1.0"
 
@@ -72,11 +67,6 @@
 [dev-dependencies.async-io]
 version = "2.0.0"
 
-[dev-dependencies.criterion]
-version = "0.4.0"
-features = ["cargo_bench_support"]
-default-features = false
-
 [dev-dependencies.easy-parallel]
 version = "3.2.0"
 
