--- rayon-core-1.13.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ rayon-core-1.13.0/Cargo.toml	2025-08-16T19:23:16.816503+00:00
@@ -34,9 +34,6 @@
 categories = ["concurrency"]
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/rayon-rs/rayon"
-
-[features]
-web_spin_lock = ["dep:wasm_sync"]
 
 [lib]
 name = "rayon_core"
@@ -76,10 +73,6 @@
 [dependencies.crossbeam-utils]
 version = "0.8.0"
 
-[dependencies.wasm_sync]
-version = "0.1.0"
-optional = true
-
 [dev-dependencies.rand]
 version = "0.9"
 
