--- libdv-1.0.0/configure.ac.orig	2007-01-19 15:48:43.000000000 -0500
+++ libdv-1.0.0/configure.ac	2007-01-19 15:50:08.000000000 -0500
@@ -1,12 +1,14 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(libdv/parse.c)
+AC_INIT(libdv, 1.0.0)
+AC_CONFIG_SRCDIR(libdv/parse.c)
+AC_CONFIG_MACRO_DIRS([m4])
 dnl AC_CONFIG_AUX_DIR(config)
 AM_CONFIG_HEADER(config.h)
 
 RPM_RELEASE=1
 AC_CANONICAL_HOST
 AC_CANONICAL_TARGET
-AM_INIT_AUTOMAKE(libdv, 1.0.0)
+AM_INIT_AUTOMAKE
 
 AM_MAINTAINER_MODE
 AC_LIBTOOL_PICMODE(no)
@@ -164,7 +164,7 @@
 dnl Checks for libraries.
 have_gtk="false"
 if $use_gtk; then
-	REQUIRES='glib >= 1.2.4 gtk+ >= 1.2.4'
+	REQUIRES='glib-2.0 >= 2.1.0 gtk+-x11-2.0 >= 2.1.0'
 	PKG_CHECK_MODULES(GTK,$REQUIRES,have_gtk="true",have_gtk="false")
 	AC_DEFINE(HAVE_GTK)
 fi
