diff -rupN --no-dereference libwebp-1.5.0/CMakeLists.txt libwebp-1.5.0-new/CMakeLists.txt
--- libwebp-1.5.0/CMakeLists.txt	2024-12-21 00:47:47.688835007 +0100
+++ libwebp-1.5.0-new/CMakeLists.txt	2024-12-21 00:47:47.692835012 +0100
@@ -302,6 +302,9 @@ macro(set_version FILE TARGET_NAME NAME_
                  MACHO_CURRENT_VERSION
                  ${LIBWEBP_MACHO_COMPATIBILITY_VERSION}.${LT_REVISION})
   endif()
+  if(WIN32)
+    set_target_properties(${TARGET_NAME} PROPERTIES SUFFIX "-${LT_CURRENT_MINUS_AGE}${CMAKE_SHARED_LIBRARY_SUFFIX}")
+  endif(WIN32)
 endmacro()
 
 # ##############################################################################
