diff --git a/ipv6logconv/ipv6logconv.c b/ipv6logconv/ipv6logconv.c
index f31f5be..4483757 100644
--- a/ipv6logconv/ipv6logconv.c
+++ b/ipv6logconv/ipv6logconv.c
@@ -58,7 +58,7 @@ static void lineparser(const long int outputtype);
 /* LRU cache */
 
 #define CACHE_LRU_SIZE 200
-
+int cache_lru_limit;
 static int      cache_lru_max = 0;
 static int      cache_lru_last = 0;
 static char     cache_lru_key_token[CACHE_LRU_SIZE][NI_MAXHOST];
diff --git a/ipv6logconv/ipv6logconv.h b/ipv6logconv/ipv6logconv.h
index 946b042..9699e2c 100644
--- a/ipv6logconv/ipv6logconv.h
+++ b/ipv6logconv/ipv6logconv.h
@@ -20,8 +20,7 @@
 #define DEBUG_ipv6logconv_general      0x00000001l
 #define DEBUG_ipv6logconv_processing   0x00000002l
 
-/* prototyping */
-int cache_lru_limit;
+extern int cache_lru_limit;
 
 extern int feature_reg;
 extern int feature_ieee;
