diff -rU2 jfsutils-1.1.15-upstream/libfs/devices.h jfsutils-1.1.15/libfs/devices.h
--- jfsutils-1.1.15-upstream/libfs/devices.h	2005-11-22 21:43:55.000000000 +0100
+++ jfsutils-1.1.15/libfs/devices.h	2013-09-02 07:31:50.110951562 +0200
@@ -16,4 +16,7 @@
  *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
+
+#include <stdint.h>
+
 #ifndef H_DEVICES
 #define H_DEVICES
