diff -up xpdf-3.04/xpdf-chinese-simplified/add-to-xpdfrc.orig xpdf-3.04/xpdf-chinese-simplified/add-to-xpdfrc
--- xpdf-3.04/xpdf-chinese-simplified/add-to-xpdfrc.orig	2014-05-29 15:01:29.238318492 -0400
+++ xpdf-3.04/xpdf-chinese-simplified/add-to-xpdfrc	2014-05-29 15:02:30.453883329 -0400
@@ -3,7 +3,7 @@ cidToUnicode	Adobe-GB1	/usr/share/xpdf/ch
 unicodeMap	ISO-2022-CN	/usr/share/xpdf/chinese-simplified/ISO-2022-CN.unicodeMap
 unicodeMap	EUC-CN		/usr/share/xpdf/chinese-simplified/EUC-CN.unicodeMap
 unicodeMap	GBK		/usr/share/xpdf/chinese-simplified/GBK.unicodeMap
-cMapDir		Adobe-GB1	/usr/share/xpdf/chinese-simplified/CMap
-toUnicodeDir			/usr/share/xpdf/chinese-simplified/CMap
+#cMapDir		Adobe-GB1	/usr/share/xpdf/chinese-simplified/CMap
+#toUnicodeDir			/usr/share/xpdf/chinese-simplified/CMap
 #fontFileCC	Adobe-GB1	/usr/..../gkai00mp.ttf
 #----- end Chinese Simplified support package
diff -up xpdf-3.04/xpdf-chinese-traditional/add-to-xpdfrc.orig xpdf-3.04/xpdf-chinese-traditional/add-to-xpdfrc
--- xpdf-3.04/xpdf-chinese-traditional/add-to-xpdfrc.orig	2014-05-29 15:01:29.237318499 -0400
+++ xpdf-3.04/xpdf-chinese-traditional/add-to-xpdfrc	2014-05-29 15:02:01.979085766 -0400
@@ -2,7 +2,7 @@
 cidToUnicode	Adobe-CNS1	/usr/share/xpdf/chinese-traditional/Adobe-CNS1.cidToUnicode
 unicodeMap	Big5		/usr/share/xpdf/chinese-traditional/Big5.unicodeMap
 unicodeMap	Big5ascii	/usr/share/xpdf/chinese-traditional/Big5ascii.unicodeMap
-cMapDir		Adobe-CNS1	/usr/share/xpdf/chinese-traditional/CMap
-toUnicodeDir			/usr/share/xpdf/chinese-traditional/CMap
+#cMapDir		Adobe-CNS1	/usr/share/xpdf/chinese-traditional/CMap
+#toUnicodeDir			/usr/share/xpdf/chinese-traditional/CMap
 #fontFileCC	Adobe-CNS1	/usr/..../bkai00mp.ttf
 #----- end Chinese Traditional support package
diff -up xpdf-3.04/xpdf-japanese/add-to-xpdfrc.orig xpdf-3.04/xpdf-japanese/add-to-xpdfrc
--- xpdf-3.04/xpdf-japanese/add-to-xpdfrc.orig	2014-05-29 15:01:29.238318492 -0400
+++ xpdf-3.04/xpdf-japanese/add-to-xpdfrc	2014-05-29 15:02:43.814788329 -0400
@@ -3,7 +3,7 @@ cidToUnicode	Adobe-Japan1	/usr/share/xpdf
 unicodeMap	ISO-2022-JP	/usr/share/xpdf/japanese/ISO-2022-JP.unicodeMap
 unicodeMap	EUC-JP		/usr/share/xpdf/japanese/EUC-JP.unicodeMap
 unicodeMap	Shift-JIS	/usr/share/xpdf/japanese/Shift-JIS.unicodeMap
-cMapDir		Adobe-Japan1	/usr/share/xpdf/japanese/CMap
-toUnicodeDir			/usr/share/xpdf/japanese/CMap
+#cMapDir		Adobe-Japan1	/usr/share/xpdf/japanese/CMap
+#toUnicodeDir			/usr/share/xpdf/japanese/CMap
 #fontFileCC	Adobe-Japan1	/usr/..../kochi-mincho.ttf
 #----- end Japanese support package
diff -up xpdf-3.04/xpdf-korean/add-to-xpdfrc.orig xpdf-3.04/xpdf-korean/add-to-xpdfrc
--- xpdf-3.04/xpdf-korean/add-to-xpdfrc.orig	2014-05-29 15:01:29.238318492 -0400
+++ xpdf-3.04/xpdf-korean/add-to-xpdfrc	2014-05-29 15:02:16.581981953 -0400
@@ -1,8 +1,8 @@
 #----- begin Korean support package (2011-sep-02)
 cidToUnicode	Adobe-Korea1	/usr/share/xpdf/korean/Adobe-Korea1.cidToUnicode
 unicodeMap	ISO-2022-KR	/usr/share/xpdf/korean/ISO-2022-KR.unicodeMap
-cMapDir		Adobe-Korea1	/usr/share/xpdf/korean/CMap
-toUnicodeDir			/usr/share/xpdf/korean/CMap
+#cMapDir		Adobe-Korea1	/usr/share/xpdf/korean/CMap
+#toUnicodeDir			/usr/share/xpdf/korean/CMap
 #fontFileCC	Adobe-Korea1	/usr/..../batang.ttf"
 #fontFileCC	Unidocs-Korea1	/usr/..../batang.ttf"
 #----- end Korean support package
