diff -ru pybind11-2.13.6/pybind11/commands.py pybind11-2.13.6_new/pybind11/commands.py
--- pybind11-2.13.6/pybind11/commands.py	2025-04-30 13:21:58.626584173 +0200
+++ pybind11-2.13.6_new/pybind11/commands.py	2025-04-30 13:22:22.270871704 +0200
@@ -2,7 +2,7 @@
 
 import os
 
-DIR = os.path.abspath(os.path.dirname(__file__))
+DIR = "/usr"
 
 
 def get_include(user: bool = False) -> str:  # noqa: ARG001
