diff -urN pybind11-2.11.1/pybind11/commands.py pybind11-2.11.1/pybind11/commands.py
--- pybind11-2.11.1/pybind11/commands.py	2024-03-18 12:25:44.994329051 -0700
+++ pybind11-2.11.1/pybind11/commands.py	2024-03-18 12:27:05.558472628 -0700
@@ -8,21 +8,13 @@
     Return the path to the pybind11 include directory. The historical "user"
     argument is unused, and may be removed.
     """
-    installed_path = os.path.join(DIR, "include")
-    source_path = os.path.join(os.path.dirname(DIR), "include")
-    return installed_path if os.path.exists(installed_path) else source_path
-
+    return '/usr/include/pybind11'
 
 def get_cmake_dir() -> str:
     """
     Return the path to the pybind11 CMake module directory.
     """
-    cmake_installed_path = os.path.join(DIR, "share", "cmake", "pybind11")
-    if os.path.exists(cmake_installed_path):
-        return cmake_installed_path
-
-    msg = "pybind11 not installed, installation required to access the CMake files"
-    raise ImportError(msg)
+    return '/usr/share/cmake/pybind11'
 
 
 def get_pkgconfig_dir() -> str:
