diff -up ntfs-3g_ntfsprogs-2011.10.9-RC/ntfsprogs/ntfsck.c.OLD ntfs-3g_ntfsprogs-2011.10.9-RC/ntfsprogs/ntfsck.c
--- ntfs-3g_ntfsprogs-2011.10.9-RC/ntfsprogs/ntfsck.c.OLD	2011-10-11 10:24:02.381335115 -0400
+++ ntfs-3g_ntfsprogs-2011.10.9-RC/ntfsprogs/ntfsck.c	2011-10-11 10:26:41.513559206 -0400
@@ -877,7 +877,11 @@ int main(int argc, char **argv)
 	if (errors)
 		return 2;
 	if (unsupported)
-		return 1;
+		ntfs_log_info("ntfsck was unable to run properly.\n");
+		// If we return 1 here, we fail for ntfs services fscking on boot just because
+		// ntfsck isn't smart enough to handle 99% of cases. So, we just return 0.
+		// return 1;
+		return 0;
 	return 0;
 }
 
