From 70ed7929018bf228c70932ba077f6d6bf12e8176 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Mon, 5 Oct 2015 15:05:17 +0200
Subject: [PATCH] Declare POD encoding
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 Multicore.pm | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Multicore.pm b/Multicore.pm
index 4b1b091..ae76ed5 100644
--- a/Multicore.pm
+++ b/Multicore.pm
@@ -1,3 +1,5 @@
+=encoding utf8
+
 =head1 NAME
 
 Coro::Multicore - make coro threads on multiple cores with specially supported modules
-- 
2.4.3

