diff -up sblim-sfcc-2.2.8/Makefile.in.orig sblim-sfcc-2.2.8/Makefile.in
--- sblim-sfcc-2.2.8/Makefile.in.orig	2018-10-11 11:54:07.504330059 +0200
+++ sblim-sfcc-2.2.8/Makefile.in	2018-10-11 11:54:40.301366853 +0200
@@ -290,7 +290,7 @@ build_vendor = @build_vendor@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/sfcc-$(VERSION)
+docdir = $(datadir)/doc/sblim-sfcc
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
@@ -355,7 +355,7 @@ dist_man_MANS = man/man3/CMCIClient.3 ma
 	man/man3/CMPIStatus.3 man/man3/CMPIString.3 \
 	man/man3/CMPIValue.3
 
-dist_doc_DATA = README AUTHORS COPYING
+dist_doc_DATA = README AUTHORS
 EXTRA_DIST = sfcc.spec
 noinst_HEADERS = \
 	backend/cimxml/cimXmlParser.h \
