--- object-0.39.1/Cargo.toml	2006-07-24T01:21:28+00:00
+++ object-0.39.1/Cargo.toml	2026-08-10T12:48:50.309518+00:00
@@ -53,7 +53,6 @@
     "build",
     "std",
     "compression",
-    "wasm",
 ]
 archive = []
 build = [
@@ -102,7 +101,6 @@
     "all",
     "unstable",
 ]
-wasm = ["dep:wasmparser"]
 write = [
     "write_std",
     "coff",
@@ -174,11 +172,6 @@
 version = "0.8.1"
 optional = true
 
-[dependencies.wasmparser]
-version = "0.247.0"
-optional = true
-default-features = false
-
 [lints.clippy]
 collapsible_else_if = "allow"
 collapsible_if = "allow"
