diff -Nurp sysfsutils-1.2.0.bak/COPYING sysfsutils-1.2.0/COPYING
--- sysfsutils-1.2.0.bak/COPYING	2004-09-13 08:45:15.000000000 -0500
+++ sysfsutils-1.2.0/COPYING	2004-10-11 13:05:00.483795976 -0500
@@ -1,11 +1,10 @@
 
-The commands and utilities under the "test" directory are licensed under the 
-GNU General Public License (GPL) Version 2, June 1991. The full text of the 
-GPL is located at:
+The commands and utilities are licensed under the GNU General Public License
+(GPL) Version 2, June 1991. The full text of the GPL is located at:
 
-sysfsutils/cmd/GPL
+/usr/share/doc/sysfsutils-{version}/GPL
 
 The sysfs library is licensed under the GNU Lesser Public License (LGPL)
 Version 2.1, February 1999. The full text of the LGPL is located at:
 
-sysfsutils/lib/LGPL
+/usr/share/doc/sysfsutils-{version}/LGPL
