diff -up GConf-3.2.6/gconf/gconftool.c.pkill-hack GConf-3.2.6/gconf/gconftool.c
--- GConf-3.2.6/gconf/gconftool.c.pkill-hack	2013-04-17 10:46:13.225426329 -0400
+++ GConf-3.2.6/gconf/gconftool.c	2013-04-17 10:48:25.963508300 -0400
@@ -916,8 +916,8 @@ main (int argc, char** argv)
 
       use_local_source = TRUE;
 
-      /* shut down daemon, this is a race condition, but will usually work. */
-      gconf_shutdown_daemon (NULL);
+      g_spawn_command_line_sync ("/usr/bin/pkill -HUP -f /usr/libexec/gconfd-2",
+                                 NULL, NULL, NULL, NULL);
     }
 
   if (config_source == NULL)
