diff -ruN a/fluent-plugin-s3.gemspec b/fluent-plugin-s3.gemspec
--- a/fluent-plugin-s3.gemspec	2022-07-07 17:38:36.513929716 -0700
+++ b/fluent-plugin-s3.gemspec	2022-07-07 17:40:03.263016858 -0700
@@ -11,9 +11,9 @@
   gem.authors     = ["Sadayuki Furuhashi", "Masahiro Nakagawa"]
   gem.email       = "frsyuki@gmail.com"
   #gem.platform    = Gem::Platform::RUBY
-  gem.files       = `git ls-files`.split("\n")
-  gem.test_files  = `git ls-files -- {test,spec,features}/*`.split("\n")
-  gem.executables = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
+  gem.files       = Dir['**/*']
+  gem.test_files  = gem.files.grep(%r{^(test|spec|features)/})
+  gem.executables = gem.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   gem.require_paths = ['lib']
 
   gem.add_dependency "fluentd", [">= 0.14.22", "< 2"]
