diff -up DBD-SQLite-1.27/Makefile.PL.old DBD-SQLite-1.27/Makefile.PL
--- DBD-SQLite-1.27/Makefile.PL.old	2009-11-23 12:08:05.000000000 +0100
+++ DBD-SQLite-1.27/Makefile.PL	2010-01-11 15:40:56.456399143 +0100
@@ -129,7 +129,7 @@ if ( $@ or DBI->VERSION < $DBI_required 
 # a system sqlite is also sophisticated enough to have a patching system
 # that can change the if ( 0 ) to if ( 1 )
 my ($sqlite_local, $sqlite_base, $sqlite_lib, $sqlite_inc);
-if ( 0 ) {
+if ( 1 ) {
 	require File::Spec;
 	if ( $sqlite_base = (grep(/SQLITE_LOCATION=.*/, @ARGV))[0] ) {
 		$sqlite_base =~ /=(.*)/;
