--- xerces-2_11_0/src/org/apache/html/dom/HTMLFrameElementImpl.java	2010-11-26 21:42:02.000000000 +0100
+++ xerces-2_11_0/src/org/apache/html/dom/HTMLFrameElementImpl.java	2017-10-03 17:28:29.096819783 +0200
@@ -17,6 +17,7 @@
 package org.apache.html.dom;
 
 import org.w3c.dom.html.HTMLFrameElement;
+import org.w3c.dom.Document;
 
 /**
  * @xerces.internal
@@ -127,6 +128,11 @@
         setAttribute( "src", src );
     }
 
+	public Document getContentDocument()
+	{
+		return null;
+	}
+
     
     /**
      * Constructor requires owner document.
--- xerces-2_11_0/src/org/apache/html/dom/HTMLIFrameElementImpl.java	2010-11-26 21:42:08.000000000 +0100
+++ xerces-2_11_0/src/org/apache/html/dom/HTMLIFrameElementImpl.java	2017-10-03 17:28:12.576819720 +0200
@@ -17,6 +17,7 @@
 package org.apache.html.dom;
 
 import org.w3c.dom.html.HTMLIFrameElement;
+import org.w3c.dom.Document;
 
 /**
  * @xerces.internal
@@ -150,6 +151,11 @@
         setAttribute( "width", width );
     }
 
+	public Document getContentDocument()
+	{
+		return null;
+	}
+
     
     /**
      * Constructor requires owner document.
--- xerces-2_11_0/src/org/apache/html/dom/HTMLObjectElementImpl.java	2010-11-26 21:42:05.000000000 +0100
+++ xerces-2_11_0/src/org/apache/html/dom/HTMLObjectElementImpl.java	2017-10-03 17:27:57.048819662 +0200
@@ -17,6 +17,7 @@
 package org.apache.html.dom;
 
 import org.w3c.dom.html.HTMLObjectElement;
+import org.w3c.dom.Document;
 
 /**
  * @xerces.internal
@@ -239,6 +240,10 @@
         setAttribute( "width", width );
     }
 
+	public Document getContentDocument()
+	{
+		return null;
+	}
     
     
     /**
