From 3c0536b120eb304dc6071c760569bdf49053960f Mon Sep 17 00:00:00 2001
From: Diego Herrera <dherrera@redhat.com>
Date: Mon, 30 Dec 2024 18:31:33 -0300
Subject: [PATCH 04/20] disable-power10

---
 cmake/onnxruntime_mlas.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cmake/onnxruntime_mlas.cmake b/cmake/onnxruntime_mlas.cmake
index 0ba4694c32..6951e6a860 100644
--- a/cmake/onnxruntime_mlas.cmake
+++ b/cmake/onnxruntime_mlas.cmake
@@ -423,7 +423,8 @@ else()
           set_source_files_properties(${MLAS_SRC_DIR}/power/QuantizePowerVSX.cpp PROPERTIES COMPILE_FLAGS "-mcpu=power9")
         endif()
 
-        check_cxx_compiler_flag("-mcpu=power10" HAS_POWER10)
+        #check_cxx_compiler_flag("-mcpu=power10" HAS_POWER10)
+        set(HAS_POWER10 OFF)
         if(HAS_POWER10)
           set(CMAKE_REQUIRED_FLAGS "-mcpu=power10")
           check_cxx_source_compiles("
-- 
2.47.1

