diff --color -Nur koji-1.27.0.orig/cli/koji.conf koji-1.27.0/cli/koji.conf
--- koji-1.27.0.orig/cli/koji.conf	2021-11-18 02:28:38.000000000 -0800
+++ koji-1.27.0/cli/koji.conf	2021-11-18 13:33:55.468796788 -0800
@@ -4,12 +4,15 @@
 
 ;url of XMLRPC server
 ;server = http://hub.example.com/kojihub
+server = https://koji.fedoraproject.org/kojihub
 
 ;url of web interface
 ;weburl = http://www.example.com/koji
+weburl = https://koji.fedoraproject.org/koji
 
 ;url of package download site
 ;pkgurl = http://www.example.com/packages
+topurl = https://kojipkgs.fedoraproject.org
 
 ;path to the koji top directory
 ;topdir = /mnt/koji
@@ -22,6 +25,8 @@
 ;the keytab to auth as for automated clients
 ;keytab = /etc/krb5.keytab
 
+; fedora uses kerberos auth
+authtype = kerberos
 
 ;configuration for SSL authentication
 
@@ -39,6 +44,8 @@
 ;[not_implemented_yet]
 ;enabled plugins for CLI, runroot and save_failed_tree are available
 ;plugins =
+; runroot plugin is enabled by default in fedora
+plugins = runroot
 
 ;timeout of XMLRPC requests by seconds, default: 60 * 60 * 12 = 43200
 ;timeout = 43200
@@ -48,3 +55,6 @@
 
 ;enforcing CLI authentication even for anonymous calls
 ;force_auth = False
+
+; use the fast upload feature of koji by default
+use_fast_upload = yes
