--- rustls-pki-types-1.15.1/Cargo.toml	2006-07-24T01:21:28+00:00
+++ rustls-pki-types-1.15.1/Cargo.toml	2026-09-29T10:20:00.766309+00:00
@@ -54,7 +54,7 @@
 alloc = ["dep:zeroize"]
 default = ["alloc"]
 std = ["alloc"]
-web = ["web-time"]
+web = []
 
 [lib]
 name = "rustls_pki_types"
@@ -66,10 +66,6 @@
 
 [dev-dependencies.criterion]
 version = "0.8"
-
-[target.'cfg(all(target_family = "wasm", target_os = "unknown"))'.dependencies.web-time]
-version = "1"
-optional = true
 
 [target.'cfg(all(target_os = "linux", target_arch = "x86_64"))'.dev-dependencies.crabgrind]
 version = "=0.1.9"
@@ -87,3 +83,4 @@
 level = "warn"
 priority = 0
 check-cfg = ["cfg(rustls_pki_types_docsrs)"]
+
