--- rustls-pki-types-1.15.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ rustls-pki-types-1.15.0/Cargo.toml	2026-06-30T16:22:07.293733+00:00
@@ -54,17 +54,13 @@
 alloc = ["dep:zeroize"]
 default = ["alloc"]
 std = ["alloc"]
-web = ["web-time"]
+web = []
 
 [lib]
 name = "rustls_pki_types"
 path = "src/lib.rs"
 
 [dependencies.zeroize]
-version = "1"
-optional = true
-
-[target.'cfg(all(target_family = "wasm", target_os = "unknown"))'.dependencies.web-time]
 version = "1"
 optional = true
 
@@ -84,3 +80,4 @@
 level = "warn"
 priority = 0
 check-cfg = ["cfg(rustls_pki_types_docsrs)"]
+
