From 948cdec79dfb087709917e2ae099de002c4fe400 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Wed, 23 Jul 2025 22:55:33 +0000
Subject: [PATCH] AZL3 -- disable signatures for unsupported rsa sizes in speed
 in fips mode

---
 apps/speed.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/apps/speed.c b/apps/speed.c
index 49c173d..0bfcaa7 100644
--- a/apps/speed.c
+++ b/apps/speed.c
@@ -2558,6 +2558,7 @@ int speed_main(int argc, char **argv)
     // TOBIASB: Disable X25519 and NID_X448 for FIPS provider because it isn't supported.
     // TOBIASB: Disable EdDSA for FIPS provider because it isn't supported. See fipsprov.c.
     // TOBIASB: Disable KEMS for unsupported RSA sizes for FIPS provider because they supported.
+    // TOBIASB: Disable signatures for unsupported RSA sizes for FIPS provider because they supported.
     if (FIPS_mode()) {
         for (i = 0; i < RSA_NUM; i++) {
             if (rsa_keys[i].bits < 2048)
@@ -2571,6 +2572,13 @@ int speed_main(int argc, char **argv)
             }
         }
 
+        for (i = 0; i < sigs_algs_len; i++) {
+            if (sigs_doit[i] && strcmp(sigs_algname[i], "rsa512") == 0
+                || strcmp(sigs_algname[i], "rsa1024") == 0) {
+                sigs_doit[i] = 0;
+            }
+        }
+
         memset(dsa_doit, 0, sizeof(dsa_doit));
         memset(eddsa_doit, 0, sizeof(eddsa_doit));
 
-- 
2.45.4

