From 14e9d11f16db6ad9a3a7e1d3a5a2e4e0ed8b5add Mon Sep 17 00:00:00 2001
From: Tobias Brick <tobiasb@microsoft.com>
Date: Mon, 22 Apr 2024 17:44:10 +0000
Subject: [PATCH] speed skip unavailable dgst

---
 apps/speed.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/apps/speed.c b/apps/speed.c
index 3f8dcca..9d504f2 100644
--- a/apps/speed.c
+++ b/apps/speed.c
@@ -777,6 +777,9 @@ static int EVP_MAC_loop(ossl_unused int algindex, void *args)
     for (count = 0; COND(c[algindex][testnum]); count++) {
         size_t outl;
 
+        if (mctx == NULL)
+            return -1;
+
         if (!EVP_MAC_init(mctx, NULL, 0, NULL)
             || !EVP_MAC_update(mctx, buf, lengths[testnum])
             || !EVP_MAC_final(mctx, mac, &outl, sizeof(mac)))
