--- djvulibre-3.5.27/desktopfiles/Makefile.am
+++ djvulibre-3.5.27/desktopfiles/Makefile.am
@@ -32,10 +32,9 @@ if HAVE_CONVERSION_INKSCAPE
 convert_icons_process = \
 s=`echo $@ | sed -e 's/[a-z]*\([0-9]*\).*/\1/'`; \
 ${INKSCAPE} \
---without-gui \
 --export-width=$${s} \
 --export-height=$${s} \
---export-png=$@ $<
+--export-filename=$@ $<
 endif
 
 if HAVE_CONVERSION_CONVERT
--- djvulibre-3.5.27/desktopfiles/Makefile.in
+++ djvulibre-3.5.27/desktopfiles/Makefile.in
@@ -306,10 +306,9 @@ PNGICONS = \
 @HAVE_CONVERSION_INKSCAPE_TRUE@convert_icons_process = \
 @HAVE_CONVERSION_INKSCAPE_TRUE@s=`echo $@ | sed -e 's/[a-z]*\([0-9]*\).*/\1/'`; \
 @HAVE_CONVERSION_INKSCAPE_TRUE@${INKSCAPE} \
-@HAVE_CONVERSION_INKSCAPE_TRUE@--without-gui \
 @HAVE_CONVERSION_INKSCAPE_TRUE@--export-width=$${s} \
 @HAVE_CONVERSION_INKSCAPE_TRUE@--export-height=$${s} \
-@HAVE_CONVERSION_INKSCAPE_TRUE@--export-png=$@ $<
+@HAVE_CONVERSION_INKSCAPE_TRUE@--export-filename=$@ $<
 
 @HAVE_CONVERSION_RSVG_TRUE@convert_icons_process = \
 @HAVE_CONVERSION_RSVG_TRUE@s=`echo $@ | sed -e 's/[a-z]*\([0-9]*\).*/\1/'`; \
