diff --git a/app/app_kas.c b/app/app_kas.c
index 0e7675a..2e879e6 100644
--- a/app/app_kas.c
+++ b/app/app_kas.c
@@ -171,6 +171,18 @@ int app_kas_ecc_handler(ACVP_TEST_CASE *test_case) {
         case ACVP_SHA512:
             md = EVP_sha512();
             break;
+        case ACVP_SHA3_224:
+            md = EVP_sha3_224();
+            break;
+        case ACVP_SHA3_256:
+            md = EVP_sha3_256();
+            break;
+        case ACVP_SHA3_384:
+            md = EVP_sha3_384();
+            break;
+        case ACVP_SHA3_512:
+            md = EVP_sha3_512();
+            break;
         case ACVP_SHA1:
         case ACVP_SHA512_224:
         case ACVP_SHA512_256:
@@ -515,6 +527,18 @@ int app_kas_ffc_handler(ACVP_TEST_CASE *test_case) {
     case ACVP_SHA512:
         md = EVP_sha512();
         break;
+    case ACVP_SHA3_224:
+        md = EVP_sha3_224();
+        break;
+    case ACVP_SHA3_256:
+        md = EVP_sha3_256();
+        break;
+    case ACVP_SHA3_384:
+        md = EVP_sha3_384();
+        break;
+    case ACVP_SHA3_512:
+        md = EVP_sha3_512();
+        break;
     case ACVP_SHA1:
     case ACVP_SHA512_224:
     case ACVP_SHA512_256:
diff --git a/app/app_main.c b/app/app_main.c
index 4acfef6..87257d1 100755
--- a/app/app_main.c
+++ b/app/app_main.c
@@ -1521,7 +1521,6 @@ end:
 
 static int enable_kas_ecc(ACVP_CTX *ctx) {
     ACVP_RESULT rv = ACVP_SUCCESS;
-#ifdef ACVP_ENABLE_DEPRECATED_VERSION
     /*
      * Enable KAS-ECC....
      */
@@ -1586,7 +1585,7 @@ static int enable_kas_ecc(ACVP_CTX *ctx) {
     CHECK_ENABLE_CAP_RV(rv);
     rv = acvp_cap_kas_ecc_set_scheme(ctx, ACVP_KAS_ECC_COMP, ACVP_KAS_ECC_MODE_COMPONENT, ACVP_KAS_ECC_EPHEMERAL_UNIFIED, ACVP_KAS_ECC_EE, ACVP_EC_CURVE_P521, ACVP_SHA512);
     CHECK_ENABLE_CAP_RV(rv);
-#endif
+
     /* Support is for ECC-SSC for hashZ only */
     rv = acvp_cap_kas_ecc_enable(ctx, ACVP_KAS_ECC_SSC, &app_kas_ecc_handler);
     CHECK_ENABLE_CAP_RV(rv);
diff --git a/src/acvp_kas_ecc.c b/src/acvp_kas_ecc.c
index eec60f9..42ebf40 100644
--- a/src/acvp_kas_ecc.c
+++ b/src/acvp_kas_ecc.c
@@ -1077,7 +1077,9 @@ static ACVP_RESULT acvp_kas_ecc_ssc(ACVP_CTX *ctx,
         }
         hash = acvp_lookup_hash_alg(hash_str);
         if (!(hash == ACVP_SHA224 || hash == ACVP_SHA256 ||
-              hash == ACVP_SHA384 || hash == ACVP_SHA512)) {
+              hash == ACVP_SHA384 || hash == ACVP_SHA512 ||
+              hash == ACVP_SHA3_224 || hash == ACVP_SHA3_256 ||
+              hash == ACVP_SHA3_384 || hash == ACVP_SHA3_512)) {
             ACVP_LOG_ERR("Server JSON invalid 'hashAlg'");
             rv = ACVP_INVALID_ARG;
             goto err;
diff --git a/src/acvp_kas_ffc.c b/src/acvp_kas_ffc.c
index 601395e..9e8b1eb 100644
--- a/src/acvp_kas_ffc.c
+++ b/src/acvp_kas_ffc.c
@@ -324,7 +324,9 @@ static ACVP_RESULT acvp_kas_ffc_comp(ACVP_CTX *ctx,
         }
         hash_alg = acvp_lookup_hash_alg(hash_str);
         if (hash_alg != ACVP_SHA224 && hash_alg != ACVP_SHA256 &&
-            hash_alg != ACVP_SHA384 && hash_alg != ACVP_SHA512) {
+            hash_alg != ACVP_SHA384 && hash_alg != ACVP_SHA512 &&
+            hash_alg != ACVP_SHA3_224 && hash_alg != ACVP_SHA3_256 &&
+            hash_alg != ACVP_SHA3_384 && hash_alg != ACVP_SHA3_512) {
             ACVP_LOG_ERR("Server JSON invalid 'hashAlg'");
             rv = ACVP_INVALID_ARG;
             goto err;
@@ -792,7 +794,9 @@ static ACVP_RESULT acvp_kas_ffc_ssc(ACVP_CTX *ctx,
         }
         hash_alg = acvp_lookup_hash_alg(hash_str);
         if (hash_alg != ACVP_SHA224 && hash_alg != ACVP_SHA256 &&
-            hash_alg != ACVP_SHA384 && hash_alg != ACVP_SHA512) {
+            hash_alg != ACVP_SHA384 && hash_alg != ACVP_SHA512 &&
+            hash_alg != ACVP_SHA3_224 && hash_alg != ACVP_SHA3_256 &&
+            hash_alg != ACVP_SHA3_384 && hash_alg != ACVP_SHA3_512) {
             ACVP_LOG_ERR("Server JSON invalid 'hashAlg'");
             rv = ACVP_INVALID_ARG;
             goto err;
