diff -up sblim-cmpi-syslog-0.9.0/Makefile.in.orig sblim-cmpi-syslog-0.9.0/Makefile.in
--- sblim-cmpi-syslog-0.9.0/Makefile.in.orig	2013-12-12 16:53:26.882151537 +0100
+++ sblim-cmpi-syslog-0.9.0/Makefile.in	2013-12-12 16:53:42.065205722 +0100
@@ -228,7 +228,7 @@ datadir = @datadir@
 datarootdir = @datarootdir@
 
 # docdir defines where the documentation goes
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
diff -up sblim-cmpi-syslog-0.9.0/syslog-conf/Makefile.in.orig sblim-cmpi-syslog-0.9.0/syslog-conf/Makefile.in
--- sblim-cmpi-syslog-0.9.0/syslog-conf/Makefile.in.orig	2013-07-24 18:53:26.000000000 +0200
+++ sblim-cmpi-syslog-0.9.0/syslog-conf/Makefile.in	2013-12-12 17:07:04.382041958 +0100
@@ -254,7 +254,7 @@ datadir = @datadir@
 datarootdir = @datarootdir@
 
 # docdir defines where the documentation goes
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
diff -up sblim-cmpi-syslog-0.9.0/syslog-log/Makefile.in.orig sblim-cmpi-syslog-0.9.0/syslog-log/Makefile.in
--- sblim-cmpi-syslog-0.9.0/syslog-log/Makefile.in.orig	2013-12-12 17:07:12.975069208 +0100
+++ sblim-cmpi-syslog-0.9.0/syslog-log/Makefile.in	2013-12-12 17:07:20.734097726 +0100
@@ -244,7 +244,7 @@ datadir = @datadir@
 datarootdir = @datarootdir@
 
 # docdir defines where the documentation goes
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
diff -up sblim-cmpi-syslog-0.9.0/syslog-service/Makefile.in.orig sblim-cmpi-syslog-0.9.0/syslog-service/Makefile.in
--- sblim-cmpi-syslog-0.9.0/syslog-service/Makefile.in.orig	2013-12-12 17:07:27.615120372 +0100
+++ sblim-cmpi-syslog-0.9.0/syslog-service/Makefile.in	2013-12-12 17:07:36.158150574 +0100
@@ -231,7 +231,7 @@ datadir = @datadir@
 datarootdir = @datarootdir@
 
 # docdir defines where the documentation goes
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
