diff -up sblim-cmpi-syslog-0.9.0/syslog-conf/mof/Syslog_ConfigurationRegistration.mof.bak sblim-cmpi-syslog-0.9.0/syslog-conf/mof/Syslog_ConfigurationRegistration.mof
--- sblim-cmpi-syslog-0.9.0/syslog-conf/mof/Syslog_ConfigurationRegistration.mof.bak	2009-05-22 05:50:48.000000000 +0200
+++ sblim-cmpi-syslog-0.9.0/syslog-conf/mof/Syslog_ConfigurationRegistration.mof	2013-08-05 16:57:17.691147468 +0200
@@ -19,7 +19,7 @@
 
 
 #pragma locale ("en_US")
-#pragma namespace ("root/PG_InterOp")
+#pragma namespace ("root/interop")
 
 // Syslog_Configuration 
 
diff -up sblim-cmpi-syslog-0.9.0/syslog-conf/provider-register.sh.bak sblim-cmpi-syslog-0.9.0/syslog-conf/provider-register.sh
--- sblim-cmpi-syslog-0.9.0/syslog-conf/provider-register.sh.bak	2009-05-22 05:50:48.000000000 +0200
+++ sblim-cmpi-syslog-0.9.0/syslog-conf/provider-register.sh	2013-08-05 16:59:33.132147468 +0200
@@ -216,7 +216,7 @@ pegasus_install()
     then
 	chatter Registering providers with $state cimserver
 	$CIMMOF -uc -I $mofpath -n $namespace $mymofs &&
-	$CIMMOF -uc -n root/PG_Interop $_REGFILENAME
+	$CIMMOF -uc -n root/interop $_REGFILENAME
     else
 	echo "Failed to build pegasus registration MOF." >&2
 	return 1
diff -up sblim-cmpi-syslog-0.9.0/syslog-log/mof/Syslog_LogRegistration.mof.bak sblim-cmpi-syslog-0.9.0/syslog-log/mof/Syslog_LogRegistration.mof
--- sblim-cmpi-syslog-0.9.0/syslog-log/mof/Syslog_LogRegistration.mof.bak	2009-05-22 05:50:49.000000000 +0200
+++ sblim-cmpi-syslog-0.9.0/syslog-log/mof/Syslog_LogRegistration.mof	2013-08-05 17:00:02.312147468 +0200
@@ -20,7 +20,7 @@
 
 
 #pragma locale ("en_US")
-#pragma namespace ("root/PG_InterOp")
+#pragma namespace ("root/interop")
 
 // Syslog_MessageLog 
 
diff -up sblim-cmpi-syslog-0.9.0/syslog-log/provider-register.sh.bak sblim-cmpi-syslog-0.9.0/syslog-log/provider-register.sh
--- sblim-cmpi-syslog-0.9.0/syslog-log/provider-register.sh.bak	2009-05-22 05:50:49.000000000 +0200
+++ sblim-cmpi-syslog-0.9.0/syslog-log/provider-register.sh	2013-08-05 16:59:45.854147468 +0200
@@ -216,7 +216,7 @@ pegasus_install()
     then
 	chatter Registering providers with $state cimserver
 	$CIMMOF -uc -I $mofpath -n $namespace $mymofs &&
-	$CIMMOF -uc -n root/PG_Interop $_REGFILENAME
+	$CIMMOF -uc -n root/interop $_REGFILENAME
     else
 	echo "Failed to build pegasus registration MOF." >&2
 	return 1
diff -up sblim-cmpi-syslog-0.9.0/syslog-service/mof/Syslog_ServiceRegistration.mof.bak sblim-cmpi-syslog-0.9.0/syslog-service/mof/Syslog_ServiceRegistration.mof
--- sblim-cmpi-syslog-0.9.0/syslog-service/mof/Syslog_ServiceRegistration.mof.bak	2009-05-22 05:50:49.000000000 +0200
+++ sblim-cmpi-syslog-0.9.0/syslog-service/mof/Syslog_ServiceRegistration.mof	2013-08-05 17:01:52.330147468 +0200
@@ -20,7 +20,7 @@
 
 
 #pragma locale ("en_US")
-#pragma namespace ("root/PG_InterOp")
+#pragma namespace ("root/interop")
 
 // Syslog_Service 
 
diff -up sblim-cmpi-syslog-0.9.0/syslog-service/provider-register.sh.bak sblim-cmpi-syslog-0.9.0/syslog-service/provider-register.sh
--- sblim-cmpi-syslog-0.9.0/syslog-service/provider-register.sh.bak	2009-05-22 05:50:49.000000000 +0200
+++ sblim-cmpi-syslog-0.9.0/syslog-service/provider-register.sh	2013-08-05 17:01:35.871147468 +0200
@@ -216,7 +216,7 @@ pegasus_install()
     then
 	chatter Registering providers with $state cimserver
 	$CIMMOF -uc -I $mofpath -n $namespace $mymofs &&
-	$CIMMOF -uc -n root/PG_Interop $_REGFILENAME
+	$CIMMOF -uc -n root/interop $_REGFILENAME
     else
 	echo "Failed to build pegasus registration MOF." >&2
 	return 1
