diff -up sblim-cmpi-syslog-0.9.0/syslog-conf/util/syslogconfutil.c.orig sblim-cmpi-syslog-0.9.0/syslog-conf/util/syslogconfutil.c
--- sblim-cmpi-syslog-0.9.0/syslog-conf/util/syslogconfutil.c.orig	2013-07-24 18:43:12.000000000 +0200
+++ sblim-cmpi-syslog-0.9.0/syslog-conf/util/syslogconfutil.c	2015-11-24 11:43:27.541610507 +0100
@@ -22,36 +22,57 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
+#include <unistd.h>
 
 #include "syslogconfutil.h"
 
+// has to be more than strlen of ConfInFile
+#define NAME_LEN 1024
+// approximate strlen of constant part in "find /etc -maxdepth 1 -name..."
+#define CMD_LEN 35
+
 typedef struct {
   FILE *fp;
-  char name[L_tmpnam];
+  char name[NAME_LEN];
 } Syslog_Control;
 
 void* Syslog_Conf_Begin_Enum()
 {
-  char cmdbuffer[1024];
-  Syslog_Control *cc = malloc(sizeof(Syslog_Control));
+  char *pathbuffer = NULL;
+  int path_len = 0;
+  char *cmdbuffer = NULL;
+  int cmd_len = 0;
+  int tfd = -1;
+  const char *const proc_path = "/proc/self/fd/";
+  char template[] = "/tmp/tmpsyslogXXXXXX";
+  Syslog_Control *cc = (Syslog_Control *) malloc(sizeof(Syslog_Control));
 
   syslog_debug(stderr, "--- Syslog_Conf_Begin_Enum() \n");
-  
-  bzero(&cmdbuffer, sizeof(cmdbuffer));
 
   // begin enumeration
-  if (cc && tmpnam(cc->name)) {
-    sprintf(cmdbuffer,
+  if (cc && (tfd = mkstemp(template)) != -1) {
+    unlink(template);
+    // get path to file associated to file descriptor
+    path_len = snprintf(NULL, 0, "%s%u", proc_path, tfd) + 1;
+    pathbuffer = (char *) malloc(path_len * sizeof(char));
+    snprintf(pathbuffer, path_len, "%s%u", proc_path, tfd);
+    // prepare command
+    cmd_len = path_len + strlen(ConfFileName) + CMD_LEN;
+    cmdbuffer = (char *) calloc(cmd_len, sizeof(char));
+    snprintf(cmdbuffer, cmd_len,
 	     "find /etc -maxdepth 1 -name %s"
              " > %s",
-             ConfFileName, cc->name);
-    if (system (cmdbuffer) != 0 ||
-        (cc->fp = fopen(cc->name,"r")) == NULL) {
+             ConfFileName, pathbuffer);
+    // run command
+    if (system(cmdbuffer) != 0 ||
+        (cc->fp = fdopen(tfd,"r")) == NULL) {
       free(cc);
       cc=NULL;
     }
   }
 
+  free(pathbuffer);
+  free(cmdbuffer);
   return cc;
 }
 
@@ -65,7 +86,7 @@ void* Syslog_Setting_Begin_Enum()
   // begin enumeration
   ret = setting_read();	  
   if (!ret && (cc->fp = fopen(ConfInFile, "r")) != NULL) {
-    strcpy(cc->name, ConfInFile);	    
+    strncpy(cc->name, ConfInFile, NAME_LEN);
   }
   else {
     syslog_debug(stderr, "Unable to sort the file\n");
@@ -141,7 +162,6 @@ void Syslog_Conf_End_Enum(void *handle)
   if (cc) {
     if (cc->fp)
       fclose(cc->fp);
-    remove(cc->name);
     free(cc);
   }
 }
