diff -up pegasus/mak/config-linux.mak.orig pegasus/mak/config-linux.mak
--- pegasus/mak/config-linux.mak.orig	2015-03-31 15:25:23.735154221 +0200
+++ pegasus/mak/config-linux.mak	2015-03-31 15:26:29.349434918 +0200
@@ -127,6 +127,7 @@ else
   # appeared in the 3.0 series of compilers.
   #
   ifeq ($(COMPILER), gnu)
+   FLAGS += -g
    # disable the strict aliasing
    ifeq ($(shell expr $(GCC_VERSION) '>=' 3.0), 1)
      PEGASUS_EXTRA_CXX_FLAGS += -fno-enforce-eh-specs -fno-strict-aliasing
