--- io-lifetimes-3.0.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ io-lifetimes-3.0.1/Cargo.toml	2026-08-14T07:21:51.418311+00:00
@@ -48,8 +48,6 @@
 [features]
 close = [
     "libc",
-    "hermit-abi",
-    "windows-sys",
 ]
 default = []
 
@@ -95,22 +93,8 @@
 version = "0.2.96"
 optional = true
 
-[target.'cfg(target_os = "hermit")'.dependencies.hermit-abi]
-version = ">=0.3, <=0.5"
-optional = true
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = ">=0.52, <=0.60"
-features = [
-    "Win32_Foundation",
-    "Win32_Storage_FileSystem",
-    "Win32_Networking_WinSock",
-    "Win32_Security",
-    "Win32_System_IO",
-]
-optional = true
-
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
 check-cfg = ["cfg(wasi_ext)"]
+
