diff --git a/tests/lib/subdir.am b/tests/lib/subdir.am
index ca74306..9b0980f 100644
--- a/tests/lib/subdir.am
+++ b/tests/lib/subdir.am
@@ -25,20 +25,6 @@ copy_script: tests/lib/script1.lua
 	$(INSTALL_SCRIPT) $< tests/lib/script1.lua
 
 ##############################################################################
-GRPC_TESTS_LDADD = mgmtd/libmgmt_be_nb.la staticd/libstatic.a grpc/libfrrgrpc_pb.la $(GRPC_LIBS) $(ALL_TESTS_LDADD) $(LIBYANG_LIBS) -lm
-
-if GRPC
-check_PROGRAMS += tests/lib/test_grpc
-endif
-tests_lib_test_grpc_CXXFLAGS = $(WERROR) $(TESTS_CXXFLAGS)
-tests_lib_test_grpc_CPPFLAGS = $(TESTS_CPPFLAGS)
-tests_lib_test_grpc_LDADD = $(GRPC_TESTS_LDADD)
-tests_lib_test_grpc_SOURCES = tests/lib/test_grpc.cpp
-nodist_tests_lib_test_grpc_SOURCES = \
-	yang/frr-bfdd.yang.c \
-	yang/frr-staticd.yang.c \
-	# end
-
 
 ##############################################################################
 if ZEROMQ
