--- a/python/mozbuild/mozbuild/frontend/emitter.py
+++ b/python/mozbuild/mozbuild/frontend/emitter.py
@@ -1127,11 +1127,6 @@
                 raise SandboxValidationError('Path specified in LOCAL_INCLUDES '
                     'does not exist: %s (resolved to %s)' % (local_include,
                     full_path), context)
-            if (full_path == context.config.topsrcdir or
-                    full_path == context.config.topobjdir):
-                raise SandboxValidationError('Path specified in LOCAL_INCLUDES '
-                    'is not allowed: %s (resolved to %s)' % (local_include,
-                    full_path), context)
             include_obj = LocalInclude(context, local_include)
             local_includes.append(include_obj.path.full_path)
             yield include_obj

