--- path-absolutize-3.1.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ path-absolutize-3.1.1/Cargo.toml	2024-10-23T21:35:24.730755+00:00
@@ -20,7 +20,6 @@
     "Cargo.toml",
     "README.md",
     "LICENSE",
-    "benches/bench.rs",
 ]
 description = "A library for extending `Path` and `PathBuf` in order to get an absolute path and remove the containing dots."
 homepage = "https://magiclen.org/path-absolutize"
@@ -39,15 +38,8 @@
 license = "MIT"
 repository = "https://github.com/magiclen/path-absolutize"
 
-[[bench]]
-name = "bench"
-harness = false
-
 [dependencies.path-dedot]
 version = "3.1.1"
-
-[dev-dependencies.bencher]
-version = "0.1.5"
 
 [features]
 lazy_static_cache = ["path-dedot/lazy_static_cache"]
