--- rusqlite-0.31.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ rusqlite-0.31.0/Cargo.toml	2025-12-18T19:46:14.319411+00:00
@@ -67,7 +67,6 @@
 name = "load_extension"
 required-features = [
     "load_extension",
-    "bundled",
     "functions",
     "trace",
 ]
@@ -110,14 +109,10 @@
 version = "0.1"
 
 [dependencies.hashlink]
-version = "0.9"
+version = ">=0.9, <0.11"
 
 [dependencies.libsqlite3-sys]
 version = "0.28.0"
-
-[dependencies.rusqlite-macros]
-version = "0.2.0"
-optional = true
 
 [dependencies.serde_json]
 version = "1.0"
@@ -143,9 +138,6 @@
 version = "1.0"
 optional = true
 
-[dev-dependencies.bencher]
-version = "0.1"
-
 [dev-dependencies.doc-comment]
 version = "0.3"
 
@@ -170,23 +162,6 @@
 backup = []
 blob = []
 buildtime_bindgen = ["libsqlite3-sys/buildtime_bindgen"]
-bundled = [
-    "libsqlite3-sys/bundled",
-    "modern_sqlite",
-]
-bundled-full = [
-    "modern-full",
-    "bundled",
-]
-bundled-sqlcipher = [
-    "libsqlite3-sys/bundled-sqlcipher",
-    "bundled",
-]
-bundled-sqlcipher-vendored-openssl = [
-    "libsqlite3-sys/bundled-sqlcipher-vendored-openssl",
-    "bundled-sqlcipher",
-]
-bundled-windows = ["libsqlite3-sys/bundled-windows"]
 collation = []
 column_decltype = []
 csvtab = [
@@ -197,13 +172,8 @@
 functions = []
 hooks = []
 i128_blob = []
-in_gecko = [
-    "modern_sqlite",
-    "libsqlite3-sys/in_gecko",
-]
 limits = []
 load_extension = []
-loadable_extension = ["libsqlite3-sys/loadable_extension"]
 modern-full = [
     "array",
     "backup",
@@ -229,7 +199,7 @@
     "vtab",
     "window",
 ]
-modern_sqlite = ["libsqlite3-sys/bundled_bindings"]
+modern_sqlite = []
 release_memory = []
 serialize = ["modern_sqlite"]
 series = ["vtab"]
@@ -237,13 +207,10 @@
     "libsqlite3-sys/session",
     "hooks",
 ]
-sqlcipher = ["libsqlite3-sys/sqlcipher"]
 trace = []
 unlock_notify = ["libsqlite3-sys/unlock_notify"]
 vtab = []
-wasm32-wasi-vfs = ["libsqlite3-sys/wasm32-wasi-vfs"]
 window = ["functions"]
-with-asan = ["libsqlite3-sys/with-asan"]
 
 [badges.appveyor]
 repository = "rusqlite/rusqlite"
