diff --git a/configure b/configure
index 29b1429..d0eb4be 100755
--- a/configure
+++ b/configure
@@ -18677,32 +18677,32 @@ else
   use_malshow=yes
 fi
 
-if test $use_malshow = yes
-then
-  { echo "$as_me:$LINENO: checking for GTK+ 2.8 or later" >&5
-echo $ECHO_N "checking for GTK+ 2.8 or later... $ECHO_C" >&6; }
-  if ! $SHELL -c 'pkg-config --atleast-version=2.8 gtk+-2.0' >/dev/null 2>/dev/null
-  then
-    use_malshow=no
-  fi
-  { echo "$as_me:$LINENO: result: $use_malshow" >&5
-echo "${ECHO_T}$use_malshow" >&6; }
-  if test $use_malshow = no
-  then
-    echo 1>&2 "*** No GTK+, version 2.8 or later, found. Disabling malshow."
-    echo 1>&2 "*** You may get GTK+ from <http://www.gtk.org>."
-  fi
-fi
-if test $use_malshow = yes
-then
-  GTK_LIBS=`pkg-config gtk+-2.0 --libs`
-  GTK_CFLAGS=`pkg-config gtk+-2.0 --cflags`
-  MALSHOW="malshow"
-else
-  GTK_LIBS=""
-  GTK_CFLAGS=""
-  MALSHOW=""
-fi
+# if test $use_malshow = yes
+# then
+#   { echo "$as_me:$LINENO: checking for GTK+ 2.8 or later" >&5
+# echo $ECHO_N "checking for GTK+ 2.8 or later... $ECHO_C" >&6; }
+#   if ! $SHELL -c 'pkg-config --atleast-version=2.8 gtk+-2.0' >/dev/null 2>/dev/null
+#   then
+#     use_malshow=no
+#   fi
+#   { echo "$as_me:$LINENO: result: $use_malshow" >&5
+# echo "${ECHO_T}$use_malshow" >&6; }
+#   if test $use_malshow = no
+#   then
+#     echo 1>&2 "*** No GTK+, version 2.8 or later, found. Disabling malshow."
+#     echo 1>&2 "*** You may get GTK+ from <http://www.gtk.org>."
+#   fi
+# fi
+# if test $use_malshow = yes
+# then
+#   GTK_LIBS=`pkg-config gtk+-2.0 --libs`
+#   GTK_CFLAGS=`pkg-config gtk+-2.0 --cflags`
+#   MALSHOW="malshow"
+# else
+#   GTK_LIBS=""
+#   GTK_CFLAGS=""
+#   MALSHOW=""
+# fi
 
 { echo "$as_me:$LINENO: checking for GLib" >&5
 echo $ECHO_N "checking for GLib... $ECHO_C" >&6; }
