diff -Nru malaga-7.12-orig/files.c malaga-7.12/files.c
--- malaga-7.12-orig/files.c	2009-06-10 18:02:28.000000000 +0300
+++ malaga-7.12/files.c	2009-06-10 18:04:38.000000000 +0300
@@ -131,10 +131,10 @@
 /*---------------------------------------------------------------------------*/
 
 void 
-map_file( string_t file_name, void **address, int_t *length )
+malaga_map_file( string_t file_name, void **address, int_t *length )
 /* Map file "file_name" into the memory. It will be available in the 
  * memory region starting at *ADDRESS and will occupy LENGTH bytes.
- * After usage, return the memory region via "unmap_file". */
+ * After usage, return the memory region via "malaga_unmap_file". */
 { 
 #ifdef POSIX
   int file_descriptor;
@@ -186,8 +186,8 @@
 /*---------------------------------------------------------------------------*/
 
 void 
-unmap_file( void **address, int_t length )
-/* Return the memory region that has been allocated by "map_file".
+malaga_unmap_file( void **address, int_t length )
+/* Return the memory region that has been allocated by "malaga_map_file".
  * The region starts at *ADDRESS and occupies LENGTH bytes. */
 { 
 #ifdef POSIX
diff -Nru malaga-7.12-orig/files.h malaga-7.12/files.h
--- malaga-7.12-orig/files.h	2009-06-10 18:02:28.000000000 +0300
+++ malaga-7.12/files.h	2009-06-10 18:05:19.000000000 +0300
@@ -45,13 +45,13 @@
  * which is connected to file FILE_NAME, into allocated memory block,
  * and return a pointer to that block. */
 
-extern void map_file( string_t file_name, void **address, int_t *length );
+extern void malaga_map_file( string_t file_name, void **address, int_t *length );
 /* Map file "file_name" into the memory. It will be available in the 
  * memory region starting at *ADDRESS and will occupy LENGTH bytes.
- * After usage, return the memory region via "unmap_file". */
+ * After usage, return the memory region via "malaga_unmap_file". */
 
-extern void unmap_file( void **address, int_t length );
-/* Return the memory region that has been allocated by "map_file".
+extern void malaga_unmap_file( void **address, int_t length );
+/* Return the memory region that has been allocated by "malaga_map_file".
  * The region starts at *ADDRESS and occupies LENGTH bytes. */
 
 /* File name operations. ====================================================*/
diff -Nru malaga-7.12-orig/lex_compiler.c malaga-7.12/lex_compiler.c
--- malaga-7.12-orig/lex_compiler.c	2009-06-10 18:02:28.000000000 +0300
+++ malaga-7.12/lex_compiler.c	2009-06-10 18:06:57.000000000 +0300
@@ -846,7 +846,7 @@
   int_t i;
 
   /* Map file into main memory. */
-  map_file( file_name, &prelex_data, &prelex_length );
+  malaga_map_file( file_name, &prelex_data, &prelex_length );
   header = (prelex_header_t *) prelex_data;
   check_header( &header->common_header, file_name,
 		PRELEX_FILE, MIN_PRELEX_CODE_VERSION, PRELEX_CODE_VERSION );
@@ -859,7 +859,7 @@
     lex_add_allo( strings + entries[i].surface, values + entries[i].feat );
 
   prelex_count = header->entry_count;
-  unmap_file( &prelex_data, prelex_length );
+  malaga_unmap_file( &prelex_data, prelex_length );
   allomorph_count = 0;
 }
 
diff -Nru malaga-7.12-orig/lexicon.c malaga-7.12/lexicon.c
--- malaga-7.12-orig/lexicon.c	2009-06-10 18:02:28.000000000 +0300
+++ malaga-7.12/lexicon.c	2009-06-10 18:06:37.000000000 +0300
@@ -90,7 +90,7 @@
   lexicon_header_t *header; /* Lexicon file header. */
 
   /* Map the lexicon file into memory. */
-  map_file( file_name, &lexicon_data, &lexicon_length );
+  malaga_map_file( file_name, &lexicon_data, &lexicon_length );
 
   /* Check lexicon header. */
   header = (lexicon_header_t *) lexicon_data;
@@ -117,7 +117,7 @@
 terminate_lexicon( void )
 /* Terminate this module. */
 { 
-  unmap_file( &lexicon_data, lexicon_length ); 
+  malaga_unmap_file( &lexicon_data, lexicon_length ); 
 }
 
 /* End of file. =============================================================*/

 	  	 
