diff -Naur malaga-7.12.orig/Makefile.in malaga-7.12/Makefile.in
--- malaga-7.12.orig/Makefile.in	2008-02-28 20:38:21.000000000 +0200
+++ malaga-7.12/Makefile.in	2010-02-10 20:59:54.506702713 +0200
@@ -243,7 +243,7 @@
 malshow: $(MALSHOW_OBJS) libmalaga.la
 	@echo "Linking malshow"
 	@$(LIBTOOL) --mode=link $(LINK) $(MALSHOW_OBJS) $(GTK_LIBS) \
-	  libmalaga.la -o malshow 
+	  -lm libmalaga.la -o malshow 
 
 # malsym ----------------------------------------------------------------------
 
