--- dom4j/src/java/org/dom4j/util/SimpleSingleton.java
+++ dom4j/src/java/org/dom4j/util/SimpleSingleton.java
@@ -35,8 +35,7 @@
         if (singletonClassName != null) {
             Class clazz = null;
             try {
-                clazz = Thread.currentThread().getContextClassLoader().loadClass(
-                        singletonClassName);
+                clazz = Class.forName(singletonClassName);
                 singletonInstance = clazz.newInstance();
             } catch (Exception ignore) {
                 try {
