--- rustc-1.96.1-src/src/tools/cargo/Cargo.lock.orig	2026-06-29 10:12:21.978285655 -0500
+++ rustc-1.96.1-src/src/tools/cargo/Cargo.lock	2026-06-29 10:13:45.744985498 -0500
@@ -3110,7 +3110,6 @@ checksum = "c9b3acc4b91781bb0b3386669d32
 dependencies = [
  "cc",
  "libc",
- "libssh2-sys",
  "libz-sys",
  "openssl-sys",
  "pkg-config",
@@ -3158,18 +3157,6 @@ dependencies = [
  "pkg-config",
  "vcpkg",
 ]
-
-[[package]]
-name = "libssh2-sys"
-version = "0.3.1"
-dependencies = [
- "cc",
- "libc",
- "libz-sys",
- "openssl-sys",
- "pkg-config",
- "vcpkg",
-]
 
 [[package]]
 name = "libz-sys"
--- rustc-1.96.1-src/src/tools/cargo/Cargo.toml.orig	2026-06-29 10:12:33.712458043 -0500
+++ rustc-1.96.1-src/src/tools/cargo/Cargo.toml	2026-06-30 11:25:16.694756549 -0500
@@ -51,7 +51,7 @@ filetime = "0.2.27"
 flate2 = { version = "1.1.9", default-features = false, features = ["zlib-rs"] }
 futures = { version = "0.3.32", default-features = false, features = ["std", "executor"]}
 futures-timer = "3.0.3"
-git2 = "0.20.4"
+git2 = { version = "0.20.4", default-features = false, features = ["https"] }
 git2-curl = "0.21.0"
 # When updating this, also see if `gix-transport` further down needs updating or some auth-related tests will fail.
 gix = { version = "0.81.0", default-features = false, features = ["sha1", "progress-tree", "parallel", "dirwalk", "status"] }
@@ -300,6 +300,3 @@ http-transport-reqwest = ["gix/blocking-
 
 [lints]
 workspace = true
-
-[patch.crates-io]
-libssh2-sys = { path = "third-party/libssh2-sys" }
