diff -up Judy-1.0.4/test/Checkit.test-shared Judy-1.0.4/test/Checkit
--- Judy-1.0.4/test/Checkit.test-shared	2007-05-14 13:02:16.000000000 -0400
+++ Judy-1.0.4/test/Checkit	2008-11-27 12:56:28.000000000 -0500
@@ -4,8 +4,9 @@ CC=${CC:-cc}
 echo
 echo "=== +++++++++ This test runs in 15 seconds on a PIII 750Mhz +++++++"
 
-LIBJUDY=`find ../ -name libJudy.a`
+LIBJUDY=`find ../ -name libJudy.so`
 JUDY_H=`find ../ -name Judy.h`
+export LD_LIBRARY_PATH=`dirname $LIBJUDY`
 
 echo "$CC -O SLcompare.c -DHASHMETHOD -o SL_Hash"
       $CC -O SLcompare.c -DHASHMETHOD -o SL_Hash
