diff -uNr flannel-0.12.0-original/Makefile flannel-0.12.0/Makefile
--- flannel-0.12.0-original/Makefile	2020-03-12 16:59:32.000000000 -0700
+++ flannel-0.12.0/Makefile	2021-01-22 10:15:47.929598160 -0800
@@ -36,9 +36,10 @@
 	rm -f dist/*.tar.gz
 	rm -f dist/qemu-*
 
+# do not build using glibc-static (remove -extldflags "-static" from go build command)
 dist/flanneld: $(shell find . -type f  -name '*.go')
 	go build -o dist/flanneld \
-	  -ldflags '-s -w -X github.com/coreos/flannel/version.Version=$(TAG) -extldflags "-static"'
+	  -ldflags '-s -w -X github.com/coreos/flannel/version.Version=$(TAG)'
 
 dist/flanneld.exe: $(shell find . -type f  -name '*.go')
 	CXX=x86_64-w64-mingw32-g++ CC=x86_64-w64-mingw32-gcc CGO_ENABLED=1 GOOS=windows go build -o dist/flanneld.exe \
