--- commons-daemon-1.2.2-src.orig/src/native/unix/configure.in
+++ commons-daemon-1.2.2-src/src/native/unix/configure.in
@@ -98,7 +98,7 @@ then
   AC_MSG_RESULT([jni_md.h found in $JAVA_HOME/$JAVA_INC])
   INCLUDES="$INCLUDES -I$JAVA_HOME/include -I$JAVA_HOME/$JAVA_INC"
 else
-  INCLUDES="$INCLUDES -I$JAVA_HOME/include -I$JAVA_HOME/include/$supported_os"
+  INCLUDES="$INCLUDES -I$JAVA_HOME/include -I$JAVA_HOME/include/$JAVA_OS"
 fi
 
 dnl -------------------------------------------------------------------------
