Index: commons-daemon-1.2.3-src/src/native/unix/support/apsupport.m4
===================================================================
--- commons-daemon-1.2.3-src.orig/src/native/unix/support/apsupport.m4
+++ commons-daemon-1.2.3-src/src/native/unix/support/apsupport.m4
@@ -181,6 +181,10 @@ AC_DEFUN(AP_SUPPORTED_HOST,[
     CFLAGS="$CFLAGS -DCPU=\\\"aarch64\\\""
     HOST_CPU=aarch64
     ;;
+  riscv64)
+    CFLAGS="$CFLAGS -DCPU=\\\"riscv64\\\""
+    HOST_CPU=$host_cpu
+    ;;
   *)
     AC_MSG_RESULT([failed])
     AC_MSG_ERROR([Unsupported CPU architecture "$host_cpu"]);;
