diff --git a/src/grammar/lexer.flex b/src/grammar/lexer.flex
index f81cd88..a607fb8 100644
--- a/src/grammar/lexer.flex
+++ b/src/grammar/lexer.flex
@@ -173,7 +173,7 @@ import java.util.*;
 	}
 
     public JFlexLexer( java.io.InputStream stream, java.io.Writer writer ) {
-       this( stream );
+       this( new java.io.InputStreamReader( stream, java.nio.charset.Charset.forName( "UTF-8" ) ) );
        this.writer = writer;
   	}
 
diff --git a/src/main/java/com/thoughtworks/qdox/library/ClassLoaderLibrary.java b/src/main/java/com/thoughtworks/qdox/library/ClassLoaderLibrary.java
index c947f9b..5b415c0 100644
--- a/src/main/java/com/thoughtworks/qdox/library/ClassLoaderLibrary.java
+++ b/src/main/java/com/thoughtworks/qdox/library/ClassLoaderLibrary.java
@@ -112,7 +112,7 @@ public class ClassLoaderLibrary
             if ( sourceStream != null )
             {
                 Builder builder = getModelBuilder();
-                JavaLexer lexer = new JFlexLexer( sourceStream );
+                JavaLexer lexer = new JFlexLexer( new java.io.InputStreamReader( sourceStream, java.nio.charset.Charset.forName( "UTF-8" ) ) );
                 Parser parser = new Parser( lexer, builder );
                 parser.setDebugLexer( debugLexer );
                 parser.setDebugParser( debugParser );
