--- a/azurelinuxagent/common/osutil/factory.py	2020-09-29 11:31:13.000000000 -0700
+++ b/azurelinuxagent/common/osutil/factory.py	2020-12-08 14:54:58.878191794 -0800
@@ -24,6 +24,7 @@
 from .arch import ArchUtil
 from .bigip import BigIpOSUtil
 from .clearlinux import ClearLinuxUtil
+from .mariner import MarinerOSUtil
 from .coreos import CoreOSUtil
 from .debian import DebianOSBaseUtil, DebianOSModernUtil
 from .default import DefaultOSUtil
@@ -57,6 +58,9 @@
 
     if "Clear Linux" in distro_full_name:
         return ClearLinuxUtil()
+    
+    if distro_name == "mariner":
+        return MarinerOSUtil()
 
     if distro_name == "ubuntu":
         if Version(distro_version) in [Version("12.04"), Version("12.10")]:
--- a/azurelinuxagent/common/osutil/mariner.py	2020-12-08 22:14:27.843117432 -0800
+++ b/azurelinuxagent/common/osutil/mariner.py	2020-12-08 22:52:05.926318033 -0800
@@ -0,0 +1,89 @@
+#
+# Copyright 2017 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.4 and Openssl 1.0
+#
+
+import os
+import re
+import pwd
+import shutil
+import socket
+import array
+import struct
+import fcntl
+import time
+import base64
+import azurelinuxagent.common.conf as conf
+import azurelinuxagent.common.logger as logger
+import azurelinuxagent.common.utils.fileutil as fileutil
+import azurelinuxagent.common.utils.shellutil as shellutil
+import azurelinuxagent.common.utils.textutil as textutil
+from azurelinuxagent.common.osutil.default import DefaultOSUtil
+
+class MarinerOSUtil(DefaultOSUtil):
+    def __init__(self):
+        super(MarinerOSUtil, self).__init__()
+        self.agent_conf_file_path = '/etc/waagent.conf'
+        self.jit_enabled = True
+
+    def is_dhcp_enabled(self):
+        return True
+
+    def start_network(self) :
+        return shellutil.run("systemctl start systemd-networkd", chk_err=False)
+
+    def restart_if(self, iface):
+        shellutil.run("systemctl restart systemd-networkd")
+
+    def restart_ssh_service(self):
+        shellutil.run("systemctl restart sshd")
+
+    def stop_dhcp_service(self):
+        return shellutil.run("systemctl stop systemd-networkd", chk_err=False)
+
+    def start_dhcp_service(self):
+        return shellutil.run("systemctl start systemd-networkd", chk_err=False)
+
+    def start_agent_service(self):
+        return shellutil.run("systemctl start waagent", chk_err=False)
+
+    def stop_agent_service(self):
+        return shellutil.run("systemctl stop waagent", chk_err=False)
+
+    def get_dhcp_pid(self):
+        ret= shellutil.run_get_output("pidof systemd-networkd")
+        return ret[1] if ret[0] == 0 else None
+
+    def conf_sshd(self, disable_password):
+        pass
+
+    def del_root_password(self):
+        try:
+            passwd_file_path = conf.get_passwd_file_path()
+            try:
+                passwd_content = fileutil.read_file(passwd_file_path)
+                if not passwd_content:
+                    raise FileNotFoundError
+            except FileNotFoundError:
+                new_passwd = ["root:*LOCK*:14600::::::"]
+            else:
+                passwd = passwd_content.split('\n')
+                new_passwd = [x for x in passwd if not x.startswith("root:")]
+                new_passwd.insert(0, "root:*LOCK*:14600::::::")
+            fileutil.write_file(passwd_file_path, "\n".join(new_passwd))
+        except IOError as e:
+            raise OSUtilError("Failed to delete root password:{0}".format(e))
+        pass
\ No newline at end of file
--- a/azurelinuxagent/common/version.py	2020-09-29 11:31:13.000000000 -0700
+++ b/azurelinuxagent/common/version.py	2020-12-08 14:58:51.840248635 -0800
@@ -100,6 +100,9 @@
 
     if os.path.exists("/etc/euleros-release"):
         osinfo[0] = "euleros"
+    
+    if os.path.exists("/etc/mariner-release"):
+        osinfo[0] = "mariner"
 
     # The platform.py lib has issue with detecting BIG-IP linux distribution.
     # Merge the following patch provided by F5.
--- a/config/mariner/waagent.conf	2020-12-08 22:15:55.038466254 -0800
+++ b/config/mariner/waagent.conf	2020-12-08 15:06:48.272503589 -0800
@@ -0,0 +1,80 @@
+# Microsoft Azure Linux Agent Configuration
+#
+
+# Specified program is invoked with the argument "Ready" when we report ready status
+# to the endpoint server.
+Role.StateConsumer=None
+
+# Specified program is invoked with XML file argument specifying role
+#  configuration.
+Role.ConfigurationConsumer=None
+
+# Specified program is invoked with XML file argument specifying role topology.
+Role.TopologyConsumer=None
+
+# Enable instance creation
+Provisioning.Enabled=n
+
+# Rely on cloud-init to provision
+Provisioning.UseCloudInit=y
+
+# Password authentication for root account will be unavailable.
+Provisioning.DeleteRootPassword=y
+
+# Generate fresh host key pair.
+Provisioning.RegenerateSshHostKeyPair=y
+
+# Supported values are "rsa", "dsa" and "ecdsa".
+Provisioning.SshHostKeyPairType=rsa
+
+# Monitor host name changes and publish changes via DHCP requests.
+Provisioning.MonitorHostName=y
+
+# Decode CustomData from Base64.
+Provisioning.DecodeCustomData=y
+
+# Execute CustomData after provisioning.
+Provisioning.ExecuteCustomData=n
+
+# Allow reset password of sys user
+Provisioning.AllowResetSysUser=n
+
+# Format if unformatted. If 'n', resource disk will not be mounted.
+ResourceDisk.Format=n
+
+# File system on the resource disk
+# Typically ext3 or ext4. FreeBSD images should use 'ufs2' here.
+ResourceDisk.Filesystem=ext4
+
+# Mount point for the resource disk
+ResourceDisk.MountPoint=/mnt/resource
+
+# Create and use swapfile on resource disk.
+ResourceDisk.EnableSwap=n
+
+# Size of the swapfile.
+ResourceDisk.SwapSizeMB=0
+
+# Enable verbose logging (y|n)
+Logs.Verbose=n
+
+# Is FIPS enabled
+OS.EnableFIPS=n
+
+# Root device timeout in seconds.
+OS.RootDeviceScsiTimeout=300
+
+# If "None", the system default version is used.
+OS.OpensslPath=None
+
+# Set the path to SSH keys and configuration files
+OS.SshDir=/etc/ssh
+
+# Enable or disable self-update, default is enabled
+AutoUpdate.Enabled=y
+AutoUpdate.GAFamily=Prod
+
+# Determine if the overprovisioning feature is enabled. If yes, hold extension
+# handling until inVMArtifactsProfile.OnHold is false.
+# Default is disabled
+# EnableOverProvisioning=n
--- a/init/mariner/waagent.service	2020-12-08 22:16:40.614123434 -0800
+++ b/init/mariner/waagent.service	2020-12-08 15:06:44.940529116 -0800
@@ -0,0 +1,16 @@
+[Unit]
+Description=Azure Linux Agent
+Wants=systemd-networkd-wait-online.service sshd.service sshd-keygen.service
+After=systemd-networkd-wait-online.service cloud-init.service
+
+ConditionFileIsExecutable=/usr/sbin/waagent
+ConditionPathExists=/etc/waagent.conf
+
+[Service]
+Type=simple
+ExecStart=/usr/bin/python -u /usr/sbin/waagent -daemon
+Restart=always
+RestartSec=5
+
+[Install]
+WantedBy=multi-user.target
--- a/setup.py	2020-09-29 11:31:13.000000000 -0700
+++ b/setup.py	2020-12-08 15:04:44.289458366 -0800
@@ -116,6 +116,12 @@
                        src=["config/clearlinux/waagent.conf"])
         set_systemd_files(data_files, dest='/usr/lib/systemd/system',
                           src=["init/clearlinux/waagent.service"])
+    elif name == 'mariner':
+        set_bin_files(data_files)
+        set_conf_files(data_files, dest="/etc",
+                       src=["config/mariner/waagent.conf"])
+        set_systemd_files(data_files, dest='/usr/lib/systemd/system',
+                          src=["init/mariner/waagent.service"])
     elif name == 'ubuntu':
         set_bin_files(data_files)
         set_conf_files(data_files, src=["config/ubuntu/waagent.conf"])
