--- autoconf-2.13/autoconf.texi.version	2007-02-15 11:33:42.000000000 +0100
+++ autoconf-2.13/autoconf.texi	2007-02-15 11:33:42.000000000 +0100
@@ -1,7 +1,7 @@
 \input texinfo @c -*-texinfo-*-
 @c %**start of header
-@setfilename autoconf.info
-@settitle Autoconf
+@setfilename autoconf213.info
+@settitle Autoconf-2.13
 @c For double-sided printing, uncomment:
 @c @setchapternewpage odd
 @c %**end of header
@@ -17,7 +17,8 @@
 @ifinfo
 @format
 START-INFO-DIR-ENTRY
-* Autoconf: (autoconf).         Create source code configuration scripts.
+* Autoconf213: (autoconf213).   Create source code configuration scripts.
+                                This is a legacy version of autoconf.
 END-INFO-DIR-ENTRY
 @end format
 
--- autoconf-2.13/Makefile.in.version	2007-02-15 11:33:42.000000000 +0100
+++ autoconf-2.13/Makefile.in	2007-02-15 11:37:18.000000000 +0100
@@ -68,8 +68,8 @@
 DISTFILES = AUTHORS COPYING ChangeLog ChangeLog.1 INSTALL \
 	Makefile.in NEWS README TODO $(M4FILES) \
 	acconfig.h acfunctions acheaders acidentifiers \
-	acmakevars acprograms autoconf.info* \
-	autoconf.sh autoconf.texi install.texi \
+	acmakevars acprograms autoconf213.info* \
+	autoconf.sh autoconf213.texi install.texi \
 	autoheader.sh autoscan.pl autoreconf.sh autoupdate.sh ifnames.sh \
 	config.guess config.sub configure configure.in \
 	install-sh mkinstalldirs texinfo.tex \
@@ -106,11 +106,11 @@
 autoconf.m4f: autoconf.m4 acgeneral.m4 acspecific.m4 acoldnames.m4
 autoheader.m4f: autoheader.m4 acgeneral.m4 acspecific.m4 acoldnames.m4
 
-info: autoconf.info @standards_info@ INSTALL
+info: autoconf213.info @standards_info@ INSTALL
 
 # Use --no-split to avoid creating filenames > 14 chars.
-autoconf.info: autoconf.texi install.texi
-	$(MAKEINFO) -I$(srcdir) $(srcdir)/autoconf.texi --no-split --output=$@
+autoconf213.info: autoconf213.texi install.texi
+	$(MAKEINFO) -I$(srcdir) $(srcdir)/autoconf213.texi --no-split --output=$@
 
 INSTALL: install.texi
 	$(MAKEINFO) -I$(srcdir) $(srcdir)/install.texi --output=$@ \
@@ -121,8 +121,8 @@
 
 dvi: autoconf.dvi @standards_dvi@
 
-autoconf.dvi: autoconf.texi
-	$(TEXI2DVI) $(srcdir)/autoconf.texi
+autoconf.dvi: autoconf213.texi
+	$(TEXI2DVI) $(srcdir)/autoconf213.texi
 
 standards.dvi: standards.texi make-stds.texi
 	$(TEXI2DVI) $(srcdir)/standards.texi
@@ -159,7 +159,7 @@
 
 # Don't cd, to avoid breaking install-sh references.
 install-info: info installdirs
-	if test -f autoconf.info; then \
+	if test -f autoconf213.info; then \
 	  for i in *.info*; do \
 	    $(INSTALL_DATA) $$i $(DESTDIR)/$(infodir)/$$i; \
 	  done; \
@@ -174,7 +174,7 @@
 	  rm -f $(bindir)/`echo $$p|sed '$(transform)'`; \
 	done
 	rm -fr $(acdatadir)
-	cd $(infodir) && rm -f autoconf.info*
+	cd $(infodir) && rm -f autoconf213.info*
 	if test -f standards.info || test -f $(srcdir)/standards.info; \
 	then cd $(infodir) && rm -f standards.info*; fi
 
