diff -U 3 -dHrN a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml	1970-01-01 01:00:00.000000000 +0100
+++ b/Cargo.toml	2025-08-03 10:56:40.895096187 +0200
@@ -13,6 +13,4 @@
 
 [dependencies.pyo3]
 version = "0.25.1"
-# To build extension for PyPy on Windows, "generate-import-lib" is needed:
-# https://github.com/PyO3/maturin-action/issues/267#issuecomment-2106844429
-features = ["extension-module", "generate-import-lib"]
+features = ["extension-module"]
