diff --git a/psm_context.c b/psm_context.c
index 235bd1b..2728fab 100644
--- a/psm_context.c
+++ b/psm_context.c
@@ -381,7 +381,7 @@ psmi_context_check_status(const psmi_context_t *contexti)
 		else
 		    errmsg = "Hardware not found";
 
-		psmi_handle_error(context->ep, err, errmsg);
+		psmi_handle_error(context->ep, err, errmsg, "%s");
 	    }
 	}
     }
diff --git a/psm_ep_connect.c b/psm_ep_connect.c
index 4e5b0a9..98294e2 100644
--- a/psm_ep_connect.c
+++ b/psm_ep_connect.c
@@ -275,7 +275,7 @@ __psm_ep_connect(psm_ep_t ep, int num_of_epid,
 	    }
 	}
 	errbuf[sizeof errbuf - 1] = '\0';
-	err = psmi_handle_error(ep, err, errbuf);
+	err = psmi_handle_error(ep, err, errbuf, "%s");
     }
 
 fail:
diff --git a/psm_utils.c b/psm_utils.c
index 895e7b5..c8651fe 100644
--- a/psm_utils.c
+++ b/psm_utils.c
@@ -1202,7 +1202,7 @@ psmi_coreopt_ctl(const void *core_obj, int optname,
   
  fail:
   /* Unrecognized/unknown option */
-  return psmi_handle_error(NULL, PSM_PARAM_ERR, err_string);
+  return psmi_handle_error(NULL, PSM_PARAM_ERR, err_string, "%s");
 }
 
 psm_error_t psmi_core_setopt(const void *core_obj, int optname, 
diff --git a/psm_diags.c b/psm_diags.c
index 90f2e17..5bfdb17 100644
--- a/psm_diags.c
+++ b/psm_diags.c
@@ -280,10 +280,11 @@ memcpy_check_size (memcpy_fn_t fn, int *p, int *f, size_t n)
   if (USE_MALLOC) {
     src = psmi_malloc(PSMI_EP_NONE, UNDEFINED, size);
     dst = psmi_malloc(PSMI_EP_NONE, UNDEFINED, size);
-    if (src == NULL || dst == NULL) 
+    if (src == NULL || dst == NULL) {
       if (src) psmi_free(src);
       if (dst) psmi_free(dst);
       return -1;
+    }
   }
   else {
     void *src_p, *dst_p;
 