diff --git a/buildflags.mak b/buildflags.mak
index e24dcff..82c567e 100644
--- a/buildflags.mak
+++ b/buildflags.mak
@@ -94,7 +94,8 @@ ifeq (1,${PSM_USE_SYS_UUID})
 endif
 
 CFLAGS += $(BASECFLAGS) $(if $(filter $(CC),gcc),-Wno-strict-aliasing) \
-	$(if $(PSM_VALGRIND:0=),-DPSM_VALGRIND,-DNVALGRIND)
+	$(if $(PSM_VALGRIND:0=),-DPSM_VALGRIND,-DNVALGRIND) \
+	-Wno-maybe-uninitialized -Wno-implicit-function-declaration
 
 .DEFAULT_GOAL := all
 # Dummy install rule for sub-directories that don't install anything
