diff --git a/Makefile b/Makefile
index d79c4bd..24fa29e 100644
--- a/Makefile
+++ b/Makefile
@@ -188,8 +188,8 @@ install: all
 		ln -sf ${TARGLIB}.so.${MAJOR}.${MINOR} ${TARGLIB}.so.${MAJOR} ; \
 		ln -sf ${TARGLIB}.so.${MAJOR} ${TARGLIB}.so) ; \
 	if [ X$(MIC) != X1 ]; then \
-		install -D psm.h ${DESTDIR}/usr/include/psm.h ; \
-		install -D psm_mq.h ${DESTDIR}/usr/include/psm_mq.h ; \
+		install -m 0644 -D psm.h ${DESTDIR}/usr/include/psm.h ; \
+		install -m 0644 -D psm_mq.h ${DESTDIR}/usr/include/psm_mq.h ; \
 	else \
 		filelist=/opt/intel/mic/psm/psm.filelist ; \
 		sed -e 's!%IPATHMAJOR%!$(IPATH_LIB_MAJOR)!g' \
