--- ceph/src/CMakeLists.txt.orig	2023-11-01 11:53:53.618167190 -0400
+++ ceph/src/CMakeLists.txt	2023-11-01 13:52:51.292643490 -0400
@@ -625,6 +625,7 @@
 add_subdirectory(perfglue)
 
 add_library(rados_snap_set_diff_obj OBJECT librados/snap_set_diff.cc)
+add_dependencies(rados_snap_set_diff_obj legacy-option-headers)
 
 option(WITH_LIBRADOSSTRIPER "build with libradosstriper support" ON)
 
@@ -881,6 +882,7 @@
     add_library(krbd STATIC krbd.cc
       $<TARGET_OBJECTS:parse_secret_objs>)
     target_link_libraries(krbd keyutils::keyutils)
+    add_dependencies(krbd legacy-option-headers)
   endif()
   add_subdirectory(librbd)
   if(WITH_FUSE)
