--- ceph-19.2.0/src/mds/CMakeLists.txt.orig	2024-12-20 17:15:47.632569537 -0500
+++ ceph-19.2.0/src/mds/CMakeLists.txt	2024-12-20 17:16:48.945481957 -0500
@@ -45,12 +45,11 @@
   QuiesceDbManager.cc
   QuiesceAgent.cc
   MDSRankQuiesce.cc
-  BoostUrlImpl.cc
   ${CMAKE_SOURCE_DIR}/src/common/TrackedOp.cc
   ${CMAKE_SOURCE_DIR}/src/common/MemoryModel.cc
   ${CMAKE_SOURCE_DIR}/src/osdc/Journaler.cc
   ${CMAKE_SOURCE_DIR}/src/mgr/MDSPerfMetricTypes.cc)
 add_library(mds STATIC ${mds_srcs})
 target_link_libraries(mds PRIVATE
-  heap_profiler cpu_profiler osdc ${LUA_LIBRARIES})
+  boost_url heap_profiler cpu_profiler osdc ${LUA_LIBRARIES})
 target_include_directories(mds PRIVATE "${LUA_INCLUDE_DIR}")
