--- hickory-proto-0.25.2/Cargo.toml	1970-01-01T00:00:01+00:00
+++ hickory-proto-0.25.2/Cargo.toml	2025-09-09T22:16:47.805185+00:00
@@ -164,9 +164,6 @@
     "rand/thread_rng",
     "ring?/std",
     "thiserror/std",
-    "tracing-subscriber/env-filter",
-    "tracing-subscriber/fmt",
-    "tracing-subscriber/std",
     "tracing/std",
     "url/std",
 ]
@@ -411,10 +408,6 @@
     "macros",
 ]
 
-[dev-dependencies.tracing-subscriber]
-version = "0.3"
-default-features = false
-
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
