diff --git a/src/crypto/crypto_context.cc b/src/crypto/crypto_context.cc
index 05fc2a00..c62d6cf5 100644
--- a/src/crypto/crypto_context.cc
+++ b/src/crypto/crypto_context.cc
@@ -1583,28 +1583,16 @@ void SecureContext::Init(const FunctionCallbackInfo<Value>& args) {
       min_version = 0;
       max_version = kMaxSupportedVersion;
       method = TLS_client_method();
-    } else if (sslmethod == "TLSv1_method") {
-      min_version = TLS1_VERSION;
-      max_version = TLS1_VERSION;
-    } else if (sslmethod == "TLSv1_server_method") {
-      min_version = TLS1_VERSION;
-      max_version = TLS1_VERSION;
-      method = TLS_server_method();
-    } else if (sslmethod == "TLSv1_client_method") {
-      min_version = TLS1_VERSION;
-      max_version = TLS1_VERSION;
-      method = TLS_client_method();
-    } else if (sslmethod == "TLSv1_1_method") {
-      min_version = TLS1_1_VERSION;
-      max_version = TLS1_1_VERSION;
-    } else if (sslmethod == "TLSv1_1_server_method") {
-      min_version = TLS1_1_VERSION;
-      max_version = TLS1_1_VERSION;
-      method = TLS_server_method();
-    } else if (sslmethod == "TLSv1_1_client_method") {
-      min_version = TLS1_1_VERSION;
-      max_version = TLS1_1_VERSION;
-      method = TLS_client_method();
+    } else if (sslmethod == "TLSv1_method" ||
+               sslmethod == "TLSv1_server_method" ||
+               sslmethod == "TLSv1_client_method") {
+      THROW_ERR_TLS_INVALID_PROTOCOL_METHOD(env, "TLSv1 methods disabled");
+      return;
+    } else if (sslmethod == "TLSv1_1_method" ||
+               sslmethod == "TLSv1_1_server_method" ||
+               sslmethod == "TLSv1_1_client_method") {
+      THROW_ERR_TLS_INVALID_PROTOCOL_METHOD(env, "TLSv1_1 methods disabled");
+      return;
     } else if (sslmethod == "TLSv1_2_method") {
       min_version = TLS1_2_VERSION;
       max_version = TLS1_2_VERSION;
