diff -ruN a/src/node_crypto.cc b/src/node_crypto.cc
--- a/src/node_crypto.cc	2020-05-04 14:32:49.481205912 -0700
+++ b/src/node_crypto.cc	2020-05-04 14:40:40.456686371 -0700
@@ -536,6 +536,8 @@
       method = SSLv23_server_method();
     } else if (strcmp(*sslmethod, "SSLv23_client_method") == 0) {
       method = SSLv23_client_method();
+// TLS1 and TSL1.1 are disabled in openssl
+#if 0
     } else if (strcmp(*sslmethod, "TLSv1_method") == 0) {
       method = TLSv1_method();
     } else if (strcmp(*sslmethod, "TLSv1_server_method") == 0) {
@@ -548,6 +550,7 @@
       method = TLSv1_1_server_method();
     } else if (strcmp(*sslmethod, "TLSv1_1_client_method") == 0) {
       method = TLSv1_1_client_method();
+#endif
     } else if (strcmp(*sslmethod, "TLSv1_2_method") == 0) {
       method = TLSv1_2_method();
     } else if (strcmp(*sslmethod, "TLSv1_2_server_method") == 0) {
