diff -ruN a/src/node_crypto.cc b/src/node_crypto.cc
--- a/src/node_crypto.cc	2020-05-04 15:00:25.891257518 -0700
+++ b/src/node_crypto.cc	2020-05-04 15:29:03.075011348 -0700
@@ -576,6 +576,8 @@
       method = SSLv23_server_method();
     } else if (strcmp(*sslmethod, "SSLv23_client_method") == 0) {
       method = SSLv23_client_method();
+// TLS1 and TSL1.1 are disabled in openssl
+#if 0
     } else if (strcmp(*sslmethod, "TLSv1_method") == 0) {
       method = TLSv1_method();
     } else if (strcmp(*sslmethod, "TLSv1_server_method") == 0) {
@@ -588,6 +590,7 @@
       method = TLSv1_1_server_method();
     } else if (strcmp(*sslmethod, "TLSv1_1_client_method") == 0) {
       method = TLSv1_1_client_method();
+#endif
     } else if (strcmp(*sslmethod, "TLSv1_2_method") == 0) {
       method = TLSv1_2_method();
     } else if (strcmp(*sslmethod, "TLSv1_2_server_method") == 0) {
