--- a/xalan-j2-MANIFEST.MF	2020-11-30 10:15:53.477571534 -0800
+++ b/xalan-j2-MANIFEST.MF	2020-11-30 10:19:41.504103264 -0800
@@ -44,7 +44,8 @@
  org.apache.xpath.operations;version="2.7.2",
  org.apache.xpath.patterns;version="2.7.2",
  org.apache.xpath.res;version="2.7.2"
-Require-Bundle: system.bundle, org.apache.xerces
+Require-Bundle: system.Bundle
+Require-Bundle: org.apache.xerces;resolution:=optional
 Eclipse-BuddyPolicy: registered
 
 Name: org/apache/xalan/
