diff -up smartmontools-5.38/smartd.conf.defaultconf smartmontools-5.38/smartd.conf
--- smartmontools-5.38/smartd.conf.defaultconf	2008-08-07 12:21:53.000000000 +0200
+++ smartmontools-5.38/smartd.conf	2008-08-07 12:24:26.000000000 +0200
@@ -20,7 +20,7 @@
 # Directives listed below, which will be applied to all devices that
 # are found.  Most users should comment out DEVICESCAN and explicitly
 # list the devices that they wish to monitor.
-DEVICESCAN
+DEVICESCAN -H -m root -M exec /usr/libexec/smartmontools/smartdnotify -n standby,10,q
 
 # Alternative setting to ignore temperature and power-on hours reports
 # in syslog.
