diff -up wrk/doc/en/cd-paranoia.1.in.wrk wrk/doc/en/cd-paranoia.1.in
--- wrk/doc/en/cd-paranoia.1.in.wrk	2013-07-31 18:43:13.071575017 +0200
+++ wrk/doc/en/cd-paranoia.1.in	2013-07-31 18:46:22.446042574 +0200
@@ -1,6 +1,6 @@
 .TH @CDPARANOIA_NAME@ 1 "version III release alpha 9.8 libcdio"
 .SH NAME
-@CDPARANOIA_NAME@ 9.8 (Paranoia release III via libcdio) \- an audio CD reading utility which includes extra data verification features
+@CDPARANOIA_NAME@ \- 9.8 (Paranoia release III via libcdio) - an audio CD reading utility which includes extra data verification features
 .SH SYNOPSIS
 .B @CDPARANOIA_NAME@
 .RB [ options ]
diff -up wrk/doc/en/cd-paranoia.1.wrk wrk/doc/en/cd-paranoia.1
--- wrk/doc/en/cd-paranoia.1.wrk	2013-07-31 18:42:52.799417871 +0200
+++ wrk/doc/en/cd-paranoia.1	2013-07-31 18:45:13.430507735 +0200
@@ -1,6 +1,6 @@
 .TH cd-paranoia 1 "version III release alpha 9.8 libcdio"
 .SH NAME
-cd-paranoia 9.8 (Paranoia release III via libcdio) \- an audio CD reading utility which includes extra data verification features
+cd-paranoia \- 9.8 (Paranoia release III via libcdio) - an audio CD reading utility which includes extra data verification features
 .SH SYNOPSIS
 .B cd-paranoia
 .RB [ options ]
