--- openssl-0.10.81/Cargo.toml	2006-07-24T01:21:28+00:00
+++ openssl-0.10.81/Cargo.toml	2026-06-18T14:26:20.575294+00:00
@@ -37,16 +37,12 @@
 repository = "https://github.com/rust-openssl/rust-openssl"
 
 [features]
-aws-lc = ["ffi/aws-lc"]
-aws-lc-fips = ["ffi/aws-lc-fips"]
 bindgen = ["ffi/bindgen"]
 default = []
-unstable_boringssl = ["ffi/unstable_boringssl"]
 v101 = []
 v102 = []
 v110 = []
 v111 = []
-vendored = ["ffi/vendored"]
 
 [lib]
 name = "openssl"
