From d45c06270576493e1537080fa43c1af667c4337a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nikola=20Forr=C3=B3?= <nforro@redhat.com>
Date: Tue, 12 Oct 2021 14:15:17 +0200
Subject: [PATCH] Avoid multiple definitions of typeinfos

---
 XMPFiles/source/XMPFiles_Impl.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/XMPFiles/source/XMPFiles_Impl.cpp b/XMPFiles/source/XMPFiles_Impl.cpp
index 2f918ac..ff45040 100644
--- a/XMPFiles/source/XMPFiles_Impl.cpp
+++ b/XMPFiles/source/XMPFiles_Impl.cpp
@@ -47,7 +47,9 @@ using namespace std;
 /// This file ...
 ///
 // =================================================================================================
-#include "public/include/XMP.incl_cpp"
+#if ! XMP_StaticBuild
+	#include "public/include/XMP.incl_cpp"
+#endif
 
 #if XMP_WinBuild
 	#pragma warning ( disable : 4290 )	// C++ exception specification ignored except to indicate a function is not __declspec(nothrow)
-- 
2.33.0

