diff --git a/.github/workflows/test.yml b/.github/workflows/test.yml
index 9820304e..321e8749 100644
--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -450,6 +450,12 @@ jobs:
         TESTOPTS: "-x"
       run: |
         make testc
+    - name: Run test of installed version
+      env:
+        PACKAGE_LEVEL: ${{ matrix.package_level }}
+        RUN_TYPE: ${{ steps.set-run-type.outputs.result }}
+      run: |
+        make testinstalled
     - name: Run safety
       env:
         PACKAGE_LEVEL: ${{ matrix.package_level }}
diff --git a/Makefile b/Makefile
index 09afaf72..80a2c8ca 100644
--- a/Makefile
+++ b/Makefile
@@ -285,7 +285,11 @@ else
 endif
 pytest_end2end_opts := -v --tb=short $(pytest_opts) --es-file $(test_dir)/end2endtest/es_server.yml --es-schema-file $(test_dir)/end2endtest/es_schema.yml
 
-pytest_cov_opts := --cov $(package_name) --cov $(mock_package_name) $(coverage_report) --cov-config .coveragerc
+ifdef TEST_INSTALLED
+  pytest_cov_opts :=
+else
+  pytest_cov_opts := --cov $(package_name) --cov $(mock_package_name) $(coverage_report) --cov-config .coveragerc
+endif
 
 ifeq ($(python_m_version),3)
   pytest_warning_opts := -W default
@@ -385,6 +389,7 @@ help:
 	@echo "  installtest - Run install tests"
 	@echo "  safety     - Run Safety for install and all"
 	@echo "  test       - Run unit and function tests (in tests/unittest and tests/functiontest)"
+	@echo "  testinstalled - Simulate the testing of an installed version of pywbem"
 	@echo "  leaktest   - Run memory leak tests (in tests/leaktest)"
 	@echo "  resourcetest - Run resource consumption tests (in tests/resourcetest)"
 	@echo "  perftest   - Run performance tests (in tests/perftest)"
@@ -560,12 +565,19 @@ endif
 develop: $(done_dir)/develop_$(pymn)_$(PACKAGE_LEVEL).done
 	@echo "Makefile: Target $@ done."
 
-$(done_dir)/develop_$(pymn)_$(PACKAGE_LEVEL).done: $(done_dir)/pip_upgrade_$(pymn)_$(PACKAGE_LEVEL).done $(done_dir)/install_basic_$(pymn)_$(PACKAGE_LEVEL).done $(done_dir)/install_$(pymn)_$(PACKAGE_LEVEL).done $(done_dir)/develop_os_$(pymn)_$(PACKAGE_LEVEL).done $(done_dir)/install_basic_$(pymn)_$(PACKAGE_LEVEL).done dev-requirements.txt test-requirements.txt
-	@echo "Makefile: Installing Python development requirements (with PACKAGE_LEVEL=$(PACKAGE_LEVEL))"
+$(done_dir)/test_$(pymn)_$(PACKAGE_LEVEL).done: $(done_dir)/pip_upgrade_$(pymn)_$(PACKAGE_LEVEL).done $(done_dir)/install_basic_$(pymn)_$(PACKAGE_LEVEL).done test-requirements.txt
+	@echo "Makefile: Installing Python test requirements (with PACKAGE_LEVEL=$(PACKAGE_LEVEL))"
+	-$(call RM_FUNC,$@)
+	$(PIP_INSTALL_CMD) $(pip_level_opts) -r test-requirements.txt
+	echo "done" >$@
+	@echo "Makefile: Done installing Python test requirements"
+
+$(done_dir)/develop_$(pymn)_$(PACKAGE_LEVEL).done: $(done_dir)/pip_upgrade_$(pymn)_$(PACKAGE_LEVEL).done $(done_dir)/install_basic_$(pymn)_$(PACKAGE_LEVEL).done $(done_dir)/install_$(pymn)_$(PACKAGE_LEVEL).done $(done_dir)/develop_os_$(pymn)_$(PACKAGE_LEVEL).done dev-requirements.txt test-requirements.txt
+	@echo "Makefile: Installing Python development/test requirements (with PACKAGE_LEVEL=$(PACKAGE_LEVEL))"
 	-$(call RM_FUNC,$@)
 	$(PIP_INSTALL_CMD) $(pip_level_opts) -r dev-requirements.txt
 	echo "done" >$@
-	@echo "Makefile: Done installing Python development requirements"
+	@echo "Makefile: Done installing Python development/test requirements"
 
 .PHONY: vendor
 vendor: $(done_dir)/vendor_$(pymn)_$(PACKAGE_LEVEL).done
@@ -875,9 +887,9 @@ endif
 endif
 
 ifdef TEST_INSTALLED
-  test_deps =
+  test_deps = $(done_dir)/test_$(pymn)_$(PACKAGE_LEVEL).done $(moftab_files)
 else
-  test_deps = $(done_dir)/develop_$(pymn)_$(PACKAGE_LEVEL).done $(moftab_files)
+  test_deps = $(done_dir)/test_$(pymn)_$(PACKAGE_LEVEL).done $(moftab_files)
 endif
 
 $(done_dir)/todo_$(pymn)_$(PACKAGE_LEVEL).done: $(done_dir)/develop_$(pymn)_$(PACKAGE_LEVEL).done Makefile $(pylint_rc_file) $(py_src_files) $(py_test_files)
@@ -915,9 +927,23 @@ endif
 .PHONY: test
 test: $(test_deps)
 	@echo "Makefile: Running unit and function tests"
+	@echo "Makefile: pywbem is imported from: $(python -c 'import pywbem; print(pywbem.__file__)')"
 	py.test --color=yes $(pytest_cov_opts) $(pytest_warning_opts) $(pytest_opts) $(test_dir)/unittest $(test_dir)/functiontest -s
 	@echo "Makefile: Done running unit and function tests"
 
+.PHONY: testinstalled
+testinstalled: $(done_dir)/install_pywbem_$(pymn)_$(PACKAGE_LEVEL).done base-requirements.txt
+	@echo "Makefile: Running short unit test in virtualenv with installed version of pywbem"
+ifeq ($(PLATFORM),Windows_native)
+	@echo "Makefile: Warning: Skipping the test on native Windows" >&2
+else
+	-$(call RM_FUNC,$(moftab_files))
+	-$(call RMDIR_R_FUNC,.virtualenv/testinstalled)
+	virtualenv .virtualenv/testinstalled
+	bash -cx "source .virtualenv/testinstalled/bin/activate; $(PYTHON_CMD) --version; $(PIP_INSTALL_CMD) $(pip_level_opts) -r base-requirements.txt; $(PIP_INSTALL_CMD) $(pip_level_opts) .; $(PIP_CMD) list; TEST_INSTALLED=1 TESTCASES=test_mof_compiler.py TESTOPTS='-x' make test"
+endif
+	@echo "Makefile: Done running short unit test in virtualenv with installed version of pywbem"
+
 .PHONY: testc
 testc: $(test_deps) $(done_dir)/installc_$(pymn)_$(PACKAGE_LEVEL).done
 	@echo "Makefile: Running unit and function tests on cythonized archive"
@@ -945,13 +971,13 @@ leaktest: $(test_deps)
 	@echo "Makefile: Done running memory leak tests"
 
 .PHONY: end2endtest
-end2endtest: $(done_dir)/develop_$(pymn)_$(PACKAGE_LEVEL).done $(moftab_files)
+end2endtest: $(test_deps)
 	@echo "Makefile: Running end2end tests"
 	py.test --color=yes $(pytest_end2end_warning_opts) $(pytest_end2end_opts) $(test_dir)/end2endtest -s
 	@echo "Makefile: Done running end2end tests"
 
 .PHONY: resourcetest
-resourcetest: $(done_dir)/develop_$(pymn)_$(PACKAGE_LEVEL).done $(moftab_files)
+resourcetest: $(test_deps)
 ifeq ($(python_m_version),2)
 	@echo "Makefile: Warning: Skipping resource consumption tests on Python $(python_version)" >&2
 else
@@ -961,7 +987,7 @@ else
 endif
 
 .PHONY: perftest
-perftest: $(done_dir)/develop_$(pymn)_$(PACKAGE_LEVEL).done $(moftab_files)
+perftest: $(test_deps)
 ifeq ($(python_m_version),2)
 	@echo "makefile: Warning: Skipping performance tests on Python $(python_version)" >&2
 else
diff --git a/base-requirements.txt b/base-requirements.txt
index 525bff7b..95aeabad 100644
--- a/base-requirements.txt
+++ b/base-requirements.txt
@@ -29,5 +29,6 @@ setuptools>=70.0.0; python_version >= '3.8'
 
 # wheel 0.36.2 fixes empty and invalid tag issues in archive file name, and supports get_platform() with args
 # wheel 0.38.1 fixes CVE issue 51499, and dropped support for Python <=3.6
+# wheel 0.44.0 (requires py>=3.8) removed bdist_wheel.get_platform(), which is now in setuptools>=71.0.0
 wheel>=0.36.2; python_version <= '3.6'
 wheel>=0.38.1,<0.44.0; python_version >= '3.7'
diff --git a/docs/development.rst b/docs/development.rst
index 91572fbf..c3da7db2 100644
--- a/docs/development.rst
+++ b/docs/development.rst
@@ -387,7 +387,7 @@ installed into the system Python:
 
    .. code-block:: bash
 
-       $ virtualenv --system-site-packages --no-setuptools --no-pip --no-wheel .virtualenv/test
+       $ virtualenv --system-site-packages .virtualenv/test
        $ source .virtualenv/test/bin/activate
 
    The pywbem project is set up so that Git ignores the ``.virtualenv``
diff --git a/minimum-constraints.txt b/minimum-constraints.txt
index 886e8436..0353f8f9 100644
--- a/minimum-constraints.txt
+++ b/minimum-constraints.txt
@@ -68,6 +68,7 @@ lxml==4.6.2; python_version >= '3.6' and python_version <= '3.9'
 lxml==4.6.4; python_version == '3.10'
 lxml==4.9.2; python_version == '3.11'
 lxml==4.9.3; python_version >= '3.12'
+
 # Coverage reporting (no imports, invoked via coveralls script):
 coverage==5.0
 pytest-cov==2.7.0
diff --git a/tests/README b/tests/README
index 998b6154..9e2b7375 100644
--- a/tests/README
+++ b/tests/README
@@ -129,7 +129,7 @@ installed into the system Python:
 2. Create and activate a virtual Python environment of the intended Python
    version, that is based on the system Python:
 
-       $ virtualenv --system-site-packages --no-setuptools --no-pip --no-wheel .virtualenv/test
+       $ virtualenv --system-site-packages .virtualenv/test
        $ source .virtualenv/test/bin/activate
 
    The pywbem project is set up so that Git ignores the ``.virtualenv``
