diff --git a/include/wvautoconf.h.in b/include/wvautoconf.h.in
index 06f5db2..decba99 100644
--- a/include/wvautoconf.h.in
+++ b/include/wvautoconf.h.in
@@ -65,7 +65,7 @@
 #undef HAVE_LIBCRYPTO
 
 /* Whether libc supports __libc_stack_end */
-#undef HAVE_LIBC_STACK_END
+/* #undef HAVE_LIBC_STACK_END */
 
 /* Define to 1 if you have the `pam' library (-lpam). */
 #undef HAVE_LIBPAM
@@ -169,7 +169,7 @@
 #undef HAVE_UNISTD_H
 
 /* Define to 1 if you have the <valgrind/memcheck.h> header file. */
-#undef HAVE_VALGRIND_MEMCHECK_H
+/* #undef HAVE_VALGRIND_MEMCHECK_H */
 
 /* Define to 1 if you have the <zlib.h> header file. */
 #undef HAVE_ZLIB_H
@@ -204,7 +204,7 @@
 #undef VERBOSE_PACKAGE_VERSION
 
 /* Extra version string. */
-#undef VER_STRING_EXTRA
+/*#define VER_STRING_EXTRA " (Red Hat Buildsystem)"*/
 
 /* Define to enable DBUS support. */
 #undef WITH_DBUS
