diff --git a/utils/wvtask.cc b/utils/wvtask.cc
index 9b0444c..628dc45 100644
--- a/utils/wvtask.cc
+++ b/utils/wvtask.cc
@@ -430,7 +430,7 @@ void WvTaskMan::_stackmaster()
 	    total = (val+1) * (size_t)1024;
 	    
             if (!use_shared_stack())
-                total = 1024; // enough to save the do_task stack frame
+                total = 2048; // enough to save the do_task stack frame
 
 	    // set up a stack frame for the new task.  This runs once
 	    // per get_stack.
