diff -up ./src/ortp_srtp.c.gcc10 ./src/ortp_srtp.c
--- ./src/ortp_srtp.c.gcc10	2020-02-03 14:07:41.018756772 -0500
+++ ./src/ortp_srtp.c	2020-02-03 14:07:07.879279726 -0500
@@ -296,6 +296,7 @@ srtp_t ortp_srtp_create_configure_sessio
 		srtp_ssrc_t incoming_ssrc;
 		srtp_policy_t policy;
 		
+		memset(&incoming_ssrc, 0, sizeof(srtp_ssrc_t));
 		memset(&policy, 0, sizeof(srtp_policy_t));
 		incoming_ssrc.type = ssrc_any_inbound;
 		
diff -up ./src/rtcp.c.gcc10 ./src/rtcp.c
--- ./src/rtcp.c.gcc10	2014-01-28 15:38:03.000000000 -0500
+++ ./src/rtcp.c	2020-02-03 14:03:53.586482838 -0500
@@ -358,7 +358,10 @@ static int rtcp_app_init(RtpSession *ses
 	rtcp_common_header_init(&app->ch,session,RTCP_APP,subtype,size);
 	app->ssrc=htonl(session->snd.ssrc);
 	memset(app->name,0,4);
-	strncpy(app->name,name,4);
+	int len = strlen(name);
+	if (len > 4) len = 4;
+	/* NOTE: used in rtcpparse.c with no 0 byte expected */
+	memcpy(app->name,name,len);
 	return sizeof(rtcp_app_t);
 }
 
