--- z-base-32-0.1.5/Cargo.toml	2006-07-24T01:21:28+00:00
+++ z-base-32-0.1.5/Cargo.toml	2026-06-18T14:16:49.507988+00:00
@@ -34,31 +34,10 @@
 repository = "https://github.com/matusf/z-base-32"
 
 [features]
-cli = ["dep:clap"]
-python = ["dep:pyo3"]
 
 [lib]
 name = "zbase32"
-crate-type = [
-    "cdylib",
-    "rlib",
-]
 path = "src/lib.rs"
-
-[[bin]]
-name = "zbase32"
-path = "src/bin/zbase32.rs"
-required-features = ["cli"]
-
-[dependencies.clap]
-version = "4"
-features = ["derive"]
-optional = true
-
-[dependencies.pyo3]
-version = "0.29"
-features = ["extension-module"]
-optional = true
 
 [dev-dependencies.quickcheck]
 version = "1"
