diff -ruN a/elasticsearch-api/elasticsearch-api.gemspec b/elasticsearch-api/elasticsearch-api.gemspec
--- a/elasticsearch-api/elasticsearch-api.gemspec	2022-07-07 09:28:16.348440261 -0700
+++ b/elasticsearch-api/elasticsearch-api.gemspec	2022-07-07 09:28:50.040817530 -0700
@@ -34,7 +34,7 @@
     'source_code_uri' => 'https://github.com/elastic/elasticsearch-ruby/tree/main/elasticsearch-api',
     'bug_tracker_uri' => 'https://github.com/elastic/elasticsearch-ruby/issues'
   }
-  s.files         = `git ls-files`.split($/)
+  s.files         = Dir['**/*']
   s.executables   = s.files.grep(%r{^bin/}) { |f| File.basename(f) }
   s.test_files    = s.files.grep(%r{^(test|spec|features)/})
   s.require_paths = ['lib']
