From dfd64ddb0c6c14ed01665ed2317c86d66d08362c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Thu, 8 Dec 2016 09:41:45 +0100
Subject: [PATCH] Include ecb.h from system
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 XS.xs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/XS.xs b/XS.xs
index 2dc62d2..704a3ad 100644
--- a/XS.xs
+++ b/XS.xs
@@ -11,7 +11,7 @@
 #include <inttypes.h>
 
 #define ECB_NO_THREADS 1
-#include "ecb.h"
+#include <ecb.h>
 
 // compatibility with perl <5.18
 #ifndef HvNAMELEN_get
-- 
2.7.4

