diff -Naur pngnq-1.1.orig/src/rwpng.c pngnq-1.1/src/rwpng.c
--- pngnq-1.1.orig/src/rwpng.c	2010-05-10 18:42:56.000000000 -0400
+++ pngnq-1.1/src/rwpng.c	2012-07-26 18:08:54.947306021 -0400
@@ -32,7 +32,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#include "png.h"        /* libpng header; includes zlib.h */
+#include "png.h"        /* libpng header */
+#include "zlib.h"       /* need a few symbols from this */
 #include "rwpng.h"      /* typedefs, common macros, public prototypes */
 
 /* future versions of libpng will provide this macro: */
