diff -up stunnel-5.55/src/options.c.system-ciphers stunnel-5.55/src/options.c
--- stunnel-5.55/src/options.c.system-ciphers	2019-09-19 14:43:00.631059024 +0200
+++ stunnel-5.55/src/options.c	2019-09-19 14:51:02.120053849 +0200
@@ -277,7 +277,7 @@ static char *option_not_found=
     "Specified option name is not valid here";
 
 static char *stunnel_cipher_list=
-    "HIGH:!aNULL:!SSLv2:!DH:!kDHEPSK";
+   "PROFILE=SYSTEM";
 
 #ifndef OPENSSL_NO_TLS1_3
 static char *stunnel_ciphersuites=
