--- gix-status-0.32.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ gix-status-0.32.0/Cargo.toml	2026-06-26T18:50:02.037551+00:00
@@ -112,9 +112,6 @@
 [dev-dependencies.pretty_assertions]
 version = "1.4.0"
 
-[target.'cfg(not(target_has_atomic = "64"))'.dependencies.portable-atomic]
-version = "1"
-
 [lints.clippy]
 bool_to_int_with_if = "allow"
 borrow_as_ptr = "allow"
@@ -193,3 +190,4 @@
 priority = -1
 
 [lints.rust]
+
