--- heapless-0.7.17/Cargo.toml	1970-01-01T00:00:01+00:00
+++ heapless-0.7.17/Cargo.toml	2024-03-25T22:10:29.067024+00:00
@@ -63,29 +63,13 @@
 
 [features]
 __trybuild = []
-cas = ["atomic-polyfill"]
+cas = []
 default = ["cas"]
 defmt-impl = ["defmt"]
 mpmc_large = []
 ufmt-impl = ["ufmt-write"]
 x86-sync-pool = []
 
-[target."cfg(target_arch = \"avr\")".dependencies.atomic-polyfill]
-version = "1"
-optional = true
-
 [target."cfg(target_arch = \"x86_64\")".dependencies.spin]
 version = "0.9.2"
 
-[target.riscv32i-unknown-none-elf.dependencies.atomic-polyfill]
-version = "1"
-
-[target.riscv32imc-unknown-none-elf.dependencies.atomic-polyfill]
-version = "1"
-
-[target.thumbv6m-none-eabi.dependencies.atomic-polyfill]
-version = "1"
-optional = true
-
-[target.xtensa-esp32s2-none-elf.dependencies.atomic-polyfill]
-version = "1"
