diff -ruN a/excon.gemspec b/excon.gemspec
--- a/excon.gemspec	2022-07-06 17:58:11.302837685 -0700
+++ b/excon.gemspec	2022-07-06 18:00:23.860442343 -0700
@@ -12,14 +12,7 @@
   s.license          = 'MIT'
   s.rdoc_options     = ["--charset=UTF-8"]
   s.extra_rdoc_files = %w[README.md CONTRIBUTORS.md CONTRIBUTING.md]
-  s.files            = `git ls-files -- data/* lib/*`.split("\n") + [
-    "CONTRIBUTING.md",
-    "CONTRIBUTORS.md",
-    "LICENSE.md",
-    "README.md",
-    "excon.gemspec"
-  ]
-
+  s.files            = Dir['**/*']
   s.add_development_dependency('rspec', '>= 3.5.0')
   s.add_development_dependency('activesupport')
   s.add_development_dependency('delorean')
