diff -purN boringssl-20190916.orig/crypto/cipher_extra/aead_test.cc boringssl-20190916/crypto/cipher_extra/aead_test.cc
--- boringssl-20190916.orig/src/crypto/cipher_extra/aead_test.cc	2020-03-10 20:57:53.383996634 +0100
+++ boringssl-20190916/src/crypto/cipher_extra/aead_test.cc	2020-03-10 20:59:03.108731229 +0100
@@ -31,7 +31,7 @@
 
 #if defined(OPENSSL_AARCH64) || defined(OPENSSL_PPC64LE)
 #define ALIGN_BYTES 16
-#elif defined(OPENSSL_S390X)
+#elif defined(OPENSSL_S390X) || defined(OPENSSL_ARM)
 #define ALIGN_BYTES 8
 #else
 #define ALIGN_BYTES 64
diff -purN boringssl-20190916.orig/crypto/poly1305/poly1305_test.cc boringssl-20190916/crypto/poly1305/poly1305_test.cc
--- boringssl-20190916.orig/src/crypto/poly1305/poly1305_test.cc	2020-03-10 20:57:53.363996424 +0100
+++ boringssl-20190916/src/crypto/poly1305/poly1305_test.cc	2020-03-10 21:08:32.606731399 +0100
@@ -25,7 +25,7 @@
 #include "../test/file_test.h"
 #include "../test/test_util.h"
 
-#if defined(OPENSSL_S390X)
+#if defined(OPENSSL_S390X) || defined(OPENSSL_ARM)
 #define ALIGN_BYTES 8
 #else
 #define ALIGN_BYTES 16
