diff -urN MIM/gu-itrans.mim.old MIM/gu-itrans.mim
--- MIM/gu-itrans.mim.old	2015-08-24 22:30:15.723875911 +0530
+++ MIM/gu-itrans.mim	2015-08-24 22:32:56.198910959 +0530
@@ -166,11 +166,13 @@
   ("RRI" (delete @-) "ૄ")		; not in ITRANS Gujarati table
   ("R^I" (delete @-) "ૄ")		; not in ITRANS Gujarati table
   (".c" (delete @-) "ૅ")
+  ("e." (delete @-) "ે.")	        ; not in ITRANS Gujarati table
   ("e.c" (delete @-) "ૅ")		; not in ITRANS Gujarati table
   ("e" (delete @-) "ે")
   ("ai" (delete @-) "ૈ")
-  ("o.c" (delete @-) "ૉ")
   ("o" (delete @-) "ો")
+  ("o." (delete @-) "ો.")		; not in ITRANS Gujarati table
+  ("o.c" (delete @-) "ૉ")
   ("au" (delete @-) "ૌ"))
 
 
