--- MIM/kn-itrans.mim.orig	2013-08-20 14:55:29.000000000 +0530
+++ MIM/kn-itrans.mim	2014-01-27 19:36:23.056606114 +0530
@@ -46,8 +46,44 @@
 Kannada l10n Team, kannada.l10n@gmail.com
   <http://kannada.sourceforge.net>
 on 18 Aug 2005. 
-")
 
+Key summary :
+
+1)	\"praa\" = pa + halant + raa
+ 	\"ಪ್ರಾ\" =  ಪ  +  ್  + ರ  +  ಾ
+	p + r + A
+ 
+2)	\"ska\" = sa+halant+ka
+ 	\"ಸ್ಕ\"  = ಸ  +  ್  +  ಕ
+	s + k + a
+ 
+3)	\"ththhaa\" = th + halanth + thhaa
+ 	\" ತ್ಥಾ\" = ತ + ್ +ಥ + ಾ
+	t + th + A
+
+4)	\"shhtya\" : shh + halat + T + halant + ya
+	\"ಷ್ಟ್ಯ\"  =  ಷ  +್   +  ಟ  +  ್  +  ಯ
+	S + h + T + ya 
+
+5)	\"dgaa\" : d + halant + gaa
+	\"ದ್ಗಾ\"  :  ದ + ್ + ಗ +ಾ
+	d + g + aa
+
+6)	\"ksha\" : k + halant + sha
+	\"ಕ್ಷ\" : ಕ + ್  + ಷ
+	k + Sh + a [OR] x
+
+7)	\"thra\": th + halant + r + a
+	\"ತ್ರ\" : ತ + ್ +ರ
+	t + r + a
+
+8)	\"jna\" : j + halant + na
+	\"ಜ್ಞ\" : ಜ + ್ + ಞ
+	G + Y + a
+
+Key summary: Shankar Prasad <svenkate@redhat.com>
+
+")
 (title "ಕ")
 
 (map
