--- MIM/bn-itrans.mim.orig	2013-05-23 18:22:04.000000000 +0530
+++ MIM/bn-itrans.mim	2014-01-27 19:27:28.818981529 +0530
@@ -127,16 +127,10 @@
   ("U" "ঊ")
   ("RRi" "ঋ")
   ("R^i" "ঋ")
-  ("LLi" "ঌ")				; not in ITRANS Bengali table
-  ("L^i" "ঌ")				; not in ITRANS Bengali table
   ("e" "এ")
   ("ai" "ঐ")
   ("o" "ও")
   ("au" "ঔ")
-  ("RRI" "ৠ")				; not in ITRANS Bengali table
-  ("R^I" "ৠ")				; not in ITRANS Bengali table
-  ("LLI" "ৡ")				; not in ITRANS Bengali table
-  ("L^I" "ৡ")				; not in ITRANS Bengali table
   (".N" "ঁ")
   (".n" "ং")
   ("M" "ং")				; not in ITRANS Bengali table
@@ -153,10 +147,6 @@
   ("7" "৭")
   ("8" "৮")
   ("9" "৯")
-  ("#" "্র")				; not in ITRANS Bengali table
-  ("$" "র্")				; not in ITRANS Bengali table
-  ("^" "ত্র")				; not in ITRANS Bengali table
-  ("*" "শ্র")				; not in ITRANS Bengali table
   ("]" "়")				; not in ITRANS Bengali table
   ((KP_1) "১")
   ((KP_2) "২")
@@ -183,16 +173,10 @@
   ("U" (delete @-) "ূ")
   ("RRi" (delete @-) "ৃ")
   ("R^i" (delete @-) "ৃ")
-  ("LLi" (delete @-) "ৢ")		; not in ITRANS Bengali table
-  ("L^i" (delete @-) "ৢ")		; not in ITRANS Bengali table
   ("e" (delete @-) "ে")
   ("ai" (delete @-) "ৈ")
   ("o" (delete @-) "ো")
-  ("au" (delete @-) "ৌ")
-  ("RRI" (delete @-) "ৄ")		; not in ITRANS Bengali table
-  ("R^I" (delete @-) "ৄ")		; not in ITRANS Bengali table
-  ("LLI" (delete @-) "ৣ")		; not in ITRANS Bengali table
-  ("L^I" (delete @-) "ৣ"))		; not in ITRANS Bengali table
+  ("au" (delete @-) "ৌ"))
 
  (return
   ((Return)))
