--- inscript2/IM/mni-inscript2-beng.mim	2016-04-23 09:41:08.000000000 +0200
+++ inscript2/IM/mni-inscript2-beng.mim	2019-10-10 11:58:18.068555240 +0200
@@ -18,7 +18,7 @@
 ;;
 ;; Author: Parag Nemade <pnemade@redhat.com>
 
-(input-method mni inscript2)
+(input-method mni inscript2-beng)
 
 (description "Manipuri input method for enhanced inscript layout.
 
--- inscript2/IM/mni-inscript2-mtei.mim	2016-04-23 09:41:10.000000000 +0200
+++ inscript2/IM/mni-inscript2-mtei.mim	2019-10-10 11:58:18.069555230 +0200
@@ -18,7 +18,7 @@
 ;;
 ;; Author: Parag Nemade <pnemade@redhat.com>
 
-(input-method mni inscript2)
+(input-method mni inscript2-mtei)
 
 (description "Manipuri Meetei Mayek input method for enhanced inscript layout.
 
@@ -87,13 +87,13 @@
   ((G-U) "ꯡ")
   ("Y" "ꯚ")
   ("]" "꯬")
-  ("a" "ꯣ "
+  ("a" "ꯣ")
   ("b" "ꯋ")
   ("c" "ꯃ")
   ((G-c) "ꯝ")
   ("d" "꯭")
   ("e" "ꯥ")
-  ("f" "ꯤ)
+  ("f" "ꯤ")
   ("g" "ꯨ")
   ("h" "ꯄ")
   ((G-h) "ꯞ")
--- inscript2/IM/sat-inscript2-deva.mim	2016-04-23 09:41:51.000000000 +0200
+++ inscript2/IM/sat-inscript2-deva.mim	2019-10-10 11:58:46.793249433 +0200
@@ -18,7 +18,7 @@
 ;;
 ;; Author: Parag Nemade <pnemade@redhat.com>
 
-(input-method sat inscript2)
+(input-method sat inscript2-deva)
 
 (description "Santali input method for enhanced inscript layout.
 
--- inscript2/IM/sat-inscript2-olck.mim	2016-04-23 09:41:49.000000000 +0200
+++ inscript2/IM/sat-inscript2-olck.mim	2019-10-10 11:58:46.793249433 +0200
@@ -18,7 +18,7 @@
 ;;
 ;; Author: Parag Nemade <pnemade@redhat.com>
 
-(input-method sat inscript2)
+(input-method sat inscript2-olck)
 
 (description "Santali Ol Chicky input method for enhanced inscript layout.
 
