diff --git a/testing/fulltests/default/T070com2sec_simple b/testing/fulltests/default/T070com2sec_simple
index 6c07f74..7df0b51 100644
--- a/testing/fulltests/default/T070com2sec_simple
+++ b/testing/fulltests/default/T070com2sec_simple
@@ -134,34 +134,30 @@ SAVECHECKAGENT '<"c406a", 255.255.255.255/255.255.255.255> => "t406a"'
 SAVECHECKAGENT 'line 30: Error:' # msg from h_strerror so it varies
 SAVECHECKAGENT 'line 31: Error:' # msg from h_strerror so it varies
 
-if false; then
-  # The two tests below have been disabled because these rely on resolving a
-  # domain name into a local IP address. Such DNS replies are filtered out by
-  # many security devices because to avoid DNS rebinding attacks. See also
-  # https://en.wikipedia.org/wiki/DNS_rebinding.
-
-  CHECKAGENT '<"c408a"'
-  if [ "$snmp_last_test_result" -eq 0 ] ; then
-    CHECKAGENT 'line 32: Error:'
-    if [ "$snmp_last_test_result" -ne 1 ] ; then
-      return_value=1
-      FINISHED
-    fi
-  elif [ "$snmp_last_test_result" -ne 1 ] ; then
+FINISHED
+
+# don't test the rest, it depends on DNS, which is not available in Koji
+
+CHECKAGENT '<"c408a"'
+if [ "$snmp_last_test_result" -eq 0 ] ; then
+  CHECKAGENT 'line 32: Error:'
+  if [ "$snmp_last_test_result" -ne 1 ] ; then
     return_value=1
     FINISHED
   fi
+elif [ "$snmp_last_test_result" -ne 1 ] ; then
+  return_value=1
+  FINISHED
+fi
 
-  CHECKAGENT '<"c408b"'
-  if [ "$snmp_last_test_result" -eq 0 ] ; then
-    CHECKAGENT 'line 33: Error:'
-    if [ "$snmp_last_test_result" -ne 1 ] ; then
-      return_value=1
-    fi
-  elif [ "$snmp_last_test_result" -ne 1 ] ; then
+CHECKAGENT '<"c408b"'
+if [ "$snmp_last_test_result" -eq 0 ] ; then
+  CHECKAGENT 'line 33: Error:'
+  if [ "$snmp_last_test_result" -ne 1 ] ; then
     return_value=1
   fi
-
+elif [ "$snmp_last_test_result" -ne 1 ] ; then
+  return_value=1
 fi
 
 FINISHED
diff --git a/testing/fulltests/default/T071com2sec6_simple b/testing/fulltests/default/T071com2sec6_simple
index 76da70b..bc2d432 100644
--- a/testing/fulltests/default/T071com2sec6_simple
+++ b/testing/fulltests/default/T071com2sec6_simple
@@ -132,30 +132,27 @@ SAVECHECKAGENT '<"c606a", ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff/ffff:ffff:ffff
 SAVECHECKAGENT 'line 27: Error:'
 SAVECHECKAGENT 'line 28: Error:'
 
-if false; then
-  # The two tests below have been disabled because these rely on resolving a
-  # domain name into a local IP address. Such DNS replies are filtered out by
-  # many security devices because to avoid DNS rebinding attacks. See also
-  # https://en.wikipedia.org/wiki/DNS_rebinding.
-
-  # 608
-  CHECKAGENT '<"c608a"'
-  if [ "$snmp_last_test_result" -eq 0 ] ; then
-    CHECKAGENT 'line 29: Error:'
-    errnum=`expr $errnum - 1`
-    if [ "$snmp_last_test_result" -ne 1 ] ; then
-      FINISHED
-    fi
-  elif [ "$snmp_last_test_result" -ne 1 ] ; then
+FINISHED
+
+# don't test the rest, it depends on DNS, which is not available in Koji
+
+# 608
+CHECKAGENT '<"c608a"'
+if [ "$snmp_last_test_result" -eq 0 ] ; then
+  CHECKAGENT 'line 29: Error:'
+  errnum=`expr $errnum - 1`
+  if [ "$snmp_last_test_result" -ne 1 ] ; then
     FINISHED
   fi
+elif [ "$snmp_last_test_result" -ne 1 ] ; then
+  FINISHED
+fi
 
-  CHECKAGENTCOUNT atleastone '<"c608b"'
-  if [ "$snmp_last_test_result" -eq 0 ] ; then
-    CHECKAGENT 'line 30: Error:'
-    if [ "$snmp_last_test_result" -eq 1 ] ; then
-      errnum=`expr $errnum - 1`
-    fi
+CHECKAGENTCOUNT atleastone '<"c608b"'
+if [ "$snmp_last_test_result" -eq 0 ] ; then
+  CHECKAGENT 'line 30: Error:'
+  if [ "$snmp_last_test_result" -eq 1 ] ; then
+    errnum=`expr $errnum - 1`
   fi
 fi
 
