From 7028523bd06796ca0afdc7d0abf4f6e0b41cf6a0 Mon Sep 17 00:00:00 2001
From: Muhammad Falak R Wani <falakreyaz@gmail.com>
Date: Tue, 26 Jul 2022 09:32:32 +0530
Subject: [PATCH] tox: add py39 environment

Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>
---
 tox.ini | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tox.ini b/tox.ini
index 352bcac..edf9e55 100644
--- a/tox.ini
+++ b/tox.ini
@@ -1,5 +1,5 @@
 [tox]
-envlist = py37,py36,py35,py34,py27,pypy3,pypy,pep8
+envlist = py39,py37,py36,py35,py34,py27,pypy3,pypy,pep8
 
 [testenv]
 setenv =
-- 
2.17.1

