From a47c412b44fcecdbb24842c149da221f171b5216 Mon Sep 17 00:00:00 2001
From: SumitJenaHCL <v-sumitjena@microsoft.com>
Date: Tue, 31 Mar 2026 20:54:39 +0530
Subject: [PATCH] modified redundant test for tmpfiles.d translation

---
 rust/src/composepost.rs | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/rust/src/composepost.rs b/rust/src/composepost.rs
index 2fafebcc..1186c39c 100644
--- a/rust/src/composepost.rs
+++ b/rust/src/composepost.rs
@@ -1189,6 +1189,11 @@ OSTREE_VERSION='33.4'
         umask(Mode::empty());
         let temp_rootfs = tempfile::tempdir().unwrap();
         let rootfs = openat::Dir::open(temp_rootfs.path()).unwrap();
+        let uid = geteuid().as_raw();
+        let gid = getegid().as_raw();
+        let uid_str = format!("{uid}");
+        let gid_str = format!("{gid}");
+        let mut expected_disk_size = 30u64;
         {
             for dirpath in &["usr/lib", "usr/etc", "var"] {
                 rootfs.ensure_dir_all(*dirpath, 0o755).unwrap();
@@ -1200,16 +1205,19 @@ OSTREE_VERSION='33.4'
                 .write_file_contents(
                     "usr/etc/passwd",
                     0o755,
-                    format!("test-user:x:{}:{}:::", geteuid(), getegid()),
+                    format!("test-user:x:{uid_str}:{gid_str}:::",),
                 )
                 .unwrap();
+            expected_disk_size += uid_str.len() as u64;
+            expected_disk_size += gid_str.len() as u64;
             rootfs
                 .write_file_contents(
                     "usr/etc/group",
                     0o755,
-                    format!("test-group:x:{}:", getegid()),
+                    format!("test-group:x:{gid_str}:")
                 )
                 .unwrap();
+            expected_disk_size += gid_str.len() as u64;
         }
 
         // Add test content.
@@ -1229,7 +1237,7 @@ OSTREE_VERSION='33.4'
         let cancellable = gio::Cancellable::new();
         assert_eq!(
             directory_size(rootfs.as_raw_fd(), cancellable.gobj_rewrap()).unwrap(),
-            42
+            expected_disk_size
         );
 
         var_to_tmpfiles(&rootfs, gio::NONE_CANCELLABLE).unwrap();
-- 
2.45.4

