diff -urNp openjade-1.3.2/grove/Makefile.sub devel/grove/Makefile.sub
--- openjade-1.3.2/grove/Makefile.sub	2002-10-20 23:47:24.000000000 +0200
+++ devel/grove/Makefile.sub	2007-07-23 15:39:23.000000000 +0200
@@ -1,4 +1,4 @@
 LTVERSION=0:1:0
 LIB=ogrove
-DEPLIBS=-lm
+DEPLIBS=-lm -lstdc++
 OBJS=Node.o LocNode.o
diff -urNp openjade-1.3.2/spgrove/Makefile.sub devel/spgrove/Makefile.sub
--- openjade-1.3.2/spgrove/Makefile.sub	2002-11-15 23:46:50.000000000 +0100
+++ devel/spgrove/Makefile.sub	2007-07-23 16:06:13.000000000 +0200
@@ -1,9 +1,10 @@
 LTVERSION=0:1:0
 LIB=ospgrove
 INCLUDE=-I$(srcdir)/../grove
-DEPLIBS=-lm -L$(TOP)/grove -L$(TOP)/grove/.libs \
+DEPLIBS=$(TOP)/grove/libogrove.la \
+       -lm -losp -L$(TOP)/grove/.libs \
 	-L$(TOP)/lib -L$(TOP)/lib/.libs \
-	$(LIB_THREADS)
+	$(LIB_THREADS) -lstdc++
 OBJS=GroveApp.o GroveBuilder.o SdNode.o
 GENSRCS=grove_inst.cxx
 
diff -urNp openjade-1.3.2/style/Makefile.sub devel/style/Makefile.sub
--- openjade-1.3.2/style/Makefile.sub	2003-04-18 19:18:10.000000000 +0200
+++ devel/style/Makefile.sub	2007-07-23 16:08:26.000000000 +0200
@@ -1,8 +1,8 @@
 LTVERSION=0:1:0
 LIB=ostyle
-DEPLIBS=-lm -L$(TOP)/grove -L$(TOP)/grove/.libs \
-  -L$(TOP)/lib -L$(TOP)/lib/.libs \
-  -L$(TOP)/spgrove -L$(TOP)/spgrove/.libs 
+DEPLIBS=$(TOP)/grove/libogrove.la $(TOP)/spgrove/libospgrove.la \
+  -lm -losp -L$(TOP)/lib -L$(TOP)/lib/.libs -L$(TOP)/grove/.libs \
+  -L$(TOP)/spgrove -L$(TOP)/spgrove/.libs -lstdc++
 OBJS=LangObj.o \
   Collector.o \
   DssslApp.o \
