diff --git a/conf/example.conf.in b/new/conf/example.conf.in
index 60f8023..19dce47 100644
--- a/conf/example.conf.in
+++ b/conf/example.conf.in
@@ -115,7 +115,7 @@
 	# Prefer the path with least value in lexicographical order.
 	#
 	# Example
+	preferred_names = [ "^/dev/mpath/", "^/dev/mapper/mpath", "^/dev/[hs]d" ]
-	# preferred_names = [ "^/dev/mpath/", "^/dev/mapper/mpath", "^/dev/[hs]d" ]
 	#
 	# This configuration option does not have a default value defined.
 
