From 807591c0367a973c011ae0a4395bc5d1b89a3898 Mon Sep 17 00:00:00 2001
From: kavyasree <kkaitepalli@microsoft.com>
Date: Mon, 9 Dec 2024 12:36:49 +0530
Subject: [PATCH] Removed vendor from gitignore list

---
 .gitignore | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/.gitignore b/.gitignore
index 7dd8d22..f4316af 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,7 +1,6 @@
 # Keep editor-specific, non-project specific ignore rules in global .gitignore:
 # https://help.github.com/articles/ignoring-files/#create-a-global-gitignore
 
-vendor
 .netrc
 .vscode
 .vs
@@ -129,8 +128,6 @@ junit-results
 # profile data
 /prof
 
-# vendored files
-/vendor
 
 # DShell Ignores
 .ash_history
-- 
2.34.1

