diff --git a/configure.inc b/configure.inc
index cdd67e2..b06b0c5 100755
--- a/configure.inc
+++ b/configure.inc
@@ -1433,9 +1433,6 @@ files)  echo "\$FULLNAME" "\$LIBNAME" "\$LIBNAME.\$MAJOR"
 install)$PROG_INSTALL -c \$FULLNAME "\$1"
 	$PROG_LN_S -f \$FULLNAME \$1/\$LIBNAME.\$MAJOR
 	$PROG_LN_S -f \$FULLNAME \$1/\$LIBNAME
-EOF
-    test "$LDCONFIG" && echo '	'$LDCONFIG '"$1"' >> librarian.sh
-    cat >> librarian.sh << EOF
 	;;
 esac
 EOF
