diff --git a/lib/Devel/Cycle.pm b/lib/Devel/Cycle.pm
index 19d3d63..2c5fdd1 100644
--- a/lib/Devel/Cycle.pm
+++ b/lib/Devel/Cycle.pm
@@ -215,7 +215,9 @@ sub _format_reference {
 sub _get_type {
   my $thingy = shift;
   return unless ref $thingy;
-  return 'SCALAR' if UNIVERSAL::isa($thingy,'SCALAR') || UNIVERSAL::isa($thingy,'REF');
+  return 'SCALAR' if UNIVERSAL::isa($thingy,'SCALAR') ||
+                     UNIVERSAL::isa($thingy,'REF') ||
+                     UNIVERSAL::isa($thingy,'Regexp');
   return 'ARRAY'  if UNIVERSAL::isa($thingy,'ARRAY');
   return 'HASH'   if UNIVERSAL::isa($thingy,'HASH');
   return 'CODE'   if UNIVERSAL::isa($thingy,'CODE');
