From b1a483b41c20bf2b87938cb5ff08ae49a0c055ae Mon Sep 17 00:00:00 2001
From: Pawel Winogrodzki <pawelwi@microsoft.com>
Date: Fri, 19 Nov 2021 17:57:32 -0800
Subject: [PATCH] Fix missing 'endif()'.

---
 configs/azure_c_shared_utilityFunctions.cmake | 22 +++++++++----------
 1 file changed, 10 insertions(+), 12 deletions(-)

diff --git a/configs/azure_c_shared_utilityFunctions.cmake b/configs/azure_c_shared_utilityFunctions.cmake
index caba758e..87224d7f 100644
--- a/configs/azure_c_shared_utilityFunctions.cmake
+++ b/configs/azure_c_shared_utilityFunctions.cmake
@@ -156,26 +156,24 @@ function(linux_unittests_add_exe whatIsBuilding)
             set(ARG_PREFIX "none")
             set(skip_to_next TRUE)
             #also unset all the other states
-
         elseif(${f} STREQUAL "VALGRIND_SUPPRESSIONS_FILE")
             SET(PARSING_ADDITIONAL_LIBS OFF)
             SET(PARSING_VALGRIND_SUPPRESSIONS_FILE ON)
             set(skip_to_next TRUE)
         else()
-
-        if(NOT skip_to_next)
-            if(PARSING_ADDITIONAL_LIBS)
-                if((${f} STREQUAL "debug") OR (${f} STREQUAL "optimized") OR (${f} STREQUAL "general"))
-                    SET(ARG_PREFIX ${f})
-                else()
-                    target_link_libraries_with_arg_prefix(${ARG_PREFIX} ${whatIsBuilding}_exe ${f})
-                    set(ARG_PREFIX "none")
+            if(NOT skip_to_next)
+                if(PARSING_ADDITIONAL_LIBS)
+                    if((${f} STREQUAL "debug") OR (${f} STREQUAL "optimized") OR (${f} STREQUAL "general"))
+                        SET(ARG_PREFIX ${f})
+                    else()
+                        target_link_libraries_with_arg_prefix(${ARG_PREFIX} ${whatIsBuilding}_exe ${f})
+                        set(ARG_PREFIX "none")
+                    endif()
+                elseif(PARSING_VALGRIND_SUPPRESSIONS_FILE)
+                    set(VALGRIND_SUPPRESSIONS_FILE_EXTRA_PARAMETER "--suppressions=${f}")
                 endif()
-            elseif(PARSING_VALGRIND_SUPPRESSIONS_FILE)
-                set(VALGRIND_SUPPRESSIONS_FILE_EXTRA_PARAMETER "--suppressions=${f}")
             endif()
         endif()
-
     endforeach()
 
     target_link_libraries(${whatIsBuilding}_exe micromock_ctest umock_c ctest)
-- 
2.17.1

