diff -up fxload-2008_10_13/Makefile.cflags fxload-2008_10_13/Makefile
--- fxload-2008_10_13/Makefile.cflags	2008-10-13 17:44:04.000000000 -0400
+++ fxload-2008_10_13/Makefile	2018-02-25 23:11:02.416103160 -0500
@@ -39,7 +39,7 @@ release:	rpms
 
 # object files
 $(PROG): $(FILES_OBJ)
-	$(CC) -o $(PROG) $(FILES_OBJ)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $(PROG) $(FILES_OBJ)
 
 %.o: %.c
 	$(CC) -c $(CFLAGS)  $< -o $@
