diff -ruN a/strptime.gemspec b/strptime.gemspec
--- a/strptime.gemspec	2020-09-25 20:57:14.000000000 -0700
+++ b/strptime.gemspec	2022-07-03 16:06:53.492017426 -0700
@@ -14,7 +14,7 @@
   spec.homepage      = "https://github.com/nurse/strptime"
   spec.license       = "BSD-2-Clause"
 
-  spec.files         = `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
+  spec.files         = Dir['**/*'].reject { |f| f.match(%r{^(test|spec|features)/}) }
   spec.bindir        = "exe"
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
