--- astring-0.8.5/test/test_char.ml.orig	2020-08-08 09:57:12.000000000 -0600
+++ astring-0.8.5/test/test_char.ml	2023-06-20 11:52:14.805901422 -0600
@@ -24,10 +24,10 @@ let misc = test "Char.{of_byte,of_int,to
 let predicates = test "Char.{equal,compare}" @@ fun () ->
   eq_bool (Char.equal ' ' ' ') true;
   eq_bool (Char.equal ' ' 'a') false;
-  eq_int (Char.compare ' ' 'a') (-1);
+  eq_int (Char.compare ' ' 'a') (-65);
   eq_int (Char.compare ' ' ' ') (0);
-  eq_int (Char.compare 'a' ' ') (1);
-  eq_int (Char.compare '\x00' ' ') (-1);
+  eq_int (Char.compare 'a' ' ') (65);
+  eq_int (Char.compare '\x00' ' ') (-32);
   ()
 
 let ascii_predicates = test "Char.Ascii.is_*" @@ fun () ->
