--- scowl/r/special/proper-names	2008-02-08 11:53:27.000000000 +0000
+++ scowl/r/special/proper-names	2008-02-08 12:06:00.000000000 +0000
@@ -18,7 +18,9 @@
 Bugzilla
 CSS
 CVS
+Camino
 Celgene
+ChatZilla
 Clojure
 Closure
 Cordova
@@ -45,6 +47,7 @@
 GitHub
 HBase
 Hadoop
+Haskell
 Heroku
 Hunspell
 Hurd
@@ -55,11 +58,13 @@
 Kotlin
 LibreOffice
 LyX
+Mandriva
 MariaDB
 Matlab
 Memcached
 MongoDB
 Mozilla
+MySpell
 MySQL
 NSA
 NVIDIA
@@ -83,10 +88,12 @@
 SVN
 Salesforce
 Scala
+SeaMonkey
 Seeger
 SharePoint
 Slackware
 Sourceforge
+Sunbird
 TensorFlow
 Thunderbird
 Troilus
