--- schemars-1.2.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ schemars-1.2.0/Cargo.toml	2025-12-29T20:14:59.356925+00:00
@@ -99,10 +99,6 @@
 path = "examples/validate.rs"
 
 [[test]]
-name = "integration"
-path = "tests/integration/main.rs"
-
-[[test]]
 name = "no_std"
 path = "tests/no_std.rs"
 
@@ -245,15 +241,6 @@
 default-features = false
 package = "either"
 
-[dev-dependencies.garde]
-version = "0.22"
-features = [
-    "derive",
-    "email",
-    "regex",
-    "url",
-]
-
 [dev-dependencies.indexmap2]
 version = "2.0"
 features = ["serde"]
@@ -264,10 +251,6 @@
 version = "0.2"
 features = ["serde"]
 package = "jiff"
-
-[dev-dependencies.jsonschema]
-version = "0.30"
-default-features = false
 
 [dev-dependencies.pretty_assertions]
 version = "1.2.1"
@@ -334,7 +317,3 @@
 features = ["serde"]
 default-features = false
 package = "uuid"
-
-[dev-dependencies.validator]
-version = "0.20"
-features = ["derive"]
