diff -up chromium-130.0.6723.44/components/media_router/common/providers/cast/channel/enum_table.h.me chromium-130.0.6723.44/components/media_router/common/providers/cast/channel/enum_table.h
--- chromium-130.0.6723.44/components/media_router/common/providers/cast/channel/enum_table.h.me	2024-10-15 09:58:39.571798463 +0200
+++ chromium-130.0.6723.44/components/media_router/common/providers/cast/channel/enum_table.h	2024-10-15 09:59:46.209147226 +0200
@@ -12,7 +12,6 @@
 
 #include <cstdint>
 #include <cstring>
-#include <new>
 #include <optional>
 #include <ostream>
 #include <string_view>
@@ -368,7 +367,8 @@ class EnumTable {
 
  private:
 #ifdef ARCH_CPU_64_BITS
-  alignas(std::hardware_destructive_interference_size)
+  // Align the data on a cache line boundary.
+  alignas(64)
 #endif
       std::initializer_list<Entry> data_;
   bool is_sorted_;
