diff -up chromium-132.0.6834.57/build/config/gcc/BUILD.gn.me chromium-132.0.6834.57/build/config/gcc/BUILD.gn
--- chromium-132.0.6834.57/build/config/gcc/BUILD.gn.me	2025-01-01 18:59:29.670002052 +0100
+++ chromium-132.0.6834.57/build/config/gcc/BUILD.gn	2025-01-01 20:51:11.798696436 +0100
@@ -32,7 +32,6 @@ declare_args() {
 # See http://gcc.gnu.org/wiki/Visibility
 config("symbol_visibility_hidden") {
   cflags = [ "-fvisibility=hidden" ]
-  rustflags = [ "-Zdefault-visibility=hidden" ]
 
   # Visibility attribute is not supported on AIX.
   if (current_os != "aix") {
