diff -up chromium-124.0.6367.155/ui/qt/BUILD.gn.me chromium-124.0.6367.155/ui/qt/BUILD.gn
--- chromium-124.0.6367.155/ui/qt/BUILD.gn.me	2024-05-08 18:15:34.178627040 +0200
+++ chromium-124.0.6367.155/ui/qt/BUILD.gn	2024-05-08 18:29:31.162513709 +0200
@@ -61,6 +61,7 @@ template("qt_shim") {
     packages = [
       "Qt" + invoker.qt_version + "Core",
       "Qt" + invoker.qt_version + "Widgets",
+      "Qt" + invoker.qt_version + "Gui",
     ]
   }
 
