description: disable building font tests
author: Michael Gilbert <mgilbert@debian.org>

--- a/skia/BUILD.gn
+++ b/skia/BUILD.gn
@@ -860,7 +860,7 @@ group("test_fonts_resources") {
   if (is_apple) {
     deps += [ ":test_fonts_bundle_data" ]
     data_deps += [ ":test_fonts_bundle_data" ]
-  } else {
+  } else if (false) {
     deps += [ "//third_party/test_fonts" ]
     data_deps += [ "//third_party/test_fonts" ]
   }
--- a/third_party/test_fonts/fontconfig/BUILD.gn
+++ b/third_party/test_fonts/fontconfig/BUILD.gn
@@ -8,9 +8,7 @@ if (is_linux || is_chromeos) {
   group("test_support") {
     testonly = true
     public_deps = [ ":fontconfig_util_linux" ]
-    data_deps = [ "//third_party/test_fonts" ]
     if (current_toolchain == host_toolchain) {
-      data_deps += [ ":do_generate_fontconfig_caches" ]
     }
   }
 
