kIndex: chromium-114.0.5735.45/base/allocator/partition_allocator/partition_alloc.gni
===================================================================
Index: chromium-144.0.7559.59/base/allocator/partition_allocator/partition_alloc.gni
===================================================================
--- chromium-144.0.7559.59.orig/base/allocator/partition_allocator/partition_alloc.gni
+++ chromium-144.0.7559.59/base/allocator/partition_allocator/partition_alloc.gni
@@ -71,7 +71,7 @@ is_clang_or_gcc = is_clang || !is_win
 # Whether 64-bit pointers are used.
 # A static_assert in partition_alloc_config.h verifies that.
 if (current_cpu == "x64" || current_cpu == "arm64" || current_cpu == "arm64e" ||
-    current_cpu == "loong64" || current_cpu == "riscv64") {
+    current_cpu == "loong64" || current_cpu == "riscv64" || current_cpu == "ppc64") {
   assert(current_cpu != "arm64e" || (is_ios && target_environment == "device"))
   has_64_bit_pointers = true
 } else if (current_cpu == "x86" || current_cpu == "arm" ||
