--- sequoia-keystore-0.7.2/Cargo.toml	1970-01-01T00:00:01+00:00
+++ sequoia-keystore-0.7.2/Cargo.toml	2026-02-10T17:08:20.114784+00:00
@@ -44,9 +44,7 @@
     "gpg-agent",
 ]
 gpg-agent = ["dep:sequoia-keystore-gpg-agent"]
-openpgp-card = ["dep:sequoia-keystore-openpgp-card"]
 softkeys = ["dep:sequoia-keystore-softkeys"]
-tpm = ["dep:sequoia-keystore-tpm"]
 
 [lib]
 name = "sequoia_keystore"
@@ -87,16 +85,8 @@
 version = "0.5"
 optional = true
 
-[dependencies.sequoia-keystore-openpgp-card]
-version = "0.2"
-optional = true
-
 [dependencies.sequoia-keystore-softkeys]
 version = "0.7"
-optional = true
-
-[dependencies.sequoia-keystore-tpm]
-version = "0.2"
 optional = true
 
 [dependencies.sequoia-openpgp]
