From 274976ff39d9142169a18655f5013e66233ac9c8 Mon Sep 17 00:00:00 2001
From: Akira TAGOH <akira@tagoh.org>
Date: Wed, 14 Feb 2018 13:20:40 +0900
Subject: [PATCH] Fix a build fail

---
 libgxim/gximcore.c       | 2 --
 libgxim/gximprotocol10.c | 2 +-
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/libgxim/gximcore.c b/libgxim/gximcore.c
index a92c413..3955d01 100644
--- a/libgxim/gximcore.c
+++ b/libgxim/gximcore.c
@@ -26,9 +26,7 @@
 #endif
 
 #include <glib/gi18n-lib.h>
-#ifdef GNOME_ENABLE_DEBUG
 #include <gdk/gdkx.h>
-#endif /* GNOME_ENABLE_DEBUG */
 #include "gximacc.h"
 #include "gximconnection.h"
 #include "gximmarshal.h"
diff --git a/libgxim/gximprotocol10.c b/libgxim/gximprotocol10.c
index 04887a5..65bd4df 100644
--- a/libgxim/gximprotocol10.c
+++ b/libgxim/gximprotocol10.c
@@ -2073,7 +2073,7 @@ g_xim_protocol10_closure_real_parser_error(GXimProtocol *proto,
 	msg = g_strdup_printf("Unable to parse the protocol %s properly",
 			      g_xim_protocol_name(major_opcode));
 
-	g_xim_messages_error(G_XIM_PROTOCOL_GET_IFACE (proto)->message, msg);
+	g_xim_messages_error(G_XIM_PROTOCOL_GET_IFACE (proto)->message, "%s", msg);
 	switch (major_opcode) {
 	    case G_XIM_CONNECT:
 	    case G_XIM_AUTH_REQUIRED:
-- 
2.14.3

