--- configure.old	2008-03-12 17:30:29.000000000 +0000
+++ configure	2008-03-12 17:34:06.000000000 +0000
@@ -10534,7 +10534,7 @@ cat >>confdefs.h <<\_ACEOF
 _ACEOF
 
       ;;
-    xpowerpc|xppc|xpowerpc64|xppc64)
+    xpowerpc|xppc)
       HAVE_POWERPC=yes
 
 cat >>confdefs.h <<\_ACEOF
@@ -10631,7 +10631,7 @@ eval "echo \"\$as_me:$LINENO: $ac_try_ec
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest.$ac_objext; then
-  flag_ok=yes
+  flag_ok=no
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
@@ -10641,7 +10641,7 @@ fi
 
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
 
-  if test "X$flag_ok" = Xyes ; then
+  if test "X$flag_ok" = Xyes -a "X$HAVE_POWERPC" = "Xyes" ; then
     HAVE_GCC_ASM_POWERPC_FPU=yes
     true
   else
