--- Newt-1.08/Newt.xs.fix	Fri Nov  6 18:34:52 1998
+++ Newt-1.08/Newt.xs	Mon Dec 17 16:13:22 2001
@@ -412,6 +412,12 @@
 #else
 	    goto not_there;
 #endif
+	if (strEQ(name, "NEWT_KEY_ESCAPE"))
+#ifdef NEWT_KEY_ESCAPE
+	    return NEWT_KEY_ESCAPE;
+#else
+	    goto not_there;
+#endif
 	if (strEQ(name, "NEWT_KEY_EXTRA_BASE"))
 #ifdef NEWT_KEY_EXTRA_BASE
 	    return NEWT_KEY_EXTRA_BASE;
--- Newt-1.08/Newt.pm.fix	Tue Jun 12 14:42:39 2001
+++ Newt-1.08/Newt.pm	Tue Jun 12 17:26:57 2001
@@ -68,6 +68,7 @@
 	NEWT_KEY_DOWN
 	NEWT_KEY_END
 	NEWT_KEY_ENTER
+	NEWT_KEY_ESCAPE
 	NEWT_KEY_EXTRA_BASE
 	NEWT_KEY_F1
 	NEWT_KEY_F10
@@ -117,6 +118,7 @@
 			    NEWT_KEY_DOWN
 			    NEWT_KEY_END
 			    NEWT_KEY_ENTER
+			    NEWT_KEY_ESCAPE
 			    NEWT_KEY_EXTRA_BASE
 			    NEWT_KEY_F1
 			    NEWT_KEY_F10
@@ -454,7 +456,7 @@
 sub Newt::Component::TakesFocus {
   my ($self, $yesorno) = @_;
 
-  Newt::newtComponentTakesFocus(self->{co}, $yesorno);
+  Newt::newtComponentTakesFocus($self->{co}, $yesorno);
   $self;
 }
 
@@ -514,12 +516,25 @@
   $self;
 }
 
+sub Newt::Form::Focus {
+  my ($self, $comp) = @_;
+
+  Newt::newtFormSetCurrent($self->{co}, $comp->{co});
+  $self;
+}
+
 sub Newt::Form::Run {
   my $self = shift;
 
   return Newt::newtFormRun($self->{co});
 }
 
+sub Newt::Form::GetCurrent {
+  my $self = shift;
+
+  return Newt::newtFormGetCurrent($self->{co});
+}
+
 sub Newt::Form::DESTROY {
   my $self = shift;
   
@@ -593,7 +608,7 @@
 
 ########################### Newt::Listbox
 
-sub Newt::Listbox::Add {
+sub Newt::Listbox::Append {
   my $self = shift;
 
   foreach (@_) {
