diff -up Newt-1.08/Newt.pm.formdestroy Newt-1.08/Newt.pm
--- Newt-1.08/Newt.pm.formdestroy	2010-06-17 16:39:32.630469243 +0100
+++ Newt-1.08/Newt.pm	2010-06-17 16:50:00.809469642 +0100
@@ -531,12 +531,6 @@ sub Newt::Form::GetCurrent {
   return Newt::newtFormGetCurrent($self->{co});
 }
 
-sub Newt::Form::DESTROY {
-  my $self = shift;
-  
-  Newt::newtFormDestroy($self->{co});
-}
-
 ########################### Newt::Label
 
 sub Newt::Label::Set {
