Changes from validator.w3.org's xml.dcl.

diff -up xhtml1-20020801/DTD/xhtml1.dcl.orig xhtml1-20020801/DTD/xhtml1.dcl
--- xhtml1-20020801/DTD/xhtml1.dcl.orig	2002-08-01 21:23:08.000000000 +0300
+++ xhtml1-20020801/DTD/xhtml1.dcl	2008-02-28 22:43:22.000000000 +0200
@@ -22,13 +22,13 @@
                 13        1      13
                 14       18  UNUSED
                 32       95      32
-               127        1  UNUSED
-               128       32  UNUSED
+               127        1     127  -- DEL, legal in XML 1.0 --
+               128       32     128  -- C1 controls, legal in XML 1.0 --
                160    55136     160
              55296     2048  UNUSED  -- surrogates --
              57344     8190   57344
              65534        2  UNUSED  -- FFFE and FFFF --
-             65536  1048576   65536
+             65536  1048576   65536  -- 16 planes outside BMP --
 
      CAPACITY NONE  -- Capacities are not restricted in XML --
 
