--- hostname-0.4.2/Cargo.toml	1970-01-01T00:00:01+00:00
+++ hostname-0.4.2/Cargo.toml	2026-03-19T14:04:36.562507+00:00
@@ -54,10 +54,6 @@
 path = "examples/hostname.rs"
 
 [[test]]
-name = "codegen"
-path = "tests/codegen.rs"
-
-[[test]]
 name = "version"
 path = "tests/version.rs"
 
@@ -70,9 +66,6 @@
 [dev-dependencies.version-sync]
 version = "0.9"
 
-[dev-dependencies.windows-bindgen]
-version = "0.65"
-
 [target.'cfg(any(unix, target_os = "redox"))'.dependencies.libc]
 version = "0.2"
 
