diff -ruN a/connection_pool.gemspec b/connection_pool.gemspec
--- a/connection_pool.gemspec	2021-04-14 08:09:49.000000000 -0700
+++ b/connection_pool.gemspec	2022-07-03 16:54:10.076388828 -0700
@@ -9,9 +9,9 @@
   s.homepage = "https://github.com/mperham/connection_pool"
   s.description = s.summary = "Generic connection pool for Ruby"
 
-  s.files = `git ls-files`.split("\n")
-  s.test_files = `git ls-files -- {test,spec,features}/*`.split("\n")
-  s.executables = `git ls-files -- bin/*`.split("\n").map { |f| File.basename(f) }
+  s.files = Dir['**/*']
+  s.test_files = s.files.grep(%r{^(test|spec|features)/}) 
+  s.executables = s.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   s.require_paths = ["lib"]
   s.license = "MIT"
   s.add_development_dependency "bundler"
