diff -urN libnvidia-container-1.5.1-orig/mk/common.mk libnvidia-container-1.5.1/mk/common.mk
--- libnvidia-container-1.5.1-orig/mk/common.mk	2021-09-27 09:27:48.290511043 -0700
+++ libnvidia-container-1.5.1/mk/common.mk	2021-09-27 09:28:29.354285161 -0700
@@ -26,7 +26,7 @@
 else
     DATE := $(shell date -u --iso-8601=minutes)
 endif
-REVISION ?= $(shell git rev-parse HEAD)
+REVISION = 0
 COMPILER := $(realpath $(shell which $(CC)))
 PLATFORM ?= $(shell uname -m)
 
