diff -urN libnvidia-container-1.9.0-orig/Makefile libnvidia-container-1.9.0/Makefile
--- libnvidia-container-1.9.0-orig/Makefile	2022-03-18 03:31:56.000000000 -0700
+++ libnvidia-container-1.9.0/Makefile	2022-03-29 15:20:11.362669600 -0700
@@ -168,6 +168,9 @@
 LIB_CPPFLAGS       += -isystem $(DEPS_DIR)$(includedir)/tirpc -DWITH_TIRPC
 LIB_LDLIBS_STATIC  += -l:libtirpc.a
 LIB_LDLIBS_SHARED  += -lpthread
+else
+CPPFLAGS           += -I/usr/include/tirpc
+LIB_LDLIBS_SHARED  += -lpthread -ltirpc
 endif
 ifeq ($(WITH_SECCOMP), yes)
 LIB_CPPFLAGS       += -DWITH_SECCOMP $(shell pkg-config --cflags libseccomp)

