diff -urN libnvidia-container-1.5.1-orig/Makefile libnvidia-container-1.5.1/Makefile
--- libnvidia-container-1.5.1-orig/Makefile	2021-09-27 09:27:48.290511043 -0700
+++ libnvidia-container-1.5.1/Makefile	2021-09-27 11:20:34.306110214 -0700
@@ -144,6 +144,9 @@
 LIB_CPPFLAGS       += -isystem $(DEPS_DIR)$(includedir)/tirpc -DWITH_TIRPC
 LIB_LDLIBS_STATIC  += -l:libtirpc.a
 LIB_LDLIBS_SHARED  += -lpthread
+else
+CPPFLAGS           += -I/usr/include/tirpc
+LIB_LDLIBS_SHARED  += -lpthread -ltirpc
 endif
 ifeq ($(WITH_SECCOMP), yes)
 LIB_CPPFLAGS       += -DWITH_SECCOMP $(shell pkg-config --cflags libseccomp)
 
