--- spin-0.12.3/Cargo.toml	2006-07-24T01:21:28+00:00
+++ spin-0.12.3/Cargo.toml	2026-08-19T17:18:49.715981+00:00
@@ -61,7 +61,6 @@
 once = []
 portable-atomic = ["dep:portable-atomic"]
 portable_atomic = ["portable-atomic"]
-rustc-dep-of-std = ["core"]
 rwlock = []
 spin_mutex = ["mutex"]
 std = []
@@ -83,17 +82,6 @@
     "rwlock",
 ]
 
-[[bench]]
-name = "mutex"
-path = "benches/mutex.rs"
-harness = false
-required-features = ["ticket_mutex"]
-
-[dependencies.core]
-version = "1.0.0"
-optional = true
-package = "rustc-std-workspace-core"
-
 [dependencies.lock_api_crate]
 version = "0.4"
 optional = true
@@ -104,6 +92,3 @@
 features = ["require-cas"]
 optional = true
 default-features = false
-
-[dev-dependencies.criterion]
-version = "0.4"
