diff -ruN a/http-hpack.gemspec b/http-hpack.gemspec
--- a/http-hpack.gemspec	2020-02-01 05:29:35.000000000 -0800
+++ b/http-hpack.gemspec	2022-07-03 15:53:19.343438913 -0700
@@ -11,7 +11,7 @@
 	spec.summary       = "A compresssor and decompressor for HTTP 2.0 HPACK."
 	spec.homepage      = "https://github.com/socketry/http-hpack"
 
-	spec.files         = `git ls-files -z`.split("\x0").reject do |f|
+	spec.files         = Dir['**/*'].reject do |f|
 		f.match(%r{^(test|spec|features)/})
 	end
 	
