diff -rupN --no-dereference Pillow-11.1.0/src/decode.c Pillow-11.1.0-new/src/decode.c
--- Pillow-11.1.0/src/decode.c	2025-01-02 06:00:59.000000000 +0100
+++ Pillow-11.1.0-new/src/decode.c	2026-02-14 19:03:46.593414472 +0100
@@ -186,7 +186,8 @@ _setimage(ImagingDecoderObject *decoder,
         state->ysize = y1 - y0;
     }
 
-    if (state->xsize <= 0 || state->xsize + state->xoff > (int)im->xsize ||
+    if (state->xoff < 0 || state->xsize <= 0 ||
+        state->xsize + state->xoff > (int)im->xsize || state->yoff < 0 ||
         state->ysize <= 0 || state->ysize + state->yoff > (int)im->ysize) {
         PyErr_SetString(PyExc_ValueError, "tile cannot extend outside image");
         return NULL;
diff -rupN --no-dereference Pillow-11.1.0/src/encode.c Pillow-11.1.0-new/src/encode.c
--- Pillow-11.1.0/src/encode.c	2025-01-02 06:00:59.000000000 +0100
+++ Pillow-11.1.0-new/src/encode.c	2026-02-14 19:03:46.593719582 +0100
@@ -253,7 +253,8 @@ _setimage(ImagingEncoderObject *encoder,
         state->ysize = y1 - y0;
     }
 
-    if (state->xsize <= 0 || state->xsize + state->xoff > im->xsize ||
+    if (state->xoff < 0 || state->xsize <= 0 ||
+        state->xsize + state->xoff > im->xsize || state->yoff < 0 ||
         state->ysize <= 0 || state->ysize + state->yoff > im->ysize) {
         PyErr_SetString(PyExc_SystemError, "tile cannot extend outside image");
         return NULL;
