diff -up imap-2007e/Makefile.system_c_client imap-2007e/Makefile
--- imap-2007e/Makefile.system_c_client	2008-06-04 13:43:35.000000000 -0500
+++ imap-2007e/Makefile	2011-06-13 14:13:04.467014334 -0500
@@ -665,9 +665,11 @@ an ua:
 	@$(MAKE) ssl$(SSLTYPE)
 	@echo Applying $@ process to sources...
 	$(TOOLS)/$@ "$(LN)" src/c-client c-client
-	$(TOOLS)/$@ "$(LN)" src/ansilib c-client
-	$(TOOLS)/$@ "$(LN)" src/charset c-client
+	$(TOOLS)/$@ "$(LN)" src/ansilib c-client
+	$(TOOLS)/$@ "$(LN)" src/charset c-client
 	$(TOOLS)/$@ "$(LN)" src/osdep/$(SYSTEM) c-client
+	cp -alf /usr/include/imap/* c-client/
+	#sleep 5
 	$(TOOLS)/$@ "$(LN)" src/mtest mtest
 	$(TOOLS)/$@ "$(LN)" src/ipopd ipopd
 	$(TOOLS)/$@ "$(LN)" src/imapd imapd
