--- reqwest-0.13.4/Cargo.toml	2006-07-24T01:21:28+00:00
+++ reqwest-0.13.4/Cargo.toml	2026-07-14T13:01:44.751538+00:00
@@ -180,7 +180,6 @@
     "tokio/fs",
     "dep:futures-util",
     "dep:tokio-util",
-    "dep:wasm-streams",
 ]
 system-proxy = ["hyper-util/client-proxy-system"]
 zstd = ["tower-http/decompression-zstd"]
@@ -241,47 +240,6 @@
 version = "0.5.2"
 features = ["limit"]
 default-features = false
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dependencies.js-sys]
-version = "0.3.77"
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dependencies.wasm-bindgen]
-version = "0.2.89"
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dependencies.wasm-bindgen-futures]
-version = "0.4.18"
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dependencies.wasm-streams]
-version = "0.5"
-optional = true
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dependencies.web-sys]
-version = "0.3.28"
-features = [
-    "AbortController",
-    "AbortSignal",
-    "Headers",
-    "Request",
-    "RequestInit",
-    "RequestMode",
-    "Response",
-    "Window",
-    "FormData",
-    "Blob",
-    "BlobPropertyBag",
-    "ServiceWorkerGlobalScope",
-    "RequestCredentials",
-    "File",
-    "ReadableStream",
-    "RequestCache",
-]
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dev-dependencies.wasm-bindgen]
-version = "0.2.89"
-features = ["serde-serialize"]
-
-[target.'cfg(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none")))'.dev-dependencies.wasm-bindgen-test]
-version = "0.3"
 
 [target.'cfg(not(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none"))))'.dependencies.cookie_crate]
 version = "0.18.0"
@@ -503,3 +461,4 @@
 level = "warn"
 priority = 0
 check-cfg = ["cfg(reqwest_unstable)"]
+
