diff -rupN --no-dereference mingw-w64-v13.0.0/mingw-w64-headers/configure mingw-w64-v13.0.0-new/mingw-w64-headers/configure
--- mingw-w64-v13.0.0/mingw-w64-headers/configure	2025-06-04 07:00:28.000000000 +0200
+++ mingw-w64-v13.0.0-new/mingw-w64-headers/configure	2025-07-13 15:32:18.523951561 +0200
@@ -2756,7 +2756,7 @@ printf "%s\n" "$with_widl" >&6; }
 
 if test "x$with_widl" = xno
 then :
-
+unset WIDL
 else case e in #(
   e)
   if test "x$srcdir" = x.
diff -rupN --no-dereference mingw-w64-v13.0.0/mingw-w64-headers/configure.ac mingw-w64-v13.0.0-new/mingw-w64-headers/configure.ac
--- mingw-w64-v13.0.0/mingw-w64-headers/configure.ac	2025-06-04 07:00:28.000000000 +0200
+++ mingw-w64-v13.0.0-new/mingw-w64-headers/configure.ac	2025-07-13 15:32:18.524596155 +0200
@@ -27,7 +27,7 @@ AC_ARG_WITH([widl],
   [AS_VAR_SET([with_widl],[no])])
 AC_MSG_RESULT([$with_widl])
 
-AS_VAR_IF([with_widl],[no],[],[
+AS_VAR_IF([with_widl],[no],[unset WIDL],[
   AS_VAR_IF([srcdir],[.],[],
     [AC_MSG_WARN([--with-widl used in out of the tree compilation.  Existing generated files will not be modified.])])
   AC_PATH_PROGS([WIDL], [$host-widl widl], [], [$with_widl])
