diff -up libaio-0.3.112/src/Makefile libaio-0.3.112/src/Makefile
--- libaio-0.3.112/src/Makefile    2020-03-17 19:00:28.855003728 +0000
+++ libaio-0.3.112/src/Makefile 2020-03-17 19:00:34.518998733 +0000
@@ -58,12 +58,12 @@ $(libname): $(libaio_sobjs) libaio.map
 	$(CC) $(SO_CFLAGS) -Wl,--version-script=libaio.map -Wl,-soname=$(soname) -o $@ $(libaio_sobjs) $(LINK_FLAGS)

 install: $(all_targets)
-	install -D -m 644 libaio.h $(includedir)/libaio.h
-	install -D -m 644 libaio.a $(libdir)/libaio.a
+	install -D -m 644 libaio.h $(destdir)$(includedir)/libaio.h
+	install -D -m 644 libaio.a $(destdir)$(usrlibdir)/libaio.a
 ifeq ($(ENABLE_SHARED),1)
-	install -D -m 755 $(libname) $(libdir)/$(libname)
-	ln -sf $(libname) $(libdir)/$(soname)
-	ln -sf $(libname) $(libdir)/libaio.so
+	install -D -m 755 $(libname) $(destdir)$(usrlibdir)/$(libname)
+	ln -sf $(libname) $(destdir)$(usrlibdir)/$(soname)
+	ln -sf $(libname) $(destdir)$(usrlibdir)/libaio.so
 endif

 $(libaio_objs): libaio.h
