From 5b7f57844682339340333034be1b8f99acafde28 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nikola=20Forr=C3=B3?= <nforro@redhat.com>
Date: Thu, 7 Jan 2016 12:26:40 +0100
Subject: [PATCH] Fix NSS hostname matching

---
 nss.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nss.c b/nss.c
index 96b0ea3..2acbd5f 100644
--- a/nss.c
+++ b/nss.c
@@ -189,7 +189,7 @@ nss_check_host(const char *server, struct sock *sp)
 					fprintf(stderr,
 						"Comparing DNS name: \"%s\"\n",
 						dn);
-				if (rfc2595_hostname_match(server, dn)
+				if ((ok = rfc2595_hostname_match(server, dn))
 						== OKAY) {
 					ac_free(dn);
 					goto out;
-- 
2.4.3

