diff -up pinfo-0.6.10/src/parse_config.h.me pinfo-0.6.10/src/parse_config.h
--- pinfo-0.6.10/src/parse_config.h.me	2020-02-24 16:01:08.119884715 +0100
+++ pinfo-0.6.10/src/parse_config.h	2020-02-24 16:01:19.396990442 +0100
@@ -85,7 +85,7 @@ typedef struct colours
 colours;
 #endif /* NO_COLOR_CURSES */
 
-int use_manual;
+extern int use_manual;
 
 int parse_config (void);
 int parse_line (char *line);
