diff --git a/Test/GccTest/CMakeLists.txt b/Test/GccTest/CMakeLists.txt
index 38bf581..9254d70 100644
--- a/Test/GccTest/CMakeLists.txt
+++ b/Test/GccTest/CMakeLists.txt
@@ -123,15 +123,6 @@ else()
 
     target_compile_options(CompileTest_gcc14_NoEH PUBLIC -Wall -std=c++14 -fno-exceptions)
 
-    add_executable(CompileTest_gcc98
-    ../CompileTest.cpp
-    ../ConstExpr.cpp 
-    ../CleanCompile.cpp
-    ../../SafeInt.hpp
-    )
-
-    target_compile_options(CompileTest_gcc98 PUBLIC -D SAFEINT_USE_CPLUSCPLUS_98 -Wall -std=c++98 -fno-exceptions)
-
     add_executable(safe_math_test_gcc
     ../c_safe_math/safe_math_test.c
     ../c_safe_math/safe_math_test.h
