--- yp-tools-yp-tools-4.2.2/src/yppasswd.c.strict-protorypes	2017-02-21 15:51:03.452034055 +0100
+++ yp-tools-yp-tools-4.2.2/src/yppasswd.c	2017-02-21 15:51:14.996030455 +0100
@@ -547,7 +547,7 @@ create_random_salt (char *salt, int num_
  * If other value is set or it is not set at all, SHA-512 is used.
  */ 
 static int
-get_env_hash_id()
+get_env_hash_id(void)
 {
   const char *v = getenv("YP_PASSWD_HASH");
   if (!v)
