From a53c942b7bc46a824c48ed9a44c3843883fdcb6c Mon Sep 17 00:00:00 2001
From: Sam Meluch <sammeluch@microsoft.com>
Date: Mon, 4 Mar 2024 20:27:29 -0800
Subject: [PATCH] add version limits

---
 setup.py | 2 +-
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/setup.py b/setup.py
index 00e7b01..c244e2f 100644
--- a/setup.py
+++ b/setup.py
@@ -63,7 +63,7 @@ if (
     sys.version_info[:2] >= (3, 6)
     and platform.python_implementation() != "PyPy"
 ):
-    EXTRAS_REQUIRE["tests_no_zope"].extend(["mypy", "pytest-mypy-plugins"])
+    EXTRAS_REQUIRE["tests_no_zope"].extend(["mypy", "pytest-mypy-plugins==3.0.0"])
 
 EXTRAS_REQUIRE["tests"] = EXTRAS_REQUIRE["tests_no_zope"] + ["zope.interface"]
 EXTRAS_REQUIRE["dev"] = (
-- 
2.34.1

