From 4363ab64a03b78d07fa281eadca792f707421b89 Mon Sep 17 00:00:00 2001
From: Sam Meluch <sammeluch@microsoft.com>
Date: Fri, 3 May 2024 13:10:32 -0700
Subject: [PATCH] Add version limit to pytest dep

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 00e7b01..e15e791 100644
--- a/setup.py
+++ b/setup.py
@@ -55,7 +55,7 @@ EXTRAS_REQUIRE = {
         "coverage[toml]>=5.0.2",
         "hypothesis",
         "pympler",
-        "pytest>=4.3.0",  # 4.3.0 dropped last use of `convert`
+        "pytest>=4.3.0,<8.0",  # 4.3.0 dropped last use of `convert`
         "six",
     ],
 }
-- 
2.34.1

