diff -up gsm-1.0-pl17/src/toast.c.warn gsm-1.0-pl17/src/toast.c
--- gsm-1.0-pl17/src/toast.c.warn	2017-06-28 02:09:38.000000000 +0200
+++ gsm-1.0-pl17/src/toast.c	2017-06-29 11:44:43.979453734 +0200
@@ -6,6 +6,9 @@
 
 /* $Header: /tmp_amd/presto/export/kbs/jutta/src/gsm/RCS/toast.c,v 1.8 1996/07/02 10:41:04 jutta Exp $ */
 
+#include	<stdio.h>
+#include	<unistd.h>
+#include	<getopt.h>
 #include	"toast.h"
 
 /*  toast -- lossy sound compression using the gsm library.
