--- rdist-6.1.5/src/client.c.hardlink	Thu Jun  3 13:56:31 1999
+++ rdist-6.1.5/src/client.c	Thu Jun  3 14:00:06 1999
@@ -342,7 +342,7 @@
 		lp->target = strdup(Tdest);
 	else
 		lp->target = NULL;
-	if (!lp->pathname || !lp->src || !(Tdest && lp->target))
+	if (!lp->pathname || !lp->src || (Tdest && !lp->target))
 		fatalerr("Cannot malloc memory in linkinfo.");
 
 	return((struct linkbuf *) NULL);
@@ -364,7 +364,7 @@
 	       "sendhardlink: rname='%s' pathname='%s' src='%s' target='%s'\n",
 		 rname, lp->pathname, lp->src, lp->target);
 		 
-	if (*lp->target == CNULL)
+	if (lp->target == NULL || *lp->target == CNULL)
 		(void) sendcmd(C_RECVHARDLINK, "%o %s %s", 
 			       opts, lp->pathname, rname);
 	else {
