--- rdist-6.1.5/rdist/defs.h.re_args	2005-09-28 17:21:11.000000000 +0200
+++ rdist-6.1.5/rdist/defs.h	2005-09-28 17:21:29.000000000 +0200
@@ -156,10 +156,6 @@
 extern char buf[];		/* general purpose buffer */
 extern int errno;		/* system error number */
 
-/* regex stuff, BSD4.2 compatibility. */
-extern char *re_comp _RE_ARGS ((const char *));
-extern int re_exec _RE_ARGS ((const char *));
-
 /* docmd.c */
 extern void docmds(char **dhosts, int argc, char **argv);
 extern void doarrow(char **filev, struct namelist *files, char *rhost, struct subcmd *cmds);
