--- rdist-6.1.5/config/config.h.oldpath	Tue Nov 10 04:59:48 1998
+++ rdist-6.1.5/config/config.h	Thu Jan 31 17:33:21 2002
@@ -58,7 +58,7 @@
  * don't have the old rdist, then uncomment the "#undef" line.
  */
 #ifndef _PATH_OLDRDIST
-#define _PATH_OLDRDIST	"/usr/ucb/oldrdist"	/* Enable compat */
+#define _PATH_OLDRDIST	"/usr/bin/oldrdist"	/* Enable compat */
 #endif
 /*#undef  _PATH_OLDRDIST*/				/* Disable compat */
 
--- rdist-6.1.5/doc/rdist.man.oldpath	Tue Nov 10 06:38:53 1998
+++ rdist-6.1.5/doc/rdist.man	Thu Jan 31 17:36:57 2002
@@ -165,10 +165,9 @@
 This option will only work if 
 .I rdist
 was compiled with the location of the old rdist
-(usually either
-.I /usr/ucb/oldrdist
-or
-.I /usr/old/rdist)
+(the path
+.I /usr/bin/oldrdist
+is used on Red Hat linux)
 and that program is available at run time.
 .PP
 .I Rdist
@@ -553,7 +552,7 @@
 should be of form
 .sp
 .RS
-\fIfacility\fB=\fItypes\fB:\fIfacility\fB=\fItypes...
+\fIfacility\fB=\fItypes\fB:\fIfacility\fB=\fItypes...\fR
 .RE
 .sp
 The valid facility names are:
--- rdist-6.1.5/README.oldpath	Fri Jul 19 19:24:09 1996
+++ rdist-6.1.5/README	Thu Jan 31 17:33:21 2002
@@ -65,12 +65,12 @@
 
 The way the old rdist started a server rdist is to run "rdist
 -Server".  If the new rdist is run with the "-Server" option, then it
-will exec a copy of the old rdist (usually /usr/old/rdist or
-/usr/ucb/oldrdist).  In this way, you get compatibility with hosts
+will exec a copy of the old rdist (the path /usr/bin/oldrdist in Red
+Hat linux).  In this way, you get compatibility with hosts
 running the old rdist attempting to rdist to a machine running new
 rdist.  If your host running new rdist wants to rdist to a host
 running the old rdist, then it must run the old rdist program
-(/usr/old/rdist or /usr/ucb/oldrdist).
+(/usr/bin/oldrdist in Red Hat linux).
 
 The definition _PATH_OLDRDIST in "config/config.h" controls the
 location of the old rdist.  If this is not defined, or the defined
