diff: rdist-6.1.5/config/mf: No such file or directory
--- rdist-6.1.5/config/os-linux.h.linux	Mon Nov  9 22:59:59 1998
+++ rdist-6.1.5/config/os-linux.h	Thu Nov 12 14:50:35 1998
@@ -77,7 +77,7 @@
 /*
  * Select the type of executable file format.
  */
-#define EXE_TYPE	EXE_AOUT
+#define EXE_TYPE	EXE_ELF
 
 /*
  * Select the type of statfs() system call (if any).
--- rdist-6.1.5/mf/Makefile.var.linux	Tue Nov 10 00:02:11 1998
+++ rdist-6.1.5/mf/Makefile.var	Thu Nov 12 14:50:35 1998
@@ -86,7 +86,7 @@
 #
 # Name of YACC.
 #
-#YACC 		= bison -y
+YACC 		= bison -y
 
 OPT		= -g
 RM		= rm
