From c53c576c09c5a3a1654b7a1f08fcc222a102499d Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Tue, 28 Feb 2017 09:00:41 -0800
Subject: [PATCH] libiscsi: fix discovery request timeout regression

---
 libiscsi/libiscsi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libiscsi/libiscsi.c b/libiscsi/libiscsi.c
index 755c18c..bb17dfc 100644
--- a/libiscsi/libiscsi.c
+++ b/libiscsi/libiscsi.c
@@ -136,6 +136,7 @@ int libiscsi_discover_sendtargets(struct libiscsi_context *context,
 
 	/* Fill the drec struct with all needed info */
 	memset(&drec, 0, sizeof drec);
+	drec.iscsid_req_tmo = -1;
 	idbm_sendtargets_defaults(&drec.u.sendtargets);
 	drec.type = DISCOVERY_TYPE_SENDTARGETS;
 	strlcpy(drec.address, address, sizeof(drec.address));
@@ -228,6 +229,7 @@ int libiscsi_discover_firmware(struct libiscsi_context *context,
 	CHECK(iface_create_ifaces_from_boot_contexts(&ifaces, &targets));
 
 	memset(&drec, 0, sizeof(drec));
+	drec.iscsid_req_tmo = -1;
 	drec.type = DISCOVERY_TYPE_FW;
 	rc = idbm_bind_ifaces_to_nodes(discovery_fw, &drec, &ifaces, &rec_list);
 	if (rc) {
-- 
2.26.2

