From cf4db608004f7c1f137ed556e3ba6b6f4d65da96 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Tue, 28 Feb 2017 10:06:42 -0800
Subject: [PATCH] libiscsi format-security build errors

---
 libiscsi/libiscsi.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libiscsi/libiscsi.c b/libiscsi/libiscsi.c
index bb17dfc..7003388 100644
--- a/libiscsi/libiscsi.c
+++ b/libiscsi/libiscsi.c
@@ -177,7 +177,8 @@ int libiscsi_discover_sendtargets(struct libiscsi_context *context,
 		*found_nodes = calloc(found, sizeof **found_nodes);
 		if (*found_nodes == NULL) {
 			snprintf(context->error_str,
-				 sizeof(context->error_str), strerror(ENOMEM));
+				 sizeof(context->error_str),
+				 "%s", strerror(ENOMEM));
 			rc = ENOMEM;
 			goto leave;
 		}
@@ -634,7 +635,7 @@ static int get_parameter_helper(void *data, node_rec_t *rec)
 	info = idbm_recinfo_alloc(MAX_KEYS);
 	if (!info) {
 		snprintf(context->error_str, sizeof(context->error_str),
-			 strerror(ENOMEM));
+			 "%s", strerror(ENOMEM));
 		return ENOMEM;
 	}
 
-- 
2.26.2

