From 55af753f593243bcd1ab4c7e82620bdee432915b Mon Sep 17 00:00:00 2001
From: Chris Leech <cleech@redhat.com>
Date: Thu, 7 Nov 2019 09:16:17 -0800
Subject: [PATCH] libiscsi: fix build against latest upstream, again

---
 libiscsi/Makefile   | 4 ++--
 libiscsi/libiscsi.c | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/libiscsi/Makefile b/libiscsi/Makefile
index f2cf248..462d666 100644
--- a/libiscsi/Makefile
+++ b/libiscsi/Makefile
@@ -10,7 +10,7 @@ WARNFLAGS ?= -Wall -Wstrict-prototypes
 CFLAGS = $(OPTFLAGS) $(WARNFLAGS) -I../include -I../usr \
 		-D$(OSNAME) -fPIC -D_GNU_SOURCE -fvisibility=hidden \
 		-I../libopeniscsiusr
-LDFLAGS = -L../libopeniscsiusr -lopeniscsiusr
+LDFLAGS = -L../libopeniscsiusr -lopeniscsiusr -lkmod -lcrypto
 LIB = libiscsi.so.0
 TESTS = tests/test_discovery_sendtargets tests/test_discovery_firmware
 TESTS += tests/test_login tests/test_logout tests/test_params
@@ -19,7 +19,7 @@ TESTS += tests/test_set_auth tests/test_get_auth
 
 COMMON_SRCS = sysdeps.o
 # sources shared between iscsid, iscsiadm and iscsistart
-ISCSI_LIB_SRCS = netlink.o transport.o cxgbi.o be2iscsi.o iscsi_timer.o initiator_common.o iscsi_err.o session_info.o iscsi_util.o io.o auth.o discovery.o login.o log.o md5.o sha1.o iface.o idbm.o sysfs.o iscsi_sysfs.o iscsi_net_util.o iscsid_req.o iser.o uip_mgmt_ipc.o
+ISCSI_LIB_SRCS = netlink.o transport.o cxgbi.o be2iscsi.o iscsi_timer.o initiator_common.o iscsi_err.o session_info.o iscsi_util.o io.o auth.o discovery.o login.o log.o iface.o idbm.o sysfs.o iscsi_sysfs.o iscsi_net_util.o iscsid_req.o iser.o uip_mgmt_ipc.o
 FW_PARAM_SRCS = fw_entry.o prom_lex.o prom_parse.tab.o fwparam_ppc.o fwparam_sysfs.o
 
 # sources shared with the userspace utils, note we build these separately
diff --git a/libiscsi/libiscsi.c b/libiscsi/libiscsi.c
index 7003388..c598aee 100644
--- a/libiscsi/libiscsi.c
+++ b/libiscsi/libiscsi.c
@@ -429,7 +429,7 @@ int libiscsi_node_login(struct libiscsi_context *context,
 
 	CHECK(idbm_for_each_iface(&nr_found, (void*)node->iface, login_helper,
 		(char *)node->name, node->tpgt,
-		(char *)node->address, node->port))
+		(char *)node->address, node->port, false))
 	if (nr_found == 0) {
 		strcpy(context->error_str, "No such node");
 		rc = ENODEV;
@@ -615,7 +615,7 @@ int libiscsi_node_set_parameter(struct libiscsi_context *context,
 
 	CHECK(idbm_for_each_iface(&nr_found, &params, idbm_node_set_param,
 		(char *)node->name, node->tpgt,
-		(char *)node->address, node->port))
+		(char *)node->address, node->port, false))
 	if (nr_found == 0) {
 		strcpy(context->error_str, "No such node");
 		rc = ENODEV;
@@ -677,7 +677,7 @@ int libiscsi_node_get_parameter(struct libiscsi_context *context,
 	   as most settings should be the same independent of the iface. */
 	CHECK(idbm_for_each_iface(&nr_found, context, get_parameter_helper,
 		(char *)node->name, node->tpgt,
-		(char *)node->address, node->port))
+		(char *)node->address, node->port, false))
 	if (nr_found == 0) {
 		strcpy(context->error_str, "No such node");
 		rc = ENODEV;
-- 
2.26.2

