From e35261316aaa598c4146a5396745cb76571e94fe Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Fri, 25 May 2018 09:39:07 -0700
Subject: [PATCH] dont install scripts

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 4ab091f..7e6b734 100644
--- a/Makefile
+++ b/Makefile
@@ -116,7 +116,7 @@ install_systemd:
 	$(INSTALL) -d $(DESTDIR)$(systemddir)
 	$(INSTALL) -m 644 $(SYSTEMDFILES) $(DESTDIR)/$(systemddir)
 
-install_programs:  $(PROGRAMS) $(SCRIPTS)
+install_programs:  $(PROGRAMS) # $(SCRIPTS)
 	$(INSTALL) -d $(DESTDIR)$(sbindir)
 	$(INSTALL) -m 755 $^ $(DESTDIR)$(sbindir)
 
-- 
2.26.2

