From b50ab4a08a12593985b14aece5f689e485647b1c Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Wed, 16 Oct 2019 23:17:20 -0700
Subject: [PATCH 1/1] fix upstream build breakage of iscsiuio LDFLAGS

---
 iscsiuio/configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iscsiuio/configure.ac b/iscsiuio/configure.ac
index 9b85448..5dc188b 100644
--- a/iscsiuio/configure.ac
+++ b/iscsiuio/configure.ac
@@ -72,7 +72,7 @@ AC_ARG_WITH([systemd],
      *)   AC_MSG_ERROR([bad value $withval for --with-systemd]) ;;
      esac],[with_libsystemd=auto])
 AS_IF([test "$with_libsystemd" != no],[
-    PKG_CHECK_MODULES([LIBSYSTEMD],[libsystemd],[LDFLAGS=$LIBSYSTEMD_LIBS],[
+    PKG_CHECK_MODULES([LIBSYSTEMD],[libsystemd],[LDFLAGS="$LDFLAGS $LIBSYSTEMD_LIBS"],[
         if test "$with_libsystemd" = yes; then
             AC_MSG_ERROR([could not find libsystemd using pkg-config])
 	else
-- 
2.26.3

