From 53886d75a93ee1747d84187a0498abcf6cebd210 Mon Sep 17 00:00:00 2001
From: Chris Leech <cleech@redhat.com>
Date: Mon, 21 Jan 2013 15:43:36 -0800
Subject: [PATCH 1/1] use Red Hat version string to match RPM package version

---
 usr/version.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr/version.h b/usr/version.h
index 3c68989..8af23a4 100644
--- a/usr/version.h
+++ b/usr/version.h
@@ -6,7 +6,7 @@
  * This may not be the same value as the kernel versions because
  * some other maintainer could merge a patch without going through us
  */
-#define ISCSI_VERSION_STR	"2.1.4"
+#define ISCSI_VERSION_STR	"6.2.1.4"
 #define ISCSI_VERSION_FILE	"/sys/module/scsi_transport_iscsi/version"
 
 #endif
-- 
2.26.3

