diff --git a/libopeniscsiusr/Makefile b/libopeniscsiusr/Makefile
index a045a45..5e51f20 100644
--- a/libopeniscsiusr/Makefile
+++ b/libopeniscsiusr/Makefile
@@ -13,11 +13,7 @@ prefix ?= /usr
 INSTALL ?= install
 
 ifndef LIB_DIR
-	ifeq ($(shell test -d /lib64 && echo 1),1)
-		LIB_DIR=$(prefix)/lib64
-	else
-		LIB_DIR=$(prefix)/lib
-	endif
+	LIB_DIR=$(prefix)/lib
 endif
 
 INCLUDE_DIR ?= $(prefix)/include