From b9c0329fb653d416b0d47f327ab3ee2480a1a009 Mon Sep 17 00:00:00 2001
From: Brian Fjeldstad <bfjelds@microsoft.com>
Date: Fri, 12 Jan 2024 00:11:22 +0000
Subject: [PATCH] fix libpath error

---
 libopeniscsiusr/Makefile | 6 +-----
 libiscsi/Makefile | 2 +-----
 2 file changed, 2 insertion(+), 6 deletions(-)

diff --git a/libopeniscsiusr/Makefile b/libopeniscsiusr/Makefile
index 0d215fc..8cc0214 100644
--- a/libopeniscsiusr/Makefile
+++ b/libopeniscsiusr/Makefile
@@ -17,11 +17,7 @@ SBINDIR ?= $(exec_prefix)/sbin
 DBROOT ?= $(etcdir)/iscsi
 
 ifndef LIB_DIR
-	ifeq ($(shell test -d /lib64 && echo 1),1)
-		LIB_DIR=$(prefix)/lib64
-	else
-		LIB_DIR=$(prefix)/lib
-	endif
+	LIB_DIR=$(prefix)/lib
 endif
 
 INCLUDE_DIR ?= $(prefix)/include
diff --git a/libiscsi/Makefile b/libiscsi/Makefile
index 9f7b034..5d29792 100644
--- a/libiscsi/Makefile
+++ b/libiscsi/Makefile
@@ -19,7 +19,7 @@ CFLAGS += -DSBINDIR=\"$(SBINDIR)\"
 CFLAGS += -DISCSI_DB_ROOT=\"$(DBROOT)\"
 CFLAGS += -DISCSI_VERSION_STR=\"$(ISCSI_VERSION_STR)\"

-CFLAGS += -L../redhat-linux-build
+CFLAGS += -L../$(RPM_ARCH)-$(TARGET_OS)-$(RPM_OS)
 LDLIBS += -lopeniscsiusr -lkmod -lcrypto

 LIB = libiscsi.so.0
-- 
2.34.1

