From d9f795c8181d2db70626b7d43ebb2e6e8d2fbed8 Mon Sep 17 00:00:00 2001
From: Rohit Rawat <xordux@gmail.com>
Date: Tue, 29 Oct 2024 14:10:20 +0000
Subject: [PATCH] Disable flaky mem defrag tests

---
 tests/unit/memefficiency.tcl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/unit/memefficiency.tcl b/tests/unit/memefficiency.tcl
index d5a6a6e..37e1711 100644
--- a/tests/unit/memefficiency.tcl
+++ b/tests/unit/memefficiency.tcl
@@ -720,6 +720,7 @@ run_solo {defrag} {
     }
     }
 
+    if {0} {
     start_cluster 1 0 {tags {"defrag external:skip cluster"} overrides {appendonly yes auto-aof-rewrite-percentage 0 save ""}} {
         test_active_defrag "cluster"
     }
@@ -727,4 +728,5 @@ run_solo {defrag} {
     start_server {tags {"defrag external:skip standalone"} overrides {appendonly yes auto-aof-rewrite-percentage 0 save ""}} {
         test_active_defrag "standalone"
     }
+    }
 } ;# run_solo
-- 
2.39.4

