From e10d46d1cb92c25d1f72735661496fe8527d499c Mon Sep 17 00:00:00 2001
From: Kanishk Bansal <kanbansal@microsoft.com>
Date: Wed, 8 Oct 2025 06:42:39 +0000
Subject: [PATCH] Disable flaky mem defrag tests

---
 tests/unit/memefficiency.tcl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/unit/memefficiency.tcl b/tests/unit/memefficiency.tcl
index 4dc04f6..7418980 100644
--- a/tests/unit/memefficiency.tcl
+++ b/tests/unit/memefficiency.tcl
@@ -721,6 +721,7 @@ run_solo {defrag} {
     }
     }
 
+    if {0} {
     start_cluster 1 0 {tags {"defrag external:skip cluster"} overrides {appendonly yes auto-aof-rewrite-percentage 0 save "" lazyfree-lazy-user-del no}} {
         test_active_defrag "cluster"
     }
@@ -728,4 +729,5 @@ run_solo {defrag} {
     start_server {tags {"defrag external:skip standalone"} overrides {appendonly yes auto-aof-rewrite-percentage 0 save "" lazyfree-lazy-user-del no}} {
         test_active_defrag "standalone"
     }
+    }
 } ;# run_solo
-- 
2.45.4

