diff --git a/valkey.conf b/valkey.conf
index 0f43f5c..f6738ad 100644
--- a/valkey.conf
+++ b/valkey.conf
@@ -351,7 +351,7 @@ loglevel notice
 # Specify the log file name. Also the empty string can be used to force
 # the server to log on the standard output. Note that if you use standard
 # output for logging but daemonize, logs will be sent to /dev/null
-logfile ""
+logfile "/var/log/valkey/valkey.log"
 
 # To enable logging to the system logger, just set 'syslog-enabled' to yes,
 # and optionally update the other syslog parameters to suit your needs.
@@ -526,7 +526,7 @@ rdb-del-sync-files no
 # 'cluster-config-file' configuration directive is a relative path.
 #
 # Note that you must specify a directory here, not a file name.
-dir ./
+dir /var/lib/valkey
 
 ################################# REPLICATION #################################
 