--- mysql-8.4.0/extra/boost/boost_1_84_0/boost/mpl/print.hpp	2024-04-10 08:26:28.000000000 +0200
+++ mysql-8.4.0/extra/boost/boost_1_84_0/boost/mpl/print.hpp_patched	2024-05-28 15:35:45.071390863 +0200
@@ -55,16 +55,15 @@ struct print
     enum { n = sizeof(T) + -1 };
 #elif defined(__MWERKS__)
     void f(int);
-#else 
-    enum {
-        n =
-# if defined(__EDG_VERSION__)
-           aux::dependent_unsigned<T>::value > -1
-# else 
-           sizeof(T) > -1
-# endif 
-        };
-#endif 
+#elif defined(__EDG_VERSION__)
+    enum { n = aux::dependent_unsigned<T>::value > -1 };
+#elif defined(BOOST_GCC)
+    enum { n1 };
+    enum { n2 };
+    enum { n = n1 != n2 };
+#else
+    enum { n = sizeof(T) > -1 };
+#endif
 };
 
 #if defined(BOOST_MSVC)
