diff --git a/src/espeak.cpp b/src/espeak.cpp
index dbd83f2..bf9c6c3 100644
--- a/src/espeak.cpp
+++ b/src/espeak.cpp
@@ -83,7 +83,8 @@ static const char *help_text =
 "--version  Shows version number and date, and location of espeak-data\n"
 "--voices=<language>\n"
 "\t   List the available voices for the specified language.\n"
-"\t   If <language> is omitted, then list all voices.\n";
+"\t   If <language> is omitted, then list all voices.\n"
+"-h, --help Show this help.\n";
 
 
 
