Index: fillup-1.42/SRC/fillup_cfg.h
===================================================================
--- fillup-1.42.orig/SRC/fillup_cfg.h
+++ fillup-1.42/SRC/fillup_cfg.h
@@ -23,9 +23,9 @@
 
 /*-------------------------------- VARIABLES ---------------------------------*/
 
-const char    * cfg_delimiter;
-const char    * cfg_commentMarker;
-const char    * cfg_quotingMarker;
+extern const char    * cfg_delimiter;
+extern const char    * cfg_commentMarker;
+extern const char    * cfg_quotingMarker;
 
 /*----------------------------------------------------------------------------*/
 
