From 1399a8999e2c9d397190417b62a35ad56e5a3898 Mon Sep 17 00:00:00 2001
From: Muhammad Falak R Wani <falakreyaz@gmail.com>
Date: Mon, 7 Mar 2022 20:21:15 +0530
Subject: [PATCH] test: make: create `DIFFERENCE` dir before testing

Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>
---
 TEST/Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/TEST/Makefile b/TEST/Makefile
index 43391cc..a2d5b28 100644
--- a/TEST/Makefile
+++ b/TEST/Makefile
@@ -9,18 +9,23 @@ testsequence: parameter replacement metadata getvariable shiftregister
 
 
 parameter:
+	mkdir -p PARAMETERS/DIFFERENCE
 	make -C PARAMETERS
 
 replacement:
+	mkdir -p REPLACEMENT/DIFFERENCE
 	make -C REPLACEMENT
 
 metadata:
+	mkdir -p METADATA/DIFFERENCE
 	make -C METADATA
 
 getvariable:
+	mkdir -p GETVARIABLE/DIFFERENCE
 	make -C GETVARIABLE
 
 shiftregister:
+	mkdir -p FCSR/DIFFERENCE
 	make -C FCSR
 
 clean:
-- 
2.17.1

