--- setup.cfg~	2019-10-26 02:53:11.000000000 -0400
+++ setup.cfg	2020-01-17 11:34:42.559534428 -0500
@@ -1,6 +1,6 @@
 [options]
 setup_requires =
-  setuptools_scm>=3.3.1
+  setuptools_scm>=1.15.7
 
 [flake8]
 max-line-length = 140
--- pyproject.toml~	2019-10-26 02:53:11.000000000 -0400
+++ pyproject.toml	2020-01-17 11:34:34.582559695 -0500
@@ -1,2 +1,2 @@
 [build-system]
-requires = ["setuptools>=30.3.0", "wheel", "setuptools_scm>=3.3.1"]
+requires = ["setuptools>=30.3.0", "wheel", "setuptools_scm>=1.15.7"]
--- setup.py~	2019-10-26 02:53:11.000000000 -0400
+++ setup.py	2020-01-17 11:43:45.380615043 -0500
@@ -60,11 +60,6 @@
 
 setup(
     name='lazy-object-proxy',
-    use_scm_version={
-        'local_scheme': 'dirty-tag',
-        'write_to': 'src/lazy_object_proxy/_version.py',
-        'fallback_version': '1.4.3',
-    },
     license='BSD-2-Clause',
     description='A fast and thorough lazy object proxy.',
     long_description='%s\n%s' % (
