diff --git a/lib/ipmi_sdr.c b/lib/ipmi_sdr.c
index fa7b082..9bc5ac2 100644
--- a/lib/ipmi_sdr.c
+++ b/lib/ipmi_sdr.c
@@ -572,6 +572,8 @@ ipmi_sdr_get_sensor_reading_ipmb(struct ipmi_intf *intf, uint8_t sensor,
 	uint32_t save_addr;
 	uint32_t save_channel;
 
+	if (target == (uint8_t) 0xb1)
+		return ipmi_sdr_get_sensor_reading(intf, sensor);
 	if ( BRIDGE_TO_SENSOR(intf, target, channel) ) {
 		lprintf(LOG_DEBUG,
 			"Bridge to Sensor "
-- 
2.1.0

