diff -urNp old/configure.ac new/configure.ac
--- old/configure.ac	2017-02-02 14:20:33.230784269 +0100
+++ new/configure.ac	2017-02-02 14:22:53.528510336 +0100
@@ -63,7 +63,7 @@ xenable_intf_imb=yes
 xenable_intf_lipmi=yes
 xenable_intf_open=yes
 #xenable_intf_serial=yes
-xenable_intf_usb=no
+xenable_intf_usb=yes
 xenable_ipmishell=yes
 
 dnl set some things so we build with GNU tools on Solaris
@@ -209,7 +209,7 @@ fi
 dnl enable IPMI USB interface
 AC_ARG_ENABLE([intf-usb],
 	[AC_HELP_STRING([--enable-intf-usb],
-			[enable IPMI USB interface [default=auto]])],
+			[enable IPMI USB interface [default=yes]])],
 		[xenable_intf_usb=$enableval],
 		[xenable_intf_usb=$xenable_intf_usb])
 if test "x$xenable_intf_usb" = "xstatic" || test "x$xenable_intf_usb" = "xplugin"; then
