From f65ea137f0d03f883219a791a49cf8ea7e16776a Mon Sep 17 00:00:00 2001
From: Vaclav Dolezal <vdolezal@redhat.com>
Date: Fri, 1 Mar 2019 14:46:12 +0100
Subject: [PATCH] Fix "ipmitool pef {status,info}" not printing final newline

Signed-off-by: Vaclav Dolezal <vdolezal@redhat.com>
---
 lib/ipmi_pef.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/ipmi_pef.c b/lib/ipmi_pef.c
index bbf25f2..4be749e 100644
--- a/lib/ipmi_pef.c
+++ b/lib/ipmi_pef.c
@@ -1183,6 +1183,7 @@ ipmi_pef2_get_info(struct ipmi_intf *intf)
 		ipmi_pef_print_guid(guid_ptr);
 	}
 	ipmi_pef_print_flags(&pef_b2s_actions, P_SUPP, pcap.actions);
+	putchar('\n');
 	return 0;
 }
 
@@ -1242,6 +1243,7 @@ ipmi_pef2_get_status(struct ipmi_intf *intf)
 		return (-1);
 	}
 	ipmi_pef_print_flags(&pef_b2s_actions, P_ACTV, rsp->data[1]);
+	putchar('\n');
 	return 0;
 }
 
-- 
2.20.1

