diff -up netpbm-10.47.04/buildtools/endiangen.c.multilib netpbm-10.47.04/buildtools/endiangen.c
--- netpbm-10.47.04/buildtools/endiangen.c.multilib	2009-10-21 13:38:54.000000000 +0200
+++ netpbm-10.47.04/buildtools/endiangen.c	2009-10-21 14:35:58.000000000 +0200
@@ -87,7 +87,6 @@ main(int argc, char **argv) {
            byteOrder() == ENDIAN_LITTLE ? "LITTLE_ENDIAN" : "BIG_ENDIAN");
     printf("#endif\n");
     printf("\n");
-    printf("#define BITS_PER_LONG %u\n", bitsPerLong());
 
     return 0;
 }
