diff -urNp a/config.mk.in b/config.mk.in
--- a/config.mk.in	2018-11-21 12:46:22.044790058 +0100
+++ b/config.mk.in	2018-11-22 13:13:10.260123268 +0100
@@ -128,7 +128,7 @@ INSTALL = $(SRCDIR)/buildtools/install.s
 
 # STRIPFLAG is the option you pass to the above install program to make it
 # strip unnecessary information out of binaries.
-STRIPFLAG = -s
+STRIPFLAG =
 # If you don't want to strip the binaries, just leave it null:
 #STRIPFLAG = 
 
@@ -482,12 +482,12 @@ JBIGLIB = $(INTERNAL_JBIGLIB)
 JBIGHDR_DIR = $(INTERNAL_JBIGHDR_DIR)
 
 # The Jasper JPEG-2000 image compression library (aka JasPer):
-JASPERLIB = $(INTERNAL_JASPERLIB)
-JASPERHDR_DIR = $(INTERNAL_JASPERHDR_DIR)
+JASPERLIB = "" 
+JASPERHDR_DIR = "/usr/include/jasper"
 # JASPERDEPLIBS is the libraries (-l options or file names) on which
 # The Jasper library depends -- i.e. what you have to link into any
 # executable that links in the Jasper library.
-JASPERDEPLIBS =
+JASPERDEPLIBS = -ljasper
 #JASPERDEPLIBS = -ljpeg
 
 # And the Utah Raster Toolkit (aka URT aka RLE) library:
diff -urNp a/converter/other/jbig/Makefile b/converter/other/jbig/Makefile
--- a/converter/other/jbig/Makefile	2018-11-21 12:46:22.075789920 +0100
+++ b/converter/other/jbig/Makefile	2018-11-22 13:13:40.837969056 +0100
@@ -11,8 +11,9 @@ include $(BUILDDIR)/config.mk
 
 # INTERNAL_JBIGLIB must be relative to the current directory, because it
 # may end up in MERGE_OBJECTS, which must be relative.
-INTERNAL_JBIGLIB = libjbig/libjbig.a
-INTERNAL_JBIGHDR_DIR = $(SRCDIR)/$(SUBDIR)/libjbig/include
+INTERNAL_JBIGLIB = 
+INTERNAL_JBIGHDR_DIR = /usr/include
+#INTERNAL_JBIGHDR_DIR = $(SRCDIR)/$(SUBDIR)/libjbig/include
 
 EXTERN_INCLUDES =
 ifneq ($(JBIGHDR_DIR),NONE)
@@ -35,7 +36,6 @@ SCRIPTS =
 
 ifeq ($(JBIGLIB),$(INTERNAL_JBIGLIB))
   JBIGLIB_DEP = $(JBIGLIB)
-  SUBDIRS += libjbig
 else
   # It's not our internal version; user's on his own to make sure it's built
 endif
@@ -49,10 +49,3 @@ include $(SRCDIR)/common.mk
 
 $(BINARIES): %: %.o $(JBIGLIB_DEP) $(LIBOPT)
 $(BINARIES): LDFLAGS_TARGET = $(shell $(LIBOPT) $(JBIGLIB))
-
-$(INTERNAL_JBIGLIB): $(BUILDDIR)/$(SUBDIR)/libjbig FORCE
-	$(MAKE) -f $(SRCDIR)/$(SUBDIR)/libjbig/Makefile \
-	  -C $(dir $@) $(notdir $@)
-
-.PHONY: FORCE
-FORCE:
