diff -ru coredns-original/Makefile coredns/Makefile
--- coredns-original/Makefile	2020-01-28 11:11:57.000000000 -0800
+++ coredns/Makefile	2021-01-25 13:06:46.445245617 -0800
@@ -1,9 +1,9 @@
 # Makefile for building CoreDNS
-GITCOMMIT:=$(shell git describe --dirty --always)
+GITCOMMIT?=$(shell git describe --dirty --always)
 BINARY:=coredns
 SYSTEM:=
 CHECKS:=check
-BUILDOPTS:=-v
+BUILDOPTS?=-v
 GOPATH?=$(HOME)/go
 MAKEPWD:=$(dir $(realpath $(firstword $(MAKEFILE_LIST))))
 CGO_ENABLED?=0
