diff -ru coredns-1.6.5-original/Makefile coredns-1.6.5/Makefile
--- coredns-1.6.5-original/Makefile	2019-11-05 05:46:54.000000000 -0800
+++ coredns-1.6.5/Makefile	2021-01-25 10:26:44.337497154 -0800
@@ -1,9 +1,9 @@
 # Makefile for building CoreDNS
-GITCOMMIT:=$(shell git describe --dirty --always)
+GITCOMMIT?=$(shell git describe --dirty --always)
 BINARY:=coredns
 SYSTEM:=
 CHECKS:=check
-BUILDOPTS:=-v
+BUILDOPTS?=-v
 GOPATH?=$(HOME)/go
 PRESUBMIT:=core coremain plugin test request
 MAKEPWD:=$(dir $(realpath $(firstword $(MAKEFILE_LIST))))
