diff -ru ntp-4.2.8p15-orig/libntp/work_thread.c ntp-4.2.8p15/libntp/work_thread.c
--- ntp-4.2.8p15-orig/libntp/work_thread.c	2018-08-14 04:51:06.000000000 -0700
+++ ntp-4.2.8p15/libntp/work_thread.c	2021-11-11 12:11:57.941827886 -0800
@@ -41,7 +41,8 @@
 #ifndef THREAD_MINSTACKSIZE
 # define THREAD_MINSTACKSIZE	(64U * 1024)
 #endif
-#ifndef __sun
+// with gcc 11.x / glibc 2.34 PTHREAD_STACK_MIN is no longer a pre-processor directive
+#if defined(__sun) && !defined(__GLIBC__)
 #if defined(PTHREAD_STACK_MIN) && THREAD_MINSTACKSIZE < PTHREAD_STACK_MIN
 # undef THREAD_MINSTACKSIZE
 # define THREAD_MINSTACKSIZE PTHREAD_STACK_MIN
 