diff -ruN a/setup.py b/setup.py
--- a/setup.py	2024-07-02 18:44:08.000000000 -0700
+++ b/setup.py	2024-07-15 17:14:43.708920246 -0700
@@ -184,6 +184,8 @@
     USR_LIB_EXEC = "usr/local/lib"
 elif os.path.isfile("/etc/redhat-release"):
     USR_LIB_EXEC = "usr/libexec"
+elif os.path.isfile("/etc/azurelinux-release"):
+    USR_LIB_EXEC = "usr/libexec"
 elif os.path.isfile("/etc/system-release-cpe"):
     with open("/etc/system-release-cpe") as f:
         cpe_data = f.read().rstrip().split(":")
