diff --git a/cloudinit/sources/DataSourceAzure.py b/cloudinit/sources/DataSourceAzure.py
index 2f3390c3..3592ec53 100755
--- a/cloudinit/sources/DataSourceAzure.py
+++ b/cloudinit/sources/DataSourceAzure.py
@@ -51,7 +51,7 @@ LOG = logging.getLogger(__name__)
 
 DS_NAME = 'Azure'
 DEFAULT_METADATA = {"instance-id": "iid-AZURE-NODE"}
-AGENT_START = ['service', 'walinuxagent', 'start']
+AGENT_START = ['systemctl', 'start', 'waagent']
 AGENT_START_BUILTIN = "__builtin__"
 BOUNCE_COMMAND_IFUP = [
     'sh', '-xc',
diff --git a/cloudinit/sources/helpers/azure.py b/cloudinit/sources/helpers/azure.py
index ad476076..c49d5c1b 100755
--- a/cloudinit/sources/helpers/azure.py
+++ b/cloudinit/sources/helpers/azure.py
@@ -9,6 +9,7 @@ import struct
 import time
 import textwrap
 import zlib
+import configobj
 from errno import ENOENT
 
 from cloudinit.settings import CFG_BUILTIN
@@ -27,6 +28,7 @@ from cloudinit import distros
 from cloudinit.reporting import events
 from cloudinit.net.dhcp import EphemeralDHCPv4
 from datetime import datetime
+from io import StringIO
 
 LOG = logging.getLogger(__name__)
 
@@ -38,6 +40,8 @@ BOOT_EVENT_TYPE = 'boot-telemetry'
 SYSTEMINFO_EVENT_TYPE = 'system-info'
 DIAGNOSTIC_EVENT_TYPE = 'diagnostic'
 COMPRESSED_EVENT_TYPE = 'compressed'
+NETWORKD_LEASES_DIR = '/run/systemd/netif/leases'
+
 # Maximum number of bytes of the cloud-init.log file that can be dumped to KVP
 # at once. This number is based on the analysis done on a large sample of
 # cloud-init.log files where the P95 of the file sizes was 537KB and the time
@@ -783,6 +787,32 @@ class WALinuxAgentShim:
                 ) from e
         return dhcp_options
 
+    @staticmethod
+    def networkd_parse_lease(content):
+        """Parse a systemd lease file content as in /run/systemd/netif/leases/
+        Parse this (almost) ini style file even though it says:
+          # This is private data. Do not parse.
+        Simply return a dictionary of key/values."""
+
+        return dict(configobj.ConfigObj(StringIO(content), list_values=False))
+
+    @staticmethod
+    def networkd_load_leases(leases_d=None):
+        """Return a dictionary of dictionaries representing each lease
+        found in lease_d.i
+        The top level key will be the filename, which is typically the ifindex."""
+
+        if leases_d is None:
+            leases_d = NETWORKD_LEASES_DIR
+
+        ret = {}
+        if not os.path.isdir(leases_d):
+            return ret
+        for lfile in os.listdir(leases_d):
+            ret[lfile] = WALinuxAgentShim.networkd_parse_lease(
+                util.load_file(os.path.join(leases_d, lfile)))
+        return ret
+
     @staticmethod
     @azure_ds_telemetry_reporter
     def _get_value_from_dhcpoptions(dhcp_options):
@@ -834,6 +864,7 @@ class WALinuxAgentShim:
                 logger_func=LOG.debug)
             dhcp_options = WALinuxAgentShim._load_dhclient_json()
             value = WALinuxAgentShim._get_value_from_dhcpoptions(dhcp_options)
+            LOG.debug('networkd value from lease %s', value)
         if value is None:
             # Fallback and check the leases file if unsuccessful
             report_diagnostic_event(
