diff --git a/tools/ds-identify b/tools/ds-identify
index 73e27c71..2dcb3f65 100755
--- a/tools/ds-identify
+++ b/tools/ds-identify
@@ -94,7 +94,7 @@ DI_MAIN=${DI_MAIN:-main}
 
 DI_BLKID_EXPORT_OUT=""
 DI_GEOM_LABEL_STATUS_OUT=""
-DI_DEFAULT_POLICY="search,found=all,maybe=all,notfound=${DI_DISABLED}"
+DI_DEFAULT_POLICY="search,found=all,maybe=all,notfound=${DI_ENABLED}"
 DI_DEFAULT_POLICY_NO_DMI="search,found=all,maybe=all,notfound=${DI_ENABLED}"
 DI_DMI_CHASSIS_ASSET_TAG=""
 DI_DMI_PRODUCT_NAME=""
