diff -ruN a/cloudinit/net/__init__.py b/cloudinit/net/__init__.py
--- a/cloudinit/net/__init__.py	2024-09-04 12:01:58.000000000 -0700
+++ b/cloudinit/net/__init__.py	2024-12-13 17:01:54.770779599 -0800
@@ -11,6 +11,7 @@
 import logging
 import os
 import re
+import time
 from typing import Any, Callable, Dict, List, Optional, Tuple
 from urllib.parse import urlparse
 
@@ -844,13 +845,20 @@
         LOG.debug("Renamed %s with ops %s", renames, ops + ups)
 
         for op, mac, new_name, params in ops + ups:
+            result_before = subp.subp(["ip", "link", "show"])
             try:
                 opmap.get(op)(*params)
+                time.sleep(2)
             except Exception as e:
+                result_after = subp.subp(["ip", "link", "show"])
                 errors.append(
                     "[unknown] Error performing %s%s for %s, %s: %s"
                     % (op, params, mac, new_name, e)
                 )
+                errors.append(
+                    "ip link show before is: %s, after is: %s"
+                    % (result_before, result_after)
+                )
 
     if len(errors):
         raise RuntimeError("\n".join(errors))
diff -ruN a/cloudinit/net/netops/iproute2.py b/cloudinit/net/netops/iproute2.py
--- a/cloudinit/net/netops/iproute2.py	2024-09-04 12:01:58.000000000 -0700
+++ b/cloudinit/net/netops/iproute2.py	2024-12-12 15:12:39.452471476 -0800
@@ -2,7 +2,7 @@
 
 from cloudinit import subp
 from cloudinit.net.netops import NetOps
-
+import time
 
 class Iproute2(NetOps):
     @staticmethod
@@ -30,6 +30,7 @@
     @staticmethod
     def link_rename(current_name: str, new_name: str):
         subp.subp(["ip", "link", "set", current_name, "name", new_name])
+        time.sleep(1)
 
     @staticmethod
     def add_route(

