diff -ruN a/cloudinit/distros/photon.py b/cloudinit/distros/photon.py
--- a/cloudinit/distros/photon.py	2023-12-14 09:17:35.000000000 -0800
+++ b/cloudinit/distros/photon.py	2024-03-13 17:59:45.237524777 -0700
@@ -151,7 +151,7 @@
 
         ret, _out, err = self.exec_cmd(cmd)
         if ret:
-            LOG.error("Error while installing packages: %s", err)
+            LOG.warning("Error while installing packages: %s", err)
 
     def update_package_sources(self):
         self._runner.run(
