diff --git a/systemd/cloud-init.service.tmpl b/systemd/cloud-init.service.tmpl
index f140344d..c58e9302 100644
--- a/systemd/cloud-init.service.tmpl
+++ b/systemd/cloud-init.service.tmpl
@@ -1,7 +1,6 @@
 ## template:jinja
 [Unit]
 Description=Initial cloud-init job (metadata service crawler)
-DefaultDependencies=no
 Wants=cloud-init-local.service
 Wants=sshd-keygen.service
 Wants=sshd.service
@@ -21,17 +20,9 @@ After=wicked.service
 After=dbus.service
 {% endif %}
 Before=network-online.target
+After=systemd-networkd-wait-online.service
 Before=sshd-keygen.service
 Before=sshd.service
-{% if variant in ["ubuntu", "unknown", "debian"] %}
-Before=sysinit.target
-Before=shutdown.target
-Conflicts=shutdown.target
-{% endif %}
-{% if variant in ["suse"] %}
-Before=shutdown.target
-Conflicts=shutdown.target
-{% endif %}
 Before=systemd-user-sessions.service
 
 [Service]
