diff -ruN a/tests/unittests/test_net.py b/cloud-init-23.2/tests/unittests/test_net.py
--- a/tests/unittests/test_net.py	2023-05-23 14:14:35.000000000 -0700
+++ b/tests/unittests/test_net.py	2023-08-11 14:18:28.278597831 -0700
@@ -7880,6 +7880,7 @@
             "eth1": "aa:aa:aa:aa:aa:01",
             "tun0": None,
         },
+        "masters": {},
         "drivers": {
             "enp0s1": "virtio_net",
             "enp0s2": "e1000",
@@ -7907,6 +7908,9 @@
     def _se_get_interface_mac(self, name):
         return self.data["macs"][name]
 
+    def _se_get_master(self, name):
+        return self.data["masters"].get(name)
+
     def _se_is_bridge(self, name):
         return name in self.data["bridges"]
 
@@ -7928,6 +7932,7 @@
         mocks = (
             "get_devicelist",
             "get_interface_mac",
+            "get_master",
             "is_bridge",
             "interface_has_own_mac",
             "is_vlan",
