--- elliptic-curve-0.13.8/Cargo.toml	1970-01-01T00:00:01+00:00
+++ elliptic-curve-0.13.8/Cargo.toml	2025-12-03T21:47:58.170589+00:00
@@ -87,7 +87,7 @@
 default-features = false
 
 [dependencies.hex-literal]
-version = "0.4"
+version = ">=0.4, <2.0"
 optional = true
 
 [dependencies.hkdf]
@@ -143,7 +143,7 @@
 default-features = false
 
 [dev-dependencies.hex-literal]
-version = "0.4"
+version = ">=0.4, <2.0"
 
 [dev-dependencies.sha2]
 version = "0.10"
