--- liboping-1.10.0/src/Makefile.am	2017-05-11 00:32:27.000000000 -0700
+++ liboping-1.10.0/src/Makefile.am.new	2018-07-24 14:29:43.400661398 -0700
@@ -3,7 +3,7 @@
 SUBDIRS = mans
 
 if COMPILER_IS_GCC
-AM_CFLAGS = -Wall -Werror
+AM_CFLAGS = -Wall
 endif
 
 include_HEADERS = oping.h
--- liboping-1.10.0/src/Makefile.in	2018-07-24 14:30:52.192553951 -0700
+++ liboping-1.10.0/src/Makefile.in.new	2018-07-24 14:30:57.281546003 -0700
@@ -388,7 +388,7 @@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign no-dependencies
 SUBDIRS = mans
-@COMPILER_IS_GCC_TRUE@AM_CFLAGS = -Wall -Werror
+@COMPILER_IS_GCC_TRUE@AM_CFLAGS = -Wall
 include_HEADERS = oping.h
 lib_LTLIBRARIES = liboping.la
 
