diff -ruN jimtcl-0.81/auto.def jimtcl-lib64/auto.def
--- jimtcl-0.81/auto.def	2021-11-27 15:06:54.000000000 -0800
+++ jimtcl-lib64/auto.def	2022-01-07 12:05:53.769713328 -0800
@@ -271,7 +271,7 @@
 cc-check-lfs
 cc-check-functions fseeko ftello
 
-define TCL_LIBRARY [get-define prefix]/lib/jim
+define TCL_LIBRARY [get-define libdir]/jim
 
 lassign [split [get-define host] -] host_cpu host_vendor host_os
 # Scrub revision from the host_os
@@ -445,7 +445,7 @@
 }
 define VERSION [format %.2f [expr {[get-define JIM_VERSION] / 100.0}]]
 define LIBSOEXT [format [get-define SH_SOEXTVER] [get-define VERSION]]
-if {[get-define libdir] ni {/lib /usr/lib}} {
+if {[get-define libdir] ni {/lib /usr/lib /usr/lib64}} {
     define SH_LINKRPATH_FLAGS [format [get-define SH_LINKRPATH] [get-define libdir]]
 } else {
     define SH_LINKRPATH_FLAGS ""
diff -ruN jimtcl-0.81/build-jim-ext.in jimtcl-lib64/build-jim-ext.in
--- jimtcl-0.81/build-jim-ext.in	2021-11-27 15:06:54.000000000 -0800
+++ jimtcl-lib64/build-jim-ext.in	2022-01-07 12:07:18.397171096 -0800
@@ -41,7 +41,7 @@
 			set cross 1
 		}
 		--install {
-			# Install to $DESTDIR/@prefix@/lib/jim
+			# Install to $DESTDIR/@libdir@/jim
 			set install 1
 		}
 		--static {
@@ -225,7 +225,7 @@
 			load ./$target
 		}
 		if {$install} {
-			set dest [env DESTDIR ""]@prefix@/lib/jim
+			set dest [env DESTDIR ""]@libdir@/jim
 			puts "Install: $target => $dest"
 			file mkdir $dest
 			file copy $target $dest/$target
