--- event-listener-5.4.2/Cargo.toml	2006-07-24T01:21:28+00:00
+++ event-listener-5.4.2/Cargo.toml	2026-08-01T21:19:27.879502+00:00
@@ -45,7 +45,6 @@
 default = ["std"]
 loom = [
     "parking?/loom",
-    "dep:loom",
 ]
 portable-atomic = [
     "portable-atomic-util",
@@ -113,18 +112,12 @@
 [dev-dependencies.waker-fn]
 version = "1"
 
-[target."cfg(loom)".dependencies.loom]
-version = "0.7"
-optional = true
-
 [target.'cfg(not(target_family = "wasm"))'.dependencies.parking]
 version = "2.0.0"
 optional = true
-
-[target.'cfg(target_family = "wasm")'.dev-dependencies.wasm-bindgen-test]
-version = "0.3"
 
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
 check-cfg = ["cfg(loom)"]
+
