--- event-listener-5.4.2/Cargo.toml	2006-07-24T01:21:28+00:00
+++ event-listener-5.4.2/Cargo.toml	2026-08-01T21:19:57.373780+00:00
@@ -43,9 +43,6 @@
 
 [features]
 default = ["std"]
-loom = [
-    "parking?/loom",
-]
 portable-atomic = [
     "portable-atomic-util",
     "portable_atomic_crate",
@@ -62,17 +59,8 @@
 path = "examples/mutex.rs"
 
 [[test]]
-name = "loom"
-path = "tests/loom.rs"
-
-[[test]]
 name = "notify"
 path = "tests/notify.rs"
-
-[[bench]]
-name = "bench"
-path = "benches/bench.rs"
-harness = false
 
 [dependencies.critical-section]
 version = "1.2.0"
@@ -93,11 +81,6 @@
 optional = true
 default-features = false
 package = "portable-atomic"
-
-[dev-dependencies.criterion]
-version = "0.8"
-features = ["cargo_bench_support"]
-default-features = false
 
 [dev-dependencies.critical-section]
 version = "1.2.0"
