diff -ruN a/fluent-plugin-prometheus.gemspec b/fluent-plugin-prometheus.gemspec
--- a/fluent-plugin-prometheus.gemspec	2022-07-07 16:39:48.391818041 -0700
+++ b/fluent-plugin-prometheus.gemspec	2022-07-07 16:40:10.208077101 -0700
@@ -8,7 +8,7 @@
   spec.homepage      = "https://github.com/fluent/fluent-plugin-prometheus"
   spec.license       = "Apache-2.0"
 
-  spec.files         = `git ls-files -z`.split("\x0")
+  spec.files         = Dir['**/*']
   spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
