diff -rupr a/Lib/platform.py b/Lib/platform.py
--- a/Lib/platform.py	2015-12-05 11:47:01.000000000 -0800
+++ b/Lib/platform.py	2017-01-20 16:39:55.628824665 -0800
@@ -261,7 +261,7 @@ _release_version = re.compile(r'([^0-9]+
 _supported_dists = (
     'SuSE', 'debian', 'fedora', 'redhat', 'centos',
     'mandrake', 'mandriva', 'rocks', 'slackware', 'yellowdog', 'gentoo',
-    'UnitedLinux', 'turbolinux')
+    'UnitedLinux', 'turbolinux', 'mariner')
 
 def _parse_release_file(firstline):
 
