diff -ruN a/parallel.gemspec b/parallel.gemspec
--- a/parallel.gemspec	2022-07-21 10:49:46.558106492 -0700
+++ b/parallel.gemspec	2022-07-21 11:06:50.875142979 -0700
@@ -13,7 +13,10 @@
     "source_code_uri"   => "https://github.com/grosser/#{name}/tree/v#{s.version}",
     "wiki_uri"          => "https://github.com/grosser/#{name}/wiki",
   }
-  s.files = `git ls-files lib MIT-LICENSE.txt`.split("\n")
+  s.files = Dir['**/*']
+  s.test_files    = s.files.grep(%r{^(test|spec|features)/})
+  s.executables   = s.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
+  s.require_paths = ["lib"]
   s.license = "MIT"
   s.required_ruby_version = '>= 2.4'
 end
