Index: wvdial-1.61/Makefile.in
===================================================================
--- wvdial-1.61.orig/Makefile.in	2011-01-10 21:49:50.000000000 +0300
+++ wvdial-1.61/Makefile.in	2011-01-10 21:50:51.000000000 +0300
@@ -34,7 +34,7 @@
 	wvdialmon.o
 
 wvdial wvdialconf papchaptest pppmon: \
-  LDFLAGS+=-luniconf -lwvstreams -lwvutils -lwvbase
+  LIBS+=-luniconf -lwvstreams -lwvutils -lwvbase
 
 wvdial wvdialconf papchaptest pppmon: wvdial.a
 
