Index: wvdial-1.61/wvrules.mk
===================================================================
--- wvdial-1.61.orig/wvrules.mk
+++ wvdial-1.61/wvrules.mk
@@ -71,7 +71,7 @@ runtests:
 
 INCFLAGS=$(addprefix -I,$(WVSTREAMS_INC) $(XPATH))
 CPPFLAGS+=$(INCFLAGS) \
-	-D_BSD_SOURCE -D_GNU_SOURCE $(OSDEFINE) \
+	-D_DEFAULT_SOURCE -D_GNU_SOURCE $(OSDEFINE) \
 	-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 \
 	-DUNSTABLE
 
