diff -up openwsman-2.6.5/src/lib/wsman-curl-client-transport.c.orig openwsman-2.6.5/src/lib/wsman-curl-client-transport.c
--- openwsman-2.6.5/src/lib/wsman-curl-client-transport.c.orig	2018-11-14 13:53:27.442138557 +0100
+++ openwsman-2.6.5/src/lib/wsman-curl-client-transport.c	2018-11-14 14:11:28.508714204 +0100
@@ -186,16 +186,23 @@ convert_to_last_error(CURLcode r)
 		return WS_LASTERR_SSL_CONNECT_ERROR;
         case CURLE_BAD_FUNCTION_ARGUMENT:
                 return WS_LASTERR_CURL_BAD_FUNCTION_ARG;
+#if LIBCURL_VERSION_NUM < 0x073E00
 	case CURLE_SSL_PEER_CERTIFICATE:
 		return WS_LASTERR_SSL_PEER_CERTIFICATE;
+#endif
 	case CURLE_SSL_ENGINE_NOTFOUND:
 		return WS_LASTERR_SSL_ENGINE_NOTFOUND;
 	case CURLE_SSL_ENGINE_SETFAILED:
 		return WS_LASTERR_SSL_ENGINE_SETFAILED;
 	case CURLE_SSL_CERTPROBLEM:
 		return WS_LASTERR_SSL_CERTPROBLEM;
+#if LIBCURL_VERSION_NUM < 0x073E00
 	case CURLE_SSL_CACERT:
 		return WS_LASTERR_SSL_CACERT;
+#else
+	case CURLE_PEER_FAILED_VERIFICATION:
+		return WS_LASTERR_SSL_PEER_CERTIFICATE;
+#endif
 #if LIBCURL_VERSION_NUM > 0x70C01
 	case CURLE_SSL_ENGINE_INITFAILED:
 		return WS_LASTERR_SSL_ENGINE_INITFAILED;
