--- ws-jaxme-0.5.2/src/js/org/apache/ws/jaxme/js/pattern/CompiledClassReflector.java	2005-07-28 21:47:58.000000000 +0200
+++ ws-jaxme-0.5.2/src/js/org/apache/ws/jaxme/js/pattern/CompiledClassReflector.java	2017-10-06 17:25:25.213569086 +0200
@@ -34,18 +34,7 @@
 	 * {@link JavaSource}.</p>
 	 */
 	protected JavaMethod getMethod(JavaSource pSource, Method pMethod) {
-		JavaMethod method = pSource.newJavaMethod(pMethod.getName(),
-				                                  JavaQNameImpl.getInstance(pMethod.getReturnType()),
-												  JavaSource.PUBLIC);
-		Class[] classes = pMethod.getParameterTypes();
-		for (int i = 0;  i < classes.length;  i++) {
-			method.addParam(classes[i], "arg" + i);
-		}
-		Class[] exceptions = pMethod.getExceptionTypes();
-		for (int i = 0;  i < exceptions.length;  i++) {
-			method.addThrows(exceptions[i]);
-		}
-		return method;
+		return pSource.newJavaMethod(pMethod);
 	}
 	
 	/** Returns the compiled class being used to gather information.
