`diff -uNr xmlstarlet-1.6.1-orig/configure.ac xmlstarlet-1.6.1/configure.ac
--- xmlstarlet-1.6.1-orig/configure.ac	2013-12-22 17:00:28.000000000 -0500
+++ xmlstarlet-1.6.1/configure.ac	2014-12-08 17:12:47.836499274 -0500
@@ -1,5 +1,5 @@
 AC_PREREQ([2.62])
-AC_INIT([XMLStarlet], [m4_esyscmd_s([git describe --tags --dirty])],
+AC_INIT([XMLStarlet], [m4_esyscmd_s([if test -d .git ; then git describe --tags --dirty ; fi])],
         [http://sourceforge.net/projects/xmlstar/support],
         [],
         [http://xmlstar.sourceforge.net/])
