--- configure.ac.gettext_version	2016-03-02 15:10:26.000000000 +0100
+++ configure.ac	2017-02-15 13:39:12.295019148 +0100
@@ -59,7 +60,6 @@ JH_CHECK_XML_CATALOG([http://docbook.sou
 AM_CONDITIONAL(ENABLE_REGENERATE_MAN, test x$enable_man != xno)
 
 dnl internationalization macros
-AM_GNU_GETTEXT_VERSION
 AM_GNU_GETTEXT([external])
 
 AC_OUTPUT(Makefile lib/Makefile src/Makefile man/Makefile po/Makefile.in)
