--- anstyle-query-1.1.5/Cargo.toml	1970-01-01T00:00:01+00:00
+++ anstyle-query-1.1.5/Cargo.toml	2025-11-15T22:09:09.944475+00:00
@@ -93,13 +93,6 @@
 name = "query"
 path = "examples/query.rs"
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = ">=0.60.2, <0.62"
-features = [
-    "Win32_System_Console",
-    "Win32_Foundation",
-]
-
 [lints.clippy]
 bool_assert_comparison = "allow"
 branches_sharing_code = "allow"
@@ -171,3 +164,4 @@
 [lints.rust.rust_2018_idioms]
 level = "warn"
 priority = -1
+
