diff -up console-setup/Keyboard/ckbcomp.orig console-setup/Keyboard/ckbcomp
--- console-setup/Keyboard/ckbcomp.orig	2012-06-26 12:04:16.612309745 +0200
+++ console-setup/Keyboard/ckbcomp	2012-06-26 12:21:06.448385970 +0200
@@ -15,7 +15,7 @@
 
 #     If you have not received a copy of the GNU General Public License
 #     along with this program, write to the Free Software Foundation, Inc.,
-#     59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#     51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 use warnings 'all';
 use strict;
