--- configure.in	2009-12-06 21:43:11.000000000 -0500
+++ configure.in.new	2009-12-17 15:13:58.794592478 -0500
@@ -31,26 +31,15 @@
 MYPCFILES="tokyocabinet.pc"
 
 # Building flags
-MYCFLAGS="-std=c99 -Wall -fPIC -fsigned-char -O2"
-MYCPPFLAGS="-I. -I\$(INCLUDEDIR) -I$HOME/include -I/usr/local/include"
+MYCFLAGS="-std=c99 -Wall -fPIC -fsigned-char"
+MYCPPFLAGS="-I. -I\$(INCLUDEDIR)"
 MYCPPFLAGS="$MYCPPFLAGS -DNDEBUG -D_GNU_SOURCE=1 -D_REENTRANT -D__EXTENSIONS__"
-MYLDFLAGS="-L. -L\$(LIBDIR) -L$HOME/lib -L/usr/local/lib"
+MYLDFLAGS="-L. -L\$(LIBDIR)"
 MYCMDLDFLAGS=""
 MYRUNPATH="\$(LIBDIR)"
 MYLDLIBPATHENV="LD_LIBRARY_PATH"
 MYPOSTCMD="true"
 
-# Building paths
-PATH="$PATH:$HOME/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin"
-PATH="$PATH:/opt/SUNWspro/bin:/usr/ccs/bin:/usr/xpg4/bin:/usr/xpg6/bin:/usr/ucb"
-CPATH="$HOME/include:/usr/local/include:$CPATH"
-LIBRARY_PATH="$HOME/lib:/usr/local/lib:$LIBRARY_PATH"
-LD_LIBRARY_PATH="$HOME/lib:/usr/local/lib:$LD_LIBRARY_PATH"
-PKG_CONFIG_PATH="$HOME/lib/pkgconfig:/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH"
-export PATH CPATH LIBRARY_PATH LD_LIBRARY_PATH PKG_CONFIG_PATH
-
-
-
 #================================================================
 # Options
 #================================================================
--- Makefile.in	2009-10-13 08:33:20.000000000 -0400
+++ Makefile.in.new	2009-12-17 15:15:07.257604314 -0500
@@ -54,8 +54,8 @@
 LDFLAGS = @MYLDFLAGS@
 CMDLDFLAGS = @MYCMDLDFLAGS@
 LIBS = @LIBS@
-LDENV = LD_RUN_PATH=/lib:/usr/lib:$(LIBDIR):$(HOME)/lib:/usr/local/lib:@MYRUNPATH@:.
-RUNENV = @MYLDLIBPATHENV@=.:/lib:/usr/lib:$(LIBDIR):$(HOME)/lib:/usr/local/lib:@MYRUNPATH@
+LDENV = 
+RUNENV = @MYLDLIBPATHENV@=.:$(LIBDIR)
 POSTCMD = @MYPOSTCMD@
 
 
--- tokyocabinet.pc.in	2008-02-18 02:25:18.000000000 -0500
+++ tokyocabinet.pc.in.new	2009-12-17 15:15:29.626591709 -0500
@@ -10,5 +10,5 @@
 Name: Tokyo Cabinet
 Description: a modern implementation of DBM
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ltokyocabinet @LIBS@
+Libs: -L${libdir} -ltokyocabinet
 Cflags: -I${includedir}
