diff -ruN a/fluent-plugin-elasticsearch.gemspec b/fluent-plugin-elasticsearch.gemspec
--- a/fluent-plugin-elasticsearch.gemspec	2022-07-07 16:33:59.519676611 -0700
+++ b/fluent-plugin-elasticsearch.gemspec	2022-07-07 16:34:32.980073686 -0700
@@ -11,7 +11,7 @@
   s.homepage      = 'https://github.com/uken/fluent-plugin-elasticsearch'
   s.license       = 'Apache-2.0'
 
-  s.files         = `git ls-files`.split($/)
+  s.files         = Dir['**/*']
   s.executables   = s.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   s.test_files    = s.files.grep(%r{^(test|spec|features)/})
   s.require_paths = ['lib']
