--- tokio-1.53.1/Cargo.toml	2006-07-24T01:21:28+00:00
+++ tokio-1.53.1/Cargo.toml	2026-07-21T12:04:06.043586+00:00
@@ -84,11 +84,6 @@
     "time",
 ]
 io-std = []
-io-uring = [
-    "libc",
-    "mio/os-poll",
-    "mio/os-ext",
-]
 io-util = ["bytes"]
 macros = ["tokio-macros"]
 net = [
@@ -108,7 +103,6 @@
 ]
 rt = []
 rt-multi-thread = ["rt"]
-schedule-latency = []
 signal = [
     "libc",
     "mio/os-poll",
@@ -117,7 +111,6 @@
     "signal-hook-registry",
 ]
 sync = []
-taskdump = []
 test-util = [
     "rt",
     "sync",
@@ -278,10 +271,6 @@
 path = "tests/io_copy.rs"
 
 [[test]]
-name = "io_copy_bidirectional"
-path = "tests/io_copy_bidirectional.rs"
-
-[[test]]
 name = "io_driver"
 path = "tests/io_driver.rs"
 
@@ -324,22 +313,6 @@
 [[test]]
 name = "io_read_exact"
 path = "tests/io_read_exact.rs"
-
-[[test]]
-name = "io_read_line"
-path = "tests/io_read_line.rs"
-
-[[test]]
-name = "io_read_to_end"
-path = "tests/io_read_to_end.rs"
-
-[[test]]
-name = "io_read_to_string"
-path = "tests/io_read_to_string.rs"
-
-[[test]]
-name = "io_read_until"
-path = "tests/io_read_until.rs"
 
 [[test]]
 name = "io_repeat"
@@ -844,14 +817,11 @@
 version = "0.3.0"
 features = ["async-await"]
 
-[dev-dependencies.futures-concurrency]
-version = "7.6.3"
-
 [dev-dependencies.futures-test]
 version = "0.3.31"
 
 [dev-dependencies.mockall]
-version = "0.13.0"
+version = ">=0.11,<0.14"
 
 [dev-dependencies.tokio-stream]
 version = "0.1"
