From 21ba8928b5dbcf69c6dbd7ff81c755268113ba40 Mon Sep 17 00:00:00 2001
From: Olivia Crain <oliviacrain@microsoft.com>
Date: Tue, 14 Dec 2021 11:50:27 -0800
Subject: [PATCH 2/3] xli

---
 lynx.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lynx.cfg b/lynx.cfg
index a56cd51..aeb75de 100644
--- a/lynx.cfg
+++ b/lynx.cfg
@@ -2310,7 +2310,7 @@ MINIMAL_COMMENTS:TRUE
 # XLOADIMAGE_COMMAND definition.
 # If you use xli, you may want to add the -quiet flag.
 #
-#XLOADIMAGE_COMMAND:xli %s &
+XLOADIMAGE_COMMAND:xli %s &
 
 .h2 VIEWER
 # MIME types may be assigned to external viewers using
-- 
2.25.1

