--- dyninst-10.1.0/testsuite-10.1.0/src/proccontrol/pcontrol_mutatee_tools.c.orig	2019-05-31 05:20:00.492379706 +0200
+++ dyninst-10.1.0/testsuite-10.1.0/src/proccontrol/pcontrol_mutatee_tools.c	2019-05-31 05:10:11.354826668 +0200
@@ -62,9 +62,9 @@
 #if !defined(os_windows_test)
 #include <poll.h>
 
-static unsigned int gettid(){
-    return (unsigned int)pthread_self();
-}
+// static unsigned int gettid(){
+//     return (unsigned int)pthread_self();
+// }
 #endif
 
 thread_t threads[MAX_POSSIBLE_THREADS];
