--- toml_edit-0.23.10+spec-1.0.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ toml_edit-0.23.10+spec-1.0.0/Cargo.toml	2026-02-23T17:26:35.118379+00:00
@@ -13,7 +13,7 @@
 edition = "2021"
 rust-version = "1.76"
 name = "toml_edit"
-version = "0.23.10+spec-1.0.0"
+version = "0.23.10"
 build = false
 include = [
     "build.rs",
@@ -259,3 +259,4 @@
 [lints.rust.rust_2018_idioms]
 level = "warn"
 priority = -1
+
