--- binutils.orig/ld/testsuite/ld-elf/linux-x86.exp	2024-01-24 11:52:35.288014542 +0000
+++ binutils-2.41/ld/testsuite/ld-elf/linux-x86.exp	2024-01-24 17:31:39.356167357 +0000
@@ -225,6 +225,10 @@ if { [check_ifunc_attribute_available] }
 # Old gcc silently ignores __attribute__ ((aligned())) with too big alignment.
 proc compiler_honours_aligned { } {
     global CC_FOR_TARGET READELF srcdir subdir
+
+    # Temporary fix for CentOS-10 kernel issue.  (RHEL-22466)
+    return 0 
+    
     ld_compile $CC_FOR_TARGET $srcdir/$subdir/p_align-1.c tmpdir/p_align-1.o
     set output [run_host_cmd "$READELF" "-SW tmpdir/p_align-1.o"]
     if { [regexp { [.]data *PROGBITS .* 8388608[\n]} $output] } {
