--- normpath-1.5.1/Cargo.toml	2006-07-24T01:21:28+00:00
+++ normpath-1.5.1/Cargo.toml	2026-05-09T13:12:56.011796+00:00
@@ -60,8 +60,6 @@
 
 [features]
 localization = [
-    "windows-sys/Win32_UI_Shell",
-    "windows-sys/Win32_UI_WindowsAndMessaging",
 ]
 
 [lib]
@@ -91,13 +89,3 @@
 [target."cfg(not(windows))".dev-dependencies.libc]
 version = "0.2"
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.61"
-features = ["Win32_Storage_FileSystem"]
-
-[target."cfg(windows)".dev-dependencies.normpath_macros]
-version = "=1.5.1"
-
-[target."cfg(windows)".dev-dependencies.windows-sys]
-version = "0.61"
-features = ["Win32_Foundation"]
