diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4ec527c..1969afc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -252,13 +252,13 @@ if(NOT WIN32)
     set(PKGCONFIG_LIBS "-L\${libdir} -l${QCA_LIB_NAME}")
   endif()

-  if(NOT BUILD_WITH_QT6)
+  # if(NOT BUILD_WITH_QT6)
     # qca2.pc uses absolute paths. So it must be there. Don't rellocate this.
     configure_file("qca2.pc.cmake" "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pkgconfig/${QCA_PC_NAME}" @ONLY)
     if(NOT DEVELOPER_MODE)
       install(FILES "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pkgconfig/${QCA_PC_NAME}" DESTINATION ${PKGCONFIG_INSTALL_PREFIX})
     endif()
-  endif()
+  # endif()
 endif()

 # strip CMAKE_INSTALL_PREFIX in all paths
