diff --git a/makedefs b/makedefs
index 186a3b9..d6a2eb8 100644
--- a/makedefs
+++ b/makedefs
@@ -682,7 +682,7 @@ EOF
 		: ${SHLIB_ENV="LD_LIBRARY_PATH=`pwd`/lib"}
 		: ${PLUGIN_LD="${CC-gcc} -shared"}
 		;;
-    Linux.[3456].*)
+    Linux.[34567].*)
 		SYSTYPE=LINUX$RELEASE_MAJOR
 		case "$CCARGS" in
 		 *-DNO_DB*) ;;
diff --git a/src/util/sys_defs.h b/src/util/sys_defs.h
index 3c1e93c..a6a2808 100644
--- a/src/util/sys_defs.h
+++ b/src/util/sys_defs.h
@@ -764,7 +764,7 @@ extern int initgroups(const char *, int);
   * LINUX.
   */
 #if defined(LINUX2) || defined(LINUX3) || defined(LINUX4) || defined(LINUX5) \
-	|| defined(LINUX6)
+	|| defined(LINUX6) || defined(LINUX7)
 #define SUPPORTED
 #define UINT32_TYPE	unsigned int
 #define UINT16_TYPE	unsigned short
