From 921d82e5db1e721515bc20b2c9e7a2a6f75d024d Mon Sep 17 00:00:00 2001
From: Marian Koncek <mkoncek@redhat.com>
Date: Mon, 11 Aug 2025 09:45:59 +0200
Subject: [PATCH] Remove usage of junit pioneer

---
 .../integration/CredentialPluginTest.java     | 77 -------------------
 1 file changed, 77 deletions(-)

diff --git a/src/test/java/org/mariadb/jdbc/integration/CredentialPluginTest.java b/src/test/java/org/mariadb/jdbc/integration/CredentialPluginTest.java
index 256326c..a031f1b 100644
--- a/src/test/java/org/mariadb/jdbc/integration/CredentialPluginTest.java
+++ b/src/test/java/org/mariadb/jdbc/integration/CredentialPluginTest.java
@@ -10,7 +10,6 @@ import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.Assumptions;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
-import org.junitpioneer.jupiter.SetEnvironmentVariable;
 
 public class CredentialPluginTest extends Common {
 
@@ -129,80 +128,4 @@ public class CredentialPluginTest extends Common {
         () -> createCon("&user=toti&credentialType=ENV&pwdKey=myPwdKey"),
         "Access denied");
   }
-
-  @Test
-  @SetEnvironmentVariable(key = "myPwdKey", value = "!Passw0rd3Works")
-  public void envsPwdTest() throws Exception {
-    Common.assertThrowsContains(
-        SQLException.class,
-        () -> createCon("&user=toto&credentialType=ENV&pwdKey=myPwdKey"),
-        "Access denied");
-
-    try (Connection conn = createCon("user=identityUser&credentialType=ENV&pwdKey=myPwdKey")) {
-      Statement stmt = conn.createStatement();
-      ResultSet rs = stmt.executeQuery("SELECT '5'");
-      assertTrue(rs.next());
-      assertEquals("5", rs.getString(1));
-    }
-  }
-
-  @Test
-  @SetEnvironmentVariable(key = "myPwdKey", value = "!Passw0rd3Works")
-  @SetEnvironmentVariable(key = "MARIADB_USER", value = "identityUser")
-  public void envsDefaultIdentityAndPwdTest() throws Exception {
-    try (Connection conn = createCon("credentialType=ENV&pwdKey=myPwdKey")) {
-      Statement stmt = conn.createStatement();
-      ResultSet rs = stmt.executeQuery("SELECT '5'");
-      assertTrue(rs.next());
-      assertEquals("5", rs.getString(1));
-    }
-  }
-
-  @Test
-  @SetEnvironmentVariable(key = "MARIADB_PWD", value = "!Passw0rd3Works")
-  @SetEnvironmentVariable(key = "MARIADB_USER", value = "identityUser")
-  public void envsIdentityDefaultPwdTest() throws Exception {
-
-    try (Connection conn = createCon("credentialType=ENV")) {
-      Statement stmt = conn.createStatement();
-      ResultSet rs = stmt.executeQuery("SELECT '5'");
-      assertTrue(rs.next());
-      assertEquals("5", rs.getString(1));
-    }
-  }
-
-  @Test
-  @SetEnvironmentVariable(key = "myPwdKey", value = "!Passw0rd3Works")
-  @SetEnvironmentVariable(key = "myUserKey", value = "identityUser")
-  public void envsIdentityAndPwdTest() throws Exception {
-
-    try (Connection conn = createCon("credentialType=ENV&userKey=myUserKey&pwdKey=myPwdKey")) {
-      Statement stmt = conn.createStatement();
-      ResultSet rs = stmt.executeQuery("SELECT '5'");
-      assertTrue(rs.next());
-      assertEquals("5", rs.getString(1));
-    }
-  }
-
-  @Test
-  // @ClearSystemProperty(key = "some key")
-  @SetEnvironmentVariable(key = "MARIADB2_USER", value = "identityUser")
-  @SetEnvironmentVariable(key = "MARIADB2_PWD", value = "!Passw0rd3Works")
-  @SuppressWarnings("unchecked")
-  public void envTestsIdentityTest() throws Exception {
-    Assumptions.assumeTrue(!isMaxscale());
-    Assumptions.assumeTrue(isMariaDBServer() && haveSsl());
-    if (!minVersion(11, 4, 1)) {
-      // Self-signed certificates error
-      assertThrows(SQLException.class, () -> createCon("credentialType=ENVTEST&sslMode=DISABLE"));
-      assertThrows(SQLException.class, () -> createCon("credentialType=ENVTEST"));
-    }
-
-    try (Connection conn = createCon("credentialType=ENVTEST&sslMode=TRUST")) {
-      Statement stmt = conn.createStatement();
-      ResultSet rs = stmt.executeQuery("SELECT '5'");
-      assertTrue(rs.next());
-      assertEquals("5", rs.getString(1));
-    }
-  }
 }
-- 
2.50.1

