--- zerovec-derive-0.11.6/Cargo.toml	2006-07-24T01:21:28+00:00
+++ zerovec-derive-0.11.6/Cargo.toml	2026-08-21T06:18:14.513832+00:00
@@ -45,24 +45,6 @@
 name = "zerovec_derive"
 path = "src/lib.rs"
 proc-macro = true
-
-[[example]]
-name = "derives"
-path = "examples/derives.rs"
-test = true
-harness = false
-
-[[example]]
-name = "make"
-path = "examples/make.rs"
-test = true
-harness = false
-
-[[example]]
-name = "make_var"
-path = "examples/make_var.rs"
-test = true
-harness = false
 
 [[test]]
 name = "derive_ule_test"
