diff -pruN qt-x11-free-3.3.6.org/src/kernel/qscriptengine_x11.cpp qt-x11-free-3.3.6/src/kernel/qscriptengine_x11.cpp
--- qt-x11-free-3.3.6.org/src/kernel/qscriptengine_x11.cpp	2007-02-13 14:41:25.000000000 +0530
+++ qt-x11-free-3.3.6/src/kernel/qscriptengine_x11.cpp	2007-02-13 14:40:02.000000000 +0530
@@ -253,7 +253,7 @@ static const unsigned char indicForms[0x
     Unknown, Unknown, Unknown, Unknown,
     Unknown, Unknown, Unknown, Unknown,
 
-    Other, Other, Other, Other,
+    Other, Other, VowelMark, VowelMark,
     Other, Other, Other, Other,
     Other, Other, Other, Other,
     Other, Other, Other, Other,
