diff -ruN a/fluent-plugin-kafka.gemspec b/fluent-plugin-kafka.gemspec
--- a/fluent-plugin-kafka.gemspec	2022-07-07 16:36:58.897805650 -0700
+++ b/fluent-plugin-kafka.gemspec	2022-07-07 16:37:27.726147887 -0700
@@ -8,7 +8,7 @@
   gem.homepage      = "https://github.com/fluent/fluent-plugin-kafka"
   gem.license       = "Apache-2.0"
 
-  gem.files         = `git ls-files`.split($\)
+  gem.files         = Dir['**/*']
   gem.executables   = gem.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.name          = "fluent-plugin-kafka"
