From 44b8e6cd6c2b962867d0e87380d2ca85d76b92e5 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 1 Jul 2014 08:48:12 -0400
Subject: [PATCH] Manually tell it we've got 64-bit files, because 32-bit OSes
 need it.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 dumpet.c    | 1 +
 1 files changed, 1 insertions(+)

diff --git a/dumpet.c b/dumpet.c
index 24b1395..18be54b 100644
--- a/dumpet.c
+++ b/dumpet.c
@@ -20,6 +20,7 @@
  */
 
 #define _GNU_SOURCE 1
+#define _FILE_OFFSET_BITS 64
 #include <stdlib.h>
 #include <stdint.h>
 #include <stdio.h>
-- 
1.9.3

