diff --git a/src/tests/p11test/p11test.c b/src/tests/p11test/p11test.c
index feecf6fb..a16282cb 100644
--- a/src/tests/p11test/p11test.c
+++ b/src/tests/p11test/p11test.c
@@ -34,6 +34,9 @@
 
 #define DEFAULT_P11LIB	"../../pkcs11/.libs/opensc-pkcs11.so"
 
+/* Global variable keeping information about token we are using */
+token_info_t token;
+
 void display_usage() {
 	fprintf(stdout,
 		" Usage:\n"
diff --git a/src/tests/p11test/p11test_common.h b/src/tests/p11test/p11test_common.h
index 2f3b238d..9eb5cede 100644
--- a/src/tests/p11test/p11test_common.h
+++ b/src/tests/p11test/p11test_common.h
@@ -84,7 +84,7 @@ typedef struct {
 	size_t  num_keygen_mechs;
 } token_info_t;
 
-token_info_t token;
+extern token_info_t token;
 
 #endif /* P11TEST_COMMON_H */
 
