From 7da90c7ea755fffbfbe9006d29524c7c8a86a925 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sun, 14 Jan 2024 16:43:15 +0100
Subject: [PATCH] * src/log.c (log_init): Redirect INFO logs to stderr with -O-

---
 src/log.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/log.c b/src/log.c
index 8b6796e9..61adde4b 100644
--- a/src/log.c
+++ b/src/log.c
@@ -226,7 +226,7 @@ void log_init(void)
 
 	// set info logging
 	wget_logger_set_func(wget_get_logger(WGET_LOGGER_INFO),
-		config.verbose && !config.quiet ? (fileno(stdout) == fileno(stderr) ? write_info_stderr : write_info_stdout) : NULL);
+		config.verbose && !config.quiet ? ((fileno(stdout) == fileno(stderr) || !wget_strcmp(config.output_document, "-")) ? write_info_stderr : write_info_stdout) : NULL);
 //	wget_logger_set_stream(wget_get_logger(WGET_LOGGER_INFO), config.verbose && !config.quiet ? stdout : NULL);
 #endif
 }
-- 
2.43.0

