--- a/Lib/cgi.py	2014-06-29 19:05:24.000000000 -0700
+++ b/Lib/cgi.py	2015-02-05 18:03:52.273922307 -0800
@@ -1,13 +1,5 @@
-#! /usr/local/bin/python
+#! /usr/bin/python3
 
-# NOTE: the above "/usr/local/bin/python" is NOT a mistake.  It is
-# intentionally NOT "/usr/bin/env python".  On many systems
-# (e.g. Solaris), /usr/local/bin is not in $PATH as passed to CGI
-# scripts, and /usr/local/bin is the default directory where Python is
-# installed, so /usr/bin/env would be unable to find python.  Granted,
-# binary installations by Linux vendors often install Python in
-# /usr/bin.  So let those vendors patch cgi.py to match their choice
-# of installation.
 
 """Support module for CGI (Common Gateway Interface) scripts.
 
