diff -rupr c/Lib/platform.py d/Lib/platform.py
--- c/Lib/platform.py	2015-12-06 17:39:08.000000000 -0800
+++ d/Lib/platform.py	2017-01-20 16:59:20.272309523 -0800
@@ -268,7 +268,7 @@ _release_version = re.compile(r'([^0-9]+
 _supported_dists = (
     'SuSE', 'debian', 'fedora', 'redhat', 'centos',
     'mandrake', 'mandriva', 'rocks', 'slackware', 'yellowdog', 'gentoo',
-    'UnitedLinux', 'turbolinux', 'arch', 'mageia')
+    'UnitedLinux', 'turbolinux', 'arch', 'mageia','mariner')
 
 def _parse_release_file(firstline):
 
