From: Mitch Lindgren <mitchl@microsoft.com>
Date: Thu, 23 Apr 2020 15:54:49 -0700
Subject: [PATCH] Replace unsupported TLS methods with TLS_method

diff -ruN a/Modules/_ssl.c b/Modules/_ssl.c
--- a/Modules/_ssl.c	2020-04-30 10:24:29.253672308 -0700
+++ b/Modules/_ssl.c	2020-05-12 11:50:31.590358951 -0700
@@ -263,6 +263,25 @@
 
 #endif /* OpenSSL < 1.1.0 or LibreSSL < 2.7.0 */
 
+// Workaround for importing ssl module failing when version-specific TLS methods
+// (LSG-Linux bug 7028652)
+// Since the version-specific methods are not present, we instead simply return TLS_method which
+// automatically negotiates the highest available version (TLS v1.2 only in our case)
+const SSL_METHOD *TLSv1_method(void)
+{
+    return TLS_method();
+}
+
+const SSL_METHOD *TLSv1_1_method(void)
+{
+    return TLS_method();
+}
+
+const SSL_METHOD *TLSv1_2_method(void)
+{
+    return TLS_method();
+}
+
 /* Default cipher suites */
 #ifndef PY_SSL_DEFAULT_CIPHERS
 #define PY_SSL_DEFAULT_CIPHERS 1
