--- gcc/config/m68k/m68k.cc.orig	2023-05-19 10:05:34.000000000 +0100
+++ gcc/config/m68k/m68k.cc	2023-06-08 12:22:09.168546535 +0100
@@ -5031,7 +5031,7 @@
   else if (letter == '/')
     asm_fprintf (file, "%R");
   else if (letter == '?')
-    asm_fprintf (file, m68k_library_id_string);
+    asm_fprintf (file, "%s", m68k_library_id_string);
   else if (letter == 'p')
     {
       output_addr_const (file, op);
