diff -ruN a/fluent-plugin-webhdfs.gemspec b/fluent-plugin-webhdfs.gemspec
--- a/fluent-plugin-webhdfs.gemspec	2022-07-07 17:45:56.526926415 -0700
+++ b/fluent-plugin-webhdfs.gemspec	2022-07-07 17:46:14.683131357 -0700
@@ -10,7 +10,7 @@
   gem.homepage      = "https://github.com/fluent/fluent-plugin-webhdfs"
   gem.license       = "Apache-2.0"
 
-  gem.files         = `git ls-files`.split($\)
+  gem.files         = Dir['**/*']
   gem.executables   = gem.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.require_paths = ["lib"]
