diff -r -u a/mkspecs/linux-g++/qmake.conf b/mkspecs/linux-g++/qmake.conf
--- a/mkspecs/linux-g++/qmake.conf	2015-10-30 06:20:01.000000000 -0200
+++ b/mkspecs/linux-g++/qmake.conf	2015-11-05 11:23:23.230741601 -0200
@@ -5,6 +5,7 @@
 MAKEFILE_GENERATOR      = UNIX
 CONFIG                 += incremental
 QMAKE_INCREMENTAL_STYLE = sublib
+QMAKE_CFLAGS_RELEASE   += -O2
 
 include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)
diff -r -u a/mkspecs/linux-g++-32/qmake.conf b/mkspecs/linux-g++-32/qmake.conf
--- a/mkspecs/linux-g++-32/qmake.conf	2015-10-30 06:20:01.000000000 -0200
+++ b/mkspecs/linux-g++-32/qmake.conf	2015-11-05 11:22:19.761494470 -0200
@@ -10,6 +10,7 @@
 
 QMAKE_CFLAGS            = -m32
 QMAKE_LFLAGS            = -m32
+QMAKE_CFLAGS_RELEASE   += -O2
 
 include(../common/gcc-base-unix.conf)
 include(../common/g++-unix.conf)
diff -r -u a/mkspecs/linux-g++-64/qmake.conf b/mkspecs/linux-g++-64/qmake.conf
--- a/mkspecs/linux-g++-64/qmake.conf	2015-10-30 06:20:01.000000000 -0200
+++ b/mkspecs/linux-g++-64/qmake.conf	2015-11-05 11:22:49.497610248 -0200
@@ -13,6 +13,7 @@
 
 QMAKE_CFLAGS            = -m64
 QMAKE_LFLAGS            = -m64
+QMAKE_CFLAGS_RELEASE   += -O2
 
 include(../common/gcc-base-unix.conf)
 include(../common/g++-unix.conf)
