diff -ruN a/fluent-config-regexp-type.gemspec b/fluent-config-regexp-type.gemspec
--- a/fluent-config-regexp-type.gemspec	2018-05-23 19:31:02.000000000 -0700
+++ b/fluent-config-regexp-type.gemspec	2022-07-07 16:14:06.445576502 -0700
@@ -11,7 +11,7 @@
   spec.description   = "The compatibility monkey patch to use regexp type "
   spec.homepage      = "https://github.com/okkez/fluent-config-regexp-type"
 
-  spec.files         = `git ls-files -z`.split("\x0").reject do |f|
+  spec.files         = Dir['**/*'].reject do |f|
     f.match(%r{^(test|spec|features)/})
   end
   spec.bindir        = "exe"
