diff -ruN a/ffi.gemspec b/ffi.gemspec
--- a/ffi.gemspec	2022-01-10 06:33:24.000000000 -0800
+++ b/ffi.gemspec	2022-07-06 18:38:21.055186019 -0700
@@ -16,20 +16,10 @@
     s.metadata['source_code_uri'] = 'https://github.com/ffi/ffi/'
     s.metadata['mailing_list_uri'] = 'http://groups.google.com/group/ruby-ffi'
   end
-  s.files = `git ls-files -z`.split("\x0").reject do |f|
+  s.files = Dir['**/*'].reject do |f|
     f =~ /^(\.|bench|gen|libtest|nbproject|spec)/
   end
 
-  # Add libffi git files
-  lfs = `git --git-dir ext/ffi_c/libffi/.git ls-files -z`.split("\x0")
-  # Add autoconf generated files of libffi
-  lfs += %w[ configure config.guess config.sub install-sh ltmain.sh missing fficonfig.h.in ]
-  # Add automake generated files of libffi
-  lfs += `git --git-dir ext/ffi_c/libffi/.git ls-files -z *.am */*.am`.gsub(".am\0", ".in\0").split("\x0")
-  s.files += lfs.map do |f|
-    File.join("ext/ffi_c/libffi", f)
-  end
-
   s.extensions << 'ext/ffi_c/extconf.rb'
   s.rdoc_options = %w[--exclude=ext/ffi_c/.*\.o$ --exclude=ffi_c\.(bundle|so)$]
   s.license = 'BSD-3-Clause'
