diff -ruN a/ffi.gemspec b/ffi.gemspec
--- a/ffi.gemspec	2022-01-10 06:33:24.000000000 -0800
+++ b/ffi.gemspec	2022-04-14 18:44:06.744012764 -0700
@@ -22,8 +22,6 @@
 
   # Add libffi git files
   lfs = `git --git-dir ext/ffi_c/libffi/.git ls-files -z`.split("\x0")
-  # Add autoconf generated files of libffi
-  lfs += %w[ configure config.guess config.sub install-sh ltmain.sh missing fficonfig.h.in ]
   # Add automake generated files of libffi
   lfs += `git --git-dir ext/ffi_c/libffi/.git ls-files -z *.am */*.am`.gsub(".am\0", ".in\0").split("\x0")
   s.files += lfs.map do |f|
