--- sequoia-openpgp-2.4.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ sequoia-openpgp-2.4.1/Cargo.toml	2026-09-23T13:07:38.792891+00:00
@@ -79,8 +79,6 @@
     "dep:cipher",
     "dep:cx448",
     "dep:eax",
-    "dep:winapi",
-    "dep:win-crypto-ng",
     "dep:ed25519",
     "dep:ed25519-dalek",
     "dep:num-bigint-dig",
@@ -90,7 +88,6 @@
     "dep:hkdf",
     "dep:sha2",
     "dep:ocb3",
-    "dep:zeroize",
 ]
 crypto-fuzzing = []
 crypto-nettle = ["dep:nettle"]
@@ -549,37 +546,6 @@
 version = ">=0.20, <0.24"
 default-features = false
 
-[target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dependencies.chrono]
-version = "0.4.10"
-features = [
-    "std",
-    "wasmbind",
-    "clock",
-]
-default-features = false
-
-[target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dependencies.getrandom]
-version = "0.2"
-features = ["js"]
-
-[target."cfg(windows)".dependencies.win-crypto-ng]
-version = "0.5.1"
-features = [
-    "rand",
-    "block-cipher",
-]
-optional = true
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.8"
-features = ["bcrypt"]
-optional = true
-default-features = false
-
-[target."cfg(windows)".dependencies.zeroize]
-version = "1"
-optional = true
-
 [lints.clippy.complexity]
 level = "allow"
 priority = -1
@@ -603,3 +569,4 @@
     'cfg(osslconf, values("OPENSSL_NO_OCB"))',
     'cfg(osslconf, values("OPENSSL_NO_DES"))',
 ]
+
