From 7610a10691e680db00edcbd4ce6767a30641cd71 Mon Sep 17 00:00:00 2001
From: Michael Simacek <msimacek@redhat.com>
Date: Tue, 6 Jun 2017 13:47:43 +0200
Subject: [PATCH 2/4] Invoke logback via reflection

---
 .../logging/impl/LogbackConfiguration.java    | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/maven-embedder/src/main/java/org/apache/maven/cli/logging/impl/LogbackConfiguration.java b/maven-embedder/src/main/java/org/apache/maven/cli/logging/impl/LogbackConfiguration.java
index 5d9fab744..ced38cb5a 100644
--- a/maven-embedder/src/main/java/org/apache/maven/cli/logging/impl/LogbackConfiguration.java
+++ b/maven-embedder/src/main/java/org/apache/maven/cli/logging/impl/LogbackConfiguration.java
@@ -35,22 +35,31 @@
     @Override
     public void setRootLoggerLevel( Level level )
     {
-        ch.qos.logback.classic.Level value;
+        String value;
         switch ( level )
         {
             case DEBUG:
-                value = ch.qos.logback.classic.Level.DEBUG;
+                value = "DEBUG";
                 break;
 
             case INFO:
-                value = ch.qos.logback.classic.Level.INFO;
+                value = "INFO";
                 break;
 
             default:
-                value = ch.qos.logback.classic.Level.ERROR;
+                value = "ERROR";
                 break;
         }
-        ( (ch.qos.logback.classic.Logger) LoggerFactory.getLogger( Logger.ROOT_LOGGER_NAME ) ).setLevel( value );
+        Logger logger = LoggerFactory.getLogger( Logger.ROOT_LOGGER_NAME );
+        try {
+            Class<?> levelClass = Class.forName("ch.qos.logback.classic.Level");
+            Object logbackLevel = levelClass.getField(value).get(null);
+            Class<?> loggerClass = Class.forName("ch.qos.logback.classic.Logger");
+            loggerClass.getMethod("setLevel", new Class<?>[] {levelClass})
+                .invoke(logger, new Object[] {logbackLevel});
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to initialize logback configuration", e);
+        }
     }
 
     @Override
-- 
2.21.0

