From 564a923a91d042e24e9259d86f69e0061f28ef4f Mon Sep 17 00:00:00 2001
From: Vladimir Panteleev <git@thecybershadow.net>
Date: Thu, 16 Jan 2020 12:11:20 +0000
Subject: [PATCH 1/2] Win::Hivex::Regedit: Accept CRLF line endings

---
 perl/lib/Win/Hivex/Regedit.pm | 1 +
 1 file changed, 1 insertion(+)

diff --git a/perl/lib/Win/Hivex/Regedit.pm b/perl/lib/Win/Hivex/Regedit.pm
index 34426f1..2b17036 100644
--- a/perl/lib/Win/Hivex/Regedit.pm
+++ b/perl/lib/Win/Hivex/Regedit.pm
@@ -144,6 +144,7 @@ sub reg_import
         # this is fairly common in pasted regedit files.
         $lineno++;
         chomp;
+        s/\r$//;
         if (s/\\\s*$//) {
             $_ .= <$fh>;
             redo unless eof ($fh);
-- 
2.24.1

