From 334324dcc1804defb1d91c021d229fe8754de8c7 Mon Sep 17 00:00:00 2001
From: Peter Hillman <peterh@wetafx.co.nz>
Date: Tue, 16 Jul 2019 20:16:09 +1200
Subject: [PATCH 1/2] don't index empty array in testMultiPartSharedAttributes

---
 OpenEXR/IlmImfTest/testMultiPartSharedAttributes.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/OpenEXR/IlmImfTest/testMultiPartSharedAttributes.cpp b/OpenEXR/IlmImfTest/testMultiPartSharedAttributes.cpp
index 29fe4561..c4eec4ab 100644
--- a/OpenEXR/IlmImfTest/testMultiPartSharedAttributes.cpp
+++ b/OpenEXR/IlmImfTest/testMultiPartSharedAttributes.cpp
@@ -152,7 +152,7 @@ testMultiPartOutputFileForExpectedFailure (const vector<Header> & headers,
     try
     {
         remove(fn.c_str());
-        MultiPartOutputFile file(fn.c_str(), &headers[0],headers.size());
+        MultiPartOutputFile file(fn.c_str(), (headers.size()>0 ? &headers[0] : NULL) , headers.size() );
         cerr << "ERROR -- " << failMessage << endl;
         assert (false);
     }

From de72d9c8cb898790fe65c42921e9f0e995e8d212 Mon Sep 17 00:00:00 2001
From: Peter Hillman <peterh@wetafx.co.nz>
Date: Wed, 17 Jul 2019 17:46:20 +1200
Subject: [PATCH 2/2] use headers.data() instead of &headers[0]

---
 OpenEXR/IlmImfTest/testMultiPartSharedAttributes.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/OpenEXR/IlmImfTest/testMultiPartSharedAttributes.cpp b/OpenEXR/IlmImfTest/testMultiPartSharedAttributes.cpp
index c4eec4ab..460edc3f 100644
--- a/OpenEXR/IlmImfTest/testMultiPartSharedAttributes.cpp
+++ b/OpenEXR/IlmImfTest/testMultiPartSharedAttributes.cpp
@@ -152,7 +152,7 @@ testMultiPartOutputFileForExpectedFailure (const vector<Header> & headers,
     try
     {
         remove(fn.c_str());
-        MultiPartOutputFile file(fn.c_str(), (headers.size()>0 ? &headers[0] : NULL) , headers.size() );
+        MultiPartOutputFile file(fn.c_str(), headers.data() , headers.size() );
         cerr << "ERROR -- " << failMessage << endl;
         assert (false);
     }
