--- ttmkfdir-3.0.9/encoding.cpp.orig   2002-12-09 17:52:48.000000000 +0900
+++ ttmkfdir-3.0.9/encoding.cpp        2004-01-31 18:16:46.617118976 +0900
@@ -4,7 +4,6 @@
 #include <cstring>
 #include <unistd.h>
 #include <zlib.h>
-#include "freetype/freetype.h"

 #include "ttmkfdir.h"
 #include "encoding.h"
--- ttmkfdir-3.0.9/encoding.h.orig     2002-12-09 17:52:48.000000000 +0900
+++ ttmkfdir-3.0.9/encoding.h  2004-01-31 18:19:18.600014064 +0900
@@ -6,7 +6,8 @@
 #include <map>
 #include <string>

-#include "freetype/freetype.h"
+#include <ft2build.h>
+#include FT_FREETYPE_H

 #include "util.h"

--- ttmkfdir-3.0.9/ttf.h.orig  2003-01-08 14:25:25.000000000 +0900
+++ ttmkfdir-3.0.9/ttf.h       2004-01-31 18:32:53.333155800 +0900
@@ -3,12 +3,13 @@
 #define TTF_H__

 #include <string>
-#include "freetype/freetype.h"
-#include "freetype/tttables.h"
-#include "freetype/ftsnames.h"
-#include "freetype/ttnameid.h"
-#include "freetype/fterrors.h"
-#include "freetype/ftmodule.h"
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include FT_SFNT_NAMES_H
+#include FT_TRUETYPE_TABLES_H
+#include FT_TRUETYPE_IDS_H
+#include FT_ERRORS_H
+#include FT_MODULE_H

 #include "util.h"
 #include "encoding.h"
