diff -up sse2neon-1.7.0/tests/impl.cpp.orig sse2neon-1.7.0/tests/impl.cpp
--- sse2neon-1.7.0/tests/impl.cpp.orig	2024-09-11 22:01:32.988454083 +0200
+++ sse2neon-1.7.0/tests/impl.cpp	2024-09-11 22:34:57.437312354 +0200
@@ -1332,7 +1332,7 @@ result_t test_mm_comieq_ss(const SSE2NEO
     // The GCC does not implement _mm_comieq_ss correctly.
     // See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98612 for more
     // information.
-#if defined(__GNUC__) && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ < 11)
     return TEST_UNIMPL;
 #else
     const float *_a = impl.mTestFloatPointer1;
@@ -1380,7 +1380,7 @@ result_t test_mm_comile_ss(const SSE2NEO
     // The GCC does not implement _mm_comile_ss correctly.
     // See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98612 for more
     // information.
-#if defined(__GNUC__) && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ < 11)
     return TEST_UNIMPL;
 #else
     const float *_a = impl.mTestFloatPointer1;
@@ -1401,7 +1401,7 @@ result_t test_mm_comilt_ss(const SSE2NEO
     // The GCC does not implement _mm_comilt_ss correctly.
     // See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98612 for more
     // information.
-#if defined(__GNUC__) && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ < 11)
     return TEST_UNIMPL;
 #else
     const float *_a = impl.mTestFloatPointer1;
@@ -1423,7 +1423,7 @@ result_t test_mm_comineq_ss(const SSE2NE
     // The GCC does not implement _mm_comineq_ss correctly.
     // See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98612 for more
     // information.
-#if defined(__GNUC__) && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ < 11)
     return TEST_UNIMPL;
 #else
     const float *_a = impl.mTestFloatPointer1;
@@ -1949,7 +1949,7 @@ result_t test_mm_extract_pi16(const SSE2
     // test when GCC fix this bug.
     // see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98495 for more
     // information
-#if defined(__clang__) || defined(_MSC_VER)
+#if defined(__clang__) || defined(_MSC_VER) || ( !defined(__clang__) && defined(__GNUC__) && (__GNUC__ >= 11) )
     uint64_t *_a = (uint64_t *) impl.mTestIntPointer1;
     const int idx = iter & 0x3;
 
@@ -4289,7 +4289,7 @@ result_t test_mm_comieq_sd(const SSE2NEO
     // The GCC does not implement _mm_comieq_sd correctly.
     // See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98612 for more
     // information.
-#if defined(__GNUC__) && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ < 11)
     return TEST_UNIMPL;
 #else
     const double *_a = (const double *) impl.mTestFloatPointer1;
@@ -4339,7 +4339,7 @@ result_t test_mm_comile_sd(const SSE2NEO
     // The GCC does not implement _mm_comile_sd correctly.
     // See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98612 for more
     // information.
-#if defined(__GNUC__) && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ < 11)
     return TEST_UNIMPL;
 #else
     const double *_a = (const double *) impl.mTestFloatPointer1;
@@ -4361,7 +4361,7 @@ result_t test_mm_comilt_sd(const SSE2NEO
     // The GCC does not implement _mm_comilt_sd correctly.
     // See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98612 for more
     // information.
-#if defined(__GNUC__) && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ < 11)
     return TEST_UNIMPL;
 #else
     const double *_a = (const double *) impl.mTestFloatPointer1;
@@ -4383,7 +4383,7 @@ result_t test_mm_comineq_sd(const SSE2NE
     // The GCC does not implement _mm_comineq_sd correctly.
     // See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98612 for more
     // information.
-#if defined(__GNUC__) && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ < 11)
     return TEST_UNIMPL;
 #else
     const double *_a = (const double *) impl.mTestFloatPointer1;
