diff -ruN a/fluent-plugin-record-modifier.gemspec b/fluent-plugin-record-modifier.gemspec
--- a/fluent-plugin-record-modifier.gemspec	2022-07-07 17:48:30.048676902 -0700
+++ b/fluent-plugin-record-modifier.gemspec	2022-07-07 17:49:49.801599978 -0700
@@ -11,9 +11,9 @@
   gem.email       = "repeatedly@gmail.com"
   #gem.platform    = Gem::Platform::RUBY
   gem.license     = 'MIT'
-  gem.files       = `git ls-files`.split("\n")
-  gem.test_files  = `git ls-files -- {test,spec,features}/*`.split("\n")
-  gem.executables = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
+  gem.files       = Dir['**/*']
+  gem.test_files  = gem.files.grep(%r{^(test|spec|features)/})
+  gem.executables = gem.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   gem.require_paths = ['lib']
 
   gem.add_dependency "fluentd", [">= 1.0", "< 2"]
