From 2c111a7766129647bac5736aa74f8a4478ccec03 Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Sat, 16 Mar 2019 00:17:41 -0400
Subject: [PATCH 1/6] Unbundle blosc

Signed-off-by: Elliott Sales de Andrade <quantum.analyst@gmail.com>
---
 MANIFEST.in |  1 -
 setup.py    | 70 ++++++++++++-----------------------------------------
 2 files changed, 16 insertions(+), 55 deletions(-)

diff --git a/MANIFEST.in b/MANIFEST.in
index 41513ec..ef51486 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,4 +1,3 @@
-recursive-include c-blosc *
 recursive-include numcodecs *.pyx
 recursive-include numcodecs *.pxd
 include numcodecs/tests/package_with_entrypoint-0.1.dist-info/entry_points.txt
diff --git a/setup.py b/setup.py
index c27fad3..b5230b2 100644
--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,8 @@
 import os
+import shlex
+import subprocess
 import sys
+from collections import namedtuple
 from distutils import ccompiler
 from distutils.command.clean import clean
 from distutils.sysconfig import customize_compiler
@@ -56,6 +59,15 @@ def error(*msg):
     print('[numcodecs]', *msg, **kwargs)
 
 
+Package = namedtuple('Package', 'cflags libs')
+
+
+def pkgconfig(name):
+    cflags = os.fsdecode(subprocess.check_output(['pkg-config', '--cflags', name]))
+    libs = os.fsdecode(subprocess.check_output(['pkg-config', '--libs', name]))
+    return Package(shlex.split(cflags), shlex.split(libs))
+
+
 def blosc_extension():
     info('setting up Blosc extension')
 
@@ -69,57 +81,9 @@ def blosc_extension():
         extra_link_args.append('-pthread')
 
     # setup blosc sources
-    blosc_sources = [f for f in glob('c-blosc/blosc/*.c') if 'avx2' not in f and 'sse2' not in f]
-    include_dirs = [os.path.join('c-blosc', 'blosc')]
-
-    # add internal complibs
-    blosc_sources += glob('c-blosc/internal-complibs/lz4*/*.c')
-    blosc_sources += glob('c-blosc/internal-complibs/snappy*/*.cc')
-    blosc_sources += glob('c-blosc/internal-complibs/zlib*/*.c')
-    blosc_sources += glob('c-blosc/internal-complibs/zstd*/common/*.c')
-    blosc_sources += glob('c-blosc/internal-complibs/zstd*/compress/*.c')
-    blosc_sources += glob('c-blosc/internal-complibs/zstd*/decompress/*.c')
-    blosc_sources += glob('c-blosc/internal-complibs/zstd*/dictBuilder/*.c')
-    include_dirs += [d for d in glob('c-blosc/internal-complibs/*') if os.path.isdir(d)]
-    include_dirs += [d for d in glob('c-blosc/internal-complibs/*/*') if os.path.isdir(d)]
-    include_dirs += [d for d in glob('c-blosc/internal-complibs/*/*/*') if os.path.isdir(d)]
-    # remove minizip because Python.h 3.8 tries to include crypt.h
-    include_dirs = [d for d in include_dirs if 'minizip' not in d]
-    define_macros += [
-        ('HAVE_LZ4', 1),
-        # ('HAVE_SNAPPY', 1),
-        ('HAVE_ZLIB', 1),
-        ('HAVE_ZSTD', 1),
-    ]
-    # define_macros += [('CYTHON_TRACE', '1')]
-
-    # SSE2
-    if have_sse2 and not disable_sse2:
-        info('compiling Blosc extension with SSE2 support')
-        extra_compile_args.append('-DSHUFFLE_SSE2_ENABLED')
-        blosc_sources += [f for f in glob('c-blosc/blosc/*.c') if 'sse2' in f]
-        if os.name == 'nt':
-            define_macros += [('__SSE2__', 1)]
-    else:
-        info('compiling Blosc extension without SSE2 support')
-
-    # AVX2
-    if have_avx2 and not disable_avx2:
-        info('compiling Blosc extension with AVX2 support')
-        extra_compile_args.append('-DSHUFFLE_AVX2_ENABLED')
-        blosc_sources += [f for f in glob('c-blosc/blosc/*.c') if 'avx2' in f]
-        if os.name == 'nt':
-            define_macros += [('__AVX2__', 1)]
-    else:
-        info('compiling Blosc extension without AVX2 support')
-
-    # include assembly files
-    if cpuinfo.platform.machine() == 'x86_64':
-        extra_objects = [
-            S[:-1] + 'o' for S in glob("c-blosc/internal-complibs/zstd*/decompress/*amd64.S")
-        ]
-    else:
-        extra_objects = []
+    blosc = pkgconfig('blosc')
+    extra_compile_args += blosc.cflags
+    extra_link_args += blosc.libs
 
     sources = ['numcodecs/blosc.pyx']
 
@@ -127,12 +91,10 @@ def blosc_extension():
     return [
         Extension(
             'numcodecs.blosc',
-            sources=sources + blosc_sources,
-            include_dirs=include_dirs,
+            sources=sources,
             define_macros=define_macros,
             extra_compile_args=extra_compile_args,
             extra_link_args=extra_link_args,
-            extra_objects=extra_objects,
         ),
     ]
 
-- 
2.50.0

