--- a/chkconfig.c	2015-06-01 05:07:44.000000000 -0700
+++ b/chkconfig.c	2015-11-20 10:31:05.364098917 -0800
@@ -700,8 +700,8 @@
     struct service s;
     poptContext optCon;
     struct poptOption optionsTable[] = {
-        {"add", '\0', 0, &addItem, 0},
-        {"del", '\0', 0, &delItem, 0},
+        {"add", 'a', 0, &addItem, 0},
+        {"del", 'd', 0, &delItem, 0},
         {"override", '\0', 0, &overrideItem, 0},
         {"no-redirect", '\0', 0, &noRedirectItem, 0},
         {"list", '\0', 0, &listItem, 0},
