diff --git a/scripts/build b/scripts/build
index 809285e9f14..c80dd4b6bce 100755
--- a/scripts/build
+++ b/scripts/build
@@ -98,15 +98,17 @@ if [ ! -x ${INSTALLBIN}/cni ]; then
     echo Building cni
     TMPDIR=$(mktemp -d)
     trap cleanup EXIT
-    WORKDIR=$TMPDIR/src/github.com/containernetworking/plugins
-    git clone -b $VERSION_CNIPLUGINS https://github.com/rancher/plugins.git $WORKDIR
-    cd $WORKDIR
-    GO111MODULE=off GOPATH=$TMPDIR CGO_ENABLED=0 "${GO}" build -tags "$TAGS" -ldflags "$LDFLAGS $STATIC" -o $INSTALLBIN/cni
+    WORKDIR=$TMPDIR/src/github.com/containernetworking
+    mkdir -p $WORKDIR
+    ls
+    cp -r vendor/plugins $WORKDIR
+    cd $WORKDIR/plugins
+    GO111MODULE=off GOPATH=$TMPDIR CGO_ENABLED=0 "${GO}" build -tags "$TAGS" -ldflags "$VERSIONFLAGS $LDFLAGS $STATIC" -o $INSTALLBIN/cni
 )
 fi
 
 echo Building k3s
-CGO_ENABLED=1 "${GO}" build -tags "$TAGS" -ldflags "$VERSIONFLAGS $LDFLAGS $STATIC" -o bin/k3s ./cmd/server/main.go
+CGO_ENABLED=1 "${GO}" build -mod=vendor -tags "$TAGS" -ldflags "$VERSIONFLAGS $LDFLAGS $STATIC" -o bin/k3s ./cmd/server/main.go
 ln -s k3s ./bin/k3s-agent
 ln -s k3s ./bin/k3s-server
 ln -s k3s ./bin/k3s-etcd-snapshot
@@ -119,16 +121,22 @@ ln -s k3s ./bin/ctr
 export GOPATH=$(pwd)/build
 
 echo Building containerd
-pushd ./build/src/github.com/containerd/containerd
+CONTAINERD_DIR=build/src/github.com/containerd
+mkdir -p $CONTAINERD_DIR
+cp -r vendor/containerd $CONTAINERD_DIR
+pushd ./$CONTAINERD_DIR/containerd
 TAGS="${TAGS/netcgo/netgo}"
-CGO_ENABLED=1 "${GO}" build -tags "$TAGS" -ldflags "$VERSIONFLAGS $LDFLAGS $STATIC" -o bin/containerd              ./cmd/containerd
-CGO_ENABLED=1 "${GO}" build -tags "$TAGS" -ldflags "$VERSIONFLAGS $LDFLAGS $STATIC" -o bin/containerd-shim-runc-v2 ./cmd/containerd-shim-runc-v2
+CGO_ENABLED=1 "${GO}" build -mod=vendor -tags "$TAGS" -ldflags "$VERSIONFLAGS $LDFLAGS $STATIC" -o bin/containerd              ./cmd/containerd
+CGO_ENABLED=1 "${GO}" build -mod=vendor -tags "$TAGS" -ldflags "$VERSIONFLAGS $LDFLAGS $STATIC" -o bin/containerd-shim-runc-v2 ./cmd/containerd-shim-runc-v2
 popd
-cp -vf ./build/src/github.com/containerd/containerd/bin/* ./bin/
+cp -vf ./$CONTAINERD_DIR/containerd/bin/* ./bin/
 
 echo Building runc
-pushd ./build/src/github.com/opencontainers/runc
+RUNC_DIR=build/src/github.com/opencontainers
+mkdir -p $RUNC_DIR
+cp -r vendor/runc $RUNC_DIR
+pushd ./$RUNC_DIR/runc
 rm -f runc
-make EXTRA_LDFLAGS="-w -s" BUILDTAGS="$RUNC_TAGS" $RUNC_STATIC
+make EXTRA_LDFLAGS="-w -s" BUILDTAGS="$RUNC_TAGS" $RUNC_STATIC GOMODVENDOR=1
 popd
-cp -vf ./build/src/github.com/opencontainers/runc/runc ./bin/
+cp -vf ./$RUNC_DIR/runc/runc ./bin/
\ No newline at end of file
diff --git a/scripts/package-cli b/scripts/package-cli
index c96a2cae515..737674c1840 100755
--- a/scripts/package-cli
+++ b/scripts/package-cli
@@ -59,7 +59,7 @@ LDFLAGS="
     -w -s
 "
 STATIC="-extldflags '-static'"
-CGO_ENABLED=0 "${GO}" build -ldflags "$LDFLAGS $STATIC" -o ${CMD_NAME} ./cmd/k3s/main.go
+CGO_ENABLED=0 "${GO}" build -mod=vendor -ldflags "$LDFLAGS $STATIC" -o ${CMD_NAME} ./cmd/k3s/main.go
 
 stat ${CMD_NAME}
 