--- ChangeLog
+++ ChangeLog
@@ -72,7 +72,7 @@
 2.18	This version needs libgd 2.0.28 or higher.
 	Fixed documentation bug in synopsis of GD::Simple.
 	Updated Polyline to version 0.20
-2.17	Added animated GIF patches from Jaakko Hyv‰tti.
+2.17	Added animated GIF patches from Jaakko Hyv√§tti.
 	Added dynamic bitmapped font loading support.
 	Added fontconfig support.
 	Added a simplified API called GD::Simple.
