diff -ruN a/serverengine.gemspec b/serverengine.gemspec
--- a/serverengine.gemspec	2022-01-13 01:45:20.000000000 -0800
+++ b/serverengine.gemspec	2022-07-03 16:44:41.589782916 -0700
@@ -11,7 +11,7 @@
   gem.homepage      = "https://github.com/fluent/serverengine"
   gem.license       = "Apache 2.0"
 
-  gem.files         = `git ls-files`.split($\)
+  gem.files         = Dir['**/*']
   gem.executables   = gem.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.require_paths = ["lib"]
