From 1cf5aef99b26a46529ca797547c0b49627fffe78 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sat, 7 Dec 2019 11:08:54 +0000
Subject: [PATCH 2/2] caml_named_value returns const value pointer in OCaml
 4.09+
 
---
 augeas-c.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
 
diff --git a/augeas-c.c b/augeas-c.c
index 9217017..679bcb5 100644
--- a/augeas-c.c
+++ b/augeas-c.c
@@ -86,7 +86,7 @@ static const int error_map_len = sizeof error_map / sizeof error_map[0];
 static void
 raise_error_and_maybe_close (augeas_t t, const char *msg, bool close_handle)
 {
-  value *exn = caml_named_value ("Augeas.Error");
+  const value *exn = caml_named_value ("Augeas.Error");
   value args[5];
   const int code = aug_error (t);
   const char *aug_err_msg;
@@ -132,7 +132,7 @@ raise_error_and_maybe_close (augeas_t t, const char *msg, bool close_handle)
 static void
 raise_init_error (const char *msg)
 {
-  value *exn = caml_named_value ("Augeas.Error");
+  const value *exn = caml_named_value ("Augeas.Error");
   value args[5];
 
   args[0] = caml_alloc (1, 0);
-- 
2.23.0
