diff -rupN --no-dereference qtbase-everywhere-src-6.10.1/cmake/QtPriHelpers.cmake qtbase-everywhere-src-6.10.1-new/cmake/QtPriHelpers.cmake
--- qtbase-everywhere-src-6.10.1/cmake/QtPriHelpers.cmake	2025-08-11 06:54:51.000000000 +0200
+++ qtbase-everywhere-src-6.10.1-new/cmake/QtPriHelpers.cmake	2025-09-02 15:30:04.326861863 +0200
@@ -551,9 +551,13 @@ QT.${config_module_name}_private.disable
         qt_path_join(private_pri_file_path "${target_path}" "${private_pri_file_name}")
         list(APPEND pri_files "${private_pri_file_path}")
 
-        set(library_prefixes ${CMAKE_SHARED_LIBRARY_PREFIX} ${CMAKE_STATIC_LIBRARY_PREFIX})
+        set(library_prefixes
+            ${CMAKE_SHARED_LIBRARY_PREFIX}
+            ${CMAKE_IMPORT_LIBRARY_PREFIX}
+            ${CMAKE_STATIC_LIBRARY_PREFIX})
         set(library_suffixes
             ${CMAKE_SHARED_LIBRARY_SUFFIX}
+            ${CMAKE_CMAKE_IMPORT_LIBRARY_SUFFIX}
             ${CMAKE_EXTRA_SHARED_LIBRARY_SUFFIXES}
             ${CMAKE_STATIC_LIBRARY_SUFFIX})
         if(MSVC)
@@ -1024,9 +1028,13 @@ CONFIG += ${private_config_joined}
         list(APPEND inputs "${preliminary_pri_root}/${cfg}/${pri_data_cmake_file}")
     endforeach()
 
-    set(library_prefixes ${CMAKE_SHARED_LIBRARY_PREFIX} ${CMAKE_STATIC_LIBRARY_PREFIX})
+    set(library_prefixes
+        ${CMAKE_SHARED_LIBRARY_PREFIX}
+        ${CMAKE_IMPORT_LIBRARY_PREFIX}
+        ${CMAKE_STATIC_LIBRARY_PREFIX})
     set(library_suffixes
         ${CMAKE_SHARED_LIBRARY_SUFFIX}
+        ${CMAKE_CMAKE_IMPORT_LIBRARY_SUFFIX}
         ${CMAKE_EXTRA_SHARED_LIBRARY_SUFFIXES}
         ${CMAKE_STATIC_LIBRARY_SUFFIX})
     if(MSVC)
diff -rupN --no-dereference qtbase-everywhere-src-6.10.1/cmake/QtPrlHelpers.cmake qtbase-everywhere-src-6.10.1-new/cmake/QtPrlHelpers.cmake
--- qtbase-everywhere-src-6.10.1/cmake/QtPrlHelpers.cmake	2025-08-11 06:54:51.000000000 +0200
+++ qtbase-everywhere-src-6.10.1-new/cmake/QtPrlHelpers.cmake	2025-09-02 15:30:04.327348071 +0200
@@ -131,11 +131,17 @@ ${prl_step1_content_libs}
          CONTENT
          "FINAL_PRL_FILE_PATH = ${final_prl_file_path}")
 
-    set(library_prefixes ${CMAKE_SHARED_LIBRARY_PREFIX} ${CMAKE_STATIC_LIBRARY_PREFIX})
+    set(library_prefixes
+        ${CMAKE_SHARED_LIBRARY_PREFIX}
+        ${CMAKE_IMPORT_LIBRARY_PREFIX}
+        ${CMAKE_STATIC_LIBRARY_PREFIX})
     set(library_suffixes
         ${CMAKE_SHARED_LIBRARY_SUFFIX}
+        ${CMAKE_CMAKE_IMPORT_LIBRARY_SUFFIX}
         ${CMAKE_EXTRA_SHARED_LIBRARY_SUFFIXES}
         ${CMAKE_STATIC_LIBRARY_SUFFIX})
+    list(REMOVE_DUPLICATES library_prefixes)
+    list(REMOVE_DUPLICATES library_suffixes)
 
     if(QT_GENERATOR_IS_MULTI_CONFIG)
         set(configs ${CMAKE_CONFIGURATION_TYPES})
diff -rupN --no-dereference qtbase-everywhere-src-6.10.1/mkspecs/common/g++-win32.conf qtbase-everywhere-src-6.10.1-new/mkspecs/common/g++-win32.conf
--- qtbase-everywhere-src-6.10.1/mkspecs/common/g++-win32.conf	2025-08-11 06:54:51.000000000 +0200
+++ qtbase-everywhere-src-6.10.1-new/mkspecs/common/g++-win32.conf	2025-09-02 15:30:04.327686606 +0200
@@ -54,6 +54,7 @@ QMAKE_PREFIX_SHLIB      =
 QMAKE_EXTENSION_SHLIB   = dll
 QMAKE_PREFIX_STATICLIB  = lib
 QMAKE_EXTENSION_STATICLIB = a
+QMAKE_EXTENSION_IMPORTLIB = dll.a
 QMAKE_LIB_EXTENSIONS    = a dll.a
 
 QMAKE_LIBS              =
diff -rupN --no-dereference qtbase-everywhere-src-6.10.1/mkspecs/features/create_cmake.prf qtbase-everywhere-src-6.10.1-new/mkspecs/features/create_cmake.prf
--- qtbase-everywhere-src-6.10.1/mkspecs/features/create_cmake.prf	2025-08-11 06:54:51.000000000 +0200
+++ qtbase-everywhere-src-6.10.1-new/mkspecs/features/create_cmake.prf	2025-09-02 15:30:04.327924560 +0200
@@ -332,8 +332,9 @@ mac {
         CMAKE_WINMAIN_FILE_LOCATION_DEBUG = libqtmain$${QT_LIBINFIX}$${debug_suffix}.a
         CMAKE_WINMAIN_FILE_LOCATION_RELEASE = libqtmain$${QT_LIBINFIX}.a
 
-        CMAKE_IMPLIB_FILE_LOCATION_DEBUG = lib$${CMAKE_QT_STEM}$${debug_suffix}.a
-        CMAKE_IMPLIB_FILE_LOCATION_RELEASE = lib$${CMAKE_QT_STEM}.a
+        isEmpty(QMAKE_EXTENSION_IMPORTLIB): QMAKE_EXTENSION_IMPORTLIB = a
+        CMAKE_IMPLIB_FILE_LOCATION_DEBUG = lib$${CMAKE_QT_STEM}$${debug_suffix}.$${QMAKE_EXTENSION_IMPORTLIB}
+        CMAKE_IMPLIB_FILE_LOCATION_RELEASE = lib$${CMAKE_QT_STEM}.$${QMAKE_EXTENSION_IMPORTLIB}
     } else {
         CMAKE_WINMAIN_FILE_LOCATION_DEBUG = qtmain$${QT_LIBINFIX}$${debug_suffix}.lib
         CMAKE_WINMAIN_FILE_LOCATION_RELEASE = qtmain$${QT_LIBINFIX}.lib
diff -rupN --no-dereference qtbase-everywhere-src-6.10.1/src/corelib/Qt6CoreMacros.cmake qtbase-everywhere-src-6.10.1-new/src/corelib/Qt6CoreMacros.cmake
--- qtbase-everywhere-src-6.10.1/src/corelib/Qt6CoreMacros.cmake	2025-08-11 06:54:51.000000000 +0200
+++ qtbase-everywhere-src-6.10.1-new/src/corelib/Qt6CoreMacros.cmake	2025-09-02 15:30:04.328387681 +0200
@@ -611,7 +611,6 @@ function(_qt_internal_apply_win_prefix_a
 
         # This should cover both MINGW with GCC and CLANG.
         if(NOT MSVC)
-            set_property(TARGET "${target}" PROPERTY IMPORT_SUFFIX ".a")
 
             get_target_property(target_type ${target} TYPE)
             if(target_type STREQUAL "STATIC_LIBRARY")
