diff -up ./photosyst.c.orig ./photosyst.c
--- ./photosyst.c.orig	2017-03-25 11:00:03.000000003 +0200
+++ ./photosyst.c	2017-12-08 01:31:13.795157092 +0200
@@ -1537,6 +1537,7 @@ static struct {
 	{ "^dasd[a-z][a-z]*$",			{0},  nullmodname, DSKTYPE, },
 	{ "^mmcblk[0-9][0-9]*$",		{0},  nullmodname, DSKTYPE, },
 	{ "^emcpower[a-z][a-z]*$",		{0},  nullmodname, DSKTYPE, },
+	{ "^nvme[0-9][a-z][0-9]*$",		{0},  nullmodname, DSKTYPE, },
 };
 
 static int
