--- src/org/iso_relax/jaxp/ValidatingDocumentBuilderFactory.java
+++ src/org/iso_relax/jaxp/ValidatingDocumentBuilderFactory.java
@@ -101,4 +101,12 @@
     { _WrappedFactory.setIgnoringElementContentWhitespace(whitespace); }
     public void setNamespaceAware(boolean awareness)
     { _WrappedFactory.setNamespaceAware(awareness); }
+    
+    //java5 build
+    public boolean getFeature(String name) throws ParserConfigurationException {
+        return _WrappedFactory.getFeature(name);
+    }
+    public void setFeature(String name, boolean value) throws ParserConfigurationException {
+        _WrappedFactory.setFeature(name, value);
+    }
 }
