diff --git a/Cython/Build/IpythonMagic.py b/Cython/Build/IpythonMagic.py
index 343c6d8..83583e2 100644
--- a/Cython/Build/IpythonMagic.py
+++ b/Cython/Build/IpythonMagic.py
@@ -565,7 +565,7 @@ __doc__ = __doc__.format(
     # rST doesn't see the -+ flag as part of an option list, so we
     # hide it from the module-level docstring.
-    CYTHON_DOC=dedent(CythonMagics.cython.__doc__\
-                                  .replace('-+, --cplus', '--cplus    ')),
+    CYTHON_DOC=dedent(
+        CythonMagics.cython.__doc__.replace('-+, --cplus', '--cplus    ')),
     CYTHON_INLINE_DOC=dedent(CythonMagics.cython_inline.__doc__),
     CYTHON_PYXIMPORT_DOC=dedent(CythonMagics.cython_pyximport.__doc__),
 )
