https://github.com/cython/cython/pull/5945/files

diff -ruN a/Cython/Build/Tests/TestCyCache.py b/Cython/Build/Tests/TestCyCache.py
--- a/Cython/Build/Tests/TestCyCache.py 2024-03-22 16:33:42.785995661 +0000
+++ b/Cython/Build/Tests/TestCyCache.py 2024-03-22 16:41:00.218527876 +0000
@@ -3,6 +3,7 @@
 import gzip
 import os
 import tempfile
+from contextlib import closing

 import Cython.Build.Dependencies
 import Cython.Utils
@@ -70,7 +71,8 @@
             f.write('pass')
         self.fresh_cythonize(a_pyx, cache=self.cache_dir)
         a_cache = os.path.join(self.cache_dir, os.listdir(self.cache_dir)[0])
-        gzip.GzipFile(a_cache, 'wb').write('fake stuff'.encode('ascii'))
+        with closing(gzip.GzipFile(a_cache, 'wb')) as gzipfile:
+            gzipfile.write('fake stuff'.encode('ascii'))
         os.unlink(a_c)
         self.fresh_cythonize(a_pyx, cache=self.cache_dir)
         with open(a_c) as f:
