diff -ruN a/src/runtime/pkg/resourcecontrol/cgroups.go b/src/runtime/pkg/resourcecontrol/cgroups.go
--- a/src/runtime/pkg/resourcecontrol/cgroups.go	2022-08-16 14:41:20.000000000 -0700
+++ b/src/runtime/pkg/resourcecontrol/cgroups.go	2022-08-26 11:00:33.841763764 -0700
@@ -57,6 +57,7 @@
 		"/dev/zero",
 		"/dev/urandom",
 		"/dev/console",
+		"/dev/ptmx",
 	}
 
 	// Processes running in a device-cgroup are constrained, they have acccess
@@ -65,6 +66,7 @@
 	// need access to certain character devices in the host, like kvm and vhost-net.
 	hypervisorDevices := []string{
 		"/dev/kvm",         // To run virtual machines
+		"/dev/mshv",         // To run virtual machines
 		"/dev/vhost-net",   // To create virtqueues
 		"/dev/vfio/vfio",   // To access VFIO devices
 		"/dev/vhost-vsock", // To interact with vsock if
