diff -up SQLAlchemy-Utils-0.41.1/setup.py.nosqla2 SQLAlchemy-Utils-0.41.1/setup.py
--- SQLAlchemy-Utils-0.41.1/setup.py.nosqla2	2024-03-25 18:48:29.557353009 +0100
+++ SQLAlchemy-Utils-0.41.1/setup.py	2024-03-25 18:49:07.211580657 +0100
@@ -74,7 +74,7 @@ setup(
     include_package_data=True,
     platforms='any',
     install_requires=[
-        'SQLAlchemy>=1.3',
+        'SQLAlchemy>=1.3,<2',
         "importlib_metadata ; python_version<'3.8'",
     ],
     extras_require=extras_require,
