From 49bad3bb6eecd589c619f1a03f0ea1ba46d8615d Mon Sep 17 00:00:00 2001
From: Yan Fu <yafu@redhat.com>
Date: Thu, 12 Jun 2025 03:27:00 -0400
Subject: [PATCH 10/14] Fedora: tempfile 3.20

Signed-off-by: Uri Lublin <uril@redhat.com>
Signed-off-by: Yan Fu <yafu@redhat.com>
---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 1b7f5b2..ac9208c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,7 +39,7 @@ serde_json = "1.0"
 serial_test = "3"
 sha2 = "0.10.9"
 strum = { version = "0.27", features = ["derive"] }
-tempfile = "3.19"
+tempfile = "3.20"
 thiserror = "2.0"
 tokio = "1.44"
 toml = "0.8.22"
-- 
2.49.0

