From 36df303ab5266bdd42ef79cdae7160f59edca587 Mon Sep 17 00:00:00 2001
From: Yan Fu <yafu@redhat.com>
Date: Thu, 30 Oct 2025 04:00:34 -0400
Subject: [PATCH 11/15] Fedora: kbs-types: 0.13.0

Signed-off-by: Uri Lublin <uril@redhat.com>
Signed-off-by: Yan Fu <yafu@redhat.com>
---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 94a0b61..41aa3d5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,7 @@ const_format = "0.2.34"
 env_logger = "0.11.8"
 hex = "0.4.3"
 hmac = "0.12.1"
-kbs-types = { "git" = "https://github.com/confidential-containers/kbs-types.git", rev = "900aa8f" }
+kbs-types = "0.13.0"
 log = "0.4.28"
 nix = "0.30"
 openssl = "0.10"
-- 
2.51.1

