--- normpath-1.5.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ normpath-1.5.0/Cargo.toml	2025-10-22T12:24:02.691103+00:00
@@ -60,8 +60,6 @@
 
 [features]
 localization = [
-    "windows-sys/Win32_UI_Shell",
-    "windows-sys/Win32_UI_WindowsAndMessaging",
 ]
 
 [lib]
@@ -90,13 +88,3 @@
 [target."cfg(not(windows))".dev-dependencies.libc]
 version = "0.2"
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.61"
-features = ["Win32_Storage_FileSystem"]
-
-[target."cfg(windows)".dev-dependencies.normpath_macros]
-version = "=1.5.0"
-
-[target."cfg(windows)".dev-dependencies.windows-sys]
-version = "0.61"
-features = ["Win32_Foundation"]
