From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Miro=20Hron=C4=8Dok?= <miro@hroncok.cz>
Date: Fri, 25 Apr 2025 09:33:37 +0200
Subject: 00461: Downstream only: Install wheel in test venvs when setuptools <
 71

---
 Lib/test/support/__init__.py | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/Lib/test/support/__init__.py b/Lib/test/support/__init__.py
index e49e3668a3..4c42234ccc 100644
--- a/Lib/test/support/__init__.py
+++ b/Lib/test/support/__init__.py
@@ -2329,9 +2329,18 @@ def setup_venv_with_pip_setuptools(venv_dir):
         else:
             python = os.path.join(venv, 'bin', python_exe)
 
+        setuptools_whl = _findwheel('setuptools')
+        whl_filename = os.path.basename(setuptools_whl)
+        setuptools_major = int(whl_filename.split('-')[1].split('.')[0])
+        if setuptools_major >= 71:  # we need 70.1+, but that's OK
+            wheels = (setuptools_whl,)
+        else:
+            wheels = (setuptools_whl, _findwheel('wheel'))
+
         cmd = (python, '-X', 'dev',
                '-m', 'pip', 'install',
-               _findwheel('setuptools'))
+               *wheels,
+               )
         if verbose:
             print()
             print('Run:', ' '.join(cmd))
