--- aspell-wa-0.4.15.orig/Makefile	2008-09-20 16:46:39.000000000 +0100
+++ aspell-wa-0.4.15/Makefile	2008-09-20 17:00:02.000000000 +0100
@@ -27,7 +27,7 @@
 	buildhash dj.lst walon.aff dj.hash
 
 walon_lst:
-	if [ ! -r walon.lst -o -n "`find mots* -type f -newer walon.lst`" ] ; \
+	if [ ! -r walon.lst -o -n "`find mots* -type f -newer walon.lst > /dev/null 2>&1`" ] ; \
 	then \
 	  grep -v '^#' mots*/* ${EXTRA} |cut -d: -f2- |cut -d'	' -f1|\
 	  tr ' ' '\n'|(LC_ALL=C sort -u) | grep -v '^$$' > walon.lst ; \
@@ -119,7 +119,7 @@
 	aspell --local-data-dir=. --lang=wa@ni create master ./wa@ni.rws < wa.lst
 
 myspell: walon_lst
-	wc -l walon.lst > wa.dic
+	cat walon.lst | wc -l > wa.dic
 	cat walon.lst >> wa.dic
 	# myspell is unable to handle comments and tabs :-(
 	grep -v '^#' wa_affix.dat | sed 's/	/    /g' > wa.aff
