--- tracing-0.1.43/Cargo.toml	1970-01-01T00:00:01+00:00
+++ tracing-0.1.43/Cargo.toml	2026-01-09T19:06:36.034222+00:00
@@ -82,7 +82,6 @@
 release_max_level_trace = []
 release_max_level_warn = []
 std = ["tracing-core/std"]
-valuable = ["tracing-core/valuable"]
 
 [lib]
 name = "tracing"
@@ -164,55 +163,6 @@
 name = "subscriber"
 path = "tests/subscriber.rs"
 
-[[bench]]
-name = "baseline"
-path = "benches/baseline.rs"
-harness = false
-
-[[bench]]
-name = "dispatch_get_clone"
-path = "benches/dispatch_get_clone.rs"
-harness = false
-
-[[bench]]
-name = "dispatch_get_ref"
-path = "benches/dispatch_get_ref.rs"
-harness = false
-
-[[bench]]
-name = "empty_span"
-path = "benches/empty_span.rs"
-harness = false
-
-[[bench]]
-name = "enter_span"
-path = "benches/enter_span.rs"
-harness = false
-
-[[bench]]
-name = "event"
-path = "benches/event.rs"
-harness = false
-
-[[bench]]
-name = "shared"
-path = "benches/shared.rs"
-
-[[bench]]
-name = "span_fields"
-path = "benches/span_fields.rs"
-harness = false
-
-[[bench]]
-name = "span_no_fields"
-path = "benches/span_no_fields.rs"
-harness = false
-
-[[bench]]
-name = "span_repeated"
-path = "benches/span_repeated.rs"
-harness = false
-
 [dependencies.log]
 version = "0.4.17"
 optional = true
@@ -226,10 +176,6 @@
 
 [dependencies.tracing-core]
 version = "0.1.35"
-default-features = false
-
-[dev-dependencies.criterion]
-version = "0.3.6"
 default-features = false
 
 [dev-dependencies.futures]
