--- piper-0.2.5/Cargo.toml	2006-07-24T01:21:28+00:00
+++ piper-0.2.5/Cargo.toml	2026-03-20T12:14:41.023605+00:00
@@ -49,11 +49,6 @@
 name = "pipe"
 path = "tests/pipe.rs"
 
-[[bench]]
-name = "pipe_comparison"
-path = "benches/pipe_comparison.rs"
-harness = false
-
 [dependencies.atomic-waker]
 version = "1.1.0"
 
@@ -85,11 +80,6 @@
 [dev-dependencies.async-io]
 version = "2.0.0"
 
-[dev-dependencies.criterion]
-version = "0.8"
-features = ["cargo_bench_support"]
-default-features = false
-
 [dev-dependencies.easy-parallel]
 version = "3.2.0"
 
