From ed0b1ecb84c0de69e7327dae6b8feee6abba551c Mon Sep 17 00:00:00 2001
From: Sharath Srikanth Chellappa <sharathsr@microsoft.com>
Date: Mon, 5 Aug 2024 11:45:42 -0700
Subject: [PATCH] Updating hotplugdetachmentGracePeriod patch with repeated lookups for disk detachment

---
 pkg/virt-launcher/virtwrap/manager.go | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/pkg/virt-launcher/virtwrap/manager.go b/pkg/virt-launcher/virtwrap/manager.go
index 63d8d43d29..eab8e2e1f5 100644
--- a/pkg/virt-launcher/virtwrap/manager.go
+++ b/pkg/virt-launcher/virtwrap/manager.go
@@ -935,6 +935,33 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, allowEmul
 			logger.Reason(err).Error("detaching device")
 			return nil, err
 		}
+		logger.V(1).Infof("Checking if disk %s, target %s is still attached", detachDisk.Alias.GetName(), detachDisk.Target.Device)
+		// Check if the disk is still attached
+		for i := 0; i < 5; i++ {
+			logger.V(1).Infof("Loop count: %d, domain name: %s", i, detachDisk.Alias.GetName())
+			dom, err = l.virConn.LookupDomainByName(domain.Spec.Name))
+			if err != nil {
+				logger.Reason(err).Error("getting domain after device detach")
+				return
+			}
+			// Check if the domain still has the detach disk name and detach disk target
+			xmlDesc, err := dom.GetXMLDesc(0)
+			if err != nil {
+				logger.Reason(err).Error("getting domain XML after device detach")
+				return
+			}
+
+			logger.V(1).Infof("Checking if disk %s, target %s is still present in the domain", detachDiskName, detachDiskTargetDevice)
+			logger.V(1).Infof("Domain XML: %s", xmlDesc)
+			if strings.Contains(xmlDesc, detachDiskName) && strings.Contains(xmlDesc, detachDiskTargetDevice) {
+				logger.V(1).Infof("Disk %s with target %s is still present in the domain", detachDiskName, detachDiskTargetDevice)
+			} else {
+				logger.V(1).Infof("Disk %s with target %s has been successfully detached", detachDiskName, detachDiskTargetDevice)
+				dom.Free()
+				break
+			}
+			time.Sleep(1 * time.Second)
+		}
 	}
 	// Look up all the disks to attach
 	for _, attachDisk := range getAttachedDisks(oldSpec.Devices.Disks, domain.Spec.Devices.Disks) {
-- 
2.45.2

