diff -Naur expect5.45.4/exp_clib.c expect5.45.4.mod/exp_clib.c
--- expect5.45.4/exp_clib.c	2018-02-02 11:15:52.000000000 -0800
+++ expect5.45.4.mod/exp_clib.c	2020-05-31 18:42:49.259968532 -0700
@@ -1476,8 +1476,8 @@
      char *str;
 {
   if (exp_is_debugging) {
-    fprintf(stderr,str);
-    if (exp_logfile) fprintf(exp_logfile,str);
+    fprintf(stderr,"%s",str);
+    if (exp_logfile) fprintf(exp_logfile,"%s",str);
   }
 }
 
