--- sequoia-sq-1.4.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ sequoia-sq-1.4.0/Cargo.toml	2026-07-08T08:28:11.296604+00:00
@@ -49,6 +49,12 @@
 ]
 license = "LGPL-2.0-or-later"
 repository = "https://gitlab.com/sequoia-pgp/sequoia-sq"
+exclude = [
+    "/deny.toml",
+    "/Dockerfile",
+    "/sq.subplot",
+    "/sq-subplot.md",
+]
 
 [badges.gitlab]
 repository = "sequoia-pgp/sequoia-sq"
@@ -57,20 +63,9 @@
 status = "actively-developed"
 
 [features]
-allow-experimental-crypto = ["sequoia-openpgp/allow-experimental-crypto"]
-allow-variable-time-crypto = ["sequoia-openpgp/allow-variable-time-crypto"]
-crypto-botan = ["sequoia-openpgp/crypto-botan"]
-crypto-botan2 = ["sequoia-openpgp/crypto-botan2"]
-crypto-cng = ["sequoia-openpgp/crypto-cng"]
 crypto-nettle = ["sequoia-openpgp/crypto-nettle"]
 crypto-openssl = ["sequoia-openpgp/crypto-openssl"]
-crypto-rust = ["sequoia-openpgp/crypto-rust"]
-default = ["crypto-nettle"]
-subplot = [
-    "culpa",
-    "subplot-build",
-    "subplotlib",
-]
+default = ["crypto-openssl"]
 
 [[bin]]
 name = "sq"
@@ -80,11 +75,6 @@
 [[test]]
 name = "integration"
 path = "tests/integration.rs"
-
-[[test]]
-name = "sq-subplot"
-path = "tests/sq-subplot.rs"
-required-features = ["subplot"]
 
 [dependencies.aho-corasick]
 version = "1"
@@ -113,10 +103,6 @@
 version = ">=0.7, <2"
 default-features = false
 
-[dependencies.culpa]
-version = "1"
-optional = true
-
 [dependencies.dirs]
 version = ">=5, <7"
 
@@ -197,10 +183,6 @@
 [dependencies.serde]
 version = "1.0.137"
 features = ["derive"]
-
-[dependencies.subplotlib]
-version = ">=0.11, <0.14"
-optional = true
 
 [dependencies.tempfile]
 version = "3.1"
@@ -303,10 +285,6 @@
 version = "1.0.137"
 features = ["derive"]
 
-[build-dependencies.subplot-build]
-version = ">=0.11, <0.14"
-optional = true
-
 [build-dependencies.terminal_size]
 version = ">=0.2.6, <0.5"
 
