diff --git a/Makefile.in b/Makefile.in
index f293649..e5d2b1a 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -518,6 +518,7 @@ distcleancheck_listfiles = find . -type f -print
 
 # These are needed due to a bug in Automake
 pkgdatadir = @pkgdatadir@
+pkgdatadir2 = @pkgdatadir2@
 pkglibdir = @pkglibdir@
 ACLOCAL = @ACLOCAL@
 AMTAR = @AMTAR@
@@ -2071,7 +2072,7 @@ gen/dirs.h: gen/mk-dirs_h.pl
 	perl ${srcdir}/gen/mk-dirs_h.pl ${prefix} ${pkgdatadir} ${pkglibdir}  ${sysconfdir} >  gen/dirs.h
 
 scripts/run-with-aspell: scripts/run-with-aspell.create
-	sh ${srcdir}/scripts/run-with-aspell.create ${pkgdatadir} > scripts/run-with-aspell
+	sh ${srcdir}/scripts/run-with-aspell.create ${pkgdatadir2} > scripts/run-with-aspell
 	chmod 755 scripts/run-with-aspell
 @PSPELL_COMPATIBILITY_TRUE@scripts/pspell-config: scripts/mkconfig
 @PSPELL_COMPATIBILITY_TRUE@	sh ${srcdir}/scripts/mkconfig ${VERSION} ${datadir} ${pkgdatadir}
diff --git a/configure b/configure
index 1236c90..8a6e697 100755
--- a/configure
+++ b/configure
@@ -709,6 +709,7 @@ CXXFLAGS
 CXX
 pkglibdir
 pkgdatadir
+pkgdatadir2
 pkgdocdir
 MAINT
 MAINTAINER_MODE_FALSE
@@ -3297,18 +3298,21 @@ pkgdatadir=undef
 # Check whether --enable-pkgdatadir was given.
 if test "${enable_pkgdatadir+set}" = set; then :
   enableval=$enable_pkgdatadir; pkgdatadir=$enable_pkgdatadir
+  pkgdatadir2=$enable_pkgdatadir
 fi
 
 
 # Check whether --enable-pkgdata-dir was given.
 if test "${enable_pkgdata_dir+set}" = set; then :
   enableval=$enable_pkgdata_dir; pkgdatadir=$enable_dict_dir
+  pkgdatadir2=$enable_dict_dir
 fi
 
 
 if test "$pkgdatadir" = "undef"
 then
   pkgdatadir=\${libdir}/aspell-0.60
+  pkgdatadir2=${exec_prefix}/lib/aspell-0.60:${exec_prefix}/lib64/aspell-0.60
 fi
 
 
