diff --git a/rhnlib.spec b/rhnlib.spec
index bcd5dc0..16e901c 100644
--- a/rhnlib.spec
+++ b/rhnlib.spec
@@ -7,7 +7,7 @@
 Summary: Python libraries for the Spacewalk project
 Name: rhnlib
 Version: 2.8.6
-Release: 1%{?dist}
+Release: 2%{?dist}
 URL:     https://github.com/spacewalkproject/spacewalk
 Source0: https://github.com/spacewalkproject/spacewalk/archive/%{name}-%{version}.tar.gz
 
@@ -87,6 +87,9 @@ make -f Makefile.rhnlib
 %endif
 
 %changelog
+* Mon Feb 19 2018 Tomas Kasparek <tkasparek@redhat.com> 2.8.6-2
+- rebuild for rhel8
+
 * Fri Feb 09 2018 Michael Mraka <michael.mraka@redhat.com> 2.8.6-1
 - remove install/clean section initial cleanup
 - removed Group from specfile
