diff --git a/rhn/connections.py b/rhn/connections.py
index df53d38..66ae4d9 100644
--- a/rhn/connections.py
+++ b/rhn/connections.py
@@ -158,6 +158,9 @@ class HTTPProxyConnection(HTTPConnection):
         enc_userpass = base64.encodestring(userpass).replace("\n", "")
         self.putheader("Proxy-Authorization", "Basic %s" % enc_userpass)
 
+    def _set_hostport(self, host, port):
+        (self.host, self.port) = self._get_hostport(host, port)
+
 class HTTPSConnection(HTTPConnection):
     response_class = HTTPResponse
     default_port = httplib.HTTPSConnection.default_port
@@ -261,4 +264,4 @@ def idn_ascii_to_puny(hostname):
         return None
     else:
         hostname = i18n.ustr(hostname)
-        return hostname.encode('idna')
+        return i18n.ustr(hostname.encode('idna'))
diff --git a/rhnlib.spec b/rhnlib.spec
index 1d197e9..36bf3dc 100644
--- a/rhnlib.spec
+++ b/rhnlib.spec
@@ -7,7 +7,7 @@
 Summary: Python libraries for the Spacewalk project
 Name: rhnlib
 Version: 2.8.6
-Release: 5%{?dist}
+Release: 6%{?dist}
 URL:     https://github.com/spacewalkproject/spacewalk
 Source0: https://github.com/spacewalkproject/spacewalk/archive/%{name}-%{version}.tar.gz
 
@@ -74,6 +74,9 @@ make -f Makefile.rhnlib
 %endif
 
 %changelog
+* Wed Dec 19 2018 Michael Mraka <michael.mraka@redhat.com> 2.8.6-6
+- Resolves: #1652859 - python3 http.client does not contain _set_hostport()
+
 * Fri Nov 02 2018 Tomas Kasparek <tkasparek@redhat.com> 2.8.6-5
 - Resolves: #1643415 - forbid old SSL versions during negotiation
   (tkasparek@redhat.com)
