diff --git a/rhnlib.spec b/rhnlib.spec
index 16e901c..f5eb7fd 100644
--- a/rhnlib.spec
+++ b/rhnlib.spec
@@ -1,4 +1,4 @@
-%if 0%{?fedora} || 0%{?suse_version} > 1320
+%if 0%{?fedora} || 0%{?suse_version} > 1320 || 0%{?rhel} >= 8
 %global build_py3   1
 %endif
 
@@ -7,7 +7,7 @@
 Summary: Python libraries for the Spacewalk project
 Name: rhnlib
 Version: 2.8.6
-Release: 2%{?dist}
+Release: 3%{?dist}
 URL:     https://github.com/spacewalkproject/spacewalk
 Source0: https://github.com/spacewalkproject/spacewalk/archive/%{name}-%{version}.tar.gz
 
@@ -87,6 +87,9 @@ make -f Makefile.rhnlib
 %endif
 
 %changelog
+* Tue Feb 20 2018 Tomas Kasparek <tkasparek@redhat.com> 2.8.6-3
+- rhel8 utilizes python3 (tkasparek@redhat.com)
+
 * Mon Feb 19 2018 Tomas Kasparek <tkasparek@redhat.com> 2.8.6-2
 - rebuild for rhel8
 
