diff -Naur ./go/src/cmd/dist/unix.c ./go_/src/cmd/dist/unix.c
--- ./go/src/cmd/dist/unix.c	2017-11-21 17:33:56.000000000 -0800
+++ ./go_/src/cmd/dist/unix.c	2020-02-05 22:40:39.501947515 -0800
@@ -706,7 +706,7 @@
 			gohostarch = "amd64";
 		else if(hassuffix(u.machine, "86"))
 			gohostarch = "386";
-		else if(contains(u.machine, "arm"))
+		else if(contains(u.machine, "arm") || contains(u.machine, "aarch64"))
 			gohostarch = "arm";
 		else
 			fatal("unknown architecture: %s", u.machine);
