diff -ur tss2-1234/utils/policymaker.c tss2-1234-new/utils/policymaker.c
--- tss2-1234/utils/policymaker.c	2018-10-18 12:16:59.742439220 -0700
+++ tss2-1234-new/utils/policymaker.c	2018-10-18 12:34:34.991755536 -0700
@@ -208,10 +208,10 @@
 	}
 	/* hash extend */
 	if ((rc == 0) && (prc != NULL)) {
-	    TSS_Hash_Generate(&digest,
-			      startSizeInBytes, (uint8_t *)&digest.digest,	/* extend */
-			      lineLength /2, lineBinary,
-			      0, NULL);
+	    rc = TSS_Hash_Generate(&digest,
+				   startSizeInBytes, (uint8_t *)&digest.digest,	/* extend */
+				   lineLength /2, lineBinary,
+				   0, NULL);
 	}
 	if ((rc == 0) && (prc != NULL)) {
 	    if (verbose) TSS_PrintAll("intermediate policy digest",
