diff -Nrup a/utils/makefile.fedora b/utils/makefile.fedora
--- a/utils/makefile.fedora	2019-12-19 16:26:06.695476012 -0700
+++ b/utils/makefile.fedora	2019-12-19 16:25:58.153501665 -0700
@@ -181,6 +181,8 @@ tsscryptoh.o: 	$(TSS_HEADERS) tsscryptoh
 		$(CC) $(CCFLAGS) $(CCLFLAGS) tsscryptoh.c
 tsscrypto.o: 	$(TSS_HEADERS) tsscrypto.c
 		$(CC) $(CCFLAGS) $(CCLFLAGS) tsscrypto.c
+tssprintcmd.o: 	$(TSS_HEADERS) tssprintcmd.c
+		$(CC) $(CCFLAGS) $(CCLFLAGS) tssprintcmd.c
 tssutils.o: 	$(TSS_HEADERS) tssutils.c
 		$(CC) $(CCFLAGS) $(CCLFLAGS) tssutils.c
 tssfile.o: 	$(TSS_HEADERS) tssfile.c
