diff -ruN a/fluent-plugin-td.gemspec b/fluent-plugin-td.gemspec
--- a/fluent-plugin-td.gemspec	2022-07-07 16:51:04.391312214 -0700
+++ b/fluent-plugin-td.gemspec	2022-07-07 16:53:52.009195355 -0700
@@ -11,9 +11,9 @@
   gem.authors     = ["Treasure Data, Inc."]
   gem.email       = "support@treasure-data.com"
   #gem.platform    = Gem::Platform::RUBY
-  gem.files       = `git ls-files`.split("\n")
-  gem.test_files  = `git ls-files -- {test,spec,features}/*`.split("\n")
-  gem.executables = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
+  gem.files       = Dir['**/*']
+  gem.test_files  = gem.files.grep(%r{^(test|spec|features)/}) 
+  gem.executables = gem.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   gem.require_paths = ['lib']
   gem.license       = "Apache-2.0"
 
