--- hyper-rustls-0.27.7/Cargo.toml	1970-01-01T00:00:01+00:00
+++ hyper-rustls-0.27.7/Cargo.toml	2025-07-17T13:51:10.956688+00:00
@@ -45,17 +45,12 @@
 ]
 
 [features]
-aws-lc-rs = ["rustls/aws_lc_rs"]
 default = [
     "native-tokio",
     "http1",
+    "ring",
     "tls12",
     "logging",
-    "aws-lc-rs",
-]
-fips = [
-    "aws-lc-rs",
-    "rustls/fips",
 ]
 http1 = ["hyper-util/http1"]
 http2 = ["hyper-util/http2"]
@@ -87,7 +82,7 @@
 [[example]]
 name = "server"
 path = "examples/server.rs"
-required-features = ["aws-lc-rs"]
+required-features = ["ring"]
 
 [[test]]
 name = "tests"
