From 6b2a33044583e892badb9ac86cd2c6252b1a532f Mon Sep 17 00:00:00 2001
From: Dmitry Belyavskiy <beldmit@gmail.com>
Date: Thu, 15 May 2025 13:43:28 +0200
Subject: [PATCH 08/50] openssh-5.1p1-askpass-progress

---
 contrib/gnome-ssh-askpass2.c | 39 +++++++++++++++++++++++++++++++++---
 1 file changed, 36 insertions(+), 3 deletions(-)

diff --git a/contrib/gnome-ssh-askpass2.c b/contrib/gnome-ssh-askpass2.c
index a62f9815..cb7152dc 100644
--- a/contrib/gnome-ssh-askpass2.c
+++ b/contrib/gnome-ssh-askpass2.c
@@ -58,6 +58,7 @@
 #include <unistd.h>
 
 #include <X11/Xlib.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkx.h>
 #include <gdk/gdkkeysyms.h>
@@ -146,6 +147,17 @@ parse_env_hex_color(const char *env, GdkColor *c)
 	return 1;
 }
 
+static void
+move_progress(GtkWidget *entry, gpointer progress)
+{
+	gdouble step;
+	g_return_if_fail(GTK_IS_PROGRESS_BAR(progress));
+	
+	step = g_random_double_range(0.03, 0.1);
+	gtk_progress_bar_set_pulse_step(GTK_PROGRESS_BAR(progress), step);
+	gtk_progress_bar_pulse(GTK_PROGRESS_BAR(progress));
+}
+
 static int
 passphrase_dialog(char *message, int prompt_type)
 {
@@ -153,7 +165,7 @@ passphrase_dialog(char *message, int prompt_type)
 	char *passphrase, *local;
 	int result, grab_tries, grab_server, grab_pointer;
 	int buttons, default_response;
-	GtkWidget *parent_window, *dialog, *entry;
+	GtkWidget *parent_window, *dialog, *entry, *progress, *hbox;
 	GdkGrabStatus status;
 	GdkColor fg, bg;
 	int fg_set = 0, bg_set = 0;
@@ -199,14 +211,19 @@ passphrase_dialog(char *message, int prompt_type)
 		gtk_widget_modify_bg(dialog, GTK_STATE_NORMAL, &bg);
 
 	if (prompt_type == PROMPT_ENTRY || prompt_type == PROMPT_NONE) {
+		hbox = gtk_hbox_new(FALSE, 0);
+		gtk_box_pack_start(GTK_BOX(GTK_DIALOG(dialog)->vbox), hbox, FALSE,
+		    FALSE, 0);
+		gtk_widget_show(hbox);
+
 		entry = gtk_entry_new();
 		if (fg_set)
 			gtk_widget_modify_fg(entry, GTK_STATE_NORMAL, &fg);
 		if (bg_set)
 			gtk_widget_modify_bg(entry, GTK_STATE_NORMAL, &bg);
 		gtk_box_pack_start(
-		    GTK_BOX(gtk_dialog_get_content_area(GTK_DIALOG(dialog))),
-		    entry, FALSE, FALSE, 0);
+		    GTK_BOX(hbox), entry, TRUE, FALSE, 0);
+		gtk_entry_set_width_chars(GTK_ENTRY(entry), 2);
 		gtk_entry_set_visibility(GTK_ENTRY(entry), FALSE);
 		gtk_widget_grab_focus(entry);
 		if (prompt_type == PROMPT_ENTRY) {
@@ -225,6 +242,22 @@ passphrase_dialog(char *message, int prompt_type)
 			g_signal_connect(G_OBJECT(entry), "key_press_event",
 			    G_CALLBACK(check_none), dialog);
 		}
+
+		hbox = gtk_hbox_new(FALSE, 0);
+		gtk_box_pack_start(GTK_BOX(GTK_DIALOG(dialog)->vbox),
+		    hbox, FALSE, FALSE, 8);
+		gtk_widget_show(hbox);
+
+		progress = gtk_progress_bar_new();
+
+		gtk_progress_bar_set_text(GTK_PROGRESS_BAR(progress),
+		    "Passphrase length hidden intentionally");
+		gtk_box_pack_start(GTK_BOX(hbox), progress, TRUE,
+		    TRUE, 5);
+		gtk_widget_show(progress);
+		g_signal_connect(G_OBJECT(entry), "changed",
+				 G_CALLBACK(move_progress), progress);
+
 	}
 
 	/* Grab focus */
-- 
2.49.0

