diff -ruN a/public_suffix.gemspec b/public_suffix.gemspec
--- a/public_suffix.gemspec	2020-09-02 02:30:59.000000000 -0700
+++ b/public_suffix.gemspec	2022-07-03 14:47:37.815903327 -0700
@@ -23,7 +23,7 @@
   s.required_ruby_version = ">= 2.3"
 
   s.require_paths    = ["lib"]
-  s.files            = `git ls-files`.split("\n")
-  s.test_files       = `git ls-files -- {test,spec,features}/*`.split("\n")
+  s.files            = Dir['**/*']
+  s.test_files       = s.files.grep(%r{^(test|spec|features)/})
   s.extra_rdoc_files = %w( LICENSE.txt )
 end
