diff --git a/Makefile b/Makefile
index 37a8673..888feea 100644
--- a/Makefile
+++ b/Makefile
@@ -16,6 +16,8 @@
 # along with this program; if not, write to the Free Software Foundation,
 # Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. 
 
+DESTDIR ?= 
+
 prefix=/usr/local
 
 bin_dir=$(prefix)/bin
@@ -29,9 +31,9 @@ man8_dir=$(man_dir)/man8
 
 etc_dir=/etc/authbind
 
-INSTALL_FILE	?= install -o root -g root -m 644 
-INSTALL_PROGRAM ?= install -o root -g root -m 755 -s
-INSTALL_DIR	?= install -o root -g root -m 755 -d
+INSTALL_FILE	?= install -m 644 -p
+INSTALL_PROGRAM ?= install -m 755 -p
+INSTALL_DIR	?= install -m 755 -p -d
 STRIP		?= strip
 
 OPTIMISE=	-O2
@@ -59,19 +61,19 @@ MANPAGES_8=		authbind-helper.8
 all:			$(TARGETS)
 
 install:		$(TARGETS)
-		$(INSTALL_DIR) $(lib_dir) $(man1_dir) $(man8_dir)
-		$(INSTALL_PROGRAM) $(BINTARGETS) $(bin_dir)/.
-		$(INSTALL_FILE) $(LIBTARGET) $(lib_dir)/.
-		$(STRIP) --strip-unneeded $(lib_dir)/$(LIBTARGET)
-		ln -sf $(LIBTARGET) $(lib_dir)/$(LIBCANON)
-		$(INSTALL_PROGRAM) $(HELPER) $(libexec_dir)/.
-		chmod u+s $(libexec_dir)/$(HELPER)
-		$(INSTALL_DIR) $(etc_dir) \
-			$(etc_dir)/byport $(etc_dir)/byaddr $(etc_dir)/byuid
+		$(INSTALL_DIR) $(DESTDIR)$(bin_dir) $(DESTDIR)$(lib_dir) $(DESTDIR)$(libexec_dir) $(DESTDIR)$(man1_dir) $(DESTDIR)$(man8_dir)
+		$(INSTALL_PROGRAM) $(BINTARGETS) $(DESTDIR)$(bin_dir)/.
+		$(INSTALL_PROGRAM) $(LIBTARGET) $(DESTDIR)$(lib_dir)/.
+		$(STRIP) --strip-unneeded $(DESTDIR)$(lib_dir)/$(LIBTARGET)
+		ln -sf $(LIBTARGET) $(DESTDIR)$(lib_dir)/$(LIBCANON)
+		$(INSTALL_PROGRAM) $(HELPER) $(DESTDIR)$(libexec_dir)/.
+		chmod u+s $(DESTDIR)$(libexec_dir)/$(HELPER)
+		$(INSTALL_DIR) $(DESTDIR)$(etc_dir) \
+			$(DESTDIR)$(etc_dir)/byport $(DESTDIR)$(etc_dir)/byaddr $(DESTDIR)$(etc_dir)/byuid
 
 install_man:		$(MANPAGES_1) $(MANPAGES_8)
-		$(INSTALL_FILE) $(MANPAGES_1) $(man1_dir)/.
-		$(INSTALL_FILE) $(MANPAGES_8) $(man8_dir)/.
+		$(INSTALL_FILE) $(MANPAGES_1) $(DESTDIR)$(man1_dir)/.
+		$(INSTALL_FILE) $(MANPAGES_8) $(DESTDIR)$(man8_dir)/.
 
 libauthbind.o:		libauthbind.c authbind.h
 		$(CC) -D_REENTRANT $(CFLAGS) $(CPPFLAGS) -c -o $@ -fPIC $<
@@ -82,7 +84,7 @@ helper:			helper.o
 helper.o authbind.o:	authbind.h
 
 $(LIBTARGET):		libauthbind.o
-		$(LD) -shared -soname $(LIBCANON) -o $@ $< $(LIBS)
+		$(CC) -nostartfiles -shared -Wl,-soname $(LIBCANON) $(LDFLAGS) -o $@ $< $(LIBS)
 
 clean distclean:
 		rm -f $(TARGETS) *.o *~ ./#*# *.bak *.new core
