--- tox.ini~	2020-01-17 12:16:20.000000000 -0500
+++ tox.ini	2020-01-17 12:16:53.631039837 -0500
@@ -14,7 +14,7 @@
 deps =
   pypy: backports.functools_lru_cache
   pypy: enum34
-  lazy-object-proxy==1.4.*
+  lazy-object-proxy
   ; we have a brain for nose
   ; we use pytest for tests
   nose
--- tox.ini~	2020-01-17 12:19:13.000000000 -0500
+++ tox.ini	2020-01-17 12:19:37.378500415 -0500
@@ -25,7 +25,7 @@
   python-dateutil
   pypy: singledispatch
   six~=1.12
-  wrapt==1.11.*
+  wrapt
   coverage
 
 setenv =
--- astroid/__pkginfo__.py~	2019-11-06 13:35:12.000000000 -0500
+++ astroid/__pkginfo__.py	2020-01-17 12:21:47.169072852 -0500
@@ -22,9 +22,9 @@
 
 extras_require = {}
 install_requires = [
-    "lazy_object_proxy==1.4.*",
+    "lazy_object_proxy",
     "six~=1.12",
-    "wrapt==1.11.*",
+    "wrapt",
     'typed-ast>=1.4.0,<1.5;implementation_name== "cpython" and python_version<"3.8"',
 ]
 
