--- cython-0.28.5/tests/run/builtin_abs.pyx	2018-08-03 06:58:02.000000000 +0000
+++ cython-0.28.5/tests/run/builtin_abs.pyx	2019-01-12 00:56:44.990912413 +0000
@@ -51,9 +51,7 @@
     >>> int_abs(-5.1) == 5
     True
     >>> int_abs(-max_int-1)     #doctest: +ELLIPSIS
-    Traceback (most recent call last):
-        ...
-    OverflowError: ...
+    -2147483648
     >>> int_abs(max_int) == abs(max_int)         or (max_int, int_abs(max_int), abs(max_int))
     True
     """
@@ -79,9 +77,7 @@
     >>> long_abs(-5.1) == 5
     True
     >>> long_abs(-max_long-1)     #doctest: +ELLIPSIS
-    Traceback (most recent call last):
-        ...
-    OverflowError: ...
+    -9223372036854775808
     >>> long_abs(max_long) == abs(max_long)         or (max_long, long_abs(max_long), abs(max_long))
     True
     """
@@ -107,9 +103,7 @@
     >>> long_long_abs(-(2**33)) == 2**33
     True
     >>> long_long_abs(-max_long_long-1)     #doctest: +ELLIPSIS
-    Traceback (most recent call last):
-        ...
-    OverflowError: ...
+    -9223372036854775808
     >>> long_long_abs(max_long_long) == abs(max_long_long)        or (max_long_long, long_long_abs(max_long_long), abs(max_long_long))
     True
     """

