From ac6be33c89e2710b1bab37e1d1950835e85703d7 Mon Sep 17 00:00:00 2001
From: MaxBrodeurUrbas <murbas98@gmail.com>
Date: Thu, 20 Jan 2022 16:05:04 -0800
Subject: [PATCH] freelocale bug fix

---
 libparted/labels/atari.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libparted/labels/atari.c b/libparted/labels/atari.c
index 2ac03d2..47ce12e 100644
--- a/libparted/labels/atari.c
+++ b/libparted/labels/atari.c
@@ -1971,5 +1971,5 @@ void
 ped_disk_atari_done ()
 {
 	ped_disk_type_unregister (&atari_disk_type);
-	freelocale(atr_c_locale);
+	PED_ASSERT (freelocale(atr_c_locale));
 }
-- 
2.17.1

