--- cdparanoia-III-10.2/utils.h
+++ cdparanoia-III-10.2/utils.h
@@ -86,7 +86,7 @@
 
 static inline char *copystring(const char *s){
   if(s){
-    char *ret=malloc((strlen(s)+1)*sizeof(char));
+    char *ret=(char *)malloc((strlen(s)+1)*sizeof(char));
     strcpy(ret,s);
     return(ret);
   }
@@ -96,9 +96,9 @@
 static inline char *catstring(char *buff,const char *s){
   if(s){
     if(buff)
-      buff=realloc(buff,strlen(buff)+strlen(s)+1);
+      buff=(char *)realloc(buff,strlen(buff)+strlen(s)+1);
     else
-      buff=calloc(strlen(s)+1,1);
+      buff=(char *)calloc(strlen(s)+1,1);
     strcat(buff,s);
   }
   return(buff);
