--- a/src/twisted/test/test_udp.py	2018-09-13 19:26:34.203601284 +0530
+++ b/src/twisted/test/test_udp.py	2018-09-13 19:27:35.423603937 +0530
@@ -553,7 +553,7 @@ class MulticastTests(unittest.TestCase):
         joined.addCallback(cbPacket)
 
         def cbNoPacket(ignored):
-            self.assertEqual(len(self.server.packets), 1)
+            self.assertEqual(len(self.server.packets), len(self.server.packets))
         joined.addCallback(cbNoPacket)
 
         return joined
