diff -up ./ext/dba/config.m4.libdb ./ext/dba/config.m4
--- ./ext/dba/config.m4.libdb	2020-04-09 14:06:11.000000000 +0200
+++ ./ext/dba/config.m4	2020-04-09 14:35:08.208605065 +0200
@@ -375,61 +375,13 @@ if test "$PHP_DB4" != "no"; then
   dbdp4="/usr/local/BerkeleyDB.4."
   dbdp5="/usr/local/BerkeleyDB.5."
   for i in $PHP_DB4 ${dbdp5}1 ${dbdp5}0 ${dbdp4}8 ${dbdp4}7 ${dbdp4}6 ${dbdp4}5 ${dbdp4}4 ${dbdp4}3 ${dbdp4}2 ${dbdp4}1 ${dbdp}0 /usr/local /usr; do
-    if test -f "$i/db5/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/db5/db.h
-      break
-    elif test -f "$i/db4/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/db4/db.h
-      break
-    elif test -f "$i/include/db5.3/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db5.3/db.h
-      break
-    elif test -f "$i/include/db5.1/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db5.1/db.h
-      break
-    elif test -f "$i/include/db5.0/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db5.0/db.h
-      break
-    elif test -f "$i/include/db4.8/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db4.8/db.h
-      break
-    elif test -f "$i/include/db4.7/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db4.7/db.h
-      break
-    elif test -f "$i/include/db4.6/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db4.6/db.h
-      break
-    elif test -f "$i/include/db4.5/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db4.5/db.h
-      break
-    elif test -f "$i/include/db4/db.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db4/db.h
-      break
-    elif test -f "$i/include/db/db4.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db/db4.h
-      break
-    elif test -f "$i/include/db4.h"; then
-      THIS_PREFIX=$i
-      THIS_INCLUDE=$i/include/db4.h
-      break
-    elif test -f "$i/include/db.h"; then
+    if test -f "$i/include/db.h"; then
       THIS_PREFIX=$i
       THIS_INCLUDE=$i/include/db.h
       break
     fi
   done
-  PHP_DBA_DB_CHECK(4, db-5.3 db-5.1 db-5.0 db-4.8 db-4.7 db-4.6 db-4.5 db-4.4 db-4.3 db-4.2 db-4.1 db-4.0 db-4 db4 db, [(void)db_create((DB**)0, (DB_ENV*)0, 0)])
+  PHP_DBA_DB_CHECK(4, db, [(void)db_create((DB**)0, (DB_ENV*)0, 0)])
 fi
 PHP_DBA_STD_RESULT(db4,Berkeley DB4)
 
diff -up ./ext/dba/dba.c.libdb ./ext/dba/dba.c
--- ./ext/dba/dba.c.libdb	2020-04-09 14:06:11.000000000 +0200
+++ ./ext/dba/dba.c	2020-04-09 14:36:30.593275190 +0200
@@ -50,6 +50,10 @@
 #include "php_lmdb.h"
 #include "dba_arginfo.h"
 
+#ifdef DB4_INCLUDE_FILE
+#include DB4_INCLUDE_FILE
+#endif
+
 PHP_MINIT_FUNCTION(dba);
 PHP_MSHUTDOWN_FUNCTION(dba);
 PHP_MINFO_FUNCTION(dba);
@@ -452,6 +456,10 @@ PHP_MINFO_FUNCTION(dba)
 
 	php_info_print_table_start();
 	php_info_print_table_row(2, "DBA support", "enabled");
+#ifdef DB_VERSION_STRING
+ 	php_info_print_table_row(2, "libdb header version", DB_VERSION_STRING);
+ 	php_info_print_table_row(2, "libdb library version", db_version(NULL, NULL, NULL));
+#endif
 	if (handlers.s) {
 		smart_str_0(&handlers);
 		php_info_print_table_row(2, "Supported handlers", ZSTR_VAL(handlers.s));
