From db200499551e386e7616c621fcbd69e350081664 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 18 Jul 2018 13:41:02 -0400
Subject: [PATCH 6/8] Honor sbindir

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index f0d13720db5..cfa8e0d60ab 100644
--- a/Makefile
+++ b/Makefile
@@ -42,14 +42,14 @@ test: all
 	@./test.sh
 
 install: all
-	mkdir -p $(DESTDIR)$(PREFIX)/sbin
+	mkdir -p $(DESTDIR)$(PREFIX)$(sbindir)
 	mkdir -p $(DESTDIR)/$(mandir)/man8
-	install -m 755 new-kernel-pkg $(DESTDIR)$(PREFIX)/sbin
+	install -m 755 new-kernel-pkg $(DESTDIR)$(PREFIX)$(sbindir)
 	install -m 644 new-kernel-pkg.8 $(DESTDIR)/$(mandir)/man8
-	install -m 755 installkernel $(DESTDIR)$(PREFIX)/sbin
+	install -m 755 installkernel $(DESTDIR)$(PREFIX)$(sbindir)
 	install -m 644 installkernel.8 $(DESTDIR)/$(mandir)/man8
 	if [ -f grubby ]; then \
-		install -m 755 grubby $(DESTDIR)$(PREFIX)/sbin ; \
+		install -m 755 grubby $(DESTDIR)$(PREFIX)$(sbindir) ; \
 		install -m 644 grubby.8 $(DESTDIR)/$(mandir)/man8 ; \
 	fi
 
-- 
2.17.1

