From d678047ee92332c3a2055be838311a5ac0913927 Mon Sep 17 00:00:00 2001
From: Daniel McIlvaney <damcilva@microsoft.com>
Date: Sun, 11 Sep 2022 11:42:54 -0700
Subject: [PATCH] Include missing rpm 4.18.0 header rpmstring.h

---
 rpm-sort.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rpm-sort.c b/rpm-sort.c
index f196356..b048c07 100644
--- a/rpm-sort.c
+++ b/rpm-sort.c
@@ -7,6 +7,7 @@
 #include <assert.h>
 #include <argp.h>
 #include <rpm/rpmlib.h>
+#include <rpm/rpmstring.h>
 #include <err.h>
 
 typedef enum {
-- 
2.17.1

