diff -up vitess-8.0.0/go/vt/topo/k8stopo/client/clientset/versioned/typed/topo/v1beta1/vitesstoponode.go.orig vitess-8.0.0/go/vt/topo/k8stopo/client/clientset/versioned/typed/topo/v1beta1/vitesstoponode.go
--- vitess-8.0.0/go/vt/topo/k8stopo/client/clientset/versioned/typed/topo/v1beta1/vitesstoponode.go.orig	2020-10-27 14:56:23.000000000 +0100
+++ vitess-8.0.0/go/vt/topo/k8stopo/client/clientset/versioned/typed/topo/v1beta1/vitesstoponode.go	2020-12-27 16:10:25.286537516 +0100
@@ -19,6 +19,7 @@ limitations under the License.
 package v1beta1
 
 import (
+	"context"
 	"time"
 
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
@@ -70,7 +71,7 @@ func (c *vitessTopoNodes) Get(name strin
 		Resource("vitesstoponodes").
 		Name(name).
 		VersionedParams(&options, scheme.ParameterCodec).
-		Do().
+		Do(context.TODO()).
 		Into(result)
 	return
 }
@@ -87,7 +88,7 @@ func (c *vitessTopoNodes) List(opts v1.L
 		Resource("vitesstoponodes").
 		VersionedParams(&opts, scheme.ParameterCodec).
 		Timeout(timeout).
-		Do().
+		Do(context.TODO()).
 		Into(result)
 	return
 }
@@ -104,7 +105,7 @@ func (c *vitessTopoNodes) Watch(opts v1.
 		Resource("vitesstoponodes").
 		VersionedParams(&opts, scheme.ParameterCodec).
 		Timeout(timeout).
-		Watch()
+		Watch(context.TODO())
 }
 
 // Create takes the representation of a vitessTopoNode and creates it.  Returns the server's representation of the vitessTopoNode, and an error, if there is any.
@@ -114,7 +115,7 @@ func (c *vitessTopoNodes) Create(vitessT
 		Namespace(c.ns).
 		Resource("vitesstoponodes").
 		Body(vitessTopoNode).
-		Do().
+		Do(context.TODO()).
 		Into(result)
 	return
 }
@@ -127,7 +128,7 @@ func (c *vitessTopoNodes) Update(vitessT
 		Resource("vitesstoponodes").
 		Name(vitessTopoNode.Name).
 		Body(vitessTopoNode).
-		Do().
+		Do(context.TODO()).
 		Into(result)
 	return
 }
@@ -139,7 +140,7 @@ func (c *vitessTopoNodes) Delete(name st
 		Resource("vitesstoponodes").
 		Name(name).
 		Body(options).
-		Do().
+		Do(context.TODO()).
 		Error()
 }
 
@@ -155,7 +156,7 @@ func (c *vitessTopoNodes) DeleteCollecti
 		VersionedParams(&listOptions, scheme.ParameterCodec).
 		Timeout(timeout).
 		Body(options).
-		Do().
+		Do(context.TODO()).
 		Error()
 }
 
@@ -168,7 +169,7 @@ func (c *vitessTopoNodes) Patch(name str
 		SubResource(subresources...).
 		Name(name).
 		Body(data).
-		Do().
+		Do(context.TODO()).
 		Into(result)
 	return
 }
