--- jiff-0.2.37/Cargo.toml	2006-07-24T01:21:28+00:00
+++ jiff-0.2.37/Cargo.toml	2026-09-13T09:12:14.531690+00:00
@@ -70,7 +70,6 @@
 alloc = [
     "jcore/alloc",
     "serde_core?/alloc",
-    "portable-atomic-util/alloc",
     "defmt?/alloc",
 ]
 arbitrary = [
@@ -92,8 +91,6 @@
     "jcore/defmt",
 ]
 js = [
-    "dep:wasm-bindgen",
-    "dep:js-sys",
 ]
 logging = [
     "dep:log",
@@ -118,14 +115,12 @@
 ]
 tz-system = [
     "std",
-    "dep:windows-link",
 ]
 tzdb-bundle-always = [
     "dep:jiff-tzdb",
     "alloc",
 ]
 tzdb-bundle-platform = [
-    "dep:jiff-tzdb-platform",
     "alloc",
 ]
 tzdb-concatenated = ["std"]
@@ -232,37 +227,14 @@
 [dev-dependencies.walkdir]
 version = "2.5.0"
 
-[target.'cfg(all(any(target_arch = "wasm32", target_arch = "wasm64"), target_os = "unknown"))'.dependencies.js-sys]
-version = "0.3.50"
-optional = true
-
-[target.'cfg(all(any(target_arch = "wasm32", target_arch = "wasm64"), target_os = "unknown"))'.dependencies.wasm-bindgen]
-version = "0.2.70"
-optional = true
-
 [target."cfg(any())".dependencies.jiff-static]
 version = "=0.2.37"
 
-[target.'cfg(any(windows, target_family = "wasm"))'.dependencies.jiff-tzdb-platform]
-version = "0.1.3"
-optional = true
-
 [target.'cfg(not(target_family = "wasm"))'.dev-dependencies.hifitime]
 version = "4.3.0"
-
-[target.'cfg(not(target_has_atomic = "ptr"))'.dependencies.portable-atomic]
-version = "1.10.0"
-default-features = false
-
-[target.'cfg(not(target_has_atomic = "ptr"))'.dependencies.portable-atomic-util]
-version = "0.2.4"
-default-features = false
-
-[target."cfg(windows)".dependencies.windows-link]
-version = "0.2.1"
-optional = true
 
 [lints.rust.unexpected_cfgs]
 level = "allow"
 priority = 0
 check-cfg = ["cfg(docsrs_jiff)"]
+
