From 4e63491c2a9ba2d9e9f5ad22360051038918c19d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Tue, 29 Oct 2019 14:49:09 +0100
Subject: [PATCH] Correct a shell bang in data/regenerate_cert.sh
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 data/regenerate_cert.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/regenerate_cert.sh b/data/regenerate_cert.sh
index 9bf1879..8980fbe 100755
--- a/data/regenerate_cert.sh
+++ b/data/regenerate_cert.sh
@@ -1,4 +1,4 @@
-#/bin/sh
+#!/bin/sh
 # regenerate server certificate for the SSL tests
 
 if [ -e openssl.cnf ]; then
-- 
2.21.0

