--- cvs-1.11.21/src/diff.c.old	2005-05-27 19:17:03.000000000 +0200
+++ cvs-1.11.21/src/diff.c	2005-12-15 15:22:05.000000000 +0100
@@ -955,14 +955,16 @@
 	    /* The first revision does not exist.  If EMPTY_FILES is
                true, treat this as an added file.  Otherwise, warn
                about the missing tag.  */
-	    if( use_rev2 == NULL || RCS_isdead( vers->srcfile, use_rev2 ) )
+	    if( use_rev2 == NULL || RCS_isdead( vers->srcfile, use_rev2 ) ) {
 		/* At least in the case where DIFF_REV1 and DIFF_REV2
 		 * are both numeric (and non-existant (NULL), as opposed to
 		 * dead?), we should be returning some kind of error (see
 		 * basicb-8a0 in testsuite).  The symbolic case may be more
 		 * complicated.
 		 */
-		return DIFF_SAME;
+		error (0, 0, "no revision in file %s or missing file %s", finfo->fullname, finfo->fullname);
+		return DIFF_ERROR;
+	    }
 	    if( empty_files )
 		return DIFF_ADDED;
 	    if( use_rev1 != NULL )
