diff -Naur orjson-3.11.7-original/Cargo.toml orjson-3.11.7/Cargo.toml
--- orjson-3.11.7-original/Cargo.toml	2026-02-02 14:52:49.000000000 +0000
+++ orjson-3.11.7/Cargo.toml	2026-02-02 20:09:00.423175389 +0000
@@ -52,7 +52,7 @@
 itoap = { version = "1", default-features = false, features = ["std", "simd"] }
 jiff = { version = "^0.2", default-features = false }
 once_cell = { version = "1", default-features = false, features = ["alloc", "race"] }
-pyo3-ffi = { version = "0.28", default-features = false }
+pyo3-ffi = { version = ">=0.27,<0.29", default-features = false }
 serde = { version = "1", default-features = false }
 serde_json = { version = "1", default-features = false, features = ["std"] }
 simdutf8 = { version = "0.1", default-features = false, features = ["std", "public_imp", "aarch64_neon"] }
@@ -64,7 +64,7 @@
 
 [build-dependencies]
 cc = { version = "1" }
-pyo3-build-config = { version = "0.28" }
+pyo3-build-config = { version = ">=0.27,<0.29" }
 version_check = { version = "0.9" }
 
 [profile.dev]
