diff -up clamav-1.5.2/clamconf/clamconf.c.default_confs clamav-1.5.2/clamconf/clamconf.c
--- clamav-1.5.2/clamconf/clamconf.c.default_confs	2026-01-01 00:00:00.000000000 +0000
+++ clamav-1.5.2/clamconf/clamconf.c	2026-01-01 00:00:00.000000000 +0000
@@ -63,9 +63,9 @@ static struct _cfgfile {
     const char *name;
     int tool;
 } cfgfile[] = {
-    {"clamd.conf", OPT_CLAMD},
+    {"clamd.d/scan.conf", OPT_CLAMD},
     {"freshclam.conf", OPT_FRESHCLAM},
-    {"clamav-milter.conf", OPT_MILTER},
+    {"mail/clamav-milter.conf", OPT_MILTER},
     {NULL, 0}};
 
 static void printopts(struct optstruct *opts, int nondef)
diff -up clamav-1.5.2/common/optparser.h.default_confs clamav-1.5.2/common/optparser.h
--- clamav-1.5.2/common/optparser.h.default_confs	2026-01-01 00:00:00.000000000 +0000
+++ clamav-1.5.2/common/optparser.h	2026-01-01 00:00:00.000000000 +0000
@@ -63,9 +63,9 @@
     #define OPT_DATADIR           DATADIR
     #define OPT_CONFDIR           CONFDIR
     #define OPT_CERTSDIR          CERTSDIR
-    #define OPT_CONFDIR_CLAMD     CONFDIR PATHSEP "clamd.conf"
+    #define OPT_CONFDIR_CLAMD     CONFDIR PATHSEP "clamd.d/scan.conf"
     #define OPT_CONFDIR_FRESHCLAM CONFDIR PATHSEP "freshclam.conf"
-    #define OPT_CONFDIR_MILTER    CONFDIR PATHSEP "clamav-milter.conf"
+    #define OPT_CONFDIR_MILTER    CONFDIR PATHSEP "mail/clamav-milter.conf"
 
 #endif
 // clang-format on
diff -up clamav-1.5.2/docs/man/clamav-milter.8.in.default_confs clamav-1.5.2/docs/man/clamav-milter.8.in
--- clamav-1.5.2/docs/man/clamav-milter.8.in.default_confs	2026-01-01 00:00:00.000000000 +0000
+++ clamav-1.5.2/docs/man/clamav-milter.8.in	2026-01-01 00:00:00.000000000 +0000
@@ -27,7 +27,7 @@ Print the version number and exit.
 Read configuration from FILE.
 .SH "FILES"
 .LP
-@CONFDIR@/clamav-milter.conf
+@CONFDIR@/mail/clamav-milter.conf
 .SH "AUTHOR"
 .LP
 aCaB <acab@clamav.net>
diff -up clamav-1.5.2/docs/man/clamav-milter.conf.5.in.default_confs clamav-1.5.2/docs/man/clamav-milter.conf.5.in
--- clamav-1.5.2/docs/man/clamav-milter.conf.5.in.default_confs	2026-01-01 00:00:00.000000000 +0000
+++ clamav-1.5.2/docs/man/clamav-milter.conf.5.in	2026-01-01 00:00:00.000000000 +0000
@@ -239,7 +239,7 @@ Default: no
 All options expressing a size are limited to max 4GB. Values in excess will be reset to the maximum.
 .SH "FILES"
 .LP
-@CONFDIR@/clamav-milter.conf
+@CONFDIR@/mail/clamav-milter.conf
 .SH "AUTHOR"
 .LP
 aCaB <acab@clamav.net>
diff -up clamav-1.5.2/docs/man/clamd.8.in.default_confs clamav-1.5.2/docs/man/clamd.8.in
--- clamav-1.5.2/docs/man/clamd.8.in.default_confs	2026-01-01 00:00:00.000000000 +0000
+++ clamav-1.5.2/docs/man/clamd.8.in	2026-01-01 00:00:00.000000000 +0000
@@ -7,7 +7,7 @@ clamd \- an anti\-virus daemon
 clamd [options]
 .SH "DESCRIPTION"
 .LP
-The daemon listens for incoming connections on Unix and/or TCP socket and scans files or directories on demand. It reads the configuration from @CONFDIR@/clamd.conf
+The daemon listens for incoming connections on Unix and/or TCP socket and scans files or directories on demand. It reads the configuration from @CONFDIR@/clamd.d/scan.conf
 .SH "COMMANDS"
 .LP
 It's recommended to prefix clamd commands with the letter \fBz\fR (eg. zSCAN) to indicate that the command will be delimited by a NULL character and that clamd should continue reading command data until a NULL character is read. The null delimiter assures that the complete command and its entire argument will be processed as a single command. Alternatively commands may be prefixed with the letter \fBn\fR (e.g. nSCAN) to use a newline character as the delimiter. Clamd replies will honour the requested terminator in turn.
@@ -149,7 +149,7 @@ Reload the signature databases.
 Perform a clean exit.
 .SH "FILES"
 .LP
-@CONFDIR@/clamd.conf
+@CONFDIR@/clamd.d/scan.conf
 .SH "CREDITS"
 Please check the full documentation for credits.
 .SH "AUTHOR"
diff -up clamav-1.5.2/docs/man/clamd.conf.5.in.default_confs clamav-1.5.2/docs/man/clamd.conf.5.in
--- clamav-1.5.2/docs/man/clamd.conf.5.in.default_confs	2026-01-01 00:00:00.000000000 +0000
+++ clamav-1.5.2/docs/man/clamd.conf.5.in	2026-01-01 00:00:00.000000000 +0000
@@ -875,7 +875,7 @@ Default: no
 All options expressing a size are limited to max 4GB. Values in excess will be reset to the maximum.
 .SH "FILES"
 .LP
-@CONFDIR@/clamd.conf
+@CONFDIR@/clamd.d/scan.conf
 .SH "AUTHORS"
 .LP
 Tomasz Kojm <tkojm@clamav.net>, Kevin Lin <klin@sourcefire.com>
