diff --git a/toolkit/tools/roast/roast.go b/toolkit/tools/roast/roast.go
index bcf437d3f..17486e97f 100644
--- a/toolkit/tools/roast/roast.go
+++ b/toolkit/tools/roast/roast.go
@@ -47,7 +47,8 @@ var (
 	inputDir  = exe.InputDirFlag(app, "A directory containing a .RAW image or a rootfs directory")
 	outputDir = exe.OutputDirFlag(app, "A destination directory for the output image")
 
-	configFile = app.Flag("config", "Path to the image config file.").Required().ExistingFile()
+	// george - comment this out.
+	// configFile = app.Flag("config", "Path to the image config file.").Required().ExistingFile()
 	tmpDir     = app.Flag("tmp-dir", "Directory to store temporary files while converting.").Required().String()
 
 	releaseVersion = app.Flag("release-version", "Release version to add to the output artifact name").String()
@@ -97,6 +98,19 @@ func main() {
 		logger.Log.Panicf("Error when creating output directory. Error: %s", err)
 	}
 
+	logger.Log.Infof("inputDir = %s", inDirPath)
+	logger.Log.Infof("outputDir = %s", outDirPath)
+	logger.Log.Infof("tmpDir = %s", tmpDirPath)
+
+	newInitrdFile, err := 
+	  convertArtifact("initrd", outDirPath, "initrd", "", inDirPath , false, true)
+	if err != nil {
+		logger.Log.Panicf("Error when creating target image. Error: %s", err)
+	}
+
+	logger.Log.Infof("newInitrdFile = %s", newInitrdFile)
+
+/*
 	config, err := configuration.Load(*configFile)
 	if err != nil {
 		logger.Log.Panicf("Failed loading image configuration. Error: %s", err)
@@ -106,6 +120,7 @@ func main() {
 	if err != nil {
 		logger.Log.Panic(err)
 	}
+*/
 }
 
 func generateImageArtifacts(workers int, inDir, outDir, releaseVersion, imageTag, tmpDir string, config configuration.Config) (err error) {
