diff -Naur plutovg-1.3.0.old/CMakeLists.txt plutovg-1.3.0/CMakeLists.txt
--- plutovg-1.3.0.old/CMakeLists.txt	2025-07-23 22:51:35.113905059 +0200
+++ plutovg-1.3.0/CMakeLists.txt	2025-07-23 22:54:00.829170783 +0200
@@ -38,6 +38,7 @@
 
 set_target_properties(plutovg PROPERTIES
     SOVERSION ${PLUTOVG_VERSION_MAJOR}
+    VERSION ${PLUTOVG_VERSION_MAJOR}.${PLUTOVG_VERSION_MINOR}.${PLUTOVG_VERSION_MICRO}
     C_VISIBILITY_PRESET hidden
     C_STANDARD_REQUIRED ON
     C_STANDARD 11
