diff -Naur plutovg-1.3.3.old/CMakeLists.txt plutovg-1.3.3/CMakeLists.txt
--- plutovg-1.3.3.old/CMakeLists.txt	2026-05-20 11:33:06.191907612 +0200
+++ plutovg-1.3.3/CMakeLists.txt	2026-05-20 11:33:13.349934031 +0200
@@ -28,9 +28,6 @@
     source/plutovg-ft-raster.h
     source/plutovg-ft-stroker.h
     source/plutovg-ft-types.h
-    source/plutovg-stb-image-write.h
-    source/plutovg-stb-image.h
-    source/plutovg-stb-truetype.h
 )
 
 add_library(plutovg ${plutovg_sources} ${plutovg_headers})
diff -Naur plutovg-1.3.3.old/source/plutovg-font.c plutovg-1.3.3/source/plutovg-font.c
--- plutovg-1.3.3.old/source/plutovg-font.c	2026-05-20 11:33:06.193583185 +0200
+++ plutovg-1.3.3/source/plutovg-font.c	2026-05-20 11:33:13.350196917 +0200
@@ -6,7 +6,7 @@
 
 #define STBTT_STATIC
 #define STB_TRUETYPE_IMPLEMENTATION
-#include "plutovg-stb-truetype.h"
+#include <stb/stb_truetype.h>
 
 static int plutovg_text_iterator_length(const void* data, plutovg_text_encoding_t encoding)
 {
diff -Naur plutovg-1.3.3.old/source/plutovg-surface.c plutovg-1.3.3/source/plutovg-surface.c
--- plutovg-1.3.3.old/source/plutovg-surface.c	2026-05-20 11:33:06.196907656 +0200
+++ plutovg-1.3.3/source/plutovg-surface.c	2026-05-20 11:33:13.350567104 +0200
@@ -3,11 +3,11 @@
 
 #define STB_IMAGE_WRITE_STATIC
 #define STB_IMAGE_WRITE_IMPLEMENTATION
-#include "plutovg-stb-image-write.h"
+#include <stb/stb_image_write.h>
 
 #define STB_IMAGE_STATIC
 #define STB_IMAGE_IMPLEMENTATION
-#include "plutovg-stb-image.h"
+#include <stb/stb_image.h>
 
 static plutovg_surface_t* plutovg_surface_create_uninitialized(size_t width, size_t height)
 {
