From 3f7321d4d8d3500eca2d98ec982433b9967b17d7 Mon Sep 17 00:00:00 2001
From: Yan Fu <yafu@redhat.com>
Date: Thu, 30 Oct 2025 03:47:46 -0400
Subject: [PATCH 09/15] Fedora: tokio 1.48

Signed-off-by: Uri Lublin <uril@redhat.com>
Signed-off-by: Yan Fu <yafu@redhat.com>
---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 30cc484..a4f5261 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -41,7 +41,7 @@ sha2 = "0.10.9"
 strum = { version = "0.27", features = ["derive"] }
 tempfile = "3.23"
 thiserror = "2.0"
-tokio = "1.47"
+tokio = "1.48"
 toml = "0.8.23"
 tonic = "0.12"
 tonic-build = "0.12"
-- 
2.51.1

