From 35bc583134bd1265726fc46734fee102e304173b Mon Sep 17 00:00:00 2001
From: Yan Fu <yafu@redhat.com>
Date: Thu, 30 Oct 2025 04:02:24 -0400
Subject: [PATCH 12/15] Fedora: serde_with: 3.12.0

Signed-off-by: Uri Lublin <uril@redhat.com>
Signed-off-by: Yan Fu <yafu@redhat.com>
---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 41aa3d5..393ae5a 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,7 +34,7 @@ resource_uri = { path = "attestation-agent/deps/resource_uri" }
 ring = "0.17"
 rstest = "0.26"
 serde = { version = "1.0", features = ["derive"] }
-serde_with = { version = "3.14.0", features = ["base64"] }
+serde_with = { version = "3.12.0", features = ["base64"] }
 serde_json = "1.0"
 serial_test = "3"
 sha2 = "0.10.9"
-- 
2.51.1

