From c40b5c532f4377a2262bfbed6ebddc18004e0d90 Mon Sep 17 00:00:00 2001
From: Yan Fu <yafu@redhat.com>
Date: Thu, 30 Oct 2025 04:05:48 -0400
Subject: [PATCH 13/15] Fedora: picky-asn1-x509: 0.14.5

Signed-off-by: Uri Lublin <uril@redhat.com>
Signed-off-by: Yan Fu <yafu@redhat.com>
---
 attestation-agent/attester/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/attestation-agent/attester/Cargo.toml b/attestation-agent/attester/Cargo.toml
index 9d30685..65aad08 100644
--- a/attestation-agent/attester/Cargo.toml
+++ b/attestation-agent/attester/Cargo.toml
@@ -41,7 +41,7 @@ env_logger = "0.11"
 num-traits = { version = "0.2" }
 picky-asn1 = { version = "0.10.1", optional = true }
 picky-asn1-der = { version = "0.5.2", optional = true }
-picky-asn1-x509 = { version = "0.14.6", optional = true }
+picky-asn1-x509 = { version = "0.14.5", optional = true }
 tss-esapi = { version = "7.5", optional = true }
 
 
-- 
2.51.1

