From 4d62e4d91b15e146de322103c3ec4dd73e5aaa64 Mon Sep 17 00:00:00 2001
From: adityadubey <adityadubey@microsoft.com>
Date: Fri, 23 Feb 2024 16:43:54 -0800
Subject: [PATCH] replace to flit

---
 pyproject.toml | 17 +++--------------
 1 file changed, 3 insertions(+), 14 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index a452025..1651b4a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,9 +1,6 @@
 [build-system]
-build-backend = "hatchling.build"
-requires = [
-  "hatch-vcs",
-  "hatchling",
-]
+build-backend = "flit_core.buildapi"
+requires = ["flit_core >=3.8.0,<4"]
 
 [project]
 name = "prettytable"
@@ -28,9 +25,7 @@ classifiers = [
   "Topic :: Text Processing",
   "Typing :: Typed",
 ]
-dynamic = [
-  "version",
-]
+version = "3.10.0"
 dependencies = [
   "wcwidth",
 ]
@@ -45,12 +40,6 @@ Changelog = "https://github.com/jazzband/prettytable/releases"
 Homepage = "https://github.com/jazzband/prettytable"
 Source = "https://github.com/jazzband/prettytable"
 
-[tool.hatch]
-version.source = "vcs"
-
-[tool.hatch.version.raw-options]
-local_scheme = "no-local-version"
-
 [tool.ruff]
 fix = true
 
-- 
2.34.1

