From af022c31e6467193069cc4a22c31d7a97d414928 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Tue, 19 Nov 2019 09:52:13 +0100
Subject: [PATCH] Normalize shebang
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 scripts/hashbase_inc.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/hashbase_inc.pl b/scripts/hashbase_inc.pl
index 8b740f7..c450ab8 100644
--- a/scripts/hashbase_inc.pl
+++ b/scripts/hashbase_inc.pl
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 use strict;
 use warnings;
 
-- 
2.21.0

