diff -up qttools-opensource-src-5.5.0/src/linguist/lrelease/main.cpp.qmake-qt5 qttools-opensource-src-5.5.0/src/linguist/lrelease/main.cpp
--- qttools-opensource-src-5.5.0/src/linguist/lrelease/main.cpp.qmake-qt5	2015-06-29 15:13:45.000000000 -0500
+++ qttools-opensource-src-5.5.0/src/linguist/lrelease/main.cpp	2015-07-16 10:23:49.664773033 -0500
@@ -310,9 +310,9 @@ int main(int argc, char **argv)
             option.qmake_abslocation = QString::fromLocal8Bit(qgetenv("QMAKE"));
             if (option.qmake_abslocation.isEmpty())
 #ifdef QT_BOOTSTRAPPED
-                option.qmake_abslocation = binDir + QLatin1String("/qmake");
+                option.qmake_abslocation = binDir + QLatin1String("/qmake-qt5");
 #else
-                option.qmake_abslocation = app.applicationDirPath() + QLatin1String("/qmake");
+                option.qmake_abslocation = app.applicationDirPath() + QLatin1String("/qmake-qt5");
 #endif
             option.initProperties();
             QMakeVfs vfs;
diff -up qttools-opensource-src-5.5.0/src/linguist/lupdate/main.cpp.qmake-qt5 qttools-opensource-src-5.5.0/src/linguist/lupdate/main.cpp
--- qttools-opensource-src-5.5.0/src/linguist/lupdate/main.cpp.qmake-qt5	2015-06-29 15:13:46.000000000 -0500
+++ qttools-opensource-src-5.5.0/src/linguist/lupdate/main.cpp	2015-07-16 10:23:49.665773055 -0500
@@ -1048,7 +1048,7 @@ int main(int argc, char **argv)
         ProFileGlobals option;
         option.qmake_abslocation = QString::fromLocal8Bit(qgetenv("QMAKE"));
         if (option.qmake_abslocation.isEmpty())
-            option.qmake_abslocation = app.applicationDirPath() + QLatin1String("/qmake");
+            option.qmake_abslocation = app.applicationDirPath() + QLatin1String("/qmake-qt5");
         option.debugLevel = proDebug;
         option.initProperties();
         option.setCommandLineArguments(QDir::currentPath(),
