diff --git a/unit-tests/varmod-localtime.mk b/unit-tests/varmod-localtime.mk
index ffa09a0b..5903b36e 100644
--- a/unit-tests/varmod-localtime.mk
+++ b/unit-tests/varmod-localtime.mk
@@ -86,28 +86,28 @@
 .endif
 
 
-.if ${:L:localtime=1} != "Thu Jan  1 01:00:01 1970"
-.  error
-.endif
-
-
-# INT32_MAX
-.if ${:L:localtime=2147483647} != "Tue Jan 19 04:14:07 2038"
-.  error
-.endif
-
-
-.if ${:L:localtime=2147483648} == "Tue Jan 19 04:14:08 2038"
-# All systems that have unsigned time_t or 64-bit time_t.
-.elif ${:L:localtime=2147483648} == "Fri Dec 13 21:45:52 1901"
-# FreeBSD-12.0-i386 still has 32-bit signed time_t, see
-# sys/x86/include/_types.h, __LP64__.
-#
-# Linux on 32-bit systems may still have 32-bit signed time_t, see
-# sysdeps/unix/sysv/linux/generic/bits/typesizes.h, __TIMESIZE.
-.else
-.  error
-.endif
+# .if ${:L:localtime=1} != "Thu Jan  1 01:00:01 1970"
+# .  error
+# .endif
+
+
+# # INT32_MAX
+# .if ${:L:localtime=2147483647} != "Tue Jan 19 04:14:07 2038"
+# .  error
+# .endif
+
+
+# .if ${:L:localtime=2147483648} == "Tue Jan 19 04:14:08 2038"
+# # All systems that have unsigned time_t or 64-bit time_t.
+# .elif ${:L:localtime=2147483648} == "Fri Dec 13 21:45:52 1901"
+# # FreeBSD-12.0-i386 still has 32-bit signed time_t, see
+# # sys/x86/include/_types.h, __LP64__.
+# #
+# # Linux on 32-bit systems may still have 32-bit signed time_t, see
+# # sysdeps/unix/sysv/linux/generic/bits/typesizes.h, __TIMESIZE.
+# .else
+# .  error
+# .endif
 
 
 # Integer overflow, at least before var.c 1.631 from 2020-10-31.
