--- notify-8.2.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ notify-8.2.0/Cargo.toml	2026-09-19T06:46:38.525526+00:00
@@ -38,7 +38,7 @@
 optional = true
 
 [dependencies.flume]
-version = "0.11.1"
+version = ">=0.11.1,<0.13"
 optional = true
 
 [dependencies.libc]
@@ -68,9 +68,6 @@
 [features]
 default = ["macos_fsevent"]
 macos_fsevent = []
-macos_kqueue = [
-    "mio",
-]
 serde = ["notify-types/serde"]
 serialization-compat-6 = ["notify-types/serialization-compat-6"]
 
