diff --git a/cmd/dotty/dotty_ui.lefty b/cmd/dotty/dotty_ui.lefty
index 6fdec8f..270c0c8 100644
--- a/cmd/dotty/dotty_ui.lefty
+++ b/cmd/dotty/dotty_ui.lefty
@@ -351,7 +351,7 @@ dotty.protovt.normal.uifuncs = [
         else
             gt.insertedge (gt, data.pobj, null, data.obj, null, null, 1);
     };
-    'rightdown' = function (data) {
+    'rightup' = function (data) {
         local vt, gt, menu, i;
 
         vt = dotty.views[data.widget];
@@ -456,7 +456,7 @@ dotty.protovt.birdseye.uifuncs = [
     'middledown' = dotty.protovt.normal.uifuncs.middledown;
     'middlemove' = dotty.protovt.normal.uifuncs.middlemove;
     'middleup' = dotty.protovt.normal.uifuncs.middleup;
-    'rightdown' = dotty.protovt.normal.uifuncs.rightdown;
+    'rightup' = dotty.protovt.normal.uifuncs.rightup;
     'keyup' = dotty.protovt.normal.uifuncs.keyup;
     'redraw' = dotty.protovt.normal.uifuncs.redraw;
     'closeview' = dotty.protovt.normal.uifuncs.closeview;
