diff --git a/cmd/lefty/dot2l/dotlex.c b/cmd/lefty/dot2l/dotlex.c
index cf738c0..65e17e2 100644
--- a/cmd/lefty/dot2l/dotlex.c
+++ b/cmd/lefty/dot2l/dotlex.c
@@ -252,7 +252,7 @@ static char *scan_token (char *p) {
     char *q;
 
     q = lexbuf;
-    if (p == '\0')
+    if (!p || *p == '\0')
         return NULL;
     while (isalnum (*p) || (*p == '_') || (!isascii (*p)))
         *q++ = *p++;
diff --git a/cmd/tools/gvgen.c b/cmd/tools/gvgen.c
index c2b166d..c9a51e9 100644
--- a/cmd/tools/gvgen.c
+++ b/cmd/tools/gvgen.c
@@ -453,6 +453,8 @@ closeOpen (void)
 	fprintf(opts.outfile, "}\ngraph {\n");
 }
 
+extern void makeTetrix(int depth, edgefn ef);
+
 int main(int argc, char *argv[])
 {
     GraphType graphType;
