diff -rupN --no-dereference leptonica-1.85.0/CMakeLists.txt leptonica-1.85.0-new/CMakeLists.txt
--- leptonica-1.85.0/CMakeLists.txt	2024-10-16 21:32:44.000000000 +0200
+++ leptonica-1.85.0-new/CMakeLists.txt	2025-07-17 21:00:46.015045444 +0200
@@ -71,7 +71,7 @@ option(ENABLE_WEBP "Enable WebP support"
 option(ENABLE_OPENJPEG "Enable OpenJPEG support" ON)
 
 set(leptonica_INSTALL_CMAKE_DIR
-    "lib/cmake/leptonica"
+    "${CMAKE_INSTALL_LIBDIR}/cmake/leptonica"
     CACHE STRING "Install destination for CMake package files")
 
 if(NOT SW_BUILD)
@@ -374,7 +374,7 @@ configure_file(lept.pc.cmake ${CMAKE_CUR
 # to resolve generator expression in OUTPUT_NAME
 file(
   GENERATE
-  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lept_$<CONFIG>.pc
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/lept.pc
   INPUT ${CMAKE_CURRENT_BINARY_DIR}/lept.pc.in)
 
 configure_file(
@@ -384,8 +384,8 @@ configure_file(
   ${CMAKE_CURRENT_SOURCE_DIR}/cmake/templates/LeptonicaConfig.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/LeptonicaConfig.cmake @ONLY)
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/lept_$<CONFIG>.pc
-        DESTINATION lib/pkgconfig)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/lept.pc
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/LeptonicaConfig.cmake
               ${CMAKE_CURRENT_BINARY_DIR}/LeptonicaConfig-version.cmake
         DESTINATION ${leptonica_INSTALL_CMAKE_DIR})
diff -rupN --no-dereference leptonica-1.85.0/prog/CMakeLists.txt leptonica-1.85.0-new/prog/CMakeLists.txt
--- leptonica-1.85.0/prog/CMakeLists.txt	2024-10-16 21:32:44.000000000 +0200
+++ leptonica-1.85.0-new/prog/CMakeLists.txt	2025-07-17 21:00:46.014577716 +0200
@@ -18,7 +18,7 @@ function(add_prog_target target)
     if(FREEBSD AND HAVE_LIBM AND ${target} MATCHES ${math_progs})
       target_link_libraries         (${target} leptonica m)
     else()
-      target_link_libraries         (${target} leptonica)
+      target_link_libraries         (${target} leptonica m)
     endif()
     set_target_properties           (${target} PROPERTIES FOLDER prog)
 endfunction(add_prog_target)
diff -rupN --no-dereference leptonica-1.85.0/src/CMakeLists.txt leptonica-1.85.0-new/src/CMakeLists.txt
--- leptonica-1.85.0/src/CMakeLists.txt	2024-10-16 21:32:44.000000000 +0200
+++ leptonica-1.85.0-new/src/CMakeLists.txt	2025-07-17 21:00:46.016291864 +0200
@@ -17,6 +17,9 @@ add_library                     (leptoni
 set_target_properties           (leptonica PROPERTIES VERSION   6.0.0)
 set_target_properties           (leptonica PROPERTIES SOVERSION 6)
 set_target_properties           (leptonica PROPERTIES OUTPUT_NAME leptonica$<$<BOOL:${MSVC}>:-${PROJECT_VERSION}$<$<CONFIG:DEBUG>:d>>)
+if(MINGW)
+set_target_properties           (leptonica PROPERTIES SUFFIX "-6${CMAKE_SHARED_LIBRARY_SUFFIX}")
+endif(MINGW)
 
 if (BUILD_SHARED_LIBS)
     target_compile_definitions  (leptonica PRIVATE -DLIBLEPT_EXPORTS)
@@ -66,7 +69,7 @@ if (UNIX)
             ${CMAKE_COMMAND} -E create_symlink "$<TARGET_FILE:leptonica>" ${old_target}
             DEPENDS ${PROJECT_NAME}
             COMMENT "Creating lept symlink")
-        install(FILES ${CMAKE_BINARY_DIR}/src/${old_target} DESTINATION lib)
+        install(FILES ${CMAKE_BINARY_DIR}/src/${old_target} DESTINATION ${CMAKE_INSTALL_LIBDIR})
     endif(SYM_LINK)
 endif()
 
@@ -89,7 +92,7 @@ else()
     export(TARGETS leptonica APPEND FILE ${CMAKE_BINARY_DIR}/LeptonicaTargets.cmake)
 endif()
 
-install(TARGETS leptonica EXPORT LeptonicaTargets RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+install(TARGETS leptonica EXPORT LeptonicaTargets RUNTIME DESTINATION bin LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(EXPORT LeptonicaTargets DESTINATION  ${leptonica_INSTALL_CMAKE_DIR})
 install(FILES
     ${hdr}
