https://rt.cpan.org/Public/Bug/Display.html?id=86060

--- lib/Mail/SPF/Util.pm
+++ lib/Mail/SPF/Util.pm
@@ -155,9 +155,9 @@ sub ipv6_address_is_ipv4_mapped {
 throws I<Mail::SPF::EInvalidOptionValue>
 
 Returns the given I<NetAddr::IP> IPv4 or IPv6 address compactly formatted as a
-I<string>.  For IPv4 addresses, this is equivalent to calling L< NetAddr::IP's
-C<addr> |NetAddr::IP/addr> method.  For IPv6 addresses, this is equivalent to
-calling L< NetAddr::IP's C<short> |NedAddr::IP/short> method.  Throws a
+I<string>.  For IPv4 addresses, this is equivalent to calling L<NetAddr::IP's
+C<addr>|NetAddr::IP/addr> method.  For IPv6 addresses, this is equivalent to
+calling L<NetAddr::IP's C<short>|NetAddr::IP/short> method.  Throws a
 I<Mail::SPF::EInvalidOptionValue> exception if the specified object is not a
 I<NetAddr::IP> IPv4 or IPv6 address object.
 
