diff -Naur plutosvg-0.0.8.old/CMakeLists.txt plutosvg-0.0.8/CMakeLists.txt
--- plutosvg-0.0.8.old/CMakeLists.txt	2026-05-20 13:20:17.959117220 +0200
+++ plutosvg-0.0.8/CMakeLists.txt	2026-05-20 13:21:03.282601241 +0200
@@ -27,6 +27,7 @@
 
 set_target_properties(plutosvg PROPERTIES
     SOVERSION ${PLUTOSVG_VERSION_MAJOR}
+    VERSION ${PLUTOSVG_VERSION_MAJOR}.${PLUTOSVG_VERSION_MINOR}.${PLUTOSVG_VERSION_MICRO}
     C_VISIBILITY_PRESET hidden
     C_STANDARD_REQUIRED ON
     C_STANDARD 99
