diff -Naur plutosvg-0.0.8.old/source/plutosvg.h plutosvg-0.0.8/source/plutosvg.h
--- plutosvg-0.0.8.old/source/plutosvg.h	2026-05-20 13:20:17.960599064 +0200
+++ plutosvg-0.0.8/source/plutosvg.h	2026-05-20 13:20:31.371860365 +0200
@@ -23,7 +23,7 @@
 #ifndef PLUTOSVG_H
 #define PLUTOSVG_H
 
-#include <plutovg.h>
+#include <plutovg/plutovg.h>
 
 #if defined(PLUTOSVG_BUILD_STATIC)
 #define PLUTOSVG_EXPORT
