diff -up sblim-testsuite-1.3.0/consistence.pm.orig sblim-testsuite-1.3.0/consistence.pm
--- sblim-testsuite-1.3.0/consistence.pm.orig	2011-06-09 16:05:22.000978911 +0200
+++ sblim-testsuite-1.3.0/consistence.pm	2011-06-09 16:06:16.382978956 +0200
@@ -52,21 +52,21 @@ sub wbem_cli {
 #                      asked for in parameter ${class} -> Provider failure ?
 #
 
-my $CLASS="";
+our $CLASS="";
 my $INPUT_FILE;
 
 my $REPORT_FILE;
 my $OLD_HDL;             # save file handle of STDOUT; necesarry when report file is generated
 
-my $keyname;
-my $keyval;
-my $inst;
-my $checkRef_ain;
-my $checkRef_gi;
-my $sourceClass;
-my $targetClass;
+our $keyname;
+our $keyval;
+our $inst;
+our $checkRef_ain;
+our $checkRef_gi;
+our $sourceClass;
+our $targetClass;
 
-my @rc = (0,"");
+our @rc = (0,"");
 
 
 #******************************************************************************#
