diff -up sblim-testsuite-1.3.0/instance.pm.orig sblim-testsuite-1.3.0/instance.pm
--- sblim-testsuite-1.3.0/instance.pm.orig	2016-05-24 14:40:39.863492483 +0200
+++ sblim-testsuite-1.3.0/instance.pm	2016-05-24 14:42:47.333201401 +0200
@@ -189,7 +189,7 @@ sub get {
 	  return @rc;
         }
 	if ( defined $volatile && 
-	     $get[1]=~/${cimom::ERROR_PREFIX}!{NOT_FOUND}/i ) {
+	     $get[1]=~/${cimom::ERROR_PREFIX}!\{NOT_FOUND}/i ) {
 	  $rc[0]=14;
 	  $rc[1]="warning - volatile class type returned with exception - $get[1]";
 	  print "... warning\n";
