diff -up sblim-testsuite-1.3.0/Makefile.in.orig sblim-testsuite-1.3.0/Makefile.in
--- sblim-testsuite-1.3.0/Makefile.in.orig	2013-08-28 11:20:07.814934872 +0200
+++ sblim-testsuite-1.3.0/Makefile.in	2013-08-28 11:20:28.085980836 +0200
@@ -143,7 +143,7 @@ datarootdir = @datarootdir@
 #
 
 # docdir defines where the documentation goes
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host_alias = @host_alias@
