diff -pruN qt-x11-free-3.3.6.org/src/kernel/qfont_x11.cpp qt-x11-free-3.3.6/src/kernel/qfont_x11.cpp
--- qt-x11-free-3.3.6.org/src/kernel/qfont_x11.cpp	2006-03-08 00:00:00.000000000 +0530
+++ qt-x11-free-3.3.6/src/kernel/qfont_x11.cpp	2006-12-04 18:49:09.000000000 +0530
@@ -709,7 +709,7 @@ int QFontMetrics::charWidth( const QStri
     if ( script >= QFont::Arabic && script <= QFont::Khmer ) {
 	// complex script shaping. Have to do some hard work
 	int from = QMAX( 0,  pos - 8 );
-	int to = QMIN( (int)str.length(), pos + 8 );
+	int to = str.length();
 	QConstString cstr( str.unicode()+from, to-from);
 	QTextEngine layout( cstr.string(), d );
 	layout.itemize( QTextEngine::WidthOnly );
