--- qt-x11-free-3.3.4/src/kernel/qpsprinter.cpp.cjk	2005-01-21 18:16:11.000000000 +0100
+++ qt-x11-free-3.3.4/src/kernel/qpsprinter.cpp	2005-01-26 15:09:20.911239907 +0100
@@ -4748,10 +4748,11 @@
 
 // GB18030 fonts on Turbolinux
 static const psfont SongGBK2K [] = {
-    { "MSung-Light-GBK2K-H", 0, 100. },
-    { "MSung-Italic-GBK2K-H", 0, 100. },
-    { "MSung-Bold-GBK2K-H", 0, 100. },
-    { "MSung-BoldItalic-GBK2K-H", 0, 100. },
+    { "MSungGBK-Light-GBK2K-H", 0, 100. },
+    { "MSungGBK-Light-GBK2K-H", 0, 100. },
+    { "MSungGBK-Light-GBK2K-H", 0, 100. },
+    { "MSungGBK-Light-GBK2K-H", 0, 100. },
+	 
 };
 static const psfont KaiGBK2K [] = {
     { "MKai-Medium-GBK2K-H", 0, 100. },
