diff --git a/Makefile.am b/Makefile.am
index 7e7d4163d..4904dd591 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -26,7 +26,7 @@ EXTRA_DIST = ChangeLog CREDITS INSTALL \
 BUILT_SOURCES =
 
 SUBDIRS = po misc
-SUBDIRS += rpmio lib sign build scripts fileattrs docs .
+SUBDIRS += rpmio lib sign build scripts fileattrs .
 
 if ENABLE_PYTHON
 SUBDIRS += python
@@ -40,7 +40,7 @@ if HAVE_FAKECHROOT
 SUBDIRS += tests
 endif
 
-DIST_SUBDIRS = po misc rpmio lib sign build python scripts fileattrs docs tests plugins
+DIST_SUBDIRS = po misc rpmio lib sign build python scripts fileattrs tests plugins
 
 pkgconfigdir = $(libdir)/pkgconfig
 
@@ -259,24 +259,6 @@ uninstall-local:
 
 MAINTAINERCLEANFILES = ChangeLog
 
-# Always build librpm docs
-if DOXYGEN
-docs/librpm.doxy: docs/librpm.doxy.in docs/librpm/Doxyheader.h Makefile.am configure.ac $(pkginclude_HEADERS)
-	files=`echo " $(pkginclude_HEADERS)" | $(SED) -e "s, , \@top_srcdir\@/,g"`; \
-	$(SED) -e "s,[@]pkginclude_HEADERS[@],$$files," \
-	  -e "s,\@top_srcdir\@/,$(top_srcdir)/,g" \
-	  -e "s,[@]HAVE_DOT[@],$(HAVE_DOT)," \
-	  -e "s,[@]PACKAGE[@],$(PACKAGE)," \
-	  -e "s,[@]VERSION[@],$(VERSION)," \
-	< $(top_srcdir)/docs/librpm.doxy.in > docs/librpm.doxy
-
-docs/librpm/html: docs/librpm.doxy
-	@DOXYGEN@ docs/librpm.doxy
-noinst_DATA += docs/librpm/html
-endif
-EXTRA_DIST += docs/librpm.doxy.in docs/librpm/Doxyheader.h
-EXTRA_DIST += docs/librpm/html
-
 .PHONY: ci
 ci:
 	podman build -t rpm -f $(top_srcdir)/ci/Dockerfile $(top_srcdir)
