From 5e4ef70a1fda792d7ca32311ecc29302c7b13ca5 Mon Sep 17 00:00:00 2001
From: Jakub Hrozek <jakub.hrozek@posteo.se>
Date: Sun, 1 Apr 2018 10:40:13 +0200
Subject: [PATCH 1/2] Disable pylint tests

---
 tests/Makefile.am | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index 0a7854eec62520014919ad3983db70c78be483e2..8c742a78e3ce8e822fbd7bd9d5735a010e2f0f80 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -22,9 +22,11 @@ TESTS = test_dict test_set test_tio test_expr test_getpeercred test_cfg \
         test_attmap test_myldap.sh test_common test_nsscmds.sh \
         test_pamcmds.sh test_manpages.sh test_clock \
         test_tio_timeout
-if HAVE_PYTHON
-  TESTS += test_pycompile.sh test_pylint.sh
-endif
+
+#if HAVE_PYTHON
+#  TESTS += test_pycompile.sh test_pylint.sh
+#endif
+
 if ENABLE_PYNSLCD
   TESTS += test_pynslcd_cache.py test_doctest.sh
 endif
-- 
2.14.3

