From 4d6d04857a257494257389cc499aea9a6fd740ee Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Thu, 21 Oct 2021 10:01:06 +0200
Subject: [PATCH] Build media-session from local tarbal

---
 subprojects/media-session.wrap | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/subprojects/media-session.wrap b/subprojects/media-session.wrap
index f6b4e62fb..a6249e5e1 100644
--- a/subprojects/media-session.wrap
+++ b/subprojects/media-session.wrap
@@ -1,4 +1,4 @@
-[wrap-git]
-url = https://gitlab.freedesktop.org/pipewire/media-session.git
-revision = head
+[wrap-file]
+source_filename = media-session-0.4.1.tar.gz
+directory = media-session-0.4.1
 
-- 
2.31.1
