--- Makefile.PL
+++ Makefile.PL
@@ -17,7 +17,7 @@ my %WriteMakefileArgs = (
   "NAME" => "Sub::Exporter::Progressive",
   "PREREQ_PM" => {},
   "TEST_REQUIRES" => {
-    "Test::More" => "0.88"
+    "Test::More" => "0.47"
   },
   "VERSION" => "0.001013",
   "test" => {
@@ -27,7 +27,7 @@ my %WriteMakefileArgs = (
 
 
 my %FallbackPrereqs = (
-  "Test::More" => "0.88"
+  "Test::More" => "0.47"
 );
 
 
--- t/all.t
+++ t/all.t
@@ -2,7 +2,7 @@
 use strict;
 use warnings;
 
-use Test::More;
+use Test::More tests => 6;
 use lib 't/lib';
 use A::JunkAll;
 
@@ -17,5 +17,3 @@ use A::JunkAll ':all';
 ok(main->can('junk1'), 'sub exported');
 ok(main->can('junk2'), 'sub exported');
 ok(! $INC{'Sub/Exporter.pm'}, 'Sub::Exporter not loaded');
-
-done_testing;
--- t/basic.t
+++ t/basic.t
@@ -2,11 +2,9 @@
 use strict;
 use warnings;
 
-use Test::More;
+use Test::More tests => 2;
 use lib 't/lib';
 use A::Junk 'junk1';
 
 ok(main->can('junk1'), 'requested sub exported');
 ok(! $INC{'Sub/Exporter.pm'}, 'Sub::Exporter not loaded');
-
-done_testing;
--- t/default.t
+++ t/default.t
@@ -2,11 +2,9 @@
 use strict;
 use warnings;
 
-use Test::More;
+use Test::More tests => 2;
 use lib 't/lib';
 use A::Junk;
 
 ok(main->can('junk2'), 'sub exported');
 ok(! $INC{'Sub/Exporter.pm'}, 'Sub::Exporter not loaded');
-
-done_testing;
--- t/release-changes_has_content.t
+++ t/release-changes_has_content.t
@@ -10,7 +10,6 @@ BEGIN {
 
 use Test::More tests => 2;
 
-note 'Checking Changes';
 my $changes_file = 'Changes';
 my $newver = '0.001013';
 my $trial_token = '-TRIAL';
@@ -22,8 +21,6 @@ SKIP: {
     ok(_get_changes($newver), "$changes_file has content for $newver");
 }
 
-done_testing;
-
 # _get_changes copied and adapted from Dist::Zilla::Plugin::Git::Commit
 # by Jerome Quelin
 sub _get_changes
--- t/sex.t
+++ t/sex.t
@@ -2,7 +2,7 @@
 use strict;
 use warnings;
 
-use Test::More;
+use Test::More tests => 2;
 BEGIN {
     plan skip_all => 'Sub::Exporter not installed'
         unless eval { require Sub::Exporter };
@@ -13,5 +13,3 @@ use A::Junk 'junk1' => { -as => 'junk' }
 
 ok(main->can('junk'), 'sub renamed with Sub::Exporter');
 ok($INC{'Sub/Exporter.pm'}, 'Sub::Exporter loaded');
-
-done_testing;
--- t/tags.t
+++ t/tags.t
@@ -1,7 +1,7 @@
 use strict;
 use warnings;
 
-use Test::More;
+use Test::More tests => 44;
 use Carp;
 use lib 't/lib';
 use A::Junk ':other';
@@ -64,5 +64,3 @@ SKIP: {
   check_tag('bar -default', [qw/foo bar/], [qw/baz/]);
 }
 
-done_testing;
-
--- t/version-check.t
+++ t/version-check.t
@@ -1,7 +1,7 @@
 
 use strict;
 use warnings;
-use Test::More;
+use Test::More tests => 5;
 
 BEGIN {
    package AAA;
@@ -38,5 +38,3 @@ ok(eval('use AAA 1; 1'), 'perl built-in
    );
 }
 
-done_testing;
-
