--- sequoia-keystore-softkeys-0.7.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ sequoia-keystore-softkeys-0.7.1/Cargo.toml	2025-12-01T22:26:17.192291+00:00
@@ -70,11 +70,6 @@
 [dev-dependencies.env_logger]
 version = ">=0.10, <0.12"
 
-[dev-dependencies.sequoia-openpgp]
-version = "2"
-features = ["compression"]
-default-features = false
-
 [dev-dependencies.tempfile]
 version = "3"
 
@@ -98,6 +93,7 @@
 [target."cfg(not(windows))".dev-dependencies.sequoia-openpgp]
 version = "2"
 features = [
+    "compression",
     "crypto-nettle",
     "__implicit-crypto-backend-for-tests",
 ]
