--- cargo-util-0.2.25/Cargo.toml	1970-01-01T00:00:01+00:00
+++ cargo-util-0.2.25/Cargo.toml	2026-02-16T14:02:19.453813+00:00
@@ -66,23 +66,8 @@
 [dependencies.walkdir]
 version = "2.5.0"
 
-[target.'cfg(target_os = "macos")'.dependencies.core-foundation]
-version = "0.10.1"
-features = ["mac_os_10_7_support"]
-
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.174"
-
-[target."cfg(windows)".dependencies.miow]
-version = "0.6.1"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.61"
-features = [
-    "Win32_Storage_FileSystem",
-    "Win32_Foundation",
-    "Win32_System_Console",
-]
 
 [lints.clippy]
 dbg_macro = "warn"
@@ -104,3 +89,4 @@
 
 [lints.rustdoc]
 private_intra_doc_links = "allow"
+
