--- ratatui-0.26.3/Cargo.toml	1970-01-01T00:00:01+00:00
+++ ratatui-0.26.3/Cargo.toml	2025-04-12T23:20:34.864507+00:00
@@ -244,7 +244,7 @@
 version = "0.3"
 
 [dependencies.compact_str]
-version = "0.7.1"
+version = ">=0.8.0, <0.10.0"
 
 [dependencies.crossterm]
 version = "0.27"
@@ -276,7 +276,7 @@
 features = ["derive"]
 
 [dependencies.termion]
-version = "3.0"
+version = "4.0"
 optional = true
 
 [dependencies.termwiz]
