diff --git a/sendmail/sendmail.h b/sendmail/sendmail.h
index 441399c..95b90b8 100644
--- a/sendmail/sendmail.h
+++ b/sendmail/sendmail.h
@@ -2551,7 +2551,7 @@ extern const SM_EXC_TYPE_T EtypeQuickAbort; /* type of a QuickAbort exception */
 #  define SOL_IP IPPROTO_IP
 # endif
 # if defined(SOL_IP) && defined(IP_TOS)
-EXTERN int	InetQoS;	/* QoS mapping */
+extern int	InetQoS;	/* QoS mapping */
 # endif
 #endif
 
