--- sendmail-8.14.9/sendmail/helpfile		2014-03-06 18:31:31.000000000 +0100
+++ sendmail-8.14.9/sendmail/helpfile.noversion	2014-05-21 17:25:29.000000000 +0200
@@ -11,9 +11,7 @@
 cpyr	forth in the LICENSE file which can be found at the top level of
 cpyr	the sendmail distribution.
 cpyr
-cpyr	$$Id: helpfile,v 8.49 2013-11-22 20:51:55 ca Exp $$
-cpyr
-smtp	This is sendmail version $v
+smtp	This is sendmail
 smtp	Topics:
 smtp		HELO	EHLO	MAIL	RCPT	DATA
 smtp		RSET	NOOP	QUIT	HELP	VRFY
