--- sendmail-8.14.4/cf/cf/Build				1999-03-02 03:37:12.000000000 +0100
+++ sendmail-8.14.4/cf/cf/Build.makemapman		2010-01-03 22:49:38.000000000 +0100
@@ -18,7 +18,7 @@
 SMROOT=${SMROOT-../..}
 BUILDTOOLS=${BUILDTOOLS-$SMROOT/devtools}
 
-M4=`sh $BUILDTOOLS/bin/find_m4.sh`
+M4=/usr/bin/m4
 ret=$?
 if [ $ret -ne 0 ]
 then
--- sendmail-8.14.4/devtools/OS/Linux			2009-01-22 03:15:42.000000000 +0100
+++ sendmail-8.14.4/devtools/OS/Linux.makemapman	2010-01-03 22:50:27.000000000 +0100
@@ -6,7 +6,7 @@
 define(`confDEPEND_TYPE', `CC-M')
 define(`confCCOPTS_SO', `-fPIC')
 define(`confSM_OS_HEADER', `sm_os_linux')
-define(`confMANROOT', `/usr/man/man')
+define(`confMANROOT', `/usr/share/man/man')
 define(`confLIBS', `-ldl')
 define(`confEBINDIR', `/usr/sbin')
 APPENDDEF(`confLIBSEARCH', `crypt nsl')
@@ -16,6 +16,8 @@
 define(`confMTLDOPTS', `-lpthread')
 define(`confLDOPTS_SO', `-shared')
 define(`confSONAME',`-soname')
+define('confSBINGRP', 'mail')
+define('confSBINMODE', '6755')
 
 ifelse(confBLDVARIANT, `DEBUG',
 dnl Debug build
--- sendmail-8.14.4/makemap/makemap.8			2008-05-03 01:07:48.000000000 +0200
+++ sendmail-8.14.4/makemap/makemap.8.makemapman	2010-01-03 22:51:04.000000000 +0100
@@ -52,12 +52,6 @@
 parameter.  
 They may be
 .TP
-dbm
-DBM format maps.  
-This requires the 
-ndbm(3) 
-library.
-.TP
 btree
 B-Tree format maps.  
 This requires the new Berkeley DB 
--- sendmail-8.14.4/rmail/rmail.c			2001-09-18 23:45:29.000000000 +0200
+++ sendmail-8.14.4/rmail/rmail.c.makemapman		2010-01-03 22:51:36.000000000 +0100
@@ -276,7 +276,6 @@
 	args[i++] = _PATH_SENDMAIL;	/* Build sendmail's argument list. */
 	args[i++] = "-G";		/* relay submission */
 	args[i++] = "-oee";		/* No errors, just status. */
-	args[i++] = "-odq";		/* Queue it, don't try to deliver. */
 	args[i++] = "-oi";		/* Ignore '.' on a line by itself. */
 
 	/* set from system and protocol used */
