diff --git a/cf/m4/cfhead.m4 b/cf/m4/cfhead.m4
index 714a3ec..3fd6c1c 100644
--- a/cf/m4/cfhead.m4
+++ b/cf/m4/cfhead.m4
@@ -260,7 +260,7 @@ ifdef(`MAIL_SETTINGS_DIR', , `define(`MAIL_SETTINGS_DIR', `/etc/mail/')')
 define(`DATABASE_MAP_TYPE', `hash')
 
 # set up default values for options
-define(`ALIAS_FILE', `MAIL_SETTINGS_DIR`'aliases')
+define(`ALIAS_FILE', `/etc/aliases')
 define(`confMAILER_NAME', ``MAILER-DAEMON'')
 define(`confFROM_LINE', `From $g $d')
 define(`confOPERATORS', `.:%@!^/[]+')
diff --git a/sendmail/aliases.0 b/sendmail/aliases.0
index cfdbe17..5ea4c28 100644
--- a/sendmail/aliases.0
+++ b/sendmail/aliases.0
@@ -63,7 +63,7 @@ DDEESSCCRRIIPPTTIIOONN
        the list of users defined in that file.
 
        This  is  only  the  raw  data file; the actual aliasing information is
-       placed into a binary format in the file /etc/mail/aliases.db using  the
+       placed into a binary format in the file /etc/aliases.db using  the
        program  newaliases(1).   A  newaliases command should be executed each
        time the aliases file is changed for the change to take effect.
 
diff --git a/sendmail/aliases.5 b/sendmail/aliases.5
index f09b49c..7b16db2 100644
--- a/sendmail/aliases.5
+++ b/sendmail/aliases.5
@@ -23,7 +23,7 @@ ID
 aliases used by 
 sendmail.
 The file resides in 
-/etc/mail 
+/etc
 and 
 is formatted as a series of lines of the form
 .IP
@@ -96,7 +96,7 @@ list of users defined in that file.
 .PP
 This is only the raw data file; the actual aliasing information is
 placed into a binary format in the file 
-/etc/mail/aliases.db
+/etc/aliases.db
 using the program 
 newaliases(1).  
 A 
diff --git a/sendmail/newaliases.0 b/sendmail/newaliases.0
index c77f401..e2a1670 100644
--- a/sendmail/newaliases.0
+++ b/sendmail/newaliases.0
@@ -10,7 +10,7 @@ SSYYNNOOPPSSIISS
 
 DDEESSCCRRIIPPTTIIOONN
        NNeewwaalliiaasseess  rebuilds  the  random access data base for the mail aliases
-       file /etc/mail/aliases.  It must be run each time this file is  changed
+       file /etc/aliases.  It must be run each time this file is  changed
        in order for the change to take effect.
 
        NNeewwaalliiaasseess is identical to ``sendmail -bi''.
@@ -22,7 +22,7 @@ DDEESSCCRRIIPPTTIIOONN
        sseennddmmaaiill..
 
 FFIILLEESS
-       /etc/mail/aliases   The mail aliases file
+       /etc/aliases   The mail aliases file
 
 SSEEEE AALLSSOO
        aliases(5), sendmail(8)
diff --git a/sendmail/newaliases.1 b/sendmail/newaliases.1
index 59dc0de..9ba8752 100644
--- a/sendmail/newaliases.1
+++ b/sendmail/newaliases.1
@@ -20,7 +20,7 @@ newaliases
 .SH DESCRIPTION
 .B Newaliases
 rebuilds the random access data base for the mail aliases file
-/etc/mail/aliases.  It must be run each time this file is changed
+/etc/aliases.  It must be run each time this file is changed
 in order for the change to take effect.
 .PP
 .B Newaliases
@@ -40,7 +40,7 @@ puts a special token into the data base that is required by
 .B sendmail.
 .SH FILES
 .TP 2i
-/etc/mail/aliases
+/etc/aliases
 The mail aliases file
 .SH SEE ALSO
 aliases(5), sendmail(8)
diff --git a/sendmail/sendmail.0 b/sendmail/sendmail.0
index 515d5f7..8236411 100644
--- a/sendmail/sendmail.0
+++ b/sendmail/sendmail.0
@@ -434,10 +434,10 @@ FFIILLEESS
        are only approximations.
 
 
-        /etc/mail/aliases
+        /etc/aliases
               raw data for alias names
 
-        /etc/mail/aliases.db
+        /etc/aliases.db
               data base of alias names
 
         /etc/mail/sendmail.cf
diff --git a/sendmail/sendmail.8 b/sendmail/sendmail.8
index 0356839..1258c26 100644
--- a/sendmail/sendmail.8
+++ b/sendmail/sendmail.8
@@ -711,10 +711,10 @@ Thus,
 these values are only approximations.
 .PP
 .TP
- /etc/mail/aliases
+ /etc/aliases
 raw data for alias names
 .TP
- /etc/mail/aliases.db
+ /etc/aliases.db
 data base of alias names
 .TP
  /etc/mail/sendmail.cf
