diff --git a/src/cleanup/cleanup_message.c b/src/cleanup/cleanup_message.c
index f2b86d0..018e39e 100644
--- a/src/cleanup/cleanup_message.c
+++ b/src/cleanup/cleanup_message.c
@@ -772,6 +772,8 @@ static void cleanup_header_done_callback(void *context)
 		    /* Normalize whitespace. */
 		    token = tok822_scan_limit(state->fullname, &dummy_token,
 					      var_token_limit);
+		    if (!token)
+				token = tok822_alloc(TOK822_QSTRING, state->fullname);
 		} else {
 		    token = tok822_alloc(TOK822_QSTRING, state->fullname);
 		}
