diff --git a/src/daemon/cgrulesengd.c b/src/daemon/cgrulesengd.c
index 51988de..3145978 100644
--- a/src/daemon/cgrulesengd.c
+++ b/src/daemon/cgrulesengd.c
@@ -653,7 +653,7 @@ close:
 
 static int cgre_create_netlink_socket_process_msg(void)
 {
-	int sk_nl = 0, sk_unix = 0, sk_max;
+	int sk_nl = -1, sk_unix = -1, sk_max;
 	enum proc_cn_mcast_op *mcop_msg;
 	struct sockaddr_nl my_nla;
 	struct sockaddr_un saddr;
