diff --git a/src/api.c b/src/api.c
index f2991a4..a372738 100644
--- a/src/api.c
+++ b/src/api.c
@@ -375,7 +375,7 @@ int cg_chmod_recursive(struct cgroup *cgroup, mode_t dir_mode, int dirm_change,
 void cgroup_set_permissions(struct cgroup *cgroup, mode_t control_dperm, mode_t control_fperm,
 			    mode_t task_fperm)
 {
-	if (!cgroup) {
+	if (!cgroup || cgroup == NULL) {
 		/* ECGROUPNOTALLOWED */
 		cgroup_err("Cgroup, operation not allowed\n");
 		return;
