--- ron-0.12.2/Cargo.toml	2006-07-24T01:21:28+00:00
+++ ron-0.12.2/Cargo.toml	2026-06-23T11:42:51.276307+00:00
@@ -115,10 +115,6 @@
 path = "tests/147_empty_sets_serialisation.rs"
 
 [[test]]
-name = "152_bitflags"
-path = "tests/152_bitflags.rs"
-
-[[test]]
 name = "203_error_positions"
 path = "tests/203_error_positions.rs"
 
@@ -433,10 +429,6 @@
 features = ["serde"]
 default-features = false
 
-[dev-dependencies.option_set]
-version = "0.3"
-default-features = false
-
 [dev-dependencies.serde]
 version = "1.0.193"
 features = [
