diff -up sblim-cmpi-fsvol-1.5.1/Makefile.in.orig sblim-cmpi-fsvol-1.5.1/Makefile.in
--- sblim-cmpi-fsvol-1.5.1/Makefile.in.orig	2013-08-08 08:58:33.498014617 +0200
+++ sblim-cmpi-fsvol-1.5.1/Makefile.in	2013-08-08 08:58:43.045063067 +0200
@@ -274,7 +274,7 @@ datadir = @datadir@
 datarootdir = @datarootdir@
 
 # docdir defines where the documentation goes
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
