From 115eca353c55587ff6dacdadf22075904adf92d6 Mon Sep 17 00:00:00 2001
From: Mayank Singh <mayansingh@microsoft.com>
Date: Thu, 22 May 2025 11:27:30 +0000
Subject: [PATCH] Extending timeout for test cases with multiple read/writes.

---
 src/libstrongswan/tests/suites/test_settings.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libstrongswan/tests/suites/test_settings.c b/src/libstrongswan/tests/suites/test_settings.c
index 8ef026b..a7b2b98 100644
--- a/src/libstrongswan/tests/suites/test_settings.c
+++ b/src/libstrongswan/tests/suites/test_settings.c
@@ -1743,6 +1743,7 @@ Suite *settings_suite_create()
 	suite_add_tcase(s, tc);
 
 	tc = tcase_create("valid/invalid data");
+	tcase_set_timeout(tc, 10);
 	tcase_add_checked_fixture(tc, setup_base_config, teardown_config);
 	tcase_add_test(tc, test_valid);
 	tcase_add_test(tc, test_invalid);
-- 
2.45.3

