diff -Naur strongswan-5.6.0-orig/src/libstrongswan/utils/utils/memory.h strongswan-5.6.0/src/libstrongswan/utils/utils/memory.h
--- strongswan-5.6.0-orig/src/libstrongswan/utils/utils/memory.h	2017-08-14 02:48:41.000000000 -0400
+++ strongswan-5.6.0/src/libstrongswan/utils/utils/memory.h	2017-09-12 01:15:29.690527667 -0400
@@ -14,6 +14,8 @@
  * for more details.
  */
 
+#include <stdint.h> /* for uintptr_t */
+
 /**
  * @defgroup memory_i memory
  * @{ @ingroup utils_i
