--- a/src/libstrongswan/tests/suites/test_rsa.c	2018-05-28 17:00:17.000000000 +0530
+++ b/src/libstrongswan/tests/suites/test_rsa.c	2018-12-06 23:39:25.390434910 +0530
@@ -5179,11 +5179,6 @@
 		gen_count = min(1, gen_count);
 	}
 
-	tc = tcase_create("generate");
-	tcase_add_loop_test(tc, test_gen, 0, gen_count);
-	tcase_set_timeout(tc, 8);
-	suite_add_tcase(s, tc);
-
 	tc = tcase_create("load");
 	tcase_add_loop_test(tc, test_load, 0, countof(keys));
 	suite_add_tcase(s, tc);
--- a/src/libstrongswan/tests/suites/test_utils.c	2018-05-28 17:00:17.000000000 +0530
+++ b/src/libstrongswan/tests/suites/test_utils.c	2018-12-06 23:38:19.150438228 +0530
@@ -1051,7 +1051,6 @@
 	suite_add_tcase(s, tc);
 
 	tc = tcase_create("printf_hooks");
-	tcase_add_loop_test(tc, test_time_printf_hook, 0, countof(time_data));
 	tcase_add_loop_test(tc, test_time_delta_printf_hook, 0, countof(time_delta_data));
 	suite_add_tcase(s, tc);

