diff -pruN imsettings-1.6.1.orig/data/imsettings-functions.in imsettings-1.6.1/data/imsettings-functions.in
--- imsettings-1.6.1.orig/data/imsettings-functions.in	2013-04-03 16:26:45.000000000 +0900
+++ imsettings-1.6.1/data/imsettings-functions.in	2013-04-03 19:20:35.000000000 +0900
@@ -160,10 +160,11 @@ function is_qt_supported() {
 }
 
 function is_xim_supported() {
-    [ -n "$IMSETTINGS_DISABLE_DESKTOP_CHECK" ] && return 0
-    if check_imsettings_capability xim; then
-	return 0
-    fi
+# XXX: Disable XIM support so far
+#    [ -n "$IMSETTINGS_DISABLE_DESKTOP_CHECK" ] && return 0
+#    if check_imsettings_capability xim; then
+#	return 0
+#    fi
 
     return 1
 }
