diff -pruN imsettings-1.8.2.orig/imsettings-daemon/imsettings-proc.c imsettings-1.8.2/imsettings-daemon/imsettings-proc.c
--- imsettings-1.8.2.orig/imsettings-daemon/imsettings-proc.c	2019-01-18 20:52:58.000000000 +0900
+++ imsettings-1.8.2/imsettings-daemon/imsettings-proc.c	2020-01-29 13:33:07.051676780 +0900
@@ -84,7 +84,7 @@ static gboolean _stop_process
 
 G_DEFINE_TYPE_WITH_PRIVATE (IMSettingsProc, imsettings_proc, G_TYPE_OBJECT);
 
-guint signals[LAST_SIGNAL] = { 0 };
+static guint signals[LAST_SIGNAL] = { 0 };
 
 /*< private >*/
 static gboolean
diff -pruN imsettings-1.8.2.orig/imsettings-daemon/imsettings-server.c imsettings-1.8.2/imsettings-daemon/imsettings-server.c
--- imsettings-1.8.2.orig/imsettings-daemon/imsettings-server.c	2019-11-14 14:08:03.000000000 +0900
+++ imsettings-1.8.2/imsettings-daemon/imsettings-server.c	2020-01-29 13:32:50.827832726 +0900
@@ -91,7 +91,7 @@ GDBusInterfaceVTable __iface_vtable = {
 	NULL,
 	NULL,
 };
-guint signals[LAST_SIGNAL];
+static guint signals[LAST_SIGNAL];
 
 G_DEFINE_TYPE_WITH_PRIVATE (IMSettingsServer, imsettings_server, G_TYPE_OBJECT);
 G_LOCK_DEFINE_STATIC (logger);
