--- os_info-3.15.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ os_info-3.15.0/Cargo.toml	2026-05-28T15:59:11.914124+00:00
@@ -73,36 +73,3 @@
 version = "0.31"
 features = ["feature"]
 
-[target.'cfg(target_os = "android")'.dependencies.android_system_properties]
-version = "0.1"
-
-[target.'cfg(target_os = "ios")'.dependencies.objc2]
-version = "0.6"
-
-[target.'cfg(target_os = "ios")'.dependencies.objc2-foundation]
-version = "0.3"
-features = ["NSString"]
-
-[target.'cfg(target_os = "ios")'.dependencies.objc2-ui-kit]
-version = "0.3"
-
-[target.'cfg(target_os = "macos")'.dependencies.objc2-foundation]
-version = "0.3"
-features = [
-    "NSData",
-    "NSError",
-    "NSEnumerator",
-    "NSString",
-]
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.61"
-features = [
-    "Win32_Foundation",
-    "Win32_System_LibraryLoader",
-    "Win32_System_Registry",
-    "Win32_System_SystemInformation",
-    "Win32_System_SystemServices",
-    "Win32_System_Threading",
-    "Win32_UI_WindowsAndMessaging",
-]
