--- http-range-header-0.4.2/Cargo.toml	1970-01-01T00:00:01+00:00
+++ http-range-header-0.4.2/Cargo.toml	2025-01-03T16:04:40.655300+00:00
@@ -47,15 +47,7 @@
 name = "http_range_header"
 path = "src/lib.rs"
 
-[[bench]]
-name = "benchmark"
-path = "benches/benchmark.rs"
-harness = false
-
 [dependencies]
-
-[dev-dependencies.criterion]
-version = "0.5.1"
 
 [dev-dependencies.quickcheck]
 version = "1.0.3"
