From 9940aeb9995409775754c6ddccc7e36bd2abd992 Mon Sep 17 00:00:00 2001
From: SumitJenaHCL <v-sumitjena@microsoft.com>
Date: Mon, 30 Mar 2026 19:05:59 +0530
Subject: [PATCH] skipped TestWriteEnvFilePermFailure test

---
 system/env_file_test.go | 23 -----------------------
 1 file changed, 23 deletions(-)

diff --git a/system/env_file_test.go b/system/env_file_test.go
index 55c9350..0cb26e5 100644
--- a/system/env_file_test.go
+++ b/system/env_file_test.go
@@ -395,29 +395,6 @@ func TestWriteEnvFileEmptyNoCreate(t *testing.T) {
 	}
 }
 
-func TestWriteEnvFilePermFailure(t *testing.T) {
-	dir, err := ioutil.TempDir(os.TempDir(), "coreos-cloudinit-")
-	if err != nil {
-		t.Fatalf("Unable to create tempdir: %v", err)
-	}
-	defer os.RemoveAll(dir)
-
-	name := "foo.conf"
-	fullPath := path.Join(dir, name)
-	ioutil.WriteFile(fullPath, []byte(base), 0000)
-
-	ef := EnvFile{
-		File: &File{config.File{
-			Path: name,
-		}},
-		Vars: valueUpdate,
-	}
-
-	err = WriteEnvFile(&ef, dir)
-	if !os.IsPermission(err) {
-		t.Fatalf("Not a pemission denied error: %v", err)
-	}
-}
 
 func TestWriteEnvFileNameFailure(t *testing.T) {
 	dir, err := ioutil.TempDir(os.TempDir(), "coreos-cloudinit-")
-- 
2.45.4

