diff -up efax-0.9a-001114/efix.c.orig efax-0.9a-001114/efix.c
--- efax-0.9a-001114/efix.c.orig	2009-09-07 15:55:23.000000000 +0200
+++ efax-0.9a-001114/efix.c	2009-09-07 15:56:34.000000000 +0200
@@ -235,7 +235,8 @@ int main( int argc, char **argv)
   float				/* values used: */
     xres = 0, yres = 0, xsz = 0, ysz = 0 ;
 
-  IFILE ifile, ovfile ;
+  IFILE ifile = { 0 };
+  IFILE ovfile = { 0 };
   OFILE ofile ;
 
   char **ifnames,  *ovfnames [ 2 ] = { 0, 0 } ;
