--- efax-0.9/efaxmsg.c.ORIG	1999-03-19 22:38:20.000000000 -0600
+++ efax-0.9/efaxmsg.c	2008-10-14 15:39:12.000000000 -0500
@@ -142,6 +142,8 @@ int msg ( char *fmt, ... ) 
 		logfile[i] ) ;
       }
       vfprintf( logfile[i], p, ap ) ;
+      va_end ( ap );
+      va_start ( ap, fmt );
       if ( flags & S ) fprintf ( logfile[i], " %s", strerror ( errno ) ) ;
       if ( ! ( flags & NOLF ) ) fputs ( "\n", logfile[i] ) ;
       atcol1[i] = flags & NOLF ? 0 : 1 ;
