--- efax-0.9/efax.c.nullptr	Sat Mar 27 09:24:10 1999
+++ efax-0.9/efax.c	Wed Jan 12 22:17:21 2000
@@ -2186,6 +2186,7 @@
   
   /* print initial message to both stderr & stdout */
   argv0 = argv[0] ;
+  memset((void *)lkfile, 0, sizeof(lkfile));  /* random effects on sparc linux */
   verb[1] = "ewia" ;
   msg ( "I " Version " " Copyright ) ;
   argv0 = efaxbasename ( argv0 ) ;
--- efax-0.9/efaxos.c.nullptr	Tue Mar  2 01:18:30 1999
+++ efax-0.9/efaxos.c	Wed Jan 12 22:16:04 2000
@@ -482,7 +482,7 @@
 { 
   int err = 0 ;
   char **p = lkfiles ;
-  while ( *p && ! err ) 
+  while ( *p && **p && ! err ) 
     if ( ( err = ttlock ( *p++, log ) ) == 3 ) err = 0 ; 
   return err ; 
 }
@@ -495,7 +495,7 @@
 { 
   int err = 0, i ;
   char **p = lkfiles ;
-  while ( *p ) 
+  while ( *p && **p ) 
     if ( ( i = ttunlock ( *p++ ) ) != 0 ) err = i ; 
   return err ; 
 }
