diff -up efax-0.9a-001114/efaxlib.c.me efax-0.9a-001114/efaxlib.c
--- efax-0.9a-001114/efaxlib.c.me	2020-01-30 15:17:45.367168505 +0100
+++ efax-0.9a-001114/efaxlib.c	2020-01-30 15:18:38.615050032 +0100
@@ -25,6 +25,8 @@ extern t4tab btab [ ( 64 + 27 + 13 ) + 1
 
 short short256 = 256 ;		/* for endian-ness detection */
 
+uchar reversebits [ 256 ], normalbits [ 256 ] ;
+
 /* Make sure printf strings have only %d escapes and n or fewer
    of them.  Returns 0 if OK, 1 on error. */
 
diff -up efax-0.9a-001114/efaxlib.h.me efax-0.9a-001114/efaxlib.h
--- efax-0.9a-001114/efaxlib.h.me	2020-01-30 15:17:07.914550036 +0100
+++ efax-0.9a-001114/efaxlib.h	2020-01-30 15:17:25.710843913 +0100
@@ -208,7 +208,7 @@ int runor ( short *a, int na, short *b,
 /* Bit reversal lookup tables (note that the `normalbits' array
    is the one actually used for the bit reversal.  */
 
-uchar reversebits [ 256 ], normalbits [ 256 ] ;
+extern uchar reversebits [ 256 ], normalbits [ 256 ] ;
 
 void initbittab(void) ;
 
