diff -ruN a/elasticsearch/elasticsearch.gemspec b/elasticsearch/elasticsearch.gemspec
--- a/elasticsearch/elasticsearch.gemspec	2022-07-07 09:24:12.069738351 -0700
+++ b/elasticsearch/elasticsearch.gemspec	2022-07-07 09:25:33.998637120 -0700
@@ -33,7 +33,7 @@
     'source_code_uri' => 'https://github.com/elastic/elasticsearch-ruby/tree/main',
     'bug_tracker_uri' => 'https://github.com/elastic/elasticsearch-ruby/issues'
   }
-  s.files = `git ls-files`.split($/)
+  s.files = Dir['**/*']
   s.executables = s.files.grep(%r{^bin/}) { |f| File.basename(f) }
   s.executables << 'elastic_ruby_console'
   s.test_files = s.files.grep(%r{^(test|spec|features)/})
