--- a/QT/PstoeditQtGui/PstoeditQtGui.orig.pro	2026-02-21 19:48:10.600329677 +0100
+++ b/QT/PstoeditQtGui/PstoeditQtGui.pro	2026-02-21 19:47:42.327765740 +0100
@@ -8,6 +8,9 @@
 # In order to do so, uncomment the following line.
 #DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0x060000    # disables all the APIs deprecated before Qt 6.0.0
 
+QMAKE_CXXFLAGS += $(OPTFLAGS)
+QMAKE_CFLAGS += $(OPTFLAGS)
+
 SOURCES += \
     PstoeditQtGuiMain.cpp \
     PstoeditQtGui.cpp
