--- gnome-session/data/hardware-compatibility	2012-03-21 16:30:06.269104695 -0700
+++ gnome-session/data/hardware-compatibility.new	2012-03-22 23:26:37.201967075 -0700
@@ -19,6 +19,9 @@
 -Mesa DRI R[12]00[^[:digit:]]
 -Mesa DRI R[12]00$
 
+# NV30 family on Nouveau: https://bugzilla.redhat.com/show_bug.cgi?id=745202
+-Gallium .* on NV3[0-9A-F]$
+
 # Old Mesa software GL renderer
 -software rasterizer
 
