--- socket2-0.6.4/Cargo.toml	2006-07-24T01:21:28+00:00
+++ socket2-0.6.4/Cargo.toml	2026-05-29T18:42:10.842419+00:00
@@ -93,12 +93,3 @@
 [target.'cfg(any(unix, target_os = "wasi"))'.dependencies.libc]
 version = "0.2.172"
 
-[target."cfg(windows)".dependencies.windows-sys]
-version = ">=0.60, <0.62"
-features = [
-    "Win32_Foundation",
-    "Win32_Networking_WinSock",
-    "Win32_System_IO",
-    "Win32_System_Threading",
-    "Win32_System_WindowsProgramming",
-]
