diff -rupN --no-dereference CGNS-4.5.1/src/ptests/CMakeLists.txt CGNS-4.5.1-new/src/ptests/CMakeLists.txt
--- CGNS-4.5.1/src/ptests/CMakeLists.txt	2025-12-29 19:02:10.000000000 +0100
+++ CGNS-4.5.1-new/src/ptests/CMakeLists.txt	2025-12-30 09:38:23.250109024 +0100
@@ -45,51 +45,15 @@ set(C_PTESTS_LIST
 	test_poly_unstructured
 	thesis_benchmark)
 
-set(pcgns_ctest_FILES pcgns_ctest.c)
-add_executable(pcgns_ctest ${pcgns_ctest_FILES})
+#
+function(add_test_exe file)
+  add_executable (${file} ${file}.c)
+  target_link_libraries (${file} PRIVATE ${PTESTS_LK_LIBS})
+endfunction ()
 
-set(cexample_FILES cexample.c)
-add_executable(cexample ${cexample_FILES})
-
-set(benchmark_FILES benchmark.c)
-add_executable(benchmark ${benchmark_FILES})
-
-set(benchmark_hdf5_FILES benchmark_hdf5.c)
-add_executable(benchmark_hdf5 ${benchmark_hdf5_FILES})
-
-set(Metadata_Stressing_FILES Metadata_Stressing.c)
-add_executable(Metadata_Stressing ${Metadata_Stressing_FILES})
-
-set(open_close_FILES open_close.c)
-add_executable(open_close ${open_close_FILES})
-
-set(comm_test_FILES comm_test.c)
-add_executable(comm_test ${comm_test_FILES})
-
-set(test_base_FILES test_base.c)
-add_executable(test_base ${test_base_FILES})
-
-set(test_unstructured_FILES test_unstructured.c)
-add_executable(test_unstructured ${test_unstructured_FILES})
-
-set(test_unstruc_quad_FILES test_unstruc_quad.c)
-add_executable(test_unstruc_quad ${test_unstruc_quad_FILES})
-
-set(test_zone_FILES test_zone.c)
-add_executable(test_zone ${test_zone_FILES})
-
-set(test_general_readwrite_FILES test_general_readwrite.c)
-add_executable(test_general_readwrite ${test_general_readwrite_FILES})
-
-set(test_poly_unstructured_FILES test_poly_unstructured.c)
-add_executable(test_poly_unstructured ${test_poly_unstructured_FILES})
-
-set(thesis_benchmark_FILES thesis_benchmark.c)
-add_executable(thesis_benchmark ${thesis_benchmark_FILES})
-
-# Linking
+# Define C tests
 foreach(TGTNAME ${C_PTESTS_LIST})
-  target_link_libraries(${TGTNAME} PRIVATE ${PTESTS_LK_LIBS})
+  add_test_exe(${TGTNAME})
 endforeach()
 
 if (CGNS_ENABLE_FORTRAN AND HAVE_FORTRAN_2003)
@@ -101,21 +65,15 @@ if (CGNS_ENABLE_FORTRAN AND HAVE_FORTRAN
     test_unstruc_quad_f90
     test_poly_unstructured_f90
   )
-  set(pcgns_ftest_FILES pcgns_ftest.F90 ${CMAKE_SOURCE_DIR}/src/tests/utilsf.F90)
-  add_executable(pcgns_ftest ${pcgns_ftest_FILES})
-  set(fexample_FILES fexample.F90)
-  add_executable(fexample ${fexample_FILES})
-  set(benchmark_hdf5_f90_FILES benchmark_hdf5_f90.F90)
-  add_executable(benchmark_hdf5_f90 ${benchmark_hdf5_f90_FILES})
-  set(test_mixed_par_ser_FILES test_mixed_par_ser.F90)
-  add_executable(test_mixed_par_ser ${test_mixed_par_ser_FILES})
-  set(test_unstruc_quad_f90_FILES test_unstruc_quad_f90.F90)
-  add_executable(test_unstruc_quad_f90 ${test_unstruc_quad_f90_FILES})
-  set(test_poly_unstructured_f90_FILES test_poly_unstructured_f90.F90 ${CMAKE_SOURCE_DIR}/src/tests/utilsf.F90)
-  add_executable(test_poly_unstructured_f90 ${test_poly_unstructured_f90_FILES})
-  # Linking
+  add_library(test_utilsf OBJECT ${CMAKE_SOURCE_DIR}/src/tests/utilsf.F90)
+
+  function (add_fortran_exe file)
+    add_executable (${file} ${file}.F90)
+    target_link_libraries (${file} PRIVATE ${PTESTS_LK_LIBS} test_utilsf)
+  endfunction ()
+
   foreach(TGTNAME ${FORTRAN_PTESTS_LIST})
-    target_link_libraries(${TGTNAME} PRIVATE ${PTESTS_LK_LIBS})
+    add_fortran_exe(${TGTNAME})
   endforeach()
 endif ()
 
