diff -up mutt-2.4.0/doc/Makefile.am.lynx_no_backscapes mutt-2.4.0/doc/Makefile.am
--- mutt-2.4.0/doc/Makefile.am.lynx_no_backscapes	2026-06-12 07:03:22.000000000 +0200
+++ mutt-2.4.0/doc/Makefile.am	2026-06-28 22:59:45.674628590 +0200
@@ -150,7 +150,7 @@ uninstall-local:
 	fi
 
 manual.txt: manual.html
-	-LC_ALL=C lynx -localhost -dump -nolist -nonumbers -with_backspaces -display_charset=us-ascii manual.html > $@ || \
+	-LC_ALL=C lynx -localhost -dump -nolist -display_charset=us-ascii manual.html > $@ || \
 	LC_ALL=C w3m -T text/html -I utf-8 -O utf-8 -dump < manual.html > $@ || \
 	LC_ALL=C elinks -dump -no-numbering -no-references manual.html | sed -e 's,\\001, ,g' > $@
 
