From 56d9b5d8ddbf7de2fdf8d68edac8822b43a5908f Mon Sep 17 00:00:00 2001
From: Diego Herrera <dherrera@redhat.com>
Date: Tue, 30 Sep 2025 01:40:00 -0300
Subject: [PATCH 17/19] disable downloading dependencies

---
 .../external/onnxruntime_external_deps.cmake  | 31 -------------------
 1 file changed, 31 deletions(-)

diff --git a/cmake/external/onnxruntime_external_deps.cmake b/cmake/external/onnxruntime_external_deps.cmake
index f16f08cc05..900e4bfa30 100644
--- a/cmake/external/onnxruntime_external_deps.cmake
+++ b/cmake/external/onnxruntime_external_deps.cmake
@@ -2,37 +2,6 @@ message(STATUS "Loading Dependencies URLs ...")
 
 include(external/helper_functions.cmake)
 
-file(STRINGS deps.txt ONNXRUNTIME_DEPS_LIST)
-foreach(ONNXRUNTIME_DEP IN LISTS ONNXRUNTIME_DEPS_LIST)
-  # Lines start with "#" are comments
-  if(NOT ONNXRUNTIME_DEP MATCHES "^#")
-    # The first column is name
-    list(POP_FRONT ONNXRUNTIME_DEP ONNXRUNTIME_DEP_NAME)
-    # The second column is URL
-    # The URL below may be a local file path or an HTTPS URL
-    list(POP_FRONT ONNXRUNTIME_DEP ONNXRUNTIME_DEP_URL)
-    set(DEP_URL_${ONNXRUNTIME_DEP_NAME} ${ONNXRUNTIME_DEP_URL})
-    # The third column is SHA1 hash value
-    set(DEP_SHA1_${ONNXRUNTIME_DEP_NAME} ${ONNXRUNTIME_DEP})
-
-    if(ONNXRUNTIME_DEP_URL MATCHES "^https://")
-      # Search a local mirror folder
-      string(REGEX REPLACE "^https://" "${REPO_ROOT}/mirror/" LOCAL_URL "${ONNXRUNTIME_DEP_URL}")
-
-      if(EXISTS "${LOCAL_URL}")
-        cmake_path(ABSOLUTE_PATH LOCAL_URL)
-        set(DEP_URL_${ONNXRUNTIME_DEP_NAME} "${LOCAL_URL}")
-      endif()
-    endif()
-  endif()
-endforeach()
-
-message(STATUS "Loading Dependencies ...")
-include(FetchContent)
-
-# ABSL should be included before protobuf because protobuf may use absl
-include(external/abseil-cpp.cmake)
-
 set(RE2_BUILD_TESTING OFF CACHE BOOL "" FORCE)
 
 onnxruntime_fetchcontent_declare(
-- 
2.51.0

