--- jna-4.5.2/native/Makefile.orig	2020-03-25 20:56:44.585855363 +0000
+++ jna-4.5.2/native/Makefile	2020-03-25 20:57:38.661808289 +0000
@@ -336,10 +336,10 @@
 ifeq ($(CC),gcc)
     GCC_MAJOR_VERSION = $(shell gcc -dumpversion | cut -f 1 -d '.')
     ifneq ($(GCC_MAJOR_VERSION),4)
-	LOC_CC_OPTS=-Wno-unknown-warning-option -Werror -Wno-clobbered -Wno-unused-variable
+	LOC_CC_OPTS=-Wno-unknown-warning-option -Wno-clobbered -Wno-unused-variable
     endif
 else
-    LOC_CC_OPTS=-Wno-unknown-warning-option -Werror -Wno-clobbered -Wno-unused-variable
+    LOC_CC_OPTS=-Wno-unknown-warning-option -Wno-clobbered -Wno-unused-variable
 endif
 
 # Enable 64-bit builds if the arch demands it
