From 4105bd51e654df8b40c53eb4b4abe7a3415783eb Mon Sep 17 00:00:00 2001
From: Tom Rix <Tom.Rix@amd.com>
Date: Fri, 31 Oct 2025 07:28:05 -0700
Subject: [PATCH] hipsparse change test download dir

---
 clients/tests/CMakeLists.txt | 22 ++++++++++++----------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/clients/tests/CMakeLists.txt b/clients/tests/CMakeLists.txt
index 15a7c21f4c30..69beb9da6208 100644
--- a/clients/tests/CMakeLists.txt
+++ b/clients/tests/CMakeLists.txt
@@ -27,11 +27,13 @@ find_package(GTest REQUIRED)
 #
 # Client matrices.
 #
+option(CMAKE_MATRICES_DIR "Directory containing test matrices" ${PROJECT_BINARY_DIR}/matrices)
+
 if(NOT EXISTS "${CMAKE_MATRICES_DIR}")
   #
   # Download.
   #
-  set(CMAKE_MATRICES_DIR ${PROJECT_BINARY_DIR}/matrices CACHE STRING "Matrices directory.")
+
   file(MAKE_DIRECTORY ${CMAKE_MATRICES_DIR})
 
   if(NOT TARGET hipsparse)
@@ -46,15 +48,15 @@ else()
   #
   # Copy.
   #
-  if(NOT CMAKE_MATRICES_DIR STREQUAL "${PROJECT_BINARY_DIR}/matrices")
-    message("Copy matrix files from ${CMAKE_MATRICES_DIR} to ${PROJECT_BINARY_DIR}/matrices")
-
-    execute_process(COMMAND cp -r ${CMAKE_MATRICES_DIR} ${PROJECT_BINARY_DIR}/matrices RESULT_VARIABLE STATUS WORKING_DIRECTORY ${CMAKE_MATRICES_DIR})
-
-    if(STATUS AND NOT STATUS EQUAL 0)
-      message(FATAL_ERROR "Failed to copy matrix .bin files, aborting.")
-    endif()
-  endif()
+  #  if(NOT CMAKE_MATRICES_DIR STREQUAL "${PROJECT_BINARY_DIR}/matrices")
+  #    message("Copy matrix files from ${CMAKE_MATRICES_DIR} to ${PROJECT_BINARY_DIR}/matrices")
+  #
+  #    execute_process(COMMAND cp -r ${CMAKE_MATRICES_DIR} ${PROJECT_BINARY_DIR}/matrices RESULT_VARIABLE STATUS WORKING_DIRECTORY ${CMAKE_MATRICES_DIR})
+  #
+  #    if(STATUS AND NOT STATUS EQUAL 0)
+  #      message(FATAL_ERROR "Failed to copy matrix .bin files, aborting.")
+  #    endif()
+  #  endif()
 
 endif()
 
-- 
2.51.0

