--- ttmkfdir-3.0.9/ttf.cpp.sopwith	Thu Aug  7 12:49:08 2003
+++ ttmkfdir-3.0.9/ttf.cpp	Thu Aug  7 12:50:01 2003
@@ -239,7 +239,7 @@
 	for (i = 0; i < n; i++) {
 	    if ((fterror = FT_Get_Sfnt_Name (face, i, &NamePtr)) != FT_Err_Ok) {
                     std::cout << "Warning: Can't SFNT name : " << FileName << "(" << fterror << ")" << std::endl;
-	            return;
+	            return NULL;
             };
             platform = NamePtr.platform_id;
             encoding = NamePtr.encoding_id;
