--- dircpy-0.3.19/Cargo.toml	1970-01-01T00:00:01+00:00
+++ dircpy-0.3.19/Cargo.toml	2024-08-09T11:23:39.610112+00:00
@@ -25,10 +25,6 @@
 license = "MIT"
 repository = "https://github.com/woelper/dircpy/"
 
-[[bench]]
-name = "copy_rustlang"
-harness = false
-
 [dependencies.jwalk]
 version = "0.8"
 optional = true
@@ -39,18 +35,8 @@
 [dependencies.walkdir]
 version = "2.5"
 
-[dev-dependencies.criterion]
-version = "0.5"
-
 [dev-dependencies.env_logger]
 version = "0.11"
 
-[dev-dependencies.reqwest]
-version = "0.12"
-features = ["blocking"]
-
-[dev-dependencies.unzip]
-version = "0.1"
-
 [features]
 default = ["jwalk"]
