From 6bf97bc3834e88febd30fbad690caddc59adc145 Mon Sep 17 00:00:00 2001
From: Olivier Lemasle <o.lemasle@gmail.com>
Date: Tue, 27 Oct 2020 21:59:41 +0100
Subject: [PATCH] Make telemetry opt-out

---
 cmd/run.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmd/run.go b/cmd/run.go
index 0283937c..12811506 100644
--- a/cmd/run.go
+++ b/cmd/run.go
@@ -206,7 +206,7 @@ To skip bundle verification, use the --skip-verify flag.
 	addConfigOverrideFiles(runCommand.Flags(), &cmdParams.rt.ConfigOverrideFiles)
 	addBundleModeFlag(runCommand.Flags(), &cmdParams.rt.BundleMode, false)
 
-	runCommand.Flags().BoolVar(&cmdParams.skipVersionCheck, "skip-version-check", false, "disables anonymous version reporting (see: https://www.openpolicyagent.org/docs/latest/privacy)")
+	runCommand.Flags().BoolVar(&cmdParams.skipVersionCheck, "skip-version-check", true, "disables anonymous version reporting (see: https://www.openpolicyagent.org/docs/latest/privacy)")
 	err := runCommand.Flags().MarkDeprecated("skip-version-check", "\"skip-version-check\" is deprecated. Use \"disable-telemetry\" instead")
 	if err != nil {
 		fmt.Println("error:", err)
