--- mio-1.1.1/Cargo.toml	2006-07-24T01:21:28+00:00
+++ mio-1.1.1/Cargo.toml	2025-12-28T17:33:21.251722+00:00
@@ -80,8 +80,6 @@
 net = []
 os-ext = [
     "os-poll",
-    "windows-sys/Win32_System_Pipes",
-    "windows-sys/Win32_Security",
 ]
 os-poll = []
 
@@ -124,31 +122,8 @@
 [dev-dependencies.rand]
 version = "0.9"
 
-[target.'cfg(target_os = "hermit")'.dependencies.libc]
-version = "0.2.178"
-
-[target.'cfg(target_os = "wasi")'.dependencies.libc]
-version = "0.2.178"
-
-[target.'cfg(target_os = "wasi")'.dependencies.wasi]
-version = "0.11.0"
-
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.178"
-
-[target."cfg(windows)".dependencies.windows-sys]
-version = "0.61"
-features = [
-    "Wdk_Foundation",
-    "Wdk_Storage_FileSystem",
-    "Wdk_System_IO",
-    "Win32_Foundation",
-    "Win32_Networking_WinSock",
-    "Win32_Storage_FileSystem",
-    "Win32_Security",
-    "Win32_System_IO",
-    "Win32_System_WindowsProgramming",
-]
 
 [lints.rust.unexpected_cfgs]
 level = "warn"
@@ -157,3 +132,4 @@
     "cfg(mio_unsupported_force_poll_poll)",
     "cfg(mio_unsupported_force_waker_pipe)",
 ]
+
