--- scc-2.4.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ scc-2.4.0/Cargo.toml	2025-10-22T12:59:37.205781+00:00
@@ -40,47 +40,12 @@
 license = "Apache-2.0"
 repository = "https://github.com/wvwwvwwv/scalable-concurrent-containers/"
 
-[features]
-loom = [
-    "dep:loom",
-    "sdd/loom",
-]
-
 [lib]
 name = "scc"
 path = "src/lib.rs"
 
-[[bench]]
-name = "bag"
-path = "benches/bag.rs"
-harness = false
-
-[[bench]]
-name = "hash_cache"
-path = "benches/hash_cache.rs"
-harness = false
-
-[[bench]]
-name = "hash_index"
-path = "benches/hash_index.rs"
-harness = false
-
-[[bench]]
-name = "hash_map"
-path = "benches/hash_map.rs"
-harness = false
-
-[[bench]]
-name = "tree_index"
-path = "benches/tree_index.rs"
-harness = false
-
 [dependencies.equivalent]
 version = "1.0"
-optional = true
-
-[dependencies.loom]
-version = "0.7"
 optional = true
 
 [dependencies.sdd]
@@ -89,9 +54,6 @@
 [dependencies.serde]
 version = "1.0"
 optional = true
-
-[dev-dependencies.criterion]
-version = "0.7"
 
 [dev-dependencies.futures]
 version = "0.3"
