From 7f63349fd0405439532f2ff0896a3697c73ccaaf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Thu, 27 Feb 2020 15:20:20 +0100
Subject: [PATCH] Fix meson_vapi_link.py to respect DESTDIR

Fixes #5
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 6d314b5..3a423b3 100644
--- a/meson.build
+++ b/meson.build
@@ -135,7 +135,7 @@ if enable_vapi
   assert(enable_gir, 'vapi support was requested, but introspection support is mandatory.')
   assert(add_languages('vala', required: false), 'vapi support was requested, but vala not found.')
 
-  meson.add_install_script('meson_vapi_link.py', libgit2_glib_datadir)
+  meson.add_install_script('meson_vapi_link.py', get_option('datadir'))
 endif
 
 # Check for libgit2 ssh support
-- 
2.24.1

