From f02a1b1a00e6c82ff280bb47df8fd7d620dc61e3 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Wed, 23 Jul 2025 17:31:23 +0000
Subject: [PATCH] AZL3 -- disable DSA in speed in fips mode

---
 apps/speed.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/apps/speed.c b/apps/speed.c
index a3d3ca8..457a662 100644
--- a/apps/speed.c
+++ b/apps/speed.c
@@ -2552,11 +2552,14 @@ int speed_main(int argc, char **argv)
     }
 
     // TOBIASB: Disable RSA sizes that aren't supported by FIPS
+    // TOBIASB: Disable DSA for FIPS provider bcecause it isn't supported.
     if (FIPS_mode()) {
         for (i = 0; i < RSA_NUM; i++) {
             if (rsa_keys[i].bits < 2048)
                 rsa_doit[i] = 0;
         }
+
+        memset(dsa_doit, 0, sizeof(dsa_doit));
     }
 
     for (i = 0; i < ALGOR_NUM; i++)
-- 
2.45.4

