From d60b9128f89fd8490ec19e265e963a2f973de180 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Tue, 26 Aug 2025 21:33:09 +0000
Subject: [PATCH] AZL3 allow normal failure for rsa in 30-test_pairwise_fail.t

---
 test/recipes/30-test_pairwise_fail.t | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/test/recipes/30-test_pairwise_fail.t b/test/recipes/30-test_pairwise_fail.t
index d87088d..a3d4ef1 100644
--- a/test/recipes/30-test_pairwise_fail.t
+++ b/test/recipes/30-test_pairwise_fail.t
@@ -31,12 +31,9 @@ run(test(["fips_version_test", "-config", $provconf, ">=3.1.0"]),
 SKIP: {
     skip "Skip RSA test because of no rsa in this build", 1
         if disabled("rsa");
-    with({ exit_checker => sub {my $val = shift; return $val == 134; } },
-    sub {
     ok(run(test(["pairwise_fail_test", "-config", $provconf,
                  "-pairwise", "rsa"])),
        "fips provider rsa keygen pairwise failure test");
-    });
 }
 
 SKIP: {
-- 
2.45.4

