diff -up libdap-3.20.3/tests/Makefile.am.orig libdap-3.20.3/tests/Makefile.am
--- libdap-3.20.3/tests/Makefile.am.orig	2019-02-14 10:13:49.000000000 -0700
+++ libdap-3.20.3/tests/Makefile.am	2019-02-17 15:27:32.426858748 -0700
@@ -120,7 +120,6 @@ $(EXPRTESTSUITE) $(DMRTESTSUITE) $(GETDA
 	$(SHELL) '$(DDSTESTSUITE)' $(TESTSUITEFLAGS)
 	$(SHELL) '$(EXPRTESTSUITE)' $(TESTSUITEFLAGS)
 	$(SHELL) '$(DMRTESTSUITE)' $(TESTSUITEFLAGS)
-	$(SHELL) '$(GETDAPTESTSUITE)' $(TESTSUITEFLAGS)
 
 # Make sure there are no spaces after the DASTESTSUITE names. jhrg 3/16/15
 # Delete the scripts here and not using CLEANFILES because this target is
diff -up libdap-3.20.3/unit-tests/Makefile.am.orig libdap-3.20.3/unit-tests/Makefile.am
--- libdap-3.20.3/unit-tests/Makefile.am.orig	2019-02-17 15:27:32.426858748 -0700
+++ libdap-3.20.3/unit-tests/Makefile.am	2019-02-17 15:28:10.725606175 -0700
@@ -78,9 +78,9 @@ D4-xml.tar.gz: D4-xml/DMR_*[0-9].xml
 if CPPUNIT
 UNIT_TESTS = marshT arrayT attrTableT structT sequenceT ddsT dasT \
 	RegexTest ArrayTest AttrTableTest ByteTest MIMEUtilTest ancT DASTest \
-	DDSTest	DDXParserTest  generalUtilTest HTTPConnectTest parserUtilTest \
+	DDSTest	DDXParserTest  generalUtilTest parserUtilTest \
 	RCReaderTest SequenceTest SignalHandlerTest  MarshallerTest \
-	HTTPCacheTest ServerFunctionsListUnitTest Int8Test Int16Test UInt16Test \
+	ServerFunctionsListUnitTest Int8Test Int16Test UInt16Test \
 	Int32Test UInt32Test Int64Test UInt64Test Float32Test Float64Test \
 	D4BaseTypeFactoryTest BaseTypeFactoryTest
 
