diff --git a/tests/Makefile.am b/tests/Makefile.am
index d325e52..123e73d 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -122,7 +122,6 @@ $(EXPRTESTSUITE) $(DMRTESTSUITE) $(GETDAPTESTSUITE)
 	$(SHELL) '$(DDSTESTSUITE)' $(TESTSUITEFLAGS)
 	$(SHELL) '$(EXPRTESTSUITE)' $(TESTSUITEFLAGS)
 	$(SHELL) '$(DMRTESTSUITE)' $(TESTSUITEFLAGS)
-	$(SHELL) '$(GETDAPTESTSUITE)' $(TESTSUITEFLAGS)
 
 # Make sure there are no spaces after the DASTESTSUITE names. jhrg 3/16/15
 # Delete the scripts here and not using CLEANFILES because this target is
diff --git a/unit-tests/Makefile.am b/unit-tests/Makefile.am
index 58ecd20..abbf860 100644
--- a/unit-tests/Makefile.am
+++ b/unit-tests/Makefile.am
@@ -85,9 +85,9 @@ D4-xml.tar.gz: D4-xml/DMR_*[0-9].xml
 if CPPUNIT
 UNIT_TESTS = marshT arrayT attrTableT structT sequenceT ddsT dasT \
 	RegexTest ArrayTest GridTest AttrTableTest ByteTest MIMEUtilTest ancT DASTest \
-	DDSTest	DDXParserTest  generalUtilTest HTTPConnectTest parserUtilTest \
+	DDSTest	DDXParserTest  generalUtilTest parserUtilTest \
 	RCReaderTest SequenceTest SignalHandlerTest  MarshallerTest \
-	HTTPCacheTest ServerFunctionsListUnitTest Int8Test Int16Test UInt16Test \
+	ServerFunctionsListUnitTest Int8Test Int16Test UInt16Test \
 	Int32Test UInt32Test Int64Test UInt64Test Float32Test Float64Test \
 	D4BaseTypeFactoryTest BaseTypeFactoryTest util_mitTest
 
