diff --git a/docs/meson.build b/docs/meson.build
index 4c5ba64..9b87e36 100644
--- a/docs/meson.build
+++ b/docs/meson.build
@@ -1,5 +1,5 @@
 docbook_docs = get_option('docbook_docs')
-if not docbook_docs.disabled()
+if docbook_docs.disabled()
   xmlto = find_program('xmlto', required: docbook_docs.enabled())
 
   if xmlto.found()
diff --git a/meson.build b/meson.build
index 088b7ed..bafeb7d 100644
--- a/meson.build
+++ b/meson.build
@@ -41,10 +41,10 @@ glib_req_version = '>= 2.26.0'
 gdk_pixbuf_dep = dependency('gdk-pixbuf-2.0')
 glib_dep = dependency('glib-2.0', version: glib_req_version)
 gio_dep = dependency('gio-2.0', version: glib_req_version)
-gtk_dep = dependency('gtk+-3.0', version: '>= 2.90', required: get_option('tests'))
+gtk_dep = []
 
 libnotify_deps = [gdk_pixbuf_dep, gio_dep, glib_dep]
-tests_deps = [gtk_dep]
+tests_deps = []
 
 conf = configuration_data()
 conf.set_quoted('VERSION', meson.project_version())
diff --git a/meson_options.txt b/meson_options.txt
index c57b364..c4b9c0c 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,6 +1,6 @@
 option('tests',
   type: 'boolean',
-  value: true,
+  value: false,
   description: 'Build tests')
 option('introspection',
   type: 'feature',
@@ -12,7 +12,7 @@ option('man',
   description: 'Enable generating the manual page (depends on xsltproc)')
 option('gtk_doc',
   type: 'boolean',
-  value: true,
+  value: false,
   description: 'Enable generating the API reference (depends on GTK-Doc)')
 option('docbook_docs',
   type: 'feature',
