--- a/libipt/test/src/ptunit-msec_cache.c	2022-01-27 09:32:59.724730213 -0800
+++ b/libipt/test/src/ptunit-msec_cache.c	2022-01-27 09:31:11.396032237 -0800
@@ -297,7 +297,7 @@
 static struct ptunit_result fill_nomap(struct test_fixture *tfix)
 {
 	const struct pt_mapped_section *msec;
-	struct pt_asid asid;
+	struct pt_asid asid = { 0 };
 	struct pt_section *section;
 	int status;
 
@@ -322,7 +322,7 @@
 {
 	const struct pt_mapped_section *msec;
 	struct pt_section *section;
-	struct pt_asid asid;
+	struct pt_asid asid = { 0 };
 	int status;
 
 	memset(&asid, 0, sizeof(asid));
