diff -up guile-1.8.7/libguile/Makefile.in.multilib guile-1.8.7/libguile/Makefile.in
--- guile-1.8.7/libguile/Makefile.in.multilib	2009-07-05 22:25:09.000000000 +0200
+++ guile-1.8.7/libguile/Makefile.in	2009-07-16 18:07:44.000000000 +0200
@@ -1717,6 +1717,11 @@ install-nodist_modincludeHEADERS: $(nodi
 	while read files; do \
 	  echo " $(INSTALL_HEADER) $$files '$(DESTDIR)$(modincludedir)'"; \
 	  $(INSTALL_HEADER) $$files "$(DESTDIR)$(modincludedir)" || exit $$?; \
+	  for f in $$files; do if [ "$$f" = "scmconfig.h" ]; then \
+ 	    grep -q "SCM_SIZEOF_LONG 4" "$(DESTDIR)$(modincludedir)/$$f" && g=32 || g=64; \
+ 	    mv "$(DESTDIR)$(modincludedir)/$$f" "$(DESTDIR)$(modincludedir)/scmconfig-$$g.h"; \
+ 	    $(INSTALL_HEADER) "$${d}scmconfig.h.mlib" "$(DESTDIR)$(modincludedir)/$$f"; \
+ 	  fi; done \
 	done
 
 uninstall-nodist_modincludeHEADERS:
diff -up guile-1.8.7/libguile/guile-snarf.in.multilib guile-1.8.7/libguile/guile-snarf.in
--- guile-1.8.7/libguile/guile-snarf.in.multilib	2009-07-04 00:19:00.000000000 +0200
+++ guile-1.8.7/libguile/guile-snarf.in	2009-07-16 18:06:48.000000000 +0200
@@ -39,7 +39,6 @@
 #   #endif
 # 
 # If the environment variable CPP is set, use its value instead of the
-# C pre-processor determined at Guile configure-time: "@CPP@".
 
 # Code:
 
@@ -75,7 +74,7 @@ tempdir="$TMPDIR/guile-snarf.$$"
 (umask 077 && mkdir $tempdir) || exit 1
 temp="$tempdir/tmp"
 
-if [ x"$CPP" = x ] ; then cpp="@CPP@" ; else cpp="$CPP" ; fi
+if [ x"$CPP" = x ] ; then cpp="gcc -E" ; else cpp="$CPP" ; fi
 
 trap "rm -rf $tempdir" 0 1 2 15
 
diff -up /dev/null guile-1.8.7/libguile/scmconfig.h.mlib
--- /dev/null	2009-06-24 16:43:53.264001725 +0200
+++ guile-1.8.7/libguile/scmconfig.h.mlib	2009-07-16 18:06:48.000000000 +0200
@@ -0,0 +1,6 @@
+#include <bits/wordsize.h>
+#if __WORDSIZE == 32
+#include "libguile/scmconfig-32.h"
+#else
+#include "libguile/scmconfig-64.h"
+#endif
