--- docbook-dtds-1.0/4.2-sgml/docbookx.dtd.euro	2003-07-15 09:07:58.000000000 +0100
+++ docbook-dtds-1.0/4.2-sgml/docbookx.dtd	2003-07-15 09:08:12.000000000 +0100
@@ -106,7 +106,7 @@
 <![%dbcent.module;[
 
 <![%sgml.features;[
-<!ENTITY euro "[euro  ]"><!-- euro sign, U+20AC NEW -->
+<!ENTITY euro SDATA "[euro  ]"><!-- euro sign, U+20AC NEW -->
 ]]>
 <![%xml.features;[
 <!ENTITY euro "&#x20AC;"><!-- euro sign, U+20AC NEW -->
--- docbook-dtds-1.0/4.2-xml/docbookx.dtd.euro	2003-07-15 09:08:02.000000000 +0100
+++ docbook-dtds-1.0/4.2-xml/docbookx.dtd	2003-07-15 09:08:21.000000000 +0100
@@ -106,7 +106,7 @@
 <![%dbcent.module;[
 
 <![%sgml.features;[
-<!ENTITY euro "[euro  ]"><!-- euro sign, U+20AC NEW -->
+<!ENTITY euro SDATA "[euro  ]"><!-- euro sign, U+20AC NEW -->
 ]]>
 <![%xml.features;[
 <!ENTITY euro "&#x20AC;"><!-- euro sign, U+20AC NEW -->
