diff -pruN nkf-2.1.4.orig/NKF.mod/NKF.pm nkf-2.1.4/NKF.mod/NKF.pm
--- nkf-2.1.4.orig/NKF.mod/NKF.pm	2015-12-12 22:51:41.000000000 +0900
+++ nkf-2.1.4/NKF.mod/NKF.pm	2018-01-29 19:23:15.836460319 +0900
@@ -53,38 +53,30 @@ __END__
 
 =head1 NAME
 
-=begin
+=begin man
 
 NKF - Perl extension for Network Kanji Filter
 
-=begin COMMAND
-
-nkf - Network Kanji Filter
-
-=end
+=end man
 
 =head1 SYNOPSIS
 
-=begin
+=begin man
 
   use NKF;
   $output = nkf("-s",$input);
 
-=begin COMMAND
-
-nkf B<[-butjnesliohrTVvwWJESZxXFfmMBOcdILg]> B<[>I<file ...>B<]>
-
-=end
+=end man
 
 =head1 DESCRIPTION
 
-=begin
+=begin man
 
 This is a Perl Extension version of nkf (Network Kanji Filter).
 It converts the last argument and return converted result. Conversion
 details are specified by flags before the last argument.
 
-=end
+=end man
 
 B<Nkf> is a yet another kanji code converter among networks, hosts and terminals.
 It converts input kanji code to designated kanji code
@@ -473,28 +465,6 @@ Unescape percent escaped characters.
 
 Decode character reference, such as "&#....;".
 
-=begin COMMAND
-
-=item B<--in-place[=>I<SUFFIX>B<]>  B<--overwrite[=>I<SUFFIX>B<]>
-
-Overwrite B<original> listed files by filtered result.
-
-B<Note> --overwrite preserves timestamps of original files.
-
-=item B<--guess=[12]>
-
-Print guessed encoding and newline. (2 is default, 1 is only encoding)
-
-=item B<--help>
-
-Print nkf's help.
-
-=item B<--version>
-
-Print nkf's version.
-
-=end
-
 =item B<-->
 
 Ignore rest of -option.
@@ -507,12 +477,12 @@ Copyright (c) 1987, Fujitsu LTD. (Itaru
 
 Copyright (c) 1996-2015, The nkf Project.
 
-=begin
+=begin man
 
 =head1 SEE ALSO
 
 perl(1).   nkf(1)
 
-=end
+=end man
 
 =cut
