diff -Naur gpm-1.20.7.orig/doc/doc.gpm.in gpm-1.20.7/doc/doc.gpm.in
--- gpm-1.20.7.orig/doc/doc.gpm.in	2012-10-26 23:21:38.000000000 +0200
+++ gpm-1.20.7/doc/doc.gpm.in	2013-07-19 19:40:33.374213536 +0200
@@ -600,7 +600,7 @@
 that one of @t{\-o dtr}, @t{\-o rts}, @t{\-o both} can be specified to
 toggle the control lines of the serial port.
 
-The following mouse type are corrently recognized:
+The following mouse type are currently recognized:
 
 @table @code
 @item bare Microsoft
@@ -621,7 +621,7 @@
         this is your case, use the @samp{bare} mouse type.  Some new
         two-button devices are ``plug and play'', and they don't play
         fair at all; in this case try @t{\-t pnp}.  Many (most)
-        three-button devices that use the microsoft protocol fail to
+        three-button devices that use the Microsoft protocol fail to
         report some middle-button events during mouse motion.  Since
         the protocol does not distinguish between the middle button
         going up and the middle button going down it would be liable
@@ -649,7 +649,7 @@
 	decoder gets into a confused state where it thinks the middle
 	button is up when it's down and vice versa. (If you get sick
 	of having to do this, please don't blame gpm; blame your buggy
-	mouse! Note that most three-button mice that do the microsoft
+	mouse! Note that most three-button mice that do the Microsoft
 	protocol can be made to do the MouseSystems protocol
 	instead. The ``3 Button Serial Mouse mini-HOWTO'' has
 	information about this.)  This mouse decoder accepts standard
