From e3963f85bf60ccb59f3bfdbe4136849a66087753 Mon Sep 17 00:00:00 2001
From: Sean Dougherty <sdougherty@microsoft.com>
Date: Sat, 17 Feb 2024 00:45:37 +0000
Subject: [PATCH] add Mariner and Azure Linux OS Versions

---
 pkg/util/helpers_linux.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pkg/util/helpers_linux.go b/pkg/util/helpers_linux.go
index b279fbe..f9f8786 100644
--- a/pkg/util/helpers_linux.go
+++ b/pkg/util/helpers_linux.go
@@ -64,6 +64,10 @@ func getOSVersion(osReleasePath string) (string, error) {
 		return getDebianVersion(osReleaseMap), nil
 	case "sles":
 		return getDebianVersion(osReleaseMap), nil
+	case "azurelinux":
+		return getDebianVersion(osReleaseMap), nil
+	case "mariner":
+		return getDebianVersion(osReleaseMap), nil
 	default:
 		return "", fmt.Errorf("Unsupported ID in /etc/os-release: %q", osReleaseMap["ID"])
 	}
-- 
2.42.0

