diff -Naur a/Makefile b/Makefile
--- a/Makefile	2021-05-14 17:05:59.000000000 -0700
+++ b/Makefile	2022-02-24 11:53:35.220842944 -0800
@@ -22,8 +22,8 @@
 all: build
 
 # PLATFORMS is the set of OS_ARCH that NPD can build against.
-LINUX_PLATFORMS=linux_amd64 linux_arm64
-PLATFORMS=$(LINUX_PLATFORMS) windows_amd64
+LINUX_PLATFORMS=linux_amd64
+PLATFORMS=$(LINUX_PLATFORMS)
 
 # VERSION is the version of the binary.
 VERSION?=$(shell if [ -d .git ]; then echo `git describe --tags --dirty`; else echo "UNKNOWN"; fi)
@@ -147,7 +147,7 @@
 
 output/linux_amd64/bin/%: $(PKG_SOURCES)
 	GOOS=linux GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
-	  CC=x86_64-linux-gnu-gcc go build \
+	  go build \
 		-mod vendor \
 		-o $@ \
 		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
@@ -157,29 +157,12 @@
 
 output/linux_amd64/test/bin/%: $(PKG_SOURCES)
 	GOOS=linux GOARCH=amd64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
-	  CC=x86_64-linux-gnu-gcc go build \
+	  go build \
 		-mod vendor \
 		-o $@ \
 		-tags "$(LINUX_BUILD_TAGS)" \
 		./test/e2e/$(subst -,,$*)
 
-output/linux_arm64/bin/%: $(PKG_SOURCES)
-	GOOS=linux GOARCH=arm64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
-	  CC=aarch64-linux-gnu-gcc go build \
-		-mod vendor \
-		-o $@ \
-		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
-		-tags "$(LINUX_BUILD_TAGS)" \
-		./cmd/$(subst -,,$*)
-	touch $@
-
-output/linux_arm64/test/bin/%: $(PKG_SOURCES)
-	GOOS=linux GOARCH=arm64 CGO_ENABLED=$(CGO_ENABLED) GO111MODULE=on \
-	  CC=aarch64-linux-gnu-gcc go build \
-		-mod vendor \
-		-o $@ \
-		-tags "$(LINUX_BUILD_TAGS)" \
-		./test/e2e/$(subst -,,$*)
 
 # In the future these targets should be deprecated.
 ./bin/log-counter: $(PKG_SOURCES)


