diff -Naur a/pkg/util/helpers_linux.go b/pkg/util/helpers_linux.go
--- a/pkg/util/helpers_linux.go	2024-02-28 17:22:37.000000000 +0000
+++ b/pkg/util/helpers_linux.go	2024-03-19 00:22:29.127451201 +0000
@@ -58,6 +58,8 @@
 		return getDebianVersion(osReleaseMap), nil
 	case "rhel":
 		return getDebianVersion(osReleaseMap), nil
+	case "mariner":
+		return getDebianVersion(osReleaseMap), nil
 	case "ol":
 		return getDebianVersion(osReleaseMap), nil
 	case "amzn":
