--- cvs-1.11.23/lib/getline.c	2005-04-04 22:46:05.000000000 +0200
+++ cvs-1.11.23/lib/getline.c.old	2008-06-03 19:06:25.000000000 +0200
@@ -154,7 +154,7 @@
   return ret;
 }
 
-int
+ssize_t
 getline (lineptr, n, stream)
      char **lineptr;
      size_t *n;
@@ -163,7 +163,7 @@
   return getstr (lineptr, n, stream, '\n', 0, GETLINE_NO_LIMIT);
 }
 
-int
+ssize_t
 getline_safe (lineptr, n, stream, limit)
      char **lineptr;
      size_t *n;
--- cvs-1.11.23/lib/getline.h	2005-04-04 22:46:05.000000000 +0200
+++ cvs-1.11.23/lib/getline.h.old	2008-06-03 19:06:27.000000000 +0200
@@ -11,9 +11,9 @@
 
 #define GETLINE_NO_LIMIT -1
 
-int
+ssize_t
   getline __PROTO ((char **_lineptr, size_t *_n, FILE *_stream));
-int
+ssize_t
   getline_safe __PROTO ((char **_lineptr, size_t *_n, FILE *_stream,
                          int limit));
 int
