diff -ru OpenBLAS-0.3.9-orig/Makefile OpenBLAS-0.3.9/Makefile
--- OpenBLAS-0.3.9-orig/Makefile	2020-03-01 15:10:20.000000000 -0800
+++ OpenBLAS-0.3.9/Makefile	2022-04-06 08:28:58.458123065 -0700
@@ -1,4 +1,10 @@
 TOPDIR	= .
+
+# parallelize the build
+MAKE_NB_JOBS = 8
+# don't build Fortran (hence don't perform tests)
+NO_FORTRAN = 1
+
 include ./Makefile.system
 
 BLASDIRS = interface driver/level2 driver/level3 driver/others
