--- ordermap-1.0.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ ordermap-1.0.0/Cargo.toml	2025-12-12T06:42:54.975466+00:00
@@ -44,9 +44,7 @@
     "arbitrary",
     "quickcheck",
     "serde",
-    "borsh",
     "rayon",
-    "sval",
 ]
 rustdoc-args = [
     "--cfg",
@@ -57,10 +55,6 @@
 arbitrary = [
     "dep:arbitrary",
     "indexmap/arbitrary",
-]
-borsh = [
-    "dep:borsh",
-    "borsh/indexmap",
 ]
 default = ["std"]
 quickcheck = [
@@ -77,10 +71,6 @@
     "indexmap/serde",
 ]
 std = ["indexmap/std"]
-sval = [
-    "dep:sval",
-    "indexmap/sval",
-]
 
 [lib]
 name = "ordermap"
@@ -103,21 +93,8 @@
 name = "tests"
 path = "tests/tests.rs"
 
-[[bench]]
-name = "bench"
-path = "benches/bench.rs"
-
-[[bench]]
-name = "faststring"
-path = "benches/faststring.rs"
-
 [dependencies.arbitrary]
 version = "1.0"
-optional = true
-default-features = false
-
-[dependencies.borsh]
-version = "1.5.6"
 optional = true
 default-features = false
 
@@ -136,11 +113,6 @@
 
 [dependencies.serde_core]
 version = "1.0.220"
-optional = true
-default-features = false
-
-[dependencies.sval]
-version = "2"
 optional = true
 default-features = false
 
