diff -ruN a/Makefile b/Makefile
--- a/Makefile	2020-06-16 14:36:12.000000000 -0700
+++ b/Makefile	2020-12-16 09:10:22.365494679 -0800
@@ -21,7 +21,7 @@
 MICDIRINTEL     = $(MICDIR)/intel-ucode
 
 all:
-	tar xf $(MICROCODE_INTEL) --strip 1 */intel-ucode/* */intel-ucode-with-caveats/* \
+	tar xf $(MICROCODE_INTEL) --wildcards --strip 1 */intel-ucode/* */intel-ucode-with-caveats/* \
 		--one-top-level=intel-ucode --strip-components=2 --backup=simple
 
 clean:
