From f230da76573e32003cf42746b32821f32b0beda6 Mon Sep 17 00:00:00 2001
From: Seth Michael Larson <seth@python.org>
Date: Tue, 3 Feb 2026 19:43:44 +0000
Subject: [PATCH] Use os.path.commonpath() instead of commonprefix()

---
 src/pyproject_hooks/_impl.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/pyproject_hooks/_impl.py b/src/pyproject_hooks/_impl.py
index 14484fa..02eb4d1 100644
--- a/src/pyproject_hooks/_impl.py
+++ b/src/pyproject_hooks/_impl.py
@@ -117,12 +117,9 @@ def norm_and_check(source_tree: str, requested: str) -> str:
 
     abs_source = os.path.abspath(source_tree)
     abs_requested = os.path.normpath(os.path.join(abs_source, requested))
-    # We have to use commonprefix for Python 2.7 compatibility. So we
-    # normalise case to avoid problems because commonprefix is a character
-    # based comparison :-(
     norm_source = os.path.normcase(abs_source)
     norm_requested = os.path.normcase(abs_requested)
-    if os.path.commonprefix([norm_source, norm_requested]) != norm_source:
+    if os.path.commonpath([norm_source, norm_requested]) != norm_source:
         raise ValueError("paths must be inside source tree")
 
     return abs_requested
