diff -up words-3.0/mwords/256772co.mpo.typos words-3.0/mwords/256772co.mpo
--- words-3.0/mwords/256772co.mpo.typos	1996-06-06 05:33:56.000000000 +0200
+++ words-3.0/mwords/256772co.mpo	2019-01-08 13:10:03.346213801 +0100
@@ -90189,7 +90189,6 @@ Ultra-byronism
 Ultra-calvinist
 Ultra-christian
 Ultra-english
-Ultra-englishUltra-french
 Ultra-french
 Ultra-gallican
 Ultra-gangetic
@@ -116323,7 +116322,6 @@ by-day
 by-dependency
 by-design
 by-doing
-by-doingby-drinking
 by-drinking
 by-dweller
 by-effect
@@ -155743,7 +155741,8 @@ half-eaten
 half-ebb
 half-educated
 half-embraced
-half-embracinghalf-embracingly
+half-embracing
+half-embracingly
 half-enamored
 half-enforced
 half-erased
@@ -208469,7 +208468,6 @@ rude-looking
 rude-made
 rude-mannered
 rude-spoken
-rude-spokenrude-spun
 rude-spun
 rude-thoughted
 rude-tongued
@@ -251103,7 +251101,6 @@ what-is-it
 what-like
 what-you-call-it
 what-you-may call-it
-what-you-may--call-it
 what-you-may-call-'em
 whatever comes
 whatever may happen
diff -up words-3.0/mwords/354984si.ngl.typos words-3.0/mwords/354984si.ngl
--- words-3.0/mwords/354984si.ngl.typos	1996-06-06 05:30:48.000000000 +0200
+++ words-3.0/mwords/354984si.ngl	2019-01-08 13:06:25.581693153 +0100
@@ -17787,7 +17787,6 @@ architecture
 architecture's
 architectures
 architecturesque
-architecure
 architis
 architraval
 architrave
@@ -110139,7 +110138,6 @@ flounderingly
 flounders
 flour
 floured
-flourescent
 flouriness
 flouring
 flourish
@@ -330243,7 +330241,6 @@ unlowly
 unloyal
 unloyally
 unloyalty
-unltraconservative
 unlubricant
 unlubricated
 unlubricating
@@ -331241,7 +331238,6 @@ unneath
 unneatly
 unneatness
 unnebulous
-unneccessary
 unnecessaries
 unnecessarily
 unnecessariness
