diff -up sblim-cmpi-nfsv4-1.1.0/provider-register.sh.orig sblim-cmpi-nfsv4-1.1.0/provider-register.sh
--- sblim-cmpi-nfsv4-1.1.0/provider-register.sh.orig	2013-08-08 14:14:54.635453999 +0200
+++ sblim-cmpi-nfsv4-1.1.0/provider-register.sh	2013-08-08 14:15:08.539526596 +0200
@@ -216,7 +216,7 @@ pegasus_install()
     then
 	chatter Registering providers with $state cimserver
 	$CIMMOF -uc -I $mofpath -n $namespace $mymofs &&
-	$CIMMOF -uc -n root/PG_Interop $_REGFILENAME
+	$CIMMOF -uc -n root/interop $_REGFILENAME
     else
 	echo "Failed to build pegasus registration MOF." >&2
 	return 1
