diff --git a/Unix/buildtool b/Unix/buildtool
--- a/Unix/buildtool	2020-09-29 10:33:36.055821162 -0700
+++ b/Unix/buildtool	2020-09-29 11:24:05.121922456 -0700
@@ -274,9 +274,9 @@
             distro=`lsb_release -i | awk -F":" '{ print $2 }'`
             distro_version=`lsb_release -r | awk -F":" '{ print $2 }'`
 	    ;;
-        arm*:Linux:*)
+        aarch64*:Linux:*)
             os=LINUX
-            arch=ARM
+            arch=AARCH64
             compiler=GNU
             distro=`lsb_release -i | awk -F":" '{ print $2 }'`
             distro_version=`lsb_release -r | awk -F":" '{ print $2 }'`
@@ -293,7 +293,7 @@
 
 case "$platform" in
 
-    LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+    LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
         compiler_version=`gcc --version | awk -F" " 'match($0, /[0-9]*\.[0-9]*\.[0-9]*/){ if (match($0,  /[0-9]*\.[0-9]*\.[0-9]*/, m)) print m[0] }'`
         compiler_major_version=`echo $compiler_version | awk -F'.' '{ print $1}'`
         compiler_minor_version=`echo $compiler_version | awk -F'.' '{ print $2}'`
@@ -573,7 +573,7 @@
 
     case "$platform" in
 
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
             echo gcc
             ;;
         MONTAVISTA_IX86_GNU)
@@ -614,7 +614,7 @@
     fi
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
             echo g++
             ;;
         MONTAVISTA_IX86_GNU)
@@ -689,7 +689,7 @@
         LINUX_IX86_GNU|LINUX_X86_64_GNU)
             echo size
             ;;
-        LINUX_ARM_GNU)
+        LINUX_AARCH64_GNU)
             echo size
             ;;
         MONTAVISTA_IX86_GNU)
@@ -778,7 +778,7 @@
     r="$r -g"
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|MONTAVISTA_IX86_GNU|NETBSD_IX86_GNU|LINUX_ARM_GNU)
+        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|MONTAVISTA_IX86_GNU|NETBSD_IX86_GNU|LINUX_AARCH64_GNU)
             if test $cxx_opt ; then
                 r="$r -std=gnu++98"
             fi    
@@ -974,7 +974,7 @@
     r=""
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
             r="$r -shared"
             test -n "$libpath_opt" && r="$r -Wl,-rpath=$libpath_opt"
             ;;
@@ -1098,7 +1098,7 @@
     r=""
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
             test -n "$libpath_opt" && r="$r -Wl,-rpath=$libpath_opt"
             ;;
         MONTAVISTA_IX86_GNU)
@@ -1181,7 +1181,7 @@
     r=""
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
             r="gcc -M"
             ;;
         MONTAVISTA_IX86_GNU)
@@ -1225,7 +1225,7 @@
     r=""
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
             r="-lpthread -ldl -lpam"
             ;;
         MONTAVISTA_IX86_GNU)
@@ -1273,7 +1273,7 @@
     args="$arg2 $arg3 $arg4 $arg5 $arg6 $arg7 $arg8 $arg9"
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
             for path in $args
             do
                 r="$r -Wl,-rpath=$path"
@@ -1360,7 +1360,7 @@
     r=""
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
             ;;
         MONTAVISTA_IX86_GNU)
             ;;
@@ -1441,7 +1441,7 @@
     fi
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
             echo "lib$arg2.so"
             ;;
         MONTAVISTA_IX86_GNU)
@@ -1485,7 +1485,7 @@
     fi
 
     case "$platform" in
-        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+        LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
             echo "so"
             ;;
         MONTAVISTA_IX86_GNU)
@@ -1577,7 +1577,7 @@
 
     if [ -z "$libdir" ]; then
         case "$platform" in
-            LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_ARM_GNU)
+            LINUX_IX86_GNU|LINUX_X86_64_GNU|LINUX_PPC_GNU|LINUX_AARCH64_GNU)
                 if [ -f "/usr/lib/libssl.so" ]; then
                     libdir=/usr/lib
                 fi
diff --git a/Unix/pal/palcommon.h b/Unix/pal/palcommon.h
--- a/Unix/pal/palcommon.h	2020-09-29 10:33:36.091821099 -0700
+++ b/Unix/pal/palcommon.h	2020-09-29 11:23:09.370299157 -0700
@@ -930,7 +930,7 @@
 }
 
 
-#elif defined(CONFIG_ARCH_SPARC) || defined(CONFIG_ARCH_ARM)
+#elif defined(CONFIG_ARCH_SPARC) || defined(CONFIG_ARCH_AARCH64)
 
 /* Sparc only does big endian */
 
