diff -up ./tests/Bootstrap.php.rpm ./tests/Bootstrap.php
--- ./tests/Bootstrap.php.rpm	2026-08-25 11:01:39.000000000 +0200
+++ ./tests/Bootstrap.php	2026-08-25 11:39:03.724218665 +0200
@@ -46,47 +46,3 @@ if (! \function_exists('vdd')) {
     }
 }
 
-(static function (string $rootDir): void {
-    $classLoader = require \implode(DIRECTORY_SEPARATOR, [$rootDir, 'vendor', 'autoload.php']);
-    if (! $classLoader instanceof ClassLoader) {
-        throw new RuntimeException('Unable to load ' . ClassLoader::class);
-    }
-
-    $hamcrestPath = \implode(
-        DIRECTORY_SEPARATOR,
-        [$rootDir, 'vendor', 'hamcrest', 'hamcrest-php', 'hamcrest', 'Hamcrest.php']
-    );
-    if (\file_exists($hamcrestPath)) {
-        require_once $hamcrestPath;
-    }
-
-    $testsDir = \implode(DIRECTORY_SEPARATOR, [$rootDir, 'tests']);
-    if (! \is_dir($testsDir)) {
-        throw new RuntimeException('Unable to find tests directory: ' . $testsDir);
-    }
-
-    $fixtureDir = \implode(DIRECTORY_SEPARATOR, [$testsDir, 'Fixture']);
-    if (! \is_dir($fixtureDir)) {
-        throw new RuntimeException('Unable to find fixture directory: ' . $fixtureDir);
-    }
-
-    // Add autoloading for the fixture classes
-    $classLoader->add('', $fixtureDir . DIRECTORY_SEPARATOR . 'Namespaced');
-    $versions = ['PHP73', 'PHP74', 'PHP80', 'PHP81', 'PHP82', 'PHP83', 'PHP84', 'PHP85', 'PHP86'];
-    foreach ($versions as $version) {
-        $classLoader->addPsr4($version . '\\', $fixtureDir . DIRECTORY_SEPARATOR . $version);
-    }
-    // $classLoader->addPsr4('Tests\\Unit\\', $testsDir . DIRECTORY_SEPARATOR . 'Unit');
-
-    $debug = false;
-    // If debug is enabled, set up a loader for Mockery to use for generated mock classes
-    if ($debug) {
-        $mocksDirectory = \implode(DIRECTORY_SEPARATOR, [$testsDir, '_mocks']);
-
-        if (! \file_exists($mocksDirectory)) {
-            \mkdir($mocksDirectory, 0777, true);
-        }
-
-        Mockery::setLoader(new RequireLoader($mocksDirectory));
-    }
-})(\dirname(__DIR__));
