--- uu_mv-0.7.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ uu_mv-0.7.0/Cargo.toml	2026-03-16T15:50:25.465087+00:00
@@ -87,16 +87,11 @@
 ]
 package = "uucore"
 
-[dev-dependencies.divan]
-version = "4.0.5"
-package = "codspeed-divan-compat"
-
 [dev-dependencies.tempfile]
 version = "3.15.0"
 
 [dev-dependencies.uucore]
 version = "0.7.0"
-features = ["benchmark"]
 package = "uucore"
 
 [target."cfg(unix)".dependencies.libc]
