From ca6804e084b8dea1e243b3fe493ba4c582fba14a Mon Sep 17 00:00:00 2001
From: kavyasree <kkaitepalli@microsoft.com>
Date: Wed, 26 Feb 2025 09:03:26 +0530
Subject: [PATCH] Remove_leave_log_after_failure_test

---
 src/tools/rust-installer/test.sh | 24 ------------------------
 1 file changed, 24 deletions(-)

diff --git a/src/tools/rust-installer/test.sh b/src/tools/rust-installer/test.sh
index 36fc60dd1..698585b78 100755
--- a/src/tools/rust-installer/test.sh
+++ b/src/tools/rust-installer/test.sh
@@ -1051,30 +1051,6 @@ create_log() {
 }
 runtest create_log
 
-leave_log_after_failure() {
-    # chmod doesn't work on windows
-    if [ ! -n "${WINDOWS-}" ]; then
-    try sh "$S/gen-installer.sh" \
-        --image-dir="$TEST_DIR/image1" \
-        --work-dir="$WORK_DIR" \
-        --output-dir="$OUT_DIR"
-    mkdir -p "$PREFIX_DIR/lib/packagelib"
-    touch "$PREFIX_DIR/lib/packagelib/components"
-    chmod u-w "$PREFIX_DIR/lib/packagelib/components"
-    expect_fail "$WORK_DIR/package/install.sh" --prefix="$PREFIX_DIR"
-    chmod u+w "$PREFIX_DIR/lib/packagelib/components"
-    try test -e "$PREFIX_DIR/lib/packagelib/install.log"
-    local _log="$(cat "$PREFIX_DIR/lib/packagelib/install.log")"
-    if [ -z "$_log" ]; then
-        fail "log is empty"
-    fi
-    # script should tell user where the logs are
-    if ! grep -q "see logs at" "$PREFIX_DIR/lib/packagelib/install.log"; then
-        fail "missing log message"
-    fi
-    fi
-}
-runtest leave_log_after_failure
 
 # https://github.com/rust-lang/rust-installer/issues/22
 help() {
-- 
2.34.1

