--- rustc-1.39-0-src/src/librustc_incremental/persist/fs.rs	2020-08-12 10:36:42.776264792 -0700
+++ rustc-1.39-0-src/src/librustc_incremental/persist/fs.rs	2020-08-12 10:42:43.797059804 -0700
@@ -880,7 +880,10 @@
 fn safe_remove_dir_all(p: &Path) -> io::Result<()> {
     if p.exists() {
         let canonicalized = p.canonicalize()?;
-        std_fs::remove_dir_all(canonicalized)
+        match std_fs::remove_dir_all(canonicalized) {
+            Err(ref err) if err.kind() == io::ErrorKind::NotFound => Ok(()),
+            result => result,
+        }
     } else {
         Ok(())
     }
