diff -Nur aiohttp-3.6.1/aiohttp/_cparser.pxd aiohttp-3.6.1.new/aiohttp/_cparser.pxd
--- aiohttp-3.6.1/aiohttp/_cparser.pxd	2019-09-19 17:52:35.000000000 +0200
+++ aiohttp-3.6.1.new/aiohttp/_cparser.pxd	2019-09-21 15:54:52.313964294 +0200
@@ -1,7 +1,7 @@
 from libc.stdint cimport uint16_t, uint32_t, uint64_t
 
 
-cdef extern from "../vendor/http-parser/http_parser.h":
+cdef extern from "http_parser.h":
     ctypedef int (*http_data_cb) (http_parser*,
                                   const char *at,
                                   size_t length) except -1
diff -Nur aiohttp-3.6.1/setup.py aiohttp-3.6.1.new/setup.py
--- aiohttp-3.6.1/setup.py	2019-09-19 17:52:35.000000000 +0200
+++ aiohttp-3.6.1.new/setup.py	2019-09-21 15:55:15.423495152 +0200
@@ -27,7 +27,6 @@
 extensions = [Extension('aiohttp._websocket', ['aiohttp/_websocket.c']),
               Extension('aiohttp._http_parser',
                         ['aiohttp/_http_parser.c',
-                         'vendor/http-parser/http_parser.c',
                          'aiohttp/_find_header.c'],
                         define_macros=[('HTTP_PARSER_STRICT', 0)],
                         ),
