--- curl-0.4.50/Cargo.toml	2006-07-24T01:21:28+00:00
+++ curl-0.4.50/Cargo.toml	2026-06-20T10:34:20.598427+00:00
@@ -146,13 +146,3 @@
 version = "0.9.64"
 optional = true
 
-[target.'cfg(target_env = "msvc")'.dependencies.schannel]
-version = "0.1.13"
-
-[target.'cfg(target_env = "msvc")'.dependencies.windows-sys]
-version = ">=0.59.0, <0.62.0"
-features = [
-    "Win32_Foundation",
-    "Win32_System_LibraryLoader",
-    "Win32_Security_Cryptography",
-]
