diff -ruN b/metaclass.gemspec c/metaclass.gemspec
--- b/metaclass.gemspec	2022-07-20 17:37:27.006892932 -0700
+++ c/metaclass.gemspec	2022-07-20 17:38:06.423381826 -0700
@@ -13,9 +13,9 @@
 
   s.rubyforge_project = "metaclass"
 
-  s.files         = `git ls-files`.split("\n")
-  s.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
-  s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
+  s.files         = Dir['**/*']
+  s.test_files    = s.files.grep(%r{^(test|spec|features)/})
+  s.executables   = s.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   s.require_paths = ["lib"]
 
   s.add_development_dependency 'minitest'
