diff -ruN a/power_assert.gemspec b/power_assert.gemspec
--- a/power_assert.gemspec	2021-08-28 19:32:29.000000000 -0700
+++ b/power_assert.gemspec	2022-07-03 14:30:07.683660284 -0700
@@ -12,9 +12,7 @@
   s.summary     = "Power Assert for Ruby"
   s.description = "Power Assert shows each value of variables and method calls in the expression. It is useful for testing, providing which value wasn't correct when the condition is not satisfied."
 
-  s.files         = `git ls-files -z`.split("\x0").reject do |f|
-    f.match(%r{\A(?:test|spec|features|benchmark|bin)/})
-  end
+  s.files         = Dir['**/*']
   s.bindir        = 'exe'
   s.executables   = s.files.grep(%r{^exe/}) { |f| File.basename(f) }
   s.require_paths = ['lib']
