--- zbus_names-4.3.4/Cargo.toml	2006-07-24T01:21:28+00:00
+++ zbus_names-4.3.4/Cargo.toml	2026-07-23T19:46:21.600593+00:00
@@ -48,11 +48,6 @@
 path = "src/lib.rs"
 bench = false
 
-[[bench]]
-name = "benchmarks"
-path = "benches/benchmarks.rs"
-harness = false
-
 [dependencies.serde]
 version = "1.0.200"
 features = ["derive"]
@@ -64,10 +59,6 @@
 version = "5.13.1"
 features = ["enumflags2"]
 
-[dev-dependencies.criterion]
-version = "5.0.0"
-package = "codspeed-criterion-compat"
-
 [lints.rust.unexpected_cfgs]
 level = "warn"
 priority = 0
