--- mariadb-connector-odbc-3.0.6-ga-src/ma_odbc.h	2018-07-31 13:39:02.000000000 +0200
+++ mariadb-connector-odbc-3.0.6-ga-src/ma_odbc.h_patched	2018-08-01 16:10:09.962798000 +0200
@@ -29,7 +29,7 @@
 
 #include <stdlib.h>
 
-#include <mysql.h>
+#include <mysql/mysql.h>
 
 #include <ma_legacy_helpers.h>
 
@@ -37,7 +37,7 @@
 #include <sqlext.h>
 #include <odbcinst.h>
 
-#include <errmsg.h>
+#include <mysql/errmsg.h>
 #include <string.h>
 #include <stdio.h>
 #include <math.h>

--- mariadb-connector-odbc-3.1.10-ga-src/ma_conv_charset.h.old	2020-10-26 11:37:51.314034870 +0100
+++ mariadb-connector-odbc-3.1.10-ga-src/ma_conv_charset.h	2020-10-26 11:38:05.860205545 +0100
@@ -23,7 +23,7 @@
 #ifndef _MA_CONV_CHARSET_H_
 #define _MA_CONV_CHARSET_H_
 
-#include "mariadb_ctype.h"
+#include <mysql/mariadb_ctype.h>
 
 size_t MADB_ConvertString(const char *from __attribute__((unused)),
                           size_t *from_len __attribute__((unused)),

