--- portable-atomic-1.13.0/Cargo.toml	1970-01-01T00:00:01+00:00
+++ portable-atomic-1.13.0/Cargo.toml	2026-01-12T16:22:03.732163+00:00
@@ -114,16 +114,6 @@
 
 [target."cfg(unix)".dev-dependencies.libc]
 version = "=0.2.163"
-
-[target."cfg(valgrind)".dev-dependencies.crabgrind]
-version = "0.1"
-
-[target."cfg(windows)".dev-dependencies.windows-sys]
-version = "0.61"
-features = [
-    "Win32_Foundation",
-    "Win32_System_Threading",
-]
 
 [lints.clippy]
 all = "warn"
@@ -266,3 +256,4 @@
 
 [profile.release]
 debug = 2
+
