--- tokio-rustls-0.26.6/Cargo.toml	2006-07-24T01:21:28+00:00
+++ tokio-rustls-0.26.6/Cargo.toml	2026-09-28T14:45:07.115092+00:00
@@ -50,16 +50,13 @@
 ]
 
 [features]
-aws-lc-rs = ["aws_lc_rs"]
-aws_lc_rs = ["rustls/aws_lc_rs"]
 brotli = ["rustls/brotli"]
 default = [
     "logging",
     "tls12",
-    "aws_lc_rs",
+    "ring",
 ]
 early-data = []
-fips = ["rustls/fips"]
 logging = ["rustls/logging"]
 ring = ["rustls/ring"]
 tls12 = ["rustls/tls12"]
