From ab066912f5b86194a10ffcdc6b349e01f3e70005 Mon Sep 17 00:00:00 2001
From: Brian Fjeldstad <bfjelds@microsoft.com>
Date: Mon, 18 Mar 2024 22:33:10 +0000
Subject: [PATCH] disable migration-tests

---
 tests/qtest/meson.build | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/tests/qtest/meson.build b/tests/qtest/meson.build
index 47dabf91d0..5177c91736 100644
--- a/tests/qtest/meson.build
+++ b/tests/qtest/meson.build
@@ -2,7 +2,6 @@ slow_qtests = {
   'ahci-test' : 60,
   'bios-tables-test' : 120,
   'boot-serial-test' : 60,
-  'migration-test' : 150,
   'npcm7xx_pwm-test': 150,
   'prom-env-test' : 60,
   'pxe-test' : 60,
@@ -99,7 +98,6 @@ qtests_i386 = \
    'cpu-plug-test',
    'q35-test',
    'vmgenid-test',
-   'migration-test',
    'test-x86-cpuid-compat',
    'numa-test'
   ]
@@ -167,7 +165,7 @@ qtests_ppc64 = \
   (slirp.found() ? ['pxe-test'] : []) +              \
   (config_all_devices.has_key('CONFIG_USB_UHCI') ? ['usb-hcd-uhci-test'] : []) +             \
   (config_all_devices.has_key('CONFIG_USB_XHCI_NEC') ? ['usb-hcd-xhci-test'] : []) +         \
-  qtests_pci + ['migration-test', 'numa-test', 'cpu-plug-test', 'drive_del-test']
+  qtests_pci + ['numa-test', 'cpu-plug-test', 'drive_del-test']
 
 qtests_sh4 = (config_all_devices.has_key('CONFIG_ISA_TESTDEV') ? ['endianness-test'] : [])
 qtests_sh4eb = (config_all_devices.has_key('CONFIG_ISA_TESTDEV') ? ['endianness-test'] : [])
@@ -222,8 +220,7 @@ qtests_aarch64 = \
    config_all_devices.has_key('CONFIG_TPM_TIS_I2C') ? ['tpm-tis-i2c-test'] : []) + \
   ['arm-cpu-features',
    'numa-test',
-   'boot-serial-test',
-   'migration-test']
+   'boot-serial-test']
 
 qtests_s390x = \
   qtests_filter + \
@@ -231,8 +228,7 @@ qtests_s390x = \
    'drive_del-test',
    'device-plug-test',
    'virtio-ccw-test',
-   'cpu-plug-test',
-   'migration-test']
+   'cpu-plug-test']
 
 qtests_riscv32 = \
   (config_all_devices.has_key('CONFIG_SIFIVE_E_AON') ? ['sifive-e-aon-watchdog-test'] : [])
@@ -308,7 +304,6 @@ qtests = {
   'dbus-vmstate-test': files('migration-helpers.c') + dbus_vmstate1,
   'erst-test': files('erst-test.c'),
   'ivshmem-test': [rt, '../../contrib/ivshmem-server/ivshmem-server.c'],
-  'migration-test': migration_files,
   'pxe-test': files('boot-sector.c'),
   'qos-test': [chardev, io, qos_test_ss.apply(config_targetos, strict: false).sources()],
   'tpm-crb-swtpm-test': [io, tpmemu_files],
-- 
2.34.1
