From 51dd9ed1eabeba1886a050313739fcf2966eda77 Mon Sep 17 00:00:00 2001
From: Pawel Winogrodzki <pawelwi@microsoft.com>
Date: Wed, 8 Dec 2021 10:56:58 -0800
Subject: [PATCH] Fixing glibc 'struct statx' usage.

---
 tools/virtiofsd/passthrough_ll.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/tools/virtiofsd/passthrough_ll.c b/tools/virtiofsd/passthrough_ll.c
index 38b2af8..f239e8c 100644
--- a/tools/virtiofsd/passthrough_ll.c
+++ b/tools/virtiofsd/passthrough_ll.c
@@ -978,11 +978,8 @@ static int do_statx(struct lo_data *lo, int dirfd, const char *pathname,
             statbuf->st_ctim.tv_sec = statxbuf.stx_ctime.tv_sec;
             statbuf->st_ctim.tv_nsec = statxbuf.stx_ctime.tv_nsec;
 
-            if (statxbuf.stx_mask & STATX_MNT_ID) {
-                *mnt_id = statxbuf.stx_mnt_id;
-            } else {
-                *mnt_id = 0;
-            }
+            *mnt_id = 0;
+
             return 0;
         } else if (errno != ENOSYS) {
             return -1;
-- 
2.17.1

