diff -urN ispell-gaeilge-5.0.old/makefile ispell-gaeilge-5.0/makefile
--- ispell-gaeilge-5.0.old/makefile	2017-12-04 02:02:36.000000000 +0530
+++ ispell-gaeilge-5.0/makefile	2018-07-08 09:16:16.483019455 +0530
@@ -20,17 +20,15 @@
 ALTAFFIXFILE=gaeilgemor.aff
 INSTALL_DATA=$(INSTALL) -m 444
 
-SORT=/usr/bin/sort -u
+SORT=sort -u
 
-hashtable: $(INSTALLATION).hash
-
-all: gaeilge.hash gaeilgelit.hash gaeilgemor.hash
+all: ga_IE.dic
 
 # grep -v filters out Malmö, São, LC_ALL=C needed!
-gaeilge.hash: $(RAWWORDS) $(AFFIXFILE) $(PERSONAL)
+ga_IE.dic: $(RAWWORDS) $(AFFIXFILE) $(PERSONAL)
 	LC_ALL=C $(SORT) $(RAWWORDS) $(PERSONAL) | LC_ALL=C grep -v "[^'a-zA-ZáéíóúÁÉÍÓÚ/-]" | iconv -f UTF-8 -t iso-8859-1 > gaeilge.focail
-	iconv -f UTF-8 -t iso-8859-1 $(AFFIXFILE) > tempaff.txt
-	$(ISPELLBIN)/buildhash gaeilge.focail tempaff.txt gaeilge.hash
+	wc -l gaeilge.focail | cut -d ' ' -f 1 > ga_IE.dic
+	LC_ALL=C $(SORT) -f -t / --key 1,1 gaeilge.focail >> ga_IE.dic
 	rm -f gaeilge.focail tempaff.txt
 
 gaeilgelit.hash: $(RAWWORDS) $(LITWORDS) gaeilgelit.aff $(PERSONAL)
