--- soplex-release-713/soplex-config.cmake.in.orig	2025-01-28 02:39:35.000000000 -0700
+++ soplex-release-713/soplex-config.cmake.in	2025-02-13 18:47:18.182015353 -0700
@@ -3,7 +3,7 @@ if(NOT TARGET libsoplex)
 endif()
 
 set(SOPLEX_LIBRARIES libsoplex)
-set(SOPLEX_PIC_LIBRARIES libsoplex-pic)
+set(SOPLEX_PIC_LIBRARIES libsoplex)
 set(SOPLEX_INCLUDE_DIRS "@CONF_INCLUDE_DIRS@")
 set(SOPLEX_WITH_MPFR @SOPLEX_WITH_MPFR@)
 set(SOPLEX_FOUND TRUE)
--- soplex-release-713/src/CMakeLists.txt.orig	2025-01-28 02:39:35.000000000 -0700
+++ soplex-release-713/src/CMakeLists.txt	2025-02-13 18:47:18.182015353 -0700
@@ -133,26 +133,7 @@ if(MSVC)
 endif()
 
 # create soplex library with pic
-add_library(libsoplex-pic STATIC ${sources})
-if(MSVC)
-    setLibProperties(libsoplex-pic "libsoplex-pic")
-else()
-   setLibProperties(libsoplex-pic "soplex-pic")
-endif()
-
-set_target_properties(libsoplex-pic PROPERTIES
-    POSITION_INDEPENDENT_CODE on
-    VERSION ${SOPLEX_VERSION_MAJOR}.${SOPLEX_VERSION_MINOR}.${SOPLEX_VERSION_PATCH}.${SOPLEX_VERSION_SUB}
-    SOVERSION ${SOPLEX_VERSION_MAJOR}.${SOPLEX_VERSION_MINOR})
-add_dependencies(libsoplex-pic soplex_update_githash)
-target_include_directories(libsoplex-pic PUBLIC
-    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-    $<INSTALL_INTERFACE:include>)
-target_link_libraries(libsoplex-pic ${libs})
-
-# create soplex library without pic
-add_library(libsoplex STATIC ${sources})
+add_library(libsoplex SHARED soplex_interface.cpp ${sources})
 if(MSVC)
     setLibProperties(libsoplex "libsoplex")
 else()
@@ -160,6 +141,7 @@ else()
 endif()
 
 set_target_properties(libsoplex PROPERTIES
+    POSITION_INDEPENDENT_CODE on
     VERSION ${SOPLEX_VERSION_MAJOR}.${SOPLEX_VERSION_MINOR}.${SOPLEX_VERSION_PATCH}.${SOPLEX_VERSION_SUB}
     SOVERSION ${SOPLEX_VERSION_MAJOR}.${SOPLEX_VERSION_MINOR})
 add_dependencies(libsoplex soplex_update_githash)
@@ -168,31 +150,9 @@ target_include_directories(libsoplex PUB
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
     $<INSTALL_INTERFACE:include>)
 target_link_libraries(libsoplex ${libs})
+set_target_properties(libsoplex PROPERTIES CXX_VISIBILITY_PRESET default)
 
-# create C shared library
-add_library(libsoplexshared SHARED ${sources})
-if(MSVC)
-    setLibProperties(libsoplexshared "libsoplexshared")
-else()
-   setLibProperties(libsoplexshared "soplexshared")
-endif()
-
-set_target_properties(libsoplexshared PROPERTIES
-    VERSION ${SOPLEX_VERSION_MAJOR}.${SOPLEX_VERSION_MINOR}.${SOPLEX_VERSION_PATCH}.${SOPLEX_VERSION_SUB}
-    SOVERSION ${SOPLEX_VERSION_MAJOR}.${SOPLEX_VERSION_MINOR})
-add_dependencies(libsoplexshared soplex_update_githash libsoplex)
-if(MSVC)
-    target_compile_options(libsoplexshared PUBLIC "/Zc:__cplusplus")
-    setLibProperties(libsoplexshared "libsoplexshared")
-endif()
-target_include_directories(libsoplexshared PUBLIC
-    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-    $<INSTALL_INTERFACE:include>)
-target_link_libraries(libsoplexshared libsoplex ${libs})
-set_target_properties(libsoplexshared PROPERTIES CXX_VISIBILITY_PRESET default)
-
-# create soplex binary using library without pic
+# create soplex binary using library with pic
 add_executable(soplex soplexmain.cpp)
 target_link_libraries(soplex LINK_PUBLIC libsoplex ${Boost_LIBRARIES})
 
@@ -209,9 +169,6 @@ endif()
 add_executable(example EXCLUDE_FROM_ALL example.cpp)
 target_link_libraries(example libsoplex)
 
-# set the install rpath to the installed destination
-set_target_properties(soplex PROPERTIES INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
-
 # install the header files of soplex
 install(FILES ${headers} ${PROJECT_BINARY_DIR}/soplex/config.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/soplex)
 install(FILES soplex.h soplex.hpp soplex_interface.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
@@ -237,14 +194,14 @@ install(FILES
   DESTINATION include/soplex/external/zstr)
 
 # install the binary and the library to appropriate lcoations and add them to an export group
-install(TARGETS soplex libsoplex libsoplex-pic libsoplexshared EXPORT soplex-targets
+install(TARGETS soplex libsoplex EXPORT soplex-targets
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 	INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 # Add library targets to the build-tree export set
-export(TARGETS libsoplex libsoplex-pic libsoplexshared
+export(TARGETS libsoplex
   FILE "${CMAKE_BINARY_DIR}/soplex-targets.cmake")
 
 #configure the config file for the build tree
--- soplex-release-713/tests/c_interface/CMakeLists.txt.orig	2025-01-28 02:39:35.000000000 -0700
+++ soplex-release-713/tests/c_interface/CMakeLists.txt	2025-02-13 18:47:18.182015353 -0700
@@ -2,4 +2,4 @@ project(soplex_testing LANGUAGES C)
 
 add_executable(soplex_c_testing "main.c")
 include_directories( "${CMAKE_CURRENT_SOURCE_DIR}/../../src/")
-target_link_libraries(soplex_c_testing libsoplexshared)
+target_link_libraries(soplex_c_testing libsoplex)
