--- rle-decode-fast-1.0.3/Cargo.toml	1970-01-01T00:00:01+00:00
+++ rle-decode-fast-1.0.3/Cargo.toml	2023-08-05T22:05:22.977258+00:00
@@ -19,15 +19,5 @@
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/WanzenBug/rle-decode-helper"
 
-[[bench]]
-name = "bench"
-harness = false
-required-features = ["bench"]
-[dependencies.criterion]
-version = "0.2"
-optional = true
-
-[features]
-bench = ["criterion"]
 [badges.maintenance]
 status = "deprecated"
