diff --git a/PAMmodule.c.orig b/PAMmodule.c
index f6730f1..7276415 100644
--- a/PAMmodule.c.orig
+++ b/PAMmodule.c
@@ -149,8 +149,8 @@ static PyObject * PyPAM_pam(PyObject *self, PyObject *args)
     Py_INCREF(Py_None);
     p->user_data = Py_None;
     
-    p->dlh1 = dlopen("libpam.so", RTLD_LAZY | RTLD_GLOBAL);
-    p->dlh2 = dlopen("libpam_misc.so", RTLD_LAZY | RTLD_GLOBAL);
+    p->dlh1 = dlopen("libpam.so.0", RTLD_LAZY | RTLD_GLOBAL);
+    p->dlh2 = dlopen("libpam_misc.so.0", RTLD_LAZY | RTLD_GLOBAL);
 
     return (PyObject *) p;
 }
