diff -up PyPAM-0.5.0/PAMmodule.c.retval PyPAM-0.5.0/PAMmodule.c
--- PyPAM-0.5.0/PAMmodule.c.retval	2012-05-04 21:47:51.000000000 +0200
+++ PyPAM-0.5.0/PAMmodule.c	2012-05-07 09:42:27.690963206 +0200
@@ -248,7 +248,7 @@ static PyObject * PyPAM_setcred(PyObject
     result = pam_setcred(_self->pamh, flags);
     
     if (result != PAM_SUCCESS) {
-        PyErr_SetString(PyPAM_Error, "Not authenticated");
+        PyPAM_Err(_self, result);
         return NULL;
     }
 
@@ -270,7 +270,7 @@ static PyObject * PyPAM_acct_mgmt(PyObje
     result = pam_acct_mgmt(_self->pamh, flags);
     
     if (result != PAM_SUCCESS) {
-        PyErr_SetString(PyPAM_Error, "Not authenticated");
+        PyPAM_Err(_self, result);
         return NULL;
     }
 
@@ -292,7 +292,7 @@ static PyObject * PyPAM_chauthtok(PyObje
     result = pam_chauthtok(_self->pamh, flags);
     
     if (result != PAM_SUCCESS) {
-        PyErr_SetString(PyPAM_Error, "Not authenticated");
+        PyPAM_Err(_self, result);
         return NULL;
     }
 
@@ -314,7 +314,7 @@ static PyObject * PyPAM_open_session(PyO
     result = pam_open_session(_self->pamh, flags);
     
     if (result != PAM_SUCCESS) {
-        PyErr_SetString(PyPAM_Error, "Not authenticated");
+        PyPAM_Err(_self, result);
         return NULL;
     }
 
@@ -336,7 +336,7 @@ static PyObject * PyPAM_close_session(Py
     result = pam_close_session(_self->pamh, flags);
     
     if (result != PAM_SUCCESS) {
-        PyErr_SetString(PyPAM_Error, "Not authenticated");
+        PyPAM_Err(_self, result);
         return NULL;
     }
 
@@ -433,7 +433,7 @@ static PyObject * PyPAM_putenv(PyObject
     result = pam_putenv(_self->pamh, val);
     
     if (result != PAM_SUCCESS) {
-        PyErr_SetString(PyPAM_Error, "Not authenticated");
+        PyPAM_Err(_self, result);
         return NULL;
     }
 
