diff -up sblim-cmpi-network-1.4.0/Makefile.in.orig sblim-cmpi-network-1.4.0/Makefile.in
--- sblim-cmpi-network-1.4.0/Makefile.in.orig	2013-08-08 10:01:45.191367016 +0200
+++ sblim-cmpi-network-1.4.0/Makefile.in	2013-08-08 10:01:53.354408931 +0200
@@ -287,7 +287,7 @@ datadir = @datadir@
 datarootdir = @datarootdir@
 
 # docdir defines where the documentation goes
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
