Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/hx509/quote.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/hx509/quote.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/hx509/quote.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 # Copyright (c) 2010 Kungliga Tekniska Högskolan
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/Makefile.am
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/Makefile.am
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/Makefile.am
@@ -89,7 +89,7 @@ idn_lookup_SOURCES = idn-lookup.c
 
 LDADD = libwind.la $(LIB_roken)
 
-PYTHON = python
+PYTHON = python2
 
 if !MAINTAINER_MODE
 skip_python = test -f $@ ||
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/UnicodeData.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/UnicodeData.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/UnicodeData.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-bidi.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/gen-bidi.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-bidi.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-combining.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/gen-combining.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-combining.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-errorlist.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/gen-errorlist.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-errorlist.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-map.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/gen-map.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-map.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-normalize.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/gen-normalize.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-normalize.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-punycode-examples.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/gen-punycode-examples.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/gen-punycode-examples.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/generate.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/generate.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/generate.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/rfc3454.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/rfc3454.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/rfc3454.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/rfc4518.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/rfc4518.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/rfc4518.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/stringprep.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/stringprep.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/stringprep.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/util.py
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/lib/wind/util.py
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/lib/wind/util.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 
 # $Id$
Index: heimdal-3e58559362dd3f485e6d6659d587a9169a131391/windows/NTMakefile.w32
===================================================================
--- heimdal-3e58559362dd3f485e6d6659d587a9169a131391.orig/windows/NTMakefile.w32
+++ heimdal-3e58559362dd3f485e6d6659d587a9169a131391/windows/NTMakefile.w32
@@ -120,7 +120,7 @@ RC=rc
 AWK_CMD=gawk.exe
 YACC_CMD=bison.exe
 LEX_CMD=flex.exe
-PYTHON=python.exe
+PYTHON=python2.exe
 PERL=perl.exe
 CMP=cmp.exe
 MAKECAT=makecat.exe
