--- a/Makefile.in
+++ b/Makefile.in
@@ -74,8 +74,8 @@ INSTALL_PREFIX	 	  = @prefix@
 INSTALL_API_HEADERS	  = $(INSTALL_PREFIX)/include/vamp
 INSTALL_SDK_HEADERS	  = $(INSTALL_PREFIX)/include/vamp-sdk
 INSTALL_HOSTSDK_HEADERS	  = $(INSTALL_PREFIX)/include/vamp-hostsdk
-INSTALL_SDK_LIBS	  = $(INSTALL_PREFIX)/lib
-INSTALL_PLUGINS		  = $(INSTALL_PREFIX)/lib/vamp
+INSTALL_SDK_LIBS	  = @libdir@
+INSTALL_PLUGINS		  = $(INSTALL_SDK_LIBS)/vamp
 INSTALL_BINARIES	  = $(INSTALL_PREFIX)/bin 
 
 INSTALL_SDK_LIBNAME	  = libvamp-sdk.so.2.10.0
@@ -90,7 +90,7 @@ INSTALL_HOSTSDK_LINK_DEV  = libvamp-host
 INSTALL_HOSTSDK_STATIC    = libvamp-hostsdk.a
 INSTALL_HOSTSDK_LA        = libvamp-hostsdk.la
 
-INSTALL_PKGCONFIG	  = $(INSTALL_PREFIX)/lib/pkgconfig
+INSTALL_PKGCONFIG	  = $(INSTALL_SDK_LIBS)/pkgconfig
 
 # Flags required to tell the compiler to create a dynamically loadable object
 #
