diff -up enscript-1.6.6/docs/enscript.man.manfixes enscript-1.6.6/docs/enscript.man
--- enscript-1.6.6/docs/enscript.man.manfixes	2011-10-30 17:48:42.000000000 +0100
+++ enscript-1.6.6/docs/enscript.man	2013-04-04 13:26:50.257945126 +0200
@@ -394,6 +394,9 @@ generate RTF (Rich Text Format)
 generate ANSI terminal control codes
 .RE
 .TP 8
+.B \-W, \-\-option=\f2app,option\f3
+Pass option \f2option\f3 to helper application \f2app\f3
+.TP 8
 .B \-X \f2name\f3, \-\-encoding=\f2name\f3
 Use the input encoding \f2name\f1.  Currently \f3enscript\f1 supports
 the following encodings:
@@ -471,6 +474,10 @@ cookies from the beginning of the file.
 .B \-\-color\f1[\f3=\f2bool\f1]\f3
 Use colors in the highlighting outputs.
 .TP 8
+.B \-\-continuous\-page\-numbers
+Count page numbers across input files. Don't restart numbering at
+beginning of each file.
+.TP 8
 .B \-\-download\-font=\f2fontname\f3
 Include the font description file (\f2.pfa\f1 or \f2.pfb\f1 file) of
 the font \f2fontname\f1 to the generated output.
diff -up enscript-1.6.6/src/main.c.manfixes enscript-1.6.6/src/main.c
--- enscript-1.6.6/src/main.c.manfixes	2013-04-04 13:21:31.926629557 +0200
+++ enscript-1.6.6/src/main.c	2013-04-04 13:21:31.953629499 +0200
@@ -2580,7 +2580,7 @@ Mandatory arguments to long options are
   -v, --verbose              tell what we are doing\n\
   -V, --version              print version number\n\
   -w, --language=LANG        set output language to LANG\n\
-  -W, --options=APP,OPTION   pass option OPTION to helper application APP\n\
+  -W, --option=APP,OPTION   pass option OPTION to helper application APP\n\
   -X, --encoding=NAME        use input encoding NAME\n\
   -z, --no-formfeed          do not interpret form feed characters\n\
   -Z, --pass-through         pass through PostScript and PCL files\n\
