--- jpegxl-sys-0.12.0+libjxl-0.11.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ jpegxl-sys-0.12.0+libjxl-0.11.1/Cargo.toml	2026-03-13T16:45:52.129716+00:00
@@ -13,7 +13,7 @@
 edition = "2021"
 rust-version = "1.85.0"
 name = "jpegxl-sys"
-version = "0.12.0+libjxl-0.11.1"
+version = "0.12.0"
 build = "build.rs"
 links = "jxl"
 autolib = false
@@ -73,3 +73,4 @@
 
 [lints.rustdoc]
 broken_intra_doc_links = "deny"
+
