diff -up suite3270-3.5/c3270/c3270.man.ibmhosts suite3270-3.5/c3270/c3270.man
--- suite3270-3.5/c3270/c3270.man.ibmhosts	2016-05-16 10:54:55.000000000 +0200
+++ suite3270-3.5/c3270/c3270.man	2016-05-16 10:55:03.000000000 +0200
@@ -3403,7 +3403,7 @@ backslash character.
 If more than one \fB\-xrm\fP option is given for the same resource,
 the last one on the command line is used.
 .SH "FILES"
-/usr/local/lib/x3270/ibm_hosts
+/etc/ibm_hosts
 .br
 $HOME/.c3270pro
 
diff -up suite3270-3.5/c3270/ibm_hosts.man.ibmhosts suite3270-3.5/c3270/ibm_hosts.man
--- suite3270-3.5/c3270/ibm_hosts.man.ibmhosts	2016-05-16 10:53:16.000000000 +0200
+++ suite3270-3.5/c3270/ibm_hosts.man	2016-05-16 10:53:38.000000000 +0200
@@ -3,7 +3,7 @@
 .SH "NAME"
 ibm_hosts \- host database for x3270 and c3270
 .SH "SYNOPSIS"
-/usr/lib/X11/x3270/ibm_hosts
+/etc/ibm_hosts
 .SH "DESCRIPTION"
 The \fBibm_hosts\fP
 file contains information regarding IBM hosts on the network.
@@ -82,7 +82,7 @@ The name \fBvm\fP will cause the 3270 em
 enter the string `3' on the second data-entry field on the screen, and
 send the Enter \s-1AID\s+1 sequence.
 .SH "FILES"
-/usr/lib/X11/x3270/ibm_hosts
+/etc/ibm_hosts
 .SH "SEE ALSO"
 x3270(1),
 c3270(1)
diff -up suite3270-3.5/x3270/ibm_hosts.man.ibmhosts suite3270-3.5/x3270/ibm_hosts.man
--- suite3270-3.5/x3270/ibm_hosts.man.ibmhosts	2016-05-16 10:54:26.000000000 +0200
+++ suite3270-3.5/x3270/ibm_hosts.man	2016-05-16 10:54:40.000000000 +0200
@@ -3,7 +3,7 @@
 .SH "NAME"
 ibm_hosts \- host database for x3270 and c3270
 .SH "SYNOPSIS"
-/usr/lib/X11/x3270/ibm_hosts
+/etc/ibm_hosts
 .SH "DESCRIPTION"
 The \fBibm_hosts\fP
 file contains information regarding IBM hosts on the network.
@@ -82,7 +82,7 @@ The name \fBvm\fP will cause the 3270 em
 enter the string `3' on the second data-entry field on the screen, and
 send the Enter \s-1AID\s+1 sequence.
 .SH "FILES"
-/usr/lib/X11/x3270/ibm_hosts
+/etc/ibm_hosts
 .SH "SEE ALSO"
 x3270(1),
 c3270(1)
diff -up suite3270-3.5/x3270/x3270.man.ibmhosts suite3270-3.5/x3270/x3270.man
--- suite3270-3.5/x3270/x3270.man.ibmhosts	2016-05-16 10:53:50.000000000 +0200
+++ suite3270-3.5/x3270/x3270.man	2016-05-16 10:54:00.000000000 +0200
@@ -5275,7 +5275,7 @@ T}
 The special types \fBsocks4a\fP and \fBsocks5d\fP can also be used to force
 the proxy server to do the hostname resolution for the SOCKS protocol.
 .SH "FILES"
-/usr/lib/X11/x3270/ibm_hosts
+/etc/ibm_hosts
 .br
 $HOME/.x3270pro
 
