diff -up suite3270-3.5/x3270/Examples/cms_logon.bash.paths suite3270-3.5/x3270/Examples/cms_logon.bash
--- suite3270-3.5/x3270/Examples/cms_logon.bash.paths	2015-09-20 01:04:56.000000000 +0200
+++ suite3270-3.5/x3270/Examples/cms_logon.bash	2016-05-12 12:00:57.000000000 +0200
@@ -1,4 +1,4 @@
-#! /usr/local/bin/bash
+#! /bin/bash
 
 # Copyright (c) 1995-2009, Paul Mattes.
 # All rights reserved.
diff -up suite3270-3.5/x3270/Examples/peer_script.expect.paths suite3270-3.5/x3270/Examples/peer_script.expect
--- suite3270-3.5/x3270/Examples/peer_script.expect.paths	2015-09-20 01:04:56.000000000 +0200
+++ suite3270-3.5/x3270/Examples/peer_script.expect	2016-05-12 12:00:57.000000000 +0200
@@ -1,4 +1,4 @@
-#! /usr/local/bin/expect
+#! /usr/bin/expect
 # TSO login script, which runs as a peer of x3270.
 # expect version
 
