--- pem-3.0.6/Cargo.toml	1970-01-01T00:00:01+00:00
+++ pem-3.0.6/Cargo.toml	2025-10-22T12:37:49.818675+00:00
@@ -49,11 +49,6 @@
 name = "pem"
 path = "src/lib.rs"
 
-[[bench]]
-name = "pem_benchmark"
-path = "benches/pem_benchmark.rs"
-harness = false
-
 [dependencies.base64]
 version = "0.22.0"
 features = ["alloc"]
@@ -64,9 +59,6 @@
 optional = true
 default-features = false
 
-[dev-dependencies.criterion]
-version = "0.3.0"
-
 [dev-dependencies.proptest]
 version = "1"
 features = ["std"]
