--- icu_locale_core-2.3.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ icu_locale_core-2.3.0/Cargo.toml	2026-08-18T06:42:23.819634+00:00
@@ -143,9 +143,6 @@
 optional = true
 default-features = false
 
-[dev-dependencies.iai]
-version = "0.1.1"
-
 [dev-dependencies.litemap]
 version = "0.8.0"
 features = ["testing"]
@@ -167,9 +164,6 @@
 
 [dev-dependencies.serde_json]
 version = "1.0.45"
-
-[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.criterion]
-version = "0.5.0"
 
 [lints.clippy]
 alloc_instead_of_core = "warn"
