--- t/also.t
+++ t/also.t
@@ -1,7 +1,7 @@
 #!/usr/bin/env perl
 use strict;
 use warnings;
-use Test::More;
+use Test::More tests => 2;
 use Test::Fatal;
 use lib 't/lib/also';
 
@@ -29,4 +29,3 @@ use lib 't/lib/also';
     );
 }
 
-done_testing;
--- t/basic.t
+++ t/basic.t
@@ -1,7 +1,7 @@
 #!/usr/bin/env perl
 use strict;
 use warnings;
-use Test::More;
+use Test::More tests => 4;
 use lib 't/lib/basic';
 
 {
@@ -44,4 +44,3 @@ use lib 't/lib/basic';
     );
 }
 
-done_testing;
--- t/conflicts.t
+++ t/conflicts.t
@@ -2,7 +2,7 @@
 use strict;
 use warnings;
 
-use Test::More;
+use Test::More tests => 12;
 use Test::Fatal;
 use lib 't/lib/conflicts';
 
@@ -109,4 +109,3 @@ use lib 't/lib/conflicts';
     ;
 }
 
-done_testing;
--- t/dist.t
+++ t/dist.t
@@ -1,7 +1,7 @@
 #!/usr/bin/env perl
 use strict;
 use warnings;
-use Test::More;
+use Test::More tests => 12;
 use Test::Fatal;
 use lib 't/lib/dist';
 
@@ -73,4 +73,3 @@ use lib 't/lib/dist';
     is(Bar::Conflicts::Bad->dist, 'Bar', "correct dist");
 }
 
-done_testing;
--- t/merge.t
+++ t/merge.t
@@ -1,7 +1,7 @@
 #!/usr/bin/env perl
 use strict;
 use warnings;
-use Test::More;
+use Test::More tests => 1;
 use lib 't/lib/merge';
 
 {
@@ -18,4 +18,3 @@ use lib 't/lib/merge';
     );
 }
 
-done_testing;
--- t/runtime.t
+++ t/runtime.t
@@ -1,7 +1,7 @@
 #!/usr/bin/env perl
 use strict;
 use warnings;
-use Test::More;
+use Test::More tests => 7;
 use lib 't/lib/runtime';
 
 use Module::Runtime 'require_module';
@@ -46,4 +46,3 @@ is(scalar(grep { ref($_) eq 'ARRAY' && @
    1,
    "only installed one \@INC hook");
 
-done_testing;
--- t/warn.t
+++ t/warn.t
@@ -1,7 +1,7 @@
 #!/usr/bin/env perl
 use strict;
 use warnings;
-use Test::More;
+use Test::More tests => 1;
 use lib 't/lib/warn';
 
 {
@@ -13,4 +13,3 @@ use lib 't/lib/warn';
     is($warning, '', "we don't see warnings from loaded modules");
 }
 
-done_testing;
--- xt/release/no-tabs.t
+++ xt/release/no-tabs.t
@@ -3,7 +3,7 @@ use warnings;
 
 # this test was generated with Dist::Zilla::Plugin::NoTabsTests 0.06
 
-use Test::More 0.88;
+use Test::More tests => 1;
 use Test::NoTabs;
 
 my @files = (
@@ -11,4 +11,3 @@ my @files = (
 );
 
 notabs_ok($_) foreach @files;
-done_testing;
