--- zerovec-derive-0.11.2/Cargo.toml	1970-01-01T00:00:01+00:00
+++ zerovec-derive-0.11.2/Cargo.toml	2025-10-28T23:46:27.079376+00:00
@@ -46,24 +46,6 @@
 path = "src/lib.rs"
 proc-macro = true
 
-[[example]]
-name = "derives"
-path = "examples/derives.rs"
-test = true
-harness = false
-
-[[example]]
-name = "make"
-path = "examples/make.rs"
-test = true
-harness = false
-
-[[example]]
-name = "make_var"
-path = "examples/make_var.rs"
-test = true
-harness = false
-
 [dependencies.proc-macro2]
 version = "1.0.61"
 
