--- squashfs-tools/mksquashfs.c.orig	2019-05-21 01:50:02.197183719 -0500
+++ squashfs-tools/mksquashfs.c	2019-05-21 01:51:05.818179903 -0500
@@ -50,6 +50,7 @@
 #include <sys/wait.h>
 #include <limits.h>
 #include <ctype.h>
+#include <sys/sysmacros.h>
 
 #ifndef linux
 #define __BYTE_ORDER BYTE_ORDER
--- squashfs-tools/unsquashfs.c.orig	2019-05-21 01:57:45.984155904 -0500
+++ squashfs-tools/unsquashfs.c	2019-05-21 01:58:39.837152674 -0500
@@ -37,6 +37,7 @@
 #include <sys/resource.h>
 #include <limits.h>
 #include <ctype.h>
+#include <sys/sysmacros.h>
 
 struct cache *fragment_cache, *data_cache;
 struct queue *to_reader, *to_inflate, *to_writer, *from_writer;
