diff -Nrup a/squashfs-tools/mksquashfs.c b/squashfs-tools/mksquashfs.c
--- a/squashfs-tools/mksquashfs.c	2019-12-17 11:18:23.336924170 -0700
+++ b/squashfs-tools/mksquashfs.c	2019-12-17 11:16:56.184159640 -0700
@@ -3030,7 +3030,7 @@ inline void alloc_inode_no(struct inode_
 }
 
 
-inline struct dir_ent *create_dir_entry(char *name, char *source_name,
+static inline struct dir_ent *create_dir_entry(char *name, char *source_name,
 	char *nonstandard_pathname, struct dir_info *dir)
 {
 	struct dir_ent *dir_ent = malloc(sizeof(struct dir_ent));
