diff -ur cheetah3-3.2.6.post1.orig/Cheetah/Template.py cheetah3-3.2.6.post1/Cheetah/Template.py
--- cheetah3-3.2.6.post1.orig/Cheetah/Template.py	2024-10-30 22:02:27.084580240 -0700
+++ cheetah3-3.2.6.post1/Cheetah/Template.py	2024-10-30 22:06:23.575516652 -0700
@@ -24,6 +24,7 @@
 try:
     import cgi  # Used by .webInput() if the template is a CGI script.
 except ImportError:  # Python 3.13+
+    from urllib.parse import parse_qs
     cgi = None
 import types
 
@@ -1919,11 +1920,17 @@
         """
         src = src.lower()
         isCgi = not self._CHEETAH__isControlledByWebKit
-        if isCgi and (cgi is not None) and src in ('f', 'v'):
+        if isCgi and src in ('f', 'v'):
             global _formUsedByWebInput
-            if _formUsedByWebInput is None:
-                _formUsedByWebInput = cgi.FieldStorage()
-            source, func = 'field', _formUsedByWebInput.getvalue
+            if cgi is None:
+                if _formUsedByWebInput is None:
+                    _formUsedByWebInput = \
+                        parse_qs(os.environ.get('QUERY_STRING', ''))
+                source, func = 'field', _formUsedByWebInput.get
+            else:
+                if _formUsedByWebInput is None:
+                    _formUsedByWebInput = cgi.FieldStorage()
+                source, func = 'field', _formUsedByWebInput.getvalue
         elif isCgi and src == 'c':
             raise RuntimeError("can't get cookies from a CGI script")
         elif isCgi and src == 's':
diff -ur cheetah3-3.2.6.post1.orig/docs/news.rst cheetah3-3.2.6.post1/docs/news.rst
--- cheetah3-3.2.6.post1.orig/docs/news.rst	2024-10-30 22:02:27.064580330 -0700
+++ cheetah3-3.2.6.post1/docs/news.rst	2024-10-30 22:07:00.025352723 -0700
@@ -13,6 +13,8 @@
      caused by ``_namemapper`` looking up a key in a non-dictionary.
    - Fixed mapping test in ``NameMapper.py``:
      Python 3.13 brough a new mapping type ``FrameLocalsProxy``.
+   - Fixed ``Template.webInput``: Use ``urllib.parse.parse_qs``
+     instead of ``cgi.FieldStorage``; Python 3.13 dropped ``cgi``.
 
 3.2.6.post1 (2021-02-22)
 ------------------------
