diff -up console-setup/Keyboard/ckbcomp.orig console-setup/Keyboard/ckbcomp
--- console-setup/Keyboard/ckbcomp.orig	2023-01-05 09:43:41.059205282 +0100
+++ console-setup/Keyboard/ckbcomp	2023-01-05 09:44:49.789082563 +0100
@@ -204,10 +204,6 @@ my %acmtable; # Unicode -> legacy code (
 
 my $KEYMAP = ''; # This variable contains the generated keymap
 
-my $broken_caps = 0; # In unicode mode Caps_Lock doesn't work for non-ASCII
-                     # letters.  1 = the keymap contains non-ascii letters.
-                     # See http://bugzilla.kernel.org/show_bug.cgi?id=7746#c21
-
 my %keycodes_table; # x keysym -> x key code
 my %aliases;        # x keysym -> x keysym
 
@@ -4400,9 +4396,6 @@ sub print_vector {
 		my $u = ord (uc (pack ("U", $v)));
 		my $c = ($v == $l ? $u : $l);
 		$capsvector[$mask] = $1 ."U+". sprintf ("%04x", $c);
-		if ($v != $c && $v > 0x7f) {
-		    $broken_caps = 1;
-		}
 	    }
 	}
 	if ($no_NoSymbol) {
@@ -4800,10 +4793,6 @@ keycode 127 =
     die "$0: Unsupported keyboard type $arch\n";
 }
 
-if ($broken_caps) {
-    $KEYMAP =~ s/Caps_Lock/CtrlL_Lock/g;
-}
-
 print $KEYMAP;
 
 if (!$compose_charmap && $charmap) {
