diff -up pyxdg-0.26/xdg/Menu.py.fixOnlyShowIn pyxdg-0.26/xdg/Menu.py
--- pyxdg-0.26/xdg/Menu.py.fixOnlyShowIn	2018-11-01 11:34:26.222123436 -0400
+++ pyxdg-0.26/xdg/Menu.py	2018-11-01 11:36:39.459203908 -0400
@@ -298,11 +298,11 @@ class Menu:
                     entry.Show = NO_EXEC
                     self.Visible -= 1
                 elif xdg.Config.windowmanager:
-                    if (entry.DesktopEntry.OnlyShowIn != [] and (
-                            xdg.Config.windowmanager not in entry.DesktopEntry.OnlyShowIn
+                    if (entry.DesktopEntry.getOnlyShowIn() != [] and (
+                            xdg.Config.windowmanager not in entry.DesktopEntry.getOnlyShowIn()
                         )
                     ) or (
-                        xdg.Config.windowmanager in entry.DesktopEntry.NotShowIn
+                        xdg.Config.windowmanager in entry.DesktopEntry.getNotShowIn()
                     ):
                         entry.Show = NOT_SHOW_IN
                         self.Visible -= 1
