diff -Naur deprecation-2.1.0-original/tests/test_deprecation.py deprecation-2.1.0/tests/test_deprecation.py
--- deprecation-2.1.0-original/tests/test_deprecation.py	2020-04-20 10:14:19.000000000 -0400
+++ deprecation-2.1.0/tests/test_deprecation.py	2024-09-12 11:46:30.659544778 -0400
@@ -12,7 +12,12 @@
 
 # As we unfortunately support Python 2.7, it lacks TestCase.subTest which
 # is in 3.4+ or in unittest2
-import unittest2
+try:
+    import unittest2
+except ImportError:
+    import unittest
+    unittest2 = unittest
+
 import warnings
 
 import deprecation
