diff -rupN libspatialite-5.0.1/spatialite.pc.in libspatialite-5.0.1-new/spatialite.pc.in
--- libspatialite-5.0.1/spatialite.pc.in	2021-02-07 16:53:46.000000000 +0100
+++ libspatialite-5.0.1-new/spatialite.pc.in	2021-02-10 09:54:09.647552994 +0100
@@ -8,5 +8,6 @@ includedir=@includedir@
 Name: spatialite
 Description: Spatial SQL database engine based on SQLite
 Version: @VERSION@
-Libs: -L${libdir} -lspatialite @LIBS@ @LIBXML2_LIBS@ -lm
+Libs.private: -L${libdir} @LIBS@ @LIBXML2_LIBS@ -lm
+Libs: -L${libdir} -lspatialite
 Cflags: -I${includedir}
