From 946380463bb9a233381fc122a8374806d77b1778 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Mon, 23 Oct 2017 09:45:48 -0400
Subject: [PATCH 2/2] Make quiet exit on missing PK/KEK not return error
 status.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/dbxtool.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/dbxtool.c b/src/dbxtool.c
index 717c991..b2cb587 100644
--- a/src/dbxtool.c
+++ b/src/dbxtool.c
@@ -541,13 +541,14 @@ check_pk_and_kek(bool force, bool quiet)
 		}
 	}
 	if (!all_found) {
-		if (!quiet) {
+		if (quiet) {
+			if (!force)
+				exit(0);
+		} else {
 			if (!force)
 				errx(1, "Not attempting to apply updates.");
 			warnx("attempting to apply updates anyway.");
 		}
-		if (!force)
-			exit(1);
 	}
 }
 
-- 
2.14.2

