diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-09-03 22:43:27.700119807 +0000
+++ b/CMakeLists.txt	2024-09-03 22:44:23.316342162 +0000
@@ -315,8 +315,7 @@
 find_program(NIHTEST nihtest)
 
 if(BUILD_REGRESS AND NOT NIHTEST)
-  message(WARNING "-- nihtest not found, regression testing disabled")
-  set(BUILD_REGRESS OFF)
+  message(WARNING "-- nihtest not found, but regression testing will still be enabled")
 endif()
 
 if(BUILD_REGRESS)
