From 1ca0e076bb669382770c16b6f3c53770cbee5513 Mon Sep 17 00:00:00 2001
From: Saul Paredes <saulparedes@microsoft.com>
Date: Mon, 26 Feb 2024 17:00:41 -0800
Subject: [PATCH] replace to flit

---
 pyproject.toml | 18 ++++--------------
 1 file changed, 4 insertions(+), 14 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 842a5b5..f4ed041 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,9 +1,6 @@
 [build-system]
-build-backend = "hatchling.build"
-requires = [
-  "hatch-vcs>=0.3",
-  "hatchling>=1.17.1",
-]
+build-backend = "flit_core.buildapi"
+requires = ["flit_core >=3.8.0,<4"]
 
 [project]
 name = "virtualenv"
@@ -14,7 +11,7 @@ keywords = [
   "isolated",
   "virtual",
 ]
-license = "MIT"
+license = {text = "MIT"}
 maintainers = [{ name = "Bernat Gabor", email = "gaborjbernat@gmail.com" }]
 requires-python = ">=3.7"
 classifiers = [
@@ -37,9 +34,7 @@ classifiers = [
   "Topic :: Software Development :: Testing",
   "Topic :: Utilities",
 ]
-dynamic = [
-  "version",
-]
+version = "3.10.0"
 dependencies = [
   "distlib<1,>=0.3.7",
   "filelock<4,>=3.12.2",
@@ -96,11 +91,6 @@ builtin = "virtualenv.discovery.builtin:Builtin"
 app-data = "virtualenv.seed.embed.via_app_data.via_app_data:FromAppData"
 pip = "virtualenv.seed.embed.pip_invoke:PipInvoke"
 
-[tool.hatch]
-build.hooks.vcs.version-file = "src/virtualenv/version.py"
-build.targets.sdist.include = ["/src", "/tests", "/tasks", "/tox.ini"]
-version.source = "vcs"
-
 [tool.black]
 line-length = 120
 
-- 
2.25.1

