--- insta-1.45.1/Cargo.toml	1970-01-01T00:00:01+00:00
+++ insta-1.45.1/Cargo.toml	2026-01-14T12:51:14.680100+00:00
@@ -117,16 +117,19 @@
 optional = true
 
 [dependencies.console]
-version = "0.15.4"
+version = "0.16.0"
 optional = true
 default-features = false
+features = [
+    "std",
+]
 
 [dependencies.csv]
 version = "1.1.6"
 optional = true
 
 [dependencies.globset]
-version = ">= 0.4.6, < 0.4.17"
+version = "0.4.6"
 optional = true
 
 [dependencies.once_cell]
@@ -150,7 +153,7 @@
 default-features = false
 
 [dependencies.ron]
-version = "0.12.0"
+version = ">=0.10.0, <0.13.0"
 optional = true
 
 [dependencies.serde]
@@ -165,7 +168,7 @@
 version = "3"
 
 [dependencies.toml_edit]
-version = "0.23.0"
+version = ">=0.23.0, <0.25.0"
 features = [
     "serde",
     "parse",
