--- insta-1.46.3/Cargo.toml	2006-07-24T01:21:28+00:00
+++ insta-1.46.3/Cargo.toml	2026-03-07T12:35:54.572687+00:00
@@ -117,9 +117,12 @@
 optional = true
 
 [dependencies.console]
-version = "0.15.4"
+version = "0.16.0"
 optional = true
 default-features = false
+features = [
+    "std",
+]
 
 [dependencies.csv]
 version = "1.1.6"
@@ -150,7 +153,7 @@
 default-features = false
 
 [dependencies.ron]
-version = "0.12.0"
+version = ">=0.10.0, <0.13.0"
 optional = true
 
 [dependencies.serde]
@@ -165,7 +168,7 @@
 version = "3"
 
 [dependencies.toml_edit]
-version = "0.23.0"
+version = ">=0.23.0, <0.25.0"
 features = [
     "serde",
     "parse",
