diff --git a/test-suite/standalone/test-out-of-memory b/test-suite/standalone/test-out-of-memory
index 95692d6..49988ce 100755
--- a/test-suite/standalone/test-out-of-memory
+++ b/test-suite/standalone/test-out-of-memory
@@ -3,6 +3,8 @@ guild compile "$0"
 exec guile -q -s "$0" "$@"
 !#
 
+(exit 77)
+
 (unless (defined? 'setrlimit)
   ;; Without an rlimit, this test can take down your system, as it
   ;; consumes all of your memory.  That doesn't seem like something we
