--- commons-collections4-4.1-src/src/test/java/org/apache/commons/collections4/collection/AbstractCollectionTest.java	2015-11-25 22:52:42.000000000 +0100
+++ commons-collections4-4.1-src/src/test/java/org/apache/commons/collections4/collection/AbstractCollectionTest.java	2018-11-19 09:48:39.052536491 +0100
@@ -1085,7 +1085,7 @@
         verify();
 
         try {
-            array = getCollection().toArray(null);
+            array = getCollection().toArray((Object[])null);
             fail("toArray(null) should raise NPE");
         } catch (final NullPointerException e) {
             // expected
