diff -up m17n-lib-1.8.0/m17n-config.in.multi m17n-lib-1.8.0RC/m17n-config.in
--- m17n-lib-1.8.0/m17n-config.in.multi	2012-04-23 02:16:56.000000000 +0200
+++ m17n-lib-1.8.0/m17n-config.in	2013-03-14 09:17:32.061122801 +0100
@@ -64,17 +64,15 @@ else
   LIBNAME="$LIBNAME$LIBSHELL$LIBFLT$LIBGUI"
 fi
 
+platform=`uname -m`
+
 case $1 in
 --version)
   echo "@PACKAGE_VERSION@";;
 
 --libs)
-  if test "@libdir@" != "/usr/lib"; then
-    echo "-L@libdir@ ${LIBNAME}"
-  else
-    echo "${LIBNAME}"
-  fi;;
-
+  echo "${LIBNAME}";;
+  
 --cflags)
   if test "@includedir@" != "/usr/include"; then
     echo "-I@includedir@"
@@ -82,7 +80,14 @@ case $1 in
 
 --libtool)
   for name in $LIBTOOLNAME; do
-    echo -n "@libdir@/lib${name}.la "
+   case $platform in
+    x86_64|ppc64|s390x|sparc64)
+      echo -n "/usr/lib64/lib${name}.la "
+      ;;
+    *)
+      echo -n "/usr/lib/lib${name}.la "
+      ;;
+   esac
   done;
   echo;;
 *)
