diff -up smc-tools-1.2.0/Makefile.s390 smc-tools-1.2.0/Makefile
--- smc-tools-1.2.0/Makefile.s390	2019-02-28 11:59:47.600106195 +0100
+++ smc-tools-1.2.0/Makefile	2019-02-28 11:59:58.758701108 +0100
@@ -129,10 +129,10 @@ endif
 	install $(INSTALL_FLAGS_BIN) smc_dbg $(DESTDIR)$(BINDIR)
 ifeq ($(shell uname -m | cut -c1-4),s390)
 	install $(INSTALL_FLAGS_BIN) smc_rnics $(DESTDIR)$(BINDIR)
+	install $(INSTALL_FLAGS_MAN) smc_rnics.8 $(DESTDIR)$(MANDIR)/man8
 endif
 	install $(INSTALL_FLAGS_MAN) af_smc.7 $(DESTDIR)$(MANDIR)/man7
 	install $(INSTALL_FLAGS_MAN) smc_run.8 $(DESTDIR)$(MANDIR)/man8
-	install $(INSTALL_FLAGS_MAN) smc_rnics.8 $(DESTDIR)$(MANDIR)/man8
 	install $(INSTALL_FLAGS_MAN) smc_pnet.8 $(DESTDIR)$(MANDIR)/man8
 	install $(INSTALL_FLAGS_MAN) smcss.8 $(DESTDIR)$(MANDIR)/man8
 
