diff -urNp docbook-utils-0.6.14-orig/bin/jw.in docbook-utils-0.6.14/bin/jw.in
--- docbook-utils-0.6.14-orig/bin/jw.in	2010-09-28 18:18:39.567895321 +0200
+++ docbook-utils-0.6.14/bin/jw.in	2010-09-28 18:23:25.849756994 +0200
@@ -80,9 +80,9 @@ SGML_BASE_DIR="@prefix@/share/sgml"
 SGML_CATALOGS_DIR="/etc/sgml"
 if [ -f "$SGML_CONF" ]
 then
-  RE='^[:space:]*SGML_BASE_DIR[:space:]*=[:space:]*'
+  RE='^[[:space:]]*SGML_BASE_DIR[[:space:]]*=[[:space:]]*'
   SGML_BASE_DIR=`grep --color=never $RE $SGML_CONF | sed "s/$RE//"`
-  RE='^[:space:]*SGML_CATALOGS_DIR[:space:]*=[:space:]*'
+  RE='^[[:space:]]*SGML_CATALOGS_DIR[[:space:]]*=[[:space:]]*'
   SGML_CATALOGS_DIR=`grep --color=never $RE $SGML_CONF | sed "s/$RE//"`
 fi
 
@@ -312,7 +312,7 @@ case $SGML_STANDARD_CATALOGS in
 	  SGML_CATALOG_FILES=$SGML_CENTRALIZED_CATALOG
 	else
 	  SGML_CATALOG_FILES=`find $SGML_BASE_DIR -name catalog`
-	  SGML_CATALOG_FILES=`echo "$SGML_CATALOG_FILES" | tr [:space:] :`
+	  SGML_CATALOG_FILES=`echo "$SGML_CATALOG_FILES" | tr [[:space:]] :`
 	fi
 	;;
   no)	SGML_CATALOG_FILES=""
