diff -ru yara-74734418a256c5304ccaf1d322c57e305ff75362/docs/conf.py yara-74734418a256c5304ccaf1d322c57e305ff75362.new/docs/conf.py
--- yara-74734418a256c5304ccaf1d322c57e305ff75362/docs/conf.py	2016-08-02 22:08:18.000000000 +0200
+++ yara-74734418a256c5304ccaf1d322c57e305ff75362.new/docs/conf.py	2016-08-08 23:08:47.637063737 +0200
@@ -102,12 +102,13 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 
-try:
-	import sphinx_rtd_theme
-	html_theme = "sphinx_rtd_theme"
-	html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
-except:
-	html_theme = "default"
+#try:
+#	import sphinx_rtd_theme
+#	html_theme = "sphinx_rtd_theme"
+#	html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
+#except:
+#	html_theme = "default"
+html_theme = "default"
 
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
