--- mt-st-1.1/mt.c.mtio	2018-05-29 08:26:16.874145539 +0200
+++ mt-st-1.1/mt.c	2018-05-29 08:32:03.940573221 +0200
@@ -22,8 +22,9 @@
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/utsname.h>
+#include <sys/sysmacros.h>
 
-#include "mtio.h"
+#include <linux/mtio.h>
 
 #ifndef DEFTAPE
 #define DEFTAPE "/dev/tape"     /* default tape device */
--- mt-st-1.1/stinit.c.mtio	2008-04-27 13:24:16.000000000 +0200
+++ mt-st-1.1/stinit.c	2008-08-04 08:24:45.000000000 +0200
@@ -22,7 +22,7 @@
 #include <linux/major.h>
 #include <scsi/sg.h>
 
-#include "mtio.h"
+#include <linux/mtio.h>
 
 #ifndef FALSE
 #define TRUE 1
