diff -up mt-st-1.1/stinit.c.options mt-st-1.1/stinit.c
--- mt-st-1.1/stinit.c.options	2013-07-23 11:55:33.000000000 +0200
+++ mt-st-1.1/stinit.c	2013-07-23 11:56:09.000000000 +0200
@@ -889,7 +889,7 @@ define_tape(int tapeno, FILE *dbf, devde
 usage(int retval)
 {
     fprintf(stderr,
-	    "Usage: stinit [-h] [-v] [-f dbname] [-p] [drivename_or_number ...]\n");
+	    "Usage: stinit [-h] [-v] [--version] [-f dbname] [-p] [-r] [drivename_or_number ...]\n");
     exit(retval);
 }
 
