diff -up ./init/fapolicyd-magic.magic-override ./init/fapolicyd-magic
--- ./init/fapolicyd-magic.magic-override	2020-11-16 20:16:10.764346043 +0100
+++ ./init/fapolicyd-magic	2020-11-16 20:18:30.354528379 +0100
@@ -1,9 +1,17 @@
 0	string/w	#!\ /usr/bin/bash	Bourne-Again shell script text executable
 !:mime	text/x-shellscript
 
+0  search/1/w  #!\ /usr/bin/env\ bash  Bourne-Again shell script text executable
+!:strength + 15
+!:mime text/x-shellscript
+
 0	string/w	#!\ /usr/bin/sh		Shell script text executable
 !:mime	text/x-shellscript
 
+0   search/1/w  #!\ /usr/bin/env\ bash  Bourne-Again shell script text executable
+!:strength + 15
+!:mime text/x-shellscript
+
 0	string/wt	#!\ /bin/rc		Plan 9 shell script text executable
 !:mime	text/x-plan9-shellscript
 
@@ -47,10 +55,18 @@
 !:strength + 15
 !:mime text/x-python
 
+0   search/1/w  #!\ /usr/bin/env\ python3   Python script text executable
+!:strength + 15
+!:mime text/x-python
+
 0	search/1/wt	#!\ /usr/bin/python2	Python script text executable
 !:strength + 15
 !:mime text/x-python
 
+0  search/1/w  #!\ /usr/bin/env\ python2   Python script text executable
+!:strength + 15
+!:mime text/x-python
+
 0	search/1/wt	#!\ /usr/bin/python	Python script text executable
 !:strength + 15
 !:mime text/x-python
