diff -up nss/tests/ssl/ssl.sh.ignore_test nss/tests/ssl/ssl.sh
--- nss/tests/ssl/ssl.sh.ignore_test	2025-04-22 10:30:10.569990667 +0200
+++ nss/tests/ssl/ssl.sh	2025-04-22 10:30:34.773327320 +0200
@@ -1600,7 +1600,7 @@ ssl_run_tests()
             if using_sql ; then
                 ssl_policy_listsuites
                 ssl_policy_selfserv
-                ssl_policy_pkix_ocsp
+#               ssl_policy_pkix_ocsp
                 ssl_policy
             fi
             ;;
