diff -urN c/examples/chronyd.service d/examples/chronyd.service
--- c/examples/chronyd.service  2022-03-08 08:58:48.624885692 +0000
+++ d/examples/chronyd.service  2022-03-08 08:59:29.500744600 +0000
@@ -10,6 +10,8 @@
 PIDFile=/run/chrony/chronyd.pid
 EnvironmentFile=-/etc/sysconfig/chronyd
 ExecStart=/usr/sbin/chronyd $OPTIONS
+ExecStartPost=/usr/libexec/chrony-helper update-daemon
+ExecStopPost=/usr/libexec/chrony-helper remove-daemon-state
 PrivateTmp=yes
 ProtectHome=yes
 ProtectSystem=full
 