diff -Naur a/examples/chronyd.service b/examples/chronyd.service
--- a/examples/chronyd.service	2023-12-05 13:22:10.000000000 +0000
+++ b/examples/chronyd.service	2024-02-26 23:50:09.535761297 +0000
@@ -10,6 +10,8 @@
 PIDFile=/run/chrony/chronyd.pid
 EnvironmentFile=-/etc/sysconfig/chronyd
 ExecStart=/usr/sbin/chronyd $OPTIONS
+ExecStartPost=/usr/libexec/chrony-helper update-daemon
+ExecStopPost=/usr/libexec/chrony-helper remove-daemon-state
 
 CapabilityBoundingSet=~CAP_AUDIT_CONTROL CAP_AUDIT_READ CAP_AUDIT_WRITE
 CapabilityBoundingSet=~CAP_BLOCK_SUSPEND CAP_KILL CAP_LEASE CAP_LINUX_IMMUTABLE
