From 6bf9e4ff0fb7eff11a02bd937045bf5dc291841a Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 6 May 2020 16:48:47 +0200
Subject: [PATCH] data: Fix pkg-config installation path

---
 data/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/meson.build b/data/meson.build
index 25ce031..e22b544 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -38,4 +38,5 @@ pkgconfig.generate(
     name: 'shared-mime-info',
     description: 'Freedesktop common MIME database',
     version: meson.project_version(),
+    install_dir: get_option('datadir') / 'pkgconfig'
 )
-- 
2.26.2

