From 0da5c5710d0bc5ae8f50fe8a50485f1309113378 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Wed, 23 Jul 2025 19:56:48 +0000
Subject: [PATCH] AZL3 -- disable eddsa in speed in fips mode

---
 apps/speed.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/apps/speed.c b/apps/speed.c
index 559df0f..d535d20 100644
--- a/apps/speed.c
+++ b/apps/speed.c
@@ -2556,6 +2556,7 @@ int speed_main(int argc, char **argv)
     // TOBIASB: Disable KMAC for FIPS provider because it isn't supported.
     // TOBIASB: Disable brainpool for FIPS provider because it isn't supported. See curve_list in ec_curve.c.
     // TOBIASB: Disable X25519 and NID_X448 for FIPS provider because it isn't supported.
+    // TOBIASB: Disable EdDSA for FIPS provider because it isn't supported. See fipsprov.c.
     if (FIPS_mode()) {
         for (i = 0; i < RSA_NUM; i++) {
             if (rsa_keys[i].bits < 2048)
@@ -2563,6 +2564,7 @@ int speed_main(int argc, char **argv)
         }
 
         memset(dsa_doit, 0, sizeof(dsa_doit));
+        memset(eddsa_doit, 0, sizeof(eddsa_doit));
 
         doit[D_KMAC128] = doit[D_KMAC256] = 0;
 
-- 
2.45.4

