diff -up tbb-2019_U8/src/test/test_task_scheduler_init.cpp.orig tbb-2019_U8/src/test/test_task_scheduler_init.cpp
--- tbb-2019_U8/src/test/test_task_scheduler_init.cpp.orig	2019-06-06 04:29:15.000000000 -0600
+++ tbb-2019_U8/src/test/test_task_scheduler_init.cpp	2019-08-26 12:08:19.547880295 -0600
@@ -285,7 +285,7 @@ namespace TestBlockingTerminateNS {
             }
         };
         void operator()() {
-            const int numThreads = 4;
+            const int numThreads = 2;
             tbb::task_scheduler_init init( numThreads );
             Harness::SpinBarrier barrier( numThreads );
             tbb::parallel_for( 0, numThreads, Body( barrier ) );
