From d898ad3920cb08d6918d11d9678bae28393a82d1 Mon Sep 17 00:00:00 2001
From: tinywrkb <tinywrkb@gmail.com>
Date: Thu, 17 Mar 2022 03:55:51 +0200
Subject: [PATCH] dist-files/Makefile.sharedlibrary: link against libm

Closes #2464.
---
 Makefile.sharedlibrary | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile.sharedlibrary b/Makefile.sharedlibrary
index 3a6c7cacc..4e20084e3 100644
--- a/Makefile.sharedlibrary
+++ b/Makefile.sharedlibrary
@@ -71,11 +71,11 @@ all: libduktape.$(SO_REALNAME_SUFFIX) libduktaped.$(SO_REALNAME_SUFFIX) duktape.
 
 libduktape.$(SO_REALNAME_SUFFIX):
 	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -shared -fPIC -Wall -Wextra -Os -Wl,$(LD_SONAME_ARG),libduktape.$(SO_SONAME_SUFFIX) \
-		-o $@ $(DUKTAPE_SRCDIR)/duktape.c
+		-o $@ $(DUKTAPE_SRCDIR)/duktape.c -lm
 
 libduktaped.$(SO_REALNAME_SUFFIX):
 	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -shared -fPIC -g -Wall -Wextra -Os -Wl,$(LD_SONAME_ARG),libduktaped.$(SO_SONAME_SUFFIX) \
-		-o $@ $(DUKTAPE_SRCDIR)/duktape.c
+		-o $@ $(DUKTAPE_SRCDIR)/duktape.c -lm
 
 duktape.pc:
 	sed -e "s|@PREFIX@|$(INSTALL_PREFIX)|" \
