From 6696712fa7336550c8740be3d8947c993cea16e0 Mon Sep 17 00:00:00 2001
From: Muhammad Falak R Wani <falakreyaz@gmail.com>
Date: Fri, 12 Aug 2022 07:27:06 +0530
Subject: [PATCH 1/3] xterm defaults

Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>
---
 XTerm.ad | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/XTerm.ad b/XTerm.ad
index 6d347a1..e4b1828 100644
--- a/XTerm.ad
+++ b/XTerm.ad
@@ -270,3 +270,11 @@
 !*allowTcapOps: false
 !*allowTitleOps: false
 !*allowWindowOps: false
+
+! Red Hat Defaults:
+*allowFontOps: false
+*allowTcapOps: false
+*VT100*eightBitInput: false
+*VT100*scrollBar: true
+*VT100*utf8Title: true
+*termName: xterm-256color
-- 
2.40.1

