diff --git a/xterm.desktop b/xterm.desktop
index b6b27d0..e9a2c6c 100644
--- a/xterm.desktop
+++ b/xterm.desktop
@@ -32,13 +32,13 @@
 # -----------------------------------------------------------------------------
 [Desktop Entry]
 Name=XTerm
-#GenericName=Terminal
+GenericName=Terminal
 Comment=standard terminal emulator for the X window system
 Exec=xterm
 Terminal=false
 Type=Application
 Encoding=UTF-8
-Icon=xterm-color_48x48
+Icon=xterm-color
 Categories=System;TerminalEmulator;
 Keywords=shell;prompt;command;commandline;cmd;
 StartupWMClass=XTerm
