From 43040291be2f48d6b2aae015592266c0b1dbd59e Mon Sep 17 00:00:00 2001
From: Muhammad Falak R Wani <falakreyaz@gmail.com>
Date: Fri, 12 Aug 2022 07:34:42 +0530
Subject: [PATCH 3/3] xterm man paths

Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>
---
 minstall.in | 8 ++++++++
 xterm.man   | 7 +++----
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/minstall.in b/minstall.in
index fe2c610..03b00b3 100644
--- a/minstall.in
+++ b/minstall.in
@@ -141,6 +141,11 @@ USE_chr1=`echo "$USE_name" | sed -e 's/^.//'`
 USE_Name=${USE_chr0}${USE_chr1}
 USE_NAME=`echo "$USE_name" | tr "$lower" "$upper"`
 
+fontpath=/usr/share/X11/fonts
+xorgcfgdir=/etc/X11
+xorgcfgfil=xorg.conf
+X_MANSECT=7
+
 sed	-e 's%__vendorversion__%"X Window System"%' \
 	-e 's%__app_version__%Patch\ \#'"$PATCH_NUM"% \
 	-e 's%__app_date__%'"$PATCH_YMD"% \
@@ -166,6 +171,9 @@ sed	-e 's%__vendorversion__%"X Window System"%' \
 	-e "s%/etc/utmp%$UTMP_PATH%g" \
 	-e 's%fIwtmp\\%fI'$WTMP_NAME'\\%g' \
 	-e "s%/etc/wtmp%$WTMP_PATH%g" \
+	-e s%__fontpath__%$fontpath%g \
+	-e s%__xorgcfgdir__%$xorgcfgdir%g \
+	-e s%__xorgcfgfil__%$xorgcfgfil%g \
 	-e 's%/\\(\*\*%/*%g' \
 	"$OLD_FILE" >$NEW_FILE
 # diff -u $OLD_FILE $NEW_FILE
diff --git a/xterm.man b/xterm.man
index f8270fd..51867fa 100644
--- a/xterm.man
+++ b/xterm.man
@@ -2572,19 +2572,18 @@ Since X11R6, bitmap fonts have been scaled.
 The font server claims to provide the bold font that \fI\*n\fP requests,
 but the result is not always readable.
 XFree86 introduced a feature which can be used to suppress the scaling.
-In the X server's configuration file (e.g., \*(``/etc/X11/XFree86\*(''
-or \*(``/etc/X11/xorg.conf\*(''), you
+In the X server's configuration file (e.g., \*(``__xorgcfgdir__/__xorgcfgfil__\*(''), you
 can add \*(``:unscaled\*('' to the end of the directory specification for the
 \*(``misc\*('' fonts,
 which comprise the fixed-pitch fonts that are used by \fI\*n\fP.
 For example
 .NS
-FontPath	"/usr/lib/X11/fonts/misc/"
+FontPath	"__fontpath__/misc/"
 .NE
 .IP
 would become
 .NS
-FontPath	"/usr/lib/X11/fonts/misc/:unscaled"
+FontPath	"__fontpath__/misc/:unscaled"
 .NE
 .IP
 Depending on your configuration, the font server may have its own configuration
-- 
2.40.1

