--- arpwatch-2.1a15/Makefile.in	2013-04-23 11:17:51.994488347 +0200
+++ arpwatch-2.1a15/Makefile.in.new	2013-04-23 11:17:24.000000000 +0200
@@ -48,12 +48,12 @@
 DEFS = -DDEBUG @DEFS@ -DARPDIR=\"$(ARPDIR)\" -DPATH_SENDMAIL=\"$(SENDMAIL)\"
 
 # Standard CFLAGS
-CFLAGS = $(CCOPT) $(DEFS) $(INCLS)
+CFLAGS = $(CCOPT) $(DEFS) $(INCLS) -pie
 
 # Standard LIBS
-LIBS = @LIBS@
+LIBS = @LIBS@ -pie -Wl,-z,relro,-z,now
 # Standard LIBS without libpcap.a
-SLIBS = @LBL_LIBS@
+SLIBS = @LBL_LIBS@ -pie -Wl,-z,relro,-z,now
 
 INSTALL = @INSTALL@
 SENDMAIL = @V_SENDMAIL@
