--- arpwatch-2.1a15/arp2ethers.scripts	2002-01-05 20:40:48.000000000 +0100
+++ arpwatch-2.1a15/arp2ethers	2006-11-09 14:34:42.000000000 +0100
@@ -13,7 +13,7 @@
 #	- sort
 #
 
-sort +2rn arp.dat | \
+sort -k 2 -rn arp.dat | \
     awk 'NF == 4 { print }' | \
     awk -f p.awk | \
     egrep -v '\.[0-9][0-9]*$' | \
--- arpwatch-2.1a15/arpfetch.scripts	2006-07-28 20:10:30.000000000 +0200
+++ arpwatch-2.1a15/arpfetch	2006-11-09 14:37:05.000000000 +0100
@@ -4,8 +4,6 @@
 # arpfetch - collect arp data from a cisco using net-snmp
 #
 
-export PATH="/usr/local/bin:${PATH}"
-
 prog=`basename $0`
 
 if [ $# -ne 2 ]; then
@@ -30,4 +28,3 @@
 	    print ea "\t" ip
     }'
 
-rm -f ${t1}
