From 6ec99107916b9d7b6486ea5fadbd8071eef1388f Mon Sep 17 00:00:00 2001
From: Muhammad Falak R Wani <falakreyaz@gmail.com>
Date: Sat, 2 Sep 2023 10:26:24 +0530
Subject: [PATCH 1/5] patch1

Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>
---
 configure.ac | 18 +++++-------------
 1 file changed, 5 insertions(+), 13 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1a12c04..5b3a883 100644
--- a/configure.ac
+++ b/configure.ac
@@ -189,7 +189,8 @@ AC_EGREP_CPP(yes,
 ], AC_NOTE(- you have a SVR4 system) AC_DEFINE(SVR4) svr4=1)
 if test -n "$svr4" ; then
 oldlibs="$LIBS"
-LIBS="$LIBS -lelf"
+# Humm we don't actually need to link against libelf for Linux
+LIBS="$LIBS"
 AC_CHECKING(SVR4)
 AC_TRY_LINK([
     #include <utmpx.h>
@@ -669,17 +670,13 @@ AC_TRY_LINK([
     tgetent((char *)0, (char *)0);
 ],,
 olibs="$LIBS"
-LIBS="-lcurses $olibs"
-AC_CHECKING(libcurses)
+LIBS="-ltinfo $olibs"
+AC_CHECKING(libtinfo)
 AC_TRY_LINK([
     #include <curses.h>
     #include <term.h>
 ],[
-#ifdef __hpux
-__sorry_hpux_libcurses_is_totally_broken_in_10_10();
-#else
 tgetent((char *)0, (char *)0);
-#endif
 ],,
 LIBS="-ltermcap $olibs"
 AC_CHECKING(libtermcap)
@@ -739,7 +736,7 @@ AC_TRY_RUN([
 int
 main()
 {
- return(strcmp(tgoto("%p1%d", 0, 1), "1") ? 0 : 1);
+    return 1;
 }], AC_NOTE(- you use the termcap database),
 AC_NOTE(- you use the terminfo database) AC_DEFINE(TERMINFO),
 AC_NOTE(- skipping check because we are cross compiling; assuming terminfo database is used) AC_DEFINE(TERMINFO))
@@ -953,11 +950,6 @@ fi
 dnl
 dnl    ****  loadav  ****
 dnl
-if test "$cross_compiling" = no ; then
-AC_CHECKING(for libutil(s))
-test -f /usr/lib/libutils.a && LIBS="$LIBS -lutils"
-test -f /usr/lib/libutil.a && LIBS="$LIBS -lutil"
-fi
 
 AC_CHECKING(getloadavg)
 AC_TRY_LINK([
-- 
2.40.1


