From 0fc13cd7b18ec571092bac9082459c5acebf65f8 Mon Sep 17 00:00:00 2001
From: Andy Grover <agrover@redhat.com>
Date: Tue, 8 Dec 2015 09:46:26 -0800
Subject: [PATCH] disable xen_pvscsi

---
 rtslib/fabric.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtslib/fabric.py b/rtslib/fabric.py
index 2887783..2eb6173 100644
--- a/rtslib/fabric.py
+++ b/rtslib/fabric.py
@@ -469,7 +469,7 @@ fabric_modules = {
     "tcm_fc": FCoEFabricModule,
 #    "usb_gadget": USBGadgetFabricModule, # very rare, don't show
     "vhost": VhostFabricModule,
-    "xen-pvscsi": XenPvScsiFabricModule,
+#    "xen-pvscsi": XenPvScsiFabricModule, # no xen on Fedora
     "ibmvscsis": IbmvscsisFabricModule,
     }
 
