diff -u -r libvdpau-1.4.orig/src/meson.build libvdpau-1.4/src/meson.build
--- libvdpau-1.4.orig/src/meson.build	2020-07-31 13:36:55.104530000 +0200
+++ libvdpau-1.4/src/meson.build	2020-07-31 14:21:02.982468000 +0200
@@ -16,4 +16,4 @@
     install : true,
 )
 
-install_data('vdpau_wrapper.cfg', install_dir : get_option('sysconfdir'))
+install_data('vdpau_wrapper.cfg', install_dir : '/usr/etc')
diff -u -r libvdpau-1.4.orig/src/vdpau_wrapper.c libvdpau-1.4/src/vdpau_wrapper.c
--- libvdpau-1.4.orig/src/vdpau_wrapper.c	2020-07-31 13:36:55.116541000 +0200
+++ libvdpau-1.4/src/vdpau_wrapper.c	2020-07-31 14:19:44.407242000 +0200
@@ -362,7 +362,10 @@
 
     fp = fopen(VDPAU_SYSCONFDIR "/vdpau_wrapper.cfg", "r");
     if (!fp) {
-        return;
+        fp = fopen("/usr/etc/vdpau_wrapper.cfg", "r");
+        if (!fp) {
+            return;
+        }
     }
 
     while (fgets(buffer, sizeof(buffer), fp) != NULL) {
