--- httpcomponents-client-4.5.6/httpclient/src/main/java/org/apache/http/client/utils/URLEncodedUtils.java	2018-11-06 09:37:41.743463818 +0100
+++ httpcomponents-client-4.5.6.new/httpclient/src/main/java/org/apache/http/client/utils/URLEncodedUtils.java	2019-03-13 10:49:52.762831489 +0100
@@ -32,6 +32,7 @@
 import java.io.InputStreamReader;
 import java.io.Reader;
 import java.net.URI;
+import java.nio.Buffer;
 import java.nio.ByteBuffer;
 import java.nio.CharBuffer;
 import java.nio.charset.Charset;
@@ -587,7 +588,7 @@
                 bb.put((byte) c);
             }
         }
-        bb.flip();
+        ((Buffer)bb).flip();
         return charset.decode(bb).toString();
     }
 
