From 31525d3855f876ddf2e29091b2e8d376f923e09e Mon Sep 17 00:00:00 2001
From: Rex Dieter <rdieter@gmail.com>
Date: Thu, 24 May 2018 14:40:53 -0500
Subject: [PATCH 2/2] xdg-open: better pcmanfm check (BR106636,BR106161)

---
 ChangeLog           | 2 ++
 scripts/xdg-open.in | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 6864f88..3eed7c4 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,7 @@
 === xdg-utils 1.1.4 ===
 
+2018-05-24
+   * xdg-open: better pcmanfm check (BR106636,BR106161)
 
 === xdg-utils 1.1.3 ===
 
diff --git a/scripts/xdg-open.in b/scripts/xdg-open.in
index 630e63e..bf9da4c 100644
--- a/scripts/xdg-open.in
+++ b/scripts/xdg-open.in
@@ -451,7 +451,7 @@ open_lxde()
 {
 
     # pcmanfm only knows how to handle file:// urls and filepaths, it seems.
-    if pcmanfm --help >/dev/null 2>&1 -a is_file_url_or_path "$1"; then
+    if pcmanfm --help >/dev/null 2>&1 && is_file_url_or_path "$1"; then
         local file="$(file_url_to_path "$1")"
 
         # handle relative paths
-- 
2.17.0

