diff -urB antlr-2.7.7/antlr/Makefile.in new/antlr/Makefile.in
--- antlr-2.7.7/antlr/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ new/antlr/Makefile.in	2010-08-02 16:52:40.515132125 +0200
@@ -321,11 +321,11 @@
 
 ## install antlr.jar in libdirectory if exists
 install ::	@ANTLR_JAR@ 
-	-$(MKDIR) -p "$(libdir)"
-	-$(MKDIR) -p "$(datadir)/$(versioneddir)"
+	-$(MKDIR) -p "$(DESTDIR)$(libdir)"
+	-$(MKDIR) -p "$(DESTDIR)$(datadir)/$(versioneddir)"
 	-test -f @ANTLR_JAR@ && {\
-	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(datadir)/$(versioneddir)" ; \
-	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(libdir)" ; \
+	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
+	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(DESTDIR)$(libdir)" ; \
 	}
 
 distclean :: clean
diff -urB antlr-2.7.7/doc/Makefile.in new/doc/Makefile.in
--- antlr-2.7.7/doc/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ new/doc/Makefile.in	2010-08-02 16:53:27.179728157 +0200
@@ -43,12 +43,12 @@
 	$(eol)
 
 install ::
-	@$(MKDIR) -p "$(antlr_doc_DIR)"
+	@$(MKDIR) -p "$(DESTDIR)$(antlr_doc_DIR)"
 	@@ECHO@ "install doc files .. "
 	@for f in $(antlr_doc_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(antlr_doc_DIR)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(antlr_doc_DIR)" ; \
 		fi ;\
 	done
 
diff -urB antlr-2.7.7/lib/cpp/antlr/CharScanner.hpp new/lib/cpp/antlr/CharScanner.hpp
--- antlr-2.7.7/lib/cpp/antlr/CharScanner.hpp	2006-11-01 22:37:17.000000000 +0100
+++ new/lib/cpp/antlr/CharScanner.hpp	2010-08-02 15:37:59.888152499 +0200
@@ -11,6 +11,8 @@
 #include <antlr/config.hpp>
 
 #include <map>
+#include <stdio.h>
+#include <string.h>
 
 #ifdef HAS_NOT_CCTYPE_H
 #include <ctype.h>
diff -urB antlr-2.7.7/lib/cpp/antlr/Makefile.in new/lib/cpp/antlr/Makefile.in
--- antlr-2.7.7/lib/cpp/antlr/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ new/lib/cpp/antlr/Makefile.in	2010-08-02 16:54:40.276072824 +0200
@@ -75,11 +75,11 @@
 
 install:
 	@@ECHO@ "install hpp files .. "
-	@$(MKDIR) -p "$(includedir)/antlr"
+	@$(MKDIR) -p "$(DESTDIR)$(includedir)/antlr"
 	@for f in $(antlr_hpp_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(includedir)/antlr" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(includedir)/antlr" ; \
 		fi ;\
 	done
 
diff -urB antlr-2.7.7/lib/cpp/src/Makefile.in new/lib/cpp/src/Makefile.in
--- antlr-2.7.7/lib/cpp/src/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ new/lib/cpp/src/Makefile.in	2010-08-02 16:54:09.396352933 +0200
@@ -114,12 +114,12 @@
   $(eol)
 
 this-install: @ANTLR_LIB@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install C++ core files .. "
 	@for f in $(antlr_lib_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
 		fi ;\
 	done
 
diff -urB antlr-2.7.7/lib/csharp/antlr.astframe/Makefile.in new/lib/csharp/antlr.astframe/Makefile.in
--- antlr-2.7.7/lib/csharp/antlr.astframe/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ new/lib/csharp/antlr.astframe/Makefile.in	2010-08-02 16:55:58.989338267 +0200
@@ -45,13 +45,13 @@
 
 ## install our target ..
 install :: @ASTFRAME_NET@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install C# core files .. "
 	@for f in @ASTFRAME_NET@  ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
-			$(INSTALL) -m 444 "$${f}" "$(datadir)/$(versioneddir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
 		fi ;\
 	done
 
diff -urB antlr-2.7.7/lib/csharp/antlr.runtime/Makefile.in new/lib/csharp/antlr.runtime/Makefile.in
--- antlr-2.7.7/lib/csharp/antlr.runtime/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ new/lib/csharp/antlr.runtime/Makefile.in	2010-08-02 16:56:29.183049014 +0200
@@ -135,13 +135,13 @@
 	@RMF@ Makefile
 
 install :: @ANTLR_NET@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install C# core files .. "
 	@for f in @ANTLR_NET@  ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
-			$(INSTALL) -m 444 "$${f}" "$(datadir)/$(versioneddir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
 		fi ;\
 	done
 
diff -urB antlr-2.7.7/lib/python/Makefile.in new/lib/python/Makefile.in
--- antlr-2.7.7/lib/python/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ new/lib/python/Makefile.in	2010-08-02 16:55:21.052695892 +0200
@@ -41,26 +41,26 @@
 extradir = $(datadir)/$(versioneddir)
 
 install-this:
-	$(MKDIR) -p "$(antlr_py_DIR)"
-	$(MKDIR) -p "$(libdir)"
+	$(MKDIR) -p "$(DESTDIR)$(antlr_py_DIR)"
+	$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install python files .. "
 	@for f in $(antlr_py_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
 		fi ;\
 	done
 	@for f in $(antlr_py1_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(antlr_py_DIR)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(antlr_py_DIR)" ; \
 		fi ;\
 	done
 	@f="$(objdir)/scripts/pyantlr.sh"; \
 	if test -f "$${f}" ; then \
 			@ECHO@ "install Python installer" ; \
 			$(MKDIR) -p "$(sbindir)" ; \
-			$(INSTALL) -m 555 "$${f}" "$(sbindir)" ; \
+			$(INSTALL) -m 555 "$${f}" "$(DESTDIR)$(sbindir)" ; \
 	fi
 
 
diff -urB antlr-2.7.7/Makefile.in new/Makefile.in
--- antlr-2.7.7/Makefile.in	2006-11-01 22:37:18.000000000 +0100
+++ new/Makefile.in	2010-08-02 16:52:05.892424224 +0200
@@ -124,16 +124,16 @@
 extradir = $(datadir)/$(versioneddir)
 
 install ::
-	$(MKDIR) -p "$(bindir)"
-	$(MKDIR) -p "$(extradir)"
-	$(MKDIR) -p "$(docdir)"
-	$(INSTALL) -m 755 scripts/run-antlr                          "$(bindir)/antlr"
-	$(INSTALL) -m 755 scripts/antlr-config                       "$(bindir)/antlr-config"
-	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-mode.el      "$(extradir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-jedit.xml    "$(extradir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/LICENSE.txt               "$(docdir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/README.txt                "$(docdir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/INSTALL.txt               "$(docdir)"
+	$(MKDIR) -p "$(DESTDIR)$(bindir)"
+	$(MKDIR) -p "$(DESTDIR)$(extradir)"
+	$(MKDIR) -p "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 755 scripts/run-antlr                          "$(DESTDIR)$(bindir)/antlr"
+	$(INSTALL) -m 755 scripts/antlr-config                       "$(DESTDIR)$(bindir)/antlr-config"
+	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-mode.el      "$(DESTDIR)$(extradir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-jedit.xml    "$(DESTDIR)$(extradir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/LICENSE.txt               "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/README.txt                "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/INSTALL.txt               "$(DESTDIR)$(docdir)"
 
 install ::
 	@ECHO@ "installation done"
