--- orc-2.2.0/c++/src/CMakeLists.txt.orig	2025-07-26 18:16:40.000000000 -0400
+++ orc-2.2.0/c++/src/CMakeLists.txt	2025-08-11 08:57:02.968852390 -0400
@@ -202,17 +202,22 @@
     BpackingAvx512.cc)
 endif(BUILD_ENABLE_AVX512)
 
-add_library (orc STATIC ${SOURCE_FILES})
+add_library (orc SHARED ${SOURCE_FILES})
+
+set_target_properties(orc PROPERTIES
+  POSITION_INDEPENDENT_CODE ON
+  VERSION 2.2.2
+  SOVERSION 2)
 
 target_link_libraries (orc
   INTERFACE
     ${ORC_INSTALL_INTERFACE_TARGETS}
   PRIVATE
-    $<BUILD_INTERFACE:orc::protobuf>
-    $<BUILD_INTERFACE:orc::zlib>
-    $<BUILD_INTERFACE:orc::snappy>
-    $<BUILD_INTERFACE:orc::lz4>
-    $<BUILD_INTERFACE:orc::zstd>
+    protobuf
+    z
+    snappy
+    lz4
+    zstd
     $<BUILD_INTERFACE:${LIBHDFSPP_LIBRARIES}>
     $<BUILD_INTERFACE:${SPARSEHASH_LIBRARIES}>
   )
@@ -249,6 +254,6 @@
 
 install(TARGETS orc EXPORT orc_targets)
 install(EXPORT orc_targets
-  DESTINATION ${ORC_INSTALL_CMAKE_DIR}
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}
   NAMESPACE "orc::"
   FILE "orcTargets.cmake")
--- orc-2.2.2/cmake_modules/ThirdpartyToolchain.cmake.orig	2026-01-12 13:20:40.873102664 -0500
+++ orc-2.2.2/cmake_modules/ThirdpartyToolchain.cmake	2026-01-12 14:01:23.763082029 -0500
@@ -20,20 +20,20 @@
 set(ORC_INSTALL_INTERFACE_TARGETS)
 
 set(ORC_FORMAT_VERSION "1.1.1")
-set(LZ4_VERSION "1.10.0")
-set(SNAPPY_VERSION "1.2.2")
-set(ZLIB_VERSION "1.3.1")
-set(GTEST_VERSION "1.17.0")
-set(PROTOBUF_VERSION "3.5.1")
-set(ZSTD_VERSION "1.5.7")
-set(SPARSEHASH_VERSION "2.11.1")
-
-option(ORC_PREFER_STATIC_PROTOBUF "Prefer static protobuf library, if available" ON)
-option(ORC_PREFER_STATIC_SNAPPY   "Prefer static snappy library, if available"   ON)
-option(ORC_PREFER_STATIC_LZ4      "Prefer static lz4 library, if available"      ON)
-option(ORC_PREFER_STATIC_ZSTD     "Prefer static zstd library, if available"     ON)
-option(ORC_PREFER_STATIC_ZLIB     "Prefer static zlib library, if available"     ON)
-option(ORC_PREFER_STATIC_GMOCK    "Prefer static gmock library, if available"    ON)
+#set(LZ4_VERSION "1.10.0")
+#set(SNAPPY_VERSION "1.2.2")
+#set(ZLIB_VERSION "1.3.1")
+#set(GTEST_VERSION "1.17.0")
+#set(PROTOBUF_VERSION "3.5.1")
+#set(ZSTD_VERSION "1.5.7")
+#set(SPARSEHASH_VERSION "2.11.1")
+
+#option(ORC_PREFER_STATIC_PROTOBUF "Prefer static protobuf library, if available" ON)
+#option(ORC_PREFER_STATIC_SNAPPY   "Prefer static snappy library, if available"   ON)
+#option(ORC_PREFER_STATIC_LZ4      "Prefer static lz4 library, if available"      ON)
+#option(ORC_PREFER_STATIC_ZSTD     "Prefer static zstd library, if available"     ON)
+#option(ORC_PREFER_STATIC_ZLIB     "Prefer static zlib library, if available"     ON)
+#option(ORC_PREFER_STATIC_GMOCK    "Prefer static gmock library, if available"    ON)
 
 # zstd requires us to add the threads
 FIND_PACKAGE(Threads REQUIRED)
@@ -145,8 +145,7 @@
   message(STATUS "Using DEFAULT URL: ${ORC_FORMAT_SOURCE_URL}")
 endif()
 ExternalProject_Add (orc-format_ep
-  URL ${ORC_FORMAT_SOURCE_URL}
-  URL_HASH SHA256=584dfe2a4202946178fd8fc7d1239be7805b9ed4596ab2042dee739e7880992b
+  URL "file:///builddir/build/SOURCES/orc-format-${ORC_FORMAT_VERSION}.tar.gz"
   CONFIGURE_COMMAND ""
   BUILD_COMMAND     ""
   INSTALL_COMMAND     ""
