--- hslua-1.2.0/hslua.cabal~	2001-09-09 09:46:40.000000000 +0800
+++ hslua-1.2.0/hslua.cabal	2021-02-15 22:34:53.477731593 +0800
@@ -213,8 +213,7 @@
     cpp-options:         -DHARDCODE_REG_KEYS
 
   if !impl(ghc >= 8.0)
-    build-depends:       fail                 >= 4.9    && < 5
-                       , semigroups           >= 0.18   && < 0.20
+    build-depends:       semigroups           >= 0.18   && < 0.20
 
   if impl(ghc < 8.8)
     build-depends:       base-compat          >= 0.10
@@ -267,8 +266,7 @@
                      , tasty-quickcheck     >= 0.8
 
   if impl(ghc < 8.0)
-    build-depends:       fail                 >= 4.9    && < 5
-                       , semigroups           >= 0.18   && < 0.20
+    build-depends:       semigroups           >= 0.18   && < 0.20
   if impl(ghc < 8.8)
     build-depends:       base-compat          >= 0.10
     hs-source-dirs:      prelude
