diff -rup a/sysdeps/linux-gnu/ppc/plt.c b/sysdeps/linux-gnu/ppc/plt.c
--- a/sysdeps/linux-gnu/ppc/plt.c	2022-01-27 20:40:52.956980433 -0500
+++ b/sysdeps/linux-gnu/ppc/plt.c	2022-01-27 20:41:37.031599771 -0500
@@ -687,8 +687,8 @@ arch_elf_add_func_entry(struct process *
 		if (libsym == NULL
 		    || library_symbol_init(libsym, addr, full_name, 1,
 					   LS_TOPLT_NONE) < 0) {
-			free(libsym);
 			delete_symbol_chain(libsym);
+			free(libsym);
 			libsym = NULL;
 			fprintf(stderr, "Couldn't add symbol %s"
 				"for tracing.\n", name);
