diff -up ./mk/rte.lib.mk.stupid ./mk/rte.lib.mk
--- ./mk/rte.lib.mk.stupid	2018-09-27 13:45:48.336632453 -0400
+++ ./mk/rte.lib.mk	2018-09-27 13:46:03.900563944 -0400
@@ -103,7 +103,7 @@ ifneq ($(CC_SUPPORTS_Z),false)
 NO_UNDEFINED := -z defs
 endif
 
-O_TO_S = $(LD) -L$(RTE_SDK_BIN)/lib $(_CPU_LDFLAGS) $(EXTRA_LDFLAGS) \
+O_TO_S = $(LD) -L$(RTE_SDK_BIN)/lib $(_CPU_LDFLAGS) $(EXTRA_LDFLAGS) -fcf-protection=full \
 	  -shared $(OBJS-y) $(NO_UNDEFINED) $(LDLIBS) -Wl,-soname,$(LIB) -o $(LIB)
 O_TO_S_STR = $(subst ','\'',$(O_TO_S)) #'# fix syntax highlight
 O_TO_S_DISP = $(if $(V),"$(O_TO_S_STR)","  LD $(@)")
