--- it_IT.dic	2010-04-29 14:31:47.000000000 +0100
+++ it_IT.dic	2010-04-29 14:32:07.000000000 +0100
@@ -1,32 +1,4 @@
-95221
-/ Dizionario italiano
-/
-/ Copyright (C) 2001, 2002 Gianluca Turconi
-/ Copyright (C) 2002, 2003, 2004 Gianluca Turconi and Davide Prina
-/ Copyright (C) 2004, 2005, 2006, 2007  Davide Prina
-/
-/ E-Mail: DavidePrina(CHIOCCIOLA)yahoo(PUNTO)com
-/ home: http://linguistico.sf.net/wiki
-/
-/ Version 2.4 (01/09/2007) (DD/MM/YYYY)
-/
-/ This file is distributed under GPL license.
-/
-/ This file is part of the "dizionario italiano".
-/
-/ The "dizionario italiano" is free software; you can redistribute it and/or
-/ modify it under the terms of the GNU General Public License as published
-/ by the Free Software Foundation; either version 3 of the License, or
-/ (at your option) any later version.
-/
-/ The "dizionario italiano" is distributed in the hope that it will be                
-/ useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
-/ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
-/ GNU General Public License for more details.
-/
-/ You should have received a copy of the GNU General Public
-/ License along with the "dizionario italiano",
-/ if not, see <http://www.gnu.org/licenses/>.
+95195
 a
 ab
 abaco/OTqr
