diff --git a/maint/config/getgrouplist.m4 b/maint/config/getgrouplist.m4
index 3e52c3001..ca97e2085 100644
--- a/maint/config/getgrouplist.m4
+++ b/maint/config/getgrouplist.m4
@@ -1,5 +1,6 @@
 AC_DEFUN([AX_GETGROUPLIST],
 [
+	AC_REQUIRE([AC_EGREP_HEADER])
 	AC_MSG_CHECKING(for getgrouplist in distro)
 	dnl if this test passes, the distro does have it
 	dnl if this test fails, the distro does not have it
