diff -ruN a/fluent-plugin-prometheus_pushgateway.gemspec b/fluent-plugin-prometheus_pushgateway.gemspec
--- a/fluent-plugin-prometheus_pushgateway.gemspec	2022-07-07 16:48:18.749481200 -0700
+++ b/fluent-plugin-prometheus_pushgateway.gemspec	2022-07-07 18:06:31.785591980 -0700
@@ -13,9 +13,7 @@
 
   # Specify which files should be added to the gem when it is released.
   # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
-  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
-    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
-  end
+  spec.files         = Dir['**/*'].reject { |f| f.match(%r{^(test|spec|features)/}) }
   spec.bindir        = "exe"
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
