--- crossbeam-utils-0.8.21/Cargo.toml	1970-01-01T00:00:01+00:00
+++ crossbeam-utils-0.8.21/Cargo.toml	2024-12-15T19:16:25.064716+00:00
@@ -80,10 +80,6 @@
 nightly = []
 std = []
 
-[target."cfg(crossbeam_loom)".dependencies.loom]
-version = "0.7.1"
-optional = true
-
 [lints.clippy.declare_interior_mutable_const]
 level = "allow"
 priority = 1
@@ -99,3 +95,4 @@
     "cfg(crossbeam_loom)",
     "cfg(crossbeam_sanitize)",
 ]
+
