diff -up zfs-fuse-0.7.2.2/src/SConstruct.py3 zfs-fuse-0.7.2.2/src/SConstruct
--- zfs-fuse-0.7.2.2/src/SConstruct.py3	2018-01-17 11:02:43.616124442 +0100
+++ zfs-fuse-0.7.2.2/src/SConstruct	2018-01-17 11:04:22.826712411 +0100
@@ -57,7 +57,7 @@ else:
 if not (('-DDEBUG' in env['CCFLAGS']) or 
 		('-DNDEBUG' in env['CCFLAGS'])):
 	print
-	print "Misconfigured debug level: Neither DEBUG or NDEBUG appears to have been defined: %s" % env['CCFLAGS']
+	print("Misconfigured debug level: Neither DEBUG or NDEBUG appears to have been defined: %s" % env['CCFLAGS'])
 	sys.exit(1)
 
 env['CPPPATH'] = []
@@ -82,7 +82,7 @@ myarch = getarch(arch)
 
 if not myarch:
 	print
-	print 'Sorry, only the x86, amd64 and sparc64 hardware architectures are supported'
+	print('Sorry, only the x86, amd64 and sparc64 hardware architectures are supported')
 	sys.exit(1)
 
 if myarch == 'sparc64':
@@ -131,7 +131,7 @@ env.Install(man_dir, '../doc/zstreamdump
 env.Install(man_dir, '../doc/zfs-fuse.8')
 
 if "tags" in sys.argv:
-    print "updating tags..."
+    print("updating tags...")
     os.system("ctags --extra=+f `find -name '*.c'` `find -name '*.h'`")
 
 env.Alias('install', [install_dir, man_dir, cfg_dir])
