diff -ruN a/cool.io.gemspec b/cool.io.gemspec
--- a/cool.io.gemspec	2021-02-16 15:13:41.000000000 -0800
+++ b/cool.io.gemspec	2022-07-03 15:56:02.777359145 -0700
@@ -13,9 +13,9 @@
   s.extensions = ["ext/cool.io/extconf.rb", "ext/iobuffer/extconf.rb"]
   s.licenses   = ["MIT"]
 
-  s.files         = `git ls-files`.split("\n")
-  s.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
-  s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
+  s.files         = Dir['**/*']
+  s.test_files    = s.files.grep(%r{^(test|spec|features)/})
+  s.executables   = s.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   s.require_paths = ["lib"]
 
   s.add_development_dependency "rake-compiler", "~> 1.0"
