From 04c6ae5109f2c315221166b501870ba067eab9a6 Mon Sep 17 00:00:00 2001
From: Chris PeBenito <Christopher.PeBenito@microsoft.com>
Date: Fri, 20 Aug 2021 17:08:38 +0000
Subject: [PATCH 1/8] Makefile: Revise relabel targets to relabel all seclabel
 filesystems.

Signed-off-by: Chris PeBenito <Christopher.PeBenito@microsoft.com>
---
 Makefile | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index b01f67bad..28403e517 100644
--- a/Makefile
+++ b/Makefile
@@ -320,8 +320,7 @@ off_mods += $(filter-out $(cmdline_off) $(cmdline_base) $(cmdline_mods), $(mod_c
 off_mods += $(filter-out $(base_mods) $(mod_mods) $(off_mods),$(notdir $(detected_mods)))
 
 # filesystems to be used in labeling targets
-filesystems = $(shell mount | grep -v "context=" | egrep -v '\((|.*,)bind(,.*|)\)' | $(AWK) '/(ext[234]|btrfs| xfs| jfs).*rw/{print $$3}';)
-fs_names := "btrfs ext2 ext3 ext4 xfs jfs"
+filesystems = $(shell mount | $(AWK) '$$NF ~ /\yseclabel\y/ {print $$3}')
 
 ########################################
 #
@@ -612,7 +611,7 @@ $(tags):
 # Filesystem labeling
 #
 checklabels:
-	@echo "Checking labels on filesystem types: $(fs_names)"
+	@echo "Checking labels on filesystems: $(filesystems)"
 	@if test -z "$(filesystems)"; then \
 		echo "No filesystems with extended attributes found!" ;\
 		false ;\
@@ -620,7 +619,7 @@ checklabels:
 	$(verbose) $(SETFILES) -E -v -n $(fcpath) $(filesystems)
 
 restorelabels:
-	@echo "Restoring labels on filesystem types: $(fs_names)"
+	@echo "Restoring labels on filesystems: $(filesystems)"
 	@if test -z "$(filesystems)"; then \
 		echo "No filesystems with extended attributes found!" ;\
 		false ;\
@@ -628,7 +627,7 @@ restorelabels:
 	$(verbose) $(SETFILES) -E -v $(fcpath) $(filesystems)
 
 relabel:
-	@echo "Relabeling filesystem types: $(fs_names)"
+	@echo "Relabeling filesystems: $(filesystems)"
 	@if test -z "$(filesystems)"; then \
 		echo "No filesystems with extended attributes found!" ;\
 		false ;\
@@ -636,7 +635,7 @@ relabel:
 	$(verbose) $(SETFILES) -E $(fcpath) $(filesystems)
 
 resetlabels:
-	@echo "Resetting labels on filesystem types: $(fs_names)"
+	@echo "Resetting labels on filesystems: $(filesystems)"
 	@if test -z "$(filesystems)"; then \
 		echo "No filesystems with extended attributes found!" ;\
 		false ;\
-- 
2.17.1

