From 9c787c65cf5647a7c51e1d6a47e05213408bb99b Mon Sep 17 00:00:00 2001
From: Chris PeBenito <Christopher.PeBenito@microsoft.com>
Date: Tue, 30 Aug 2022 13:46:15 +0000
Subject: [PATCH 05/14] Temp kubernetes fix

---
 policy/modules/services/container.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/modules/services/container.te b/policy/modules/services/container.te
index 096d6c23d..ca844558a 100644
--- a/policy/modules/services/container.te
+++ b/policy/modules/services/container.te
@@ -1191,3 +1191,6 @@ optional_policy(`
        unconfined_domain_noaudit(spc_user_t)
        domain_ptrace_all_domains(spc_user_t)
 ')
+
+# temp kubernetes fix
+type_transition initrc_t var_run_t:dir container_runtime_t "k8s.cni.cncf.io";
-- 
2.43.2

