From 22df79fe65d1eb8199a6c7e50e569d2d21460ecb Mon Sep 17 00:00:00 2001
From: Chris PeBenito <Christopher.PeBenito@microsoft.com>
Date: Mon, 23 May 2022 14:45:00 +0000
Subject: [PATCH 34/37] files: Make etc_runtime_t a config file.

Signed-off-by: Chris PeBenito <Christopher.PeBenito@microsoft.com>
---
 policy/modules/kernel/files.te | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/policy/modules/kernel/files.te b/policy/modules/kernel/files.te
index c1775f65d..2691a8611 100644
--- a/policy/modules/kernel/files.te
+++ b/policy/modules/kernel/files.te
@@ -72,7 +72,7 @@ optional_policy(`
 # files in /etc that are automatically
 # generated during initialization.
 #
-type etc_runtime_t;
+type etc_runtime_t, configfile;
 files_type(etc_runtime_t)
 
 #
-- 
2.25.1

