From e9c6d23736e5005d3f46c1b501fa5f62818df4be Mon Sep 17 00:00:00 2001
From: Chris PeBenito <Christopher.PeBenito@microsoft.com>
Date: Mon, 29 Aug 2022 19:21:11 +0000
Subject: [PATCH 46/47] init: Allow nnp/nosuid transitions from systemd
 initrc_t.

Signed-off-by: Chris PeBenito <Christopher.PeBenito@microsoft.com>
---
 policy/modules/system/init.if | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/init.if b/policy/modules/system/init.if
index 8bc28e4ab..ee37afcca 100644
--- a/policy/modules/system/init.if
+++ b/policy/modules/system/init.if
@@ -551,6 +551,8 @@ interface(`init_system_domain',`
 
 	ifdef(`init_systemd',`
 		init_domain($1, $2)
+
+		allow initrc_t $1:process2 { nnp_transition nosuid_transition };
 	')
 ')
 
-- 
2.25.1

