From 4bc0a53200c73ff4d97fe096336fdbf916d341a9 Mon Sep 17 00:00:00 2001
From: Chris PeBenito <Christopher.PeBenito@microsoft.com>
Date: Mon, 29 Aug 2022 19:22:38 +0000
Subject: [PATCH 47/47] selinuxutil: Semanage reads policy for export.

Signed-off-by: Chris PeBenito <Christopher.PeBenito@microsoft.com>
---
 policy/modules/system/selinuxutil.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/selinuxutil.te b/policy/modules/system/selinuxutil.te
index 0ffc0f9b1..8bc39cbc8 100644
--- a/policy/modules/system/selinuxutil.te
+++ b/policy/modules/system/selinuxutil.te
@@ -517,6 +517,7 @@ mls_file_read_all_levels(semanage_t)
 selinux_validate_context(semanage_t)
 selinux_get_enforce_mode(semanage_t)
 selinux_getattr_fs(semanage_t)
+selinux_read_policy(semanage_t)
 # for setsebool:
 selinux_set_all_booleans(semanage_t)
 
-- 
2.25.1

