From 547272fdee2dee9b598bf17b9e92770724586f49 Mon Sep 17 00:00:00 2001
From: Chris PeBenito <Christopher.PeBenito@microsoft.com>
Date: Mon, 29 Aug 2022 20:07:54 +0000
Subject: [PATCH 48/48] sysnetwork: ifconfig searches debugfs.

Signed-off-by: Chris PeBenito <Christopher.PeBenito@microsoft.com>
---
 policy/modules/system/sysnetwork.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/sysnetwork.te b/policy/modules/system/sysnetwork.te
index e6c1a453f..fe729c75a 100644
--- a/policy/modules/system/sysnetwork.te
+++ b/policy/modules/system/sysnetwork.te
@@ -321,6 +321,7 @@ kernel_read_network_state(ifconfig_t)
 kernel_request_load_module(ifconfig_t)
 kernel_search_network_sysctl(ifconfig_t)
 kernel_rw_net_sysctls(ifconfig_t)
+kernel_search_debugfs(ifconfig_t)
 
 corenet_rw_tun_tap_dev(ifconfig_t)
 
-- 
2.25.1

