From d7f9b9c1cd5275e566df5bc6fa6f509bc461a961 Mon Sep 17 00:00:00 2001
From: Chris PeBenito <Christopher.PeBenito@microsoft.com>
Date: Mon, 29 Aug 2022 20:22:37 +0000
Subject: [PATCH 49/49] usermanage: Add dac_read_search to useradd.

Signed-off-by: Chris PeBenito <Christopher.PeBenito@microsoft.com>
---
 policy/modules/admin/usermanage.te | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/policy/modules/admin/usermanage.te b/policy/modules/admin/usermanage.te
index 0b3013181..9c288323c 100644
--- a/policy/modules/admin/usermanage.te
+++ b/policy/modules/admin/usermanage.te
@@ -468,7 +468,7 @@ optional_policy(`
 # Useradd local policy
 #
 
-allow useradd_t self:capability { chown dac_override fowner fsetid kill setuid sys_resource };
+allow useradd_t self:capability { chown dac_read_search dac_override fowner fsetid kill setuid sys_resource };
 dontaudit useradd_t self:capability { net_admin sys_tty_config };
 dontaudit useradd_t self:cap_userns sys_ptrace;
 allow useradd_t self:process { transition signal_perms getsched setsched getsession getpgid setpgid getcap setcap share getattr setfscreate noatsecure siginh rlimitinh dyntransition setkeycreate setsockcreate getrlimit };
-- 
2.25.1

