--- jflex/examples/byaccj/README.orig	2003-01-07 16:30:17.000000000 +0100
+++ jflex/examples/byaccj/README	2005-10-06 21:08:38.000000000 +0200
@@ -1,8 +1,8 @@
 This directory contains an interoperability example for BYacc/J and JFlex
 (a small calculator)
 
-You need the parser generator byacc/j from
-http://troi.lincom-asg.com/~rjamison/byacc/
+You need the parser generator byacc/j from
+http://byaccj.sourceforge.net/
 for this example to work (version byacc/j 1.1 has been tested).
 
 Use the Makefile to generate the lexer and parser.
