Index: jflex-1.4.3/src/JFlex/tests/AllTests.java
===================================================================
--- jflex-1.4.3.orig/src/JFlex/tests/AllTests.java
+++ jflex-1.4.3/src/JFlex/tests/AllTests.java
@@ -32,7 +32,7 @@ import junit.framework.TestSuite;
 public class AllTests {
 
   public static void main(String[] args) {
-    junit.textui.TestRunner.run(AllTests.class);
+    //junit.textui.TestRunner.run(AllTests.class);
   }
 
   public static Test suite() {
