diff -ru nvidia-container-toolkit-1.9.0-orig/go.mod nvidia-container-toolkit-1.9.0/go.mod
--- nvidia-container-toolkit-1.9.0-orig/go.mod	2022-03-30 18:10:08.868848600 -0700
+++ nvidia-container-toolkit-1.9.0/go.mod	2022-03-30 18:12:12.471443600 -0700
@@ -8,7 +8,6 @@
 	github.com/opencontainers/runtime-spec v1.0.3-0.20211101234015-a3c33d663ebc
 	github.com/pelletier/go-toml v1.9.3
 	github.com/sirupsen/logrus v1.8.1
-	github.com/stretchr/testify v1.7.0
 	github.com/tsaikd/KDGoLib v0.0.0-20191001134900-7f3cf518e07d
 	github.com/urfave/cli/v2 v2.3.0
 	golang.org/x/mod v0.3.0
Only in nvidia-container-toolkit-1.9.0-orig: vendor
