diff -ur nvidia-container-toolkit-1.5.1-orig/go.mod nvidia-container-toolkit-1.5.1/go.mod
--- nvidia-container-toolkit-1.5.1-orig/go.mod	2021-09-29 08:36:04.432397867 -0700
+++ nvidia-container-toolkit-1.5.1/go.mod	2021-09-29 16:37:20.464834034 -0700
@@ -4,6 +4,5 @@
 
 require (
 	github.com/BurntSushi/toml v0.3.1
-	github.com/stretchr/testify v1.7.0
 	golang.org/x/mod v0.3.0
 )
Only in nvidia-container-toolkit-1.5.1-orig: vendor
