diff -up mrtg-2.17.7/bin/mrtg.orig mrtg-2.17.7/bin/mrtg
--- mrtg-2.17.7/bin/mrtg.orig	2018-08-08 13:09:59.682296406 +0200
+++ mrtg-2.17.7/bin/mrtg	2018-08-08 13:11:01.089173582 +0200
@@ -302,7 +302,7 @@ sub main {
     if ($cfg{enableipv6} eq 'yes') {
         if ( eval {local $SIG{__DIE__};require Socket; require Socket6; require IO::Socket::INET6;}) {
             import Socket;
-            import Socket6;
+            Socket6->import(qw(inet_pton getaddrinfo));
             debug('base', "IPv6 libraries found, IPv6 enabled.");
         } else {
             warn "$NOW: WARNING: IPv6 libraries not found, IPv6 disabled.\n";
diff -up mrtg-2.17.7/lib/mrtg2/SNMP_Session.pm.orig mrtg-2.17.7/lib/mrtg2/SNMP_Session.pm
--- mrtg-2.17.7/lib/mrtg2/SNMP_Session.pm.orig	2018-07-13 08:09:56.000000000 +0200
+++ mrtg-2.17.7/lib/mrtg2/SNMP_Session.pm	2018-08-08 13:13:15.361988807 +0200
@@ -146,7 +146,7 @@ BEGIN {
 
     if (eval {local $SIG{__DIE__};require Socket6;} &&
        eval {local $SIG{__DIE__};require IO::Socket::INET6; IO::Socket::INET6->VERSION("1.26");}) {
-       Socket6->import(qw(pack_sockaddr_in6 inet_pton getaddrinfo));
+       Socket6->import(qw(inet_pton getaddrinfo inet_ntop));
 	$ipv6_addr_len = length(pack_sockaddr_in6(161, inet_pton(AF_INET6(), "::1")));
 	$SNMP_Session::ipv6available = 1;
     }
@@ -601,7 +601,7 @@ use Carp;
 BEGIN {
     if($SNMP_Session::ipv6available) {
 	    import IO::Socket::INET6;
-        Socket6->import(qw(pack_sockaddr_in6 inet_pton getaddrinfo));
+        Socket6->import(qw(inet_pton getaddrinfo inet_ntop));
     }
 }
 
