diff --git a/src/conf/alsa.conf b/src/conf/alsa.conf
index 18427ec6..1915af4e 100644
--- a/src/conf/alsa.conf
+++ b/src/conf/alsa.conf
@@ -80,8 +80,7 @@ defaults.pcm.nonblock 1
 defaults.pcm.compat 0
 defaults.pcm.minperiodtime 5000		# in us
 defaults.pcm.ipc_key 5678293
-defaults.pcm.ipc_gid audio
-defaults.pcm.ipc_perm 0660
+defaults.pcm.ipc_perm 0600
 defaults.pcm.tstamp_type default
 defaults.pcm.dmix.max_periods 0
 defaults.pcm.dmix.channels 2
diff --git a/src/conf/pcm/dmix.conf b/src/conf/pcm/dmix.conf
index 50e573da..70523f29 100644
--- a/src/conf/pcm/dmix.conf
+++ b/src/conf/pcm/dmix.conf
@@ -48,10 +48,6 @@ pcm.!dmix {
 		@func refer
 		name defaults.pcm.ipc_key
 	}
-	ipc_gid {
-		@func refer
-		name defaults.pcm.ipc_gid
-	}
 	ipc_perm {
 		@func refer
 		name defaults.pcm.ipc_perm
diff --git a/src/conf/pcm/dsnoop.conf b/src/conf/pcm/dsnoop.conf
index f4336e5f..60b9f212 100644
--- a/src/conf/pcm/dsnoop.conf
+++ b/src/conf/pcm/dsnoop.conf
@@ -41,10 +41,6 @@ pcm.!dsnoop {
 		@func refer
 		name defaults.pcm.ipc_key
 	}	
-	ipc_gid {
-		@func refer
-		name defaults.pcm.ipc_gid
-	}
 	ipc_perm {
 		@func refer
 		name defaults.pcm.ipc_perm
