--- signal-hook-mio-0.2.5/Cargo.toml	1970-01-01T00:00:01+00:00
+++ signal-hook-mio-0.2.5/Cargo.toml	2025-10-29T10:35:16.492565+00:00
@@ -48,25 +48,12 @@
 repository = "vorner/signal-hook"
 
 [features]
-support-v0_6 = [
-    "mio-0_6",
-    "mio-uds",
-]
-support-v0_7 = ["mio-0_7"]
 support-v0_8 = ["mio-0_8"]
 support-v1_0 = ["mio-1_0"]
 
 [lib]
 name = "signal_hook_mio"
 path = "src/lib.rs"
-
-[[test]]
-name = "mio_0_6"
-path = "tests/mio_0_6.rs"
-
-[[test]]
-name = "mio_0_7"
-path = "tests/mio_0_7.rs"
 
 [[test]]
 name = "mio_0_8"
@@ -78,20 +65,6 @@
 
 [dependencies.libc]
 version = "~0.2"
-
-[dependencies.mio-0_6]
-version = "~0.6"
-optional = true
-package = "mio"
-
-[dependencies.mio-0_7]
-version = "~0.7"
-features = [
-    "os-util",
-    "uds",
-]
-optional = true
-package = "mio"
 
 [dependencies.mio-0_8]
 version = "~0.8"
@@ -111,21 +84,8 @@
 optional = true
 package = "mio"
 
-[dependencies.mio-uds]
-version = "~0.6"
-optional = true
-
 [dependencies.signal-hook]
 version = "~0.3"
 
-[dev-dependencies.mio-0_7]
-version = "~0.7"
-features = [
-    "os-util",
-    "os-poll",
-    "uds",
-]
-package = "mio"
-
 [dev-dependencies.serial_test]
 version = "~3"
