--- reqsign-aws-v4-3.0.2/Cargo.toml	2006-07-24T01:21:28+00:00
+++ reqsign-aws-v4-3.0.2/Cargo.toml	2026-07-10T16:57:58.513308+00:00
@@ -95,19 +95,6 @@
 [dependencies.sha1]
 version = "0.11"
 
-[dev-dependencies.aws-credential-types]
-version = "1.1.8"
-
-[dev-dependencies.aws-sigv4]
-version = "1.2.0"
-
-[dev-dependencies.criterion]
-version = "0.8.1"
-features = [
-    "async_tokio",
-    "html_reports",
-]
-
 [dev-dependencies.dotenvy]
 version = "0.15"
 
