--- reqsign-aws-v4-3.3.1/Cargo.toml	2006-07-24T01:21:28+00:00
+++ reqsign-aws-v4-3.3.1/Cargo.toml	2026-09-26T07:01:06.070790+00:00
@@ -60,7 +60,7 @@
 version = "0.4"
 
 [dependencies.quick-xml]
-version = "0.42.0"
+version = ">=0.41.0, <0.43.0"
 features = ["serialize"]
 
 [dependencies.reqsign-aws-core]
@@ -75,19 +75,6 @@
 
 [dev-dependencies.anyhow]
 version = "1"
-
-[dev-dependencies.aws-credential-types]
-version = "1.1.8"
-
-[dev-dependencies.aws-sigv4]
-version = "1.2.0"
-
-[dev-dependencies.criterion]
-version = "0.8.1"
-features = [
-    "async_tokio",
-    "html_reports",
-]
 
 [dev-dependencies.dotenvy]
 version = "0.15"
