diff --git a/src/rf64.c b/src/rf64.c
index 876cd45..0059382 100644
--- a/src/rf64.c
+++ b/src/rf64.c
@@ -742,9 +742,11 @@ rf64_write_header (SF_PRIVATE *psf, int calc_length)
 #endif
 
 	/* Padding may be needed if string data sizes change. */
-	pad_size = psf->dataoffset - 16 - psf->header.indx ;
-	if (pad_size >= 0)
-		psf_binheader_writef (psf, "m4z", BHWm (PAD_MARKER), BHW4 ((unsigned int) pad_size), BHWz (pad_size)) ;
+	if (psf->header.indx + 8 < psf->dataoffset)
+	{	/* Add PAD data if necessary. */
+		int k = psf->dataoffset - 16 - psf->header.indx ;
+		psf_binheader_writef (psf, "m4z", BHWm (PAD_MARKER), BHW4 ((unsigned int) k), BHWz (k)) ;
+		} ;
 
 	if (wpriv->rf64_downgrade && (psf->filelength < RIFF_DOWNGRADE_BYTES))
 		psf_binheader_writef (psf, "tm8", BHWm (data_MARKER), BHW8 (psf->datalength)) ;
