diff --git a/pyproject.toml b/pyproject.toml
index 83d8bee..2c07851 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,10 +1,34 @@
-[build-system]
-requires = [
-    'setuptools>=44',
-    'wheel>=0.30.0',
-    'six',
+[project]
+name = 'pam'
+version = '2.0.2'
+authors = [{name = "David Ford", email = "david@blue-labs.org"}]
+description = 'Python PAM module using ctypes, py3'
+readme = 'README.md'
+license = {text = "MIT License"}
+
+classifiers = [
+    'Development Status :: 6 - Mature',
+    'Environment :: Plugins',
+    'Intended Audience :: Developers',
+    'Intended Audience :: Information Technology',
+    'Intended Audience :: System Administrators',
+    'License :: OSI Approved :: MIT License',
+    'Operating System :: POSIX',
+    'Operating System :: POSIX :: Linux',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 2',
+    'Programming Language :: Python :: 3',
+    'Topic :: Security',
+    'Topic :: System :: Systems Administration :: Authentication/Directory',
 ]
-build-backend = 'setuptools.build_meta'
+
+[project.urls]
+Source = 'https://github.com/FirefighterBlu3/python-pam'
+BugTracker = 'https://github.com/FirefighterBlu3/python-pam/issues'
+
+[build-system]
+requires = ['flit_core>=3.3']
+build-backend = 'flit_core.buildapi'
 
 # ignore the tox documentation, it IS NOT supported yet
 # https://github.com/tox-dev/tox/issues/2148
