--- zerovec-0.11.8/Cargo.toml	2006-07-24T01:21:28+00:00
+++ zerovec-0.11.8/Cargo.toml	2026-08-22T09:52:32.895786+00:00
@@ -80,11 +80,6 @@
 path = "src/lib.rs"
 bench = false
 
-[[example]]
-name = "zv_serde"
-path = "examples/zv_serde.rs"
-required-features = ["serde"]
-
 [[bench]]
 name = "vzv"
 path = "benches/vzv.rs"
@@ -156,28 +151,13 @@
 [dev-dependencies.bincode]
 version = "1.3.1"
 
-[dev-dependencies.getrandom]
-version = "0.3"
-features = ["wasm_js"]
-
-[dev-dependencies.iai]
-version = "0.1.1"
-
 [dev-dependencies.insta]
 version = "1.43.2"
 features = ["json"]
 
-[dev-dependencies.postcard]
-version = "1.0.3"
-features = ["use-std"]
-default-features = false
-
 [dev-dependencies.rand]
 version = "0.9"
 
-[dev-dependencies.rand_distr]
-version = "0.5"
-
 [dev-dependencies.rand_pcg]
 version = "0.9"
 
@@ -196,9 +176,6 @@
 version = "0.8.3"
 features = ["derive"]
 default-features = false
-
-[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.criterion]
-version = "0.5.0"
 
 [lints.clippy]
 alloc_instead_of_core = "warn"
