--- tracing-subscriber-0.3.23/Cargo.toml	2006-07-24T01:21:28+00:00
+++ tracing-subscriber-0.3.23/Cargo.toml	2026-03-14T23:01:34.583267+00:00
@@ -99,8 +99,6 @@
 ]
 valuable = [
     "tracing-core/valuable",
-    "valuable_crate",
-    "valuable-serde",
     "tracing-serde/valuable",
 ]
 
@@ -346,17 +344,6 @@
 
 [dev-dependencies.tracing-log]
 version = "0.2.0"
-
-[target."cfg(tracing_unstable)".dependencies.valuable-serde]
-version = "0.1.0"
-optional = true
-default-features = false
-
-[target."cfg(tracing_unstable)".dependencies.valuable_crate]
-version = "0.1.0"
-optional = true
-default-features = false
-package = "valuable"
 
 [lints.rust.unexpected_cfgs]
 level = "warn"
@@ -366,3 +353,4 @@
     "cfg(tracing_unstable)",
     "cfg(unsound_local_offset)",
 ]
+
