diff -rupr b/finger/lprint.c c/finger/lprint.c
--- b/finger/lprint.c	1999-09-14 03:51:11.000000000 -0700
+++ c/finger/lprint.c	2016-12-07 13:24:43.785452865 -0800
@@ -48,7 +48,7 @@ char lprint_rcsid[] =
 #include <sys/types.h>
 #include <sys/file.h>
 #include <sys/stat.h>
-#include <sys/time.h>
+#include <time.h>
 #include "finger.h"
 
 static void lprint(PERSON *pn);
diff -rupr b/finger/sprint.c c/finger/sprint.c
--- b/finger/sprint.c	1999-12-12 10:59:33.000000000 -0800
+++ c/finger/sprint.c	2016-12-07 13:24:56.856976108 -0800
@@ -40,7 +40,7 @@ char sprint_rcsid[] = "$Id: sprint.c,v 1
 #endif /* not lint */
 
 #include <sys/types.h>
-#include <sys/time.h>
+#include <time.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
