diff -up librdkafka-2.14.1/mklove/modules/configure.libssl.orig librdkafka-2.14.1/mklove/modules/configure.libssl
--- librdkafka-2.14.1/mklove/modules/configure.libssl.orig	2025-12-23 00:00:00.000000000 +0000
+++ librdkafka-2.14.1/mklove/modules/configure.libssl	2025-12-23 00:00:00.000000000 +0000
@@ -16,7 +16,7 @@
 # SSL that is the feature, not crypto.

 mkl_toggle_option "Feature" ENABLE_SSL "--enable-ssl" "Enable SSL support" "try"
-
+mkl_toggle_option "Feature" ENABLE_SSL_ENGINE "--enable-ssl-engine" "Enable SSL engine support" "n"

 function manual_checks {
     case "$ENABLE_SSL" in
diff -up librdkafka-2.14.1/src/rdkafka_conf.h.orig librdkafka-2.14.1/src/rdkafka_conf.h
--- librdkafka-2.14.1/src/rdkafka_conf.h.orig	2025-12-23 00:00:00.000000000 +0000
+++ librdkafka-2.14.1/src/rdkafka_conf.h	2025-12-23 00:00:00.000000000 +0000
@@ -34,7 +34,8 @@
 #include "rdkafka_cert.h"

 #if WITH_SSL && OPENSSL_VERSION_NUMBER >= 0x10100000 &&                        \
-    !defined(OPENSSL_NO_ENGINE)
+    !defined(OPENSSL_NO_ENGINE) &&                                             \
+    ENABLE_SSL_ENGINE
 #define WITH_SSL_ENGINE 1
 /* Deprecated in OpenSSL 3 */
 #include <openssl/engine.h>
