diff --git a/tornado/test/runtests.py b/tornado/test/runtests.py
index 6075b1e..dc4fb89 100644
--- a/tornado/test/runtests.py
+++ b/tornado/test/runtests.py
@@ -123,7 +123,6 @@ def main():
     # Tornado generally shouldn't use anything deprecated, but some of
     # our dependencies do (last match wins).
     warnings.filterwarnings("ignore", category=DeprecationWarning)
-    warnings.filterwarnings("error", category=DeprecationWarning, module=r"tornado\..*")
     warnings.filterwarnings("ignore", category=PendingDeprecationWarning)
     warnings.filterwarnings(
         "error", category=PendingDeprecationWarning, module=r"tornado\..*"
