From 58f264d25c9dc427733a6a72fedc7b0e3b9c8e06 Mon Sep 17 00:00:00 2001
From: Lumir Balhar <lbalhar@redhat.com>
Date: Thu, 16 May 2019 13:50:14 +0200
Subject: [PATCH] Do not turn DeprecationWarning into Exception

---
 tornado/test/runtests.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tornado/test/runtests.py b/tornado/test/runtests.py
index dab76a2..117fd1c 100644
--- a/tornado/test/runtests.py
+++ b/tornado/test/runtests.py
@@ -124,7 +124,6 @@ def main():
     # Tornado generally shouldn't use anything deprecated, but some of
     # our dependencies do (last match wins).
     warnings.filterwarnings("ignore", category=DeprecationWarning)
-    warnings.filterwarnings("error", category=DeprecationWarning, module=r"tornado\..*")
     warnings.filterwarnings("ignore", category=PendingDeprecationWarning)
     warnings.filterwarnings(
         "error", category=PendingDeprecationWarning, module=r"tornado\..*"
-- 
2.21.0

