diff -up enscript-1.6.4/src/mkafmmap.c.rh457720 enscript-1.6.4/src/mkafmmap.c
--- enscript-1.6.4/src/mkafmmap.c.rh457720	2008-08-08 11:33:47.000000000 +0200
+++ enscript-1.6.4/src/mkafmmap.c	2008-08-08 11:33:58.000000000 +0200
@@ -139,7 +139,7 @@ main (int argc, char *argv[])
       int option_index = 0;
       int c;
 
-      c = getopt_long (argc, argv, "p:h", long_options, &option_index);
+      c = getopt_long (argc, argv, "p:hV", long_options, &option_index);
       if (c == -1)
 	break;
 
