diff -up enscript-1.6.6/afm/Makefile.am.rh477382 enscript-1.6.6/afm/Makefile.am
--- enscript-1.6.6/afm/Makefile.am.rh477382	2012-09-20 00:17:34.000000000 +0200
+++ enscript-1.6.6/afm/Makefile.am	2012-09-26 13:42:09.223974485 +0200
@@ -29,9 +29,12 @@ hvnbo.afm hvno.afm hvo.afm ncb.afm ncbi.
 pobi.afm poi.afm por.afm sy.afm tib.afm tibi.afm tii.afm tir.afm	\
 zcmi.afm zd.afm
 
-public_fonts = matrix.afm matrix.pfa
+public_fonts = matrix.afm matrix.eps
 
 afmdir = $(pkgdatadir)/afm
 dist_afm_DATA = font.map $(default_afm) $(public_fonts) MustRead.html
 
 EXTRA_DIST = ChangeLog.old
+
+matrix.eps: matrix.pfa
+	cp matrix.pfa matrix.eps
diff -up enscript-1.6.6/src/util.c.rh477382 enscript-1.6.6/src/util.c
--- enscript-1.6.6/src/util.c.rh477382	2011-10-30 17:48:42.000000000 +0100
+++ enscript-1.6.6/src/util.c	2012-09-26 13:40:53.336866463 +0200
@@ -1084,9 +1084,16 @@ download_font (char *name)
       buffer_append (&fname, ".pfb");
       if (stat (buffer_ptr (&fname), &stat_st) != 0)
 	{
-	  /* Couldn't find font description file, nothing to download. */
-	  buffer_uninit (&fname);
-	  return;
+	  /* .eps */
+	  buffer_clear (&fname);
+	  buffer_append (&fname, prefix);
+	  buffer_append (&fname, ".eps");
+	  if (stat (buffer_ptr (&fname), &stat_st) != 0)
+	    {
+	      /* Couldn't find font description file, nothing to download. */
+	      buffer_uninit (&fname);
+	      return;
+	    }
 	}
     }
 
